---
title: 'Preprocessing: Method development lung cancer patient cohort'
author: "Qian-Wu Liao"
output:
  workflowr::wflow_html:
    toc: yes
    toc_float: yes
    code_folding: hide
---

<font size='4'> Description: Preprocess Tissue and Plasma Targeted Metabolomics
data generated by AG Hell and Tissue and Plasma DDA Proteomics from AG Krijgsveld,
which includes data cleansing, data normalization, and in the end, storing all needed
information of each dataset (4 in total) in SummarizedExperiment objects for further
analyses. </font>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.height = 8, fig.width = 10)
knitr::opts_knit$set(root.dir = '/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev')
```

Load libraries
```{r library loading, message = FALSE}
library('readr')
library('readxl')
library('vsn')
library('pwr')
library('SummarizedExperiment')
library('ggrepel')
library('tidyverse')
# Load user-defined functions
source('./code/misc.R')

# Set plot theme
th <- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = 'bold'),
        axis.text = element_text(face = 'bold'),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))
```

```{r message = F}
# Prepare patient metadata table
# Retrieve patient recurrence information
recurAnno <- readxl::read_excel(
  './data/AG_Hell/Thorax cohort_not-normalized_subset plasma.xlsx',
  skip = 1) %>%
  filter(!is.na(parents)) %>%
  dplyr::select(c(Individual_ID, Status)) %>%
  dplyr::rename(Patient = 'Individual_ID',
                Condition = 'Status') %>%
  dplyr::filter(Condition != 'BASELINE_SURGERY') %>%
  dplyr::mutate(Recurrence = dplyr::case_when(
    Condition == 'FOLLOW-UP' ~ 'No',
    Condition == 'RECURRENCE' ~ 'Yes'
  )) %>%
  dplyr::select(-Condition)
# Clean up patient metadata information (adjuvant chemotherapy can be used later)
metadataTab <- readr::read_tsv('./data/patient_metadata.tsv') %>%
  dplyr::select(Code, Gender, `Age at diagnosis`, `Smoking status`, `Pathological stage`) %>%
  dplyr::rename(Patient = Code, Age = `Age at diagnosis`, Smoking = `Smoking status`,
                Stage = `Pathological stage`) %>%
  dplyr::filter(Patient %in% recurAnno$Patient) %>%
  dplyr::mutate(Gender = dplyr::case_when(Gender == 'MALE' ~ 'Male',
                                          Gender == 'FEMALE' ~ 'Female'),
                Smoking = dplyr::case_when(Smoking == 'SMOKER' ~ 'Smoker',
                                           Smoking == 'EX-SMOKER' ~ 'Ex-smoker',
                                           Smoking == 'NON-SMOKER' ~ 'Non-smoker'))
# Combine all patient information into a table
metadataTab <- dplyr::left_join(recurAnno, metadataTab, by = 'Patient')
```

# Plasma Metabolomics

```{r message = F}
# Clean up data and convert data to long data
metaPlasmaTab <- readxl::read_excel(
  './data/AG_Hell/Thorax cohort_not-normalized_subset plasma.xlsx',
  skip = 1) %>%
  filter(!is.na(parents)) %>%
  dplyr::select(-c(Sample_identification_cut, parents)) %>%
  rename(Sample = 'patients_tissue',
         Patient = 'Individual_ID',
         Condition = 'Status',
         Identifier = 'Sample Identification') %>%
  pivot_longer(cols = -c('Sample', 'Patient', 'Condition', 'Identifier'),
               names_to = 'Metabolite',
               values_to = 'Abundance') %>%
  mutate(Condition = case_when(Condition == 'BASELINE_SURGERY' ~ 'Baseline',
                               Condition == 'FOLLOW-UP' ~ 'Follow-up',
                               Condition == 'RECURRENCE' ~ 'Recurrence'),
         Abundance = as.numeric(Abundance)) %>%
  left_join(metadataTab, by = 'Patient')

# Convert long data to SE object
metaPlasma <- df2SummExp(metaPlasmaTab, row_id = 'Metabolite', col_id = 'Sample',
                         values = 'Abundance', col_anno = c('Condition', 'Patient',
                                                            'Recurrence', 'Gender',
                                                            'Age', 'Smoking', 'Stage',
                                                            'Identifier'))
```

Display dimensions of processed data (40 samples and 630 features)
```{r}
dim(metaPlasma)
```

Display distribution of original data
```{r warning = F}
metaPlasma4Plot <- summExp2df(metaPlasma, assay = 'Abundance',
                              row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaPlasma4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

# Zero-valued data points are removed.
```

## Perform VSN

Perform VSN (variance stabilizing normalization) on original data
```{r}
exprMat <- as.matrix(assay(metaPlasma))
fit <- vsnMatrix(exprMat)
metaPlasmaNorm <- metaPlasma
assay(metaPlasmaNorm) <- predict(fit, exprMat)
# save vsn normalized data
# saveRDS(metaPlasmaNorm, './data/AG_Hell/metaPlasmaNorm.rds')

metaPlasmaNorm4Plot <- summExp2df(metaPlasmaNorm, assay = 'Abundance',
                                  row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaPlasmaNorm4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

```{r eval = F}
# Store median normalized data for computing log2(FC) later
exprMat <- as.matrix(assay(metaPlasma))
sampleAnno <- tibble::as_tibble(colData(metaPlasma), rownames = 'Sample')
metaPlasmaMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance+0.0001)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample')

ggplot(metaPlasmaMediTab, aes(x=Sample, y=mediAbundance)) +
  geom_boxplot() +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

metaPlasmaMedi <- df2SummExp(metaPlasmaMediTab, row_id = 'Feature', col_id = 'Sample',
                             values = 'mediAbundance', col_anno = c('Condition', 'Patient',
                                                                    'Recurrence', 'Gender',
                                                                    'Age', 'Smoking', 'Stage',
                                                                    'Identifier'))
# saveRDS(metaPlasmaMedi, './data/AG_Hell/metaPlasmaMedi.rds')
```

## Perform PCA

Use only top 100 metabolites that have high variance
```{r}
exprMat <- assay(metaPlasmaNorm)
sd <- rowSds(exprMat)
exprMatSub <- exprMat[order(sd, decreasing = TRUE)[1:100],]
resPC <- prcomp(t(exprMatSub), center = TRUE, scale. = FALSE)

anno <- distinct(metaPlasmaTab, Sample, Patient, Condition)
metaPlasmaNorm4Plot <- resPC$x[,1:10] %>%
  as.data.frame() %>%
  rownames_to_column('Sample') %>%
  left_join(anno, by = 'Sample')

ggplot(metaPlasmaNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
<br/>
<br/>

Use all metabolites
```{r}
resPC2 <- prcomp(t(exprMat), center = TRUE, scale. = FALSE)

metaPlasmaNorm4Plot2 <- resPC2$x[, 1:10] %>%
  as.data.frame() %>%
  rownames_to_column('Sample')%>%
  left_join(anno, by = 'Sample')

ggplot(metaPlasmaNorm4Plot2, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```




# Tissue Metabolomics

```{r message = F, warning = F}
# Clean up data and convert data to long data
metaTissueTab <- readxl::read_excel(
  './data/AG_Hell/Thorax cohort_not-normalized_subset tissue.xlsx',
  skip = 1) %>%
  dplyr::filter(!is.na(parents)) %>%
  dplyr::select(-c(Sample_identification_cut, parents)) %>%
  dplyr::rename(Sample = patients_tissue,
                Patient = Individual_ID,
                Condition = Status,
                Identifier = `Sample Identification`) %>%
  tidyr::pivot_longer(cols = -c('Sample', 'Patient', 'Condition', 'Identifier'),
                      names_to = 'Metabolite', values_to = 'Abundance') %>%
  dplyr::mutate(Condition = dplyr::case_when(Condition == 'Normalgewebe' ~ 'Normal',
                                             Condition == 'Tumorgewebe' ~ 'Tumor'),
                Abundance = as.numeric(Abundance)) %>%
  dplyr::left_join(metadataTab, by = 'Patient')

# Convert long data to SE object
metaTissue <- df2SummExp(metaTissueTab, row_id = 'Metabolite', col_id = 'Sample',
                         values = 'Abundance', col_anno = c('Condition', 'Patient',
                                                            'Recurrence', 'Gender',
                                                            'Age', 'Smoking', 'Stage',
                                                            'Identifier'))
```

Display dimensions of processed data (40 samples and 630 features)
```{r}
dim(metaTissue)
```

Display 3 NA data points in this dataset
```{r}
dplyr::filter(metaTissueTab, is.na(Abundance)) %>%
  dplyr::select(-c(Identifier, Recurrence, Gender, Smoking, Stage))

# Values of these data points were infinite (character) before converting them to
# numeric class.
```

Display distribution of original data
```{r warning = F}
metaTissue4Plot <- summExp2df(metaTissue, assay = 'Abundance',
                              row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaTissue4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

# There are 3 NA data points and 756 zeros.
```

## Perform VSN

```{r warning = F}
exprMat <- as.matrix(assay(metaTissue))
fit <- vsnMatrix(exprMat)
metaTissueNorm <- metaTissue
assay(metaTissueNorm) <- predict(fit, exprMat)
# Save vsn normalized data
# saveRDS(metaTissueNorm, './data/AG_Hell/metaTissueNorm.rds')

metaTissueNorm4Plot <- summExp2df(metaTissueNorm, assay = 'Abundance',
                                  row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaTissueNorm4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

```{r eval = F}
# Store median normalized data for computing log2(FC)
exprMat <- as.matrix(assay(metaTissue))
sampleAnno <- tibble::as_tibble(colData(metaTissue), rownames = 'Sample')
metaTissueMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance+0.0001)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample')

ggplot(metaTissueMediTab, aes(x=Sample, y=mediAbundance)) +
  geom_boxplot() +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

metaTissueMedi <- df2SummExp(metaTissueMediTab, row_id = 'Feature', col_id = 'Sample',
                             values = 'mediAbundance', col_anno = c('Condition', 'Patient',
                                                                    'Recurrence', 'Gender',
                                                                    'Age', 'Smoking', 'Stage',
                                                                    'Identifier'))
# saveRDS(metaTissueMedi, './data/AG_Hell/metaTissueMedi.rds')
```

## Perform PCA
Take an initial look at dataset for quality control
<br/>
<br/>

Use only top 100 metabolites that have high variance
```{r}
exprMat <- assay(metaTissueNorm)
# Remove the metabolites containing non-applicable values (try to impute the
# missing values later)
exprMat <- exprMat[which(rownames(exprMat) != c('Ala', 'Histamine')),]
sd <- rowSds(exprMat)
exprMatSub <- exprMat[order(sd, decreasing = TRUE)[1:100],]
resPC <- prcomp(t(exprMatSub), center = TRUE, scale. = FALSE) # Each row is an
# observation, i.e., samples

anno <- dplyr::distinct(metaTissueTab, Sample, Patient, Condition)
metaTissueNorm4Plot <- resPC$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('Sample') %>%
  dplyr::left_join(anno, by = 'Sample')

ggplot(metaTissueNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
<br/>
<br/>

Use all metabolites except those three containing missing values
```{r}
resPC2 <- prcomp(t(exprMat), center = TRUE, scale. = FALSE)

metaTissueNorm4Plot2 <- resPC2$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('Sample') %>%
  dplyr::left_join(anno, by = 'Sample')

ggplot(metaTissueNorm4Plot2, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```

```{r include = F, eval = F}
# Perform power analysis to estimate the required number of samples for varied
# statistical power
dTab <- summExp2df(metaTissueNorm, assay = 'Abundance',
                   row_id = 'Metabolite', col_id = 'Sample') %>%
  dplyr::select(-Sample) %>%
  tidyr::pivot_wider(names_from = 'Condition', values_from = 'Value') #%>%
  dplyr::mutate(diff = abs(Normal - Tumor)) %>%
  dplyr::group_by(Metabolite) %>%
  dplyr::summarise(d = mean(diff) / sd(diff))

dTab4Plot <- lapply(seq(0.01, 1.5, length.out = 100), function(i){
  tibble(nCall = nrow(dplyr::filter(dTab, d > i)),
         nSample = pwr.t.test(d = i, sig.level = 0.05,
                              power = 0.8, type = 'paired')$n)}) %>%
  dplyr::bind_rows() %>%
  dplyr::arrange(nCall) %>%
  dplyr::filter(nCall > 0)

ggplot(dTab4Plot, aes(x=nCall, y=nSample)) +
  geom_line() +
  geom_point() +
  ylim(0, 100)
```




# Plasma Proteomics

```{r message = F}
# Abundance data
proPlasmaTab <- readxl::read_excel(
  './data/AG_Krijgsveld/20211119_133842_20211018_TMMS_TRX_plasma_Report.xlsx',
  sheet = 1) %>%
  dplyr::select(PG.Genes, PG.ProteinGroups, contains('oecf4')) %>%
  pivot_longer(cols = -c('PG.Genes', 'PG.ProteinGroups'),
               names_to = 'Sample_IDs',
               values_to = 'Abundance') %>%
  mutate(Abundance = as.numeric(Abundance)) %>%
  group_by(PG.Genes, PG.ProteinGroups, Sample_IDs) %>%
  summarise(Abundance = mean(Abundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(!is.na(Abundance))

# Column PG.ProteinGroups has no overlap and column PG.Genes has few overlaps (NA:5).
```

```{r}
# Sample conditions of 'LH32F07U_P_R' and 'E04NMF53_P_R' were not recorded (follow-up
# or recurrence), yet, annotations can be recovered according to Plasma Metabolomics
# due to same patient cohort.

# Annotation data
proPlasmaAnno <- readxl::read_excel(
  './data/AG_Krijgsveld/20211119_133842_20211018_TMMS_TRX_plasma_Report.xlsx',
  sheet = 2) %>%
  dplyr::select(Sample_IDs, Condition, parents, Identifier) %>%
  mutate(Patient = str_remove_all(parents, '/THRX_SPACE/THRX_DB/SC_T_S_|_P_B|_P_R'),
         Sample = str_remove(parents, '/THRX_SPACE/THRX_DB/SC_T_S_'),
         Identifier = str_remove(parents, '/THRX_SPACE/THRX_DB/')) %>%
  dplyr::select(-parents) %>%
  left_join(metadataTab, by = 'Patient')

# Unify plasma annotations for Plasma Proteomics and Metabolomics
newCondition <- proPlasmaAnno$Condition
newCondition[which(newCondition == 'Pretherapeutic')] <- 'Baseline'
newCondition[which(proPlasmaAnno$Sample == 'LH32F07U_P_R')] <- 'Follow-up'
newCondition[which(proPlasmaAnno$Sample == 'E04NMF53_P_R')] <- 'Recurrence'
proPlasmaAnno$Condition <- newCondition
```

Show dimensions of processed data (40 samples and 435 features)
```{r}
proPlasmaTab <- left_join(proPlasmaTab, proPlasmaAnno, by = 'Sample_IDs')
proPlasma <- df2SummExp(proPlasmaTab, row_id = 'PG.ProteinGroups', col_id = 'Sample',
                        values = 'Abundance', row_anno = 'PG.Genes',
                        col_anno = c('Condition', 'Patient', 'Recurrence', 'Gender',
                                     'Age', 'Smoking', 'Stage', 'Identifier'))
dim(proPlasma)
```

Display distribution of original data
```{r}
proPlasma4Plot <- summExp2df(proPlasma, assay = 'Abundance',
                             row_id = 'PG.ProteinGroups', col_id = 'Sample')
ggplot(proPlasma4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Protein Abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

## perform VSN

```{r}
exprMat <- as.matrix(assay(proPlasma))
fit <- vsnMatrix(exprMat)
proPlasmaNorm <- proPlasma
assay(proPlasmaNorm) <- predict(fit, exprMat)
# Save vsn normalized data
# saveRDS(proPlasmaNorm, './data/AG_Krijgsveld/proPlasmaNorm.rds')

proPlasmaNorm4Plot <- summExp2df(proPlasmaNorm, assay = 'Abundance',
                                 row_id = 'PG.ProteinGroups', col_id = 'Sample')
ggplot(proPlasmaNorm4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Protein Abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

```{r eval = F}
# Store median normalized data for computing log2(FC)
exprMat <- as.matrix(assay(proPlasma))
sampleAnno <- tibble::as_tibble(colData(proPlasma), rownames = 'Sample')
featAnno <- tibble::as_tibble(rowData(proPlasma), rownames = 'Feature')
proPlasmaMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample') %>%
  dplyr::left_join(featAnno, by = 'Feature')

ggplot(proPlasmaMediTab, aes(x=Sample, y=mediAbundance)) +
  geom_boxplot() +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

proPlasmaMedi <- df2SummExp(proPlasmaMediTab, row_id = 'Feature', col_id = 'Sample',
                            values = 'mediAbundance', row_anno = 'PG.Genes',
                            col_anno = c('Condition', 'Patient', 'Recurrence', 'Gender',
                                         'Age', 'Smoking', 'Stage', 'Identifier'))
# saveRDS(proPlasmaMedi, './data/AG_Krijgsveld/proPlasmaMedi.rds')
```

## Perform PCA

```{r}
exprMat <- assay(proPlasmaNorm)
resPC <- prcomp(t(exprMat), center = TRUE, scale. = FALSE)

anno <- dplyr::distinct(proPlasmaTab, Sample, Patient, Condition)
proPlasmaNorm4Plot <- resPC$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('Sample') %>%
  dplyr::left_join(anno, by = 'Sample')

ggplot(proPlasmaNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```




# Tissue Proteomics

```{r message = F}
# Abundance data
proTissueTab <- readxl::read_excel(
  './data/AG_Krijgsveld/20211119_151605_20211104_TMMS_Tissue_Report.xlsx',
  sheet = 1) %>%
  dplyr::select(PG.Genes, PG.ProteinGroups, contains('oecf4')) %>%
  # Convert messy data to tidy table
  tidyr::pivot_longer(cols = -c('PG.Genes', 'PG.ProteinGroups'),
                      names_to = 'Sample_IDs',
                      values_to = 'Abundance') %>%
  dplyr::mutate(Abundance = as.numeric(Abundance)) %>%
  dplyr::group_by(PG.Genes, PG.ProteinGroups, Sample_IDs) %>%
  # Determine a single numeric value for a protein of a sample if multiple
  # identical protein measures of a sample exist
  dplyr::summarise(Abundance = mean(Abundance, na.rm = TRUE)) %>%
  dplyr::ungroup() %>%
  dplyr::filter(!is.na(Abundance))

# Column PG.ProteinGroups has no overlap and column PG.Genes has few overlaps (
# TMPO:2 and NA:5) because some genes can encode multiple protein isoforms through
# RNA splicing.
# 19 protein groups are not present in any samples (Abundance = NaN).
```

```{r message = F}
# Annotation data
proTissueAnno <- readxl::read_excel(
  './data/AG_Krijgsveld/20211119_151605_20211104_TMMS_Tissue_Report.xlsx',
  sheet = 2) %>%
  dplyr::select(Sample_IDs, Condition, parents, Identifier) %>%
  dplyr::mutate(Patient = str_remove_all(parents, '/THRX_SPACE/THRX_DB/SC_T_S_|_TG|_NG'),
                Sample = str_remove(parents, '/THRX_SPACE/THRX_DB/SC_T_S_'),
                Identifier = str_remove(Identifier, '/THRX_SPACE/'),
                id = paste0('smp', seq(nrow(.)))) %>%
  dplyr::select(-parents) %>%
  dplyr::left_join(metadataTab, by = 'Patient')
```

Show dimensions of processed data (40 samples and 5193 features)
```{r}
# Combine processed Abundance and annotation data
proTissueTab <- dplyr::left_join(proTissueTab, proTissueAnno, by = 'Sample_IDs')
# Convert tidy table to value matrix for further analysis
proTissue <- df2SummExp(proTissueTab, row_id = 'PG.ProteinGroups', col_id = 'id',
                        values = 'Abundance', row_anno = 'PG.Genes',
                        col_anno = c('Condition', 'Patient', 'Recurrence', 'Gender',
                                     'Age', 'Smoking', 'Stage', 'Sample', 'Identifier'))
dim(proTissue)
```

View distribution of original data to see if there is technical noise
```{r}
# Convert value matrix to long data for making plot
proTissue4Plot <- summExp2df(proTissue, assay = 'Abundance',
                             row_id = 'PG.ProteinGroups', col_id = 'id')
ggplot(proTissue4Plot, aes(x=id, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Protein abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

```{r include = F, eval = F}
# Log2-transform original data
ggplot(proTissue4Plot, aes(x=id, y=log2(Value))) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Protein abundance') +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

## Perform VSN

```{r}
exprMat <- as.matrix(assay(proTissue))
fit <- vsnMatrix(exprMat)
proTissueNorm <- proTissue
assay(proTissueNorm) <- predict(fit, exprMat)

proTissueNorm4Plot <- summExp2df(proTissueNorm, assay = 'Abundance',
                                 row_id = 'PG.ProteinGroups', col_id = 'id')
ggplot(proTissueNorm4Plot, aes(x=id, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Protein abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

## Perform PCA
Take an initial look at dataset for quality control

```{r}
# prcomp()'s parameter *center* is set to TRUE (Check effects of parameters *center*
# and *scale.*)

exprMat <- assay(proTissueNorm)
sd <- rowSds(exprMat)
exprMatSub <- exprMat[order(sd, decreasing = TRUE)[1:1000],]
resPC <- prcomp(t(exprMatSub), center = TRUE, scale. = FALSE) # Each row is an
# observation, i.e., samples

anno <- dplyr::distinct(proTissueTab, id, Patient, Condition)
proTissueNorm4Plot <- resPC$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('id') %>%
  dplyr::left_join(anno, by = 'id')

ggplot(proTissueNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_text_repel(aes(label=id), show.legend = F) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
Patient "LH32F07U" has two tumor samples (smp24, smp29) and one normal sample (smp9).\
Patient "F04ERF3M" does not have a paired tumor sample (smp22).\
It is very probably mislabeling. If not, check correlation between two tumor samples
from the same patient to get further idea.
<br/>
<br/>

```{r}
# Manually modify potential mislabeling for now
ids <- colnames(proTissueNorm)
colnames(proTissueNorm) <- colData(proTissueNorm)$Sample
colnames(proTissueNorm)[which(ids == 'smp29')] <- 'F04ERF3M_TG'
colData(proTissueNorm)$Sample <- colnames(proTissueNorm)
colData(proTissueNorm)$Patient[which(ids == 'smp29')] <- 'F04ERF3M'
colData(proTissueNorm)$Recurrence[which(ids == 'smp29')] <- 'Yes'
colData(proTissueNorm)$Age[which(ids == 'smp29')] <- 68
# Remove redundant Sample column in column annotation table
colData(proTissueNorm) <- colData(proTissueNorm)[, -which(colnames(colData(proTissueNorm)) == 'Sample')]

# Save vsn normalized data
# saveRDS(proTissueNorm, './data/AG_Krijgsveld/proTissueNorm.rds')
```

```{r eval = F}
# Store median normalized data for computing log2(FC)
# Manually modify potential mislabeling for now
ids <- colnames(proTissue)
colnames(proTissue) <- colData(proTissue)$Sample
colnames(proTissue)[which(ids == 'smp29')] <- 'F04ERF3M_TG'
colData(proTissue)$Sample <- colnames(proTissue)
colData(proTissue)$Patient[which(ids == 'smp29')] <- 'F04ERF3M'
colData(proTissue)$Recurrence[which(ids == 'smp29')] <- 'Yes'
colData(proTissue)$Age[which(ids == 'smp29')] <- 68

exprMat <- as.matrix(assay(proTissue))
sampleAnno <- tibble::as_tibble(colData(proTissue))
featAnno <- tibble::as_tibble(rowData(proTissue), rownames = 'Feature')
proTissueMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample') %>%
  dplyr::left_join(featAnno, by = 'Feature')

ggplot(proTissueMediTab, aes(x=Sample, y=mediAbundance)) +
  geom_boxplot() +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

proTissueMedi <- df2SummExp(proTissueMediTab, row_id = 'Feature', col_id = 'Sample',
                            values = 'mediAbundance', row_anno = 'PG.Genes',
                            col_anno = c('Condition', 'Patient', 'Recurrence', 'Gender',
                                         'Age', 'Smoking', 'Stage', 'Identifier'))
# saveRDS(proTissueMedi, './data/AG_Krijgsveld/proTissueMedi.rds')
```

```{r include = F, eval = F}
# prcomp()'s parameter *center* is set to FALSE

resPC2 <- prcomp(t(exprMatSub), center = FALSE, scale. = FALSE)

proTissueNorm4Plot2 <- resPC2$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('id') %>%
  dplyr::left_join(anno, by = 'id')

ggplot(proTissueNorm4Plot2, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed')
```

```{r include = F, eval = F}
### Perform power analysis

compD <- function(tab, n = 40){
  #' Compute effect size for each variable
  
  meanDiff <- abs(tab$avg[1] - tab$avg[2])
  fc <- abs(log(tab$avg[1]/tab$avg[2]))
  sdPool <- sqrt((tab$sd[1]^2 + tab$sd[2]^2) / 2)
  corFac <- ((n-3)/(n-2.25))*sqrt((n-2)/n) # Correction factor for small datasets
  d = corFac * meanDiff / sdPool
  return(tibble(d = d, logFC = fc))
}
```

```{r message = F, include = F, eval = F}
proTissueNormTab <- summExp2df(proTissueNorm, assay = 'Abundance',
                               row_id = 'PG.ProteinGroups', col_id = 'id') %>%
  dplyr::group_by(Condition, PG.ProteinGroups) %>%
  dplyr::summarise(avg = mean(Value), sd = sd(Value))

dTab <- dplyr::group_by(proTissueNormTab, PG.ProteinGroups) %>%
  tidyr::nest() %>%
  dplyr::mutate(results = map(data, compD)) %>%
  tidyr::unnest(results) %>%
  dplyr::select(-data)
```

```{r include = F, eval = F}
# Estimate required number of samples for varied statistical power

dTab4Plot <- lapply(seq(0.01, 1.5, length.out = 100), function(i){
  tibble(nCall = nrow(dplyr::filter(dTab, d > i)),
         nSample = pwr.t.test(d = i, sig.level = 0.05, power = 0.8,
                              type = 'two.sample')$n)}) %>%
  dplyr::bind_rows() %>%
  dplyr::arrange(nCall) %>%
  dplyr::filter(nCall > 0)

ggplot(dTab4Plot, aes(x=nCall, y=nSample)) +
  geom_line() +
  geom_point() +
  scale_y_log10() +
  scale_x_log10()

# nSample is number of samples required in each group.
```




# Abundance comparisons
Tissue vs Plasma:\
Investigate overall differences of Metabolomics and Proteomics abundance between
Tissue and Plasma samples
```{r}
# Load normalized data
metaTissueNorm <- readRDS('./data/AG_Hell/metaTissueNorm.rds')
metaPlasmaNorm <- readRDS('./data/AG_Hell/metaPlasmaNorm.rds')
proTissueNorm <- readRDS('./data/AG_Krijgsveld/proTissueNorm.rds')
proPlasmaNorm <- readRDS('./data/AG_Krijgsveld/proPlasmaNorm.rds')
```

## Metabolomics

```{r warning = F}
# Convert SummarizedExperiment objects to tables for plotting
# Tissue Metabolomics
metaTissueNorm4Plot <- summExp2df(metaTissueNorm, assay = 'Abundance',
                                  row_id = 'Metabolite', col_id = 'Sample') %>%
  dplyr::select(c('Metabolite', 'Sample', 'Value')) %>%
  dplyr::mutate(Sample_type = 'Tissue', Sample = paste0('Tissue_', Sample))

# Plasma Metabolomics
metaPlasmaNorm4Plot <- summExp2df(metaPlasmaNorm, assay = 'Abundance',
                                  row_id = 'Metabolite', col_id = 'Sample') %>%
  dplyr::select(c('Metabolite', 'Sample', 'Value')) %>%
  dplyr::mutate(Sample_type = 'Plasma', Sample = paste0('Plasma_', Sample))

# Concatenate two tables
metaTab4Plot <- rbind(metaTissueNorm4Plot, metaPlasmaNorm4Plot)

ggplot(metaTab4Plot, aes(x=Sample, y=Value, col=Sample_type)) +
  geom_boxplot(width = 0.7) +
  labs(y = 'Metabolite abundance', title = 'Metabolomics', col = 'Sample type') +
  th + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

# ggplot(metaTab4Plot, aes(x=Sample_type, y=Value, col=Sample_type)) +
#   geom_boxplot() +
#   labs(x = 'Sample type', y = 'Metabolite abundance', title = 'Metabolomics') +
#   theme(legend.position = 'none') +
#   ggpubr::stat_compare_means(method = 'wilcox.test', paired = F)

# When sample size is large, even minor differences are coming with highly significant
# p-values. Statistical meaningfulness does not equal biological meaningfulness.
# In this case, a complementary approach, effect size, should be reported.
```

## Proteomics

```{r}
# Tissue Proteomics
proTissueNorm4Plot <- summExp2df(proTissueNorm, assay = 'Abundance',
                                 row_id = 'PG.ProteinGroups', col_id = 'Sample') %>%
  dplyr::select(c('PG.ProteinGroups', 'Sample', 'Value')) %>%
  dplyr::mutate(Sample_type = 'Tissue', Sample = paste0('Tissue_', Sample))

# Plasma Proteomics
proPlasmaNorm4Plot <- summExp2df(proPlasmaNorm, assay = 'Abundance',
                                 row_id = 'PG.ProteinGroups', col_id = 'Sample') %>%
  dplyr::select(c('PG.ProteinGroups', 'Sample', 'Value')) %>%
  dplyr::mutate(Sample_type = 'Plasma', Sample = paste0('Plasma_', Sample))

# Concatenate two tables
proTab4Plot <- rbind(proTissueNorm4Plot, proPlasmaNorm4Plot)

ggplot(proTab4Plot, aes(x=Sample, y=Value, col=Sample_type)) +
  geom_boxplot(width = 0.7) +
  labs(y = 'Protein abundance', title = 'Proteomics', col = 'Sample type') +
  th + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

# ggplot(proTab4Plot, aes(x=Sample_type, y=Value, col=Sample_type)) +
#   geom_boxplot() +
#   labs(x = 'Sample type', y = 'Protein abundance', title = 'Proteomics') +
#   theme(legend.position = 'none') +
#   ggpubr::stat_compare_means(method = 'wilcox.test', paired = F)
```
The overall protein level of Tissue samples is higher than that of Plasma samples.
