<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Comparisons between DIA and DDA proteomics datasets</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparisons between DIA and DDA proteomics
datasets</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree717cb2321703f047d6659449429f744bc5e74644targetblank717cb23a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/717cb2321703f047d6659449429f744bc5e74644" target="_blank">717cb23</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree717cb2321703f047d6659449429f744bc5e74644targetblank717cb23a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/717cb2321703f047d6659449429f744bc5e74644" target="_blank">717cb23</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/feature_selection.Rmd
    Untracked:  analysis/mofa_downstream_analysis.Rmd
    Untracked:  code/Junyan/
    Untracked:  code/about.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/feature_selection_old.Rmd
    Untracked:  code/license.Rmd
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/mofa_varied_omics_combinations.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Klingmuller/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/MethodDev_LungCancer_Datasets.RData
    Untracked:  data/mofa/
    Untracked:  data/patient_metadata.tsv
    Untracked:  data/rf/
    Untracked:  data/stats/
    Untracked:  output/TAC_2023/
    Untracked:  output/YIG_seminar/
    Untracked:  output/preliminary_analysis_untargeted.html

Unstaged changes:
    Deleted:    analysis/mofa_varied_omics_combinations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/comparisons_DIA_DDA.Rmd</code>)
and HTML (<code>docs/comparisons_DIA_DDA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/9a0840b76aceb404bc0147e8caca1cd3a5a42beb/analysis/comparisons_DIA_DDA.Rmd" target="_blank">9a0840b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-08-04
</td>
<td>
Redo feature filtering for DIA proteomics and rerun subsequent analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/83a849a244396dfe32e55476a8e333eebc667fb4/analysis/comparisons_DIA_DDA.Rmd" target="_blank">83a849a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-31
</td>
<td>
Do single-omics analysis of new DIA plasma and tissue proteomics from AG
Krijgsveld and compare it with other proteomics we have
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/128dd12ec84dc9456c33f1481db143f6784d230d/docs/comparisons_DIA_DDA.html" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/4d5db4a46e8ac975848741857077394c93902c56/analysis/comparisons_DIA_DDA.Rmd" target="_blank">4d5db4a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
<td>
Reanalyze filtered untargeted metabolomics and lipidomics with PPCA and
additional patient metadata
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/comparisons_DIA_DDA.html" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/b0836405d1d46e75d85da360f9aa744d48e4ae9c/analysis/comparisons_DIA_DDA.Rmd" target="_blank">b083640</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-27
</td>
<td>
Preprocess DIA tissue proteomics from AG Klingmüller and compare it with
DIA and DDA data from AG Krijgsveld
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/comparisons_DIA_DDA.html" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/7cee36ee4cca86c197f370aed6285533cfeed9be/analysis/comparisons_DIA_DDA.Rmd" target="_blank">7cee36e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Compute correlations of common features between DIA and DDA Proteomics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Compare DIA and DDA Plasma and Tissue
Proteomics generated by AG Krijgsveld. Basically, correlations of common
features between DIA and DDA Proteomics were computed to see data
reproducibility using different data acquisition methods.<br />
</p>
<p>Update: Compare DIA Tissue Proteomics generated by AG Klingmüller
with aforementioned DIA and DDA Tissue Proteomics from AG Krijgsveld. In
addition, significant cancer recurrence-related proteins from each
dataset are also summarized.<br />
</p>
<p>Update: Compare New DIA Plasma and Tissue Proteomics from AG
Krijgsveld with all aforementioned datasets. Old DIA Tissue Proteomics
from AG Krijgsveld may have something wrong (bad data quality)!
</font></p>
<p>Load libraries</p>
<pre class="r"><code>library(SummarizedExperiment)
library(ggrepel)
library(igraph)
library(ggraph)
library(tidyverse)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<pre class="r"><code>calcFeatCorr &lt;- function(data1, data2, show_nFeats = F) {
  #&#39; Systematically compute feature correlations between two PROTEOMICS (for now) datasets
  #&#39; 
  #&#39; Parameters
  #&#39; data1, data2: SummarizedExperiment objects that contain data matrices
  #&#39; 
  #&#39; Return
  #&#39; corrRes: A table of correlations of overlapped features between two datasets
  
  # Extract data matrix
  dat1 &lt;- SummarizedExperiment::assay(data1)
  dat2 &lt;- SummarizedExperiment::assay(data2)
  
  # Make two matrices share same sample and feature spaces and in same row and column order
  # Keep only common samples
  commonSmps &lt;- intersect(colnames(dat1), colnames(dat2))
  dat1 &lt;- dat1[, commonSmps]
  dat2 &lt;- dat2[, commonSmps]
  # Keep only common features
  # Select first protein as representative if there are more than one
  rownames(dat1) &lt;- stringr::str_remove(rownames(dat1), &#39;;.*&#39;)
  rownames(dat2) &lt;- stringr::str_remove(rownames(dat2), &#39;;.*&#39;)
  commonFeats &lt;- intersect(rownames(dat1), rownames(dat2))
  dat1 &lt;- dat1[commonFeats,]
  dat2 &lt;- dat2[commonFeats,]
  if (show_nFeats) {
    cat(&#39;Features in data1: &#39;, nrow(data1), &#39;\n&#39;, &#39;Features in data2: &#39;, nrow(data2), &#39;\n&#39;,
        &#39;Features in common: &#39;, length(commonFeats), sep = &#39;&#39;)
  }
  
  # Compute feature correlations between two proteomics datasets
  # Do sanity check
  if (identical(rownames(dat1), rownames(dat2)) &amp;
      identical(colnames(dat1), colnames(dat2))) {
    corrs &lt;- sapply(seq_len(nrow(dat1)), function(i) {
      corr &lt;- cor.test(dat1[i,], dat2[i,], method = &#39;pearson&#39;, use = &quot;pairwise.complete.obs&quot;)
      corr$estimate
    })
    corrRes &lt;- data.frame(Feature = rownames(dat1), Correlation = corrs) %&gt;%
      dplyr::arrange(Correlation)
  }
  
  return(corrRes)
}

doFeatAssoTest &lt;- function(data, condition) {
  #&#39; Conduct univariate association test between each feature and CANCER RECURRENCE (for now)
  #&#39; 
  #&#39; Parameters
  #&#39; data: A SummarizedExperiment object that contains the data matrix and metadata
  #&#39; condition: A character specifying the sample metadata to split the dataset
  #&#39; 
  #&#39; Return
  #&#39; assoTestRes: A table of feature significance test results
  
  # Subset samples
  subsetIdx &lt;- which(colData(data)$Condition == condition)
  datSub &lt;- data[, subsetIdx]
  datMat &lt;- SummarizedExperiment::assay(datSub)
  
  if (all(complete.cases(datMat))) {
    # Perform analysis
    soaRes &lt;- doSOA(datSub, meta_var = &#39;Recurrence&#39;)
    assoTestRes &lt;- soaRes$tFeatRes %&gt;%
      dplyr::select(Var1, pVal) %&gt;%
      dplyr::rename(Feature = Var1)
  } else {
    smpAnno &lt;- tibble::as_tibble(colData(datSub), rownames = &#39;Sample&#39;)
    # Perform analysis
    # Fit linear probabilistic dropout model to normalized data
    fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
    assoTestRes &lt;- proDA::test_diff(fit,
                                 contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                 sort_by = &#39;pval&#39;) %&gt;%
      dplyr::select(name, pval) %&gt;%
      dplyr::rename(Feature = name, pVal = pval)
  }
  
  return(assoTestRes)
}

combineFeatAsso &lt;- function(data1, data2, name1, name2, condition, prot_gene_tbl, alpha = 0.05) {
  #&#39; Conduct univariate association tests between features and CANCER RECURRENCE (for now)
  #&#39; and combine and tidy up results and information into a table for making plots
  #&#39; 
  #&#39; Parameters
  #&#39; data1, data2: SummarizedExperiment objects that contain the data matrices and metadata
  #&#39; name1, name2: Characters indicating the dataset names in the plot legend
  #&#39; condition: A character specifying the sample metadata to split the dataset
  #&#39; prot_gene_tbl: A table containing proteins and their encoding genes for plot
  #&#39; labeling. The column for proteins should be named &#39;Proteins&#39; and for genes
  #&#39; should be named &#39;Genes&#39;. Furthermore, only the first protein and gene should
  #&#39; be kept if there are more than one
  #&#39; #&#39; alpha: A numeric value specifying the significance level for hypothesis testing
  #&#39; 
  #&#39; Return:
  #&#39; combinedAssoTestRes: A table containing association test results, labeling
  #&#39; information, etc. for making plots
  
  # Extract common features between two datasets
  # Select first protein as representative if there are more than one
  commonFeats &lt;- intersect(stringr::str_remove(rownames(data1), &#39;;.*&#39;),
                           stringr::str_remove(rownames(data2), &#39;;.*&#39;))
  
  # Compute associations between features and cancer recurrence and tidy up association test results
  assoTestRes_dat1 &lt;- doFeatAssoTest(data1, condition = condition) %&gt;%
    dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.*&#39;),
                  Sig_dat1 = dplyr::case_when(pVal &lt;= alpha ~ &#39;Yes&#39;,
                                              pVal &gt; alpha ~ &#39;No&#39;)) %&gt;%
    dplyr::filter(Feature %in% commonFeats) %&gt;%
    dplyr::rename(pVal_dat1 = pVal)
  assoTestRes_dat2 &lt;- doFeatAssoTest(data2, condition = condition) %&gt;%
    dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.*&#39;),
                  Sig_dat2 = dplyr::case_when(pVal &lt;= alpha ~ &#39;Yes&#39;,
                                              pVal &gt; alpha ~ &#39;No&#39;)) %&gt;%
    dplyr::filter(Feature %in% commonFeats) %&gt;%
    dplyr::rename(pVal_dat2 = pVal)
  
  # Combine association test results into a table and prepare color and label information
  combinedAssoTestRes &lt;- dplyr::left_join(assoTestRes_dat1, assoTestRes_dat2, by = &#39;Feature&#39;) %&gt;%
    dplyr::mutate(Significance = dplyr::case_when(Sig_dat1 == &#39;Yes&#39; &amp; Sig_dat2 == &#39;Yes&#39; ~ &#39;Sig. in Both&#39;,
                                                  Sig_dat1 == &#39;Yes&#39; &amp; Sig_dat2 == &#39;No&#39; ~ paste(&#39;Sig. in&#39;, name1),
                                                  Sig_dat1 == &#39;No&#39; &amp; Sig_dat2 == &#39;Yes&#39; ~ paste(&#39;Sig. in&#39;, name2),
                                                  Sig_dat1 == &#39;No&#39; &amp; Sig_dat2 == &#39;No&#39; ~ &#39;Not sig.&#39;),
                  Gene = plyr::mapvalues(Feature,
                                         from = prot_gene_tbl$Proteins,
                                         to = prot_gene_tbl$Genes,
                                         warn_missing = F),
                  Label = dplyr::case_when(Significance == &#39;Sig. in Both&#39; ~ Gene))
  # Reorder points in graph
  significance &lt;- combinedAssoTestRes$Significance
  combinedAssoTestRes &lt;- combinedAssoTestRes[c(which(significance == &#39;Not sig.&#39;),
                                               which(significance == &#39;Sig. in Both&#39;),
                                               which(significance == paste(&#39;Sig. in&#39;, name1)),
                                               which(significance == paste(&#39;Sig. in&#39;, name2))),]
  
  return(combinedAssoTestRes)
}

findOverlapSigFeats &lt;- function(DIA_data1, DIA_data2, DDA_data = NULL, condition,
                                prot_gene_tbl, alpha = 0.05) {
  #&#39; Extract overlaps of significant features across datasets
  #&#39; 
  #&#39; Parameters
  #&#39; DIA_data1, DIA_data2: SummarizedExperiment objects that contain the data matrices and metadata
  #&#39; DDA_data: An SE object containing the data matrix and metadata, which is used
  #&#39; as an extra layer of evidence to support found significant features. Defualt is NULL
  #&#39; condition: A character specifying the sample metadata to split the dataset
  #&#39; prot_gene_tbl: A table containing proteins and their encoding genes for plot
  #&#39; labeling. The column for proteins should be named &#39;Proteins&#39; and for genes
  #&#39; should be named &#39;Genes&#39;. Furthermore, only the first protein and gene should
  #&#39; be kept if there are more than one
  #&#39; alpha: A numeric value specifying the significance level for hypothesis testing
  #&#39; 
  #&#39; Return
  #&#39; A list containing the following components:
  #&#39; overlapFeatTab: A table containing summarized results of significance tests
  #&#39; across datasets. The columns of &#39;pVal1&#39; and &#39;pVal2&#39; hold feature significance
  #&#39; from input data1 and data2. Score is the product of -log10(pVal1) and -log10(pVal2)
  #&#39; sigFeatTab1, sigFeatTab2: Summarized information of common significant features
  #&#39; and needed sample and feature metadata
  
  # Extract common features between two datasets
  # Select first protein as representative if there are more than one
  cmnFeats&lt;- intersect(stringr::str_remove(rownames(DIA_data1), &#39;;.*&#39;),
                       stringr::str_remove(rownames(DIA_data2), &#39;;.*&#39;))
  
  # Do association test between each feature and cancer recurrence and organize
  # significant features from each dataset
  sigFeats1_DIA &lt;- doFeatAssoTest(DIA_data1, condition = condition) %&gt;%
    dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.*&#39;),
                  Occurrence = dplyr::case_when(pVal &lt;= alpha ~ 1,
                                                pVal &gt; alpha ~ 0)) %&gt;%
    dplyr::filter(Feature %in% cmnFeats)
  sigFeats2_DIA &lt;- doFeatAssoTest(DIA_data2, condition = condition) %&gt;%
    dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.*&#39;),
                  Occurrence = dplyr::case_when(pVal &lt;= alpha ~ 1,
                                                pVal &gt; alpha ~ 0)) %&gt;%
    dplyr::filter(Feature %in% cmnFeats)
  
  # Summarize all information and pinpoint most promising recurrence-related features
  summSigFeats &lt;- dplyr::left_join(sigFeats1_DIA, sigFeats2_DIA, by = &#39;Feature&#39;) %&gt;%
    dplyr::mutate(Occurrence = Occurrence.x + Occurrence.y) %&gt;%
    dplyr::filter(Occurrence == 2) %&gt;%
    # dplyr::mutate(Score = ((-log10(pVal.x))+(-log10(pVal.y)))/2) %&gt;%
    dplyr::mutate(Score = (-log10(pVal.x) * -log10(pVal.y))) %&gt;%
    dplyr::arrange(dplyr::desc(Score)) %&gt;%
    dplyr::mutate(Gene = plyr::mapvalues(Feature,
                                         from = prot_gene_tbl$Proteins,
                                         to = prot_gene_tbl$Genes,
                                         warn_missing = F),
                  Gene = factor(Gene, levels = Gene)) %&gt;%
    dplyr::rename(Protein = Feature, pVal1 = pVal.x, pVal2 = pVal.y) %&gt;%
    dplyr::select(-c(Occurrence, Occurrence.x, Occurrence.y))
  
  # Add extra layer of evidence if provided
  if (!is.null(DDA_data)) {
    sigFeats_DDA &lt;- doFeatAssoTest(DDA_data, condition = condition) %&gt;%
      dplyr::filter(pVal &lt;= alpha) %&gt;%
      dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.*&#39;)) %&gt;%
      dplyr::pull(Feature)
    summSigFeats &lt;- dplyr::mutate(summSigFeats,
                                  inDDA = dplyr::case_when(Protein %in% sigFeats_DDA ~ &#39;Yes&#39;,
                                                           !(Protein %in% sigFeats_DDA) ~ &#39;No&#39;))
  }
  
  # Retrieve abundances of overlapped significant features from each dataset and
  # combine all needed sample and feature information into a table
  # Prepare patient recurrence information
  recurInfo &lt;- tibble::as_tibble(colData(DDA_data), rownames = &#39;Sample&#39;) %&gt;%
    dplyr::select(Sample, Recurrence)
  # Tidy up all needed information
  DIA_dataSub1 &lt;- DIA_data1[, which(colData(DIA_data1)$Condition == condition)]
  rownames(DIA_dataSub1) &lt;- stringr::str_remove(rownames(DIA_dataSub1), &#39;;.*&#39;)
  sigFeats1 &lt;- t(SummarizedExperiment::assay(DIA_dataSub1)) %&gt;%
    tibble::as_tibble(rownames = &#39;Sample&#39;) %&gt;%
    dplyr::select(c(Sample, summSigFeats$Protein)) %&gt;%
    tidyr::pivot_longer(cols = -&#39;Sample&#39;, names_to = &#39;Protein&#39;, values_to = &#39;Abundance&#39;) %&gt;%
    dplyr::left_join(summSigFeats[, c(&#39;Protein&#39;, &#39;Gene&#39;, &#39;pVal1&#39;, &#39;inDDA&#39;)], by = &#39;Protein&#39;) %&gt;%
    dplyr::left_join(recurInfo, by = &#39;Sample&#39;) %&gt;%
    dplyr::mutate(Gene = paste0(Gene, &#39;\n(p = &#39;, round(pVal1, 4), &#39;)&#39;),
                  Gene = dplyr::case_when(inDDA == &#39;Yes&#39; ~ paste0(&#39;*&#39;, Gene),
                                          inDDA == &#39;No&#39; ~ Gene),
                  Gene = factor(Gene, levels = unique(Gene)))
  DIA_dataSub2 &lt;- DIA_data2[, which(colData(DIA_data2)$Condition == condition)]
  rownames(DIA_dataSub2) &lt;- stringr::str_remove(rownames(DIA_dataSub2), &#39;;.*&#39;)
  sigFeats2 &lt;- t(SummarizedExperiment::assay(DIA_dataSub2)) %&gt;%
    tibble::as_tibble(rownames = &#39;Sample&#39;) %&gt;%
    dplyr::select(c(Sample, summSigFeats$Protein)) %&gt;%
    tidyr::pivot_longer(cols = -&#39;Sample&#39;, names_to = &#39;Protein&#39;, values_to = &#39;Abundance&#39;) %&gt;%
    dplyr::left_join(summSigFeats[, c(&#39;Protein&#39;, &#39;Gene&#39;, &#39;pVal2&#39;, &#39;inDDA&#39;)], by = &#39;Protein&#39;) %&gt;%
    dplyr::left_join(recurInfo, by = &#39;Sample&#39;) %&gt;%
    dplyr::mutate(Gene = paste0(Gene, &#39;\n(p = &#39;, round(pVal2, 4), &#39;)&#39;),
                  Gene = dplyr::case_when(inDDA == &#39;Yes&#39; ~ paste0(&#39;*&#39;, Gene),
                                          inDDA == &#39;No&#39; ~ Gene),
                  Gene = factor(Gene, levels = unique(Gene)))
  
  return(list(overlapFeatTab = summSigFeats, sigFeatTab1 = sigFeats1, sigFeatTab2 = sigFeats2))
}</code></pre>
<div id="plasma-proteomics" class="section level1">
<h1>Plasma Proteomics</h1>
<pre class="r"><code># Load preprocessed data
proPlasma_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm_DIA.rds&#39;)
proPlasma_DDA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm.rds&#39;)
new_proPlasma_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/new_proPlasmaNorm_DIA.rds&#39;)</code></pre>
<div id="two-dia-both-from-ag-krij." class="section level2">
<h2>Two DIA both from AG Krij.</h2>
<p>Compute Pearson correlations of all common features between Plasma
Proteomics datasets. Note that both Baseline and Follow-up samples were
used.<br />
New DIA (590) vs DIA (421) Plasma Proteomics both from AG Krijgsveld
=&gt; 387 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(new_proPlasma_DIA, proPlasma_DIA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.25, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;Two DIA Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display associations of feature recurrence predictive power between
Plasma Proteomics datasets<br />
Two DIA Plasma Proteomics both from AG Krijgsveld (significance level =
0.05)<br />
<strong>Baseline Plasma</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proPlasma_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(new_proPlasma_DIA, proPlasma_DIA, &#39;New DIA&#39;, &#39;DIA&#39;,
                                       condition = &#39;Baseline&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of New DIA&#39;, y = &#39;-log10(pVal) of DIA&#39;,
       title = &#39;Two DIA Baseline Plasma Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="new-dia-vs-dda-both-from-ag-krij." class="section level2">
<h2>New DIA vs DDA both from AG Krij.</h2>
<p>Compute Pearson correlations of all common features. Note that both
Baseline and Follow-up samples were used.<br />
New DIA (590) vs DDA (435) Plasma Proteomics both from AG Krijgsveld
=&gt; 354 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(new_proPlasma_DIA, proPlasma_DDA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.4, 1, 0.2))) +
  labs(y = &#39;Frequency&#39;, title = &#39;New DIA vs DDA Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-5-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display associations of feature recurrence predictive power<br />
New DIA vs DDA Plasma Proteomics both from AG Krijgsveld (significance
level = 0.05)<br />
<strong>Baseline Plasma</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proPlasma_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(new_proPlasma_DIA, proPlasma_DDA, &#39;New DIA&#39;, &#39;DDA&#39;,
                                       condition = &#39;Baseline&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of New DIA&#39;, y = &#39;-log10(pVal) of DDA&#39;,
       title = &#39;New DIA vs DDA Baseline Plasma Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="dia-vs-dda-both-from-ag-krij." class="section level2">
<h2>DIA vs DDA both from AG Krij.</h2>
<p>Compute Pearson correlations of all common features between Plasma
Proteomics datasets. Note that both Baseline and Follow-up samples were
used.<br />
DIA (421) vs DDA (435) Plasma Proteomics both from AG Krijgsveld =&gt;
341 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(proPlasma_DIA, proPlasma_DDA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.5, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA vs DDA Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-7-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display associations of feature recurrence predictive power between
Plasma Proteomics datasets<br />
DIA vs DDA Plasma Proteomics both from AG Krijgsveld (significance level
= 0.05)<br />
<strong>Baseline Plasma</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(proPlasma_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proPlasma_DIA, proPlasma_DDA, &#39;DIA&#39;, &#39;DDA&#39;,
                                       condition = &#39;Baseline&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA&#39;, y = &#39;-log10(pVal) of DDA&#39;,
       title = &#39;DIA vs DDA Baseline Plasma Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="short-summary" class="section level2">
<h2>Short summary</h2>
<p>Display summarized information of common significant features between
two DIA datasets. Columns ‘pVal1’ and ‘pVal2’ present feature
significance from input DIA data1 and data2. Score is product of
-log10(pVal1) and -log10(pVal2) and inDDA indicates whether features are
identified and tested significant in DDA data.<br />
<strong>Baseline Plasma</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proPlasma_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Pinpoint significant features overlapped across datasets
summSigFeats &lt;- findOverlapSigFeats(proPlasma_DIA, new_proPlasma_DIA, proPlasma_DDA,
                                    condition = &#39;Baseline&#39;, prot_gene_tbl = featInfo)
summSigFeats$overlapFeatTab</code></pre>
<pre><code>     Protein       pVal1        pVal2    Score       Gene inDDA
1 A0A075B6Q5 0.033449266 0.0001022974 5.887897   IGHV3-64   Yes
2 A0A0B4J1U3 0.002811172 0.0081712684 5.325985   IGLV1-36   Yes
3     Q08380 0.012355979 0.0089038068 3.912461   LGALS3BP   Yes
4     P01833 0.009400026 0.0410112931 2.811466       PIGR   Yes
5 A0A0J9YXX1 0.029229374 0.0157511456 2.765649 IGHV5-10-1    No</code></pre>
<p>Visualize top common significant features. Gene name starting with
asterisk indicates that feature is supported by DDA dataset.</p>
<pre class="r"><code># Visualize most promising significant features
ggplot(summSigFeats$sigFeatTab1, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  labs(title = &#39;DIA Baseline Plasma Proteomics (AG Krijgsveld)&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summSigFeats$sigFeatTab2, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  labs(title = &#39;New DIA Baseline Plasma Proteomics (AG Krijgsveld)&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-11-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tissue-proteomics" class="section level1">
<h1>Tissue Proteomics</h1>
<pre class="r"><code># Load preprocessed data
proTissue_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm_DIA.rds&#39;)
proTissue_DDA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm.rds&#39;)
proTissue_DIA_Klin &lt;- readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)
# Extract only Method Development samples
proTissue_DIA_Klin &lt;- proTissue_DIA_Klin[, colnames(proTissue_DIA_Klin) %in% colnames(proTissue_DDA)]
new_proTissue_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/new_proTissueNorm_DIA.rds&#39;)</code></pre>
<div id="dia-from-ag-klin.-vs-new-dia-from-krij."
class="section level2">
<h2>DIA from AG Klin. vs New DIA from Krij.</h2>
<p>Compute Pearson correlations of all common features between Tissue
Proteomics datasets. Note that both Tumor and Normal samples were
used.<br />
DIA Tissue Proteomics from AG Klingmüller (7693) vs New DIA Tissue
Proteomics from AG Krijgsveld (6065) =&gt; 5763 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(proTissue_DIA_Klin, new_proTissue_DIA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.5, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA Proteomics from AG Klingmüller vs New DIA from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display associations of feature recurrence predictive power between
Tissue Proteomics datasets<br />
DIA Tissue Proteomics from AG Klingmüller vs New DIA Tissue Proteomics
from AG Krijgsveld (significance level = 0.05)<br />
<strong>Tumor Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proTissue_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proTissue_DIA_Klin, new_proTissue_DIA, &#39;DIA (Klin.)&#39;, &#39;DIA (Krij.)&#39;,
                                       condition = &#39;Tumor&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA (Klin.)&#39;, y = &#39;-log10(pVal) of DIA (Krij.)&#39;,
       title = &#39;DIA Tumor Tissue Proteomics from AG Klingmüller vs New DIA from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normal Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proTissue_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proTissue_DIA_Klin, new_proTissue_DIA, &#39;DIA (Klin.)&#39;, &#39;DIA (Krij.)&#39;,
                                       condition = &#39;Normal&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 2) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 4, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA (Klin.)&#39;, y = &#39;-log10(pVal) of DIA (Krij.)&#39;,
       title = &#39;DIA Normal Tissue Proteomics from AG Klingmüller vs New DIA from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="dia-from-ag-klin.-vs-dda-from-krij." class="section level2">
<h2>DIA from AG Klin. vs DDA from Krij.</h2>
<p>Compute Pearson correlations of all common features. Note that both
Tumor and Normal samples were used.<br />
DIA Tissue Proteomics from AG Klingmüller (7693) vs DDA Tissue
Proteomics from AG Krijgsveld (5193) =&gt; 4955 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(proTissue_DIA_Klin, proTissue_DDA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.5, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA Proteomics from AG Klingmüller vs DDA from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display associations of feature recurrence predictive power<br />
DIA Tissue Proteomics from AG Klingmüller vs DDA Tissue Proteomics from
AG Krijgsveld (significance level = 0.05)<br />
<strong>Tumor Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(proTissue_DDA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, PG.Genes) %&gt;%
  dplyr::rename(Genes = PG.Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proTissue_DIA_Klin, proTissue_DDA, &#39;DIA (Klin.)&#39;, &#39;DDA (Krij.)&#39;,
                                       condition = &#39;Tumor&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA (Klin.)&#39;, y = &#39;-log10(pVal) of DDA (Krij.)&#39;,
       title = &#39;DIA Tumor Tissue Proteomics from AG Klingmüller vs DDA from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normal Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(proTissue_DDA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, PG.Genes) %&gt;%
  dplyr::rename(Genes = PG.Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proTissue_DIA_Klin, proTissue_DDA, &#39;DIA (Klin.)&#39;, &#39;DDA (Krij.)&#39;,
                                       condition = &#39;Normal&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA (Klin.)&#39;, y = &#39;-log10(pVal) of DDA (Krij.)&#39;,
       title = &#39;DIA Normal Tissue Proteomics from AG Klingmüller vs DDA from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="new-dia-vs-dda-both-from-ag-krij.-1" class="section level2">
<h2>New DIA vs DDA both from AG Krij.</h2>
<p>Compute Pearson correlations of all common features. Note that both
Tumor and Normal samples were used.<br />
New DIA (6065) vs DDA (5193) Tissue Proteomics both from AG Krijgsveld
=&gt; 4693 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(new_proTissue_DIA, proTissue_DDA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.5, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;New DIA Proteomics vs DDA both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display associations of feature recurrence predictive power<br />
New DIA vs DDA Tissue Proteomics both from AG Krijgsveld (significance
level = 0.05)<br />
<strong>Tumor Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proTissue_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proTissue_DIA_Klin, proTissue_DDA, &#39;DIA&#39;, &#39;DDA&#39;,
                                       condition = &#39;Tumor&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA&#39;, y = &#39;-log10(pVal) of DDA&#39;,
       title = &#39;New DIA vs DDA Tumor Tissue Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normal Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proTissue_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Prepare association test results including color and label information
combinedAssoTestRes &lt;- combineFeatAsso(proTissue_DIA_Klin, proTissue_DDA, &#39;DIA&#39;, &#39;DDA&#39;,
                                       condition = &#39;Normal&#39;, prot_gene_tbl = featInfo)

ggplot(combinedAssoTestRes, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance, label=Label)) +
  geom_point(size = 3) +
  scale_color_manual(values = c(&#39;grey&#39;, &#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;)) +
  geom_text_repel(size = 5, show.legend = F) +
  labs(x = &#39;-log10(pVal) of DIA&#39;, y = &#39;-log10(pVal) of DDA&#39;,
       title = &#39;DIA vs DDA Normal Tissue Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="short-summary-1" class="section level2">
<h2>Short summary</h2>
<p>Display summarized information of common significant features between
two DIA datasets. Columns ‘pVal1’ and ‘pVal2’ present feature
significance from input DIA data1 and data2. Score is product of
-log10(pVal1) and -log10(pVal2) and inDDA indicates whether features are
identified and tested significant in DDA data.<br />
<strong>Tumor Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proTissue_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Pinpoint significant features overlapped across datasets
summSigFeats &lt;- findOverlapSigFeats(proTissue_DIA_Klin, new_proTissue_DIA, proTissue_DDA,
                                    condition = &#39;Tumor&#39;, prot_gene_tbl = featInfo)
summSigFeats$overlapFeatTab</code></pre>
<pre><code>   Protein      pVal1        pVal2    Score    Gene inDDA
1   Q9NWW5 0.03895378 0.0007931469 4.370207    CLN6    No
2   Q9BZK7 0.03942589 0.0022047024 3.730517 TBL1XR1   Yes
3   P54687 0.03951762 0.0040669261 3.354700   BCAT1   Yes
4   O00330 0.02652668 0.0106858133 3.107224    PDHX   Yes
5   Q9UDW1 0.04087258 0.0066321570 3.024780  UQCR10    No
6   Q07960 0.02967816 0.0239376295 2.476056 ARHGAP1   Yes
7   Q9HAV7 0.04652870 0.0139831517 2.470571  GRPEL1   Yes
8   Q9Y6N5 0.04832079 0.0137933978 2.447943    SQOR    No
9   Q14318 0.04721250 0.0311698835 1.997221   FKBP8    No
10  P99999 0.03901655 0.0466208726 1.875639    CYCS   Yes</code></pre>
<p>Visualize top common significant features. Gene name starting with
asterisk indicates that feature is supported by DDA dataset.</p>
<pre class="r"><code># Visualize most promising significant features
ggplot(summSigFeats$sigFeatTab1, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;, nrow = 2) +
  labs(title = &#39;DIA Tumor Tissue Proteomics (AG Klingmüller)&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(summSigFeats$sigFeatTab2, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;, nrow = 2) +
  labs(title = &#39;New DIA Tumor Tissue Proteomics (AG Krijgsveld)&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-23-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display summarized information of common significant features between
two DIA datasets<br />
<strong>Normal Tissue</strong></p>
<pre class="r"><code># Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(new_proTissue_DIA), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::select(Proteins, Genes) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))

# Pinpoint significant features overlapped across datasets
summSigFeats &lt;- findOverlapSigFeats(proTissue_DIA_Klin, new_proTissue_DIA, proTissue_DDA,
                                    condition = &#39;Normal&#39;, prot_gene_tbl = featInfo)
summSigFeats$overlapFeatTab</code></pre>
<pre><code>   Protein        pVal1        pVal2    Score      Gene inDDA
1   Q14515 0.0006768175 0.0077668531 6.686928   SPARCL1    No
2   Q9UKU9 0.0076051935 0.0008064958 6.554569   ANGPTL2   Yes
3   P52789 0.0012205888 0.0118559218 5.611456       HK2   Yes
4   Q8WW59 0.0121711868 0.0013118904 5.518267    SPRYD4    No
5   P24821 0.0020087112 0.0095673804 5.445968       TNC    No
6   P11310 0.0038833653 0.0058614182 5.380881     ACADM    No
7   Q9BTZ2 0.0098764345 0.0031593559 5.014305     DHRS4   Yes
8   Q9NXR1 0.0271796502 0.0006973055 4.942430      NDE1    No
9   Q7Z4W1 0.0079422922 0.0054055701 4.761156      DCXR   Yes
10  Q8N4T8 0.0030891945 0.0148174888 4.591639      CBR4    No
11  P55263 0.0093529476 0.0058705737 4.527462       ADK    No
12  Q86YM7 0.0025874080 0.0225389044 4.261185    HOMER1    No
13  Q92506 0.0096270736 0.0080788990 4.219836   HSD17B8    No
14  O00264 0.0150121756 0.0073741831 3.888344    PGRMC1    No
15  Q8NBQ5 0.0053042987 0.0197610772 3.877665  HSD17B11   Yes
16  Q8N3P4 0.0079974330 0.0197740506 3.573172      VPS8    No
17  P07942 0.0257530247 0.0060200928 3.528592     LAMB1   Yes
18  Q5VWJ9 0.0031328548 0.0436099499 3.406559     SNX30    No
19  Q6BCY4 0.0257297302 0.0078343659 3.347617    CYB5R2    No
20  P43405 0.0069996334 0.0281261356 3.342051       SYK    No
21  P15924 0.0047061595 0.0391473370 3.275251       DSP    No
22  Q05823 0.0038040566 0.0480160608 3.190719    RNASEL    No
23  Q9Y3B7 0.0308911885 0.0080078417 3.166038    MRPL11    No
24  Q5S007 0.0048602097 0.0435204444 3.149172     LRRK2    No
25  Q9H8L6 0.0058782826 0.0424523344 3.060808     MMRN2    No
26  Q99805 0.0055609120 0.0484352562 2.964769    TM9SF2    No
27  P01700 0.0126801800 0.0282956779 2.936893  IGLV1-47    No
28  Q9Y3E5 0.0332268326 0.0107104817 2.912949     PTRH2    No
29  Q9P0M6 0.0302211478 0.0126557398 2.883933 MACROH2A2    No
30  Q99417 0.0165837122 0.0245683631 2.865643     MYCBP    No
31  Q8IX01 0.0490233872 0.0076138360 2.774245     SUGP2    No
32  P41236 0.0262502075 0.0177567716 2.767523    PPP1R2    No
33  P02647 0.0084218909 0.0467017930 2.760588     APOA1    No
34  O14683 0.0139476883 0.0331611639 2.744968   TP53I11    No
35  Q96PE3 0.0386223785 0.0114629755 2.742525    INPP4A    No
36  Q99828 0.0308591483 0.0153509298 2.740054      CIB1    No
37  Q9H7Z7 0.0326141421 0.0155086038 2.689884    PTGES2    No
38  P98194 0.0131569834 0.0380486439 2.670160    ATP2C1    No
39  Q9UHP3 0.0416723982 0.0129141634 2.607015     USP25    No
40  P84157 0.0227309266 0.0281828317 2.547269     MXRA7    No
41  Q14151 0.0184321495 0.0368201933 2.487015     SAFB2    No
42  Q13137 0.0228479992 0.0308971118 2.478274  CALCOCO2    No
43  P45954 0.0271209586 0.0277261915 2.439516    ACADSB    No
44  Q5RI15 0.0220187875 0.0342142453 2.429122     COX20    No
45  Q9Y6A9 0.0184374901 0.0424739780 2.379244     SPCS1    No
46  P04632 0.0176657276 0.0457318935 2.348460    CAPNS1    No
47  Q9H2M9 0.0267129550 0.0332319760 2.326003  RAB3GAP2    No
48  P37235 0.0452452321 0.0203542439 2.273890    HPCAL1    No
49  Q6P1X6 0.0281029306 0.0342983148 2.272156   C8orf82    No
50  Q13753 0.0407253138 0.0234158183 2.266603     LAMC2   Yes
51  Q13751 0.0295065208 0.0332626799 2.261527     LAMB3   Yes
52  O95817 0.0338361463 0.0319509492 2.199335      BAG3   Yes
53  Q92947 0.0246449976 0.0456861971 2.155429      GCDH    No
54  Q9NY61 0.0229812501 0.0490308255 2.145832      AATF    No
55  P28068 0.0272974588 0.0431862040 2.134154   HLA-DMB    No
56  P0DP23 0.0308535976 0.0423845116 2.073870     CALM1    No
57  Q8NDA8 0.0444605772 0.0307217623 2.045011     MROH1    No
58  Q9Y3C6 0.0453521794 0.0316421538 2.014745     PPIL1    No
59  Q9NVA1 0.0345402571 0.0452373183 1.965226     UQCC1    No
60  Q6NZI2 0.0316368954 0.0499793333 1.951562    CAVIN1    No</code></pre>
<p>Visualize top common significant features. Gene name starting with
asterisk indicates that feature is supported by DDA dataset.</p>
<pre class="r"><code># Extract top features
sigFeats &lt;- summSigFeats$overlapFeatTab$Protein[1:6]
sigFeatTab1 &lt;- dplyr::filter(summSigFeats$sigFeatTab1, Protein %in% sigFeats)
sigFeatTab2 &lt;- dplyr::filter(summSigFeats$sigFeatTab2, Protein %in% sigFeats)

# Visualize most promising significant features
ggplot(sigFeatTab1, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  labs(title = &#39;DIA Tumor Tissue Proteomics (AG Klingmüller)&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(sigFeatTab2, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  labs(title = &#39;New DIA Tumor Tissue Proteomics (AG Krijgsveld)&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-25-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="dia-from-ag-klin.-vs-dia-from-krij." class="section level2">
<h2>DIA from AG Klin. vs DIA from Krij.</h2>
<p><font size=4> <strong>Old DIA Tissue Proteomics from AG Krijgsveld
got bad data quality.</strong> </font></p>
<p>Compute Pearson correlations of all common features. Note that both
Tumor and Normal samples were used.<br />
DIA Tissue Proteomics from AG Klingmüller (7693) vs DIA Tissue
Proteomics from AG Krijgsveld (5057) =&gt; 4893 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(proTissue_DIA_Klin, proTissue_DIA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.8, 0.6, 0.2))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA Proteomics from AG Klingmüller vs AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="dia-vs-dda-both-from-ag-krij.-1" class="section level2">
<h2>DIA vs DDA both from AG Krij.</h2>
<p><font size=4> <strong>Old DIA Tissue Proteomics from AG Krijgsveld
got bad data quality.</strong> </font></p>
<p>Compute Pearson correlations of all common features. Note that both
Tumor and Normal samples were used.<br />
DIA (5057) vs DDA (5193) Tissue Proteomics both from AG Krijgsveld =&gt;
4358 common features</p>
<pre class="r"><code># Compute correlations of common features between two datasets
corrRes &lt;- calcFeatCorr(proTissue_DIA, proTissue_DDA)

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(-0.6, -0.4, -0.2, 0, 0.2, 0.4, 0.6)) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA vs DDA Tissue Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.2                
 [5] purrr_1.0.1                 readr_2.1.4                
 [7] tidyr_1.3.0                 tibble_3.2.1               
 [9] tidyverse_2.0.0             ggraph_2.1.0               
[11] igraph_1.5.0                ggrepel_0.9.3              
[13] ggplot2_3.4.2               SummarizedExperiment_1.30.2
[15] Biobase_2.60.0              GenomicRanges_1.52.0       
[17] GenomeInfoDb_1.36.1         IRanges_2.34.1             
[19] S4Vectors_0.38.1            BiocGenerics_0.46.0        
[21] MatrixGenerics_1.12.2       matrixStats_1.0.0          
[23] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0        viridisLite_0.4.2       farver_2.1.1           
 [4] viridis_0.6.3           bitops_1.0-7            fastmap_1.1.1          
 [7] RCurl_1.98-1.12         tweenr_2.0.2            promises_1.2.0.1       
[10] digest_0.6.33           timechange_0.2.0        lifecycle_1.0.3        
[13] processx_3.8.2          magrittr_2.0.3          compiler_4.3.0         
[16] rlang_1.1.1             sass_0.4.7              tools_4.3.0            
[19] utf8_1.2.3              yaml_2.3.7              knitr_1.43             
[22] labeling_0.4.2          S4Arrays_1.0.4          graphlayouts_1.0.0     
[25] DelayedArray_0.26.6     plyr_1.8.8              withr_2.5.0            
[28] grid_4.3.0              polyclip_1.10-4         fansi_1.0.4            
[31] git2r_0.32.0            colorspace_2.1-0        scales_1.2.1           
[34] MASS_7.3-60             cli_3.6.1               rmarkdown_2.23         
[37] crayon_1.5.2            generics_0.1.3          rstudioapi_0.15.0      
[40] tzdb_0.4.0              httr_1.4.6              cachem_1.0.8           
[43] ggforce_0.4.1           zlibbioc_1.46.0         XVector_0.40.0         
[46] vctrs_0.6.3             Matrix_1.6-0            jsonlite_1.8.7         
[49] callr_3.7.3             hms_1.1.3               jquerylib_0.1.4        
[52] glue_1.6.2              ps_1.7.5                stringi_1.7.12         
[55] gtable_0.3.3            later_1.3.1             munsell_0.5.0          
[58] pillar_1.9.0            htmltools_0.5.5         proDA_1.14.0           
[61] GenomeInfoDbData_1.2.10 R6_2.5.1                tidygraph_1.2.3        
[64] rprojroot_2.0.3         evaluate_0.21           lattice_0.21-8         
[67] extraDistr_1.9.1        highr_0.10              httpuv_1.6.11          
[70] bslib_0.5.0             Rcpp_1.0.11             gridExtra_2.3          
[73] whisker_0.4.1           xfun_0.39               fs_1.6.2               
[76] getPass_0.2-2           pkgconfig_2.0.3        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
