<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Comparisons between DIA and DDA proteomics datasets</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparisons between DIA and DDA proteomics
datasets</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree8088fce3349a9e05c998f7283f2e7c6ffe55b67ftargetblank8088fcea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/8088fce3349a9e05c998f7283f2e7c6ffe55b67f" target="_blank">8088fce</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree8088fce3349a9e05c998f7283f2e7c6ffe55b67ftargetblank8088fcea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/8088fce3349a9e05c998f7283f2e7c6ffe55b67f" target="_blank">8088fce</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/feature_selection.Rmd
    Untracked:  code/about.Rmd
    Untracked:  code/functions.R
    Untracked:  code/license.Rmd
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Klingmuller/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/mofa/
    Untracked:  data/patient_metadata.tsv
    Untracked:  data/stats/
    Untracked:  output/TAC_2023/
    Untracked:  output/group_meeting/
    Untracked:  output/preliminary_analysis_DIA.html

Unstaged changes:
    Modified:   analysis/preliminary_analysis_untargeted.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/comparisons_DIA_DDA.Rmd</code>)
and HTML (<code>docs/comparisons_DIA_DDA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/b0836405d1d46e75d85da360f9aa744d48e4ae9c/analysis/comparisons_DIA_DDA.Rmd" target="_blank">b083640</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-27
</td>
<td>
Preprocess DIA tissue proteomics from AG Klingmüller and compare it with
DIA and DDA data from AG Krijgsveld
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/comparisons_DIA_DDA.html" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/7cee36ee4cca86c197f370aed6285533cfeed9be/analysis/comparisons_DIA_DDA.Rmd" target="_blank">7cee36e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Compute correlations of common features between DIA and DDA Proteomics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Compare DIA and DDA Plasma and Tissue
Proteomics generated by AG Krijgsveld. Basically, correlations of common
features between DIA and DDA Proteomics were computed to see data
reproducibility using different data acquisition methods.<br />
</p>
<p>Update: Compare DIA Tissue Proteomics generated by AG Klingmüller
with aforementioned DIA and DDA Tissue Proteomics from AG Krijgsveld.
</font></p>
<p>Load libraries</p>
<pre class="r"><code>library(SummarizedExperiment)
library(ggrepel)
library(igraph)
library(ggraph)
library(tidyverse)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<div id="plasma-proteomics" class="section level1">
<h1>Plasma Proteomics</h1>
<p>Compute Pearson correlations of all common features between Plasma
Proteomics datasets. Note that both Baseline and Follow-up samples were
used.<br />
</p>
<p>DIA (445) vs DDA (435) Plasma Proteomics both from AG Krijgsveld
=&gt; 351 common features</p>
<pre class="r"><code># Load preprocessed data
proPlasma_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm_DIA.rds&#39;)
proPlasma_DDA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm.rds&#39;)
# Extract data matrix
datMat_DIA &lt;- SummarizedExperiment::assay(proPlasma_DIA)
datMat_DDA &lt;- SummarizedExperiment::assay(proPlasma_DDA)
# Make two matrices share same sample and feature spaces and in same row and column order
# Remove patient samples not measured in DIA proteomics from DDA proteomics
commonSmps &lt;- colnames(datMat_DIA)
datMat_DIA &lt;- datMat_DIA[, commonSmps]
datMat_DDA &lt;- datMat_DDA[, commonSmps]
# Keep only common features between DIA and DDA proteomics
# Select first protein as representative if there are more than one
rownames(datMat_DIA) &lt;- stringr::str_remove(rownames(datMat_DIA), &#39;;.*&#39;)
rownames(datMat_DDA) &lt;- stringr::str_remove(rownames(datMat_DDA), &#39;;.*&#39;)
commonFeats &lt;- intersect(rownames(datMat_DIA), rownames(datMat_DDA))
datMat_DIA &lt;- datMat_DIA[commonFeats,]
datMat_DDA &lt;- datMat_DDA[commonFeats,]

# Compute feature correlations between DIA and DDA proteomics
if (identical(rownames(datMat_DIA), rownames(datMat_DDA)) &amp;
    identical(colnames(datMat_DIA), colnames(datMat_DDA))) {
  corrs &lt;- sapply(seq_len(nrow(datMat_DIA)), function(i) {
    corr &lt;- cor.test(datMat_DIA[i,], datMat_DDA[i,], method = &#39;pearson&#39;,
                     use = &quot;pairwise.complete.obs&quot;)
    corr$estimate
  })
  corrRes &lt;- data.frame(Feature = rownames(datMat_DIA), Correlation = corrs) %&gt;%
    dplyr::arrange(Correlation) %&gt;%
    dplyr::mutate(Feature = factor(Feature, levels = Feature))
}

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.5, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA vs DDA Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-1-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggplot(corrRes, aes(x=Feature, y=Correlation)) +
#   geom_point(size = 2, color = &#39;grey60&#39;) +
#   geom_hline(yintercept = 0.7, color = &#39;red&#39;, linewidth = 0.7) +
#   geom_text(x=-7, y=0.7, label = &#39;0.7&#39;, size = 4.5, color = &#39;red&#39;) +
#   coord_cartesian(clip = &#39;off&#39;) +
#   th + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),
#              panel.grid.major.x = element_blank())</code></pre>
<pre class="r"><code># Display significant associations between features and cancer recurrence
# DIA
# Prepare protein-gene table for changing protein names to gene names
featInfo_DIA &lt;- tibble::as_tibble(rowData(proPlasma_DIA), rownames = &#39;Protein.Groups&#39;)
# Subset baseline samples
smpBaseIdx &lt;- which(colData(proPlasma_DIA)$Condition == &#39;Baseline&#39;)
proBase_DIA &lt;- proPlasma_DIA[, smpBaseIdx]
datMat_DIA &lt;- SummarizedExperiment::assay(proBase_DIA)
smpAnno_DIA &lt;- tibble::as_tibble(colData(proBase_DIA), rownames = &#39;Sample&#39;)

# Perform analysis
# Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat_DIA, design = ~ smpAnno_DIA$Recurrence)
all_proDAProBaseRes_DIA &lt;- proDA::test_diff(fit,
                                            contrast = Intercept + `smpAnno_DIA$RecurrenceYes` - Intercept,
                                            sort_by = &#39;pval&#39;) %&gt;%
  dplyr::select(-c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::rename(Protein = name, pVal = pval, pValAdj = adj_pval, tStat = t_statistic) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(Protein,
                                       from = featInfo_DIA$Protein.Groups,
                                       to = featInfo_DIA$Genes,
                                       warn_missing = F))
# Identify significant differentially abundant entities
proDAProBaseRes_DIA &lt;- dplyr::filter(all_proDAProBaseRes_DIA, pVal &lt; 0.05)
# Display features significantly separating recurrence and non-recurrence patients
proDAProBaseRes_DIA

# DDA
# Prepare protein-gene table for changing protein names to gene names
featInfo_DDA &lt;- tibble::as_tibble(rowData(proPlasma_DDA), rownames = &#39;PG.Groups&#39;)
# Subset baseline samples
smpBaseIdx &lt;- which(colData(proPlasma_DDA)$Condition == &#39;Baseline&#39;)
proBase_DDA &lt;- proPlasma_DDA[, smpBaseIdx]

# Perform analysis
proBaseRes_DDA &lt;- doSOA(proBase_DDA, meta_var = &#39;Recurrence&#39;, num_PCfeats = 30)
tFeatSigRes_DDA &lt;- proBaseRes_DDA$tFeatSigRes %&gt;%
  dplyr::select(-c(Test, Var2)) %&gt;%
  dplyr::rename(Protein = Var1, tStat = Stat)

# Display features that can significantly separate recurrence and non-recurrence
# patients
dplyr::mutate(tFeatSigRes_DDA, Gene = plyr::mapvalues(Protein,
                                                      from = featInfo_DDA$PG.Groups,
                                                      to = featInfo_DDA$PG.Genes,
                                                      warn_missing = F))

# Display overlaps between DIA and DDA significant proteins
# Select first protein as representative if there are more than one
sigFeats_DIA &lt;- stringr::str_remove(proDAProBaseRes_DIA$Protein, &#39;;.*&#39;)
sigFeats_DDA &lt;- stringr::str_remove(tFeatSigRes_DDA$Protein, &#39;;.*&#39;)
cat(&#39;Overlaps of significant features:&#39;,
    plyr::mapvalues(intersect(sigFeats_DIA, sigFeats_DDA),
                    from = featInfo_DIA$Protein.Groups,
                    to = featInfo_DIA$Genes,
                    warn_missing = F), &#39;\n&#39;,
    sep = &#39;  &#39;)

# Compute correlation of p-values between DIA and DDA roteomics
pVal_DIA &lt;- dplyr::select(all_proDAProBaseRes_DIA, c(Protein, pVal)) %&gt;%
  dplyr::mutate(Protein = stringr::str_remove(Protein, &#39;;.*&#39;)) %&gt;%
  dplyr::filter(Protein %in% commonFeats) %&gt;%
  dplyr::arrange(Protein)
pVal_DDA &lt;- dplyr::select(proBaseRes_DDA$tFeatRes, c(Var1, pVal)) %&gt;%
  dplyr::rename(Protein = Var1) %&gt;%
  dplyr::mutate(Protein = stringr::str_remove(Protein, &#39;;.*&#39;)) %&gt;%
  dplyr::filter(Protein %in% commonFeats) %&gt;%
  dplyr::arrange(Protein)
if (identical(pVal_DDA$Protein, pVal_DIA$Protein)) {
  cat(&#39;Correlation of p-values:&#39;, cor.test(pVal_DIA$pVal, pVal_DDA$pVal, method = &#39;pearson&#39;,
                                           use = &quot;pairwise.complete.obs&quot;)$estimate)
}</code></pre>
</div>
<div id="tissue-proteomics" class="section level1">
<h1>Tissue Proteomics</h1>
<p>Compute Pearson correlations of all common features between Tissue
Proteomics datasets. Note that both Baseline and Follow-up samples were
used.</p>
<pre class="r"><code># Load preprocessed data
proTissue_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm_DIA.rds&#39;)
proTissue_DDA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm.rds&#39;)
proTissue_DIA_klin &lt;- readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)</code></pre>
<p>DIA (5658) vs DDA (5193) Tissue Proteomics both from AG Krijgsveld
=&gt; 4636 common features</p>
<pre class="r"><code># Extract data matrix
datMat_DIA &lt;- SummarizedExperiment::assay(proTissue_DIA)
datMat_DDA &lt;- SummarizedExperiment::assay(proTissue_DDA)

# Make two matrices share same sample and feature spaces and in same row and column order
# Remove patient samples not measured in DIA proteomics from DDA proteomics
commonSmps &lt;- colnames(datMat_DIA)
datMat_DIA &lt;- datMat_DIA[, commonSmps]
datMat_DDA &lt;- datMat_DDA[, commonSmps]
# Keep only common features between DIA and DDA proteomics
# Select first protein as representative if there are more than one
rownames(datMat_DIA) &lt;- stringr::str_remove(rownames(datMat_DIA), &#39;;.*&#39;)
rownames(datMat_DDA) &lt;- stringr::str_remove(rownames(datMat_DDA), &#39;;.*&#39;)
commonFeats &lt;- intersect(rownames(datMat_DIA), rownames(datMat_DDA))
datMat_DIA &lt;- datMat_DIA[commonFeats,]
datMat_DDA &lt;- datMat_DDA[commonFeats,]

# Compute feature correlations between DIA and DDA proteomics
if (identical(rownames(datMat_DIA), rownames(datMat_DDA)) &amp;
    identical(colnames(datMat_DIA), colnames(datMat_DDA))) {
  corrs &lt;- sapply(seq_len(nrow(datMat_DIA)), function(i) {
    corr &lt;- cor.test(datMat_DIA[i,], datMat_DDA[i,], method = &#39;pearson&#39;,
                     use = &quot;pairwise.complete.obs&quot;)
    corr$estimate
  })
  corrRes &lt;- data.frame(Feature = rownames(datMat_DIA), Correlation = corrs) %&gt;%
    dplyr::arrange(Correlation) %&gt;%
    dplyr::mutate(Feature = factor(Feature, levels = Feature))
}

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(-0.6, -0.4, -0.2, 0, 0.2, 0.4, 0.6)) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA vs DDA Tissue Proteomics both from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br/> <br/></p>
<p>DIA Tissue Proteomics from AG Klingmüller (8257) vs DDA Tissue
Proteomics from AG Krijgsveld (5193) =&gt; 5022 common features</p>
<pre class="r"><code># Extract data matrix
dat1 &lt;- SummarizedExperiment::assay(proTissue_DIA_klin)
dat2 &lt;- SummarizedExperiment::assay(proTissue_DDA)

# Make two matrices share same sample and feature spaces and in same row and column order
# Keep only common samples
commonSmps &lt;- intersect(colnames(dat1), colnames(dat2))
dat1 &lt;- dat1[, commonSmps]
dat2 &lt;- dat2[, commonSmps]
# Keep only common features
# Select first protein as representative if there are more than one
rownames(dat1) &lt;- stringr::str_remove(rownames(dat1), &#39;;.*&#39;)
rownames(dat2) &lt;- stringr::str_remove(rownames(dat2), &#39;;.*&#39;)
commonFeats &lt;- intersect(rownames(dat1), rownames(dat2))
dat1 &lt;- dat1[commonFeats,]
dat2 &lt;- dat2[commonFeats,]

# Compute feature correlations between DIA and DDA proteomics
if (identical(rownames(dat1), rownames(dat2)) &amp;
    identical(colnames(dat1), colnames(dat2))) {
  corrs &lt;- sapply(seq_len(nrow(dat1)), function(i) {
    corr &lt;- cor.test(dat1[i,], dat2[i,], method = &#39;pearson&#39;, use = &quot;pairwise.complete.obs&quot;)
    corr$estimate
  })
  corrRes &lt;- data.frame(Feature = rownames(dat1), Correlation = corrs) %&gt;%
    dplyr::arrange(Correlation) %&gt;%
    dplyr::mutate(Feature = factor(Feature, levels = Feature))
}

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.5, 1, 0.25))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA Proteomics from AG Klingmüller vs DDA Proteomics from AG Krijgsveld&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" />
<br/> <br/></p>
<p>DIA Tissue Proteomics from AG Krijgsveld (5658) vs AG Klingmüller
(8257) =&gt; 5451 common features</p>
<pre class="r"><code># Extract data matrix
dat1 &lt;- SummarizedExperiment::assay(proTissue_DIA)
dat2 &lt;- SummarizedExperiment::assay(proTissue_DIA_klin)

# Make two matrices share same sample and feature spaces and in same row and column order
# Keep only common samples
commonSmps &lt;- intersect(colnames(dat1), colnames(dat2))
dat1 &lt;- dat1[, commonSmps]
dat2 &lt;- dat2[, commonSmps]
# Keep only common features
# Select first protein as representative if there are more than one
rownames(dat1) &lt;- stringr::str_remove(rownames(dat1), &#39;;.*&#39;)
rownames(dat2) &lt;- stringr::str_remove(rownames(dat2), &#39;;.*&#39;)
commonFeats &lt;- intersect(rownames(dat1), rownames(dat2))
dat1 &lt;- dat1[commonFeats,]
dat2 &lt;- dat2[commonFeats,]

# Compute feature correlations between DIA and DDA proteomics
if (identical(rownames(dat1), rownames(dat2)) &amp;
    identical(colnames(dat1), colnames(dat2))) {
  corrs &lt;- sapply(seq_len(nrow(dat1)), function(i) {
    corr &lt;- cor.test(dat1[i,], dat2[i,], method = &#39;pearson&#39;, use = &quot;pairwise.complete.obs&quot;)
    corr$estimate
  })
  corrRes &lt;- data.frame(Feature = rownames(dat1), Correlation = corrs) %&gt;%
    dplyr::arrange(Correlation) %&gt;%
    dplyr::mutate(Feature = factor(Feature, levels = Feature))
}

ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-0.8, 0.6, 0.2))) +
  labs(y = &#39;Frequency&#39;, title = &#39;DIA Proteomics from AG Krijgsveld vs AG Klingmüller&#39;) +
  th</code></pre>
<p><img src="figure/comparisons_DIA_DDA.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.2                
 [5] purrr_1.0.1                 readr_2.1.4                
 [7] tidyr_1.3.0                 tibble_3.2.1               
 [9] tidyverse_2.0.0             ggraph_2.1.0               
[11] igraph_1.4.3                ggrepel_0.9.3              
[13] ggplot2_3.4.2               SummarizedExperiment_1.30.1
[15] Biobase_2.60.0              GenomicRanges_1.52.0       
[17] GenomeInfoDb_1.36.0         IRanges_2.34.0             
[19] S4Vectors_0.38.1            BiocGenerics_0.45.3        
[21] MatrixGenerics_1.12.0       matrixStats_1.0.0          
[23] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0        viridisLite_0.4.2       farver_2.1.1           
 [4] viridis_0.6.3           bitops_1.0-7            fastmap_1.1.1          
 [7] RCurl_1.98-1.12         tweenr_2.0.2            promises_1.2.0.1       
[10] digest_0.6.31           timechange_0.2.0        lifecycle_1.0.3        
[13] processx_3.8.1          magrittr_2.0.3          compiler_4.3.0         
[16] rlang_1.1.1             sass_0.4.6              tools_4.3.0            
[19] utf8_1.2.3              yaml_2.3.7              knitr_1.43             
[22] labeling_0.4.2          S4Arrays_1.0.4          graphlayouts_1.0.0     
[25] DelayedArray_0.26.3     withr_2.5.0             grid_4.3.0             
[28] polyclip_1.10-4         fansi_1.0.4             git2r_0.32.0           
[31] colorspace_2.1-0        scales_1.2.1            MASS_7.3-60            
[34] cli_3.6.1               rmarkdown_2.22          crayon_1.5.2           
[37] generics_0.1.3          rstudioapi_0.14         tzdb_0.4.0             
[40] httr_1.4.6              cachem_1.0.8            ggforce_0.4.1          
[43] zlibbioc_1.46.0         XVector_0.40.0          vctrs_0.6.2            
[46] Matrix_1.5-4.1          jsonlite_1.8.4          callr_3.7.3            
[49] hms_1.1.3               jquerylib_0.1.4         glue_1.6.2             
[52] ps_1.7.5                stringi_1.7.12          gtable_0.3.3           
[55] later_1.3.1             munsell_0.5.0           pillar_1.9.0           
[58] htmltools_0.5.5         GenomeInfoDbData_1.2.10 R6_2.5.1               
[61] tidygraph_1.2.3         rprojroot_2.0.3         evaluate_0.21          
[64] lattice_0.21-8          highr_0.10              httpuv_1.6.11          
[67] bslib_0.4.2             Rcpp_1.0.10             gridExtra_2.3          
[70] whisker_0.4.1           xfun_0.39               fs_1.6.2               
[73] getPass_0.2-2           pkgconfig_2.0.3        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
