<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>MOFA: Downstream analysis</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MOFA: Downstream analysis</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree9a5191e865f9c1d40a49ea46a7c1c020e986c53etargetblank9a5191ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/9a5191e865f9c1d40a49ea46a7c1c020e986c53e" target="_blank">9a5191e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree9a5191e865f9c1d40a49ea46a7c1c020e986c53etargetblank9a5191ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/9a5191e865f9c1d40a49ea46a7c1c020e986c53e" target="_blank">9a5191e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/comparison_funcs.R
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R
    Untracked:  data/Discovery/
    Untracked:  data/MethodDev/
    Untracked:  data/aliquot_metadata.xlsx
    Untracked:  data/patient_metadata.xlsx
    Untracked:  data/sample_metadata.xlsx
    Untracked:  output/Discovery/
    Untracked:  output/MethodDev/
    Untracked:  output/SC_meeting/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Discovery_mofa_downstream.Rmd</code>) and HTML
(<code>docs/Discovery_mofa_downstream.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/9a5191e865f9c1d40a49ea46a7c1c020e986c53e/analysis/Discovery_mofa_downstream.Rmd" target="_blank">9a5191e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-05-15
</td>
<td>
Update analysis report
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/b83f6736534c624c72cb444130bf22766780a846/analysis/Discovery_mofa_downstream.Rmd" target="_blank">b83f673</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-05-15
</td>
<td>
Just for quick update of analysis report webpage
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/df818f2365f9e09d63fc82c5fb78ed7a7c78749c/analysis/Discovery_mofa_downstream.Rmd" target="_blank">df818f2</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-04-21
</td>
<td>
Redo investigation of impacts of sample missing levels on data and
retrain MOFA models using only common samples
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/2842b9a63f3f105b3e9571476d97dcac9e2f89da/analysis/Discovery_mofa_downstream.Rmd" target="_blank">2842b9a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-04-14
</td>
<td>
Train MOFA models on Klin. NTP and Untar. Lipid. and refine functions
for MOFA training and analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Do multi-omics integration using MOFA.
</font></p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;MOFA2&#39;)
library(&#39;reticulate&#39;)
# use_python(&#39;/Users/qianwu/opt/anaconda3/bin/python&#39;)
library(&#39;AnnotationDbi&#39;)
library(&#39;org.Hs.eg.db&#39;)
# This annotation object is accessed using AnnotationDbi:select()
hs &lt;- org.Hs.eg.db
library(&#39;msigdbr&#39;)
library(&#39;clusterProfiler&#39;)
library(&#39;ggrepel&#39;)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)
source(&#39;./code/mofa_funcs.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<pre class="r"><code># Prepare data tables to train MOFA models
# Load normalized datasets
comb_proTissue_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsnBC.rds&#39;)
comb_proNormal_Klin &lt;- comb_proTissue_Klin[, colData(comb_proTissue_Klin)$Condition == &#39;Normal&#39;]

comb_proTissue_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)

dis_lipTissue_Hopf &lt;- readRDS(&#39;./data/Discovery/AG_Hopf/lipTissueVsn_WBC25.rds&#39;)
dis_lipNormal_Hopf &lt;- dis_lipTissue_Hopf[, colData(dis_lipTissue_Hopf)$Condition == &#39;Normal&#39;]

mDev_lipTissue_Hopf &lt;- readRDS(&#39;./data/MethodDev/AG_Hopf/lipTissueVsn.rds&#39;)
mDev_lipNormal_Hopf &lt;- mDev_lipTissue_Hopf[, colData(mDev_lipTissue_Hopf)$Condition == &#39;Normal&#39;]

dis_lipBase_Hopf &lt;- readRDS(&#39;./data/Discovery/AG_Hopf/lipBaseVsn_B1WBC25.rds&#39;)

mDev_lipPlasma_Hopf &lt;- readRDS(&#39;./data/MethodDev/AG_Hopf/lipPlasmaVsn.rds&#39;)
mDev_lipBase_Hopf &lt;- mDev_lipPlasma_Hopf[, colData(mDev_lipPlasma_Hopf)$TimePoint == &#39;Baseline&#39;]

# Sanity check dimensionality of each dataset (Feature x Sample)
dataList &lt;- list(Comb_Tissue_Proteomics_Klin = comb_proTissue_Klin,
                 Comb_Tissue_Proteomics_Krij = comb_proTissue_Krij,
                 Dis_Tissue_Lipidomics_Hopf = dis_lipTissue_Hopf,
                 MDev_Tissue_Lipidomics_Hopf = mDev_lipTissue_Hopf,
                 Dis_Baseline_Lipidomics_Hopf = dis_lipBase_Hopf,
                 MDev_Plasma_Lipidomics_Hopf = mDev_lipPlasma_Hopf)
# lapply(dataList, dim)

# Prepare significant features for feature selection
# soaRes &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proNormalRes_Klin_SV1.rds&#39;)
# sigFeats_comb_proNormal_Klin &lt;- soaRes$sig.feat.tab$Var1
# soaRes &lt;- doSOA(dis_lipNormal_Hopf, meta_var = &#39;Recurrence&#39;, use_proDA = T)
# sigFeats_dis_lipNormal_Hopf &lt;- soaRes$featSigAssoRes$Var1
# soaRes &lt;- doSOA(mDev_lipNormal_Hopf, meta_var = &#39;Recurrence&#39;, use_proDA = T)
# sigFeats_mDev_lipNormal_Hopf &lt;- soaRes$featSigAssoRes$Var1
# soaRes &lt;- doSOA(dis_lipBase_Hopf, meta_var = &#39;Recurrence&#39;, use_proDA = T)
# sigFeats_dis_lipBase_Hopf &lt;- soaRes$featSigAssoRes$Var1
# soaRes &lt;- doSOA(mDev_lipBase_Hopf, meta_var = &#39;Recurrence&#39;, use_proDA = T)
# sigFeats_mDev_lipBase_Hopf &lt;- soaRes$featSigAssoRes$Var1

# Convert SE objects to long data for creating MOFA object through &#39;create_mofa_from_df&#39;
# to include metadata
# comb_proNormalTab_Klin &lt;- mofa_summExp2df(comb_proNormal_Klin, smp_type = &#39;Tissue&#39;,
#                                           data_acqui = &#39;Combined DIA&#39;, data_modal = &#39;Proteomics&#39;) %&gt;%
#   dplyr::filter(feature %in% sigFeats_comb_proNormal_Klin)
# dis_lipNormalTab_Hopf &lt;- mofa_summExp2df(dis_lipNormal_Hopf, smp_type = &#39;Tissue&#39;,
#                                          data_acqui = &#39;Discovery Untargeted&#39;, data_modal = &#39;Lipidomics&#39;) %&gt;%
#   dplyr::filter(feature %in% sigFeats_dis_lipNormal_Hopf)
# mDev_lipNormalTab_Hopf &lt;- mofa_summExp2df(mDev_lipNormal_Hopf, smp_type = &#39;Tissue&#39;,
#                                           data_acqui = &#39;MethodDev Untargeted&#39;, data_modal = &#39;Lipidomics&#39;) %&gt;%
#   dplyr::filter(feature %in% sigFeats_mDev_lipNormal_Hopf)
# dis_lipBaseTab_Hopf &lt;- mofa_summExp2df(dis_lipBase_Hopf, smp_type = &#39;Plasma&#39;,
#                                        data_acqui = &#39; Discovery Untargeted&#39;, data_modal = &#39;Lipidomics&#39;) %&gt;%
#   dplyr::filter(feature %in% sigFeats_dis_lipBase_Hopf)
# mDev_lipBaseTab_Hopf &lt;- mofa_summExp2df(mDev_lipBase_Hopf, smp_type = &#39;Plasma&#39;,
#                                         data_acqui = &#39;MethodDev Untargeted&#39;, data_modal = &#39;Lipidomics&#39;) %&gt;%
#   dplyr::filter(feature %in% sigFeats_mDev_lipBase_Hopf)

# Prepare sample spaces from different datasets to remove unwanted samples later 
smps_CombNTP &lt;- stringr::str_replace(colnames(comb_proNormal_Klin), &#39;_NG&#39;, &#39;_B&#39;)
smps_DisNTL &lt;- stringr::str_replace(colnames(dis_lipNormal_Hopf), &#39;_NG&#39;, &#39;_B&#39;)
smps_MDevNTL &lt;- stringr::str_replace(colnames(mDev_lipNormal_Hopf), &#39;_NG&#39;, &#39;_B&#39;)
smps_DisBPL &lt;- stringr::str_replace(colnames(dis_lipBase_Hopf), &#39;_V1&#39;, &#39;_B&#39;)
smps_MDevBPL &lt;- colnames(mDev_lipBase_Hopf)</code></pre>
<div id="comb.-dia-ntp-klin.-all-unt.-ntl" class="section level1">
<h1>Comb. DIA NTP (Klin.) + All Unt. NTL</h1>
<p>Combined DIA Normal Tissue Proteomics (MethodDev + Discovery) from AG
Klingmüller + Discovery and MethodDev Untargeted Normal Tissue
Lipidomics<br />
(since features defined by RT/CCS/MZ have difficulty being combined
together from different cohorts)<br />
<strong>Note that only statistically significant recurrence-related
features were used to train a MOFA model</strong></p>
<pre class="r"><code># Train MOFA model
# Remove unwanted samples
# smpViewCounts &lt;- table(c(smps_CombNTP, smps_DisNTL, smps_MDevNTL))
# rmSmps &lt;- names(smpViewCounts)[smpViewCounts == 1]
# tmp_dis_lipNormalTab_Hopf &lt;- dplyr::filter(dis_lipNormalTab_Hopf, !sample %in% rmSmps)
# tmp_mDev_lipNormalTab_Hopf &lt;- dplyr::filter(mDev_lipNormalTab_Hopf, !sample %in% rmSmps)
# 
# mofaObject &lt;- trainMOFA(list(comb_proNormalTab_Klin, tmp_dis_lipNormalTab_Hopf,
#                              tmp_mDev_lipNormalTab_Hopf),
#                         view_data = T, train_mofa = T, num_factors = 10,
#                         save_path = &#39;./data/Discovery/mofa/sigFeats_CombDIANTPKlin_DisMDevUntNTL&#39;)

# Load trained model
mofaObject &lt;- readRDS(&#39;./data/Discovery/mofa/sigFeats_CombDIANTPKlin_DisMDevUntNTL.rds&#39;)</code></pre>
<p>Overview model<br />
Note that samples in only one view were removed from model training</p>
<pre class="r"><code># Overview model
MOFA2::plot_data_overview(mofaObject)</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
MOFA2::plot_factor_cor(mofaObject)</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject, max_r2 = 15)</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-4-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display and visualize significant recurrence-related factors</p>
<pre class="r"><code># Display significant associations between learned factors (Var1) and cancer recurrence
# (Var2) and visualize significant recurrence-related factors
sigFactor &lt;- mofa_vizSigFactor(mofaObject, smpGroup = &#39;Baseline Group&#39;, show_res = F)
sigFactor$sigAssoRes</code></pre>
<pre><code>     Var1       Var2         pVal      pValAdj     Stat   Test
1 Factor1 Recurrence 2.528678e-05 0.0002528678 4.810616 T-test
2 Factor4 Recurrence 3.151088e-02 0.1575544104 2.235392 T-test</code></pre>
<pre class="r"><code>facTab &lt;- dplyr::mutate(sigFactor$tab4Plot,
                        Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                        Label = dplyr::case_when(Recurrence %in% &#39;Yes&#39; &amp; Factor1 &gt; 0 ~ sample,
                                                 Recurrence %in% &#39;No&#39; &amp; Factor1 &lt; 0 ~ sample))

ggplot(facTab, aes(x=Recurrence, y=Factor1, col=Recurrence, fill=Recurrence, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 4, show.legend = F) +
  geom_text_repel(show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             size = 9, show.legend = F, hjust = 0.4, vjust = 1) +
  scale_color_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  scale_fill_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  th +
  theme(axis.title = element_text(size = 28),
        axis.text = element_text(size = 16),
        legend.title = element_text(size = 26), legend.text = element_text(size = 24))</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(facTab, aes(x=Recurrence, y=Factor4, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 4, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             show.legend = F) +
  scale_color_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  scale_fill_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  th</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-5-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(sigFactor$tab4Plot, aes(x=Factor1, y=Factor4, col=Recurrence)) +
  geom_point(size = 4) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  th</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-5-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Compute ROC-AUC score
response &lt;- ifelse(facTab$Recurrence == &#39;Yes&#39;, yes = &#39;1&#39;, no = &#39;0&#39;) %&gt;%
  factor(levels = c(&#39;1&#39;, &#39;0&#39;))
predictor &lt;- facTab$Factor1
rocRes &lt;- suppressMessages(
  pROC::roc(response = response, predictor = predictor, plot = T,
            legacy.axes = T, print.auc = T, print.auc.x = 0.4,
            xlab = &#39;False positive rate&#39;, ylab = &#39;True positive rate&#39;,
            main = &#39;ROC Curve for Factor1&#39;,
            cex.lab = 1.2, cex.main = 1.1, col = &#39;#377eb8&#39;, lwd = 4,
            direction = &#39;&lt;&#39;)
)</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-5-4.png" width="960" style="display: block; margin: auto;" />
=&gt; Factor1 is highly correlated with PC1 from Klin. NTP and explains
variance also in MDev. NTL, but in Dis. NTL, which implies that
significant features captured from MDev. and Dis. NTL may provide
distinct information probably due to biological variations in different
patient cohorts. In addition, Dis. Tissue Lipidomics got worse sample or
data quality than MDev., which is shown by their metadata-assisted
quality controls (Tumor vs Normal).</p>
<p>Compare Factor1 with recurrence-related PC1 learned from Combined
Klin. Normal Tissue Proteomics</p>
<pre class="r"><code># Visualize correlation between significant factor (MOFA) and significant PC (PCA)
# Prepare significant factor table
facTab &lt;- sigFactor$tab4Plot %&gt;%
  dplyr::select(sample, Factor1, Recurrence) %&gt;%
  dplyr::mutate(sample = stringr::str_replace(sample, &#39;_B&#39;, &#39;_NG&#39;))
# Prepare significant PC table
proNormalRes_Klin &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proNormalRes_Klin_SV1.rds&#39;)
# Show significant PCs
# proNormalRes_Klin$sig.pc.tab
pcTab &lt;- proNormalRes_Klin$SOA.res$pcTab %&gt;%
  dplyr::select(Sample, `PC1 (11.1%)`) %&gt;%
  dplyr::rename(sample = Sample)
# Combine all information
combinedTab &lt;- dplyr::left_join(facTab, pcTab, by = &#39;sample&#39;)
# Make a plot and compute correlation
ggplot(combinedTab, aes(x=Factor1, y=`PC1 (11.1%)`)) +
  geom_point(aes(col=Recurrence), size = 4) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 7) +
  th</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize enrichment analysis results</p>
<pre class="r"><code># Prepare ranked feature list
# Tidy up weight matrix and feature metadata to reduce information loss
proGeneTab &lt;- rowData(comb_proNormal_Klin) %&gt;%
  tibble::as_tibble(rownames = &#39;Proteins&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Proteins&#39;)
tidyWeiMat &lt;- mofa_keepUniFeats(mofaObject, factor = 1, view = &#39;Combined DIA Normal Proteomics&#39;,
                                feat_anno = proGeneTab, to_genes = T)
# Rank feature weights
rankedGeneList &lt;- mofa_rankFeatList(tidyWeiMat, factor = 1)
# Make Recurrence group positive weights
rankedGeneList &lt;- sort(-rankedGeneList, decreasing = T)

# Do enrichment analysis - Hallmark gene sets (H)
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)
# Run GSEA
gseaResH &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabH,
                                  minGSSize = 10, maxGSSize = 500,
                                  pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                  by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.6, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;, title = &#39;MSigDB:H - Factor1&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># =&gt; Factor4: HALLMARK_MYC_TARGETS_V1</code></pre>
</div>
<div id="comb.-dia-ntp-klin.-all-unt.-bpl" class="section level1">
<h1>Comb. DIA NTP (Klin.) + All Unt. BPL</h1>
<p>Combined DIA Normal Tissue Proteomics (MethodDev + Discovery) from AG
Klingmüller + Discovery and MethodDev Untargeted Baseline Plasma
Lipidomics (since features defined by RT/CCS/MZ have difficulty being
combined together from different cohorts)<br />
<strong>Note that only statistically significant recurrence-related
features were used to train a MOFA model</strong></p>
<pre class="r"><code># Train MOFA model
# Remove unwanted samples
# smpViewCounts &lt;- table(c(smps_CombNTP, smps_DisBPL, smps_MDevBPL))
# rmSmps &lt;- names(smpViewCounts)[smpViewCounts == 1]
# tmp_dis_lipBaseTab_Hopf &lt;- dplyr::filter(dis_lipBaseTab_Hopf, !sample %in% rmSmps)
# tmp_mDev_lipBaseTab_Hopf &lt;- dplyr::filter(mDev_lipBaseTab_Hopf, !sample %in% rmSmps)
# 
# mofaObject &lt;- trainMOFA(list(comb_proNormalTab_Klin, tmp_dis_lipBaseTab_Hopf,
#                              tmp_mDev_lipBaseTab_Hopf),
#                         view_data = T, train_mofa = T, num_factors = 10,
#                         save_path = &#39;./data/Discovery/mofa/sigFeats_CombDIANTPKlin_DisMDevUntBPL&#39;)

# Load trained model
mofaObject &lt;- readRDS(&#39;./data/Discovery/mofa/sigFeats_CombDIANTPKlin_DisMDevUntBPL.rds&#39;)
# views_names(mofaObject) &lt;- c(&#39;Dis. Unt. BPL&#39;, &#39;Comb. DIA NTP&#39;, &#39;MDev. Unt. BPL&#39;)</code></pre>
<p>Overview model<br />
Note that samples in only one view were removed from model training</p>
<pre class="r"><code># Overview model
MOFA2::plot_data_overview(mofaObject) #+</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(text = element_text(size = 28))
# ggsave(&#39;./output/Discovery/2nd_TAC/mofa_data_UntBPL_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# MOFA2::plot_factor_cor(mofaObject)

# Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject, max_r2 = 15) #+</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># theme(axis.text.x = element_text(size = 22), axis.text.y = element_text(size =20),
#       legend.title = element_text(size = 22), legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/2nd_TAC/mofa_varExplain_UntBPL_combined_proNormal_Klin.png&#39;,
       # device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p>Display and visualize significant recurrence-related factors</p>
<pre class="r"><code># Display significant associations between learned factors (Var1) and cancer recurrence
# (Var2) and visualize significant recurrence-related factors
sigFactor &lt;- mofa_vizSigFactor(mofaObject, smpGroup = &#39;Baseline Group&#39;, show_res = F)
sigFactor$sigAssoRes</code></pre>
<pre><code>     Var1       Var2        pVal     pValAdj     Stat   Test
1 Factor1 Recurrence 2.35568e-05 0.000235568 4.833669 T-test</code></pre>
<pre class="r"><code>facTab &lt;- dplyr::mutate(sigFactor$tab4Plot,
                        Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                        Label = dplyr::case_when(Recurrence %in% &#39;Yes&#39; &amp; Factor1 &gt; 0 ~ sample,
                                                 Recurrence %in% &#39;No&#39; &amp; Factor1 &lt; 0 ~ sample))

ggplot(facTab, aes(x=Recurrence, y=Factor1, col=Recurrence, fill=Recurrence, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 4, show.legend = F) +
  geom_text_repel(show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             size = 9, show.legend = F, hjust = 0.4, vjust = 1) +
  scale_color_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  scale_fill_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  th +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 20),
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/mofa_fac1_UntBPL_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)

# Compute ROC-AUC score
response &lt;- ifelse(facTab$Recurrence == &#39;Yes&#39;, yes = &#39;1&#39;, no = &#39;0&#39;) %&gt;%
  factor(levels = c(&#39;1&#39;, &#39;0&#39;))
predictor &lt;- facTab$Factor1
rocRes &lt;- pROC::roc(response = response, predictor = predictor, plot = T,
                    legacy.axes = T, print.auc = T, print.auc.x = 0.4,
                    xlab = substitute(paste(bold(&#39;False positive rate&#39;))),
                    ylab = substitute(paste(bold(&#39;True positive rate&#39;))),
                    main = &#39;ROC Curve for Factor1&#39;,
                    cex.lab = 1.8, cex.axis = 1.2, cex.main = 1.8, col = &#39;#377eb8&#39;,
                    lwd = 6, direction = &#39;&lt;&#39;, quiet = T)</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-12-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># pROC::ci.auc(rocRes)</code></pre>
<p>Compare Factor1 with recurrence-related PC1 learned from Combined
Klin. Normal Tissue Proteomics and another Factor1 from model built on
Klin. NTP and Unt. NTL</p>
<pre class="r"><code># Visualize correlation between significant factors (MOFA) and significant PC (PCA)
# Prepare significant factor table
facTab &lt;- sigFactor$tab4Plot %&gt;%
  dplyr::select(sample, Factor1, Recurrence) %&gt;%
  dplyr::mutate(sample = stringr::str_replace(sample, &#39;_B&#39;, &#39;_NG&#39;))
# Prepare significant PC table
proNormalRes_Klin &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proNormalRes_Klin_SV1.rds&#39;)
# Show significant PCs
# proNormalRes_Klin$sig.pc.tab
pcTab &lt;- proNormalRes_Klin$SOA.res$pcTab %&gt;%
  dplyr::select(Sample, `PC1 (11.1%)`) %&gt;%
  dplyr::rename(sample = Sample)
# Combine all information
combinedTab &lt;- dplyr::left_join(facTab, pcTab, by = &#39;sample&#39;) %&gt;%
  dplyr::mutate(Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
# Make a plot and compute correlation
ggplot(combinedTab, aes(x=Factor1, y=`PC1 (11.1%)`)) +
  geom_point(aes(col=Recurrence), size = 4) +
  scale_color_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  scale_fill_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 9) +
  th +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 20),
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/mofa_fac1_vs_pc1_UntBPL_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)

# Prepare significant factor tables
facTab &lt;- sigFactor$tab4Plot %&gt;%
  dplyr::select(sample, Factor1, Recurrence) %&gt;%
  dplyr::rename(Factor1_BPL = Factor1)
mofaObj &lt;- readRDS(&#39;./data/Discovery/mofa/sigFeats_CombDIANTPKlin_DisMDevUntNTL.rds&#39;)
facTab2 &lt;- get_factors(mofaObj)[[&#39;Baseline Group&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor1) %&gt;%
  dplyr::rename(Factor1_NTL = Factor1)
# Combine all information and remove unwanted samples
combinedTab &lt;- dplyr::left_join(facTab, facTab2, by = &#39;sample&#39;)
# Make a plot and compute correlation
ggplot(combinedTab, aes(x=Factor1_BPL, y=Factor1_NTL)) +
  geom_point(aes(col=Recurrence), size = 4) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 7) +
  th</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-13-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize enrichment analysis results</p>
<pre class="r"><code># Prepare ranked feature list
# Tidy up weight matrix and feature metadata to reduce information loss
proGeneTab &lt;- rowData(comb_proNormal_Klin) %&gt;%
  tibble::as_tibble(rownames = &#39;Proteins&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Proteins&#39;)
tidyWeiMat &lt;- mofa_keepUniFeats(mofaObject, factor = 1, view = &#39;Combined DIA Normal Proteomics&#39;,
                                feat_anno = proGeneTab, to_genes = T)
# Rank feature weights
rankedGeneList &lt;- mofa_rankFeatList(tidyWeiMat, factor = 1)
# Make Recurrence group positive weights
rankedGeneList &lt;- sort(-rankedGeneList, decreasing = T)

# Do enrichment analysis - Hallmark gene sets (H)
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)
# Run GSEA
gseaResH &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabH,
                                  minGSSize = 10, maxGSSize = 500,
                                  pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                  by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.6, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;, title = &#39;MSigDB:H - Factor1&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/Discovery_mofa_downstream.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] proDA_1.16.0                limma_3.58.1               
 [3] pcaMethods_1.94.0           lubridate_1.9.3            
 [5] forcats_1.0.0               stringr_1.5.1              
 [7] dplyr_1.1.4                 purrr_1.0.2                
 [9] readr_2.1.5                 tidyr_1.3.1                
[11] tibble_3.2.1                tidyverse_2.0.0            
[13] SummarizedExperiment_1.32.0 GenomicRanges_1.54.1       
[15] GenomeInfoDb_1.38.8         MatrixGenerics_1.14.0      
[17] matrixStats_1.3.0           ggrepel_0.9.5              
[19] ggplot2_3.5.1               clusterProfiler_4.10.1     
[21] msigdbr_7.5.1               org.Hs.eg.db_3.18.0        
[23] AnnotationDbi_1.64.1        IRanges_2.36.0             
[25] S4Vectors_0.40.2            Biobase_2.62.0             
[27] BiocGenerics_0.48.1         reticulate_1.36.1          
[29] MOFA2_1.12.1                workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] splines_4.3.2           later_1.3.2             bitops_1.0-7           
  [4] ggplotify_0.1.2         filelock_1.0.3          polyclip_1.10-6        
  [7] basilisk.utils_1.14.1   pROC_1.18.5             lifecycle_1.0.4        
 [10] rstatix_0.7.2           rprojroot_2.0.4         processx_3.8.4         
 [13] lattice_0.22-6          MASS_7.3-60.0.1         backports_1.4.1        
 [16] magrittr_2.0.3          sass_0.4.9              rmarkdown_2.26         
 [19] jquerylib_0.1.4         yaml_2.3.8              httpuv_1.6.15          
 [22] cowplot_1.1.3           DBI_1.2.2               RColorBrewer_1.1-3     
 [25] abind_1.4-5             zlibbioc_1.48.2         Rtsne_0.17             
 [28] ggraph_2.2.1            RCurl_1.98-1.14         yulab.utils_0.1.4      
 [31] tweenr_2.0.3            git2r_0.33.0            GenomeInfoDbData_1.2.11
 [34] enrichplot_1.22.0       tidytree_0.4.6          pheatmap_1.0.12        
 [37] codetools_0.2-20        DelayedArray_0.28.0     DOSE_3.28.2            
 [40] ggforce_0.4.2           tidyselect_1.2.1        aplot_0.2.2            
 [43] farver_2.1.1            viridis_0.6.5           jsonlite_1.8.8         
 [46] tidygraph_1.3.1         tools_4.3.2             treeio_1.26.0          
 [49] Rcpp_1.0.12             glue_1.7.0              gridExtra_2.3          
 [52] SparseArray_1.2.4       xfun_0.43               qvalue_2.34.0          
 [55] HDF5Array_1.30.1        withr_3.0.0             fastmap_1.1.1          
 [58] basilisk_1.14.3         rhdf5filters_1.14.1     fansi_1.0.6            
 [61] callr_3.7.6             digest_0.6.35           timechange_0.3.0       
 [64] R6_2.5.1                gridGraphics_0.5-1      colorspace_2.1-0       
 [67] GO.db_3.18.0            RSQLite_2.3.6           utf8_1.2.4             
 [70] generics_0.1.3          data.table_1.15.4       graphlayouts_1.1.1     
 [73] httr_1.4.7              S4Arrays_1.2.1          scatterpie_0.2.2       
 [76] whisker_0.4.1           uwot_0.2.2              pkgconfig_2.0.3        
 [79] gtable_0.3.5            blob_1.2.4              XVector_0.42.0         
 [82] shadowtext_0.1.3        htmltools_0.5.8.1       carData_3.0-5          
 [85] fgsea_1.28.0            scales_1.3.0            png_0.1-8              
 [88] corrplot_0.92           ggfun_0.1.4             knitr_1.46             
 [91] rstudioapi_0.16.0       tzdb_0.4.0              reshape2_1.4.4         
 [94] nlme_3.1-164            cachem_1.0.8            rhdf5_2.46.1           
 [97] parallel_4.3.2          HDO.db_0.99.1           pillar_1.9.0           
[100] grid_4.3.2              vctrs_0.6.5             ggpubr_0.6.0           
[103] promises_1.3.0          car_3.1-2               evaluate_0.23          
[106] cli_3.6.2               compiler_4.3.2          rlang_1.1.3            
[109] crayon_1.5.2            ggsignif_0.6.4          labeling_0.4.3         
[112] ps_1.7.6                getPass_0.2-4           plyr_1.8.9             
[115] fs_1.6.3                stringi_1.8.3           viridisLite_0.4.2      
[118] BiocParallel_1.36.0     babelgene_22.9          munsell_0.5.1          
[121] Biostrings_2.70.3       lazyeval_0.2.2          GOSemSim_2.28.1        
[124] Matrix_1.6-5            dir.expiry_1.10.0       hms_1.1.3              
[127] patchwork_1.2.0         bit64_4.0.5             Rhdf5lib_1.24.2        
[130] statmod_1.5.0           KEGGREST_1.42.0         highr_0.10             
[133] broom_1.0.5             igraph_2.0.3            memoise_2.0.1          
[136] bslib_0.7.0             ggtree_3.10.1           fastmatch_1.1-4        
[139] bit_4.0.5               ape_5.8                 gson_0.1.0             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
