<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Preprocessing: DIA Proteomics of Discovery cohort (AG Krijgsveld)</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Preprocessing: DIA Proteomics of Discovery
cohort (AG Krijgsveld)</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree282daae0fdbb7078dd8d66521af4c5d654c63cfbtargetblank282daaea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/282daae0fdbb7078dd8d66521af4c5d654c63cfb" target="_blank">282daae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree282daae0fdbb7078dd8d66521af4c5d654c63cfbtargetblank282daaea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/282daae0fdbb7078dd8d66521af4c5d654c63cfb" target="_blank">282daae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Discovery/
    Ignored:    data/MethodDev/
    Ignored:    data/aliquot_metadata.xlsx
    Ignored:    data/metadata_latest/
    Ignored:    data/patient_metadata.xlsx
    Ignored:    data/sample_metadata.xlsx
    Ignored:    output/.DS_Store
    Ignored:    output/Discovery/
    Ignored:    output/MethodDev/
    Ignored:    output/SC_meeting/

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/data_availability.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R

Unstaged changes:
    Modified:   analysis/FeatSelection_ML.Rmd
    Modified:   analysis/MDev_Dis_02_soa_annotated_untargeted.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Discovery_01_preprocessing_DIA_Krij.Rmd</code>) and HTML
(<code>docs/Discovery_01_preprocessing_DIA_Krij.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/e957730a0dc718d91bfb403da6c3e1c1f3239143/analysis/Discovery_01_preprocessing_DIA_Krij.Rmd" target="_blank">e957730</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-02
</td>
<td>
Preprocess combined Method Dev and Discovery Tissue Proteomics from AG
Klin. and rename and rearrange source code and analysis results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Preprocess Discovery Tissue and Plasma
DIA Proteomics generated by Karim Aljakouch from AG Krijgsveld,
including data cleansing, filtering, and normalization (VSN). All needed
information was then stored in SummarizedExperiment objects for further
analyses. </font></p>
<p>Q: There is no ‘1HTNWJ_TU’ and got extra Tumor sample
‘6EY5JO_II_TU’?</p>
<p>Load libraries</p>
<pre class="r"><code>library(readr)
library(readxl)
library(vsn)
library(limma)
library(visdat)
library(ggrepel)
library(SummarizedExperiment)
library(tidyverse)
# Load user-defined functions
source(&#39;./code/analysis_pipeline.R&#39;)
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<pre class="r"><code># Load sample metadata
smpMetadat &lt;- readxl::read_excel(&#39;./data/sample_metadata.xlsx&#39;)
colnames(smpMetadat) &lt;- smpMetadat[3,, drop = F]
smpMetadat &lt;- dplyr::slice(smpMetadat, -c(1:3)) %&gt;%
  dplyr::select(c(Code, Parents, Visit, `Material submitted`, `Date and time of collection or surgery`,
                  `SMART-CARE cohort identifier`)) %&gt;%
  dplyr::rename(Sample = Code, Patient = Parents, TimePoint = Visit, SmpType = `Material submitted`,
                Date = `Date and time of collection or surgery`, Cohort = `SMART-CARE cohort identifier`) %&gt;%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, &#39;^SC_T_S_|^SC_DIS_S_|^SC_S_|_P&#39;),
                Patient = stringr::str_remove(Patient, &#39;^/THRX_SPACE/THRX_DB/&#39;),
                TimePoint = dplyr::case_when(TimePoint == &#39;PRETHERAPEUTIC&#39; ~ &#39;Baseline&#39;,
                                             TimePoint == &#39;FOLLOW-UP&#39; ~ &#39;Follow-up&#39;,
                                             TimePoint == &#39;RECURRENCE&#39; ~ &#39;Recurrence&#39;),
                SmpType = dplyr::case_when(SmpType == &#39;EDTA_PLASMA&#39; ~ &#39;Plasma&#39;,
                                           SmpType == &#39;FRESH_FROZEN_TISSUE&#39; ~ &#39;Tissue&#39;),
                Condition = dplyr::case_when(grepl(&#39;_TU|_TG&#39;, Sample) ~ &#39;Tumor&#39;,
                                             grepl(&#39;_NG&#39;, Sample) ~ &#39;Normal&#39;,
                                             !grepl(&#39;_TU|_TG|_NG&#39;, Sample) ~ TimePoint),
                TimePoint = dplyr::case_when(TimePoint %in% &#39;Recurrence&#39; ~ &#39;Follow-up&#39;,
                                             !TimePoint %in% &#39;Recurrence&#39; ~ TimePoint),
                Date = stringr::str_extract(Date, &#39;^\\d+-\\d+-\\d+&#39;),
                Cohort = dplyr::case_when(Cohort == &#39;DISCOVERY_COHORT&#39; ~ &#39;Discovery&#39;,
                                          Cohort == &#39;METHOD_DEVELOPMENT_COHORT&#39; ~ &#39;MethodDev&#39;),
                Date = as.Date(Date, format = &#39;%Y-%m-%d&#39;))

# Load patient metadata
patientMetadat &lt;- readxl::read_excel(&#39;./data/patient_metadata.xlsx&#39;)
colnames(patientMetadat) &lt;- patientMetadat[3,, drop = F]
patientMetadat &lt;- dplyr::slice(patientMetadat, -c(1:3)) %&gt;%
  dplyr::select(c(Code, Gender, `Age at diagnosis`, `Pathological stage`, `Smoking status`,
                  `Adjuvant chemotherapy`,)) %&gt;%
  dplyr::rename(Patient = Code, Age = `Age at diagnosis`, Stage = `Pathological stage`,
                Smoking = `Smoking status`, Adjuvant = `Adjuvant chemotherapy`) %&gt;%
  dplyr::mutate(Gender = dplyr::case_when(Gender == &#39;MALE&#39; ~ &#39;Male&#39;,
                                          Gender == &#39;FEMALE&#39; ~ &#39;Female&#39;),
                Smoking = dplyr::case_when(Smoking == &#39;SMOKER&#39; ~ &#39;Smoker&#39;,
                                           Smoking == &#39;EX-SMOKER&#39; ~ &#39;Ex-smoker&#39;,
                                           Smoking == &#39;NON-SMOKER&#39; ~ &#39;Non-smoker&#39;),
                Adjuvant = dplyr::case_when(Adjuvant == &#39;true&#39; ~ &#39;True&#39;,
                                            Adjuvant == &#39;false&#39; ~ &#39;False&#39;),
                Age = as.numeric(Age))
# Include patient recurrence information in patient metadata
recurPats &lt;- dplyr::filter(smpMetadat, Condition == &#39;Recurrence&#39;) %&gt;%
  dplyr::pull(Patient)
patientMetadat &lt;- dplyr::mutate(patientMetadat, Recurrence = dplyr::case_when(Patient %in% recurPats ~ &#39;Yes&#39;,
                                                                              !Patient %in% recurPats ~ &#39;No&#39;))

# Load aliquot metadata
aliquotMetadat &lt;- readxl::read_excel(&#39;./data/aliquot_metadata.xlsx&#39;)
colnames(aliquotMetadat) &lt;- aliquotMetadat[3,, drop = F]
aliquotMetadat &lt;- dplyr::slice(aliquotMetadat, -c(1:3)) %&gt;%
  dplyr::select(Code, Parents, `Tumor Cell Content (%)`, `Cohort Identifier`, `Submission to`,
                `Delivered?`) %&gt;%
  dplyr::rename(Aliquot = Code, Sample = Parents, TumorPurity = `Tumor Cell Content (%)`,
                Cohort = `Cohort Identifier`, To = `Submission to`, Delivered = `Delivered?`) %&gt;%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_T_S_|_P&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_DIS_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_DIS_&#39;),
                Cohort = dplyr::case_when(Cohort == &#39;method establishment&#39; ~ &#39;MethodDev&#39;,
                                          Cohort == &#39;DISCOVERY_COHORT&#39; ~ &#39;Discovery&#39;),
                TumorPurity = as.numeric(TumorPurity)) %&gt;%
  dplyr::filter(To == &#39;KRIJGSVELD&#39;)
# Prepare tumor purity information
tumorPurityInfo &lt;- dplyr::select(aliquotMetadat, Sample, TumorPurity)

# Combine all needed metadata
summMetadat &lt;- dplyr::left_join(smpMetadat, patientMetadat, by = &#39;Patient&#39;) %&gt;%
  dplyr::left_join(tumorPurityInfo, by = &#39;Sample&#39;)

# List sample annotations to keep in SE objects
smpAnno &lt;- c(&#39;Patient&#39;, &#39;SmpType&#39;, &#39;TimePoint&#39;, &#39;Date&#39;, &#39;Cohort&#39;, &#39;Condition&#39;,
             &#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;)</code></pre>
<div id="tissue-proteomics" class="section level1">
<h1>Tissue Proteomics</h1>
<div id="discovery-cohort" class="section level2">
<h2>Discovery cohort</h2>
<pre class="r"><code># Load unprocessed data
proTissueTab &lt;- readr::read_tsv(&#39;./data/Discovery/AG_Krijgsveld/20230815_MarcS_Discovery_Tissue.pg_matrix.tsv&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Sample = stringr::str_remove_all(Sample, &#39;^.+SC_A_|^.+SC_DIS_A_|4\\.raw$&#39;))
# Include summarized metadata
# Manually add an extra sample to summarized metadata
tmp_summMetadat &lt;- summMetadat
smp &lt;- summMetadat[summMetadat$Sample == &#39;6EY5JO_TU&#39;,]
smp[1, 1] &lt;- &#39;6EY5JO_II_TU&#39;
tmp_summMetadat &lt;- rbind(tmp_summMetadat, smp)
proTissueTab &lt;- dplyr::left_join(proTissueTab, tmp_summMetadat, by = &#39;Sample&#39;)</code></pre>
<p>Show dimensions of original data (86 samples and 7986 features)</p>
<pre class="r"><code># Preprocess data
proTissue &lt;- doPreprocessing(proTissueTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                             val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                             smpAnno = c(smpAnno, &#39;TumorPurity&#39;), do_featFilt = T,
                             cutoff = 0.5, viz_miss = T)#, save_path = &#39;./data/Discovery/AG_Krijgsveld/proTissue&#39;)
dim(proTissue$ori.data)</code></pre>
<pre><code>[1] 7986   86</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code>proTissue$ori.data.miss +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show distribution of original data</p>
<pre class="r"><code>proTissue$ori.data.dist +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (86 samples and 6077 features)</p>
<pre class="r"><code>dim(proTissue$filt.data)</code></pre>
<pre><code>[1] 6077   86</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>proTissue$filt.data.miss +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code>proTissue$vsn.data.dist +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>proTissue$vsn.feat.mean.var</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Do metadata-assisted quality control</strong></p>
<pre class="r"><code># Do single-omics analysis
proTissueSOA &lt;- doSOA(proTissue$vsn.data, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
                      do_onlyPCA = T)
# Prepare labels for samples of interest
pcTab &lt;- proTissueSOA$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;,
                                                       &#39;MJMTYR_TU&#39;, &#39;XFKHGP_TU&#39;,
                                                       &#39;XFKHGP_NG&#39;, &#39;MJMTYR_NG&#39;,
                                                       &#39;I0HVOL_NG&#39;, &#39;7EAOX7_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (31.5%)`, col=Condition, fill=Condition, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F, size = 5) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th #+ theme(axis.title = element_text(size = 28),</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># axis.text.x = element_text(size = 24), axis.text.y = element_text(size = 18),
# legend.title = element_text(size = 26), legend.text = element_text(size = 24))
# ggsave(&#39;./output/Discovery/group_meeting/quality_control_pca_Discovery_proTissue_Krij_condi.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p>=&gt; Tumor purity: I0HVOL_TU - 87% / 7EAOX7_TU - 70% / MJMTYR_TU -
40% / XFKHGP_TU - 100%<br />
(Mean of sample tumor cell contents is 74%, ranging from 40% to
100%.)</p>
<p><strong>Investigate impacts of sample missing levels on observed
features</strong><br />
Display relationship between sample missing levels and median
expressions computed by fully observed features</p>
<pre class="r"><code># Normalize unfiltered data
proTissueUnfiltVsn &lt;- doPreprocessing(proTissueTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                      val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                      smpAnno = c(smpAnno, &#39;TumorPurity&#39;), do_featFilt = F)$vsn.data
# Prepare long data for computing sample missing levels and median expressions
proTissueUnfiltVsnTab &lt;- summExp2df(proTissueUnfiltVsn, assay = &#39;Abundance&#39;,
                                    row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;) %&gt;%
  # Prepare information of sample missing levels
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(OriMissLevel = sum(is.na(Value)) / length(Value)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Prepare information of sample median expressions computed by fully observed features
  dplyr::group_by(Protein.Group) %&gt;%
  dplyr::mutate(Keep = dplyr::case_when(!any(is.na(Value)) ~ &#39;Yes&#39;,
                                        any(is.na(Value)) ~ &#39;No&#39;)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::filter(Keep %in% &#39;Yes&#39;) %&gt;%
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(MedianVsnAbun = median(Value)) %&gt;%
  dplyr::ungroup()
# Create SE object containing information of sample missing levels and median expressions
colAnno &lt;- dplyr::select(proTissueUnfiltVsnTab, Sample, OriMissLevel, MedianVsnAbun) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
proTissueUnfiltVsn &lt;- SummarizedExperiment(assays = list(Abundance = assay(proTissueUnfiltVsn)),
                                           colData = colAnno)

# Visualize relationship between sample missing levels and median expressions
ggplot(proTissueUnfiltVsnTab, aes(x=OriMissLevel, y=MedianVsnAbun)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(x = &#39;Missing level&#39;, y = &#39;Median expression&#39;) +
  th</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show significant associations between data and sample missing
levels</p>
<pre class="r"><code># Do PCA
pcaRes &lt;- doSOA(proTissueUnfiltVsn, meta_var = &#39;OriMissLevel&#39;, pca_method = &#39;pca&#39;,
                num_PCs = 20, do_onlyPCA = T)
# Show significant associations between PCs and sample missing levels
pcaRes$pcSigAssoRes %&gt;%
  dplyr::filter(grepl(&#39;PC[1-9] \\(&#39;, Var1))</code></pre>
<pre><code>         Var1         Var2     pVal  pValAdj   Stat        Test
1  PC3 (6.8%) OriMissLevel 1.36e-12 1.17e-10 -0.672 Correlation
2   PC1 (38%) OriMissLevel 2.23e-05 9.57e-04 -0.440 Correlation
3  PC4 (4.2%) OriMissLevel 1.91e-03 5.47e-02  0.330 Correlation
4  PC8 (1.9%) OriMissLevel 2.98e-02 5.59e-01  0.234 Correlation
5 PC2 (10.4%) OriMissLevel 3.25e-02 5.59e-01  0.231 Correlation</code></pre>
<pre class="r"><code># Visualize relationship between PCs and sample missing levels
pcTab &lt;- pcaRes$pcTab
ggplot(pcTab, aes(x=`PC3 (6.8%)`, y=OriMissLevel)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(y = &#39;Missing level&#39;) +
  th</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=`PC1 (38%)`, y=OriMissLevel)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(y = &#39;Missing level&#39;) +
  th</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-11-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plasma-proteomics" class="section level1">
<h1>Plasma Proteomics</h1>
<p>Due to complexity of plasma samples, we will be focusing on analysis
of Baseline samples. 63 out of 81 patients got Baseline samples and they
were extracted and preprocessed to preserve maximal amounts of reliable
features.</p>
<pre class="r"><code># Load unprocessed data
proPlasmaTab &lt;- readr::read_tsv(&#39;./data/Discovery/AG_Krijgsveld/20230811_MarcS_Discovery_Plasma.pg_matrix.tsv&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Sample = stringr::str_remove_all(Sample, &#39;^.+SC_DIS_A_|_4\\.raw$|\\.raw$|^.+KA_|_P&#39;),
                Sample = stringr::str_replace(Sample, &#39;P_&#39;, &#39;_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^SC_A_&#39;),
                Sample = dplyr::case_when(Sample == &#39;FD01075852&#39; ~ &#39;R1WTOP_V1&#39;,
                                          Sample == &#39;FD05926483&#39; ~ &#39;R1WTOP_V2&#39;,
                                          Sample == &#39;FD01282286&#39; ~ &#39;R1WTOP_V3&#39;,
                                          Sample == &#39;FD07174643&#39; ~ &#39;R1WTOP_V4&#39;,
                                          !Sample %in% c(&#39;FD01075852&#39;, &#39;FD05926483&#39;,
                                                         &#39;FD01282286&#39;, &#39;FD07174643&#39;) ~ Sample),
                Sample = stringr::str_remove_all(Sample, &#39;_FD\\d+|_LV\\d+&#39;),
                # Manually update sample names from data
                Sample = dplyr::case_when(Sample == &#39;1TK71I_V4&#39; ~ &#39;1TK71I_V3&#39;,
                                          Sample == &#39;Y4W4L7_V2&#39; ~ &#39;D1174FWT_V2&#39;,
                                          !Sample %in% c(&#39;1TK71I_V4&#39;, &#39;Y4W4L7_V2&#39;) ~ Sample))
# Include summarized metadata
proPlasmaTab &lt;- dplyr::left_join(proPlasmaTab, summMetadat, by = &#39;Sample&#39;)</code></pre>
<p>Explore time points and exact dates of plasma samples</p>
<pre class="r"><code># Explore time points of plasma samples
dplyr::select(proPlasmaTab, Sample, TimePoint, Date) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  DT::datatable()</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-19143815d351c1d2b521" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-19143815d351c1d2b521">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289"],["00VK00F1_V1","00VK00F1_V4","00VK00F1_V5","00VK00F1_V7","00VK00F1_V9","00VK00F1_V11","00VK00F1_V12","00VK00F1_V13","1MKZXR_V2","1MKZXR_V3","1N9H5W_V3","1N9H5W_V4","1TK71I_V1","1TK71I_V3","1TK71I_V6","1TK71I_V8","1TK71I_V11","1UKCKK_V2","1UKCKK_V3","1UKCKK_V4","1UKCKK_V5","1VEUNL_V2","1VEUNL_V3","1VEUNL_V4","1VGGL0_V5","1VGGL0_V6","1WFPXH_V4","1WFPXH_V5","1WFPXH_V6","7ZY8GL_V2","7ZY8GL_V4","7ZY8GL_V5","7ZY8GL_V6","8HX560EJ_V1","8HX560EJ_V2","8HX560EJ_V3","8HX560EJ_V4","8HX560EJ_V5","8HX560EJ_V6","8HX560EJ_V7","15BOK5_V3","15BOK5_V4","15BOK5_V5","20W9MFP9_V1","20W9MFP9_V2","20W9MFP9_V3","30X94F6U_V1","30X94F6U_V2","30X94F6U_V3","50VF27F5_V1","50VF27F5_V2","50VF27F5_V3","50VF27F5_V4","50VF27F5_V5","50VF27F5_V6","50VF27F5_V7","60U2R73L_V1","60U2R73L_V2","60U2R73L_V4","60U2R73L_V5","60U2R73L_V6","70W5P7RX_V1","70W5P7RX_V2","70W5P7RX_V4","70W5P7RX_V7","70W5P7RX_V10","70W5P7RX_V11","A0VA40QC_V1","A0VA40QC_V2","A0VA40QC_V3","A0VA40QC_V4","A0VA40QC_V5","A0VA40QC_V6","A0VA40QC_V7","BROG55_V3","BROG55_V4","BROG55_V5","BROG55_V6","BROG55_V7","BROG55_V9","E0XRWFH5_V1","E0XRWFH5_V2","E0XRWFH5_V3","E0XRWFH5_V4","F0WTDFML_V1","F0WTDFML_V2","F0WTDFML_V3","F0X18FJY_V1","F0X18FJY_V3","F0X18FJY_V4","F0X18FJY_V7","F0X18FJY_V8","FPNP1C_V2","FPNP1C_V4","FPNP1C_V6","FPNP1C_V8","FPNP1C_V10","JCDMVS_V2","JCDMVS_V3","JCDMVS_V4","KHXMY0PM_V1","KHXMY0PM_V2","KHXMY0PM_V4","KHXMY0PM_V6","KHXMY0PM_V9","KHXMY0PM_V10","L0WNF0W7_V1","L0WNF0W7_V2","L0WNF0W7_V3","L0WNF0W7_V4","L0WNF0W7_V5","L1FHD0D9_V1","L1FHD0D9_V3","L1FHD0D9_V5","L1FHD0D9_V8","L1FHD0D9_V10","LHWXA0P3_V1","LHWXA0P3_V2","LHWXA0P3_V3","M3TGDA_V3","M3TGDA_V5","M3TGDA_V7","O1WHE5_V2","O1WHE5_V3","O1WHE5_V5","OGYDRC_V3","OGYDRC_V4","OGYDRC_V5","R0WJHFA9_V1","R0WJHFA9_V3","R0WJHFA9_V5","R0WJHFA9_V7","R0WJHFA9_V10","TYVKOK_V1","TYVKOK_V2","TYVKOK_V3","TYVKOK_V4","TYVKOK_V5","V0VNY7R3_V1","V0VNY7R3_V3","V0VNY7R3_V4","V0VNY7R3_V5","V0VNY7R3_V6","V0VNY7R3_V8","VE0D4R_V2","VE0D4R_V4","VE0D4R_V6","VE0D4R_V7","WP27XA_V2","WP27XA_V4","WP27XA_V5","WP27XA_V6","WP27XA_V7","WP27XA_V9","WP27XA_V10","WP27XA_V11","WP27XA_V12","X0WEK0UW_V1","X0WEK0UW_V2","X0WEK0UW_V3","X0WEK0UW_V4","XHXE2026_V1","XHXE2026_V3","XHXE2026_V6","XHXE2026_V10","D1174FWT_V2","1HTNWJ_V1","1HTNWJ_V2","1HTNWJ_V3","1HTNWJ_V4","1KI5TS_V1","1KI5TS_V2","1KI5TS_V3","1KI5TS_V4","1KXTYW_V1","1KXTYW_V3","1KXTYW_V4","1KXTYW_V6","1Q4K8B_V1","1Q4K8B_V2","1Q4K8B_V3","1YWDW4_V1","1YWDW4_V2","1YWDW4_V3","2DPEFF_V1","2DPEFF_V3","2DPEFF_V6","2DPEFF_V8","6EY5JO_V1","6EY5JO_V2","6EY5JO_V3","6EY5JO_V4","13ILD5_V1","13ILD5_V2","13ILD5_V3","13ILD5_V4","19XOBE_V1","19XOBE_V4","19XOBE_V6","19XOBE_V8","111M7W_V1","111M7W_V3","111M7W_V6","111M7W_V9","167LKH_V1","167LKH_V2","167LKH_V4","167LKH_V6","R1WTOP_V1","R1WTOP_V3","R1WTOP_V2","R1WTOP_V4","H02GMZ_V1","H02GMZ_V2","I0HVOL_V1","I0HVOL_V2","I0HVOL_V3","I0HVOL_V5","I0HVOL_V8","KBEQL9_V1","KBEQL9_V2","KBEQL9_V3","LLGHRV_V1","LLGHRV_V2","LLGHRV_V3","O1EGCK_V1","O1EGCK_V2","O1EGCK_V3","O1EGCK_V4","QCUYXD_V1","QCUYXD_V2","QCUYXD_V6","QCUYXD_V8","V332YT_V1","V332YT_V2","V332YT_V3","V332YT_V4","XFKHGP_V1","XFKHGP_V2","XFKHGP_V6","XFKHGP_V9","ZSOOQ4_V1","ZSOOQ4_V2","ZSOOQ4_V4","ZSOOQ4_V5","ZSOOQ4_V7","1AHBZG_V1","1AHBZG_V2","1E7E5F_V1","1E7E5F_V2","1H3IW7_V1","1H3IW7_V2","1IHGAC_V1","1IHGAC_V2","1JV272_V2","1LJXIN_V1","1LJXIN_V2","1NL5BV_V1","1NL5BV_V2","1P4KLZ_V1","1P4KLZ_V2","1VVXHY_V1","1VVXHY_V2","1XGTDZ_V1","1XGTDZ_V2","1XUOU0_V1","1XUOU0_V2","7EAOX7_V1","7EAOX7_V2","9R9CV0_V1","9R9CV0_V2","11BMIX_V1","11BMIX_V2","12J5EO_V1","12J5EO_V2","CMVZEW_V1","CMVZEW_V2","MJMTYR_V1","MJMTYR_V2","NVJH6R_V1","NVJH6R_V2","Q7B5U6_V1","Q7B5U6_V2","Q89YQK_V1","Q89YQK_V2","QQHUB5_V1","QQHUB5_V2","Z4BEK1_V1","Z4BEK1_V2"],["Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Follow-up","Follow-up","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up","Baseline","Follow-up"],["2007-05-30","2008-01-14","2008-06-21","2009-02-05","2009-11-12","2010-08-26","2011-03-09","2013-06-09","2006-10-09","2007-01-15","2007-12-10","2008-11-17","2007-02-01","2007-07-09","2008-04-22","2008-10-17","2009-08-05","2007-06-19","2008-03-18","2008-09-22","2009-02-17","2007-06-23","2008-01-29","2008-06-05","2010-02-04","2010-09-06","2007-06-18","2007-10-10","2008-01-17","2007-01-25","2007-06-11","2007-10-24","2008-01-23","2008-06-04","2008-10-14","2009-01-20","2009-06-02","2009-09-08","2009-12-08","2010-03-16","2006-08-16","2006-12-07","2007-01-22","2008-02-15","2008-06-30","2008-08-11","2008-06-15","2008-10-29","2010-08-19","2007-05-29","2007-10-24","2008-01-29","2008-10-28","2009-04-28","2009-10-27","2009-11-12","2006-08-04","2007-01-23","2008-03-27","2009-06-11","2010-10-18","2008-02-04","2008-06-09","2009-03-30","2010-01-25","2011-03-17","2013-06-11","2007-03-29","2007-04-26","2008-02-26","2008-10-08","2009-01-19","2010-04-13","2010-11-12","2007-11-29","2008-03-03","2008-06-13","2008-12-05","2009-05-15","2010-05-28","2009-02-19","2009-08-07","2010-03-01","2010-09-15","2008-04-04","2009-01-27","2009-09-11","2008-06-01","2008-12-01","2009-06-10","2009-12-07","2010-06-14","2008-09-22","2009-03-16","2009-09-21","2010-03-17","2010-12-15","2009-05-26","2009-10-12","2010-04-12","2008-12-04","2009-03-19","2009-09-10","2010-03-11","2010-12-14","2013-06-04","2008-03-26","2008-06-21","2008-10-27","2009-06-25","2010-06-14","2008-11-20","2009-05-08","2009-11-02","2010-05-17","2010-11-08","2008-04-29","2008-06-02","2008-10-02","2007-06-05","2008-01-03","2008-06-21","2008-04-14","2009-02-03","2010-06-10","2007-06-11","2008-01-08","2008-05-05","2008-03-04","2008-10-29","2009-06-08","2010-01-11","2011-03-14","2007-06-18","2008-03-18","2008-11-11","2009-06-08","2010-08-31","2007-06-08","2007-10-16","2008-01-22","2008-06-29","2009-01-27","2010-11-11","2006-09-12","2007-03-02","2007-09-05","2008-01-16","2007-06-26","2008-01-17","2008-04-17","2008-08-20","2008-11-24","2009-06-08","2009-12-14","2010-06-07","2010-12-16","2008-02-28","2008-06-30","2008-11-14","2009-02-27","2008-10-01","2009-04-07","2010-01-20","2011-02-14","2006-11-13","2007-10-02","2007-11-19","2010-03-15","2010-08-23","2008-02-21","2008-05-15","2008-08-21","2008-10-13","2007-09-10","2008-03-04","2008-08-20","2009-09-09","2007-09-04","2007-12-17","2008-04-01","2013-08-22","2013-11-06","2014-02-28","2007-06-01","2007-11-08","2008-11-06","2009-05-07","2007-09-05","2008-05-20","2009-02-02","2009-05-20","2008-09-22","2009-05-19","2010-04-12","2012-03-15","2008-03-14","2008-10-27","2009-04-15","2010-02-24","2008-03-25","2008-11-03","2009-04-23","2010-02-18","2008-10-30","2009-02-05","2009-12-03","2010-11-11","2017-04-06","2018-03-02","2018-01-31","2018-04-10","2009-01-19","2010-02-16","2006-05-29","2006-09-27","2007-01-29","2007-08-08","2008-08-26","2006-06-29","2006-10-11","2007-01-02","2010-06-01","2011-05-03","2013-08-20","2008-02-21","2008-09-03","2009-02-17","2009-11-10","2007-12-21","2008-02-26","2009-03-25","2009-10-14","2008-08-29","2008-12-01","2009-03-02","2009-03-19","2008-03-31","2008-06-10","2009-04-23","2010-02-02","2006-06-26","2006-09-22","2007-04-11","2007-08-01","2007-06-04","2017-08-04","2020-03-06","2017-11-22","2020-05-25","2017-12-12","2020-01-20","2018-06-19","2020-08-18","2017-09-11","2019-01-04","2021-01-13","2018-10-23","2020-11-23","2015-08-17","2017-02-10","2017-02-13","2021-10-12","2016-06-08","2017-10-05","2018-09-05","2020-09-14","2013-06-23","2018-03-19","2018-12-18","2021-01-13","2019-02-14","2020-06-27","2021-01-22","2021-10-26","2017-01-23","2020-02-12","2017-06-06","2020-11-24","2013-11-18","2018-10-11","2014-05-05","2016-06-26","2018-01-04","2018-11-22","2014-04-24","2015-12-03","2018-04-18","2020-02-12"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>TimePoint<\/th>\n      <th>Date<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Sample","targets":1},{"name":"TimePoint","targets":2},{"name":"Date","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># Visualize exact dates of plasma samples
tmp_proPlasmaTab &lt;- dplyr::mutate(proPlasmaTab,
                                  Label = dplyr::case_when(TimePoint == &#39;Baseline&#39; ~ &#39;Baseline&#39;,
                                                           TimePoint != &#39;Baseline&#39; ~ &#39;Follow-up&#39;))
ggplot(tmp_proPlasmaTab, aes(x=Date, y=Patient, color=Label)) +
  geom_point(size = 2) +
  labs(title = &#39;Exact Dates&#39;) +
  scale_x_date(date_breaks = &#39;year&#39;, date_labels = &#39;%Y&#39;) +
  scale_color_manual(name = &#39;Time Point&#39;, values = c(&#39;red2&#39;, &#39;grey40&#39;)) +
  theme_minimal() +
  theme(axis.title = element_text(size = 12, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 10, angle = 60, vjust = 1, hjust = 1),
        plot.title = element_text(size = 14, face = &#39;bold&#39;, hjust = 0.5),
        legend.title = element_text(size = 12), legend.text = element_text(size = 10))</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-14-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#   theme(axis.title = element_text(size = 24, face = &#39;bold&#39;),
#         axis.text.x = element_text(size = 12, angle = 60, vjust = 1, hjust = 1),
#         legend.title = element_text(size = 22), legend.text = element_text(size = 20))
# ggsave(&#39;./output/group_meeting/smp_dates_proPlasma_Krij.png&#39;, device = &#39;png&#39;, dpi = 400, height = 10, width = 12)</code></pre>
<p>Show dimensions of original data (289 samples and 969 features)</p>
<pre class="r"><code># Preprocess data
proPlasma &lt;- doPreprocessing(proPlasmaTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                             val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                             smpAnno = smpAnno, do_featFilt = T, cutoff = 0.5,
                             viz_miss = T, bins = 20)#, save_path = &#39;./data/Discovery/AG_Krijgsveld/proPlasma&#39;)
# Show dimensions of original data (289 samples and 969 features)
dim(proPlasma$ori.data)</code></pre>
<pre><code>[1] 969 289</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code># Show missingness of original data
proPlasma$ori.data.miss +
  labs(x = &#39;Samples&#39;) +
  theme(axis.text.x = element_blank(), axis.title = element_text(size = 13, face = &#39;bold&#39;),
        panel.grid.major.y = element_blank())</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show sample mean-sd relationship of original data</p>
<pre class="r"><code>proPlasma$ori.smp.mean.var</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (289 samples and 607 features)</p>
<pre class="r"><code>dim(proPlasma$filt.data)</code></pre>
<pre><code>[1] 607 289</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>proPlasma$filt.data.miss +
  labs(x = &#39;Samples&#39;) +
  theme(axis.text.x = element_blank(), axis.title = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normalize filtered data by VSN</strong><br />
Show sample mean-sd relationship of vsn normalized data</p>
<pre class="r"><code>proPlasma$vsn.smp.mean.var +
  ylim(2.4, 3.6)</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show feature mean-sd relationship of vsn normalized data</p>
<pre class="r"><code>proPlasma$vsn.feat.mean.var</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Investigate impacts of sample missing levels on observed
features</strong><br />
Display relationship between sample missing levels and median
expressions computed by fully observed features</p>
<pre class="r"><code># Normalize unfiltered data
proPlasmaUnfiltVsn &lt;- doPreprocessing(proPlasmaTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                      val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                      smpAnno = smpAnno, do_featFilt = F)$vsn.data
# Prepare long data for computing sample missing levels and median expressions
proPlasmaUnfiltVsnTab &lt;- summExp2df(proPlasmaUnfiltVsn, assay = &#39;Abundance&#39;,
                                    row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;) %&gt;%
  # Prepare information of sample missing levels
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(OriMissLevel = sum(is.na(Value)) / length(Value)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Prepare information of sample median expressions computed by fully observed features
  dplyr::group_by(Protein.Group) %&gt;%
  dplyr::mutate(Keep = dplyr::case_when(!any(is.na(Value)) ~ &#39;Yes&#39;,
                                        any(is.na(Value)) ~ &#39;No&#39;)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::filter(Keep %in% &#39;Yes&#39;) %&gt;%
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(MedianVsnAbun = median(Value)) %&gt;%
  dplyr::ungroup()
# Create SE object containing information of sample missing levels and median expressions
colAnno &lt;- dplyr::select(proPlasmaUnfiltVsnTab, Sample, OriMissLevel, MedianVsnAbun) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
proPlasmaUnfiltVsn &lt;- SummarizedExperiment(assays = list(Abundance = assay(proPlasmaUnfiltVsn)),
                                           colData = colAnno)

# Visualize relationship between sample missing levels and median expressions
ggplot(proPlasmaUnfiltVsnTab, aes(x=OriMissLevel, y=MedianVsnAbun)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(x = &#39;Missing level&#39;, y = &#39;Median expression&#39;) +
  th</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show significant associations between data and sample missing
levels</p>
<pre class="r"><code># Do PCA
pcaRes &lt;- doSOA(proPlasmaUnfiltVsn, meta_var = &#39;OriMissLevel&#39;, pca_method = &#39;pca&#39;,
                num_PCs = 20, do_onlyPCA = T)
# Show significant associations between PCs and sample missing levels
pcaRes$pcSigAssoRes %&gt;%
  dplyr::filter(grepl(&#39;PC[1-9] \\(&#39;, Var1))</code></pre>
<pre><code>         Var1         Var2     pVal  pValAdj   Stat        Test
1 PC1 (23.8%) OriMissLevel 2.41e-59 5.17e-57  0.776 Correlation
2  PC3 (6.9%) OriMissLevel 3.48e-06 3.72e-04  0.269 Correlation
3  PC6 (4.5%) OriMissLevel 3.10e-05 2.21e-03 -0.242 Correlation
4  PC5 (4.8%) OriMissLevel 3.02e-03 1.61e-01  0.174 Correlation</code></pre>
<pre class="r"><code># Visualize relationship between PCs and sample missing levels
pcTab &lt;- pcaRes$pcTab
ggplot(pcTab, aes(x=`PC1 (23.8%)`, y=OriMissLevel)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(y = &#39;Missing level&#39;) +
  th</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="baseline-samples" class="section level2">
<h2>Baseline samples</h2>
<p><em>Preprocess only Baseline Plasma samples, instead of all Plasma
samples, to obtain larger feature space</em></p>
<p>Show dimensions of original data (63 samples and 969 features)</p>
<pre class="r"><code># Subset certain samples
proBaseTab &lt;- dplyr::filter(proPlasmaTab, TimePoint == &#39;Baseline&#39;)
# Preprocess data
proBase &lt;- doPreprocessing(proBaseTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                           val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                           smpAnno = smpAnno, do_featFilt = T, cutoff = 0.5,
                           viz_miss = T, bins = 20)#, save_path = &#39;./data/Discovery/AG_Krijgsveld/proBase&#39;)
dim(proBase$ori.data)</code></pre>
<pre><code>[1] 969  63</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code>proBase$ori.data.miss +
  theme(panel.grid.minor.y = element_blank())</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show distribution of original data</p>
<pre class="r"><code>proBase$ori.data.dist</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (63 samples and 613 features)</p>
<pre class="r"><code>dim(proBase$filt.data)</code></pre>
<pre><code>[1] 613  63</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>proBase$filt.data.miss</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code>proBase$vsn.data.dist</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>proBase$vsn.feat.mean.var</code></pre>
<p><img src="figure/Discovery_01_preprocessing_DIA_Krij.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Warsaw
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] sva_3.50.0                  BiocParallel_1.36.0        
 [3] genefilter_1.84.0           mgcv_1.9-1                 
 [5] nlme_3.1-164                proDA_1.16.0               
 [7] pcaMethods_1.94.0           pheatmap_1.0.12            
 [9] ggplotify_0.1.2             lubridate_1.9.3            
[11] forcats_1.0.0               stringr_1.5.1              
[13] dplyr_1.1.4                 purrr_1.0.2                
[15] tidyr_1.3.1                 tibble_3.2.1               
[17] tidyverse_2.0.0             SummarizedExperiment_1.32.0
[19] GenomicRanges_1.54.1        GenomeInfoDb_1.38.8        
[21] IRanges_2.36.0              S4Vectors_0.40.2           
[23] MatrixGenerics_1.14.0       matrixStats_1.3.0          
[25] ggrepel_0.9.5               ggplot2_3.5.1              
[27] visdat_0.6.0                limma_3.58.1               
[29] vsn_3.70.0                  Biobase_2.62.0             
[31] BiocGenerics_0.48.1         readxl_1.4.3               
[33] readr_2.1.5                 workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.16.0       jsonlite_1.8.8         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.27         
  [7] fs_1.6.4                zlibbioc_1.48.2         vctrs_0.6.5            
 [10] memoise_2.0.1           RCurl_1.98-1.14         rstatix_0.7.2          
 [13] htmltools_0.5.8.1       S4Arrays_1.2.1          broom_1.0.5            
 [16] cellranger_1.1.0        SparseArray_1.2.4       gridGraphics_0.5-1     
 [19] sass_0.4.9              bslib_0.7.0             htmlwidgets_1.6.4      
 [22] cachem_1.1.0            whisker_0.4.1           lifecycle_1.0.4        
 [25] pkgconfig_2.0.3         Matrix_1.6-5            R6_2.5.1               
 [28] fastmap_1.2.0           GenomeInfoDbData_1.2.11 digest_0.6.35          
 [31] colorspace_2.1-0        AnnotationDbi_1.64.1    ps_1.7.6               
 [34] rprojroot_2.0.4         crosstalk_1.2.1         RSQLite_2.3.6          
 [37] ggpubr_0.6.0            labeling_0.4.3          fansi_1.0.6            
 [40] timechange_0.3.0        httr_1.4.7              abind_1.4-5            
 [43] compiler_4.3.2          bit64_4.0.5             withr_3.0.0            
 [46] backports_1.4.1         carData_3.0-5           DBI_1.2.2              
 [49] hexbin_1.28.3           highr_0.11              ggsignif_0.6.4         
 [52] DelayedArray_0.28.0     tools_4.3.2             httpuv_1.6.15          
 [55] glue_1.7.0              callr_3.7.6             promises_1.3.0         
 [58] grid_4.3.2              getPass_0.2-4           generics_0.1.3         
 [61] gtable_0.3.5            tzdb_0.4.0              preprocessCore_1.64.0  
 [64] hms_1.1.3               car_3.1-2               utf8_1.2.4             
 [67] XVector_0.42.0          pillar_1.9.0            vroom_1.6.5            
 [70] yulab.utils_0.1.4       later_1.3.2             splines_4.3.2          
 [73] lattice_0.22-6          survival_3.5-8          bit_4.0.5              
 [76] annotate_1.80.0         tidyselect_1.2.1        locfit_1.5-9.9         
 [79] Biostrings_2.70.3       knitr_1.47              git2r_0.33.0           
 [82] edgeR_4.0.16            xfun_0.44               statmod_1.5.0          
 [85] DT_0.33                 stringi_1.8.4           yaml_2.3.8             
 [88] evaluate_0.23           codetools_0.2-20        BiocManager_1.30.22    
 [91] cli_3.6.2               affyio_1.72.0           xtable_1.8-4           
 [94] munsell_0.5.1           processx_3.8.4          jquerylib_0.1.4        
 [97] Rcpp_1.0.12             png_0.1-8               XML_3.99-0.16.1        
[100] parallel_4.3.2          blob_1.2.4              bitops_1.0-7           
[103] scales_1.3.0            affy_1.80.0             crayon_1.5.2           
[106] rlang_1.1.4             KEGGREST_1.42.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
