<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Single-omics analysis: Combined DIA Proteomics (Discovery + MethodDev)</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Single-omics analysis: Combined DIA
Proteomics (Discovery + MethodDev)</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree9a5191e865f9c1d40a49ea46a7c1c020e986c53etargetblank9a5191ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/9a5191e865f9c1d40a49ea46a7c1c020e986c53e" target="_blank">9a5191e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree9a5191e865f9c1d40a49ea46a7c1c020e986c53etargetblank9a5191ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/9a5191e865f9c1d40a49ea46a7c1c020e986c53e" target="_blank">9a5191e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/comparison_funcs.R
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R
    Untracked:  data/Discovery/
    Untracked:  data/MethodDev/
    Untracked:  data/aliquot_metadata.xlsx
    Untracked:  data/patient_metadata.xlsx
    Untracked:  data/sample_metadata.xlsx
    Untracked:  output/Discovery/
    Untracked:  output/MethodDev/
    Untracked:  output/SC_meeting/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Discovery_02_soa_DIA_combined.Rmd</code>) and HTML
(<code>docs/Discovery_02_soa_DIA_combined.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/b83f6736534c624c72cb444130bf22766780a846/analysis/Discovery_02_soa_DIA_combined.Rmd" target="_blank">b83f673</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-05-15
</td>
<td>
Just for quick update of analysis report webpage
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/67817ea971be09a45d10861552effe81cde6879b/analysis/Discovery_02_soa_DIA_combined.Rmd" target="_blank">67817ea</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-03-06
</td>
<td>
Train XGBoost and lasso logistic regression models on combined Klin.
normal tissue DIA proteomics and tidy up code for analysis of combined
proteomics data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/b0e008fde5b36ce4a776a51e6c876449470b26cb/analysis/Discovery_02_soa_DIA_combined.Rmd" target="_blank">b0e008f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-01-11
</td>
<td>
Do single-omics analysis of combined DIA tissue proteomics data
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Conduct univariate (t-test) and
multivariate (PCA) analyses on combined Tissue DIA Proteomics from AG
Krijgsveld and AG Klingmüller (2 datasets) to have overview of data and
take initial look at data power in terms of predicting patient cancer
recurrence. </font></p>
<p><strong>Metadata variables</strong><br />
Patients who got Tissue samples (n = 63):<br />
Recurrence -&gt; Cancer recurrences, Yes:No = 29:34<br />
Gender -&gt; Female:Male = 25:38<br />
Age -&gt; Diagnosis ages ranging from 51 to 84<br />
Smoking -&gt; Smoker:Ex-smoker:Non-Smoker = 14:40:9<br />
Stage -&gt; Pathological stages, IB:IIA:IIB = 36:14:13<br />
Adjuvant -&gt; Adjuvant chemotherapy, True:False = 21:42<br />
Samples:<br />
Condition -&gt; Tumor:Normal = 1:1</p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;pcaMethods&#39;)
library(&#39;sva&#39;)
library(&#39;limma&#39;)
library(&#39;proDA&#39;)
library(&#39;pheatmap&#39;)
library(&#39;ggplotify&#39;)
library(&#39;ggrepel&#39;)
library(&#39;ggvenn&#39;)
library(&#39;AnnotationDbi&#39;)
library(&#39;org.Hs.eg.db&#39;)
# This annotation object is accessed using AnnotationDbi:select()
hs &lt;- org.Hs.eg.db
library(&#39;msigdbr&#39;)
library(&#39;clusterProfiler&#39;)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/analysis_pipeline.R&#39;)
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
    theme(axis.title = element_text(face = &#39;bold&#39;),
          axis.text = element_text(face = &#39;bold&#39;),
          axis.ticks = element_line(linewidth = 0.8),
          legend.text = element_text(size = 15))</code></pre>
<p>Display associations between cancer recurrence and the other patient
metadata variables to identify potential predictors (or confounders)</p>
<pre class="r"><code># Test associations between cancer recurrence and the other patient metadata variables
# Prepare patient metadata
proTissue &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
patientMetadat &lt;- tibble::as_tibble(colData(proTissue)) %&gt;%
  dplyr::filter(!duplicated(Patient)) %&gt;%
  dplyr::select(Patient, Recurrence, Gender, Age, Smoking, Stage, Adjuvant)
patientRecur &lt;- dplyr::select(patientMetadat, Patient, Recurrence)
patientMetadat &lt;- dplyr::select(patientMetadat, -Recurrence)
testAsso(patientMetadat, patientRecur, cmn_col = &#39;Patient&#39;) %&gt;%
  dplyr::select(-Stat)</code></pre>
<pre><code>      Var1       Var2       pVal   pValAdj       Test
1  Smoking Recurrence 0.07501072 0.1990297 Chi-square
2    Stage Recurrence 0.07961190 0.1990297 Chi-square
3   Gender Recurrence 0.29953357 0.4069731 Chi-square
4 Adjuvant Recurrence 0.32557848 0.4069731 Chi-square
5      Age Recurrence 0.85820530 0.8582053     T-test</code></pre>
<div id="tissue-proteomics-ag-krijgsveld" class="section level1">
<h1>Tissue Proteomics (AG Krijgsveld)</h1>
<p>Discovery + MethodDev</p>
<pre class="r"><code># Load preprocessed data
proTissue_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::select(Protein, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)

# Perform single-omics analysis
proTissueRes_Krij &lt;- doSOA(proTissue_Krij, meta_var = c(&#39;Condition&#39;, &#39;Cohort&#39;), pca_method = &#39;ppca&#39;, do_onlyPCA = T)</code></pre>
<p><strong>Do metadata-assisted quality control</strong><br />
After testing associations between PCs and tissue sample conditions, we
found that PC1 significantly separates Tumor and Normal samples, which
indicates decent data quality. Yet, there are 4 Tumor samples
misclassified, reported as follows:<br />
=&gt; Tumor purity: I0HVOL_TU - 87% / 7EAOX7_TU - 70% / MJMTYR_TU - 40%
/ XFKHGP_TU - 100%<br />
(Mean of sample tumor cell contents is 74%, ranging from 40% to
100%.)</p>
<pre class="r"><code>pcTab &lt;- proTissueRes_Krij$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;,
                                                       &#39;MJMTYR_TU&#39;, &#39;XFKHGP_TU&#39;,
                                                       &#39;XFKHGP_NG&#39;, &#39;MJMTYR_NG&#39;,
                                                       &#39;I0HVOL_NG&#39;, &#39;7EAOX7_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (38%)`, color=Condition, fill=Condition, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="tumor-samples" class="section level2">
<h2>Tumor samples</h2>
<p>Note that <strong>tumor purity</strong> and <strong>pathological
state</strong> are also included in linear model for more reliable
t-test results because they are significantly associated with top
PCs.</p>
<pre class="r"><code># Subset certain samples
proTumor_Krij &lt;- proTissue_Krij[, which(colData(proTissue_Krij)$Condition == &#39;Tumor&#39;)]
# Perform single-omics analysis
proTumorRes_Krij &lt;- doSingleOmicsAnalysis(proTumor_Krij, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
                                          pca_method = &#39;ppca&#39;, num_PCs = 40, num_PCfeats = 30,
                                          use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
                                          feat_conv = T, plot_title = NULL, show_rownames = F,
                                          pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                          &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;,
                                                          &#39;TumorPurity&#39;, &#39;Cohort&#39;),
                                          soa_unwantVar = c(&#39;TumorPurity&#39;, &#39;Stage&#39;))</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proTumorRes_Krij$data.var.source</code></pre>
<pre><code>          Var1        Var2     pVal pValAdj   Stat        Test
1   PC4 (5.3%) TumorPurity 0.000029  0.0093  0.501 Correlation
2   PC2 (7.7%) TumorPurity 0.003280  0.3900  0.365 Correlation
3   PC5 (4.9%)         Age 0.006250  0.3900 -0.341 Correlation
4  PC35 (0.9%)      Cohort 0.007340  0.3900  2.770      T-test
5  PC10 (3.4%)       Stage 0.008400  0.3900  5.180       ANOVA
6  PC21 (1.8%)    Adjuvant 0.008750  0.3900  2.710      T-test
7   PC9 (3.6%)       Stage 0.009720  0.3900  5.010       ANOVA
8   PC1 (9.1%)       Stage 0.009750  0.3900  5.010       ANOVA
9  PC14 (2.6%)      Cohort 0.011800  0.4050  2.600      T-test
10 PC11 (3.1%)    Adjuvant 0.013800  0.4050 -2.540      T-test
11 PC26 (1.4%)    Adjuvant 0.013900  0.4050  2.530      T-test
12 PC23 (1.5%)      Gender 0.015400  0.4110  2.490      T-test
13   PC33 (1%)  Recurrence 0.016800  0.4130  2.460      T-test
14 PC11 (3.1%)  Recurrence 0.018200  0.4130 -2.430      T-test
15 PC29 (1.2%)      Cohort 0.020400  0.4130  2.380      T-test
16 PC12 (2.8%)      Gender 0.020700  0.4130 -2.380      T-test
17  PC1 (9.1%) TumorPurity 0.022600  0.4210 -0.287 Correlation
18 PC21 (1.8%)         Age 0.024300  0.4210 -0.284 Correlation
19  PC7 (4.2%)    Adjuvant 0.025000  0.4210 -2.300      T-test
20 PC27 (1.3%)         Age 0.032400  0.5190 -0.270 Correlation
21 PC28 (1.3%)         Age 0.036300  0.5300  0.264 Correlation
22  PC8 (3.8%)       Stage 0.036500  0.5300  3.500       ANOVA
23 PC17 (2.1%)     Smoking 0.040800  0.5680  3.380       ANOVA
24 PC23 (1.5%)       Stage 0.046400  0.6070  3.230       ANOVA
25 PC31 (1.1%)         Age 0.049200  0.6070  0.249 Correlation</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2). Owing to missing values, there may be situation that
only few observations in sample groups, resulting in unreliable
t-statistics. To this end, we used probabilistic dropout model (proDA)
to account for data missingness.</p>
<pre class="r"><code>proTumorRes_Krij$sig.feat.tab</code></pre>
<pre><code>          Var1       Var2     pVal pValAdj  Stat           Test     Gene
1       Q9BVT8 Recurrence 0.000594       1  3.63 T-test (proDA)    TMUB1
2       Q93034 Recurrence 0.001460       1  3.34 T-test (proDA)     CUL5
3       P10515 Recurrence 0.001940       1  3.25 T-test (proDA)     DLAT
4       Q15024 Recurrence 0.002580       1 -3.15 T-test (proDA)   EXOSC7
5       Q53LP3 Recurrence 0.002770       1  3.13 T-test (proDA)   SOWAHC
6       P09622 Recurrence 0.004340       1  2.97 T-test (proDA)      DLD
7       Q9Y572 Recurrence 0.004970       1  2.92 T-test (proDA)    RIPK3
8       Q6P1N0 Recurrence 0.005860       1 -2.86 T-test (proDA)   CC2D1A
9       Q8NBM8 Recurrence 0.006380       1 -2.83 T-test (proDA)  PCYOX1L
10      Q9Y2J2 Recurrence 0.006740       1 -2.81 T-test (proDA)  EPB41L3
11      P49768 Recurrence 0.007350       1  2.78 T-test (proDA)    PSEN1
12      P19174 Recurrence 0.007690       1 -2.76 T-test (proDA)    PLCG1
13      Q9UKK3 Recurrence 0.007780       1  2.76 T-test (proDA)    PARP4
14      Q9Y6C9 Recurrence 0.007980       1  2.75 T-test (proDA)    MTCH2
15      P43304 Recurrence 0.008350       1  2.73 T-test (proDA)     GPD2
16      Q9BXB5 Recurrence 0.008390       1 -2.73 T-test (proDA)  OSBPL10
17      Q6IC98 Recurrence 0.009260       1  2.69 T-test (proDA)   GRAMD4
18      Q9UII2 Recurrence 0.010600       1  2.64 T-test (proDA)  ATP5IF1
19      Q99614 Recurrence 0.010800       1 -2.63 T-test (proDA)     TTC1
20      Q6UWE0 Recurrence 0.010900       1  2.63 T-test (proDA)   LRSAM1
21      Q9ULR0 Recurrence 0.011100       1 -2.62 T-test (proDA)     ISY1
22      O43823 Recurrence 0.011500       1 -2.61 T-test (proDA)    AKAP8
23      Q9UN86 Recurrence 0.011500       1 -2.61 T-test (proDA)    G3BP2
24      P40222 Recurrence 0.011600       1 -2.61 T-test (proDA)    TXLNA
25      O14841 Recurrence 0.011700       1  2.60 T-test (proDA)    OPLAH
26      Q14534 Recurrence 0.012300       1  2.58 T-test (proDA)     SQLE
27      P33991 Recurrence 0.012600       1 -2.57 T-test (proDA)     MCM4
28      Q9HD40 Recurrence 0.012700       1  2.57 T-test (proDA)  SEPSECS
29      P56181 Recurrence 0.013100       1  2.56 T-test (proDA)   NDUFV3
30      Q01484 Recurrence 0.013500       1 -2.55 T-test (proDA)     ANK2
31      Q9NXR1 Recurrence 0.013500       1  2.55 T-test (proDA)     NDE1
32      Q9UNF0 Recurrence 0.013900       1  2.54 T-test (proDA)  PACSIN2
33      O14495 Recurrence 0.014200       1 -2.53 T-test (proDA)    PLPP3
34      Q8N755 Recurrence 0.014400       1  2.52 T-test (proDA)  SLC66A3
35      O00461 Recurrence 0.015100       1 -2.51 T-test (proDA)   GOLIM4
36      Q9Y6D6 Recurrence 0.015200       1  2.50 T-test (proDA)  ARFGEF1
37      P19367 Recurrence 0.015500       1  2.49 T-test (proDA)      HK1
38      Q14318 Recurrence 0.015600       1 -2.49 T-test (proDA)    FKBP8
39      P61513 Recurrence 0.015800       1 -2.49 T-test (proDA)   RPL37A
40      Q9Y5Q8 Recurrence 0.015900       1 -2.48 T-test (proDA)   GTF3C5
41      A6NDU8 Recurrence 0.016200       1 -2.48 T-test (proDA)  C5orf51
42      Q8TD16 Recurrence 0.016200       1 -2.48 T-test (proDA)    BICD2
43      Q9BTZ2 Recurrence 0.016400       1  2.47 T-test (proDA)    DHRS4
44      P30536 Recurrence 0.016500       1  2.47 T-test (proDA)     TSPO
45      Q8TE77 Recurrence 0.016600       1  2.47 T-test (proDA)     SSH3
46      P24752 Recurrence 0.017000       1  2.46 T-test (proDA)    ACAT1
47      Q8IY81 Recurrence 0.017700       1 -2.44 T-test (proDA)    FTSJ3
48      Q96K19 Recurrence 0.017700       1  2.44 T-test (proDA)   RNF170
49      Q96QD9 Recurrence 0.018300       1 -2.43 T-test (proDA)   FYTTD1
50      P15169 Recurrence 0.018400       1 -2.43 T-test (proDA)     CPN1
51      Q6PJ69 Recurrence 0.018900       1  2.41 T-test (proDA)   TRIM65
52      O75306 Recurrence 0.019000       1  2.41 T-test (proDA)   NDUFS2
53      P49447 Recurrence 0.019000       1  2.41 T-test (proDA)   CYB561
54      O75251 Recurrence 0.019100       1  2.41 T-test (proDA)   NDUFS7
55      Q9C0B1 Recurrence 0.019400       1 -2.40 T-test (proDA)      FTO
56      P56962 Recurrence 0.019500       1 -2.40 T-test (proDA)    STX17
57      Q15334 Recurrence 0.019500       1 -2.40 T-test (proDA)    LLGL1
58      Q96B26 Recurrence 0.019500       1 -2.40 T-test (proDA)   EXOSC8
59      Q9BXB4 Recurrence 0.019500       1 -2.40 T-test (proDA)  OSBPL11
60      P46527 Recurrence 0.019700       1 -2.40 T-test (proDA)   CDKN1B
61      Q6ICG6 Recurrence 0.019900       1  2.39 T-test (proDA) KIAA0930
62      Q9H3Q1 Recurrence 0.020300       1 -2.39 T-test (proDA) CDC42EP4
63      Q9HCC0 Recurrence 0.020300       1  2.39 T-test (proDA)    MCCC2
64      P35269 Recurrence 0.020600       1 -2.38 T-test (proDA)   GTF2F1
65      P42696 Recurrence 0.020800       1 -2.38 T-test (proDA)    RBM34
66      Q96GS4 Recurrence 0.020800       1  2.38 T-test (proDA)   BORCS6
67      Q969X1 Recurrence 0.021000       1  2.37 T-test (proDA)   TMBIM1
68      Q8WUQ7 Recurrence 0.021300       1 -2.37 T-test (proDA)   CACTIN
69      Q8WV41 Recurrence 0.021600       1  2.36 T-test (proDA)    SNX33
70      Q96RQ3 Recurrence 0.022100       1  2.35 T-test (proDA)    MCCC1
71      P62314 Recurrence 0.023200       1 -2.33 T-test (proDA)   SNRPD1
72      Q9NRN7 Recurrence 0.023200       1  2.33 T-test (proDA) AASDHPPT
73      Q92878 Recurrence 0.023700       1  2.32 T-test (proDA)    RAD50
74      Q96Q05 Recurrence 0.023700       1 -2.32 T-test (proDA)  TRAPPC9
75      P03950 Recurrence 0.024200       1 -2.31 T-test (proDA)      ANG
76      Q13123 Recurrence 0.024300       1 -2.31 T-test (proDA)       IK
77      O14920 Recurrence 0.025600       1  2.29 T-test (proDA)    IKBKB
78      Q9H1C4 Recurrence 0.025600       1  2.29 T-test (proDA)  UNC93B1
79      Q16585 Recurrence 0.025800       1 -2.29 T-test (proDA)     SGCB
80      Q5TBA9 Recurrence 0.026000       1 -2.29 T-test (proDA)      FRY
81      O00330 Recurrence 0.026300       1  2.28 T-test (proDA)     PDHX
82      Q9P246 Recurrence 0.026500       1  2.28 T-test (proDA)    STIM2
83      Q07960 Recurrence 0.027100       1  2.27 T-test (proDA)  ARHGAP1
84      Q9H444 Recurrence 0.027800       1  2.26 T-test (proDA)   CHMP4B
85      Q8N3P4 Recurrence 0.028300       1  2.25 T-test (proDA)     VPS8
86      O94966 Recurrence 0.028500       1  2.25 T-test (proDA)    USP19
87      Q8TB36 Recurrence 0.028500       1  2.25 T-test (proDA)    GDAP1
88      Q6P1X6 Recurrence 0.028600       1  2.25 T-test (proDA)  C8orf82
89      Q14160 Recurrence 0.028800       1  2.24 T-test (proDA)    SCRIB
90      Q8N5C1 Recurrence 0.029600       1 -2.23 T-test (proDA)   CALHM5
91      P49959 Recurrence 0.030100       1  2.22 T-test (proDA)    MRE11
92      Q00613 Recurrence 0.031900       1  2.20 T-test (proDA)     HSF1
93      P30626 Recurrence 0.032000       1  2.20 T-test (proDA)      SRI
94      Q9H977 Recurrence 0.032400       1 -2.19 T-test (proDA)    WDR54
95      Q9H0F7 Recurrence 0.032500       1 -2.19 T-test (proDA)     ARL6
96      Q6P1M3 Recurrence 0.033600       1  2.18 T-test (proDA)    LLGL2
97      Q96RE7 Recurrence 0.033800       1 -2.17 T-test (proDA)    NACC1
98      O95573 Recurrence 0.034000       1  2.17 T-test (proDA)    ACSL3
99      Q86VU5 Recurrence 0.034400       1  2.17 T-test (proDA)   COMTD1
100     O75475 Recurrence 0.034600       1 -2.16 T-test (proDA)    PSIP1
101     Q9Y6N5 Recurrence 0.034600       1  2.16 T-test (proDA)     SQOR
102     Q8WTT2 Recurrence 0.035100       1 -2.16 T-test (proDA)    NOC3L
103     Q9UBS4 Recurrence 0.035600       1 -2.15 T-test (proDA)  DNAJB11
104     O14681 Recurrence 0.035800       1  2.15 T-test (proDA)     EI24
105     Q5RKV6 Recurrence 0.036800       1 -2.14 T-test (proDA)   EXOSC6
106     Q9NWS0 Recurrence 0.037000       1  2.13 T-test (proDA)   PIH1D1
107     P35237 Recurrence 0.037100       1  2.13 T-test (proDA) SERPINB6
108     P36404 Recurrence 0.037300       1 -2.13 T-test (proDA)     ARL2
109     Q96HL8 Recurrence 0.037700       1 -2.13 T-test (proDA)   SH3YL1
110     Q9P2D3 Recurrence 0.038000       1  2.12 T-test (proDA)  HEATR5B
111     P45973 Recurrence 0.038400       1 -2.12 T-test (proDA)     CBX5
112     Q8NFU3 Recurrence 0.038600       1  2.12 T-test (proDA)    TSTD1
113     P42785 Recurrence 0.038900       1 -2.11 T-test (proDA)     PRCP
114     Q9BY42 Recurrence 0.038900       1 -2.11 T-test (proDA)     RTF2
115     Q99538 Recurrence 0.039100       1  2.11 T-test (proDA)     LGMN
116     Q9NQT8 Recurrence 0.039400       1  2.11 T-test (proDA)   KIF13B
117     Q8TB24 Recurrence 0.039700       1  2.10 T-test (proDA)     RIN3
118     Q96PC5 Recurrence 0.039700       1  2.10 T-test (proDA)     MIA2
119     O75592 Recurrence 0.039800       1 -2.10 T-test (proDA)   MYCBP2
120     P40939 Recurrence 0.039900       1  2.10 T-test (proDA)    HADHA
121     P49184 Recurrence 0.039900       1  2.10 T-test (proDA) DNASE1L1
122     P08133 Recurrence 0.040000       1 -2.10 T-test (proDA)    ANXA6
123     P63218 Recurrence 0.040000       1  2.10 T-test (proDA)     GNG5
124     P78316 Recurrence 0.040400       1 -2.10 T-test (proDA)    NOP14
125     Q5TEC6 Recurrence 0.040500       1 -2.10 T-test (proDA)     H3-2
126     Q86SK9 Recurrence 0.040700       1 -2.09 T-test (proDA)     SCD5
127     Q8NDH3 Recurrence 0.041100       1  2.09 T-test (proDA)   NPEPL1
128     Q3V6T2 Recurrence 0.041300       1 -2.09 T-test (proDA)  CCDC88A
129     P27694 Recurrence 0.041800       1 -2.08 T-test (proDA)     RPA1
130     Q9UK41 Recurrence 0.041800       1  2.08 T-test (proDA)    VPS28
131     P26358 Recurrence 0.042400       1 -2.08 T-test (proDA)    DNMT1
132     Q7Z4V5 Recurrence 0.043000       1 -2.07 T-test (proDA)   HDGFL2
133     Q53H12 Recurrence 0.043300       1  2.07 T-test (proDA)      AGK
134     Q9BXK5 Recurrence 0.043500       1  2.06 T-test (proDA)  BCL2L13
135     O14617 Recurrence 0.043600       1  2.06 T-test (proDA)    AP3D1
136     P21964 Recurrence 0.043800       1  2.06 T-test (proDA)     COMT
137     P61962 Recurrence 0.043800       1 -2.06 T-test (proDA)    DCAF7
138 A0A024RBG1 Recurrence 0.043900       1 -2.06 T-test (proDA)   NUDT4B
139     Q9H6S3 Recurrence 0.044000       1  2.06 T-test (proDA)   EPS8L2
140     P30837 Recurrence 0.045200       1 -2.05 T-test (proDA)  ALDH1B1
141     Q9NQP4 Recurrence 0.045200       1 -2.05 T-test (proDA)    PFDN4
142     Q8NG11 Recurrence 0.045300       1  2.05 T-test (proDA)  TSPAN14
143     Q96EV8 Recurrence 0.045600       1  2.04 T-test (proDA)   DTNBP1
144     Q5VU97 Recurrence 0.045800       1 -2.04 T-test (proDA)   CACHD1
145     Q9NW97 Recurrence 0.046000       1  2.04 T-test (proDA)   TMEM51
146     Q96ST2 Recurrence 0.046600       1 -2.03 T-test (proDA)     IWS1
147     Q53H96 Recurrence 0.047600       1  2.02 T-test (proDA)    PYCR3
148     Q86W50 Recurrence 0.047600       1  2.02 T-test (proDA)  METTL16
149     Q9NTJ3 Recurrence 0.047600       1 -2.02 T-test (proDA)     SMC4
150     P04792 Recurrence 0.049400       1  2.01 T-test (proDA)    HSPB1
151     O95202 Recurrence 0.049600       1  2.01 T-test (proDA)    LETM1
152     Q9P2K3 Recurrence 0.049600       1 -2.01 T-test (proDA)    RCOR3</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proTumorRes_Krij$sig.feat.heat</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Tumor samples (ns: p
&gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Krij$data
smpAnnoTab &lt;- proTissueRes_Krij$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proTumorRes_Krij$sig.feat.tab
num_sigFeats &lt;- 6
gpLevel = c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;, &#39;Yes_Normal&#39;, &#39;No_Normal&#39;)
condiCol = c(&#39;grey50&#39;, &#39;firebrick&#39;) #c(Normal, Tumor)
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(newVar1 = paste0(Var1, &#39; (&#39;, Gene, &#39;)&#39;),
                newVar1 = factor(newVar1, levels = unique(newVar1)))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Recur_Condi = paste0(Recurrence, &#39;_&#39;, Condition),
                Recur_Condi = factor(Recur_Condi, levels = gpLevel))
ggplot(topSigFeatDat, aes(x=Recur_Condi, y=Abundance, col=Condition, fill=Recurrence)) +
  geom_boxplot(alpha = 1, outlier.shape = NA, linewidth = 1) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             comparisons = list(c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;), c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;)),
                             label = &#39;p.signif&#39;, tip.length = 0.015, bracket.size = 0.7, size = 4) +
  labs(x = &#39;Recurrence&#39;) +
  scale_color_manual(values = condiCol) +
  scale_fill_manual(values = c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(newVar1), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;),
             axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proTumorRes_Krij$pval.hist</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proTumorRes_Krij$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2   pVal pValAdj  Stat   Test
1   PC33 (1%) Recurrence 0.0168   0.363  2.46 T-test
2 PC11 (3.1%) Recurrence 0.0182   0.363 -2.43 T-test</code></pre>
<p>Visualize significant PCs and display their top features with
greatest absolute loadings</p>
<pre class="r"><code>proTumorRes_Krij$sig.pc.dist$PC33</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Krij$pc.top.feat.tab$PC33</code></pre>
<pre><code>   Feature Loading    Gene
1   Q12891 -0.1810   HYAL2
2   P30838  0.1140 ALDH3A1
3   Q99453  0.1140  PHOX2B
4   P02008 -0.1110     HBZ
5   P46439 -0.1110   GSTM5
6   P11684 -0.0932 SCGB1A1
7   Q9BYB0 -0.0907  SHANK3
8   P56856 -0.0895  CLDN18
9   Q13421  0.0879    MSLN
10  P05976  0.0849    MYL1
11  O14756 -0.0839 HSD17B6
12  P04271 -0.0804   S100B
13  Q8TDL5 -0.0791  BPIFB1
14  Q9Y342 -0.0783    PLLP
15  P29972 -0.0741    AQP1
16  Q5TBA9 -0.0731     FRY
17  O14578 -0.0726     CIT
18  Q92859 -0.0707    NEO1
19  O00767  0.0683     SCD
20  P13942 -0.0683 COL11A2
21  Q14642 -0.0673  INPP5A
22  P55064  0.0668    AQP5
23  Q92954  0.0660    PRG4
24  P04278  0.0647    SHBG
25  Q86TI0 -0.0646  TBC1D1
26  Q13938 -0.0633    CAPS
27  Q8TC71 -0.0625 SPATA18
28  Q9BW30 -0.0624   TPPP3
29  P07451 -0.0617     CA3
30  O94911  0.0609   ABCA8</code></pre>
<pre class="r"><code>proTumorRes_Krij$sig.pc.dist$PC11</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-12-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Krij$pc.top.feat.tab$PC11</code></pre>
<pre><code>      Feature Loading     Gene
1      P30838 -0.1680  ALDH3A1
2      Q9UK22 -0.1280    FBXO2
3      P11684 -0.1130  SCGB1A1
4  A0A0C4DH31  0.0970 IGHV1-18
5      P13929  0.0951     ENO3
6      P11686 -0.0900    SFTPC
7      P21695  0.0891     GPD1
8      P31327 -0.0842     CPS1
9      Q9NVM6  0.0802  DNAJC17
10     P56470  0.0793   LGALS4
11     O00626  0.0771    CCL22
12     Q8TC26 -0.0766  TMEM163
13     Q86VY4 -0.0764   TSPYL5
14     O94811 -0.0761     TPPP
15     Q5H8C1  0.0746    FREM1
16     Q99453 -0.0737   PHOX2B
17     O43927  0.0726   CXCL13
18     Q13075 -0.0718     NAIP
19     Q9Y6R1 -0.0715   SLC4A4
20     P58166  0.0712    INHBE
21     P13726 -0.0698       F3
22     Q96T58  0.0695     SPEN
23     O15056  0.0671    SYNJ2
24     P01743  0.0669 IGHV1-46
25     Q9NVV5 -0.0667     AIG1
26     P78524 -0.0654  DENND2B
27     Q8N135  0.0653     LGI4
28     Q9UPW5 -0.0643  AGTPBP1
29     O43310 -0.0635     CTIF
30     Q9UHP3  0.0630    USP25</code></pre>
</div>
<div id="normal-samples" class="section level2">
<h2>Normal samples</h2>
<pre class="r"><code># Subset certain samples
proNormal_Krij &lt;- proTissue_Krij[, which(colData(proTissue_Krij)$Condition == &#39;Normal&#39;)]
# Perform single-omics analysis
proNormalRes_Krij &lt;- doSingleOmicsAnalysis(proNormal_Krij, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
                                           pca_method = &#39;ppca&#39;, num_PCs = 40, num_PCfeats = 30,
                                           use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
                                           feat_conv = T, plot_title = NULL, show_rownames = F,
                                           pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                           &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;))
# proNormalRes_Krij &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proNormalRes_Krij.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proNormalRes_Krij$data.var.source</code></pre>
<pre><code>          Var1     Var2    pVal pValAdj   Stat        Test
1   PC7 (4.7%)    Stage 0.00113   0.272  7.610       ANOVA
2   PC8 (4.5%)   Gender 0.00530   0.393  2.890      T-test
3  PC39 (0.4%)  Smoking 0.00645   0.393  5.490       ANOVA
4  PC30 (0.7%)   Gender 0.00655   0.393  2.820      T-test
5     PC9 (4%) Adjuvant 0.01200   0.508 -2.590      T-test
6  PC31 (0.7%)      Age 0.01330   0.508  0.310 Correlation
7  PC21 (1.7%)      Age 0.01580   0.508  0.303 Correlation
8  PC30 (0.7%)  Smoking 0.01690   0.508  4.370       ANOVA
9  PC18 (1.9%) Adjuvant 0.02430   0.614 -2.310      T-test
10 PC30 (0.7%)    Stage 0.02560   0.614  3.900       ANOVA
11 PC36 (0.4%)      Age 0.03550   0.679  0.265 Correlation
12 PC35 (0.5%) Adjuvant 0.03670   0.679  2.140      T-test
13 PC21 (1.7%)    Stage 0.03710   0.679  3.480       ANOVA
14 PC36 (0.4%)   Gender 0.04000   0.679  2.100      T-test
15  PC6 (5.3%)  Smoking 0.04500   0.679  3.270       ANOVA
16 PC39 (0.4%) Adjuvant 0.04520   0.679 -2.040      T-test</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proNormalRes_Krij$sig.feat.tab</code></pre>
<pre><code>      Var1       Var2     pVal pValAdj  Stat           Test      Gene
1   Q96LD4 Recurrence 0.000409   0.677  3.74 T-test (proDA)    TRIM47
2   P23921 Recurrence 0.000439   0.677 -3.72 T-test (proDA)      RRM1
3   Q9UK53 Recurrence 0.000501   0.677 -3.68 T-test (proDA)      ING1
4   Q96T58 Recurrence 0.000509   0.677 -3.67 T-test (proDA)      SPEN
5   P18887 Recurrence 0.000685   0.677  3.58 T-test (proDA)     XRCC1
6   Q9NX05 Recurrence 0.000913   0.677 -3.49 T-test (proDA)   FAM120C
7   P29317 Recurrence 0.000979   0.677 -3.46 T-test (proDA)     EPHA2
8   Q6ZVM7 Recurrence 0.001020   0.677 -3.45 T-test (proDA)    TOM1L2
9   O43708 Recurrence 0.001160   0.677  3.41 T-test (proDA)     GSTZ1
10  P37235 Recurrence 0.001160   0.677 -3.41 T-test (proDA)    HPCAL1
11  Q8IYS1 Recurrence 0.001220   0.677 -3.39 T-test (proDA)    PM20D2
12  Q32P28 Recurrence 0.001350   0.688  3.36 T-test (proDA)      P3H1
13  P47895 Recurrence 0.001520   0.715  3.32 T-test (proDA)   ALDH1A3
14  P28702 Recurrence 0.001780   0.778  3.27 T-test (proDA)      RXRB
15  Q01433 Recurrence 0.002120   0.789  3.21 T-test (proDA)     AMPD2
16  Q6P3W7 Recurrence 0.002240   0.789  3.19 T-test (proDA)     SCYL2
17  Q15811 Recurrence 0.002620   0.789  3.14 T-test (proDA)     ITSN1
18  P50454 Recurrence 0.002630   0.789  3.14 T-test (proDA)  SERPINH1
19  Q6ZWT7 Recurrence 0.002630   0.789 -3.14 T-test (proDA)    MBOAT2
20  Q9Y2Q9 Recurrence 0.002690   0.789 -3.13 T-test (proDA)    MRPS28
21  P48059 Recurrence 0.002700   0.789 -3.13 T-test (proDA)     LIMS1
22  Q9BV38 Recurrence 0.003560   0.851  3.03 T-test (proDA)     WDR18
23  O94979 Recurrence 0.003930   0.851  3.00 T-test (proDA)    SEC31A
24  Q12929 Recurrence 0.004060   0.851  2.99 T-test (proDA)      EPS8
25  O15460 Recurrence 0.004760   0.851  2.93 T-test (proDA)     P4HA2
26  Q9BUJ2 Recurrence 0.004970   0.851  2.92 T-test (proDA)  HNRNPUL1
27  O15439 Recurrence 0.004980   0.851  2.91 T-test (proDA)     ABCC4
28  O14548 Recurrence 0.005390   0.851 -2.89 T-test (proDA)   COX7A2L
29  Q9P107 Recurrence 0.005640   0.851  2.87 T-test (proDA)      GMIP
30  Q00341 Recurrence 0.005680   0.851  2.87 T-test (proDA)     HDLBP
31  Q9NRW3 Recurrence 0.006020   0.851  2.85 T-test (proDA)  APOBEC3C
32  O60784 Recurrence 0.006220   0.851  2.83 T-test (proDA)      TOM1
33  P61923 Recurrence 0.006930   0.851  2.79 T-test (proDA)     COPZ1
34  Q5VW36 Recurrence 0.006950   0.851  2.79 T-test (proDA)     FOCAD
35  O94855 Recurrence 0.006990   0.851  2.79 T-test (proDA)    SEC24D
36  Q14554 Recurrence 0.007100   0.851  2.79 T-test (proDA)     PDIA5
37  Q92508 Recurrence 0.007310   0.851 -2.78 T-test (proDA)    PIEZO1
38  P39656 Recurrence 0.007330   0.851  2.77 T-test (proDA)     DDOST
39  Q8N1F7 Recurrence 0.007860   0.851  2.75 T-test (proDA)     NUP93
40  O75695 Recurrence 0.007870   0.851 -2.75 T-test (proDA)       RP2
41  O43924 Recurrence 0.008130   0.851  2.74 T-test (proDA)     PDE6D
42  Q12872 Recurrence 0.008280   0.851 -2.73 T-test (proDA)    SFSWAP
43  Q9BUT1 Recurrence 0.008400   0.851  2.72 T-test (proDA)      BDH2
44  P15529 Recurrence 0.008480   0.851 -2.72 T-test (proDA)      CD46
45  Q96HY7 Recurrence 0.008500   0.851  2.72 T-test (proDA)    DHTKD1
46  Q96RF0 Recurrence 0.008550   0.851  2.72 T-test (proDA)     SNX18
47  Q9H488 Recurrence 0.008760   0.851  2.71 T-test (proDA)    POFUT1
48  O15379 Recurrence 0.008880   0.851 -2.70 T-test (proDA)     HDAC3
49  P14625 Recurrence 0.008900   0.851  2.70 T-test (proDA)   HSP90B1
50  P52948 Recurrence 0.008900   0.851  2.70 T-test (proDA)     NUP98
51  Q7KZF4 Recurrence 0.008920   0.851  2.70 T-test (proDA)      SND1
52  P49755 Recurrence 0.009110   0.851  2.69 T-test (proDA)    TMED10
53  O15321 Recurrence 0.009150   0.851  2.69 T-test (proDA)    TM9SF1
54  O43237 Recurrence 0.009380   0.851 -2.68 T-test (proDA)  DYNC1LI2
55  Q70UQ0 Recurrence 0.009420   0.851  2.68 T-test (proDA)     IKBIP
56  P49773 Recurrence 0.009460   0.851 -2.68 T-test (proDA)     HINT1
57  Q9H0R3 Recurrence 0.009580   0.851 -2.67 T-test (proDA)   TMEM222
58  Q9Y584 Recurrence 0.010000   0.851  2.66 T-test (proDA)    TIMM22
59  Q969G3 Recurrence 0.010200   0.851  2.65 T-test (proDA)   SMARCE1
60  Q96BP3 Recurrence 0.010500   0.851  2.64 T-test (proDA)     PPWD1
61  P78406 Recurrence 0.010700   0.851  2.64 T-test (proDA)      RAE1
62  Q9UBG0 Recurrence 0.010700   0.851  2.63 T-test (proDA)      MRC2
63  Q13308 Recurrence 0.010800   0.851  2.63 T-test (proDA)      PTK7
64  Q9Y5M8 Recurrence 0.010800   0.851  2.63 T-test (proDA)     SRPRB
65  P62263 Recurrence 0.010900   0.851  2.63 T-test (proDA)     RPS14
66  Q14152 Recurrence 0.011000   0.851  2.62 T-test (proDA)     EIF3A
67  Q9Y4P3 Recurrence 0.011000   0.851  2.62 T-test (proDA)      TBL2
68  Q01658 Recurrence 0.011100   0.851 -2.62 T-test (proDA)       DR1
69  Q96IJ6 Recurrence 0.011300   0.851  2.61 T-test (proDA)     GMPPA
70  O95136 Recurrence 0.011500   0.851  2.61 T-test (proDA)     S1PR2
71  Q15061 Recurrence 0.011500   0.851  2.60 T-test (proDA)     WDR43
72  Q9BSR8 Recurrence 0.011500   0.851  2.60 T-test (proDA)     YIPF4
73  P10599 Recurrence 0.011600   0.851 -2.60 T-test (proDA)       TXN
74  P49257 Recurrence 0.011700   0.851  2.60 T-test (proDA)     LMAN1
75  O43819 Recurrence 0.011800   0.851  2.60 T-test (proDA)      SCO2
76  P00338 Recurrence 0.011800   0.851  2.60 T-test (proDA)      LDHA
77  P11137 Recurrence 0.012100   0.851 -2.59 T-test (proDA)      MAP2
78  Q92572 Recurrence 0.012200   0.851  2.58 T-test (proDA)     AP3S1
79  Q9H7P6 Recurrence 0.012200   0.851 -2.58 T-test (proDA)    MVB12B
80  Q9NNX6 Recurrence 0.012900   0.851  2.56 T-test (proDA)     CD209
81  P61803 Recurrence 0.013100   0.851  2.56 T-test (proDA)      DAD1
82  O95167 Recurrence 0.013500   0.851  2.54 T-test (proDA)    NDUFA3
83  Q12769 Recurrence 0.013500   0.851  2.55 T-test (proDA)    NUP160
84  Q8TCZ2 Recurrence 0.013700   0.851 -2.54 T-test (proDA)    CD99L2
85  Q9P2B2 Recurrence 0.013700   0.851  2.54 T-test (proDA)    PTGFRN
86  O94819 Recurrence 0.013800   0.851  2.54 T-test (proDA)   KBTBD11
87  Q9Y2Q0 Recurrence 0.014000   0.851  2.53 T-test (proDA)    ATP8A1
88  O60333 Recurrence 0.014100   0.851 -2.53 T-test (proDA)     KIF1B
89  Q96FW1 Recurrence 0.014100   0.851  2.53 T-test (proDA)     OTUB1
90  Q07065 Recurrence 0.014400   0.851  2.52 T-test (proDA)     CKAP4
91  Q92896 Recurrence 0.014600   0.851  2.51 T-test (proDA)      GLG1
92  O43752 Recurrence 0.014700   0.851  2.51 T-test (proDA)      STX6
93  O60760 Recurrence 0.014700   0.851  2.51 T-test (proDA)     HPGDS
94  P48595 Recurrence 0.014800   0.851 -2.51 T-test (proDA) SERPINB10
95  Q14938 Recurrence 0.014800   0.851  2.51 T-test (proDA)      NFIX
96  Q9NPG4 Recurrence 0.014800   0.851 -2.51 T-test (proDA)    PCDH12
97  Q02809 Recurrence 0.014900   0.851  2.51 T-test (proDA)     PLOD1
98  Q06265 Recurrence 0.014900   0.851  2.51 T-test (proDA)    EXOSC9
99  O76003 Recurrence 0.015000   0.851 -2.50 T-test (proDA)     GLRX3
100 P05386 Recurrence 0.015800   0.851 -2.48 T-test (proDA)     RPLP1
101 P09651 Recurrence 0.015800   0.851  2.48 T-test (proDA)   HNRNPA1
102 Q9UHB9 Recurrence 0.015900   0.851  2.48 T-test (proDA)     SRP68
103 Q96IZ7 Recurrence 0.016100   0.851 -2.47 T-test (proDA)     RSRC1
104 Q96KR1 Recurrence 0.016200   0.851  2.47 T-test (proDA)       ZFR
105 Q9NZI7 Recurrence 0.016300   0.851  2.47 T-test (proDA)      UBP1
106 P50151 Recurrence 0.016400   0.851 -2.47 T-test (proDA)     GNG10
107 P29972 Recurrence 0.016800   0.851 -2.46 T-test (proDA)      AQP1
108 Q6BCY4 Recurrence 0.016800   0.851  2.46 T-test (proDA)    CYB5R2
109 Q9UI10 Recurrence 0.017000   0.851  2.45 T-test (proDA)    EIF2B4
110 O94788 Recurrence 0.017100   0.851  2.45 T-test (proDA)   ALDH1A2
111 Q8N9N7 Recurrence 0.017300   0.851 -2.45 T-test (proDA)    LRRC57
112 Q9H1Y0 Recurrence 0.017400   0.851  2.45 T-test (proDA)      ATG5
113 P12268 Recurrence 0.017600   0.851  2.44 T-test (proDA)    IMPDH2
114 Q9NYL4 Recurrence 0.017800   0.851  2.44 T-test (proDA)    FKBP11
115 Q9UNE7 Recurrence 0.017800   0.851 -2.44 T-test (proDA)     STUB1
116 P08670 Recurrence 0.018200   0.851  2.43 T-test (proDA)       VIM
117 Q9H7L9 Recurrence 0.018200   0.851  2.43 T-test (proDA)     SUDS3
118 P46734 Recurrence 0.018400   0.851  2.42 T-test (proDA)    MAP2K3
119 Q8WWY3 Recurrence 0.018500   0.851  2.42 T-test (proDA)    PRPF31
120 O75427 Recurrence 0.018800   0.851  2.41 T-test (proDA)     LRCH4
121 P35606 Recurrence 0.018900   0.851  2.41 T-test (proDA)     COPB2
122 O60264 Recurrence 0.019100   0.851  2.41 T-test (proDA)   SMARCA5
123 Q9H6Z4 Recurrence 0.019100   0.851  2.41 T-test (proDA)    RANBP3
124 Q92900 Recurrence 0.019400   0.851  2.40 T-test (proDA)      UPF1
125 Q9BX97 Recurrence 0.019500   0.851  2.40 T-test (proDA)     PLVAP
126 P01599 Recurrence 0.019600   0.851 -2.40 T-test (proDA)  IGKV1-17
127 P26641 Recurrence 0.019800   0.851  2.39 T-test (proDA)     EEF1G
128 P09758 Recurrence 0.019900   0.851 -2.39 T-test (proDA)   TACSTD2
129 Q96CS3 Recurrence 0.019900   0.851  2.39 T-test (proDA)      FAF2
130 P12429 Recurrence 0.020000   0.851 -2.39 T-test (proDA)     ANXA3
131 P30626 Recurrence 0.020000   0.851 -2.39 T-test (proDA)       SRI
132 Q8WYP5 Recurrence 0.020000   0.851  2.39 T-test (proDA)    AHCTF1
133 Q9NZC9 Recurrence 0.020000   0.851 -2.39 T-test (proDA)  SMARCAL1
134 P53992 Recurrence 0.020100   0.851  2.39 T-test (proDA)    SEC24C
135 O60826 Recurrence 0.020300   0.851  2.38 T-test (proDA)    CCDC22
136 Q9Y6R4 Recurrence 0.020300   0.851  2.38 T-test (proDA)    MAP3K4
137 Q7Z392 Recurrence 0.020500   0.851  2.38 T-test (proDA)  TRAPPC11
138 Q9UJV9 Recurrence 0.020500   0.851  2.38 T-test (proDA)     DDX41
139 Q7Z4Q2 Recurrence 0.020700   0.851  2.38 T-test (proDA)    HEATR3
140 Q9NRY5 Recurrence 0.021500   0.851  2.36 T-test (proDA)  FAM114A2
141 O14976 Recurrence 0.021700   0.851  2.36 T-test (proDA)       GAK
142 Q9Y5L4 Recurrence 0.021800   0.851 -2.35 T-test (proDA)    TIMM13
143 P52434 Recurrence 0.021900   0.851  2.35 T-test (proDA)    POLR2H
144 Q13561 Recurrence 0.022500   0.851 -2.34 T-test (proDA)     DCTN2
145 Q13356 Recurrence 0.022600   0.851  2.34 T-test (proDA)     PPIL2
146 Q9BY50 Recurrence 0.022700   0.851  2.34 T-test (proDA)    SEC11C
147 P53618 Recurrence 0.023000   0.851  2.33 T-test (proDA)     COPB1
148 Q15024 Recurrence 0.023100   0.851  2.33 T-test (proDA)    EXOSC7
149 P53621 Recurrence 0.023500   0.851  2.32 T-test (proDA)      COPA
150 P22059 Recurrence 0.023600   0.851  2.32 T-test (proDA)      OSBP
151 Q9BUR5 Recurrence 0.023600   0.851 -2.32 T-test (proDA)      APOO
152 Q99615 Recurrence 0.023700   0.851  2.32 T-test (proDA)    DNAJC7
153 A2RRP1 Recurrence 0.024000   0.851  2.31 T-test (proDA)      NBAS
154 Q9BSD7 Recurrence 0.024400   0.851 -2.31 T-test (proDA)     NTPCR
155 O00461 Recurrence 0.024700   0.851  2.30 T-test (proDA)    GOLIM4
156 O60701 Recurrence 0.024800   0.851  2.30 T-test (proDA)      UGDH
157 P49959 Recurrence 0.024900   0.851  2.30 T-test (proDA)     MRE11
158 O15523 Recurrence 0.025000   0.851 -2.30 T-test (proDA)     DDX3Y
159 Q92623 Recurrence 0.025300   0.851 -2.29 T-test (proDA)      TTC9
160 P51911 Recurrence 0.025500   0.851 -2.29 T-test (proDA)      CNN1
161 Q14764 Recurrence 0.025700   0.851  2.29 T-test (proDA)       MVP
162 Q92665 Recurrence 0.025700   0.851  2.29 T-test (proDA)    MRPS31
163 P24844 Recurrence 0.025900   0.851 -2.28 T-test (proDA)      MYL9
164 Q9BY43 Recurrence 0.025900   0.851  2.28 T-test (proDA)    CHMP4A
165 Q8N4Q0 Recurrence 0.026200   0.851  2.28 T-test (proDA)     ZADH2
166 Q96FJ2 Recurrence 0.026200   0.851 -2.28 T-test (proDA)    DYNLL2
167 Q15056 Recurrence 0.026300   0.851 -2.28 T-test (proDA)     EIF4H
168 Q9NWW5 Recurrence 0.026300   0.851  2.28 T-test (proDA)      CLN6
169 P49588 Recurrence 0.026500   0.851  2.27 T-test (proDA)     AARS1
170 Q5TAQ9 Recurrence 0.026600   0.851  2.27 T-test (proDA)     DCAF8
171 Q9Y2I2 Recurrence 0.026600   0.851 -2.27 T-test (proDA)     NTNG1
172 Q05707 Recurrence 0.027100   0.851  2.26 T-test (proDA)   COL14A1
173 Q13724 Recurrence 0.027200   0.851  2.26 T-test (proDA)      MOGS
174 P38919 Recurrence 0.027300   0.851  2.26 T-test (proDA)    EIF4A3
175 Q5T653 Recurrence 0.027300   0.851 -2.26 T-test (proDA)     MRPL2
176 Q6ZXV5 Recurrence 0.027500   0.851  2.26 T-test (proDA)     TMTC3
177 Q9UHX1 Recurrence 0.027800   0.851  2.25 T-test (proDA)     PUF60
178 P13639 Recurrence 0.027900   0.851  2.25 T-test (proDA)      EEF2
179 Q92759 Recurrence 0.028000   0.851  2.25 T-test (proDA)    GTF2H4
180 Q8N335 Recurrence 0.028300   0.851 -2.25 T-test (proDA)     GPD1L
181 Q96AY3 Recurrence 0.028300   0.851  2.25 T-test (proDA)    FKBP10
182 Q86VP3 Recurrence 0.028400   0.851 -2.25 T-test (proDA)     PACS2
183 Q14697 Recurrence 0.028600   0.851  2.24 T-test (proDA)     GANAB
184 P55795 Recurrence 0.028700   0.851  2.24 T-test (proDA)   HNRNPH2
185 P56589 Recurrence 0.028700   0.851 -2.24 T-test (proDA)      PEX3
186 Q4KWH8 Recurrence 0.028900   0.851  2.24 T-test (proDA)     PLCH1
187 P49821 Recurrence 0.029000   0.851  2.24 T-test (proDA)    NDUFV1
188 Q9BZE9 Recurrence 0.029300   0.851  2.23 T-test (proDA)   ASPSCR1
189 Q9UHD2 Recurrence 0.029300   0.851  2.23 T-test (proDA)      TBK1
190 Q9Y2L5 Recurrence 0.029600   0.851  2.23 T-test (proDA)   TRAPPC8
191 Q8TCD5 Recurrence 0.029700   0.851 -2.23 T-test (proDA)      NT5C
192 Q14331 Recurrence 0.029800   0.851  2.22 T-test (proDA)      FRG1
193 Q9BTZ2 Recurrence 0.029800   0.851  2.23 T-test (proDA)     DHRS4
194 P26358 Recurrence 0.030000   0.851  2.22 T-test (proDA)     DNMT1
195 Q96PU8 Recurrence 0.030000   0.851 -2.22 T-test (proDA)       QKI
196 O60763 Recurrence 0.031000   0.851  2.21 T-test (proDA)      USO1
197 Q5VWJ9 Recurrence 0.031000   0.851  2.21 T-test (proDA)     SNX30
198 Q9H6Y2 Recurrence 0.031100   0.851  2.21 T-test (proDA)     WDR55
199 Q7L4I2 Recurrence 0.031200   0.851 -2.21 T-test (proDA)     RSRC2
200 Q96J84 Recurrence 0.031300   0.851  2.20 T-test (proDA)   KIRREL1
201 Q00722 Recurrence 0.031600   0.851  2.20 T-test (proDA)     PLCB2
202 Q96CT7 Recurrence 0.031600   0.851  2.20 T-test (proDA)   CCDC124
203 P78527 Recurrence 0.031700   0.851  2.20 T-test (proDA)     PRKDC
204 Q9H1Z4 Recurrence 0.031700   0.851  2.20 T-test (proDA)     WDR13
205 Q9UQ80 Recurrence 0.031800   0.851  2.20 T-test (proDA)     PA2G4
206 Q15843 Recurrence 0.031900   0.851 -2.20 T-test (proDA)     NEDD8
207 Q0VGL1 Recurrence 0.032100   0.851  2.19 T-test (proDA)   LAMTOR4
208 P48444 Recurrence 0.032200   0.851  2.19 T-test (proDA)     ARCN1
209 P62829 Recurrence 0.032300   0.851  2.19 T-test (proDA)     RPL23
210 Q6NWY9 Recurrence 0.032300   0.851 -2.19 T-test (proDA)   PRPF40B
211 Q8TF01 Recurrence 0.032300   0.851 -2.19 T-test (proDA)     PNISR
212 Q8WVM0 Recurrence 0.032300   0.851  2.19 T-test (proDA)     TFB1M
213 P11233 Recurrence 0.032400   0.851 -2.19 T-test (proDA)      RALA
214 P17706 Recurrence 0.032400   0.851 -2.19 T-test (proDA)     PTPN2
215 P42566 Recurrence 0.032400   0.851  2.19 T-test (proDA)     EPS15
216 P37268 Recurrence 0.032700   0.851  2.19 T-test (proDA)     FDFT1
217 O43760 Recurrence 0.032800   0.851 -2.18 T-test (proDA)    SYNGR2
218 P33316 Recurrence 0.032800   0.851 -2.18 T-test (proDA)       DUT
219 O43293 Recurrence 0.033100   0.851 -2.18 T-test (proDA)     DAPK3
220 Q9NUJ3 Recurrence 0.033100   0.851 -2.18 T-test (proDA)   TCP11L1
221 O60443 Recurrence 0.033200   0.851  2.18 T-test (proDA)     GSDME
222 P52597 Recurrence 0.033200   0.851  2.18 T-test (proDA)    HNRNPF
223 P78347 Recurrence 0.033200   0.851  2.18 T-test (proDA)     GTF2I
224 Q12905 Recurrence 0.033300   0.851  2.18 T-test (proDA)      ILF2
225 Q14289 Recurrence 0.033300   0.851  2.18 T-test (proDA)     PTK2B
226 Q9UBS8 Recurrence 0.033300   0.851 -2.18 T-test (proDA)     RNF14
227 P78524 Recurrence 0.033400   0.851 -2.18 T-test (proDA)   DENND2B
228 P33527 Recurrence 0.033500   0.851  2.18 T-test (proDA)     ABCC1
229 Q9NVZ3 Recurrence 0.033500   0.851  2.18 T-test (proDA)    NECAP2
230 P46777 Recurrence 0.033600   0.851  2.17 T-test (proDA)      RPL5
231 Q6Y288 Recurrence 0.033800   0.851  2.17 T-test (proDA)    B3GLCT
232 Q9NUU6 Recurrence 0.033800   0.851 -2.17 T-test (proDA)   OTULINL
233 Q9NZN3 Recurrence 0.033900   0.851 -2.17 T-test (proDA)      EHD3
234 P19623 Recurrence 0.034000   0.851  2.17 T-test (proDA)       SRM
235 Q5T2T1 Recurrence 0.034000   0.851  2.17 T-test (proDA)      MPP7
236 P60228 Recurrence 0.034100   0.851  2.17 T-test (proDA)     EIF3E
237 P09429 Recurrence 0.034200   0.851  2.17 T-test (proDA)     HMGB1
238 O15116 Recurrence 0.034400   0.851  2.16 T-test (proDA)      LSM1
239 O95394 Recurrence 0.034500   0.851  2.16 T-test (proDA)      PGM3
240 Q15436 Recurrence 0.034700   0.851  2.16 T-test (proDA)    SEC23A
241 Q9UK45 Recurrence 0.035000   0.851 -2.16 T-test (proDA)      LSM7
242 Q10567 Recurrence 0.035100   0.851  2.16 T-test (proDA)     AP1B1
243 Q9BQB6 Recurrence 0.035100   0.851  2.16 T-test (proDA)    VKORC1
244 P63173 Recurrence 0.035200   0.851  2.15 T-test (proDA)     RPL38
245 Q9P2L0 Recurrence 0.035200   0.851 -2.15 T-test (proDA)     WDR35
246 O95833 Recurrence 0.035500   0.851 -2.15 T-test (proDA)     CLIC3
247 P30530 Recurrence 0.035500   0.851  2.15 T-test (proDA)       AXL
248 Q96EY5 Recurrence 0.035500   0.851 -2.15 T-test (proDA)    MVB12A
249 O94776 Recurrence 0.035600   0.851  2.15 T-test (proDA)      MTA2
250 Q96C01 Recurrence 0.035600   0.851  2.15 T-test (proDA)   FAM136A
251 Q9HD20 Recurrence 0.035600   0.851  2.15 T-test (proDA)   ATP13A1
252 P61247 Recurrence 0.035800   0.851  2.15 T-test (proDA)     RPS3A
253 P62136 Recurrence 0.035800   0.851 -2.15 T-test (proDA)    PPP1CA
254 Q16576 Recurrence 0.036400   0.851  2.14 T-test (proDA)     RBBP7
255 P43405 Recurrence 0.036500   0.851  2.14 T-test (proDA)       SYK
256 Q9NZZ3 Recurrence 0.036500   0.851 -2.14 T-test (proDA)     CHMP5
257 Q9BST9 Recurrence 0.036600   0.851  2.14 T-test (proDA)      RTKN
258 Q9NVA2 Recurrence 0.036700   0.851  2.14 T-test (proDA)  SEPTIN11
259 Q01813 Recurrence 0.036800   0.851  2.13 T-test (proDA)      PFKP
260 Q9Y5X1 Recurrence 0.036900   0.851  2.13 T-test (proDA)      SNX9
261 P61006 Recurrence 0.037000   0.851  2.13 T-test (proDA)     RAB8A
262 Q02040 Recurrence 0.037000   0.851 -2.13 T-test (proDA)   AKAP17A
263 Q9Y2H0 Recurrence 0.037000   0.851 -2.13 T-test (proDA)    DLGAP4
264 P36269 Recurrence 0.037100   0.851  2.13 T-test (proDA)      GGT5
265 P78559 Recurrence 0.037200   0.851 -2.13 T-test (proDA)     MAP1A
266 P0CAP1 Recurrence 0.037300   0.851 -2.13 T-test (proDA)     MYZAP
267 O95772 Recurrence 0.037400   0.851  2.13 T-test (proDA)  STARD3NL
268 Q14118 Recurrence 0.037500   0.851 -2.13 T-test (proDA)      DAG1
269 Q9ULH1 Recurrence 0.037500   0.851  2.13 T-test (proDA)     ASAP1
270 O43427 Recurrence 0.037600   0.851  2.13 T-test (proDA)      FIBP
271 O43822 Recurrence 0.037700   0.851 -2.12 T-test (proDA)   CFAP410
272 P33908 Recurrence 0.037900   0.853  2.12 T-test (proDA)    MAN1A1
273 P98172 Recurrence 0.038100   0.853 -2.12 T-test (proDA)     EFNB1
274 Q8TD43 Recurrence 0.038100   0.853 -2.12 T-test (proDA)     TRPM4
275 Q8IXK2 Recurrence 0.038400   0.853  2.12 T-test (proDA)   GALNT12
276 Q5JRX3 Recurrence 0.038700   0.853  2.11 T-test (proDA)    PITRM1
277 P08174 Recurrence 0.038800   0.853 -2.11 T-test (proDA)      CD55
278 Q86U44 Recurrence 0.038800   0.853  2.11 T-test (proDA)    METTL3
279 Q96I15 Recurrence 0.039000   0.853  2.11 T-test (proDA)      SCLY
280 Q9BWF3 Recurrence 0.039000   0.853  2.11 T-test (proDA)      RBM4
281 P29373 Recurrence 0.039300   0.854  2.11 T-test (proDA)    CRABP2
282 P54136 Recurrence 0.039300   0.854  2.11 T-test (proDA)     RARS1
283 Q9Y5B0 Recurrence 0.039600   0.857  2.10 T-test (proDA)     CTDP1
284 Q5SRE7 Recurrence 0.040100   0.861  2.10 T-test (proDA)    PHYHD1
285 Q9UJY4 Recurrence 0.040300   0.861  2.09 T-test (proDA)      GGA2
286 Q9NYU2 Recurrence 0.040600   0.861  2.09 T-test (proDA)     UGGT1
287 Q96B23 Recurrence 0.040700   0.861 -2.09 T-test (proDA)  C18orf25
288 Q9H269 Recurrence 0.040700   0.861  2.09 T-test (proDA)     VPS16
289 P48723 Recurrence 0.041100   0.861  2.09 T-test (proDA)    HSPA13
290 P63244 Recurrence 0.041100   0.861  2.09 T-test (proDA)     RACK1
291 O15357 Recurrence 0.041200   0.861  2.09 T-test (proDA)    INPPL1
292 Q8TCE6 Recurrence 0.041400   0.861  2.08 T-test (proDA)   DENND10
293 Q9UJU6 Recurrence 0.041400   0.861  2.08 T-test (proDA)      DBNL
294 Q07954 Recurrence 0.041500   0.861  2.08 T-test (proDA)      LRP1
295 P49459 Recurrence 0.041600   0.861 -2.08 T-test (proDA)     UBE2A
296 Q9H4M9 Recurrence 0.041600   0.861 -2.08 T-test (proDA)      EHD1
297 O95248 Recurrence 0.042000   0.861  2.08 T-test (proDA)      SBF1
298 O95298 Recurrence 0.042100   0.861  2.08 T-test (proDA)    NDUFC2
299 Q96JH7 Recurrence 0.042100   0.861  2.08 T-test (proDA)    VCPIP1
300 Q96A73 Recurrence 0.042400   0.861 -2.07 T-test (proDA)  KIAA1191
301 Q99715 Recurrence 0.042600   0.861 -2.07 T-test (proDA)   COL12A1
302 A4D1P6 Recurrence 0.043000   0.861  2.07 T-test (proDA)     WDR91
303 Q92796 Recurrence 0.043000   0.861 -2.07 T-test (proDA)      DLG3
304 Q9Y6U3 Recurrence 0.043000   0.861  2.07 T-test (proDA)      SCIN
305 Q96SB4 Recurrence 0.043100   0.861 -2.07 T-test (proDA)     SRPK1
306 Q9P2I0 Recurrence 0.043400   0.861  2.06 T-test (proDA)     CPSF2
307 Q5S007 Recurrence 0.043500   0.861  2.06 T-test (proDA)     LRRK2
308 Q96JK2 Recurrence 0.043900   0.861 -2.06 T-test (proDA)     DCAF5
309 O43759 Recurrence 0.044200   0.861  2.05 T-test (proDA)    SYNGR1
310 Q92621 Recurrence 0.044300   0.861  2.05 T-test (proDA)    NUP205
311 Q9Y5X2 Recurrence 0.044300   0.861  2.05 T-test (proDA)      SNX8
312 Q15005 Recurrence 0.044600   0.861  2.05 T-test (proDA)     SPCS2
313 Q96Q05 Recurrence 0.044800   0.861 -2.05 T-test (proDA)   TRAPPC9
314 Q9BVK2 Recurrence 0.044800   0.861  2.05 T-test (proDA)      ALG8
315 P26639 Recurrence 0.044900   0.861  2.05 T-test (proDA)     TARS1
316 P27361 Recurrence 0.045000   0.861 -2.05 T-test (proDA)     MAPK3
317 Q13085 Recurrence 0.045500   0.861  2.04 T-test (proDA)     ACACA
318 Q8WUF8 Recurrence 0.045500   0.861  2.04 T-test (proDA)   FAM172A
319 Q969L2 Recurrence 0.045500   0.861 -2.04 T-test (proDA)      MAL2
320 Q9BZQ6 Recurrence 0.045500   0.861 -2.04 T-test (proDA)     EDEM3
321 P37108 Recurrence 0.045600   0.861  2.04 T-test (proDA)     SRP14
322 Q10589 Recurrence 0.045700   0.861 -2.04 T-test (proDA)      BST2
323 P15941 Recurrence 0.045900   0.861  2.04 T-test (proDA)      MUC1
324 Q9NP66 Recurrence 0.045900   0.861  2.04 T-test (proDA)    HMG20A
325 Q8N6L1 Recurrence 0.046000   0.861  2.04 T-test (proDA)   KRTCAP2
326 P42167 Recurrence 0.046300   0.861 -2.03 T-test (proDA)      TMPO
327 Q05513 Recurrence 0.046300   0.861 -2.03 T-test (proDA)     PRKCZ
328 Q13601 Recurrence 0.046300   0.861  2.03 T-test (proDA)      KRR1
329 Q9Y342 Recurrence 0.046300   0.861 -2.03 T-test (proDA)      PLLP
330 O60488 Recurrence 0.046400   0.861 -2.03 T-test (proDA)     ACSL4
331 P04216 Recurrence 0.046600   0.861  2.03 T-test (proDA)      THY1
332 Q08397 Recurrence 0.046700   0.861 -2.03 T-test (proDA)     LOXL1
333 Q92959 Recurrence 0.047400   0.865 -2.02 T-test (proDA)   SLCO2A1
334 Q9BYT8 Recurrence 0.047400   0.865  2.02 T-test (proDA)       NLN
335 Q9BVL2 Recurrence 0.047700   0.865 -2.02 T-test (proDA)     NUP58
336 Q8N4T8 Recurrence 0.047800   0.865  2.02 T-test (proDA)      CBR4
337 Q8TB03 Recurrence 0.047900   0.865 -2.02 T-test (proDA)   CXorf38
338 Q9NX14 Recurrence 0.048000   0.865  2.02 T-test (proDA)   NDUFB11
339 Q86YL7 Recurrence 0.048200   0.865 -2.02 T-test (proDA)      PDPN
340 Q8TDD1 Recurrence 0.048500   0.865  2.01 T-test (proDA)     DDX54
341 P10398 Recurrence 0.048700   0.865  2.01 T-test (proDA)      ARAF
342 Q8WXX5 Recurrence 0.048700   0.865 -2.01 T-test (proDA)    DNAJC9
343 Q8WXA3 Recurrence 0.048900   0.865  2.01 T-test (proDA)     RUFY2
344 Q9P2J5 Recurrence 0.048900   0.865  2.01 T-test (proDA)     LARS1
345 O60869 Recurrence 0.049200   0.865  2.01 T-test (proDA)      EDF1
346 Q92888 Recurrence 0.049300   0.865  2.01 T-test (proDA)   ARHGEF1
347 P46100 Recurrence 0.049400   0.865  2.01 T-test (proDA)      ATRX
348 Q5JVF3 Recurrence 0.049500   0.865  2.00 T-test (proDA)     PCID2
349 O00267 Recurrence 0.049700   0.865  2.00 T-test (proDA)    SUPT5H
350 Q15773 Recurrence 0.049800   0.865 -2.00 T-test (proDA)      MLF2
351 O60716 Recurrence 0.049900   0.865 -2.00 T-test (proDA)    CTNND1
352 Q9Y680 Recurrence 0.049900   0.865  2.00 T-test (proDA)     FKBP7
353 Q3KQV9 Recurrence 0.050000   0.865  2.00 T-test (proDA)    UAP1L1</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proNormalRes_Krij$sig.feat.heat</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Normal samples (ns:
p &gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01, <span
class="math inline">\(***\)</span>: p &lt;= 0.001)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Krij$data
smpAnnoTab &lt;- proTissueRes_Krij$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proNormalRes_Krij$sig.feat.tab
num_sigFeats &lt;- 6
gpLevel = c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;, &#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;)
condiCol = c(&#39;firebrick&#39;, &#39;grey50&#39;) #c(Normal, Tumor)
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(newVar1 = paste0(Var1, &#39; (&#39;, Gene, &#39;)&#39;),
                newVar1 = factor(newVar1, levels = unique(newVar1)))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Recur_Condi = paste0(Recurrence, &#39;_&#39;, Condition),
                Recur_Condi = factor(Recur_Condi, levels = gpLevel))
ggplot(topSigFeatDat, aes(x=Recur_Condi, y=Abundance, col=Condition, fill=Recurrence)) +
  geom_boxplot(alpha = 1, outlier.shape = NA, linewidth = 1) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             comparisons = list(c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;), c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;)),
                             label = &#39;p.signif&#39;, tip.length = 0.015, bracket.size = 0.7, size = 4) +
  labs(x = &#39;Recurrence&#39;) +
  scale_color_manual(values = condiCol) +
  scale_fill_manual(values = c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(newVar1), scales = &#39;free&#39;) +
  th +
  theme(strip.text = element_text(size = 13, face = &#39;bold&#39;),
             axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># theme(strip.text = element_text(size = 14, face = &#39;bold&#39;),
#       axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 20),
#       axis.title = element_text(size = 26),
#       legend.title = element_text(size = 24), legend.text = element_text(size = 22))</code></pre>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proNormalRes_Krij$pval.hist +
  labs(title = &#39;Krij. Combined Normal Tissue DIA Proteomics&#39;) +
  theme(axis.title = element_text(size = 28),
        axis.text = element_text(size = 16))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/group_meeting/pVals_histogram_combined_proNormal_Krij.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 6, width = 11)</code></pre>
<p>There is no significant association between PCs (Var1) and cancer
recurrence (Var2).</p>
<pre class="r"><code>proNormalRes_Krij$sig.pc.tab</code></pre>
<pre><code>[1] Var1    Var2    pVal    pValAdj Stat    Test   
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>Enrichment analysis<br />
Reference: Hallmark gene sets<br />
<strong>t-values</strong></p>
<pre class="r"><code># Prepare ranked gene list
sigRes &lt;- proNormalRes_Krij$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, Stat) %&gt;%
  tibble::column_to_rownames(&#39;Var1&#39;)
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::select(Protein, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)

rankedGeneList &lt;- prepRankedFeatList(statRes = sigRes, featAnno = featAnnoTab, to_gene = T)</code></pre>
<pre class="r"><code># Hallmark gene sets
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)

# Run GSEA
gseaResH &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabH,
                                  minGSSize = 10, maxGSSize = 500,
                                  pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                  by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.6, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tissue-proteomics-ag-klingmüller" class="section level1">
<h1>Tissue Proteomics (AG Klingmüller)</h1>
<p>Discovery (partial) + MethodDev</p>
<pre class="r"><code># Load preprocessed data
proTissue_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsnBC.rds&#39;)
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::select(Protein, PG.Genes) %&gt;%
  dplyr::rename(Gene = PG.Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)

# Perform single-omics analysis
proTissueRes_Klin &lt;- doSOA(proTissue_Klin, meta_var = c(&#39;Condition&#39;, &#39;Cohort&#39;), pca_method = &#39;ppca&#39;, do_onlyPCA = T)</code></pre>
<p><strong>Do metadata-assisted quality control</strong><br />
PC1 significantly separates Tumor and Normal samples, which indicates
decent data quality.</p>
<pre class="r"><code>pcTab &lt;- proTissueRes_Klin$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;,
                                                       &#39;MJMTYR_TU&#39;, &#39;XFKHGP_TU&#39;,
                                                       &#39;XFKHGP_NG&#39;, &#39;MJMTYR_NG&#39;,
                                                       &#39;I0HVOL_NG&#39;, &#39;7EAOX7_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (31.8%)`, color=Condition, fill=Condition, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 20),
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/boxplot_quality_control_combined_proTissue_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 9, width = 11)</code></pre>
<pre class="r"><code>proTissueCondiRes &lt;- doSOA(proTissue_Klin, meta_var = &#39;Condition&#39;, use_proDA = T)
datTab &lt;- t(proTissueCondiRes$data) %&gt;%
  as_tibble(rownames = &#39;Sample&#39;) %&gt;%
  dplyr::select(Sample, P63104) %&gt;%
  left_join(proTissueCondiRes$smpMetadata, by = &#39;Sample&#39;)

ggplot(datTab, aes(x=Condition, y=P63104, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 9, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;, y = &#39;P63104 (YWHAZ)&#39;) +
  th +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 20),
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))
# ggsave(&#39;./output/Discovery/2nd_TAC/boxplot_YWHAZ_combined_proTissue_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)
# =&gt; Tumor P63104 abundances are not associated with pathological stages or tumor purity.
# Difference in mean abundances between Normal and Tumor samples might be larger
# if patients are at advanced cancer stage.

# tmp_datTab &lt;- dplyr::filter(datTab, Condition %in% &#39;Tumor&#39;)
# ggplot(tmp_datTab, aes(x=Stage, y=P63104, col=Stage, fill=Stage)) +
#   geom_boxplot(alpha = 0.7, outlier.shape = NA) +
#   geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
#   ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
#                              comparisons = list(c(&#39;IB&#39;, &#39;IIA&#39;), c(&#39;IIA&#39;, &#39;IIB&#39;), c(&#39;IB&#39;, &#39;IIB&#39;)),
#                              label = &#39;p.signif&#39;, tip.length = 0.02, bracket.size = 0.7, size = 6) +
#   labs(x = &#39;Pathological stage&#39;, y = &#39;P63104 (YWHAZ)&#39;) +
#   th
# 
# ggplot(tmp_datTab, aes(x=P63104, y=TumorPurity, col=Stage)) +
#   geom_point(size = 4) +
#   labs(x = &#39;P63104 (YWHAZ)&#39;, y = &#39;Tumor purity (%)&#39;) +
#   th</code></pre>
<div id="tumor-samples-1" class="section level2">
<h2>Tumor samples</h2>
<p>Display significant associations between SVs and sample metadata
variables. SVs (surrogate variables) will be included in linear model
for differential analysis</p>
<pre class="r"><code># Subset certain samples
proTumor_Klin &lt;- proTissue_Klin[, which(colData(proTissue_Klin)$Condition == &#39;Tumor&#39;)]

# Do SVA to identify latent factors that explain unknown variance
# Prepare data matrix and sample metadata table for creating model matrices
datMat &lt;- assay(proTumor_Klin)
# Keep only complete features
keptFeats &lt;- apply(datMat, 1, function(featVec) {
  !any(is.na(featVec))
})
datMat &lt;- datMat[keptFeats,]
metadatTab &lt;- colData(proTumor_Klin) %&gt;%
  tibble::as_tibble(rownames = &#39;Sample&#39;)
# Create full model matrix including both adjustment variables and variables of interest
modelMat &lt;- model.matrix(~ Recurrence, data = metadatTab)
# Create null model matrix including only adjustment variables
modelMat0 &lt;- model.matrix(~ 1, data = metadatTab)

# Perform SVA
# Identify number of latent factors that need to be estimated
numSV = sva::num.sv(datMat, modelMat, method = &#39;be&#39;)
svObj = sva(datMat, modelMat, modelMat0, n.sv = numSV)</code></pre>
<pre><code>Number of significant surrogate variables is:  8 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code># Do association tests between SVs and patient metadata
# Prepare surrogate variable table
svTab &lt;- svObj$sv
rownames(svTab) &lt;- metadatTab$Sample
colnames(svTab) &lt;- paste0(&#39;SV&#39;, seq_len(ncol(svTab)))
svTab &lt;- tibble::as_tibble(svTab, rownames = &#39;Sample&#39;)
# Prepare patient metadata table
metaTab &lt;- dplyr::select(metadatTab, Sample, Recurrence, Gender, Age, Smoking, Stage,
                         Adjuvant)
# Do association tests
testAsso(svTab, metaTab, cmn_col = &#39;Sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05)</code></pre>
<pre><code>  Var1  Var2      pVal   pValAdj     Stat  Test
1  SV2 Stage 0.0274647 0.8628037 3.978965 ANOVA</code></pre>
<pre class="r"><code># Include surrogate variables into SE object
proTumorTab &lt;- summExp2df(proTumor_Klin, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;)
proTumorTab &lt;- dplyr::left_join(proTumorTab, svTab, by = &#39;Sample&#39;)
proTumor_Klin &lt;- df2SummExp(proTumorTab, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;,
                             values = &#39;Value&#39;, row_anno = colnames(rowData(proTumor_Klin)),
                             col_anno = c(colnames(colData(proTumor_Klin)),
                                          paste0(&#39;SV&#39;, seq_len(ncol(svTab)-1))))
assay(proTumor_Klin) &lt;- as.matrix(assay(proTumor_Klin))</code></pre>
<pre class="r"><code># Perform single-omics analysis
proTumorRes_Klin &lt;- doSingleOmicsAnalysis(proTumor_Klin, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
                                          pca_method = &#39;ppca&#39;, num_PCs = 37, num_PCfeats = 30,
                                          use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
                                          feat_conv = T, plot_title = NULL, show_rownames = F,
                                          pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                          &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;,
                                                          &#39;TumorPurity&#39;, &#39;Cohort&#39;),
                                          soa_unwantVar = paste0(&#39;SV&#39;, seq_len(8)))</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proTumorRes_Klin$data.var.source</code></pre>
<pre><code>          Var1        Var2     pVal pValAdj   Stat        Test
1  PC19 (1.8%)      Gender 0.000142   0.042 -4.240      T-test
2  PC19 (1.8%)  Recurrence 0.010200   0.785 -2.710      T-test
3  PC18 (1.9%)    Adjuvant 0.011500   0.785  2.660      T-test
4  PC11 (2.8%)     Smoking 0.012600   0.785  4.950       ANOVA
5  PC35 (1.1%) TumorPurity 0.013300   0.785 -0.393 Correlation
6  PC34 (1.1%)  Recurrence 0.015900   0.785 -2.530      T-test
7  PC20 (1.8%)       Stage 0.029600   0.966  3.890       ANOVA
8  PC14 (2.2%)         Age 0.030800   0.966  0.346 Correlation
9  PC29 (1.3%)     Smoking 0.033700   0.966  3.730       ANOVA
10 PC29 (1.3%) TumorPurity 0.034600   0.966 -0.339 Correlation
11   PC2 (10%)       Stage 0.037600   0.966  3.600       ANOVA
12 PC10 (2.9%)    Adjuvant 0.041900   0.966 -2.110      T-test
13    PC6 (4%)      Gender 0.043000   0.966  2.100      T-test
14 PC18 (1.9%)     Smoking 0.045700   0.966  3.370       ANOVA</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2). Owing to missing values, there may be situation that
only few observations in sample groups, resulting in unreliable
t-statistics. To this end, we used probabilistic dropout model (proDA)
to account for data missingness.</p>
<pre class="r"><code>dplyr::select(proTumorRes_Klin$sig.feat.tab, -Test)</code></pre>
<pre><code>                    Var1       Var2     pVal pValAdj  Stat              Gene
1                 Q96G91 Recurrence 0.000110   0.903  4.47            P2RY11
2                 P00739 Recurrence 0.000402   0.999 -4.00               HPR
3                 A5D8V6 Recurrence 0.000469   0.999  3.94            VPS37C
4                 P49326 Recurrence 0.000921   0.999 -3.69              FMO5
5                 Q13637 Recurrence 0.001080   0.999  3.63             RAB32
6                 P30626 Recurrence 0.001440   0.999  3.52               SRI
7                 O75475 Recurrence 0.001460   0.999 -3.52             PSIP1
8                 Q03468 Recurrence 0.001520   0.999 -3.50             ERCC6
9                 P48595 Recurrence 0.001560   0.999 -3.49         SERPINB10
10                A7KAX9 Recurrence 0.001740   0.999  3.45          ARHGAP32
11                P0C0L5 Recurrence 0.001750   0.999 -3.45             C4B_2
12                O94811 Recurrence 0.001800   0.999 -3.44              TPPP
13                Q53ET0 Recurrence 0.002040   0.999  3.39             CRTC2
14                P27169 Recurrence 0.002080   0.999 -3.38              PON1
15                Q9UNX4 Recurrence 0.002400   0.999  3.33              WDR3
16                P10515 Recurrence 0.002880   0.999  3.25              DLAT
17                Q6DHV7 Recurrence 0.003120   0.999 -3.22              ADAL
18                O75531 Recurrence 0.003180   0.999  3.22             BANF1
19                O95445 Recurrence 0.003450   0.999 -3.19              APOM
20                Q06278 Recurrence 0.003460   0.999 -3.18              AOX1
21                Q8N4T8 Recurrence 0.003510   0.999  3.18              CBR4
22                P48449 Recurrence 0.003560   0.999  3.17               LSS
23            A0A0C4DH36 Recurrence 0.003640   0.999 -3.16          IGHV3-38
24                Q9C0B1 Recurrence 0.003640   0.999 -3.16               FTO
25                P28065 Recurrence 0.003660   0.999 -3.16             PSMB9
26                P23508 Recurrence 0.003740   0.999 -3.15               MCC
27                Q7Z434 Recurrence 0.004050   0.999  3.12              MAVS
28                P16930 Recurrence 0.004060   0.999 -3.12               FAH
29                O14791 Recurrence 0.004110   0.999 -3.12             APOL1
30                Q9BPX7 Recurrence 0.004680   0.999 -3.06           C7orf25
31                O60934 Recurrence 0.004740   0.999  3.06               NBN
32                Q13541 Recurrence 0.005060   0.999  3.03          EIF4EBP1
33                Q96I59 Recurrence 0.005230   0.999  3.02             NARS2
34                O00330 Recurrence 0.005400   0.999  3.01              PDHX
35                P06276 Recurrence 0.005680   0.999 -2.99              BCHE
36                Q9Y5J9 Recurrence 0.005700   0.999  2.99            TIMM8B
37                Q9ULR0 Recurrence 0.005770   0.999 -2.98              ISY1
38                P11217 Recurrence 0.005800   0.999 -2.98              PYGM
39                P08236 Recurrence 0.005990   0.999 -2.97              GUSB
40                Q9UK76 Recurrence 0.006020   0.999  2.96              JPT1
41                Q9BUK6 Recurrence 0.006110   0.999 -2.96             MSTO1
42                Q13126 Recurrence 0.006210   0.999 -2.95              MTAP
43                Q15828 Recurrence 0.006240   0.999  2.95              CST6
44                Q9UNN5 Recurrence 0.006430   0.999 -2.94              FAF1
45                Q8N6H7 Recurrence 0.006440   0.999  2.94           ARFGAP2
46                Q14202 Recurrence 0.006520   0.999 -2.93             ZMYM3
47                O15230 Recurrence 0.006780   0.999  2.92             LAMA5
48                Q9BZ29 Recurrence 0.006830   0.999  2.91             DOCK9
49                P21980 Recurrence 0.006840   0.999  2.91              TGM2
50                P01011 Recurrence 0.007400   0.999 -2.88          SERPINA3
51                Q9Y3B8 Recurrence 0.007550   0.999  2.87             REXO2
52                Q8NHG8 Recurrence 0.007930   0.999  2.85             ZNRF2
53                P0C870 Recurrence 0.008020   0.999 -2.85             JMJD7
54                A1IGU5 Recurrence 0.008060   0.999 -2.85          ARHGEF37
55                Q13835 Recurrence 0.008080   0.999 -2.84              PKP1
56                Q68CR1 Recurrence 0.008170   0.999  2.84            SEL1L3
57                Q6NUK1 Recurrence 0.008250   0.999  2.84          SLC25A24
58                Q02224 Recurrence 0.008360   0.999  2.83             CENPE
59                Q9NVH0 Recurrence 0.008410   0.999  2.83              EXD2
60                O15066 Recurrence 0.008670   0.999  2.82             KIF3B
61                Q6WKZ4 Recurrence 0.008860   0.999  2.81         RAB11FIP1
62                O75182 Recurrence 0.008940   0.999 -2.80             SIN3B
63                Q96T68 Recurrence 0.009540   0.999  2.78            SETDB2
64                P55268 Recurrence 0.009560   0.999  2.77             LAMB2
65                Q86UT6 Recurrence 0.009720   0.999  2.77             NLRX1
66                Q5MNZ9 Recurrence 0.009750   0.999 -2.77             WIPI1
67                P27824 Recurrence 0.009840   0.999 -2.76              CANX
68                Q5T3J3 Recurrence 0.009920   0.999 -2.76             LRIF1
69                Q6P1X6 Recurrence 0.010000   0.999  2.75           C8orf82
70                Q3KP31 Recurrence 0.010100   0.999  2.75            ZNF791
71                Q86UP3 Recurrence 0.010100   0.999 -2.75             ZFHX4
72                P25788 Recurrence 0.010200   0.999 -2.75             PSMA3
73                O60716 Recurrence 0.010300   0.999  2.74            CTNND1
74                Q8N1F8 Recurrence 0.010300   0.999  2.75           STK11IP
75                Q8NEV8 Recurrence 0.010400   0.999  2.74             EXPH5
76                P58546 Recurrence 0.010500   0.999  2.74              MTPN
77                Q96RS6 Recurrence 0.010500   0.999  2.74            NUDCD1
78                P02766 Recurrence 0.010800   0.999 -2.72               TTR
79                Q96RF0 Recurrence 0.010900   0.999  2.72             SNX18
80                P12830 Recurrence 0.011200   0.999  2.71              CDH1
81                P25685 Recurrence 0.011200   0.999  2.71            DNAJB1
82                Q9BXB4 Recurrence 0.011300   0.999 -2.71           OSBPL11
83                A1L4H1 Recurrence 0.011400   0.999 -2.70             SSC5D
84                O14975 Recurrence 0.011500   0.999 -2.70           SLC27A2
85                Q9UPR0 Recurrence 0.011500   0.999 -2.70             PLCL2
86                P67870 Recurrence 0.011600   0.999  2.70            CSNK2B
87                Q9HC16 Recurrence 0.011600   0.999 -2.69          APOBEC3G
88                O00237 Recurrence 0.011900   0.999 -2.69            RNF103
89                Q16643 Recurrence 0.012200   0.999 -2.67              DBN1
90                Q96BR1 Recurrence 0.012200   0.999  2.67              SGK3
91                Q16625 Recurrence 0.012700   0.999  2.66              OCLN
92                Q96AY4 Recurrence 0.012700   0.999  2.66             TTC28
93                P28070 Recurrence 0.012900   0.999 -2.65             PSMB4
94            A0A0B4J1V2 Recurrence 0.013000   0.999 -2.65          IGHV2-26
95                O15228 Recurrence 0.013100   0.999  2.64             GNPAT
96                P05546 Recurrence 0.013100   0.999 -2.64          SERPIND1
97                Q9H9F9 Recurrence 0.013200   0.999  2.64             ACTR5
98                Q86TV6 Recurrence 0.013300   0.999 -2.64             TTC7B
99                Q9NTI5 Recurrence 0.013500   0.999 -2.63             PDS5B
100               P21246 Recurrence 0.013700   0.999 -2.62               PTN
101               Q9BXI6 Recurrence 0.014200   0.999  2.61          TBC1D10A
102               Q9NS73 Recurrence 0.014600   0.999  2.60              MBIP
103               Q8NFH3 Recurrence 0.014800   0.999 -2.59             NUP43
104               P0C0L4 Recurrence 0.015000   0.999 -2.59               C4A
105               Q14789 Recurrence 0.015100   0.999 -2.58            GOLGB1
106               Q03591 Recurrence 0.015200   0.999 -2.58             CFHR1
107               Q8N755 Recurrence 0.015300   0.999  2.58           SLC66A3
108               Q92530 Recurrence 0.015600   0.999  2.57             PSMF1
109               P04180 Recurrence 0.015700   0.999 -2.57              LCAT
110               Q16549 Recurrence 0.015700   0.999 -2.57             PCSK7
111               Q9BQL6 Recurrence 0.015800   0.999  2.56            FERMT1
112               Q7Z3B1 Recurrence 0.016000   0.999 -2.56             NEGR1
113               Q53H12 Recurrence 0.016300   0.999  2.55               AGK
114               Q9UHR6 Recurrence 0.016300   0.999 -2.55            ZNHIT2
115               P49189 Recurrence 0.016800   0.999 -2.54           ALDH9A1
116               P80108 Recurrence 0.016800   0.999 -2.54             GPLD1
117               Q8IXT5 Recurrence 0.016800   0.999  2.54            RBM12B
118               Q9P270 Recurrence 0.017000   0.999  2.53            SLAIN2
119               Q92572 Recurrence 0.017100   0.999  2.53             AP3S1
120               O43731 Recurrence 0.017300   0.999  2.52            KDELR3
121               Q6KCM7 Recurrence 0.017300   0.999  2.53          SLC25A25
122               Q96KP4 Recurrence 0.017500   0.999 -2.52             CNDP2
123               Q9Y6X8 Recurrence 0.017500   0.999 -2.52              ZHX2
124               Q92878 Recurrence 0.017600   0.999  2.52             RAD50
125               Q15493 Recurrence 0.017900   0.999 -2.51               RGN
126               Q9BXN1 Recurrence 0.018100   0.999  2.51              ASPN
127               Q9NYH9 Recurrence 0.018200   0.999  2.50              UTP6
128               O43252 Recurrence 0.018300   0.999  2.50            PAPSS1
129               Q8WUN7 Recurrence 0.018500   0.999 -2.49             UBTD2
130               P02654 Recurrence 0.018700   0.999 -2.49             APOC1
131               Q9UHL4 Recurrence 0.018800   0.999 -2.49              DPP7
132               P04075 Recurrence 0.018900   0.999 -2.49             ALDOA
133               P49720 Recurrence 0.018900   0.999 -2.49             PSMB3
134               P54578 Recurrence 0.019000   0.999 -2.49             USP14
135               Q14624 Recurrence 0.019100   0.999 -2.48             ITIH4
136               P29558 Recurrence 0.019400   0.999  2.48             RBMS1
137               Q15386 Recurrence 0.019700   0.999  2.47             UBE3C
138               O95807 Recurrence 0.019900   0.999  2.46           TMEM50A
139               P01019 Recurrence 0.020000   0.999 -2.46               AGT
140               P20036 Recurrence 0.020000   0.999 -2.46          HLA-DPA1
141               Q9H936 Recurrence 0.020000   0.999  2.46          SLC25A22
142               Q14331 Recurrence 0.020200   0.999 -2.46              FRG1
143               P05186 Recurrence 0.020700   0.999 -2.45              ALPL
144               P04114 Recurrence 0.020800   0.999 -2.45              APOB
145               Q03252 Recurrence 0.020800   0.999  2.44             LMNB2
146               O75843 Recurrence 0.020900   0.999  2.44             AP1G2
147               Q70J99 Recurrence 0.021100   0.999  2.44            UNC13D
148               Q8WUY3 Recurrence 0.021100   0.999 -2.44            PRUNE2
149               P56589 Recurrence 0.021200   0.999  2.44              PEX3
150               P35573 Recurrence 0.021300   0.999  2.43               AGL
151               Q330K2 Recurrence 0.021400   0.999  2.43           NDUFAF6
152               Q9HCK4 Recurrence 0.021400   0.999 -2.43             ROBO2
153               O14493 Recurrence 0.021500   0.999  2.43             CLDN4
154               Q14165 Recurrence 0.021800   0.999 -2.42              MLEC
155               O95487 Recurrence 0.021900   0.999  2.42            SEC24B
156               Q96C90 Recurrence 0.022100   0.999  2.42          PPP1R14B
157               P11047 Recurrence 0.022200   0.999  2.42             LAMC1
158               P25815 Recurrence 0.022200   0.999  2.42             S100P
159               Q9H6S3 Recurrence 0.022300   0.999  2.42            EPS8L2
160               Q9NP99 Recurrence 0.022300   0.999  2.41             TREM1
161               O00186 Recurrence 0.022400   0.999  2.41            STXBP3
162               P60891 Recurrence 0.022400   0.999 -2.41             PRPS1
163               P40189 Recurrence 0.022600   0.999 -2.41             IL6ST
164               O95696 Recurrence 0.022700   0.999 -2.41              BRD1
165               P51689 Recurrence 0.022700   0.999 -2.41              ARSD
166               P56199 Recurrence 0.022700   0.999 -2.41             ITGA1
167               O15234 Recurrence 0.022900   0.999 -2.40             CASC3
168               P00450 Recurrence 0.023000   0.999 -2.40                CP
169               P43251 Recurrence 0.023000   0.999 -2.40               BTD
170               Q9BY41 Recurrence 0.023100   0.999 -2.40             HDAC8
171               Q99615 Recurrence 0.023300   0.999 -2.39            DNAJC7
172               Q9H9S4 Recurrence 0.023700   0.999 -2.39            CAB39L
173               P56524 Recurrence 0.024100   0.999 -2.38             HDAC4
174               P01009 Recurrence 0.024200   0.999 -2.38          SERPINA1
175               Q96EU7 Recurrence 0.024200   0.999  2.38         C1GALT1C1
176               P06703 Recurrence 0.024500   0.999  2.37            S100A6
177               P20618 Recurrence 0.024500   0.999 -2.37             PSMB1
178               P30536 Recurrence 0.024500   0.999  2.37              TSPO
179               P46937 Recurrence 0.024700   0.999  2.37              YAP1
180               Q96N67 Recurrence 0.024800   0.999  2.37             DOCK7
181               Q99828 Recurrence 0.024800   0.999  2.37              CIB1
182               P01023 Recurrence 0.024900   0.999 -2.36               A2M
183               Q7L8L6 Recurrence 0.025000   0.999  2.36           FASTKD5
184               Q8N7R7 Recurrence 0.025000   0.999  2.36            CCNYL1
185               Q9BSY9 Recurrence 0.025100   0.999  2.36             DESI2
186               Q9UK45 Recurrence 0.025100   0.999  2.36              LSM7
187               P25391 Recurrence 0.025200   0.999 -2.36             LAMA1
188               P25787 Recurrence 0.025200   0.999 -2.36             PSMA2
189               Q08174 Recurrence 0.025500   0.999 -2.35             PCDH1
190               A3KMH1 Recurrence 0.025800   0.999 -2.35              VWA8
191               Q7L1Q6 Recurrence 0.025800   0.999 -2.35              BZW1
192               Q7Z3D4 Recurrence 0.025900   0.999  2.35            LYSMD3
193               P28907 Recurrence 0.026600   0.999 -2.34              CD38
194               P06748 Recurrence 0.027000   0.999 -2.33              NPM1
195               Q9NUL5 Recurrence 0.027000   0.999  2.33              SHFL
196               P61160 Recurrence 0.027100   0.999  2.33             ACTR2
197               Q96HC4 Recurrence 0.027400   0.999  2.32            PDLIM5
198 P0DTE7;P0DTE8;P0DUB6 Recurrence 0.027500   0.999  2.32 AMY1B;AMY1C;AMY1A
199               Q86SQ0 Recurrence 0.027500   0.999 -2.32            PHLDB2
200               Q16537 Recurrence 0.027700   0.999  2.32           PPP2R5E
201               P30043 Recurrence 0.027800   0.999 -2.32             BLVRB
202               P02656 Recurrence 0.028100   0.999 -2.31             APOC3
203               P24752 Recurrence 0.028200   0.999  2.31             ACAT1
204               Q9BRP4 Recurrence 0.028300   0.999 -2.31             PAAF1
205               P53007 Recurrence 0.028500   0.999  2.31           SLC25A1
206               P25786 Recurrence 0.028700   0.999 -2.30             PSMA1
207               Q9UBK9 Recurrence 0.028800   0.999  2.30               UXT
208               P28062 Recurrence 0.029100   0.999 -2.30             PSMB8
209               Q99549 Recurrence 0.029100   0.999 -2.30          MPHOSPH8
210               A6NDB9 Recurrence 0.029400   0.999  2.29             PALM3
211               Q9Y305 Recurrence 0.029400   0.999  2.29             ACOT9
212               Q5T5C0 Recurrence 0.029500   0.999  2.29            STXBP5
213               Q9UBS4 Recurrence 0.029500   0.999 -2.29           DNAJB11
214               Q04656 Recurrence 0.029600   0.999 -2.29             ATP7A
215               Q9UIA9 Recurrence 0.029600   0.999  2.29              XPO7
216               P25789 Recurrence 0.029800   0.999 -2.28             PSMA4
217               Q03701 Recurrence 0.030000   0.999  2.28             CEBPZ
218               P06744 Recurrence 0.030300   0.999 -2.28               GPI
219               Q4G0P3 Recurrence 0.030300   0.999  2.28             HYDIN
220               P40306 Recurrence 0.030400   0.999 -2.28            PSMB10
221               Q8IZP0 Recurrence 0.030400   0.999  2.28              ABI1
222               O75112 Recurrence 0.030600   0.999 -2.27              LDB3
223               P60660 Recurrence 0.030600   0.999  2.27              MYL6
224               Q4J6C6 Recurrence 0.030800   0.999 -2.27             PREPL
225               Q6ZNB6 Recurrence 0.031100   0.999  2.27             NFXL1
226               Q8IWY8 Recurrence 0.031200   0.999  2.26           ZSCAN29
227               Q99614 Recurrence 0.031200   0.999 -2.26              TTC1
228               P02647 Recurrence 0.031400   0.999 -2.26             APOA1
229               Q5XPI4 Recurrence 0.031800   0.999 -2.26            RNF123
230               Q7KZ85 Recurrence 0.031900   0.999 -2.25            SUPT6H
231               Q8TBZ3 Recurrence 0.032300   0.999  2.25             WDR20
232               Q96J02 Recurrence 0.032300   0.999  2.25              ITCH
233               Q99988 Recurrence 0.033200   0.999  2.24             GDF15
234               Q9Y2Q3 Recurrence 0.033300   0.999  2.23             GSTK1
235               O75896 Recurrence 0.033500   0.999  2.23             TUSC2
236               Q9ULP9 Recurrence 0.033700   0.999  2.23           TBC1D24
237               Q53SF7 Recurrence 0.033800   0.999  2.23            COBLL1
238               Q8NEF9 Recurrence 0.033900   0.999 -2.23            SRFBP1
239               Q9Y2G2 Recurrence 0.034000   0.999 -2.22             CARD8
240               O75071 Recurrence 0.034100   0.999 -2.22           EFCAB14
241               P30613 Recurrence 0.034100   0.999 -2.22              PKLR
242               P52272 Recurrence 0.034100   0.999  2.22            HNRNPM
243               Q9Y2R5 Recurrence 0.034200   0.999  2.22            MRPS17
244               Q15063 Recurrence 0.034300   0.999  2.22             POSTN
245               Q16718 Recurrence 0.034400   0.999  2.22            NDUFA5
246               O95858 Recurrence 0.035200   0.999  2.21           TSPAN15
247               Q8TAE6 Recurrence 0.035300   0.999  2.21          PPP1R14C
248               P85037 Recurrence 0.035400   0.999  2.21             FOXK1
249               Q9H307 Recurrence 0.035400   0.999 -2.21               PNN
250               P04406 Recurrence 0.035600   0.999 -2.20             GAPDH
251               P11172 Recurrence 0.035600   0.999 -2.20              UMPS
252               Q13884 Recurrence 0.035900   0.999  2.20             SNTB1
253               Q96P70 Recurrence 0.035900   0.999 -2.20              IPO9
254               Q8TER0 Recurrence 0.036300   0.999  2.20             SNED1
255               Q9NXN4 Recurrence 0.036400   0.999  2.19             GDAP2
256               Q14678 Recurrence 0.036600   0.999 -2.19             KANK1
257               Q99538 Recurrence 0.036900   0.999  2.19              LGMN
258               P0C626 Recurrence 0.037200   0.999  2.18             OR5G3
259               P61326 Recurrence 0.037200   0.999  2.18             MAGOH
260               Q13214 Recurrence 0.037400   0.999 -2.18            SEMA3B
261               Q9Y575 Recurrence 0.037800   0.999 -2.18              ASB3
262               Q9Y250 Recurrence 0.037900   0.999 -2.17             LZTS1
263               Q8WXA9 Recurrence 0.038100   0.999 -2.17             SREK1
264               Q14160 Recurrence 0.038300   0.999  2.17             SCRIB
265               O00522 Recurrence 0.038400   0.999  2.17             KRIT1
266               Q9Y4F3 Recurrence 0.038500   0.999 -2.17             MARF1
267               P03952 Recurrence 0.038600   0.999 -2.17             KLKB1
268               P24941 Recurrence 0.038600   0.999 -2.17              CDK2
269               Q9GZT4 Recurrence 0.038700   0.999 -2.17               SRR
270               O15382 Recurrence 0.038800   0.999  2.16             BCAT2
271               O75608 Recurrence 0.038800   0.999  2.16            LYPLA1
272               O14818 Recurrence 0.039000   0.999 -2.16             PSMA7
273               Q96E39 Recurrence 0.039200   0.999  2.16            RBMXL1
274               Q9NUM3 Recurrence 0.039200   0.999 -2.16           SLC39A9
275               P04233 Recurrence 0.039400   0.999 -2.16              CD74
276               P20718 Recurrence 0.039400   0.999 -2.16              GZMH
277               Q8TB73 Recurrence 0.039500   0.999 -2.16              NDNF
278               Q6IA69 Recurrence 0.039600   0.999  2.16           NADSYN1
279               Q969X1 Recurrence 0.039900   0.999  2.15            TMBIM1
280               O75185 Recurrence 0.040000   0.999  2.15            ATP2C2
281               P15907 Recurrence 0.040000   0.999 -2.15           ST6GAL1
282               Q9P2L0 Recurrence 0.040200   0.999 -2.15             WDR35
283               Q9Y5X1 Recurrence 0.040300   0.999  2.15              SNX9
284               Q9H1A4 Recurrence 0.040400   0.999 -2.15            ANAPC1
285               Q96HP4 Recurrence 0.040500   0.999  2.14            OXNAD1
286               Q9BZW5 Recurrence 0.040500   0.999  2.14            TM6SF1
287               P62191 Recurrence 0.040600   0.999 -2.14             PSMC1
288               Q8N5G0 Recurrence 0.040600   0.999  2.14            SMIM20
289               Q92729 Recurrence 0.040600   0.999  2.14             PTPRU
290               P49959 Recurrence 0.040800   0.999  2.14             MRE11
291               Q9UJY1 Recurrence 0.041100   0.999  2.14             HSPB8
292               Q9UDR5 Recurrence 0.041300   0.999  2.13              AASS
293               O96019 Recurrence 0.041500   0.999 -2.13            ACTL6A
294               Q02985 Recurrence 0.041900   0.999 -2.13             CFHR3
295               Q04837 Recurrence 0.042300   0.999  2.12             SSBP1
296               Q86TH1 Recurrence 0.042300   0.999 -2.12          ADAMTSL2
297               Q6P1M3 Recurrence 0.042400   0.999  2.12             LLGL2
298               P05155 Recurrence 0.043100   0.999 -2.12          SERPING1
299               Q8N2G8 Recurrence 0.043100   0.999 -2.11              GHDC
300               Q9NZN5 Recurrence 0.043400   0.999  2.11          ARHGEF12
301               Q6ZS30 Recurrence 0.043500   0.999  2.11            NBEAL1
302               Q9Y3I0 Recurrence 0.043500   0.999 -2.11              RTCB
303               Q9Y2D2 Recurrence 0.043800   0.999  2.11           SLC35A3
304               P02655 Recurrence 0.043900   0.999 -2.11             APOC2
305               Q99487 Recurrence 0.043900   0.999  2.11            PAFAH2
306               Q92934 Recurrence 0.044000   0.999  2.11               BAD
307               Q09028 Recurrence 0.044200   0.999 -2.10             RBBP4
308               Q9BYG5 Recurrence 0.044200   0.999  2.10            PARD6B
309               Q86YT9 Recurrence 0.044300   0.999  2.10              JAML
310               Q6NUM9 Recurrence 0.044600   0.999  2.10            RETSAT
311               P40692 Recurrence 0.044700   0.999 -2.10              MLH1
312               Q99611 Recurrence 0.044800   0.999  2.10            SEPHS2
313               A0JNW5 Recurrence 0.044900   0.999 -2.10            BLTP3B
314               P50570 Recurrence 0.045000   0.999  2.10              DNM2
315               Q9ULH7 Recurrence 0.045000   0.999 -2.09             MRTFB
316               Q9Y365 Recurrence 0.045100   0.999  2.09           STARD10
317               Q96P16 Recurrence 0.045400   0.999 -2.09            RPRD1A
318               P11908 Recurrence 0.045500   0.999 -2.09             PRPS2
319               P23921 Recurrence 0.045700   0.999 -2.09              RRM1
320               Q8IWP9 Recurrence 0.045700   0.999 -2.09           CCDC28A
321               Q9H0X4 Recurrence 0.045700   0.999 -2.09           FAM234A
322               Q9UQN3 Recurrence 0.045800   0.999  2.09            CHMP2B
323               Q8IWE5 Recurrence 0.045900   0.999  2.09           PLEKHM2
324               Q9NRY5 Recurrence 0.045900   0.999 -2.09          FAM114A2
325               O00170 Recurrence 0.046200   0.999  2.08               AIP
326               Q9BSE4 Recurrence 0.046200   0.999  2.08           HERPUD2
327               Q9Y6Y1 Recurrence 0.046600   0.999 -2.08            CAMTA1
328               Q7LBR1 Recurrence 0.046700   0.999 -2.08            CHMP1B
329               Q15046 Recurrence 0.046800   0.999 -2.08             KARS1
330               P40123 Recurrence 0.047000   0.999 -2.07              CAP2
331               Q9NRY6 Recurrence 0.047000   0.999  2.07            PLSCR3
332               O76076 Recurrence 0.047100   0.999  2.07              CCN5
333               P52848 Recurrence 0.047200   0.999 -2.07             NDST1
334               Q9HBU6 Recurrence 0.047200   0.999  2.07             ETNK1
335               P17174 Recurrence 0.047300   0.999 -2.07              GOT1
336               Q9NNW5 Recurrence 0.047300   0.999 -2.07              WDR6
337               P40818 Recurrence 0.047700   0.999 -2.07              USP8
338               Q13393 Recurrence 0.048000   0.999  2.06              PLD1
339               O60232 Recurrence 0.048300   0.999  2.06             ZNRD2
340               P51153 Recurrence 0.048600   0.999 -2.06             RAB13
341               Q07812 Recurrence 0.048700   0.999  2.06               BAX
342               Q96S97 Recurrence 0.048700   0.999  2.06             MYADM
343               Q76FK4 Recurrence 0.048800   0.999 -2.06              NOL8
344               Q8NBI5 Recurrence 0.048900   0.999  2.06           SLC43A3
345               O43157 Recurrence 0.049000   0.999 -2.05            PLXNB1
346               Q8NCU1 Recurrence 0.049000   0.999  2.05           CCDC197
347               Q92692 Recurrence 0.049000   0.999 -2.05           NECTIN2
348               Q9UM54 Recurrence 0.049100   0.999  2.05              MYO6
349               Q96FJ0 Recurrence 0.049400   0.999 -2.05          STAMBPL1
350               P09104 Recurrence 0.049800   0.999 -2.05              ENO2
351               Q9BXT4 Recurrence 0.049900   0.999 -2.05             TDRD1</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proTumorRes_Klin$sig.feat.heat</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Tumor samples (ns: p
&gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Klin$data
smpAnnoTab &lt;- proTissueRes_Klin$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proTumorRes_Klin$sig.feat.tab
num_sigFeats &lt;- 6
gpLevel = c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;, &#39;Yes_Normal&#39;, &#39;No_Normal&#39;)
condiCol = c(&#39;grey50&#39;, &#39;firebrick&#39;) #c(Normal, Tumor)
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(newVar1 = paste0(Var1, &#39; (&#39;, Gene, &#39;)&#39;),
                newVar1 = factor(newVar1, levels = unique(newVar1)))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Recur_Condi = paste0(Recurrence, &#39;_&#39;, Condition),
                Recur_Condi = factor(Recur_Condi, levels = gpLevel))
ggplot(topSigFeatDat, aes(x=Recur_Condi, y=Abundance, col=Condition, fill=Recurrence)) +
  geom_boxplot(alpha = 1, outlier.shape = NA, linewidth = 1) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             comparisons = list(c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;), c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;)),
                             label = &#39;p.signif&#39;, tip.length = 0.015, bracket.size = 0.7, size = 4) +
  labs(x = &#39;Recurrence&#39;) +
  scale_color_manual(values = condiCol) +
  scale_fill_manual(values = c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(newVar1), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;),
             axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proTumorRes_Klin$pval.hist</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2   pVal pValAdj  Stat   Test
1 PC19 (1.8%) Recurrence 0.0102   0.294 -2.71 T-test
2 PC34 (1.1%) Recurrence 0.0159   0.294 -2.53 T-test</code></pre>
<p>Visualize significant PCs and display their top features with
greatest absolute loadings</p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.dist$PC19</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Klin$pc.top.feat.tab$PC19</code></pre>
<pre><code>         Feature Loading         Gene
1         O15550 -0.0985        KDM6A
2         P52895 -0.0802       AKR1C2
3         P01704 -0.0763     IGLV2-14
4         P20142 -0.0762          PGC
5         P69892 -0.0758         HBG2
6  P0CG29;P0CG30 -0.0742 GSTT2;GSTT2B
7         P29034 -0.0690       S100A2
8         Q99758  0.0623        ABCA3
9         P79483  0.0617     HLA-DRB3
10        Q15223 -0.0609      NECTIN1
11        Q6ZT12 -0.0569         UBR3
12        P56277 -0.0547         CMC4
13        Q96S86 -0.0543       HAPLN3
14        P69891 -0.0534         HBG1
15        P15428 -0.0530         HPGD
16        P11686 -0.0530        SFTPC
17        P13762  0.0503     HLA-DRB4
18        P01909  0.0499     HLA-DQA1
19        Q9H8P0  0.0496       SRD5A3
20        Q5T8I9 -0.0496       HENMT1
21        Q96MX0  0.0493        CMTM3
22        Q30154  0.0492     HLA-DRB5
23        Q4G0P3  0.0489        HYDIN
24        P14091 -0.0479         CTSE
25        P47895 -0.0474      ALDH1A3
26        Q9NRX3 -0.0470     NDUFA4L2
27        P01040 -0.0468         CSTA
28        Q03013  0.0465        GSTM4
29        P42771  0.0464       CDKN2A
30        O75635 -0.0463     SERPINB7</code></pre>
<pre class="r"><code>proTumorRes_Klin$sig.pc.dist$PC34</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-33-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Klin$pc.top.feat.tab$PC34</code></pre>
<pre><code>      Feature Loading     Gene
1      Q9H799 -0.0786  CPLANE1
2      P01704 -0.0656 IGLV2-14
3      P11684 -0.0621  SCGB1A1
4  A0A087WSZ0  0.0617 IGKV1D-8
5      Q9Y6R7 -0.0602    FCGBP
6      Q9BXJ4  0.0597  C1QTNF3
7      Q9UKT5  0.0595    FBXO4
8      P16112 -0.0594     ACAN
9      P55064 -0.0588     AQP5
10     P12074  0.0577   COX6A1
11     P09488 -0.0563    GSTM1
12     Q9BQB6 -0.0560   VKORC1
13     P30838 -0.0558  ALDH3A1
14     P22413  0.0552    ENPP1
15     Q9BY67 -0.0541    CADM1
16     Q9Y2R5 -0.0538   MRPS17
17     Q9NZT1  0.0535   CALML5
18     Q6P1A2 -0.0521   LPCAT3
19     P40199  0.0518  CEACAM6
20     Q16570  0.0517    ACKR1
21 A0A0B4J1Y9 -0.0517 IGHV3-72
22     P15090  0.0517    FABP4
23     P01909 -0.0513 HLA-DQA1
24     O75635  0.0512 SERPINB7
25     P50226  0.0506  SULT1A2
26     P60602 -0.0506    ROMO1
27     A0M8Q6  0.0505    IGLC7
28 A0A0B4J1Y8 -0.0504 IGLV9-49
29     Q8N465 -0.0504   D2HGDH
30     P16050  0.0503   ALOX15</code></pre>
</div>
<div id="normal-samples-1" class="section level2">
<h2>Normal samples</h2>
<p>Display significant associations between SVs and sample metadata
variables. SVs (surrogate variables) will be included in linear model
for differential analysis</p>
<pre class="r"><code># Subset certain samples
proNormal_Klin &lt;- proTissue_Klin[, which(colData(proTissue_Klin)$Condition == &#39;Normal&#39;)]

# Do SVA to identify latent factors that explain unknown variance
# Prepare data matrix and sample metadata table for creating model matrices
datMat &lt;- assay(proNormal_Klin)
# Keep only complete features
keptFeats &lt;- apply(datMat, 1, function(featVec) {
  !any(is.na(featVec))
})
datMat &lt;- datMat[keptFeats,]
metadatTab &lt;- colData(proNormal_Klin) %&gt;%
  tibble::as_tibble(rownames = &#39;Sample&#39;)
# Create full model matrix including both adjustment variables and variables of interest
modelMat &lt;- model.matrix(~ Recurrence, data = metadatTab)
# Create null model matrix including only adjustment variables
modelMat0 &lt;- model.matrix(~ 1, data = metadatTab)

# Perform SVA
# Identify number of latent factors that need to be estimated
numSV = sva::num.sv(datMat, modelMat, method = &#39;leek&#39;)
svObj = sva(datMat, modelMat, modelMat0, n.sv = numSV)</code></pre>
<pre><code>Number of significant surrogate variables is:  1 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code># Do association tests between SVs and patient metadata
# Prepare surrogate variable table
svTab &lt;- svObj$sv
rownames(svTab) &lt;- metadatTab$Sample
colnames(svTab) &lt;- paste0(&#39;SV&#39;, seq_len(ncol(svTab)))
svTab &lt;- tibble::as_tibble(svTab, rownames = &#39;Sample&#39;)
# Prepare patient metadata table
metaTab &lt;- dplyr::select(metadatTab, Sample, Recurrence, Gender, Age, Smoking, Stage,
                         Adjuvant)
# Do association tests
testAsso(svTab, metaTab, cmn_col = &#39;Sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05)</code></pre>
<pre><code>  Var1  Var2       pVal   pValAdj     Stat  Test
1  SV1 Stage 0.04131006 0.2478604 3.486135 ANOVA</code></pre>
<pre class="r"><code># Include surrogate variables into SE object
proNormalTab &lt;- summExp2df(proNormal_Klin, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;)
proNormalTab &lt;- dplyr::left_join(proNormalTab, svTab, by = &#39;Sample&#39;)
proNormal_Klin &lt;- df2SummExp(proNormalTab, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;,
                             values = &#39;Value&#39;, row_anno = colnames(rowData(proNormal_Klin)),
                             col_anno = c(colnames(colData(proNormal_Klin)),
                                          paste0(&#39;SV&#39;, seq_len(ncol(svTab)-1))))
assay(proNormal_Klin) &lt;- as.matrix(assay(proNormal_Klin))
# SVs will be incorporated into our linear model as covariates</code></pre>
<pre class="r"><code># Perform single-omics analysis
# proNormalRes_Klin &lt;- doSingleOmicsAnalysis(proNormal_Klin, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                            pca_method = &#39;ppca&#39;, num_PCs = 37, num_PCfeats = 30,
#                                            use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                            feat_conv = T, plot_title = NULL, show_rownames = F,
#                                            pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                            &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;),
#                                            soa_unwantVar = &#39;SV1&#39;)
                                           # fontsize = 15, fontsize_col = 14
proNormalRes_Klin &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proNormalRes_Klin_SV1.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proNormalRes_Klin$data.var.source</code></pre>
<pre><code>          Var1       Var2    pVal pValAdj   Stat        Test
1  PC33 (1.2%)      Stage 0.00102   0.117  8.390       ANOVA
2     PC7 (4%)     Gender 0.00106   0.117  3.550      T-test
3   PC8 (3.6%)      Stage 0.00332   0.245  6.720       ANOVA
4  PC1 (11.1%) Recurrence 0.00485   0.269 -3.000      T-test
5  PC12 (2.7%)     Gender 0.00767   0.341  2.820      T-test
6   PC9 (3.2%) Recurrence 0.02690   0.775 -2.300      T-test
7  PC21 (1.8%)      Stage 0.02860   0.775  3.930       ANOVA
8   PC5 (4.6%)        Age 0.02920   0.775 -0.349 Correlation
9  PC29 (1.4%)    Smoking 0.03140   0.775  3.820       ANOVA
10  PC3 (6.3%)   Adjuvant 0.03790   0.778  2.150      T-test
11   PC19 (2%)        Age 0.04300   0.778 -0.326 Correlation
12 PC13 (2.6%)        Age 0.04530   0.778 -0.322 Correlation</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>dplyr::select(proNormalRes_Klin$sig.feat.tab, -Test)</code></pre>
<pre><code>              Var1       Var2     pVal pValAdj  Stat            Gene
1           P41208 Recurrence 7.69e-06  0.0629 -5.22           CETN2
2           Q9NSC5 Recurrence 1.52e-05  0.0629 -5.00          HOMER3
3           P63104 Recurrence 3.07e-05  0.0678 -4.77           YWHAZ
4           Q9NVZ3 Recurrence 4.93e-05  0.0678  4.61          NECAP2
5           Q99417 Recurrence 5.82e-05  0.0678 -4.55           MYCBP
6           Q9H9H4 Recurrence 6.13e-05  0.0678 -4.54          VPS37B
7           Q9Y512 Recurrence 6.57e-05  0.0678  4.51          SAMM50
8           P11441 Recurrence 7.62e-05  0.0678 -4.46           UBL4A
9           P07951 Recurrence 8.44e-05  0.0678 -4.43            TPM2
10          P09493 Recurrence 9.82e-05  0.0678 -4.38            TPM1
11          Q53S58 Recurrence 1.11e-04  0.0678  4.34         TMEM177
12          Q9H5N1 Recurrence 1.15e-04  0.0678 -4.33          RABEP2
13          Q9UG63 Recurrence 1.18e-04  0.0678  4.32           ABCF2
14          Q5S007 Recurrence 1.23e-04  0.0678  4.30           LRRK2
15          Q9Y6Y8 Recurrence 1.23e-04  0.0678  4.30         SEC23IP
16          O94855 Recurrence 1.55e-04  0.0798  4.23          SEC24D
17          Q13561 Recurrence 1.89e-04  0.0828 -4.16           DCTN2
18          P55036 Recurrence 2.03e-04  0.0828 -4.13           PSMD4
19          Q13190 Recurrence 2.03e-04  0.0828  4.13            STX5
20          Q96E40 Recurrence 2.07e-04  0.0828 -4.13          SPACA9
21          Q9NQP4 Recurrence 2.32e-04  0.0828 -4.09           PFDN4
22          Q05823 Recurrence 2.36e-04  0.0828  4.08          RNASEL
23          Q9Y608 Recurrence 2.37e-04  0.0828 -4.08         LRRFIP2
24          P02511 Recurrence 2.51e-04  0.0828 -4.06           CRYAB
25          P84243 Recurrence 2.51e-04  0.0828 -4.06           H3-3B
26          Q13464 Recurrence 2.88e-04  0.0877 -4.02           ROCK1
27          Q14137 Recurrence 3.01e-04  0.0877  4.00            BOP1
28          P31946 Recurrence 3.08e-04  0.0877 -3.99           YWHAB
29          Q9Y5X2 Recurrence 3.14e-04  0.0877  3.99            SNX8
30          O95865 Recurrence 3.19e-04  0.0877 -3.98           DDAH2
31          Q12923 Recurrence 3.81e-04  0.1010  3.92          PTPN13
32          Q96LZ7 Recurrence 4.34e-04  0.1010  3.87           RMDN2
33          Q9NXJ5 Recurrence 4.48e-04  0.1010  3.86          PGPEP1
34          Q99816 Recurrence 4.80e-04  0.1010 -3.84          TSG101
35          P30622 Recurrence 4.87e-04  0.1010 -3.83           CLIP1
36          Q9Y281 Recurrence 5.34e-04  0.1010 -3.80            CFL2
37          Q5SY16 Recurrence 5.40e-04  0.1010  3.80            NOL9
38          Q6P9B9 Recurrence 5.49e-04  0.1010  3.79           INTS5
39          P51911 Recurrence 5.70e-04  0.1010 -3.78            CNN1
40          P21953 Recurrence 5.71e-04  0.1010  3.78          BCKDHB
41          Q8NBQ5 Recurrence 5.72e-04  0.1010  3.78        HSD17B11
42          P67936 Recurrence 6.06e-04  0.1010 -3.76            TPM4
43          Q8IWP9 Recurrence 6.15e-04  0.1010 -3.75         CCDC28A
44          Q9H936 Recurrence 6.26e-04  0.1010  3.75        SLC25A22
45          Q14118 Recurrence 6.33e-04  0.1010 -3.74            DAG1
46          Q9UNE7 Recurrence 6.44e-04  0.1010 -3.74           STUB1
47          P15311 Recurrence 6.49e-04  0.1010 -3.73             EZR
48          Q96C36 Recurrence 6.54e-04  0.1010  3.73           PYCR2
49          O00233 Recurrence 6.59e-04  0.1010 -3.73           PSMD9
50          P62875 Recurrence 6.74e-04  0.1010  3.72          POLR2L
51          Q5TZA2 Recurrence 6.80e-04  0.1010 -3.72           CROCC
52          O75818 Recurrence 6.96e-04  0.1010  3.71           RPP40
53          Q14203 Recurrence 6.97e-04  0.1010 -3.71           DCTN1
54          Q96CV9 Recurrence 7.03e-04  0.1010 -3.71            OPTN
55          Q96P48 Recurrence 7.15e-04  0.1010  3.70           ARAP1
56          Q8NI36 Recurrence 7.19e-04  0.1010  3.70           WDR36
57          P30049 Recurrence 7.45e-04  0.1010 -3.69         ATP5F1D
58          Q8N131 Recurrence 7.49e-04  0.1010 -3.68         TMEM123
59          Q92851 Recurrence 7.64e-04  0.1010  3.68          CASP10
60          P06753 Recurrence 7.69e-04  0.1010 -3.67            TPM3
61          O76024 Recurrence 7.70e-04  0.1010 -3.67            WFS1
62          Q15036 Recurrence 7.87e-04  0.1010  3.67           SNX17
63          Q5M775 Recurrence 7.88e-04  0.1010 -3.67          SPECC1
64          P07205 Recurrence 7.91e-04  0.1010  3.67            PGK2
65          Q9Y3A3 Recurrence 7.94e-04  0.1010  3.66            MOB4
66          Q9UJ83 Recurrence 8.61e-04  0.1010  3.64           HACL1
67          Q05682 Recurrence 8.65e-04  0.1010 -3.63           CALD1
68          P62258 Recurrence 8.66e-04  0.1010 -3.63           YWHAE
69          Q330K2 Recurrence 8.99e-04  0.1010  3.62         NDUFAF6
70          Q13409 Recurrence 9.00e-04  0.1010 -3.62         DYNC1I2
71          Q86UP2 Recurrence 9.05e-04  0.1010 -3.62            KTN1
72          O43556 Recurrence 9.16e-04  0.1010 -3.61            SGCE
73          Q96DI7 Recurrence 9.24e-04  0.1010  3.61         SNRNP40
74          Q14BN4 Recurrence 9.32e-04  0.1010 -3.61           SLMAP
75          P12277 Recurrence 9.34e-04  0.1010 -3.61             CKB
76          Q8WZA0 Recurrence 9.59e-04  0.1010 -3.60            LZIC
77          P17302 Recurrence 9.72e-04  0.1010 -3.59            GJA1
78          P52789 Recurrence 9.81e-04  0.1010  3.59             HK2
79          Q9UN79 Recurrence 9.86e-04  0.1010 -3.59           SOX13
80          Q10570 Recurrence 9.89e-04  0.1010  3.59           CPSF1
81          P50502 Recurrence 9.92e-04  0.1010 -3.59            ST13
82          P32321 Recurrence 1.02e-03  0.1010  3.57            DCTD
83          Q9H7X7 Recurrence 1.02e-03  0.1010 -3.58           IFT22
84          P18206 Recurrence 1.03e-03  0.1020 -3.57             VCL
85          Q9H8L6 Recurrence 1.08e-03  0.1030 -3.55           MMRN2
86          O60762 Recurrence 1.09e-03  0.1030  3.55            DPM1
87          Q9Y5Z4 Recurrence 1.09e-03  0.1030 -3.55           HEBP2
88          Q8IXQ6 Recurrence 1.10e-03  0.1030  3.55           PARP9
89          P61978 Recurrence 1.13e-03  0.1030 -3.54          HNRNPK
90          Q08554 Recurrence 1.14e-03  0.1030 -3.54            DSC1
91          Q69YQ0 Recurrence 1.14e-03  0.1030 -3.54         SPECC1L
92          Q5SNT2 Recurrence 1.15e-03  0.1030  3.53         TMEM201
93          P61966 Recurrence 1.16e-03  0.1030  3.53           AP1S1
94          O60760 Recurrence 1.20e-03  0.1030  3.52           HPGDS
95          P02533 Recurrence 1.22e-03  0.1030 -3.51           KRT14
96          Q99497 Recurrence 1.23e-03  0.1030 -3.51           PARK7
97          Q6IAN0 Recurrence 1.24e-03  0.1030  3.51          DHRS7B
98          Q8TEB9 Recurrence 1.25e-03  0.1030 -3.50          RHBDD1
99          Q96BR1 Recurrence 1.26e-03  0.1030  3.50            SGK3
100         P08240 Recurrence 1.31e-03  0.1030  3.48           SRPRA
101         Q9BQ13 Recurrence 1.31e-03  0.1030  3.49          KCTD14
102         Q15404 Recurrence 1.32e-03  0.1030 -3.48            RSU1
103         Q76N32 Recurrence 1.33e-03  0.1030 -3.48           CEP68
104         P09917 Recurrence 1.34e-03  0.1030  3.48           ALOX5
105         P16949 Recurrence 1.34e-03  0.1030 -3.48           STMN1
106         O94925 Recurrence 1.35e-03  0.1030  3.47             GLS
107         Q6NXT1 Recurrence 1.35e-03  0.1030 -3.47         ANKRD54
108         P42167 Recurrence 1.37e-03  0.1030 -3.47            TMPO
109         O14972 Recurrence 1.39e-03  0.1030  3.46          VPS26C
110         Q6NY19 Recurrence 1.41e-03  0.1030 -3.46           KANK3
111         Q9BZL4 Recurrence 1.41e-03  0.1030 -3.46        PPP1R12C
112         O60941 Recurrence 1.42e-03  0.1030 -3.46            DTNB
113         P08294 Recurrence 1.44e-03  0.1030 -3.45            SOD3
114         Q9BXP2 Recurrence 1.44e-03  0.1030  3.45         SLC12A9
115         P19878 Recurrence 1.45e-03  0.1030  3.45            NCF2
116         Q14515 Recurrence 1.45e-03  0.1030 -3.45         SPARCL1
117         Q9Y490 Recurrence 1.47e-03  0.1040 -3.44            TLN1
118         Q9Y2T2 Recurrence 1.54e-03  0.1080  3.43           AP3M1
119         O75781 Recurrence 1.55e-03  0.1080 -3.42            PALM
120         O96008 Recurrence 1.57e-03  0.1080  3.42          TOMM40
121         O14974 Recurrence 1.63e-03  0.1110 -3.41        PPP1R12A
122         O95292 Recurrence 1.64e-03  0.1110 -3.40            VAPB
123         Q9BR39 Recurrence 1.66e-03  0.1110 -3.40            JPH2
124         Q9UH62 Recurrence 1.67e-03  0.1110  3.40          ARMCX3
125         Q9UPN3 Recurrence 1.70e-03  0.1120 -3.39           MACF1
126         P62937 Recurrence 1.71e-03  0.1120 -3.39            PPIA
127         Q96PE3 Recurrence 1.73e-03  0.1120  3.39          INPP4A
128         O60343 Recurrence 1.74e-03  0.1120  3.38          TBC1D4
129         Q8N1F8 Recurrence 1.77e-03  0.1130  3.38         STK11IP
130         Q9NP74 Recurrence 1.78e-03  0.1130 -3.38           PALMD
131         P16403 Recurrence 1.81e-03  0.1130 -3.37            H1-2
132         Q9H7B2 Recurrence 1.82e-03  0.1130  3.37            RPF2
133         Q969V6 Recurrence 1.83e-03  0.1130 -3.36           MRTFA
134         O14656 Recurrence 1.85e-03  0.1130  3.36           TOR1A
135         O60237 Recurrence 1.85e-03  0.1130 -3.36        PPP1R12B
136         Q9BQI0 Recurrence 1.87e-03  0.1140 -3.36           AIF1L
137         A1A4S6 Recurrence 1.92e-03  0.1150  3.35        ARHGAP10
138         Q6UWP8 Recurrence 1.93e-03  0.1150 -3.35            SBSN
139         P05141 Recurrence 1.94e-03  0.1150  3.34         SLC25A5
140         P02538 Recurrence 2.04e-03  0.1150 -3.33           KRT6A
141         P02649 Recurrence 2.04e-03  0.1150 -3.33            APOE
142         Q01995 Recurrence 2.04e-03  0.1150 -3.33           TAGLN
143         O75312 Recurrence 2.06e-03  0.1150 -3.32            ZPR1
144         Q15436 Recurrence 2.06e-03  0.1150  3.32          SEC23A
145         Q6UVK1 Recurrence 2.08e-03  0.1150 -3.32           CSPG4
146         P39656 Recurrence 2.09e-03  0.1150  3.32           DDOST
147         P53618 Recurrence 2.09e-03  0.1150  3.32           COPB1
148         P61956 Recurrence 2.09e-03  0.1150 -3.32           SUMO2
149         Q6KB66 Recurrence 2.10e-03  0.1150 -3.31           KRT80
150         Q9Y266 Recurrence 2.10e-03  0.1150 -3.32            NUDC
151         Q96EQ0 Recurrence 2.12e-03  0.1160 -3.31            SGTB
152         Q9Y6I3 Recurrence 2.16e-03  0.1170 -3.30            EPN1
153         P10620 Recurrence 2.18e-03  0.1180  3.30           MGST1
154         Q14257 Recurrence 2.20e-03  0.1180 -3.30            RCN2
155         Q9NXN4 Recurrence 2.23e-03  0.1180  3.29           GDAP2
156         Q9UBX3 Recurrence 2.23e-03  0.1180  3.29        SLC25A10
157         Q12996 Recurrence 2.26e-03  0.1180  3.29           CSTF3
158         Q9UKD2 Recurrence 2.26e-03  0.1180  3.29           MRTO4
159         P62736 Recurrence 2.28e-03  0.1180 -3.28           ACTA2
160         O75190 Recurrence 2.32e-03  0.1190 -3.28          DNAJB6
161         Q9Y679 Recurrence 2.33e-03  0.1190  3.28            AUP1
162         Q9NQ48 Recurrence 2.36e-03  0.1200 -3.27          LZTFL1
163         Q6KCM7 Recurrence 2.44e-03  0.1230  3.26        SLC25A25
164         Q9H4M9 Recurrence 2.47e-03  0.1240 -3.25            EHD1
165         O94979 Recurrence 2.48e-03  0.1240  3.25          SEC31A
166         P08729 Recurrence 2.51e-03  0.1250 -3.25            KRT7
167         P38159 Recurrence 2.52e-03  0.1250 -3.25            RBMX
168         Q8WXX5 Recurrence 2.57e-03  0.1250 -3.24          DNAJC9
169         Q9NWV4 Recurrence 2.57e-03  0.1250 -3.24            CZIB
170         O43427 Recurrence 2.58e-03  0.1250  3.24            FIBP
171         P24844 Recurrence 2.63e-03  0.1260 -3.23            MYL9
172         O60825 Recurrence 2.65e-03  0.1260  3.23          PFKFB2
173         Q9BZE4 Recurrence 2.67e-03  0.1260  3.23          GTPBP4
174         Q9NXC5 Recurrence 2.69e-03  0.1260  3.22            MIOS
175         Q969V3 Recurrence 2.72e-03  0.1260  3.22            NCLN
176         Q86T03 Recurrence 2.75e-03  0.1260  3.22          PIP4P1
177         Q99733 Recurrence 2.76e-03  0.1260 -3.21          NAP1L4
178         P12429 Recurrence 2.82e-03  0.1260 -3.21           ANXA3
179         Q86VS8 Recurrence 2.89e-03  0.1260 -3.20           HOOK3
180         Q9NYH9 Recurrence 2.90e-03  0.1260  3.20            UTP6
181         P61952 Recurrence 2.92e-03  0.1260 -3.19           GNG11
182         O15173 Recurrence 2.94e-03  0.1260 -3.19          PGRMC2
183         O75368 Recurrence 2.94e-03  0.1260 -3.19         SH3BGRL
184         P12235 Recurrence 2.94e-03  0.1260  3.19         SLC25A4
185         Q9Y696 Recurrence 2.94e-03  0.1260 -3.19           CLIC4
186         Q9Y5Q8 Recurrence 2.99e-03  0.1260  3.18          GTF3C5
187         P58340 Recurrence 3.00e-03  0.1260 -3.18            MLF1
188         P35241 Recurrence 3.01e-03  0.1260 -3.18             RDX
189         Q9Y5U9 Recurrence 3.01e-03  0.1260  3.18         IER3IP1
190         O14924 Recurrence 3.02e-03  0.1260  3.18           RGS12
191         P14649 Recurrence 3.03e-03  0.1260 -3.18           MYL6B
192         Q9BUH6 Recurrence 3.03e-03  0.1260 -3.18            PAXX
193         P17980 Recurrence 3.04e-03  0.1260 -3.18           PSMC3
194         P05067 Recurrence 3.06e-03  0.1260 -3.18             APP
195         P30101 Recurrence 3.06e-03  0.1260 -3.18           PDIA3
196         P62191 Recurrence 3.07e-03  0.1260 -3.17           PSMC1
197         P08134 Recurrence 3.08e-03  0.1260 -3.17            RHOC
198         Q12933 Recurrence 3.09e-03  0.1260 -3.17           TRAF2
199         O60925 Recurrence 3.13e-03  0.1260 -3.17           PFDN1
200         Q92629 Recurrence 3.13e-03  0.1260 -3.17            SGCD
201         Q9Y624 Recurrence 3.13e-03  0.1260 -3.17            F11R
202         O75347 Recurrence 3.20e-03  0.1260 -3.16            TBCA
203         P17661 Recurrence 3.20e-03  0.1260 -3.16             DES
204         Q8NBJ5 Recurrence 3.20e-03  0.1260  3.16        COLGALT1
205         P33240 Recurrence 3.21e-03  0.1260 -3.16           CSTF2
206         Q16822 Recurrence 3.23e-03  0.1260  3.16            PCK2
207         P48681 Recurrence 3.26e-03  0.1260 -3.15             NES
208         Q8N766 Recurrence 3.28e-03  0.1260  3.15            EMC1
209         P20810 Recurrence 3.29e-03  0.1260 -3.15            CAST
210         Q99961 Recurrence 3.29e-03  0.1260 -3.15          SH3GL1
211         Q9BV10 Recurrence 3.29e-03  0.1260  3.15           ALG12
212         Q9H267 Recurrence 3.29e-03  0.1260  3.15          VPS33B
213         Q9Y6A9 Recurrence 3.29e-03  0.1260  3.15           SPCS1
214         Q9Y3E5 Recurrence 3.31e-03  0.1260  3.15           PTRH2
215         P49321 Recurrence 3.34e-03  0.1260 -3.14            NASP
216         Q0ZGT2 Recurrence 3.35e-03  0.1260 -3.14            NEXN
217         P51452 Recurrence 3.40e-03  0.1260 -3.14           DUSP3
218         Q6P3W7 Recurrence 3.40e-03  0.1260  3.14           SCYL2
219         P10909 Recurrence 3.41e-03  0.1260 -3.14             CLU
220         B0I1T2 Recurrence 3.42e-03  0.1260  3.13           MYO1G
221         Q6ZT21 Recurrence 3.44e-03  0.1260  3.13           TMPPE
222         O15382 Recurrence 3.47e-03  0.1260  3.13           BCAT2
223         Q9HAN9 Recurrence 3.47e-03  0.1260  3.13          NMNAT1
224         O75367 Recurrence 3.48e-03  0.1260  3.13       MACROH2A1
225         Q96GY0 Recurrence 3.48e-03  0.1260 -3.13         ZC2HC1A
226         Q9UNZ2 Recurrence 3.48e-03  0.1260 -3.13          NSFL1C
227         Q8TC05 Recurrence 3.49e-03  0.1260 -3.13            MDM1
228         P21333 Recurrence 3.50e-03  0.1260 -3.13            FLNA
229         Q14642 Recurrence 3.51e-03  0.1260 -3.12          INPP5A
230         Q96IW7 Recurrence 3.54e-03  0.1270  3.12          SEC22A
231         P33151 Recurrence 3.57e-03  0.1270 -3.12            CDH5
232         Q12955 Recurrence 3.58e-03  0.1270  3.12            ANK3
233         Q13813 Recurrence 3.59e-03  0.1270 -3.12          SPTAN1
234         Q99828 Recurrence 3.60e-03  0.1270 -3.12            CIB1
235         P63173 Recurrence 3.61e-03  0.1270  3.11           RPL38
236         Q7Z3U7 Recurrence 3.64e-03  0.1270  3.11            MON2
237         O14874 Recurrence 3.68e-03  0.1280  3.11           BCKDK
238         P60660 Recurrence 3.71e-03  0.1280 -3.10            MYL6
239         Q9Y277 Recurrence 3.71e-03  0.1280  3.10           VDAC3
240         P17544 Recurrence 3.75e-03  0.1280  3.10            ATF7
241         Q9HD20 Recurrence 3.79e-03  0.1280  3.10         ATP13A1
242         O60711 Recurrence 3.80e-03  0.1280  3.09            LPXN
243         O43765 Recurrence 3.82e-03  0.1280 -3.09            SGTA
244         Q13601 Recurrence 3.83e-03  0.1280  3.09            KRR1
245         P25686 Recurrence 3.88e-03  0.1280 -3.09          DNAJB2
246         Q16540 Recurrence 3.88e-03  0.1280  3.09          MRPL23
247         Q9NZT1 Recurrence 3.88e-03  0.1280 -3.09          CALML5
248         Q9Y6N9 Recurrence 3.89e-03  0.1280 -3.09           USH1C
249         P23497 Recurrence 3.90e-03  0.1280 -3.09           SP100
250         O15018 Recurrence 3.91e-03  0.1280 -3.08           PDZD2
251         Q92621 Recurrence 3.93e-03  0.1280  3.08          NUP205
252         Q9H930 Recurrence 3.93e-03  0.1280 -3.08          SP140L
253         Q99747 Recurrence 3.94e-03  0.1280 -3.08            NAPG
254     A0A0B4J1U3 Recurrence 3.98e-03  0.1290 -3.08        IGLV1-36
255         Q96JB2 Recurrence 3.98e-03  0.1290  3.08            COG3
256         Q86Y79 Recurrence 4.01e-03  0.1290  3.07           PTRH1
257         P04844 Recurrence 4.06e-03  0.1300  3.07            RPN2
258         Q16643 Recurrence 4.08e-03  0.1300 -3.07            DBN1
259         P53680 Recurrence 4.09e-03  0.1300  3.07           AP2S1
260         Q9UJY4 Recurrence 4.11e-03  0.1300  3.06            GGA2
261         Q7KZF4 Recurrence 4.12e-03  0.1300  3.06            SND1
262         Q08174 Recurrence 4.14e-03  0.1300 -3.06           PCDH1
263         Q15293 Recurrence 4.17e-03  0.1300 -3.06            RCN1
264         P33121 Recurrence 4.18e-03  0.1300  3.06           ACSL1
265         Q6P4A7 Recurrence 4.21e-03  0.1310  3.06           SFXN4
266         Q9Y4C5 Recurrence 4.21e-03  0.1310 -3.06           CHST2
267         Q96RU3 Recurrence 4.23e-03  0.1310 -3.05           FNBP1
268         Q9HCE6 Recurrence 4.24e-03  0.1310  3.05       ARHGEF10L
269         Q15370 Recurrence 4.29e-03  0.1310 -3.05            ELOB
270         P42892 Recurrence 4.31e-03  0.1310 -3.05            ECE1
271         Q9NZU5 Recurrence 4.32e-03  0.1310 -3.05           LMCD1
272         P22897 Recurrence 4.38e-03  0.1330  3.04            MRC1
273         Q96QA5 Recurrence 4.39e-03  0.1330 -3.04           GSDMA
274         P11532 Recurrence 4.40e-03  0.1330 -3.04             DMD
275         Q8IY95 Recurrence 4.43e-03  0.1330  3.04         TMEM192
276         Q96P11 Recurrence 4.43e-03  0.1330  3.04           NSUN5
277         Q9ULW3 Recurrence 4.50e-03  0.1340  3.03            ABT1
278         Q14444 Recurrence 4.53e-03  0.1350 -3.03         CAPRIN1
279         P04843 Recurrence 4.57e-03  0.1350  3.02            RPN1
280         Q9UGP8 Recurrence 4.57e-03  0.1350  3.03           SEC63
281  O14950;P19105 Recurrence 4.62e-03  0.1360 -3.02   MYL12B;MYL12A
282         Q7Z4V5 Recurrence 4.64e-03  0.1360 -3.02          HDGFL2
283         P61326 Recurrence 4.69e-03  0.1360 -3.01           MAGOH
284         P55290 Recurrence 4.70e-03  0.1360 -3.01           CDH13
285         O75822 Recurrence 4.71e-03  0.1360 -3.01           EIF3J
286         P04632 Recurrence 4.73e-03  0.1370 -3.01          CAPNS1
287         P53992 Recurrence 4.79e-03  0.1370  3.01          SEC24C
288         Q14376 Recurrence 4.79e-03  0.1370  3.01            GALE
289         P10301 Recurrence 4.81e-03  0.1370 -3.01            RRAS
290         P15924 Recurrence 4.81e-03  0.1370 -3.00             DSP
291         Q96N66 Recurrence 4.89e-03  0.1390  3.00          MBOAT7
292         Q92600 Recurrence 4.91e-03  0.1390  3.00           CNOT9
293         Q9NVP1 Recurrence 4.93e-03  0.1390  3.00           DDX18
294         Q01082 Recurrence 4.95e-03  0.1390 -2.99          SPTBN1
295         Q96A00 Recurrence 5.02e-03  0.1400 -2.99        PPP1R14A
296         Q96HY6 Recurrence 5.04e-03  0.1400 -2.99          DDRGK1
297         P13797 Recurrence 5.11e-03  0.1420 -2.98            PLS3
298         Q8WVV4 Recurrence 5.13e-03  0.1420 -2.98           POF1B
299         O00423 Recurrence 5.16e-03  0.1420 -2.98            EML1
300         Q7L591 Recurrence 5.17e-03  0.1420  2.98            DOK3
301         A0PJW6 Recurrence 5.19e-03  0.1420  2.98         TMEM223
302         Q9Y4D1 Recurrence 5.28e-03  0.1440 -2.97           DAAM1
303         A0AV96 Recurrence 5.30e-03  0.1440  2.97           RBM47
304         P18085 Recurrence 5.33e-03  0.1450  2.97            ARF4
305         Q13107 Recurrence 5.38e-03  0.1460  2.96            USP4
306         P53609 Recurrence 5.41e-03  0.1460 -2.96          PGGT1B
307         Q86YM7 Recurrence 5.43e-03  0.1460 -2.96          HOMER1
308         Q00341 Recurrence 5.62e-03  0.1500  2.95           HDLBP
309         O43707 Recurrence 5.64e-03  0.1500 -2.94           ACTN4
310         P09382 Recurrence 5.65e-03  0.1500 -2.94          LGALS1
311         Q7Z3J2 Recurrence 5.67e-03  0.1500  2.94          VPS35L
312         O00291 Recurrence 5.71e-03  0.1510 -2.94            HIP1
313         Q8WXF7 Recurrence 5.73e-03  0.1510  2.94            ATL1
314         P54886 Recurrence 5.77e-03  0.1510  2.94        ALDH18A1
315         Q8TDR0 Recurrence 5.78e-03  0.1510 -2.93        TRAF3IP1
316         P51659 Recurrence 5.81e-03  0.1510  2.93         HSD17B4
317         Q9Y487 Recurrence 5.81e-03  0.1510  2.93        ATP6V0A2
318         Q9BZF9 Recurrence 5.85e-03  0.1520 -2.93            UACA
319         Q5D862 Recurrence 5.89e-03  0.1520 -2.93            FLG2
320  P0DMV8;P0DMV9 Recurrence 5.93e-03  0.1530 -2.93   HSPA1A;HSPA1B
321         Q7L014 Recurrence 5.94e-03  0.1530  2.92           DDX46
322         P21730 Recurrence 6.00e-03  0.1530  2.92           C5AR1
323         P35221 Recurrence 6.00e-03  0.1530 -2.92          CTNNA1
324         Q49A26 Recurrence 6.01e-03  0.1530  2.92           GLYR1
325         P33176 Recurrence 6.03e-03  0.1530 -2.92           KIF5B
326         Q9BUN8 Recurrence 6.05e-03  0.1530  2.92           DERL1
327         Q13404 Recurrence 6.06e-03  0.1530 -2.92          UBE2V1
328         O75044 Recurrence 6.08e-03  0.1530 -2.91          SRGAP2
329         Q6IA69 Recurrence 6.13e-03  0.1540  2.91         NADSYN1
330         O14832 Recurrence 6.15e-03  0.1540  2.91            PHYH
331         Q93050 Recurrence 6.16e-03  0.1540  2.91        ATP6V0A1
332         Q9UPU5 Recurrence 6.19e-03  0.1540  2.91           USP24
333         Q9P0M6 Recurrence 6.24e-03  0.1540  2.90       MACROH2A2
334         P54289 Recurrence 6.25e-03  0.1540 -2.90        CACNA2D1
335         Q8NBJ9 Recurrence 6.25e-03  0.1540  2.90           SIDT2
336         P63165 Recurrence 6.28e-03  0.1540 -2.90           SUMO1
337         O00443 Recurrence 6.34e-03  0.1550  2.90         PIK3C2A
338         P08727 Recurrence 6.35e-03  0.1550 -2.90           KRT19
339         Q8N4S9 Recurrence 6.39e-03  0.1560 -2.90        MARVELD2
340         P68133 Recurrence 6.46e-03  0.1570 -2.89           ACTA1
341         Q96RQ1 Recurrence 6.48e-03  0.1570  2.89          ERGIC2
342         P30876 Recurrence 6.57e-03  0.1580  2.89          POLR2B
343         Q14435 Recurrence 6.57e-03  0.1580  2.89          GALNT3
344         Q9UMY4 Recurrence 6.58e-03  0.1580 -2.88           SNX12
345         Q9P0J0 Recurrence 6.61e-03  0.1580  2.88         NDUFA13
346         P54198 Recurrence 6.65e-03  0.1580  2.88            HIRA
347         O75116 Recurrence 6.67e-03  0.1580 -2.88           ROCK2
348         Q96CD0 Recurrence 6.67e-03  0.1580  2.88           FBXL8
349         Q9Y678 Recurrence 6.70e-03  0.1580  2.88           COPG1
350         Q8IXS6 Recurrence 6.71e-03  0.1580 -2.88           PALM2
351         P62333 Recurrence 6.74e-03  0.1580 -2.87           PSMC6
352         P40123 Recurrence 6.75e-03  0.1580 -2.87            CAP2
353         P43405 Recurrence 6.80e-03  0.1580  2.87             SYK
354         P98194 Recurrence 6.81e-03  0.1580  2.87          ATP2C1
355         Q9NZM1 Recurrence 6.82e-03  0.1580  2.87            MYOF
356         P11234 Recurrence 6.83e-03  0.1580 -2.87            RALB
357         Q13740 Recurrence 6.91e-03  0.1580 -2.87           ALCAM
358         P30086 Recurrence 6.92e-03  0.1580 -2.87           PEBP1
359         Q969P0 Recurrence 6.92e-03  0.1580 -2.86           IGSF8
360         P05556 Recurrence 6.94e-03  0.1580 -2.86           ITGB1
361         Q07617 Recurrence 6.95e-03  0.1580 -2.86           SPAG1
362         Q8NB16 Recurrence 6.95e-03  0.1580  2.86            MLKL
363         P27816 Recurrence 6.97e-03  0.1580 -2.86            MAP4
364         Q96KG9 Recurrence 6.98e-03  0.1580  2.86           SCYL1
365         Q8TDB4 Recurrence 7.00e-03  0.1580 -2.86           MGARP
366         Q9UMS6 Recurrence 7.01e-03  0.1580 -2.86          SYNPO2
367         Q9Y3A4 Recurrence 7.08e-03  0.1580  2.86           RRP7A
368         P22061 Recurrence 7.09e-03  0.1580 -2.86           PCMT1
369         P55285 Recurrence 7.09e-03  0.1580 -2.86            CDH6
370         Q9Y2G9 Recurrence 7.15e-03  0.1590  2.85           SBNO2
371         P50402 Recurrence 7.16e-03  0.1590 -2.85             EMD
372         Q14258 Recurrence 7.16e-03  0.1590  2.85          TRIM25
373         Q9UDT6 Recurrence 7.28e-03  0.1610 -2.85           CLIP2
374         P20839 Recurrence 7.35e-03  0.1620  2.84          IMPDH1
375         Q16831 Recurrence 7.36e-03  0.1620  2.84            UPP1
376         Q9UH65 Recurrence 7.40e-03  0.1620 -2.84          SWAP70
377         Q9NVD7 Recurrence 7.42e-03  0.1620 -2.84           PARVA
378         P04406 Recurrence 7.44e-03  0.1620 -2.84           GAPDH
379         Q9P0K7 Recurrence 7.50e-03  0.1630 -2.83           RAI14
380         Q92604 Recurrence 7.55e-03  0.1640  2.83          LPGAT1
381         Q9H2D6 Recurrence 7.56e-03  0.1640 -2.83          TRIOBP
382         O00533 Recurrence 7.69e-03  0.1660 -2.82            CHL1
383         Q9BW83 Recurrence 7.73e-03  0.1660 -2.82           IFT27
384         Q10589 Recurrence 7.76e-03  0.1660 -2.82            BST2
385         Q8IY31 Recurrence 7.77e-03  0.1660 -2.82           IFT20
386         Q96GQ5 Recurrence 7.78e-03  0.1660  2.82           RUSF1
387         O14683 Recurrence 7.85e-03  0.1660  2.82         TP53I11
388         P23469 Recurrence 7.86e-03  0.1660  2.82           PTPRE
389         P16402 Recurrence 7.87e-03  0.1660 -2.81            H1-3
390         Q13367 Recurrence 7.87e-03  0.1660  2.81           AP3B2
391         Q9Y6D5 Recurrence 7.88e-03  0.1660  2.81         ARFGEF2
392         Q9NX20 Recurrence 7.90e-03  0.1660  2.81          MRPL16
393         Q9UBY9 Recurrence 7.90e-03  0.1660 -2.81           HSPB7
394         P06727 Recurrence 7.96e-03  0.1660 -2.81           APOA4
395         P55209 Recurrence 7.96e-03  0.1660 -2.81          NAP1L1
396         H7BZ55 Recurrence 7.97e-03  0.1660 -2.81          CROCC2
397         Q9UKM7 Recurrence 7.97e-03  0.1660  2.81          MAN1B1
398         Q8NI37 Recurrence 8.07e-03  0.1670  2.80           PPTC7
399         P01111 Recurrence 8.09e-03  0.1670 -2.80            NRAS
400         P53602 Recurrence 8.10e-03  0.1670  2.80             MVD
401         O75436 Recurrence 8.22e-03  0.1690  2.80          VPS26A
402         Q9UBC2 Recurrence 8.23e-03  0.1690 -2.80         EPS15L1
403         Q12849 Recurrence 8.29e-03  0.1700 -2.79           GRSF1
404         Q15437 Recurrence 8.33e-03  0.1700  2.79          SEC23B
405         Q9Y6R1 Recurrence 8.37e-03  0.1700  2.79          SLC4A4
406         P08590 Recurrence 8.39e-03  0.1700 -2.79            MYL3
407         O75616 Recurrence 8.40e-03  0.1700  2.79           ERAL1
408         Q9H6F5 Recurrence 8.40e-03  0.1700 -2.79          CCDC86
409         Q9GZL7 Recurrence 8.43e-03  0.1700  2.79           WDR12
410         P62942 Recurrence 8.45e-03  0.1700 -2.79          FKBP1A
411         P54727 Recurrence 8.47e-03  0.1700 -2.79          RAD23B
412         Q96JC1 Recurrence 8.54e-03  0.1710  2.78           VPS39
413         P30511 Recurrence 8.58e-03  0.1710 -2.78           HLA-F
414         Q16825 Recurrence 8.60e-03  0.1710 -2.78          PTPN21
415         Q96RQ3 Recurrence 8.64e-03  0.1710  2.78           MCCC1
416         P50454 Recurrence 8.65e-03  0.1710  2.78        SERPINH1
417         A6NI72 Recurrence 8.74e-03  0.1720  2.77           NCF1B
418         P31948 Recurrence 8.75e-03  0.1720 -2.77           STIP1
419         Q8IYU8 Recurrence 8.78e-03  0.1720  2.77           MICU2
420         Q32MZ4 Recurrence 8.79e-03  0.1720 -2.77         LRRFIP1
421         Q10471 Recurrence 8.82e-03  0.1720  2.77          GALNT2
422         Q6BCY4 Recurrence 8.82e-03  0.1720  2.77          CYB5R2
423         Q9Y606 Recurrence 8.82e-03  0.1720  2.77            PUS1
424         P56199 Recurrence 8.83e-03  0.1720 -2.77           ITGA1
425         Q96G23 Recurrence 8.85e-03  0.1720  2.77           CERS2
426         Q9NRA2 Recurrence 8.89e-03  0.1720  2.77         SLC17A5
427         Q9HBL7 Recurrence 8.93e-03  0.1730  2.76          PLGRKT
428         O75475 Recurrence 9.01e-03  0.1730 -2.76           PSIP1
429         O15446 Recurrence 9.02e-03  0.1730 -2.76          POLR1G
430         Q93033 Recurrence 9.05e-03  0.1740  2.76           CD101
431         Q8NES3 Recurrence 9.06e-03  0.1740 -2.76            LFNG
432         P62745 Recurrence 9.09e-03  0.1740 -2.76            RHOB
433         Q15329 Recurrence 9.13e-03  0.1740  2.76            E2F5
434         P28068 Recurrence 9.14e-03  0.1740  2.76         HLA-DMB
435         Q9H3E2 Recurrence 9.16e-03  0.1740  2.75           SNX25
436         Q9BWS9 Recurrence 9.20e-03  0.1740  2.75           CHID1
437         Q16795 Recurrence 9.23e-03  0.1740  2.75          NDUFA9
438         P55196 Recurrence 9.28e-03  0.1740 -2.75            AFDN
439         Q9BZ95 Recurrence 9.29e-03  0.1740  2.75            NSD3
440         Q92947 Recurrence 9.31e-03  0.1740  2.75            GCDH
441         Q86SQ0 Recurrence 9.34e-03  0.1750 -2.75          PHLDB2
442         Q9Y2D5 Recurrence 9.37e-03  0.1750 -2.75           AKAP2
443         P55160 Recurrence 9.41e-03  0.1750  2.74         NCKAP1L
444         P53708 Recurrence 9.43e-03  0.1750 -2.74           ITGA8
445         Q9NRG9 Recurrence 9.50e-03  0.1760  2.74            AAAS
446         Q5RHP9 Recurrence 9.52e-03  0.1760 -2.74          ERICH3
447         Q8N5G0 Recurrence 9.55e-03  0.1760  2.74          SMIM20
448         Q6ZVM7 Recurrence 9.60e-03  0.1760 -2.74          TOM1L2
449         O00592 Recurrence 9.62e-03  0.1760 -2.74           PODXL
450         Q9Y6D6 Recurrence 9.63e-03  0.1760  2.73         ARFGEF1
451         Q9BZE1 Recurrence 9.70e-03  0.1770  2.73          MRPL37
452         P35052 Recurrence 9.72e-03  0.1770 -2.73            GPC1
453         P49006 Recurrence 9.79e-03  0.1780 -2.73        MARCKSL1
454         Q13084 Recurrence 9.80e-03  0.1780  2.73          MRPL28
455         P00973 Recurrence 9.87e-03  0.1790  2.72            OAS1
456         Q9P2D1 Recurrence 9.91e-03  0.1790 -2.72            CHD7
457         Q7Z4W1 Recurrence 9.94e-03  0.1790  2.72            DCXR
458         Q9NZB2 Recurrence 1.00e-02  0.1800  2.72         FAM120A
459         Q9Y5J5 Recurrence 1.01e-02  0.1810  2.72          PHLDA3
460         Q53FV1 Recurrence 1.02e-02  0.1820  2.71          ORMDL2
461         Q9Y2X3 Recurrence 1.02e-02  0.1830  2.71           NOP58
462         Q5T7N3 Recurrence 1.04e-02  0.1850 -2.71           KANK4
463         Q8WTW3 Recurrence 1.04e-02  0.1850  2.70            COG1
464         P29536 Recurrence 1.05e-02  0.1870 -2.70           LMOD1
465         Q709C8 Recurrence 1.05e-02  0.1860  2.70          VPS13C
466         P13747 Recurrence 1.06e-02  0.1870 -2.70           HLA-E
467         Q496M5 Recurrence 1.06e-02  0.1870 -2.70            PLK5
468         Q96QE2 Recurrence 1.06e-02  0.1870  2.69         SLC2A13
469         Q9H0A0 Recurrence 1.06e-02  0.1870  2.70           NAT10
470         O60341 Recurrence 1.07e-02  0.1870  2.69           KDM1A
471         Q01780 Recurrence 1.07e-02  0.1870  2.69         EXOSC10
472         Q96GC9 Recurrence 1.07e-02  0.1870  2.69            VMP1
473         Q9NX74 Recurrence 1.07e-02  0.1870 -2.69            DUS2
474         P16118 Recurrence 1.08e-02  0.1870  2.69          PFKFB1
475         Q9UPN9 Recurrence 1.08e-02  0.1870  2.69          TRIM33
476         Q8TEA8 Recurrence 1.09e-02  0.1880 -2.69            DTD1
477         Q9H9B1 Recurrence 1.09e-02  0.1890  2.68           EHMT1
478         Q03426 Recurrence 1.10e-02  0.1900  2.68             MVK
479         Q6WCQ1 Recurrence 1.10e-02  0.1900 -2.68           MPRIP
480         Q9Y3B2 Recurrence 1.10e-02  0.1890  2.68          EXOSC1
481         O60907 Recurrence 1.11e-02  0.1900  2.68           TBL1X
482         Q8IX04 Recurrence 1.11e-02  0.1900  2.68           UEVLD
483         Q99459 Recurrence 1.11e-02  0.1900 -2.68           CDC5L
484         Q969Q0 Recurrence 1.12e-02  0.1910 -2.67         RPL36AL
485         Q9Y2V7 Recurrence 1.12e-02  0.1900  2.68            COG6
486         Q6JBY9 Recurrence 1.13e-02  0.1920 -2.67           RCSD1
487         P04259 Recurrence 1.14e-02  0.1920 -2.67           KRT6B
488         Q6ZSS7 Recurrence 1.14e-02  0.1920 -2.67           MFSD6
489         Q92900 Recurrence 1.14e-02  0.1920  2.67            UPF1
490         O95816 Recurrence 1.15e-02  0.1930 -2.66            BAG2
491         P02647 Recurrence 1.15e-02  0.1930 -2.66           APOA1
492         P13693 Recurrence 1.15e-02  0.1930 -2.66            TPT1
493         Q8TBP5 Recurrence 1.15e-02  0.1930 -2.66         FAM174A
494         Q8TCJ2 Recurrence 1.15e-02  0.1930  2.66           STT3B
495         P37198 Recurrence 1.16e-02  0.1930 -2.66           NUP62
496         P78345 Recurrence 1.16e-02  0.1930  2.66           RPP38
497         Q16651 Recurrence 1.16e-02  0.1930 -2.66           PRSS8
498         Q99426 Recurrence 1.16e-02  0.1930 -2.66            TBCB
499         Q9H2V7 Recurrence 1.16e-02  0.1930  2.66           SPNS1
500         P21291 Recurrence 1.17e-02  0.1930 -2.66           CSRP1
501         Q00059 Recurrence 1.17e-02  0.1930 -2.65            TFAM
502         Q8NCE2 Recurrence 1.17e-02  0.1930  2.66          MTMR14
503         Q9NYL9 Recurrence 1.17e-02  0.1930 -2.66           TMOD3
504         P41236 Recurrence 1.18e-02  0.1930 -2.65          PPP1R2
505         Q9NWW6 Recurrence 1.18e-02  0.1940  2.65           NMRK1
506         O60271 Recurrence 1.20e-02  0.1960 -2.64           SPAG9
507         P39060 Recurrence 1.20e-02  0.1960 -2.65         COL18A1
508         Q8NBM4 Recurrence 1.21e-02  0.1960  2.64           UBAC2
509         P19447 Recurrence 1.22e-02  0.1980  2.64           ERCC3
510         O95833 Recurrence 1.23e-02  0.1990 -2.64           CLIC3
511         Q99547 Recurrence 1.23e-02  0.1990 -2.64        MPHOSPH6
512         Q8N129 Recurrence 1.24e-02  0.2000 -2.63           CNPY4
513         Q8NE86 Recurrence 1.24e-02  0.2000  2.63             MCU
514         Q12846 Recurrence 1.25e-02  0.2000 -2.63            STX4
515         Q5JPH6 Recurrence 1.25e-02  0.2000  2.63           EARS2
516         Q9ULW0 Recurrence 1.25e-02  0.2000 -2.63            TPX2
517         O43464 Recurrence 1.26e-02  0.2000  2.63           HTRA2
518         P54920 Recurrence 1.26e-02  0.2000 -2.63            NAPA
519         Q5VWJ9 Recurrence 1.26e-02  0.2000  2.63           SNX30
520         Q86SX6 Recurrence 1.26e-02  0.2000 -2.63           GLRX5
521         Q9UHB9 Recurrence 1.26e-02  0.2000  2.63           SRP68
522         O95347 Recurrence 1.27e-02  0.2000 -2.62            SMC2
523         P49754 Recurrence 1.27e-02  0.2000  2.62           VPS41
524         Q16637 Recurrence 1.27e-02  0.2000  2.62            SMN2
525         O00161 Recurrence 1.28e-02  0.2010 -2.62          SNAP23
526         O15254 Recurrence 1.28e-02  0.2010  2.62           ACOX3
527         O00264 Recurrence 1.29e-02  0.2010 -2.62          PGRMC1
528         O15400 Recurrence 1.29e-02  0.2010 -2.62            STX7
529         Q15276 Recurrence 1.29e-02  0.2010 -2.62          RABEP1
530         Q6S8J3 Recurrence 1.29e-02  0.2010 -2.62           POTEE
531         P01700 Recurrence 1.30e-02  0.2010 -2.61        IGLV1-47
532         Q9BYV8 Recurrence 1.30e-02  0.2010 -2.61           CEP41
533         Q9Y279 Recurrence 1.30e-02  0.2010  2.61           VSIG4
534         P21796 Recurrence 1.31e-02  0.2010  2.61           VDAC1
535         P26038 Recurrence 1.31e-02  0.2010 -2.61             MSN
536         P35579 Recurrence 1.31e-02  0.2010 -2.61            MYH9
537         P57740 Recurrence 1.31e-02  0.2010  2.61          NUP107
538         Q96T76 Recurrence 1.31e-02  0.2010  2.61           MMS19
539         P45877 Recurrence 1.32e-02  0.2010  2.61            PPIC
540         Q8N3P4 Recurrence 1.32e-02  0.2010  2.61            VPS8
541         Q13099 Recurrence 1.33e-02  0.2010 -2.60           IFT88
542         Q8NI60 Recurrence 1.33e-02  0.2010  2.60           COQ8A
543         Q9HC35 Recurrence 1.33e-02  0.2010  2.60            EML4
544         P20930 Recurrence 1.34e-02  0.2010 -2.60             FLG
545         P30626 Recurrence 1.34e-02  0.2010 -2.60             SRI
546         Q02413 Recurrence 1.34e-02  0.2010 -2.60            DSG1
547         Q8TB40 Recurrence 1.34e-02  0.2010  2.60           ABHD4
548         Q9NWH9 Recurrence 1.34e-02  0.2010 -2.60            SLTM
549         Q9P2B4 Recurrence 1.34e-02  0.2010 -2.60       CTTNBP2NL
550         P10398 Recurrence 1.35e-02  0.2010  2.60            ARAF
551         Q6UWJ1 Recurrence 1.35e-02  0.2010  2.60           TMCO3
552         Q7Z4S6 Recurrence 1.35e-02  0.2010 -2.60          KIF21A
553         Q8NDH2 Recurrence 1.35e-02  0.2010 -2.60         CCDC168
554         Q96HV5 Recurrence 1.35e-02  0.2010  2.60         TMEM41A
555         O75879 Recurrence 1.36e-02  0.2010 -2.59            GATB
556         O94956 Recurrence 1.36e-02  0.2010  2.60         SLCO2B1
557         O95639 Recurrence 1.36e-02  0.2010  2.59           CPSF4
558         P55287 Recurrence 1.36e-02  0.2010 -2.60           CDH11
559         Q15642 Recurrence 1.36e-02  0.2010 -2.59          TRIP10
560         Q8NEY8 Recurrence 1.36e-02  0.2010 -2.59          PPHLN1
561         Q9H553 Recurrence 1.36e-02  0.2010  2.59            ALG2
562         P13647 Recurrence 1.37e-02  0.2010 -2.59            KRT5
563         Q96DP5 Recurrence 1.37e-02  0.2010  2.59           MTFMT
564         O00154 Recurrence 1.38e-02  0.2010  2.59           ACOT7
565         P78367 Recurrence 1.38e-02  0.2010 -2.59          NKX3-2
566         Q15080 Recurrence 1.39e-02  0.2020  2.59            NCF4
567         Q15848 Recurrence 1.39e-02  0.2020 -2.58          ADIPOQ
568         Q9Y2Z4 Recurrence 1.39e-02  0.2020  2.59           YARS2
569         G2XKQ0 Recurrence 1.40e-02  0.2020 -2.58         SUMO1P1
570         P82930 Recurrence 1.40e-02  0.2020  2.58          MRPS34
571         Q8N4T8 Recurrence 1.40e-02  0.2020  2.58            CBR4
572         Q96M27 Recurrence 1.40e-02  0.2020  2.58           PRRC1
573         P11217 Recurrence 1.41e-02  0.2020 -2.58            PYGM
574         O15247 Recurrence 1.42e-02  0.2020 -2.58           CLIC2
575         P27348 Recurrence 1.42e-02  0.2020 -2.58           YWHAQ
576         P46087 Recurrence 1.42e-02  0.2020  2.58            NOP2
577         P53367 Recurrence 1.42e-02  0.2020 -2.58          ARFIP1
578         Q587I9 Recurrence 1.42e-02  0.2020  2.58          SFT2D3
579         Q8IY17 Recurrence 1.42e-02  0.2020  2.58          PNPLA6
580         Q9Y4W2 Recurrence 1.42e-02  0.2020  2.58           LAS1L
581         O60603 Recurrence 1.43e-02  0.2020  2.58            TLR2
582         P49792 Recurrence 1.44e-02  0.2040  2.57          RANBP2
583         Q5TEJ8 Recurrence 1.44e-02  0.2040  2.57         THEMIS2
584         O75165 Recurrence 1.45e-02  0.2040  2.57         DNAJC13
585         P38571 Recurrence 1.45e-02  0.2040  2.57            LIPA
586         Q96SB3 Recurrence 1.45e-02  0.2040 -2.57         PPP1R9B
587         Q8IZ81 Recurrence 1.46e-02  0.2050  2.57          ELMOD2
588         P07332 Recurrence 1.47e-02  0.2060  2.56             FES
589         Q2TAA2 Recurrence 1.47e-02  0.2060 -2.56            IAH1
590         Q9BX66 Recurrence 1.47e-02  0.2060 -2.56          SORBS1
591         P12236 Recurrence 1.48e-02  0.2060  2.56         SLC25A6
592         Q8TDD1 Recurrence 1.48e-02  0.2060  2.56           DDX54
593         Q9BU89 Recurrence 1.48e-02  0.2060 -2.56            DOHH
594         Q9NPA0 Recurrence 1.48e-02  0.2060  2.56            EMC7
595         Q7Z794 Recurrence 1.49e-02  0.2070 -2.56           KRT77
596         Q96L92 Recurrence 1.49e-02  0.2060  2.56           SNX27
597         O00116 Recurrence 1.50e-02  0.2070  2.55            AGPS
598         Q96PU5 Recurrence 1.50e-02  0.2070 -2.55          NEDD4L
599         Q9H1A3 Recurrence 1.50e-02  0.2070  2.56          METTL9
600         Q07065 Recurrence 1.51e-02  0.2070 -2.55           CKAP4
601         A4D1P6 Recurrence 1.52e-02  0.2080  2.55           WDR91
602         P13727 Recurrence 1.52e-02  0.2080  2.55            PRG2
603         Q9BQG0 Recurrence 1.52e-02  0.2080  2.55         MYBBP1A
604         Q9NVJ2 Recurrence 1.52e-02  0.2080  2.55           ARL8B
605         Q9UPQ0 Recurrence 1.52e-02  0.2080 -2.55          LIMCH1
606         P11310 Recurrence 1.53e-02  0.2080  2.55           ACADM
607         Q8WXK1 Recurrence 1.53e-02  0.2080  2.55           ASB15
608         Q9NQX4 Recurrence 1.54e-02  0.2090  2.54           MYO5C
609         O95486 Recurrence 1.55e-02  0.2100  2.54          SEC24A
610         Q5VW38 Recurrence 1.55e-02  0.2100  2.54          GPR107
611         Q9UEW3 Recurrence 1.57e-02  0.2110  2.54           MARCO
612         Q9Y6I8 Recurrence 1.57e-02  0.2110  2.54           PXMP4
613         Q03001 Recurrence 1.58e-02  0.2120 -2.53             DST
614         O43852 Recurrence 1.59e-02  0.2140 -2.53            CALU
615         O94808 Recurrence 1.60e-02  0.2150  2.53           GFPT2
616         Q9H078 Recurrence 1.60e-02  0.2150  2.53            CLPB
617         Q92616 Recurrence 1.61e-02  0.2150  2.53            GCN1
618         P35580 Recurrence 1.62e-02  0.2150 -2.52           MYH10
619         Q92522 Recurrence 1.62e-02  0.2150 -2.52           H1-10
620         Q92945 Recurrence 1.62e-02  0.2150 -2.52           KHSRP
621         Q96EE3 Recurrence 1.62e-02  0.2150  2.52           SEH1L
622         O00750 Recurrence 1.63e-02  0.2150 -2.52         PIK3C2B
623         O15258 Recurrence 1.63e-02  0.2150  2.52            RER1
624         P49815 Recurrence 1.63e-02  0.2150  2.52            TSC2
625         Q6ZV73 Recurrence 1.63e-02  0.2150 -2.52            FGD6
626         Q8N9B5 Recurrence 1.63e-02  0.2150 -2.52             JMY
627         P31483 Recurrence 1.64e-02  0.2160  2.52            TIA1
628         Q07075 Recurrence 1.64e-02  0.2160 -2.52           ENPEP
629         P11233 Recurrence 1.65e-02  0.2160 -2.52            RALA
630         P62829 Recurrence 1.65e-02  0.2160  2.52           RPL23
631         Q00325 Recurrence 1.65e-02  0.2160  2.51         SLC25A3
632         Q13310 Recurrence 1.65e-02  0.2160  2.52          PABPC4
633         P43686 Recurrence 1.66e-02  0.2170 -2.51           PSMC4
634         P05362 Recurrence 1.67e-02  0.2170 -2.51           ICAM1
635         P28288 Recurrence 1.67e-02  0.2170  2.51           ABCD3
636         Q14728 Recurrence 1.67e-02  0.2170  2.51          MFSD10
637         O00148 Recurrence 1.69e-02  0.2180  2.51          DDX39A
638         P52799 Recurrence 1.69e-02  0.2180 -2.50           EFNB2
639         P12814 Recurrence 1.70e-02  0.2180 -2.50           ACTN1
640         P16066 Recurrence 1.70e-02  0.2180  2.50            NPR1
641         Q96LR5 Recurrence 1.70e-02  0.2180  2.50          UBE2E2
642         Q9NWS8 Recurrence 1.70e-02  0.2180  2.50           RMND1
643         P04049 Recurrence 1.71e-02  0.2180  2.50            RAF1
644         P10412 Recurrence 1.71e-02  0.2180 -2.50            H1-4
645         Q13488 Recurrence 1.71e-02  0.2180  2.50          TCIRG1
646         Q96AE7 Recurrence 1.71e-02  0.2180 -2.50           TTC17
647         Q99567 Recurrence 1.71e-02  0.2180  2.50           NUP88
648         P63010 Recurrence 1.72e-02  0.2190  2.50           AP2B1
649         P63244 Recurrence 1.72e-02  0.2180  2.50           RACK1
650         P58546 Recurrence 1.73e-02  0.2190 -2.49            MTPN
651         Q9H0R4 Recurrence 1.73e-02  0.2190 -2.50           HDHD2
652         Q9NPH3 Recurrence 1.73e-02  0.2190 -2.50          IL1RAP
653         O95810 Recurrence 1.74e-02  0.2190 -2.49          CAVIN2
654         Q7Z3B1 Recurrence 1.74e-02  0.2190 -2.49           NEGR1
655         Q9BQ52 Recurrence 1.74e-02  0.2190  2.49           ELAC2
656         Q9NVM9 Recurrence 1.74e-02  0.2190  2.49          INTS13
657         P78559 Recurrence 1.75e-02  0.2190 -2.49           MAP1A
658         Q7Z4Q2 Recurrence 1.75e-02  0.2190  2.49          HEATR3
659         Q99653 Recurrence 1.75e-02  0.2190 -2.49            CHP1
660         Q9HB90 Recurrence 1.75e-02  0.2190  2.49           RRAGC
661         P46977 Recurrence 1.76e-02  0.2190  2.49           STT3A
662         Q14571 Recurrence 1.76e-02  0.2190  2.49           ITPR2
663         Q6UW02 Recurrence 1.76e-02  0.2190  2.49         CYP20A1
664         Q86VX2 Recurrence 1.76e-02  0.2190  2.49          COMMD7
665         Q9NQ92 Recurrence 1.76e-02  0.2190 -2.49           COPRS
666         O14920 Recurrence 1.77e-02  0.2190  2.49           IKBKB
667         P01599 Recurrence 1.77e-02  0.2190 -2.49        IGKV1-17
668         Q71UM5 Recurrence 1.77e-02  0.2190  2.49          RPS27L
669         Q9UBQ0 Recurrence 1.77e-02  0.2190  2.49           VPS29
670         P12830 Recurrence 1.79e-02  0.2200 -2.48            CDH1
671         P16422 Recurrence 1.79e-02  0.2200 -2.48           EPCAM
672         P39687 Recurrence 1.79e-02  0.2200 -2.48          ANP32A
673         Q02809 Recurrence 1.79e-02  0.2200  2.48           PLOD1
674         O00560 Recurrence 1.80e-02  0.2200  2.48           SDCBP
675         O75306 Recurrence 1.80e-02  0.2200  2.48          NDUFS2
676         O95994 Recurrence 1.80e-02  0.2200 -2.48            AGR2
677         Q96AC1 Recurrence 1.81e-02  0.2200 -2.48          FERMT2
678         Q9ULP9 Recurrence 1.81e-02  0.2200  2.48         TBC1D24
679         P49406 Recurrence 1.82e-02  0.2210  2.47          MRPL19
680         Q15061 Recurrence 1.82e-02  0.2210  2.48           WDR43
681         P30837 Recurrence 1.83e-02  0.2210 -2.47         ALDH1B1
682         O00182 Recurrence 1.84e-02  0.2210  2.47          LGALS9
683         P05026 Recurrence 1.84e-02  0.2210 -2.47          ATP1B1
684         Q8IZV5 Recurrence 1.84e-02  0.2210  2.47           RDH10
685         P20290 Recurrence 1.85e-02  0.2210 -2.47            BTF3
686         Q6ZT12 Recurrence 1.85e-02  0.2210 -2.47            UBR3
687         Q96F07 Recurrence 1.85e-02  0.2210  2.47          CYFIP2
688         Q9NTZ6 Recurrence 1.85e-02  0.2210  2.47           RBM12
689         O00308 Recurrence 1.86e-02  0.2210 -2.46            WWP2
690         O95630 Recurrence 1.86e-02  0.2210 -2.46          STAMBP
691         P13646 Recurrence 1.86e-02  0.2210 -2.46           KRT13
692         P15880 Recurrence 1.86e-02  0.2210  2.46            RPS2
693         P28161 Recurrence 1.86e-02  0.2210 -2.47           GSTM2
694         Q6P4E1 Recurrence 1.86e-02  0.2210 -2.46           GOLM2
695         Q7L2H7 Recurrence 1.86e-02  0.2210  2.46           EIF3M
696         Q8TD16 Recurrence 1.86e-02  0.2210 -2.47           BICD2
697         P98172 Recurrence 1.87e-02  0.2210 -2.46           EFNB1
698         Q09666 Recurrence 1.87e-02  0.2210 -2.46           AHNAK
699         Q9Y6E2 Recurrence 1.87e-02  0.2210  2.46            BZW2
700         P01889 Recurrence 1.88e-02  0.2210 -2.46           HLA-B
701         P51398 Recurrence 1.88e-02  0.2210  2.46            DAP3
702         Q9UMN6 Recurrence 1.88e-02  0.2210  2.46           KMT2B
703         O60264 Recurrence 1.89e-02  0.2210  2.46         SMARCA5
704         Q8IXL7 Recurrence 1.89e-02  0.2210 -2.46           MSRB3
705         Q8TDY2 Recurrence 1.89e-02  0.2210 -2.46          RB1CC1
706         Q9ULX6 Recurrence 1.89e-02  0.2210  2.46          AKAP8L
707         P61960 Recurrence 1.90e-02  0.2210 -2.46            UFM1
708         Q8TBA6 Recurrence 1.90e-02  0.2210 -2.46          GOLGA5
709         Q9H7B4 Recurrence 1.90e-02  0.2210  2.46           SMYD3
710         P16284 Recurrence 1.91e-02  0.2220 -2.45          PECAM1
711         A1X283 Recurrence 1.92e-02  0.2220  2.45        SH3PXD2B
712         P00441 Recurrence 1.92e-02  0.2220 -2.45            SOD1
713         Q7Z406 Recurrence 1.92e-02  0.2220 -2.45           MYH14
714         Q9NRP2 Recurrence 1.92e-02  0.2220  2.45            CMC2
715         Q9Y385 Recurrence 1.93e-02  0.2220  2.45          UBE2J1
716         O15061 Recurrence 1.94e-02  0.2230 -2.45            SYNM
717         O75427 Recurrence 1.94e-02  0.2230  2.45           LRCH4
718         P48729 Recurrence 1.95e-02  0.2230  2.45         CSNK1A1
719         Q8IZQ1 Recurrence 1.95e-02  0.2230 -2.44           WDFY3
720         Q92538 Recurrence 1.95e-02  0.2230  2.45            GBF1
721         Q93100 Recurrence 1.95e-02  0.2230  2.44            PHKB
722         P55072 Recurrence 1.96e-02  0.2230 -2.44             VCP
723         Q12929 Recurrence 1.96e-02  0.2230  2.44            EPS8
724         O94875 Recurrence 1.97e-02  0.2250 -2.44          SORBS2
725         Q13526 Recurrence 1.98e-02  0.2250 -2.44            PIN1
726         Q9NRY2 Recurrence 1.98e-02  0.2250 -2.44            INIP
727         P42858 Recurrence 1.99e-02  0.2250  2.44             HTT
728         Q03135 Recurrence 1.99e-02  0.2250 -2.44            CAV1
729         Q8IY33 Recurrence 1.99e-02  0.2250 -2.44         MICALL2
730         Q96SK2 Recurrence 1.99e-02  0.2250  2.44         TMEM209
731         O15347 Recurrence 2.01e-02  0.2270 -2.43           HMGB3
732         Q8N1F7 Recurrence 2.01e-02  0.2270  2.43           NUP93
733         Q9BQ95 Recurrence 2.01e-02  0.2260  2.43           ECSIT
734         P09960 Recurrence 2.02e-02  0.2270  2.43           LTA4H
735         P51178 Recurrence 2.03e-02  0.2270 -2.43           PLCD1
736         P58499 Recurrence 2.03e-02  0.2270  2.43           FAM3B
737         Q99571 Recurrence 2.03e-02  0.2270  2.43           P2RX4
738         Q9Y3E7 Recurrence 2.03e-02  0.2270 -2.43           CHMP3
739         O43291 Recurrence 2.04e-02  0.2270 -2.43          SPINT2
740         P49810 Recurrence 2.04e-02  0.2270  2.43           PSEN2
741         P61619 Recurrence 2.04e-02  0.2270  2.43         SEC61A1
742         P62310 Recurrence 2.05e-02  0.2270 -2.42            LSM3
743         Q53GD3 Recurrence 2.05e-02  0.2280 -2.42         SLC44A4
744         Q86X02 Recurrence 2.05e-02  0.2270 -2.42           CDR2L
745         O95436 Recurrence 2.06e-02  0.2280  2.42         SLC34A2
746         P46108 Recurrence 2.08e-02  0.2300 -2.42             CRK
747         P17066 Recurrence 2.09e-02  0.2300  2.42           HSPA6
748         Q8N3C0 Recurrence 2.09e-02  0.2300  2.42           ASCC3
749         Q9NP98 Recurrence 2.09e-02  0.2300 -2.42           MYOZ1
750         Q9UP83 Recurrence 2.09e-02  0.2300  2.42            COG5
751         O75817 Recurrence 2.10e-02  0.2300  2.41            POP7
752         Q15629 Recurrence 2.10e-02  0.2300  2.41           TRAM1
753         Q5TD94 Recurrence 2.10e-02  0.2300 -2.41          RSPH4A
754         Q9Y3B7 Recurrence 2.10e-02  0.2300  2.41          MRPL11
755         O60832 Recurrence 2.12e-02  0.2310  2.41            DKC1
756         Q5TH69 Recurrence 2.12e-02  0.2310  2.41         ARFGEF3
757         Q9BV38 Recurrence 2.12e-02  0.2310  2.41           WDR18
758         O15143 Recurrence 2.13e-02  0.2310  2.41          ARPC1B
759         O75398 Recurrence 2.14e-02  0.2310 -2.41           DEAF1
760         Q9NSY2 Recurrence 2.14e-02  0.2310  2.41          STARD5
761         Q9NY47 Recurrence 2.14e-02  0.2310 -2.41        CACNA2D2
762         Q13938 Recurrence 2.15e-02  0.2310 -2.40            CAPS
763         Q99836 Recurrence 2.15e-02  0.2310  2.40           MYD88
764         O43633 Recurrence 2.16e-02  0.2310 -2.40          CHMP2A
765         O60684 Recurrence 2.16e-02  0.2310  2.40           KPNA6
766         P54652 Recurrence 2.16e-02  0.2310 -2.40           HSPA2
767         P81605 Recurrence 2.16e-02  0.2310 -2.40             DCD
768         Q9BTT0 Recurrence 2.16e-02  0.2310 -2.40          ANP32E
769         P20827 Recurrence 2.17e-02  0.2310 -2.40           EFNA1
770         Q07866 Recurrence 2.17e-02  0.2310 -2.40            KLC1
771         Q6P996 Recurrence 2.17e-02  0.2310  2.40          PDXDC1
772         Q8IWX8 Recurrence 2.17e-02  0.2310  2.40           CHERP
773         Q96CM8 Recurrence 2.17e-02  0.2310  2.40           ACSF2
774         O75947 Recurrence 2.18e-02  0.2310 -2.40          ATP5PD
775         P51812 Recurrence 2.18e-02  0.2310  2.40         RPS6KA3
776         Q15459 Recurrence 2.18e-02  0.2310 -2.40           SF3A1
777         Q8IV08 Recurrence 2.18e-02  0.2310  2.40            PLD3
778         Q8TC71 Recurrence 2.18e-02  0.2310 -2.40         SPATA18
779         Q96EP5 Recurrence 2.18e-02  0.2310  2.40          DAZAP1
780         Q53QV2 Recurrence 2.19e-02  0.2320 -2.40             LBH
781         Q6ICB0 Recurrence 2.19e-02  0.2320 -2.40           DESI1
782         O15120 Recurrence 2.20e-02  0.2320  2.39          AGPAT2
783         P09758 Recurrence 2.20e-02  0.2320 -2.39         TACSTD2
784         O00244 Recurrence 2.21e-02  0.2330 -2.39           ATOX1
785         P55212 Recurrence 2.22e-02  0.2330  2.39           CASP6
786         Q86Y07 Recurrence 2.22e-02  0.2330  2.39            VRK2
787         P34741 Recurrence 2.23e-02  0.2340 -2.39            SDC2
788         Q86U38 Recurrence 2.23e-02  0.2340  2.39            NOP9
789         A5YKK6 Recurrence 2.24e-02  0.2340  2.39           CNOT1
790         P14314 Recurrence 2.24e-02  0.2340 -2.39          PRKCSH
791         Q7Z6M1 Recurrence 2.24e-02  0.2340 -2.39          RABEPK
792         Q8IWL1 Recurrence 2.24e-02  0.2340 -2.39          SFTPA2
793         Q92901 Recurrence 2.25e-02  0.2340  2.38           RPL3L
794         O00142 Recurrence 2.26e-02  0.2340  2.38             TK2
795         O95487 Recurrence 2.26e-02  0.2340  2.38          SEC24B
796         Q16706 Recurrence 2.26e-02  0.2340  2.38          MAN2A1
797         Q6UXY8 Recurrence 2.26e-02  0.2340 -2.38            TMC5
798         Q86X83 Recurrence 2.26e-02  0.2340  2.38          COMMD2
799         Q96NB2 Recurrence 2.27e-02  0.2350  2.38           SFXN2
800         Q9BW61 Recurrence 2.27e-02  0.2340  2.38            DDA1
801         Q96BY6 Recurrence 2.28e-02  0.2350  2.38          DOCK10
802         Q9BQ67 Recurrence 2.28e-02  0.2350  2.38           GRWD1
803         Q9Y312 Recurrence 2.28e-02  0.2350  2.38            AAR2
804         Q5T7V8 Recurrence 2.29e-02  0.2350 -2.38           GORAB
805         Q96KA5 Recurrence 2.29e-02  0.2350  2.38         CLPTM1L
806         Q9Y5K6 Recurrence 2.31e-02  0.2360 -2.37           CD2AP
807         O15357 Recurrence 2.32e-02  0.2370  2.37          INPPL1
808         P19388 Recurrence 2.32e-02  0.2370  2.37          POLR2E
809         O75896 Recurrence 2.33e-02  0.2370 -2.37           TUSC2
810         Q641Q2 Recurrence 2.33e-02  0.2370 -2.37         WASHC2A
811         Q8NFC6 Recurrence 2.33e-02  0.2370 -2.37          BOD1L1
812         O95678 Recurrence 2.34e-02  0.2370 -2.37           KRT75
813         P48736 Recurrence 2.34e-02  0.2370  2.37          PIK3CG
814         Q9H270 Recurrence 2.34e-02  0.2370  2.37           VPS11
815         P00488 Recurrence 2.35e-02  0.2370  2.37           F13A1
816         Q06547 Recurrence 2.35e-02  0.2370 -2.37          GABPB1
817         Q15599 Recurrence 2.35e-02  0.2370 -2.37        SLC9A3R2
818         Q8NAP3 Recurrence 2.35e-02  0.2370 -2.37          ZBTB38
819         O43493 Recurrence 2.36e-02  0.2370 -2.36          TGOLN2
820         P62070 Recurrence 2.36e-02  0.2370 -2.36           RRAS2
821         Q8N2K0 Recurrence 2.36e-02  0.2370  2.36          ABHD12
822         Q9Y6K5 Recurrence 2.39e-02  0.2400  2.36            OAS3
823         O95716 Recurrence 2.40e-02  0.2410 -2.36           RAB3D
824         Q9NYF8 Recurrence 2.41e-02  0.2410 -2.35          BCLAF1
825         A6NJ78 Recurrence 2.42e-02  0.2420  2.35         METTL15
826         Q6NUK4 Recurrence 2.42e-02  0.2420  2.35           REEP3
827         O14569 Recurrence 2.43e-02  0.2420  2.35        CYB561D2
828         Q5TBB1 Recurrence 2.43e-02  0.2420  2.35        RNASEH2B
829         Q9UBC5 Recurrence 2.43e-02  0.2420 -2.35           MYO1A
830         Q5JTZ9 Recurrence 2.44e-02  0.2420  2.35           AARS2
831         Q6UWE0 Recurrence 2.44e-02  0.2420  2.35          LRSAM1
832         Q7L3B6 Recurrence 2.45e-02  0.2430  2.35         CDC37L1
833         Q96BR5 Recurrence 2.46e-02  0.2440  2.35            COA7
834         P26583 Recurrence 2.47e-02  0.2440 -2.34           HMGB2
835         Q9BTU6 Recurrence 2.47e-02  0.2440  2.34          PI4K2A
836         Q9BYB4 Recurrence 2.47e-02  0.2440  2.34           GNB1L
837         Q9NV88 Recurrence 2.48e-02  0.2440  2.34           INTS9
838         Q9NXR1 Recurrence 2.48e-02  0.2440 -2.34            NDE1
839         O95359 Recurrence 2.49e-02  0.2440 -2.34           TACC2
840         P83916 Recurrence 2.49e-02  0.2440 -2.34            CBX1
841         Q01105 Recurrence 2.49e-02  0.2440 -2.34             SET
842         Q01433 Recurrence 2.49e-02  0.2440  2.34           AMPD2
843         O94819 Recurrence 2.50e-02  0.2450  2.34         KBTBD11
844         P11021 Recurrence 2.51e-02  0.2450 -2.34           HSPA5
845         O00567 Recurrence 2.52e-02  0.2450  2.34           NOP56
846         P43121 Recurrence 2.52e-02  0.2450 -2.34            MCAM
847         Q9H2A7 Recurrence 2.52e-02  0.2450  2.34          CXCL16
848         Q9Y5Y2 Recurrence 2.52e-02  0.2450  2.34           NUBP2
849         O43772 Recurrence 2.54e-02  0.2460  2.33        SLC25A20
850         Q06187 Recurrence 2.54e-02  0.2460  2.33             BTK
851         Q86WT1 Recurrence 2.54e-02  0.2460 -2.33          TTC30A
852         Q9H6T0 Recurrence 2.54e-02  0.2460  2.33           ESRP2
853         Q9NTM9 Recurrence 2.54e-02  0.2460  2.33            CUTC
854         Q9UBI6 Recurrence 2.54e-02  0.2460 -2.33           GNG12
855         Q9Y6M5 Recurrence 2.54e-02  0.2460  2.33         SLC30A1
856         Q96AJ9 Recurrence 2.55e-02  0.2460 -2.33           VTI1A
857         P17405 Recurrence 2.56e-02  0.2460  2.33           SMPD1
858         P48444 Recurrence 2.56e-02  0.2460  2.33           ARCN1
859         O00203 Recurrence 2.57e-02  0.2460  2.33           AP3B1
860         P37802 Recurrence 2.58e-02  0.2470 -2.33          TAGLN2
861         P98170 Recurrence 2.58e-02  0.2470  2.33            XIAP
862         Q8TC07 Recurrence 2.58e-02  0.2470  2.33         TBC1D15
863         Q9HD42 Recurrence 2.58e-02  0.2470 -2.33          CHMP1A
864         O76021 Recurrence 2.59e-02  0.2470  2.32          RSL1D1
865         Q15773 Recurrence 2.60e-02  0.2480 -2.32            MLF2
866         Q86Y82 Recurrence 2.61e-02  0.2480 -2.32           STX12
867         O60279 Recurrence 2.62e-02  0.2480 -2.32           SUSD5
868         P52943 Recurrence 2.62e-02  0.2480 -2.32           CRIP2
869         Q58FG1 Recurrence 2.62e-02  0.2480  2.32       HSP90AA4P
870         Q8N5K1 Recurrence 2.62e-02  0.2480  2.32           CISD2
871         Q8TBQ9 Recurrence 2.62e-02  0.2480  2.32        TMEM167A
872         O95219 Recurrence 2.63e-02  0.2480  2.32            SNX4
873         Q02880 Recurrence 2.63e-02  0.2480  2.32           TOP2B
874         Q86UW6 Recurrence 2.63e-02  0.2490 -2.32           N4BP2
875         Q8IVH4 Recurrence 2.64e-02  0.2490  2.32            MMAA
876         P14923 Recurrence 2.65e-02  0.2500 -2.31             JUP
877         Q5JTH9 Recurrence 2.65e-02  0.2500  2.31           RRP12
878         Q9HAT2 Recurrence 2.66e-02  0.2500  2.31            SIAE
879         P01703 Recurrence 2.67e-02  0.2510 -2.31        IGLV1-40
880         Q9Y2S7 Recurrence 2.67e-02  0.2510  2.31         POLDIP2
881         P50570 Recurrence 2.69e-02  0.2520  2.31            DNM2
882         Q5JVS0 Recurrence 2.69e-02  0.2520  2.31           HABP4
883         Q9H410 Recurrence 2.69e-02  0.2520 -2.31            DSN1
884         Q3YBM2 Recurrence 2.70e-02  0.2520  2.31        TMEM176B
885         Q9UBB9 Recurrence 2.70e-02  0.2520  2.30          TFIP11
886         Q9Y371 Recurrence 2.70e-02  0.2520 -2.31         SH3GLB1
887         Q86WU2 Recurrence 2.71e-02  0.2520  2.30            LDHD
888         Q9P2R3 Recurrence 2.71e-02  0.2520  2.30          ANKFY1
889         P35606 Recurrence 2.72e-02  0.2520  2.30           COPB2
890         Q8IWB9 Recurrence 2.72e-02  0.2520  2.30            TEX2
891         Q8NBI5 Recurrence 2.72e-02  0.2520  2.30         SLC43A3
892         O75352 Recurrence 2.74e-02  0.2530  2.30           MPDU1
893         P04156 Recurrence 2.74e-02  0.2530 -2.30            PRNP
894         Q8WUH2 Recurrence 2.75e-02  0.2540  2.30        TGFBRAP1
895         P32969 Recurrence 2.76e-02  0.2540  2.30          RPL9P9
896         P53814 Recurrence 2.77e-02  0.2550 -2.29            SMTN
897         Q15746 Recurrence 2.77e-02  0.2550 -2.30            MYLK
898         Q7Z6J4 Recurrence 2.77e-02  0.2550  2.29            FGD2
899         Q8WZA9 Recurrence 2.78e-02  0.2550 -2.29            IRGQ
900         P26440 Recurrence 2.79e-02  0.2550  2.29             IVD
901         Q56VL3 Recurrence 2.80e-02  0.2560  2.29          OCIAD2
902         Q96CU9 Recurrence 2.80e-02  0.2560  2.29         FOXRED1
903         Q9HBR0 Recurrence 2.80e-02  0.2560 -2.29        SLC38A10
904         Q04695 Recurrence 2.81e-02  0.2560 -2.29           KRT17
905         Q8NBJ7 Recurrence 2.82e-02  0.2560 -2.29           SUMF2
906         Q9Y4P3 Recurrence 2.82e-02  0.2560  2.29            TBL2
907         P13861 Recurrence 2.83e-02  0.2560 -2.28         PRKAR2A
908         P53985 Recurrence 2.83e-02  0.2560  2.28         SLC16A1
909         P61586 Recurrence 2.83e-02  0.2560 -2.28            RHOA
910         Q6NUM9 Recurrence 2.83e-02  0.2560  2.28          RETSAT
911         Q96Q42 Recurrence 2.83e-02  0.2560  2.29            ALS2
912         Q9H0J9 Recurrence 2.83e-02  0.2560  2.29          PARP12
913         O60568 Recurrence 2.84e-02  0.2560  2.28           PLOD3
914         O94829 Recurrence 2.84e-02  0.2560 -2.28           IPO13
915         Q96S97 Recurrence 2.84e-02  0.2560  2.28           MYADM
916         P08709 Recurrence 2.85e-02  0.2560 -2.28              F7
917         Q93052 Recurrence 2.86e-02  0.2570 -2.28             LPP
918         Q9BUK0 Recurrence 2.86e-02  0.2570  2.28          CHCHD7
919         Q9NWM3 Recurrence 2.86e-02  0.2570 -2.28          CUEDC1
920         Q9Y223 Recurrence 2.86e-02  0.2570  2.28             GNE
921         O75694 Recurrence 2.87e-02  0.2570  2.28          NUP155
922         Q8IWU2 Recurrence 2.87e-02  0.2570 -2.28           LMTK2
923         Q8N0U4 Recurrence 2.87e-02  0.2570  2.28         FAM185A
924         O43865 Recurrence 2.88e-02  0.2570  2.28          AHCYL1
925         P46778 Recurrence 2.88e-02  0.2570  2.28           RPL21
926         Q9Y485 Recurrence 2.88e-02  0.2570  2.28           DMXL1
927         O00499 Recurrence 2.89e-02  0.2570 -2.28            BIN1
928         O15264 Recurrence 2.89e-02  0.2570  2.28          MAPK13
929         Q92563 Recurrence 2.90e-02  0.2570 -2.28          SPOCK2
930         Q96A72 Recurrence 2.90e-02  0.2570  2.27          MAGOHB
931         Q96J01 Recurrence 2.90e-02  0.2570  2.27           THOC3
932         P51689 Recurrence 2.91e-02  0.2580 -2.27            ARSD
933         P78346 Recurrence 2.91e-02  0.2570  2.27           RPP30
934         O00273 Recurrence 2.92e-02  0.2580 -2.27            DFFA
935         O75251 Recurrence 2.92e-02  0.2580  2.27          NDUFS7
936         O15228 Recurrence 2.93e-02  0.2580  2.27           GNPAT
937         P22695 Recurrence 2.93e-02  0.2580  2.27          UQCRC2
938         Q9Y5S9 Recurrence 2.93e-02  0.2580 -2.27           RBM8A
939         Q13242 Recurrence 2.94e-02  0.2580 -2.27           SRSF9
940         Q9Y2Q0 Recurrence 2.94e-02  0.2580  2.27          ATP8A1
941         P42356 Recurrence 2.95e-02  0.2580  2.27           PI4KA
942         Q13576 Recurrence 2.95e-02  0.2580  2.27          IQGAP2
943         Q9NYB0 Recurrence 2.95e-02  0.2580 -2.27         TERF2IP
944         P49589 Recurrence 2.96e-02  0.2590 -2.27           CARS1
945         P47895 Recurrence 2.98e-02  0.2600  2.26         ALDH1A3
946         P50895 Recurrence 2.98e-02  0.2600 -2.26            BCAM
947         P60983 Recurrence 2.98e-02  0.2600 -2.26            GMFB
948         P62979 Recurrence 2.99e-02  0.2600 -2.26          RPS27A
949         Q9NRP0 Recurrence 2.99e-02  0.2600  2.26            OSTC
950         P09496 Recurrence 3.01e-02  0.2610 -2.26            CLTA
951         P53677 Recurrence 3.01e-02  0.2610  2.26           AP3M2
952  Q6DN03;Q6DRA6 Recurrence 3.01e-02  0.2610 -2.26 H2BC20P;H2BC19P
953         P98164 Recurrence 3.02e-02  0.2610  2.26            LRP2
954         Q9NXS2 Recurrence 3.02e-02  0.2610  2.26           QPCTL
955         Q8IXB1 Recurrence 3.05e-02  0.2630  2.25         DNAJC10
956         Q96S44 Recurrence 3.05e-02  0.2630  2.25          TP53RK
957         O75954 Recurrence 3.06e-02  0.2630 -2.25          TSPAN9
958         O94919 Recurrence 3.06e-02  0.2630 -2.25          ENDOD1
959         P78527 Recurrence 3.06e-02  0.2630  2.25           PRKDC
960         P56556 Recurrence 3.07e-02  0.2630  2.25          NDUFA6
961         P62701 Recurrence 3.07e-02  0.2630  2.25           RPS4X
962         Q13085 Recurrence 3.07e-02  0.2630  2.25           ACACA
963         Q15819 Recurrence 3.07e-02  0.2630 -2.25          UBE2V2
964         Q96EY5 Recurrence 3.07e-02  0.2630  2.25          MVB12A
965         Q9BSF4 Recurrence 3.07e-02  0.2630  2.25          TIMM29
966         P51572 Recurrence 3.08e-02  0.2630 -2.25          BCAP31
967         Q9NR50 Recurrence 3.08e-02  0.2630  2.25          EIF2B3
968         Q96AT9 Recurrence 3.09e-02  0.2630 -2.25             RPE
969         Q9BY89 Recurrence 3.09e-02  0.2630 -2.25        KIAA1671
970         O94880 Recurrence 3.10e-02  0.2630  2.24           PHF14
971         O96005 Recurrence 3.10e-02  0.2630  2.25          CLPTM1
972         Q12769 Recurrence 3.10e-02  0.2630  2.25          NUP160
973         P53621 Recurrence 3.11e-02  0.2630  2.24            COPA
974         Q7Z3B4 Recurrence 3.11e-02  0.2640  2.24           NUP54
975         O00422 Recurrence 3.13e-02  0.2650 -2.24           SAP18
976         P61421 Recurrence 3.14e-02  0.2650  2.24        ATP6V0D1
977         Q99996 Recurrence 3.14e-02  0.2650 -2.24           AKAP9
978         O60812 Recurrence 3.15e-02  0.2650  2.24        HNRNPCL1
979         O95528 Recurrence 3.15e-02  0.2650 -2.24         SLC2A10
980         Q96IG2 Recurrence 3.15e-02  0.2650  2.24          FBXL20
981         Q9NQT4 Recurrence 3.15e-02  0.2650  2.24          EXOSC5
982         Q8NCN5 Recurrence 3.16e-02  0.2650  2.24            PDPR
983         Q9NRH2 Recurrence 3.16e-02  0.2650 -2.24            SNRK
984         P52948 Recurrence 3.17e-02  0.2650  2.24           NUP98
985         P62277 Recurrence 3.17e-02  0.2650  2.24           RPS13
986         Q8IZ69 Recurrence 3.17e-02  0.2650  2.23          TRMT2A
987         Q9Y6W5 Recurrence 3.17e-02  0.2650 -2.24           WASF2
988         P53582 Recurrence 3.18e-02  0.2650  2.23          METAP1
989         P55899 Recurrence 3.18e-02  0.2650  2.23           FCGRT
990         O14925 Recurrence 3.19e-02  0.2650  2.23          TIMM23
991         Q96C90 Recurrence 3.19e-02  0.2650 -2.23        PPP1R14B
992         Q96MH2 Recurrence 3.19e-02  0.2650 -2.23          HEXIM2
993         Q9H5Z1 Recurrence 3.19e-02  0.2650  2.23           DHX35
994         O76094 Recurrence 3.20e-02  0.2650  2.23           SRP72
995         P51858 Recurrence 3.20e-02  0.2650 -2.23            HDGF
996         Q7Z7H8 Recurrence 3.21e-02  0.2660  2.23          MRPL10
997         Q9UPU7 Recurrence 3.21e-02  0.2660  2.23         TBC1D2B
998         P35908 Recurrence 3.22e-02  0.2660 -2.23            KRT2
999         Q9H8H3 Recurrence 3.22e-02  0.2660  2.23         METTL7A
1000        O95382 Recurrence 3.23e-02  0.2660  2.23          MAP3K6
1001        Q16644 Recurrence 3.23e-02  0.2660  2.23        MAPKAPK3
1002        Q8N1N4 Recurrence 3.23e-02  0.2660 -2.23           KRT78
1003        Q9P2G1 Recurrence 3.23e-02  0.2660 -2.23          ANKIB1
1004        P13928 Recurrence 3.25e-02  0.2670 -2.22           ANXA8
1005        Q8NFG4 Recurrence 3.25e-02  0.2670  2.22            FLCN
1006        P60174 Recurrence 3.27e-02  0.2680 -2.22            TPI1
1007        P61769 Recurrence 3.27e-02  0.2680 -2.22             B2M
1008        Q12768 Recurrence 3.27e-02  0.2680  2.22          WASHC5
1009        Q15365 Recurrence 3.28e-02  0.2680  2.22           PCBP1
1010        Q5T0N5 Recurrence 3.28e-02  0.2680 -2.22          FNBP1L
1011        Q9NPB8 Recurrence 3.28e-02  0.2680  2.22          GPCPD1
1012        P19256 Recurrence 3.30e-02  0.2690 -2.22            CD58
1013        P39880 Recurrence 3.30e-02  0.2690 -2.22            CUX1
1014        Q9BYN0 Recurrence 3.31e-02  0.2690 -2.22           SRXN1
1015        Q6P4A8 Recurrence 3.32e-02  0.2700  2.21           PLBD1
1016        Q7Z4F1 Recurrence 3.32e-02  0.2700  2.21           LRP10
1017        P23193 Recurrence 3.33e-02  0.2700 -2.21           TCEA1
1018        P37108 Recurrence 3.33e-02  0.2700 -2.21           SRP14
1019        Q15075 Recurrence 3.33e-02  0.2700 -2.21            EEA1
1020        P29728 Recurrence 3.34e-02  0.2700  2.21            OAS2
1021        Q96QK1 Recurrence 3.34e-02  0.2700  2.21           VPS35
1022        Q9P1Y5 Recurrence 3.34e-02  0.2700  2.21         CAMSAP3
1023        P54253 Recurrence 3.35e-02  0.2700 -2.21           ATXN1
1024        Q16585 Recurrence 3.35e-02  0.2700 -2.21            SGCB
1025        Q96DR5 Recurrence 3.35e-02  0.2700  2.21          BPIFA2
1026 P14678;P63162 Recurrence 3.36e-02  0.2700  2.21     SNRPB;SNRPN
1027        O43572 Recurrence 3.37e-02  0.2710  2.21          AKAP10
1028        Q9NWZ3 Recurrence 3.38e-02  0.2710  2.21           IRAK4
1029        Q9UHR4 Recurrence 3.38e-02  0.2710 -2.21        BAIAP2L1
1030        Q4G0F5 Recurrence 3.39e-02  0.2710  2.21          VPS26B
1031        Q8NCW6 Recurrence 3.39e-02  0.2710  2.21         GALNT11
1032        Q6NUK1 Recurrence 3.40e-02  0.2720  2.20        SLC25A24
1033        P54252 Recurrence 3.41e-02  0.2720 -2.20           ATXN3
1034        Q15006 Recurrence 3.41e-02  0.2720  2.20            EMC2
1035        P29374 Recurrence 3.42e-02  0.2720 -2.20          ARID4A
1036        Q9NXF1 Recurrence 3.42e-02  0.2720  2.20           TEX10
1037        Q9Y4Y9 Recurrence 3.42e-02  0.2720 -2.20            LSM5
1038        P80303 Recurrence 3.43e-02  0.2720 -2.20           NUCB2
1039        Q8IVE3 Recurrence 3.43e-02  0.2720 -2.20         PLEKHH2
1040        Q8TB24 Recurrence 3.43e-02  0.2720  2.20            RIN3
1041        Q6PI48 Recurrence 3.44e-02  0.2720  2.20           DARS2
1042        O14737 Recurrence 3.45e-02  0.2720 -2.20           PDCD5
1043        P43353 Recurrence 3.46e-02  0.2720 -2.20         ALDH3B1
1044        Q5VYK3 Recurrence 3.46e-02  0.2720  2.20           ECPAS
1045        Q7L5Y1 Recurrence 3.46e-02  0.2720  2.20          ENOSF1
1046        Q8NF50 Recurrence 3.46e-02  0.2720  2.20           DOCK8
1047        Q92608 Recurrence 3.46e-02  0.2720  2.20           DOCK2
1048        Q8TB61 Recurrence 3.47e-02  0.2730  2.19         SLC35B2
1049        Q93091 Recurrence 3.47e-02  0.2720  2.20          RNASE6
1050        Q99704 Recurrence 3.47e-02  0.2720  2.20            DOK1
1051        Q9Y2P8 Recurrence 3.47e-02  0.2720  2.20            RCL1
1052        O43837 Recurrence 3.51e-02  0.2750  2.19           IDH3B
1053        Q07157 Recurrence 3.52e-02  0.2760 -2.19            TJP1
1054        Q7Z422 Recurrence 3.53e-02  0.2760 -2.19           SZRD1
1055        Q9NQ89 Recurrence 3.53e-02  0.2760 -2.19         C12orf4
1056        Q9UNF1 Recurrence 3.53e-02  0.2760 -2.19          MAGED2
1057        Q08345 Recurrence 3.55e-02  0.2760 -2.18            DDR1
1058        Q8IUX7 Recurrence 3.55e-02  0.2760 -2.18           AEBP1
1059        P08648 Recurrence 3.56e-02  0.2760 -2.18           ITGA5
1060        Q13315 Recurrence 3.56e-02  0.2760  2.18             ATM
1061        Q3KQZ1 Recurrence 3.56e-02  0.2760  2.18        SLC25A35
1062        Q8IVM0 Recurrence 3.56e-02  0.2760 -2.18          CCDC50
1063        Q9NUP1 Recurrence 3.56e-02  0.2760  2.18         BLOC1S4
1064        O95273 Recurrence 3.57e-02  0.2760 -2.18         CCNDBP1
1065        Q9NWU1 Recurrence 3.57e-02  0.2760  2.18            OXSM
1066        P52630 Recurrence 3.58e-02  0.2760  2.18           STAT2
1067        Q96AA3 Recurrence 3.58e-02  0.2760  2.18            RFT1
1068        Q9Y315 Recurrence 3.58e-02  0.2760  2.18            DERA
1069        P09497 Recurrence 3.59e-02  0.2760 -2.18            CLTB
1070        Q5T5Y3 Recurrence 3.59e-02  0.2760 -2.18         CAMSAP1
1071        Q6UXB2 Recurrence 3.59e-02  0.2760  2.18          CXCL17
1072        Q9UBK8 Recurrence 3.59e-02  0.2760  2.18            MTRR
1073        Q9Y5X1 Recurrence 3.59e-02  0.2760  2.18            SNX9
1074        Q5RI15 Recurrence 3.61e-02  0.2770  2.18           COX20
1075        O94776 Recurrence 3.62e-02  0.2780  2.18            MTA2
1076        P28289 Recurrence 3.63e-02  0.2780 -2.18           TMOD1
1077        Q9H0V1 Recurrence 3.63e-02  0.2780  2.17         TMEM168
1078        Q9NZC7 Recurrence 3.63e-02  0.2780 -2.17            WWOX
1079        Q9BRX8 Recurrence 3.64e-02  0.2780  2.17          PRXL2A
1080        O14939 Recurrence 3.65e-02  0.2790  2.17            PLD2
1081        Q96CN9 Recurrence 3.65e-02  0.2790 -2.17            GCC1
1082        P57772 Recurrence 3.66e-02  0.2790  2.17          EEFSEC
1083        Q86UT6 Recurrence 3.66e-02  0.2790  2.17           NLRX1
1084        Q9Y4C8 Recurrence 3.66e-02  0.2790  2.17           RBM19
1085        Q08722 Recurrence 3.67e-02  0.2790 -2.17            CD47
1086        P06310 Recurrence 3.68e-02  0.2790 -2.17        IGKV2-30
1087        Q8N6R0 Recurrence 3.68e-02  0.2790  2.17         METTL13
1088        Q9H063 Recurrence 3.68e-02  0.2790 -2.17            MAF1
1089        Q9NQG5 Recurrence 3.68e-02  0.2790 -2.17          RPRD1B
1090        A4QPH2 Recurrence 3.69e-02  0.2790  2.17         PI4KAP2
1091        P15529 Recurrence 3.69e-02  0.2790 -2.17            CD46
1092        Q4L180 Recurrence 3.69e-02  0.2790 -2.17         FILIP1L
1093        O15371 Recurrence 3.70e-02  0.2790  2.17           EIF3D
1094        P63261 Recurrence 3.73e-02  0.2810 -2.16           ACTG1
1095        Q9H3U7 Recurrence 3.74e-02  0.2810  2.16           SMOC2
1096    A0A0B4J1V0 Recurrence 3.75e-02  0.2820 -2.16        IGHV3-15
1097        P17568 Recurrence 3.75e-02  0.2820  2.16          NDUFB7
1098        P19838 Recurrence 3.75e-02  0.2820  2.16           NFKB1
1099        Q99805 Recurrence 3.75e-02  0.2820  2.16          TM9SF2
1100        Q14011 Recurrence 3.76e-02  0.2820 -2.16           CIRBP
1101        Q9BSJ5 Recurrence 3.76e-02  0.2820  2.16        C17orf80
1102        O43290 Recurrence 3.77e-02  0.2820 -2.16           SART1
1103        Q6P2E9 Recurrence 3.77e-02  0.2820  2.16            EDC4
1104        Q9P253 Recurrence 3.78e-02  0.2830  2.16           VPS18
1105        Q7L3T8 Recurrence 3.79e-02  0.2830  2.16           PARS2
1106        Q9HAS0 Recurrence 3.79e-02  0.2830  2.16        C17orf75
1107        P08581 Recurrence 3.80e-02  0.2830  2.15             MET
1108        P55263 Recurrence 3.80e-02  0.2830  2.15             ADK
1109        Q92843 Recurrence 3.80e-02  0.2830 -2.15          BCL2L2
1110        Q12979 Recurrence 3.81e-02  0.2830  2.15             ABR
1111        P51674 Recurrence 3.82e-02  0.2830 -2.15           GPM6A
1112        Q6PIU2 Recurrence 3.82e-02  0.2830  2.15           NCEH1
1113        O95400 Recurrence 3.83e-02  0.2840 -2.15          CD2BP2
1114        Q9H9J2 Recurrence 3.83e-02  0.2840  2.15          MRPL44
1115        Q9BWM7 Recurrence 3.84e-02  0.2840  2.15           SFXN3
1116        Q9NRS6 Recurrence 3.84e-02  0.2840 -2.15           SNX15
1117        Q7Z478 Recurrence 3.85e-02  0.2840  2.15           DHX29
1118        Q9Y6N5 Recurrence 3.85e-02  0.2840  2.15            SQOR
1119        Q5SWX8 Recurrence 3.86e-02  0.2850  2.15            ODR4
1120        P02766 Recurrence 3.87e-02  0.2850 -2.15             TTR
1121        Q9GZV5 Recurrence 3.87e-02  0.2850 -2.15           WWTR1
1122        O14734 Recurrence 3.92e-02  0.2880  2.14           ACOT8
1123        P26196 Recurrence 3.92e-02  0.2880  2.14            DDX6
1124        Q9H444 Recurrence 3.92e-02  0.2880 -2.14          CHMP4B
1125        P18669 Recurrence 3.93e-02  0.2880 -2.14           PGAM1
1126        Q9UPN4 Recurrence 3.93e-02  0.2880 -2.14          CEP131
1127        Q8WYA0 Recurrence 3.94e-02  0.2890 -2.14           IFT81
1128        O00267 Recurrence 3.95e-02  0.2890  2.14          SUPT5H
1129        O15355 Recurrence 3.95e-02  0.2890 -2.14           PPM1G
1130        P98082 Recurrence 3.96e-02  0.2890  2.14            DAB2
1131        Q8TCF1 Recurrence 3.96e-02  0.2890  2.14          ZFAND1
1132        Q9Y3Q8 Recurrence 3.96e-02  0.2890 -2.14         TSC22D4
1133        P01920 Recurrence 3.97e-02  0.2890  2.13        HLA-DQB1
1134        Q9BYD3 Recurrence 3.97e-02  0.2890  2.13           MRPL4
1135        Q2M2I3 Recurrence 3.98e-02  0.2890  2.13          FAM83E
1136        Q08211 Recurrence 3.99e-02  0.2890  2.13            DHX9
1137        Q8IUR7 Recurrence 3.99e-02  0.2890  2.13           ARMC8
1138        Q8WV41 Recurrence 3.99e-02  0.2890  2.13           SNX33
1139        Q08AM6 Recurrence 4.00e-02  0.2890  2.13           VAC14
1140        Q6NZY4 Recurrence 4.00e-02  0.2890 -2.13          ZCCHC8
1141        Q9Y584 Recurrence 4.01e-02  0.2900  2.13          TIMM22
1142        Q12913 Recurrence 4.02e-02  0.2900 -2.13           PTPRJ
1143        Q96J02 Recurrence 4.02e-02  0.2900  2.13            ITCH
1144        Q7L5L3 Recurrence 4.05e-02  0.2920 -2.13           GDPD3
1145        Q96HW7 Recurrence 4.06e-02  0.2920  2.12           INTS4
1146        O75122 Recurrence 4.07e-02  0.2920 -2.12          CLASP2
1147        Q15067 Recurrence 4.07e-02  0.2920  2.12           ACOX1
1148        Q9H2G2 Recurrence 4.07e-02  0.2920 -2.12             SLK
1149        Q9NPY3 Recurrence 4.07e-02  0.2920 -2.12            CD93
1150        Q13546 Recurrence 4.08e-02  0.2920  2.12           RIPK1
1151        Q9H9A5 Recurrence 4.08e-02  0.2920  2.12          CNOT10
1152        Q9P2J5 Recurrence 4.08e-02  0.2920  2.12           LARS1
1153        Q14847 Recurrence 4.09e-02  0.2920 -2.12           LASP1
1154        Q6UXV4 Recurrence 4.09e-02  0.2920  2.12           APOOL
1155        P04062 Recurrence 4.10e-02  0.2930  2.12            GBA1
1156        Q8IZP0 Recurrence 4.10e-02  0.2930 -2.12            ABI1
1157        Q5VTL8 Recurrence 4.11e-02  0.2930  2.12         PRPF38B
1158        P52895 Recurrence 4.12e-02  0.2930 -2.12          AKR1C2
1159        Q5EBL4 Recurrence 4.12e-02  0.2930 -2.12          RILPL1
1160        Q9NUI1 Recurrence 4.13e-02  0.2930  2.12           DECR2
1161        Q9Y5Z7 Recurrence 4.13e-02  0.2930  2.12           HCFC2
1162        O15020 Recurrence 4.14e-02  0.2930 -2.12          SPTBN2
1163        P62857 Recurrence 4.14e-02  0.2930 -2.12           RPS28
1164        Q9NRW3 Recurrence 4.14e-02  0.2930  2.12        APOBEC3C
1165        Q9UI08 Recurrence 4.14e-02  0.2930  2.11             EVL
1166        Q96DT6 Recurrence 4.15e-02  0.2940  2.11           ATG4C
1167        Q9UBB5 Recurrence 4.16e-02  0.2940 -2.11            MBD2
1168        Q15124 Recurrence 4.17e-02  0.2940 -2.11            PGM5
1169        Q460N5 Recurrence 4.17e-02  0.2940  2.11          PARP14
1170        Q8N4P3 Recurrence 4.17e-02  0.2940 -2.11           HDDC3
1171        Q6DN90 Recurrence 4.19e-02  0.2950  2.11          IQSEC1
1172        Q15643 Recurrence 4.20e-02  0.2950 -2.11          TRIP11
1173        Q8WXH0 Recurrence 4.20e-02  0.2950 -2.11           SYNE2
1174        O43731 Recurrence 4.21e-02  0.2950  2.11          KDELR3
1175        Q53FZ2 Recurrence 4.21e-02  0.2950  2.11           ACSM3
1176        Q969E8 Recurrence 4.21e-02  0.2950 -2.11            TSR2
1177        Q9BV36 Recurrence 4.21e-02  0.2950  2.11            MLPH
1178        P30085 Recurrence 4.22e-02  0.2950 -2.11           CMPK1
1179        Q0VGL1 Recurrence 4.22e-02  0.2950  2.11         LAMTOR4
1180        Q3KQU3 Recurrence 4.22e-02  0.2950 -2.11          MAP7D1
1181        Q9ULR0 Recurrence 4.22e-02  0.2950 -2.11            ISY1
1182        Q6NUQ4 Recurrence 4.23e-02  0.2950  2.11         TMEM214
1183        P57087 Recurrence 4.24e-02  0.2950 -2.10            JAM2
1184        Q9H7D7 Recurrence 4.24e-02  0.2950  2.10           WDR26
1185        Q9HD45 Recurrence 4.24e-02  0.2950  2.10          TM9SF3
1186        Q9NQW6 Recurrence 4.24e-02  0.2950 -2.10            ANLN
1187        P24928 Recurrence 4.25e-02  0.2950  2.10          POLR2A
1188        Q15057 Recurrence 4.25e-02  0.2950  2.10           ACAP2
1189        O75976 Recurrence 4.26e-02  0.2950 -2.10             CPD
1190        Q3SY69 Recurrence 4.26e-02  0.2950  2.10         ALDH1L2
1191        Q969N2 Recurrence 4.26e-02  0.2950  2.10            PIGT
1192        Q7Z5R6 Recurrence 4.27e-02  0.2950  2.10         APBB1IP
1193        Q9BS92 Recurrence 4.27e-02  0.2950 -2.10       NIPSNAP3B
1194        Q9NY59 Recurrence 4.27e-02  0.2950  2.10           SMPD3
1195        P28347 Recurrence 4.28e-02  0.2950 -2.10           TEAD1
1196        Q6NZI2 Recurrence 4.28e-02  0.2950 -2.10          CAVIN1
1197        Q9BXR0 Recurrence 4.28e-02  0.2950  2.10           QTRT1
1198        Q8TCD5 Recurrence 4.29e-02  0.2950 -2.10            NT5C
1199        O75023 Recurrence 4.30e-02  0.2960  2.10          LILRB5
1200        Q5VZ89 Recurrence 4.30e-02  0.2960  2.10         DENND4C
1201        Q969Q5 Recurrence 4.32e-02  0.2960  2.10           RAB24
1202        Q9BQL6 Recurrence 4.32e-02  0.2960 -2.10          FERMT1
1203        Q9NR56 Recurrence 4.32e-02  0.2960  2.10           MBNL1
1204        P12268 Recurrence 4.33e-02  0.2970  2.09          IMPDH2
1205        Q15147 Recurrence 4.34e-02  0.2970 -2.09           PLCB4
1206        Q8IWA4 Recurrence 4.35e-02  0.2970  2.09            MFN1
1207        Q7L5N7 Recurrence 4.36e-02  0.2980  2.09          LPCAT2
1208        Q9H1Y0 Recurrence 4.36e-02  0.2980  2.09            ATG5
1209        P31942 Recurrence 4.37e-02  0.2980 -2.09         HNRNPH3
1210        Q96NC0 Recurrence 4.37e-02  0.2980 -2.09           ZMAT2
1211        Q9H1H9 Recurrence 4.37e-02  0.2980 -2.09          KIF13A
1212        O75937 Recurrence 4.38e-02  0.2980 -2.09          DNAJC8
1213        Q9HD26 Recurrence 4.39e-02  0.2980  2.09            GOPC
1214        Q9Y3A6 Recurrence 4.39e-02  0.2980  2.09           TMED5
1215        Q96CW1 Recurrence 4.40e-02  0.2980  2.09           AP2M1
1216        O75146 Recurrence 4.41e-02  0.2980 -2.09           HIP1R
1217        P35749 Recurrence 4.41e-02  0.2980 -2.09           MYH11
1218        P78368 Recurrence 4.41e-02  0.2980 -2.09         CSNK1G2
1219        Q6GMV2 Recurrence 4.41e-02  0.2980  2.09           SMYD5
1220        Q8IXH6 Recurrence 4.41e-02  0.2980 -2.09        TP53INP2
1221        P11597 Recurrence 4.42e-02  0.2980 -2.09            CETP
1222        Q32NB8 Recurrence 4.42e-02  0.2980  2.08            PGS1
1223        P45954 Recurrence 4.43e-02  0.2980  2.08          ACADSB
1224        Q14980 Recurrence 4.43e-02  0.2980 -2.08           NUMA1
1225        Q6P589 Recurrence 4.43e-02  0.2980  2.08       TNFAIP8L2
1226        P07741 Recurrence 4.45e-02  0.2990  2.08            APRT
1227        Q86UU1 Recurrence 4.45e-02  0.2990 -2.08          PHLDB1
1228        Q8TCT9 Recurrence 4.45e-02  0.2990  2.08            HM13
1229        Q16352 Recurrence 4.48e-02  0.3010 -2.08             INA
1230        O00625 Recurrence 4.49e-02  0.3010 -2.08             PIR
1231        P0DTE1 Recurrence 4.49e-02  0.3010 -2.08      IGHV3-38-3
1232        Q14012 Recurrence 4.49e-02  0.3010 -2.08           CAMK1
1233        Q8TF71 Recurrence 4.49e-02  0.3010  2.08        SLC16A10
1234        Q9NXE4 Recurrence 4.50e-02  0.3010  2.08           SMPD4
1235        P22307 Recurrence 4.51e-02  0.3010  2.08            SCP2
1236        Q8TEV9 Recurrence 4.51e-02  0.3010  2.08           SMCR8
1237        O95352 Recurrence 4.52e-02  0.3010  2.08            ATG7
1238        P54108 Recurrence 4.52e-02  0.3010 -2.08          CRISP3
1239        O43150 Recurrence 4.53e-02  0.3010 -2.07           ASAP2
1240        P29692 Recurrence 4.53e-02  0.3020 -2.07           EEF1D
1241        P00352 Recurrence 4.54e-02  0.3020 -2.07         ALDH1A1
1242        P28827 Recurrence 4.54e-02  0.3020 -2.07           PTPRM
1243        O95163 Recurrence 4.55e-02  0.3020  2.07            ELP1
1244        P17152 Recurrence 4.55e-02  0.3020  2.07          TMEM11
1245        P22087 Recurrence 4.55e-02  0.3020  2.07             FBL
1246        P49411 Recurrence 4.56e-02  0.3020  2.07            TUFM
1247        Q7Z7M0 Recurrence 4.58e-02  0.3030 -2.07           MEGF8
1248        O00187 Recurrence 4.60e-02  0.3040 -2.07           MASP2
1249        O76070 Recurrence 4.60e-02  0.3040 -2.07            SNCG
1250        P42702 Recurrence 4.60e-02  0.3040 -2.07            LIFR
1251        P51617 Recurrence 4.62e-02  0.3040  2.06           IRAK1
1252        P61758 Recurrence 4.62e-02  0.3040 -2.07            VBP1
1253        Q7LGA3 Recurrence 4.62e-02  0.3040  2.07          HS2ST1
1254        Q9NS69 Recurrence 4.62e-02  0.3040  2.06          TOMM22
1255        O75340 Recurrence 4.63e-02  0.3040 -2.06           PDCD6
1256        P57088 Recurrence 4.63e-02  0.3040  2.06          TMEM33
1257        Q8N5D0 Recurrence 4.63e-02  0.3040 -2.06           WDTC1
1258        Q9UDR5 Recurrence 4.63e-02  0.3040  2.06            AASS
1259        P18846 Recurrence 4.64e-02  0.3040 -2.06            ATF1
1260        Q9ULT0 Recurrence 4.64e-02  0.3040  2.06           TTC7A
1261        P49736 Recurrence 4.66e-02  0.3050 -2.06            MCM2
1262        Q9H4X1 Recurrence 4.67e-02  0.3050 -2.06            RGCC
1263        O95782 Recurrence 4.68e-02  0.3050  2.06           AP2A1
1264        Q9H3K2 Recurrence 4.68e-02  0.3050  2.06           GHITM
1265        Q9Y365 Recurrence 4.68e-02  0.3050  2.06         STARD10
1266        Q5TC12 Recurrence 4.69e-02  0.3060  2.06          ATPAF1
1267        Q9UN36 Recurrence 4.69e-02  0.3050 -2.06           NDRG2
1268        O95477 Recurrence 4.70e-02  0.3060 -2.06           ABCA1
1269        P61981 Recurrence 4.70e-02  0.3060 -2.06           YWHAG
1270        P82909 Recurrence 4.70e-02  0.3060 -2.06          MRPS36
1271        P53007 Recurrence 4.73e-02  0.3070  2.05         SLC25A1
1272        P20592 Recurrence 4.74e-02  0.3070  2.05             MX2
1273        P29972 Recurrence 4.75e-02  0.3080 -2.05            AQP1
1274        Q9HAV4 Recurrence 4.75e-02  0.3080  2.05            XPO5
1275        Q9NZN3 Recurrence 4.77e-02  0.3090 -2.05            EHD3
1276        P12270 Recurrence 4.79e-02  0.3100 -2.05             TPR
1277        Q53GG5 Recurrence 4.80e-02  0.3100 -2.05          PDLIM3
1278        O60828 Recurrence 4.81e-02  0.3100 -2.05           PQBP1
1279        Q8WTS6 Recurrence 4.81e-02  0.3100 -2.05           SETD7
1280        Q96MW5 Recurrence 4.81e-02  0.3100  2.05            COG8
1281        Q7L0Y3 Recurrence 4.82e-02  0.3100  2.04         TRMT10C
1282        Q8TBP6 Recurrence 4.82e-02  0.3100  2.05        SLC25A40
1283 B7ZAQ6;P0CG08 Recurrence 4.83e-02  0.3100  2.04   GPR89A;GPR89B
1284        Q14562 Recurrence 4.83e-02  0.3100  2.04            DHX8
1285        Q8TBX8 Recurrence 4.83e-02  0.3100  2.04         PIP4K2C
1286        P10644 Recurrence 4.85e-02  0.3110 -2.04         PRKAR1A
1287        P18031 Recurrence 4.85e-02  0.3110  2.04           PTPN1
1288        Q99487 Recurrence 4.85e-02  0.3110  2.04          PAFAH2
1289        Q9UNX4 Recurrence 4.85e-02  0.3110  2.04            WDR3
1290        Q9H4Z3 Recurrence 4.86e-02  0.3110  2.04           PCIF1
1291        Q92560 Recurrence 4.87e-02  0.3110 -2.04            BAP1
1292        Q9BY50 Recurrence 4.88e-02  0.3120  2.04          SEC11C
1293        Q9H5X1 Recurrence 4.88e-02  0.3120  2.04          CIAO2A
1294        O43854 Recurrence 4.90e-02  0.3120 -2.04           EDIL3
1295        Q58WW2 Recurrence 4.90e-02  0.3120 -2.04           DCAF6
1296        Q9NV31 Recurrence 4.90e-02  0.3120  2.04            IMP3
1297        O14578 Recurrence 4.91e-02  0.3120 -2.04             CIT
1298        Q13888 Recurrence 4.91e-02  0.3120  2.04          GTF2H2
1299        Q68DK7 Recurrence 4.92e-02  0.3130 -2.04            MSL1
1300        P42285 Recurrence 4.93e-02  0.3130  2.03           MTREX
1301        Q92859 Recurrence 4.93e-02  0.3130 -2.03            NEO1
1302        Q9P2E9 Recurrence 4.93e-02  0.3130 -2.03           RRBP1
1303        Q96IZ0 Recurrence 4.94e-02  0.3130 -2.03            PAWR
1304        Q6P1X6 Recurrence 4.95e-02  0.3130  2.03         C8orf82
1305        Q9HC38 Recurrence 4.95e-02  0.3130 -2.03           GLOD4
1306        Q9NWS0 Recurrence 4.98e-02  0.3140 -2.03          PIH1D1
1307        P36551 Recurrence 4.99e-02  0.3150  2.03            CPOX
1308        Q6P1A2 Recurrence 4.99e-02  0.3150  2.03          LPCAT3
1309        Q96BP2 Recurrence 4.99e-02  0.3150  2.03          CHCHD1
1310        P09467 Recurrence 5.00e-02  0.3150  2.03            FBP1
1311        Q9UI12 Recurrence 5.00e-02  0.3150  2.03         ATP6V1H</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proNormalRes_Klin$sig.feat.heat</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/heatmap_sigFeats_combined_proNormal_Klin.png&#39;, a,
#        device = &#39;png&#39;, dpi = 400, height = 10, width = 12)</code></pre>
<p>Visualize top significant features identified in Normal samples (ns:
p &gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01, <span
class="math inline">\(***\)</span>: p &lt;= 0.001)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Klin$data
smpAnnoTab &lt;- proTissueRes_Klin$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proNormalRes_Klin$sig.feat.tab
num_sigFeats &lt;- 6
gpLevel = c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;, &#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;)
condiCol = c(&#39;firebrick&#39;, &#39;grey50&#39;) #c(Normal, Tumor)
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(newVar1 = paste0(Gene, &#39; (&#39;, Var1, &#39;)&#39;), #\n
                newVar1 = factor(newVar1, levels = unique(newVar1)))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Recur_Condi = paste0(Recurrence, &#39;_&#39;, Condition),
                Recur_Condi = factor(Recur_Condi, levels = gpLevel),
                Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
ggplot(topSigFeatDat, aes(x=Recur_Condi, y=Abundance, col=Condition, fill=Recurrence)) +
  geom_boxplot(alpha = 1, outlier.shape = NA, linewidth = 1) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             comparisons = list(c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;), c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;)),
                             label = &#39;p.signif&#39;, tip.length = 0.015, bracket.size = 0.7, size = 4) +
  labs(x = &#39;Recurrence&#39;) +
  scale_color_manual(values = condiCol) +
  scale_fill_manual(values = c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  facet_wrap(vars(newVar1), scales = &#39;free&#39;) +
  th +
  # theme(strip.text = element_text(size = 13, face = &#39;bold&#39;),
  #            axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))
  theme(strip.text = element_text(size = 14, face = &#39;bold&#39;), #size = 26, margin = margin(0.5,0,0.5,0, &#39;cm&#39;)
        axis.title = element_text(size = 28),
        axis.text.x = element_text(angle = 60, hjust = 1, vjust = 1, size = 14), #size = 24
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/boxplot_topSigFeats_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 9, width = 16)</code></pre>
<pre class="r"><code># Prepare proteins of interest and recurrence information
datTab &lt;- t(proNormalRes_Klin$SOA.res$data) %&gt;%
  as_tibble(rownames = &#39;Sample&#39;) %&gt;%
  left_join(proNormalRes_Klin$SOA.res$smpMetadata) %&gt;%
  dplyr::select(Sample, P41208, P63104, Recurrence)

# Compute ROC-AUC score
response &lt;- ifelse(datTab$Recurrence == &#39;Yes&#39;, yes = &#39;1&#39;, no = &#39;0&#39;) %&gt;%
  factor(levels = c(&#39;1&#39;, &#39;0&#39;))
predictor &lt;- datTab$P41208
rocRes &lt;- pROC::roc(response = response, predictor = predictor, plot = T,
                    legacy.axes = T, print.auc = T, print.auc.x = 0.4,
                    xlab = substitute(paste(bold(&#39;False positive rate&#39;))),
                    ylab = substitute(paste(bold(&#39;True positive rate&#39;))),
                    main = &#39;ROC Curve for P41208 (CETN2)&#39;,
                    cex.lab = 1.8, cex.axis = 1.2, cex.main = 1.8, col = &#39;#377eb8&#39;,
                    lwd = 6, direction = &#39;&lt;&#39;, quiet = T)</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># pROC::ci.auc(rocRes)

predictor &lt;- datTab$P63104
rocRes &lt;- pROC::roc(response = response, predictor = predictor, plot = T,
                    legacy.axes = T, print.auc = T, print.auc.x = 0.4,
                    xlab = substitute(paste(bold(&#39;False positive rate&#39;))),
                    ylab = substitute(paste(bold(&#39;True positive rate&#39;))),
                    main = &#39;ROC Curve for P63104 (YWHAZ)&#39;,
                    cex.lab = 1.8, cex.axis = 1.2, cex.main = 1.8, col = &#39;#377eb8&#39;,
                    lwd = 6, direction = &#39;&lt;&#39;, quiet = T)</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-40-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># pROC::ci.auc(rocRes)</code></pre>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proNormalRes_Klin$pval.hist +
  labs(title = &#39;Klin. Combined Normal Tissue DIA Proteomics&#39;) +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 18))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-41-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/histogram_pVals_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 9, width = 11)</code></pre>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proNormalRes_Klin$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2    pVal pValAdj Stat   Test
1 PC1 (11.1%) Recurrence 0.00485   0.179 -3.0 T-test
2  PC9 (3.2%) Recurrence 0.02690   0.498 -2.3 T-test</code></pre>
<p>Visualize significant PCs and display their top features with
greatest absolute loadings</p>
<pre class="r"><code>pcTab &lt;- proNormalRes_Klin$SOA.res$pcTab %&gt;%
  dplyr::mutate(Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                Label = dplyr::case_when(Recurrence %in% &#39;Yes&#39; &amp; `PC1 (11.1%)` &gt; 0 ~ Sample,
                                         Recurrence %in% &#39;No&#39; &amp; `PC1 (11.1%)` &lt; 0 ~ Sample))
# pcTab$Label[!is.na(pcTab$Label)]
# Yes: ZSOOQ4_NG | No: CH51FFJ5_NG

ggplot(pcTab, aes(x=Recurrence, y=`PC1 (11.1%)`, col=Recurrence, fill=Recurrence, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 4, show.legend = F) +
  # geom_text_repel(show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             size = 9, show.legend = F, hjust = 0.4, vjust = 1) +
  scale_color_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  scale_fill_manual(values=c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  th +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 20),
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/boxplot_pca_PC1_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)

# Compute ROC-AUC score
response &lt;- ifelse(pcTab$Recurrence == &#39;Yes&#39;, yes = &#39;1&#39;, no = &#39;0&#39;) %&gt;%
  factor(levels = c(&#39;1&#39;, &#39;0&#39;))
predictor &lt;- pcTab$`PC1 (11.1%)`
rocRes &lt;- pROC::roc(response = response, predictor = predictor, plot = T,
                    legacy.axes = T, print.auc = T, print.auc.x = 0.4,
                    xlab = substitute(paste(bold(&#39;False positive rate&#39;))),
                    ylab = substitute(paste(bold(&#39;True positive rate&#39;))),
                    main = &#39;ROC Curve for PC1&#39;,
                    cex.lab = 1.8, cex.axis = 1.2, cex.main = 1.8, col = &#39;#377eb8&#39;,
                    lwd = 4, direction = &#39;&lt;&#39;, quiet = T)</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-43-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pROC::ci.auc(rocRes)</code></pre>
<pre><code>95% CI: 0.5914-0.9007 (DeLong)</code></pre>
<pre class="r"><code>proNormalRes_Klin$pc.top.feat.tab$PC1</code></pre>
<pre><code>                Feature Loading              Gene
1                Q53QV2  0.0624               LBH
2                Q9BXV9  0.0602              GON7
3                Q96RS0  0.0551              TGS1
4  P0DP23;P0DP24;P0DP25  0.0521 CALM1;CALM2;CALM3
5                P27482  0.0506            CALML3
6                Q9H4X1  0.0484              RGCC
7                Q6ZTK2  0.0480              &lt;NA&gt;
8                Q14257  0.0473              RCN2
9                Q9NQP4  0.0473             PFDN4
10               P03973 -0.0470              SLPI
11               Q9NSB2  0.0466             KRT84
12               Q9NZD4  0.0465              AHSP
13               Q96D15  0.0445              RCN3
14               Q9H799  0.0442           CPLANE1
15               Q5T8I9  0.0441            HENMT1
16               Q9P121  0.0433               NTM
17               P62760  0.0429             VSNL1
18               P84157  0.0423             MXRA7
19               O43852  0.0421              CALU
20               A5D8V6  0.0419            VPS37C
21           A0A0C4DH43  0.0412         IGHV2-70D
22               P67936  0.0411              TPM4
23               P69892  0.0410              HBG2
24               P50120  0.0409              RBP2
25               P12829  0.0401              MYL4
26               O60925  0.0401             PFDN1
27               P06454  0.0394              PTMA
28               P14555 -0.0393           PLA2G2A
29               P07951  0.0390              TPM2
30               P20962  0.0389              PTMS</code></pre>
<pre class="r"><code>proNormalRes_Klin$sig.pc.dist$PC9</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-43-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proNormalRes_Klin$pc.top.feat.tab$PC9</code></pre>
<pre><code>   Feature Loading     Gene
1   Q96SL4  0.1190     GPX7
2   Q13075  0.1080     NAIP
3   P35527  0.1000     KRT9
4   P04264  0.0976     KRT1
5   P08779  0.0975    KRT16
6   P50120 -0.0932     RBP2
7   P19012 -0.0904    KRT15
8   Q13206  0.0849    DDX10
9   Q8N699  0.0839    MYCT1
10  P04259  0.0832    KRT6B
11  Q02763 -0.0823      TEK
12  O43915 -0.0822    VEGFD
13  Q96RS0 -0.0792     TGS1
14  P13647  0.0776     KRT5
15  P13645  0.0744    KRT10
16  Q8TB72  0.0743     PUM2
17  Q04695  0.0740    KRT17
18  Q9NZT1  0.0706   CALML5
19  P35908  0.0686     KRT2
20  Q86YZ3  0.0684     HRNR
21  Q15800 -0.0653    MSMO1
22  Q9UJV9 -0.0633    DDX41
23  P52292 -0.0629    KPNA2
24  P01127  0.0624    PDGFB
25  O15550  0.0623    KDM6A
26  Q9HC52 -0.0621     CBX8
27  Q9UMF0  0.0620    ICAM5
28  Q5T749  0.0605     KPRP
29  Q14449 -0.0600    GRB14
30  P80748  0.0596 IGLV3-21</code></pre>
<p>Enrichment analysis<br />
Reference: Hallmark gene sets<br />
<strong>t-values</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proNormalRes_Klin$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, Stat) %&gt;%
  tibble::column_to_rownames(&#39;Var1&#39;)
featAnno_Klin &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::rename(Gene = PG.Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)

# Hallmark gene sets
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)
# Run GSEA
gseaResH_tVal_Klin &lt;- clusterProfiler::GSEA(geneList = rankedGeneList_Klin, TERM2GENE = msigTabH,
                                            minGSSize = 10, maxGSSize = 500,
                                            pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                            by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH_tVal_Klin@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 10) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;),
                Description = stringr::str_remove(Description, &#39;HALLMARK_&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.7, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12),
        title = element_text(face = &#39;bold&#39;, size = 12))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
      # axis.text.x = element_text(size = 18),
      # axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
      # legend.title = element_text(size = 20),
      # legend.text = element_text(size = 16),
      # panel.grid.minor = element_blank())
# ggsave(&#39;./output/Discovery/2nd_TAC/gsea_tVal_combined_proNormal_Klin_H.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p><strong>PC1 loadings</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proNormalRes_Klin$SOA.res$pcaRes@loadings %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, PC1) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
featAnno_Klin &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::rename(Gene = PG.Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)
rankedGeneList_Klin &lt;- sort(-rankedGeneList_Klin, decreasing = T)

# Hallmark gene sets
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)
# Run GSEA
gseaResH_pc1_Klin &lt;- clusterProfiler::GSEA(geneList = rankedGeneList_Klin, TERM2GENE = msigTabH,
                                           minGSSize = 10, maxGSSize = 500,
                                           pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                           by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH_pc1_Klin@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 10) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;),
                Description = stringr::str_remove(Description, &#39;HALLMARK_&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.7, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12),
        title = element_text(face = &#39;bold&#39;, size = 12))</code></pre>
<p><img src="figure/Discovery_02_soa_DIA_combined.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
  #       axis.text.x = element_text(size = 18),
  #       axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
  #       legend.title = element_text(size = 20),
  #       legend.text = element_text(size = 16),
  #       panel.grid.minor = element_blank())
# ggsave(&#39;./output/Discovery/2nd_TAC/gsea_pc1_combined_proNormal_Klin_H.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] visdat_0.6.0                vsn_3.70.0                 
 [3] lubridate_1.9.3             forcats_1.0.0              
 [5] stringr_1.5.1               purrr_1.0.2                
 [7] readr_2.1.5                 tidyr_1.3.1                
 [9] tibble_3.2.1                tidyverse_2.0.0            
[11] SummarizedExperiment_1.32.0 GenomicRanges_1.54.1       
[13] GenomeInfoDb_1.38.8         MatrixGenerics_1.14.0      
[15] matrixStats_1.3.0           clusterProfiler_4.10.1     
[17] msigdbr_7.5.1               org.Hs.eg.db_3.18.0        
[19] AnnotationDbi_1.64.1        IRanges_2.36.0             
[21] S4Vectors_0.40.2            ggvenn_0.1.10              
[23] dplyr_1.1.4                 ggrepel_0.9.5              
[25] ggplot2_3.5.1               ggplotify_0.1.2            
[27] pheatmap_1.0.12             proDA_1.16.0               
[29] limma_3.58.1                sva_3.50.0                 
[31] BiocParallel_1.36.0         genefilter_1.84.0          
[33] mgcv_1.9-1                  nlme_3.1-164               
[35] pcaMethods_1.94.0           Biobase_2.62.0             
[37] BiocGenerics_0.48.1         workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] splines_4.3.2           later_1.3.2             bitops_1.0-7           
  [4] polyclip_1.10-6         preprocessCore_1.64.0   pROC_1.18.5            
  [7] XML_3.99-0.16.1         lifecycle_1.0.4         rstatix_0.7.2          
 [10] edgeR_4.0.16            rprojroot_2.0.4         processx_3.8.4         
 [13] lattice_0.22-6          MASS_7.3-60.0.1         backports_1.4.1        
 [16] magrittr_2.0.3          sass_0.4.9              rmarkdown_2.26         
 [19] jquerylib_0.1.4         yaml_2.3.8              httpuv_1.6.15          
 [22] cowplot_1.1.3           DBI_1.2.2               RColorBrewer_1.1-3     
 [25] abind_1.4-5             zlibbioc_1.48.2         ggraph_2.2.1           
 [28] RCurl_1.98-1.14         yulab.utils_0.1.4       tweenr_2.0.3           
 [31] git2r_0.33.0            GenomeInfoDbData_1.2.11 enrichplot_1.22.0      
 [34] tidytree_0.4.6          annotate_1.80.0         codetools_0.2-20       
 [37] DelayedArray_0.28.0     DOSE_3.28.2             ggforce_0.4.2          
 [40] tidyselect_1.2.1        aplot_0.2.2             farver_2.1.1           
 [43] viridis_0.6.5           jsonlite_1.8.8          tidygraph_1.3.1        
 [46] survival_3.5-8          tools_4.3.2             treeio_1.26.0          
 [49] Rcpp_1.0.12             glue_1.7.0              gridExtra_2.3          
 [52] SparseArray_1.2.4       xfun_0.43               qvalue_2.34.0          
 [55] withr_3.0.0             BiocManager_1.30.22     fastmap_1.1.1          
 [58] fansi_1.0.6             callr_3.7.6             digest_0.6.35          
 [61] timechange_0.3.0        R6_2.5.1                gridGraphics_0.5-1     
 [64] colorspace_2.1-0        GO.db_3.18.0            RSQLite_2.3.6          
 [67] utf8_1.2.4              generics_0.1.3          data.table_1.15.4      
 [70] graphlayouts_1.1.1      httr_1.4.7              S4Arrays_1.2.1         
 [73] scatterpie_0.2.2        whisker_0.4.1           pkgconfig_2.0.3        
 [76] gtable_0.3.5            blob_1.2.4              XVector_0.42.0         
 [79] shadowtext_0.1.3        htmltools_0.5.8.1       carData_3.0-5          
 [82] fgsea_1.28.0            scales_1.3.0            png_0.1-8              
 [85] ggfun_0.1.4             knitr_1.46              rstudioapi_0.16.0      
 [88] tzdb_0.4.0              reshape2_1.4.4          cachem_1.0.8           
 [91] parallel_4.3.2          HDO.db_0.99.1           pillar_1.9.0           
 [94] vctrs_0.6.5             ggpubr_0.6.0            promises_1.3.0         
 [97] car_3.1-2               xtable_1.8-4            evaluate_0.23          
[100] cli_3.6.2               locfit_1.5-9.9          compiler_4.3.2         
[103] rlang_1.1.3             crayon_1.5.2            ggsignif_0.6.4         
[106] labeling_0.4.3          ps_1.7.6                affy_1.80.0            
[109] getPass_0.2-4           plyr_1.8.9              fs_1.6.3               
[112] stringi_1.8.3           viridisLite_0.4.2       babelgene_22.9         
[115] munsell_0.5.1           Biostrings_2.70.3       lazyeval_0.2.2         
[118] GOSemSim_2.28.1         Matrix_1.6-5            hms_1.1.3              
[121] patchwork_1.2.0         bit64_4.0.5             KEGGREST_1.42.0        
[124] statmod_1.5.0           extraDistr_1.10.0       highr_0.10             
[127] broom_1.0.5             igraph_2.0.3            memoise_2.0.1          
[130] affyio_1.72.0           bslib_0.7.0             ggtree_3.10.1          
[133] fastmatch_1.1-4         bit_4.0.5               ape_5.8                
[136] gson_0.1.0             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
