<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />

<meta name="date" content="2024-10-08" />

<title>Feature selection: Machine learning models (with feature filtering)</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Feature selection: Machine learning models
(with feature filtering)</h1>
<h4 class="author">Qian-Wu Liao</h4>
<h4 class="date">2024-10-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertreed14c3b30dc8e1ed76939ad796689988321a5f8b6targetblankd14c3b3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/d14c3b30dc8e1ed76939ad796689988321a5f8b6" target="_blank">d14c3b3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertreed14c3b30dc8e1ed76939ad796689988321a5f8b6targetblankd14c3b3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/d14c3b30dc8e1ed76939ad796689988321a5f8b6" target="_blank">d14c3b3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Discovery/
    Ignored:    data/MethodDev/
    Ignored:    data/aliquot_metadata.xlsx
    Ignored:    data/metadata_latest/
    Ignored:    data/patient_metadata.xlsx
    Ignored:    data/sample_metadata.xlsx
    Ignored:    output/.DS_Store
    Ignored:    output/Discovery/
    Ignored:    output/Junyan/
    Ignored:    output/MethodDev/
    Ignored:    output/SC_meeting/

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/data_availability.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R

Unstaged changes:
    Modified:   analysis/MDev_Dis_02_soa_annotated_untargeted.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/FeatSelection_ML_FeatFilt.Rmd</code>) and HTML
(<code>docs/FeatSelection_ML_FeatFilt.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/209dfece89b54d399918912772cec601439b8d97/analysis/FeatSelection_ML_FeatFilt.Rmd" target="_blank">209dfec</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-10-10
</td>
<td>
Compare model performance trained on data either unadjusted or adjusted
by SVs and do additional feature filtering on subdatasets before model
training -&gt; Data adjusted by SVs constructed from whole tissue
datasets with interaction term (Condition:Recurrence) included is chosen
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Do feature selection with machine
learning models trained on combined data of cohorts (Method Development
and Discovery) to identify potential biomarkers for predicting lung
cancer recurrence. Additional feature filtering is conducted on
subdatasets to exclude risk of biases introduced by imputation method
(missForest) used in our workflow, which can impede true but weaker
signals contributed by features more complete. ML models used currently
include lasso logistic regression, random forest, and XGBoost.
</font></p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;spsUtil&#39;)
library(ggh4x)
library(DT)
library(&#39;randomForest&#39;)
library(&#39;missForest&#39;)
library(xgboost)
library(&#39;caret&#39;)
library(&#39;glmnet&#39;)
library(ggvenn)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)
source(&#39;./code/ml_funcs.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<div id="ttp-ag-klin." class="section level1">
<h1>TTP (AG Klin.)</h1>
<p>Tumor Tissue DIA Proteomics from AG Klingmüller</p>
<div id="lasso-logistic-regression" class="section level2">
<h2>Lasso logistic regression</h2>
<p><font size='5'> <strong>Stability selection</strong> </font></p>
<p><strong>Performance statistics of 1000 models</strong></p>
<pre class="r"><code># Load preprocessed data
proTissue_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
proTumor_Klin &lt;- proTissue_Klin[, colData(proTissue_Klin)$Condition %in% &#39;Tumor&#39;]
# Load SOA results
proTumorRes_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVsByProTissueInt.rds&#39;)
# Prepare adjusted data
adjProTumor_Klin &lt;- proTumor_Klin
assay(adjProTumor_Klin) &lt;- proTumorRes_Klin$SOA.res$dataCorrect
# Do additional feature filtering to remove features observed in less than 50% of samples
ifRmFeats &lt;- apply(assay(adjProTumor_Klin), 1, function(featVec) {
  sum(is.na(featVec)) / length(featVec) &gt; 0.5
})
adjProTumor_Klin &lt;- adjProTumor_Klin[!ifRmFeats,]

# Load imputed adjusted data
impuAdjProTumor_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/impuData/featFilt_impuAdjProTumorVsn_SVsByProTissueInt.rds&#39;)

# Train models iteratively
# Prepare significant feature list
sigFeatList &lt;- proTumorRes_Klin$sig.feat.tab$Var1
# lrRes &lt;- iterLogisR(adjProTumor_Klin, doImputation = T, doInitFeatSelection = T, sigFeatList = sigFeatList,
#                     doFeatClustering = T, iter = 1000, cvFold = 5, cvMeasure = &#39;auc&#39;)
# saveRDS(lrRes, &#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProTumor_0.9r_randSplit_1000_5fold_auc_lmin_Klin.rds&#39;)
# saveRDS(lrRes$impuSE, &#39;./data/Discovery/AG_Klingmueller/impuData/featFilt_impuAdjProTumorVsn_SVsByProTissueInt.rds&#39;)
lrRes &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProTumor_0.9r_randSplit_1000_5fold_auc_lmin_Klin.rds&#39;)

# Report summarized scores
# Prepare predictions and ground truths where results from useless models are removed
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  pred &lt;- lrRes$lrRes$y_pred[-uselessModels]
  truth &lt;- lrRes$lrRes$y_truth[-uselessModels]
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  pred &lt;- lrRes$lrRes$y_pred
  truth &lt;- lrRes$lrRes$y_truth
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
suppressWarnings(summarizePredPower(pred, truth, auc_roc))</code></pre>
<pre><code>Mean AUC-ROC score: 0.95
95% CI: [0.7666667, 1]</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Top important features that are most frequently
selected</strong></p>
<pre class="r"><code># Visualize top important features
# Prepare feature annotation table
featAnno &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Feature&#39;)
topImpoFeatViz &lt;- vizTopImpoFeatsLR(lrRes, featAnno = featAnno, num_p1TopFeats = 30)
topImpoFeatViz$pick</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Full feature importance table</strong></p>
<pre class="r"><code>DT::datatable(topImpoFeatViz$fullImpoFeatTab)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-fe366aef0ca255b13f2e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-fe366aef0ca255b13f2e">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116"],["P15090","P49326","P51808","P53420","P08236","P00750","Q8IUE6","Q6KB66","Q5XKP0","Q9UGM5","P05186","A0M8Q6","P35442","O15014","O75112","P21397","P50238","P01599","Q6UY14","P80188","Q14914","Q08426","Q9UKZ4","P06310","Q13509","Q5T6X4","Q8IVH2","Q9UDX4","Q99959","Q6FIF0","Q8NB90","Q9BW30","P20036","Q06278","Q9UGU5","Q9Y625","Q9C086","P51687","P10515","Q7L804","P57087","Q86WI3","P53677","Q15063","Q9H0F6","P20742","Q53ET0","Q9P0M6","Q92604","P26440","A0A075B6K0","Q96FN9","Q5T5C0","P14780","P01903","Q58FG1","P08473","P23141","Q15493","Q99943","Q9H082","Q12767","Q7LG56","P01742","Q6ZSY5","Q99988","Q9NPB8","Q9Y653","Q8IV50","Q9UHL4","Q8WXG6","Q9NWY4","O15021","Q9Y296","Q8NDV1","Q96AB3","Q8TEB9","Q96EK6","Q9NPG8","P16930","Q13822","Q15124","Q15283","P51784","Q96PZ0","Q9P2T1","O75923","P30047","Q9BYT8","O75071","P30626","Q99538","P21964","Q16627","Q96D46","Q96DU7","Q96HJ9","O75128","O94956","P22105","P29474","P60033","Q03405","Q4G0N4","Q6ZNL6","Q8IUZ5","Q92526","Q96CD0","Q9H0N0","Q9NP66","Q9P291","Q9UKY3","O60237","P63096","Q14520","Q9UBP9"],["FABP4","FMO5","DYNLT3","COL4A4","GUSB","PLAT","H2AC21","KRT80","MICOS13","FETUB","ALPL","IGLC7","THBS2","ZNF609","LDB3","MAOA","CRIP1","IGKV1-17","ADAMTSL4","LCN2","PTGR1","EHHADH","TENM1","IGKV2-30","TUBB3","FAM162B","FOXP4","SEC14L3","PKP2","ZFAND6","SPATA5","TPPP3","HLA-DPA1","AOX1","HMGXB4","GPC6","INO80B","SUOX","DLAT","RAB11FIP2","JAM2","NLRC5","AP3M2","POSTN","SHARPIN","PZP","CRTC2","MACROH2A2","LPGAT1","IVD","IGLV3-16","DTD2","STXBP5","MMP9","HLA-DRA","HSP90AA4P","MME","CES1","RGN","AGPAT1","RAB33B","TMEM94","RRM2B","IGHV1-69","PPP1R3F","GDF15","GPCPD1","ADGRG1","LYSMD2","DPP7","MADD","HPF1","MAST4","TRAPPC4","ST6GALNAC3","ISOC2","RHBDD1","GNPNAT1","ZDHHC4","FAH","ENPP2","PGM5","RASA2","USP11","PUS7","GMPR2","DYSF","GCHFR","NLN","EFCAB14","SRI","LGMN","COMT","CCL14","NMD3","ITPKC","FMC1","COBL","SLCO2B1","TNXB","NOS3","CD81","PLAUR","NADK2","FGD5","PHYKPL","CCT6B","FBXL8","RAB6C","HMG20A","ARMCX1","CES1P1","PPP1R12B","GNAI1","HABP2","GULP1"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"P01906",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[0.981,0.953,0.922,0.881,0.88,0.868,0.868,0.854,0.845,0.836,0.821,0.819,0.796,0.795,0.75,0.738,0.6840000000000001,0.65,0.637,0.62,0.596,0.551,0.547,0.539,0.519,0.511,0.48,0.463,0.44,0.399,0.396,0.385,0.374,0.346,0.318,0.265,0.252,0.232,0.225,0.219,0.213,0.212,0.202,0.196,0.165,0.161,0.157,0.14,0.133,0.124,0.111,0.1,0.092,0.08799999999999999,0.08,0.075,0.07099999999999999,0.066,0.062,0.06,0.053,0.046,0.037,0.031,0.028,0.028,0.024,0.021,0.02,0.019,0.018,0.016,0.015,0.013,0.012,0.012,0.011,0.011,0.011,0.008999999999999999,0.008999999999999999,0.008999999999999999,0.008,0.007,0.006,0.006,0.005,0.005,0.005,0.004,0.004,0.004,0.003,0.003,0.003,0.003,0.003,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.001,0.001,0.001,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Feature<\/th>\n      <th>Annotation<\/th>\n      <th>Cluster<\/th>\n      <th>PickRate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Feature","targets":1},{"name":"Annotation","targets":2},{"name":"Cluster","targets":3},{"name":"PickRate","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><font size='5'> <strong>Systematic training and evaluation</strong>
</font></p>
<p><strong>Performance of models trained with different numbers of top
important features</strong></p>
<pre class="r"><code># Systematically train models with different numbers of top important features
# sysTrainModels &lt;- doSysTrainLogisR(lrRes, impuAdjProTumor_Klin, max_numTopFeats = 50)
# saveRDS(sysTrainModels, &#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Klin.rds&#39;)
sysTrainModels &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Klin.rds&#39;)

# Visualize systematic training
vizSysTrainModelPerf(sysTrainModels, numTopFeats = 30)</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_Stats_proTumor.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p>=&gt; Top 8 important features are selected, resulting in model with
AUC of 0.99 [0.92,1].<br />
=&gt; Top 14 important features are selected, resulting in model with
AUC of 0.988 [0.92,1].<br />
=&gt; Top 18 important features are selected, resulting in model with
AUC of 0.986 [0.92,1].</p>
<p><strong>Noninformative models that demonstrate effects of top
important features</strong></p>
<pre class="r"><code>se &lt;- impuAdjProTumor_Klin
# Train models without top important features, aka noninformative models
numTopImpoFeats &lt;- c(5, 10, 15, 20)
# for (num in numTopImpoFeats) {
#   topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:num]
#   # Train lasso logistic regression model
#   reducedData &lt;- lrRes$featClusters$reducedData
#   x &lt;- t(reducedData[!rownames(reducedData) %in% topImpoFeats,])
#   y &lt;- colData(se)$Recurrence
#   lrResNonInfo &lt;- runLogisR(x, y, targetClass = &#39;Yes&#39;, iter = 100, regularized_method = &#39;lasso&#39;,
#                             cvFold = 5, cvMeasure = &#39;auc&#39;, used_lambda = &#39;lambda.min&#39;,
#                             trainSet_ratio = 0.8, split_method = &#39;random split&#39;,
#                             plot_ROC = F, save_model = T)
#   saveRDS(lrResNonInfo, paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
#                                &#39;_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Klin.rds&#39;))
# }

# Display performance of noninformative models
meanAUC &lt;- c()
lowerAUC &lt;- c()
upperAUC &lt;- c()
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
ci &lt;- calcCI(auc_roc, bootstrap = T)
lowerAUC &lt;- c(lowerAUC, ci[1])
upperAUC &lt;- c(upperAUC, ci[2])
for (num in numTopImpoFeats) {
  lrResNonInfo &lt;- readRDS(paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
                                 &#39;_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Klin.rds&#39;))
  uselessModels &lt;- which(lrResNonInfo$nNonZero == 0)
  if (length(uselessModels) != 0) {
    auc_roc &lt;- lrResNonInfo$auc_roc[-uselessModels]
  } else {
    auc_roc &lt;- lrResNonInfo$auc_roc
  }
  meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
  ci &lt;- calcCI(auc_roc, bootstrap = T)
  lowerAUC &lt;- c(lowerAUC, ci[1])
  upperAUC &lt;- c(upperAUC, ci[2])
}
aucStatTab &lt;- data.frame(FeatSpace = c(&#39;Complete&#39;, paste0(&#39;-Top &#39;, numTopImpoFeats)),
                         MeanAUC = meanAUC, UpperAUC = upperAUC, LowerAUC = lowerAUC) %&gt;%
  dplyr::mutate(FeatSpace = factor(FeatSpace, levels = FeatSpace))

ggplot(aucStatTab, aes(x=FeatSpace, y=MeanAUC)) +
  geom_bar(stat = &#39;identity&#39;, position = position_dodge(), alpha = 0.9) +
  geom_errorbar(aes(ymin=LowerAUC, ymax=UpperAUC), width = 0.4) +
  labs(x = &#39;Feature space used to train 100 models&#39;, y = &#39;Mean AUC-ROC&#39;,
       title = &#39;Noninformative models&#39;) +
  th</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='5'> <strong>Investigation of core features</strong>
</font><br />
<em>Keep TGFBR3 in mind</em></p>
<p><font size='3'> Imputed data: </font><br />
<strong>Heatmap where samples are ordered by linear regression
scores</strong></p>
<pre class="r"><code>coreFeatPlotsImpu &lt;- vizCoreFeats(lrRes, sysTrainModels, impuAdjProTumor_Klin,
                                  heatmap_numCoreFeats = 8, colFeatAnno = &#39;Gene&#39;,
                                  scatter_twoCoreFeats = c(&#39;FABP4&#39;, &#39;FMO5&#39;),
                                  fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                                  cellwidth = 6, cellheight = 18)
coreFeatPlotsImpu$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_heatmap_proNormal_impu.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><strong>Scatterplot of two core features</strong></p>
<pre class="r"><code>coreFeatPlotsImpu$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_top1&amp;3_proTumor_impu.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><font size='3'> Original data: </font><br />
<strong>Heatmap</strong></p>
<pre class="r"><code>coreFeatPlots &lt;- vizCoreFeats(lrRes, sysTrainModels, adjProTumor_Klin,
                              heatmap_numCoreFeats = 8, colFeatAnno = &#39;Gene&#39;,
                              scatter_twoCoreFeats = c(&#39;FABP4&#39;, &#39;DYNLT3&#39;),
                              fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                              cellwidth = 6, cellheight = 18)
coreFeatPlots$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_heatmap_proTumor_ori.png&#39;, plot = a, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><strong>Scatterplot</strong></p>
<pre class="r"><code>coreFeatPlots$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Missingness and availability of core features</strong></p>
<pre class="r"><code>numTopImpoFeats &lt;- 8
# Prepare top important feature list
topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
topImpoFeatAnnos &lt;- topImpoFeatViz$fullImpoFeatTab$Annotation[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)

# Show data missingness
proTumor_Klin &lt;- proTissue_Klin[, colData(proTissue_Klin)$Condition %in% &#39;Tumor&#39;]
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Gene = stringr::str_remove(Gene, &#39;;.+&#39;))
featAvailTab &lt;- summExp2df(proTumor_Klin, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;) %&gt;%
  dplyr::select(Feature, Value, Recurrence) %&gt;%
  dplyr::mutate(Value = dplyr::case_when(!is.na(Value) ~ &#39;Observed&#39;,
                                         is.na(Value) ~ &#39;Missing&#39;),
                Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Recurrence = dplyr::case_when(Recurrence %in% &#39;Yes&#39; ~ &#39;Recurrence&#39;,
                                              Recurrence %in% &#39;No&#39; ~ &#39;Non-Recurrence&#39;)) %&gt;%
  dplyr::filter(Feature %in% topImpoFeats) %&gt;%
  dplyr::group_by(Recurrence, Feature) %&gt;%
  dplyr::mutate(SmpNumber = length(Feature)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::group_by(Recurrence, SmpNumber, Feature, Value) %&gt;%
  dplyr::summarise(Count = dplyr::n()) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = topImpoFeatAnnos),
                Recurrence = factor(Recurrence, levels = c(&#39;Recurrence&#39;, &#39;Non-Recurrence&#39;)),
                Value = factor(Value, levels = c(&#39;Missing&#39;, &#39;Observed&#39;)),
                Proportion = Count/SmpNumber)

ggplot(featAvailTab, aes(x=Gene, y=Proportion, fill=Value)) +
  geom_col(position = &#39;stack&#39;) +
  facet_wrap(vars(Recurrence), scales = &#39;free&#39;) +
  scale_fill_manual(values = c(Missing = &#39;grey&#39;, Observed = &#39;black&#39;)) +
  labs(title = &#39;Normal Tissue Proteomics&#39;) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 90, hjust = 1, vjust = 0.5),
        axis.text.y = element_text(size = 12, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_blank(), legend.text = element_text(size = 18),
        strip.text = element_text(size = 18, face = &#39;bold&#39;),
        plot.title = element_text(size = 22, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Show availability of top important features in other datasets
# Prepare feature space of other datasets
# Klingmüller Preprocessed Combined dataset
old_proTissue_Combined_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
featSpace_old_Prepro_Combined_Klin &lt;- rownames(old_proTissue_Combined_Klin) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Klingmüller Original Combined dataset
featSpace_old_Ori_Combined_Klin &lt;- readr::read_tsv(paste0(&#39;./data/MethodDev/AG_Klingmueller/&#39;,
                                                          &#39;20230309_103228_SmartCare_TumorFree_Tumor_Yang001_Report.tsv&#39;)) %&gt;%
  dplyr::pull(PG.ProteinGroups) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Krijgsveld Preprocessed Combined dataset
proTissue_Combined_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
featSpace_Prepro_Combined_Krij &lt;- rownames(proTissue_Combined_Krij) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Krijgsveld Original Discovery dataset
featSpace_Ori_Discovery_Krij &lt;- readr::read_tsv(paste0(&#39;./data/Discovery/AG_Krijgsveld/&#39;,
                                                       &#39;20230815_MarcS_Discovery_Tissue.pg_matrix.tsv&#39;)) %&gt;%
  dplyr::pull(Protein.Group) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Krijgsveld Original Method Development dataset
featSpace_Ori_MethodDev_Krij &lt;- readr::read_delim(paste0(&#39;./data/MethodDev/AG_Krijgsveld/&#39;,
                                                         &#39;20230726_Thorax_Method_Est_tissue.txt&#39;)) %&gt;%
  dplyr::pull(Protein.Group) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Collect all feature spaces into a list
featSpaceList &lt;- list(Ori_Old_Klin = featSpace_old_Ori_Combined_Klin, Prepro_Old_Klin = featSpace_old_Prepro_Combined_Klin,
                      Ori_MDev_Krij = featSpace_Ori_MethodDev_Krij, Ori_Dis_Krij = featSpace_Ori_Discovery_Krij,
                      Prepro_Comb_Krij = featSpace_Prepro_Combined_Krij)
# Store feature availability in a matrix
featAvailMat &lt;- matrix(nrow = 5, ncol = numTopImpoFeats,
                       dimnames = list(c(&#39;Ori_Old_Klin&#39;, &#39;Prepro_Old_Klin&#39;, &#39;Ori_MDev_Krij&#39;,
                                         &#39;Ori_Dis_Krij&#39;, &#39;Prepro_Comb_Krij&#39;), topImpoFeats))
for (i in seq_len(nrow(featAvailMat))) {
  for (j in seq_len(ncol(featAvailMat))) {
    if (colnames(featAvailMat)[j] %in% featSpaceList[[rownames(featAvailMat)[i]]]) {
      featAvailMat[i, j] &lt;- &#39;Yes&#39;
    } else {
      featAvailMat[i, j] &lt;- &#39;No&#39;
    }
  }
}
# Convert matrix to long data for visualization
featAvailMat &lt;- tibble::as_tibble(featAvailMat, rownames = &#39;Dataset&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Dataset&#39;, names_to = &#39;Feature&#39;, values_to = &#39;Captured&#39;) %&gt;%
  dplyr::mutate(Dataset = factor(Dataset, levels = rev(names(featSpaceList))),
                Captured = factor(Captured, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                Feature = plyr::mapvalues(Feature, from = topImpoFeats, to = topImpoFeatAnnos),
                Feature = factor(Feature, levels = topImpoFeatAnnos))

ggplot(featAvailMat, aes(x=Feature, y=Dataset, fill=Captured)) +
  geom_tile() +
  scale_fill_manual(values = c(&#39;black&#39;, &#39;grey80&#39;)) +
  labs(title = &#39;Tumor Tissue Proteomics&#39;) +
  theme(axis.ticks = element_blank(), panel.grid = element_blank(), panel.background = element_blank(),
        axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_text(size = 20), legend.text = element_text(size = 18),
        plot.title = element_text(size = 28, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-10-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="rf" class="section level2">
<h2>RF</h2>
<!-- <font size='5'> **Stability selection** </font> -->
<!-- **Performance statistics of 1000 models** -->
<!-- ```{r} -->
<!-- # Load imputed adjusted data -->
<!-- impuAdjProTumor_Klin <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProTumorVsn_SVs.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- proTumorRes_Klin <- readRDS('./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs.rds') -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- proTumorRes_Klin$sig.feat.tab$Var1 -->
<!-- # rfRes <- iterRF(impuAdjProTumor_Klin, doImputation = F, doInitFeatSelection = T, -->
<!-- #                 sigFeatList = sigFeatList, doFeatClustering = T, iter = 1000) -->
<!-- # saveRDS(rfRes, './data/Discovery/rf/rfResSOA_SVs_impuAdjProTumor_0.9r_randSplit_1000_Klin.rds') -->
<!-- rfRes <- readRDS('./data/Discovery/rf/rfResSOA_SVs_impuAdjProTumor_0.9r_randSplit_1000_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- rfRes$rfRes$y_pred -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- **Top important features with highest median ranks** -->
<!-- ```{r} -->
<!-- # Visualize top important features -->
<!-- # Prepare feature annotation table -->
<!-- proTissue_Klin <- readRDS('./data/Discovery/AG_Klingmueller/proTissueVsn.rds') -->
<!-- featAnno <- tibble::as_tibble(rowData(proTissue_Klin), rownames = 'Feature') -->
<!-- topImpoFeatViz <- vizTopImpoFeatsRF(rfRes, featAnno = featAnno, num_p1TopFeats = 30) -->
<!-- topImpoFeatViz$rank -->
<!-- ``` -->
<!-- **Full feature importance table** -->
<!-- ```{r} -->
<!-- DT::datatable(topImpoFeatViz$fullImpoFeatTab) -->
<!-- ``` -->
<!-- <font size='5'> **Systematic training and evaluation** </font> -->
<!-- **Performance of models trained with different numbers of top important features** -->
<!-- ```{r} -->
<!-- # Systematically train models with different numbers of top important features -->
<!-- # sysTrainModels <- doSysTrainRF(rfRes, impuAdjProTumor_Klin, max_numTopFeats = 100) -->
<!-- # saveRDS(sysTrainModels, './data/Discovery/rf/sysTrain_rfResSOA_SVs_impuAdjProTumor_randSplit_100_Klin.rds') -->
<!-- sysTrainModels <- readRDS('./data/Discovery/rf/sysTrain_rfResSOA_SVs_impuAdjProTumor_randSplit_100_Klin.rds') -->
<!-- # Visualize systematic training -->
<!-- vizSysTrainModelPerf(sysTrainModels, numTopFeats = 40) -->
<!-- ``` -->
<!-- => Top 21, resulting in model with AUC of 0.999 [1,1]\ -->
<!-- => Top 28, resulting in model with AUC of 1 [1,1] ???? -->
</div>
<div id="xgboost" class="section level2">
<h2>XGBoost</h2>
<!-- <font size='5'> **Linear booster** </font> -->
<!-- **Models trained with core features selected by lasso logistic regression** -->
<!-- ```{r} -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProTumorVsn_SVs.rds') -->
<!-- # Load trained models -->
<!-- lrRes <- readRDS('./data/Discovery/logisR/lrResSOA_SVs_impuAdjProTumor_0.9r_randSplit_1000_5fold_auc_lmin_Klin.rds') -->
<!-- # Specify number of core features selected by lasso logistic regression -->
<!-- numTopImpoFeats <- 5 -->
<!-- ################################################################################ -->
<!-- # Retrieve core feature list -->
<!-- coreFeats <- vizTopImpoFeatsLR(lrRes)$fullImpoFeatTab$Feature[1:numTopImpoFeats] -->
<!-- # Subset core features from data -->
<!-- coreSE <- se[rownames(se) %in% coreFeats,] -->
<!-- # Train models iteratively -->
<!-- # linearBoostRes <- iterXGBoost(coreSE, booster = 'gblinear', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(linearBoostRes, './data/Discovery/xgboost/linear_xgbRes_SVs_coreImpuAdjProTumor_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- linearBoostRes <- readRDS('./data/Discovery/xgboost/linear_xgbRes_SVs_coreImpuAdjProTumor_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- linearBoostRes$xgbRes$y_pred -->
<!-- truth <- linearBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- linearBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- **Models trained with grouped recurrence-related features** -->
<!-- ```{r} -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProTumorVsn_SVs.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- soaRes <- readRDS('./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs.rds') -->
<!-- ################################################################################ -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- soaRes$sig.feat.tab$Var1 -->
<!-- # linearBoostRes <- iterXGBoost(se, doImputation = F, doInitFeatSelection = T, -->
<!-- #                               sigFeatList = sigFeatList, doFeatClustering = T, -->
<!-- #                               booster = 'gblinear', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(linearBoostRes, './data/Discovery/xgboost/linear_xgbRes_SVs_impuAdjProTumor_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- linearBoostRes <- readRDS('./data/Discovery/xgboost/linear_xgbRes_SVs_impuAdjProTumor_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- linearBoostRes$xgbRes$y_pred -->
<!-- truth <- linearBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- linearBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ```{r eval=F, include=F} -->
<!-- # <font size='5'> **Tree booster** </font> -->
<!-- #  -->
<!-- # **Models trained with core features selected by random forest** -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProTumorVsn_SVs.rds') -->
<!-- # Load trained models -->
<!-- rfRes <- readRDS('./data/Discovery/rf/rfResSOA_SVs_impuAdjProTumor_0.9r_randSplit_1000_Klin.rds') -->
<!-- # Specify number of core features selected by random forest -->
<!-- numTopImpoFeats <- 34 -->
<!-- ################################################################################ -->
<!-- # Retrieve core feature list -->
<!-- coreFeats <- vizTopImpoFeatsRF(rfRes)$fullImpoFeatTab$Feature[1:numTopImpoFeats] -->
<!-- # Subset core features from data -->
<!-- coreSE <- se[rownames(se) %in% coreFeats,] -->
<!-- # Train models iteratively -->
<!-- # treeBoostRes <- iterXGBoost(coreSE, booster = 'gbtree', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(treeBoostRes, './data/Discovery/xgboost/tree_xgbRes_SVs_coreImpuAdjProTumor_randSplit_auc_nR1000_iter100_Klin.rds') -->
<!-- treeBoostRes <- readRDS('./data/Discovery/xgboost/tree_xgbRes_SVs_coreImpuAdjProTumor_randSplit_auc_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- treeBoostRes$xgbRes$y_pred -->
<!-- truth <- treeBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- treeBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ```{r eval=F, include=F} -->
<!-- # **Models trained with grouped recurrence-related features** -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProTumorVsn_SVs.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- soaRes <- readRDS('./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs.rds') -->
<!-- ################################################################################ -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- soaRes$sig.feat.tab$Var1 -->
<!-- # treeBoostRes <- iterXGBoost(se, doImputation = F, doInitFeatSelection = T, -->
<!-- #                             sigFeatList = sigFeatList, doFeatClustering = T, -->
<!-- #                             booster = 'gbtree', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(treeBoostRes, './data/Discovery/xgboost/tree_xgbRes_SVs_impuAdjProTumor_randSplit_auc_nR1000_iter100_Klin.rds') -->
<!-- treeBoostRes <- readRDS('./data/Discovery/xgboost/tree_xgbRes_SVs_impuAdjProTumor_randSplit_auc_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- treeBoostRes$xgbRes$y_pred -->
<!-- truth <- treeBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- treeBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
</div>
</div>
<div id="ntp-ag-klin." class="section level1">
<h1>NTP (AG Klin.)</h1>
<p>Normal Tissue DIA Proteomics from AG Klingmüller</p>
<div id="lasso-logistic-regression-1" class="section level2">
<h2>Lasso logistic regression</h2>
<p><font size='5'> <strong>Stability selection</strong> </font></p>
<p><strong>Performance statistics of 1000 models</strong></p>
<pre class="r"><code># Load preprocessed data
proTissue_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
proNormal_Klin &lt;- proTissue_Klin[, colData(proTissue_Klin)$Condition %in% &#39;Normal&#39;]
# # Load SOA results
proNormalRes_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_SVsByProTissueInt.rds&#39;)
# # Prepare adjusted data
adjProNormal_Klin &lt;- proNormal_Klin
assay(adjProNormal_Klin) &lt;- proNormalRes_Klin$SOA.res$dataCorrect
# Do additional feature filtering to remove features observed in less than 50% of samples
ifRmFeats &lt;- apply(assay(adjProNormal_Klin), 1, function(featVec) {
  sum(is.na(featVec)) / length(featVec) &gt; 0.5
})
adjProNormal_Klin &lt;- adjProNormal_Klin[!ifRmFeats,]

# Load imputed adjusted data
impuAdjProNormal_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/impuData/featFilt_impuAdjProNormalVsn_SVsByProTissueInt.rds&#39;)

# Train models iteratively
# Prepare significant feature list
sigFeatList &lt;- proNormalRes_Klin$sig.feat.tab$Var1
# &#39;auc&#39; is not applicable for CV when too few (&lt; 10) observations per fold. Instead,
# &#39;deviance&#39; will be used.
# lrRes &lt;- iterLogisR(adjProNormal_Klin, doImputation = T, doInitFeatSelection = T, sigFeatList = sigFeatList,
#                     doFeatClustering = T, iter = 1000, cvFold = 5, cvMeasure = &#39;auc&#39;)
# saveRDS(lrRes, &#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProNormal_0.9r_randSplit_1000_5fold_auc_lmin_Klin.rds&#39;)
# saveRDS(lrRes$impuSE, &#39;./data/Discovery/AG_Klingmueller/impuData/featFilt_impuAdjProNormalVsn_SVsByProTissueInt.rds&#39;)
lrRes &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProNormal_0.9r_randSplit_1000_5fold_auc_lmin_Klin.rds&#39;)

# Pinpoint useless models that will be removed, i.e., models with no any non-zero beta
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0) #higher lambda results in simpler model
# for (i in uselessModels) {
#   pred &lt;- lrRes$lrRes$y_pred[[i]]
#   truth &lt;- lrRes$lrRes$y_truth[[i]]
#   scores &lt;- caret::confusionMatrix(data = pred, reference = truth, positive = &#39;1&#39;)
#   cat(scores$overall[[&#39;Accuracy&#39;]], &#39;\n&#39;)
# } # Models that got only intercept make random predictions, 0.5 accuracy and AUC-ROC

# Report summarized scores
# Prepare predictions and ground truths where results from useless models are removed
if (length(uselessModels) != 0) {
  pred &lt;- lrRes$lrRes$y_pred[-uselessModels]
  truth &lt;- lrRes$lrRes$y_truth[-uselessModels]
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  pred &lt;- lrRes$lrRes$y_pred
  truth &lt;- lrRes$lrRes$y_truth
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
suppressWarnings(summarizePredPower(pred, truth, auc_roc)) #warning due to sometimes only pos or neg predictions</code></pre>
<pre><code>Mean AUC-ROC score: 0.981
95% CI: [0.8636364, 1]</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Top important features that are most frequently
selected</strong></p>
<pre class="r"><code># Visualize top important features
# Prepare feature annotation table
featAnno &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Feature&#39;)
topImpoFeatViz &lt;- vizTopImpoFeatsLR(lrRes, featAnno = featAnno, num_p1TopFeats = 30)
topImpoFeatViz$pick</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Full feature importance table</strong></p>
<pre class="r"><code># Display full feature importance table
DT::datatable(topImpoFeatViz$fullImpoFeatTab)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8bc16376cdb945bc438e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8bc16376cdb945bc438e">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109"],["A6NET4","Q9NPR2","P01920","Q8IWU2","P16402","Q01974","Q9HCY8","O43318","Q9C0A1","Q9UGM3","Q13387","P84243","A0A0B4J1V2","Q5T280","Q14562","Q8N9T8","Q9H9E3","O60462","Q5W0Z9","P09038","Q96E52","Q96HR9","Q9Y2J4","A0A0B4J1V0","P20231;Q15661","Q9H3G5","H7BZ55","O75420","O75054","A0A075B6P5;P01615","P43897","Q14938","Q96PP9","Q86WJ1","O14817","Q9H469","Q6ZRS2","A7KAX9","Q9Y3D3","P29372","Q6PL18","O95760","O94788","Q7Z589","O95071","P16870","P84022","Q02539","Q9Y250","Q9Y3E0","P08590","O43374","Q96DH6","Q96GX9","O14733","Q5UIP0","Q7Z6J4","Q8WVQ1","P02461","P20908","O15264","Q9P2Y4","Q02224","O43157","Q6P1M0","Q86W50","O14957","P01889","Q9H082","Q6P4Q7","Q9H814","O00443","Q9H910","P32856","Q9Y3M8","Q9UKD2","P28068","P23219","Q17R31","Q6Q4G3","Q9H6Y5","P14209","P17936","Q9H0L4","P33992","P47914","Q9Y296","P24941","Q5SRE7","Q96TA2","Q9UD71","Q9Y2Q5","P05186","P24311","P54289","Q6ZT21","Q7L775","Q8IX04","Q99952","O75525","O95278","P24844","P78411","Q08623","Q71SY5","Q9NRX3","Q9UBU9","Q9UHK6","Q9UKZ9"],["OR5K3","SEMA4B","HLA-DQB1","LMTK2","H1-3","ROR2","S100A14","MAP3K7","ZFHX2","DMBT1","MAPK8IP2","H3-3B","IGHV2-26","SPOUT1","DHX8","KRI1","COG4","NRP2","ZDHHC20","FGF2","OMA1","REEP6","AMOTL2","IGHV3-15","TPSB2;TPSAB1","CPVL","CROCC2","GIGYF1","IGSF3","IGKV2-28;IGKV2D-28","TSFM","NFIX","GBP4","CHD1L","TSPAN4","FBXL15","SRCAP","ARHGAP32","MRPS16","MPG","ATAD2","IL33","ALDH1A2","EMSY","UBR5","CPE","SMAD3","H1-1","LZTS1","GOLT1B","MYL3","RASA4","MSI2","APIP","MAP2K7","RIF1","FGD2","CANT1","COL3A1","COL5A1","MAPK13","ZNF219","CENPE","PLXNB1","SLC27A4","METTL16","UQCR11","HLA-B","RAB33B","CNNM4","PHAX","PIK3C2A","JPT2","STX2","STARD13","MRTO4","HLA-DMB","PTGS1","TATDN3","LVRN","MAGIX","CD99","IGFBP3","CSTF2T","MCM5","RPL29","TRAPPC4","CDK2","PHYHD1","YME1L1","PPP1R1B","LAMTOR2","ALPL","COX7B","CACNA2D1","TMPPE","EPM2AIP1","UEVLD","PTPN18","KHDRBS3","EPM2A","MYL9","IRX5","PUDP","MED25","NDUFA4L2","NXF1","AMACR","PCOLCE2"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[0.998,0.991,0.985,0.983,0.981,0.969,0.9379999999999999,0.926,0.914,0.862,0.855,0.835,0.828,0.721,0.713,0.713,0.711,0.649,0.635,0.579,0.536,0.526,0.467,0.449,0.449,0.421,0.361,0.345,0.329,0.325,0.318,0.302,0.3,0.243,0.212,0.162,0.145,0.141,0.134,0.129,0.129,0.118,0.117,0.114,0.113,0.113,0.112,0.112,0.111,0.11,0.093,0.07199999999999999,0.07199999999999999,0.059,0.057,0.056,0.051,0.042,0.039,0.038,0.037,0.035,0.032,0.028,0.025,0.023,0.02,0.02,0.019,0.017,0.017,0.016,0.016,0.014,0.013,0.011,0.01,0.008,0.008,0.008,0.008,0.007,0.006,0.006,0.005,0.004,0.004,0.003,0.003,0.003,0.003,0.003,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Feature<\/th>\n      <th>Annotation<\/th>\n      <th>Cluster<\/th>\n      <th>PickRate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Feature","targets":1},{"name":"Annotation","targets":2},{"name":"Cluster","targets":3},{"name":"PickRate","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><font size='5'> <strong>Systematic training and evaluation</strong>
</font></p>
<p><strong>Performance of models trained with different numbers of top
important features</strong></p>
<pre class="r"><code># Systematically train models with different numbers of top important features
# sysTrainModels &lt;- doSysTrainLogisR(lrRes, impuAdjProNormal_Klin, max_numTopFeats = 50)
# saveRDS(sysTrainModels, &#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Klin.rds&#39;)
sysTrainModels &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Klin.rds&#39;)

# Visualize systematic training
vizSysTrainModelPerf(sysTrainModels, numTopFeats = 30)</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_Stats_proNormal.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p>=&gt; Top 13 important features are selected, resulting in model with
AUC of 0.992 [0.94,1].<br />
=&gt; Top 21 important features are selected, resulting in model with
AUC of 0.994 [0.95,1].<br />
=&gt; Top 29 important features are selected, resulting in model with
AUC of 0.994 [0.95,1].</p>
<p><strong>Noninformative models that demonstrate effects of top
important features</strong></p>
<pre class="r"><code>se &lt;- impuAdjProNormal_Klin
# Train models without top important features, aka noninformative models
numTopImpoFeats &lt;- c(5, 10, 15, 20, 25, 30)
# for (num in numTopImpoFeats) {
#   topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:num]
#   # Train lasso logistic regression model
#   reducedData &lt;- lrRes$featClusters$reducedData
#   x &lt;- t(reducedData[!rownames(reducedData) %in% topImpoFeats,])
#   y &lt;- colData(se)$Recurrence
#   lrResNonInfo &lt;- runLogisR(x, y, targetClass = &#39;Yes&#39;, iter = 100, regularized_method = &#39;lasso&#39;,
#                             cvFold = 5, cvMeasure = &#39;auc&#39;, used_lambda = &#39;lambda.min&#39;,
#                             trainSet_ratio = 0.8, split_method = &#39;random split&#39;,
#                             plot_ROC = F, save_model = T)
#   saveRDS(lrResNonInfo, paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
#                                &#39;_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Klin.rds&#39;))
# }

# Display performance of noninformative models
meanAUC &lt;- c()
lowerAUC &lt;- c()
upperAUC &lt;- c()
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
ci &lt;- calcCI(auc_roc, bootstrap = T)
lowerAUC &lt;- c(lowerAUC, ci[1])
upperAUC &lt;- c(upperAUC, ci[2])
for (num in numTopImpoFeats) {
  lrResNonInfo &lt;- readRDS(paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
                                 &#39;_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Klin.rds&#39;))
  uselessModels &lt;- which(lrResNonInfo$nNonZero == 0)
  if (length(uselessModels) != 0) {
    auc_roc &lt;- lrResNonInfo$auc_roc[-uselessModels]
  } else {
    auc_roc &lt;- lrResNonInfo$auc_roc
  }
  meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
  ci &lt;- calcCI(auc_roc, bootstrap = T)
  lowerAUC &lt;- c(lowerAUC, ci[1])
  upperAUC &lt;- c(upperAUC, ci[2])
}
aucStatTab &lt;- data.frame(FeatSpace = c(&#39;Complete&#39;, paste0(&#39;-Top &#39;, numTopImpoFeats)),
                         MeanAUC = meanAUC, UpperAUC = upperAUC, LowerAUC = lowerAUC) %&gt;%
  dplyr::mutate(FeatSpace = factor(FeatSpace, levels = FeatSpace))

ggplot(aucStatTab, aes(x=FeatSpace, y=MeanAUC)) +
  geom_bar(stat = &#39;identity&#39;, position = position_dodge(), alpha = 0.9) +
  geom_errorbar(aes(ymin=LowerAUC, ymax=UpperAUC), width = 0.4) +
  labs(x = &#39;Feature space used to train 100 models&#39;, y = &#39;Mean AUC-ROC&#39;,
       title = &#39;Noninformative models&#39;) +
  th</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='5'> <strong>Investigation of core features</strong>
</font></p>
<p><font size='3'> Imputed data: </font><br />
<strong>Heatmap where samples are ordered by linear regression
scores</strong></p>
<pre class="r"><code>coreFeatPlotsImpu &lt;- vizCoreFeats(lrRes, sysTrainModels, impuAdjProNormal_Klin,
                                  heatmap_numCoreFeats = 13, colFeatAnno = &#39;Gene&#39;,
                                  scatter_twoCoreFeats = c(&#39;OR5K3&#39;, &#39;SEMA4B&#39;),
                                  fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                                  cellwidth = 6, cellheight = 18)
coreFeatPlotsImpu$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_heatmap_proNormal_impu.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><strong>Scatterplot of two core features</strong></p>
<pre class="r"><code>coreFeatPlotsImpu$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_top2_proNormal_impu.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><font size='3'> Original data: </font><br />
<strong>Heatmap</strong></p>
<pre class="r"><code>coreFeatPlots &lt;- vizCoreFeats(lrRes, sysTrainModels, adjProNormal_Klin,
                              heatmap_numCoreFeats = 13, colFeatAnno = &#39;Gene&#39;,
                              scatter_twoCoreFeats = c(&#39;OR5K3&#39;, &#39;H1-3&#39;),
                              fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                              cellwidth = 6, cellheight = 18)
coreFeatPlots$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_heatmap_proNormal_ori.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><strong>Scatterplot</strong></p>
<pre class="r"><code>coreFeatPlots$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Junyan/stabSelec_top2_proNormal_ori.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><strong>Missingness and availability of core features</strong></p>
<pre class="r"><code>numTopImpoFeats &lt;- 13
# Prepare top important feature list
topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
topImpoFeatAnnos &lt;- topImpoFeatViz$fullImpoFeatTab$Annotation[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)

# Show data missingness
proNormal_Klin &lt;- proTissue_Klin[, colData(proTissue_Klin)$Condition %in% &#39;Normal&#39;]
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Klin), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Gene = stringr::str_remove(Gene, &#39;;.+&#39;))
featAvailTab &lt;- summExp2df(proNormal_Klin, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;) %&gt;%
  dplyr::select(Feature, Value, Recurrence) %&gt;%
  dplyr::mutate(Value = dplyr::case_when(!is.na(Value) ~ &#39;Observed&#39;,
                                         is.na(Value) ~ &#39;Missing&#39;),
                Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Recurrence = dplyr::case_when(Recurrence %in% &#39;Yes&#39; ~ &#39;Recurrence&#39;,
                                              Recurrence %in% &#39;No&#39; ~ &#39;Non-Recurrence&#39;)) %&gt;%
  dplyr::filter(Feature %in% topImpoFeats) %&gt;%
  dplyr::group_by(Recurrence, Feature) %&gt;%
  dplyr::mutate(SmpNumber = length(Feature)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::group_by(Recurrence, SmpNumber, Feature, Value) %&gt;%
  dplyr::summarise(Count = dplyr::n()) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = topImpoFeatAnnos),
                Recurrence = factor(Recurrence, levels = c(&#39;Recurrence&#39;, &#39;Non-Recurrence&#39;)),
                Value = factor(Value, levels = c(&#39;Missing&#39;, &#39;Observed&#39;)),
                Proportion = Count/SmpNumber)

ggplot(featAvailTab, aes(x=Gene, y=Proportion, fill=Value)) +
  geom_col(position = &#39;stack&#39;) +
  facet_wrap(vars(Recurrence), scales = &#39;free&#39;) +
  scale_fill_manual(values = c(Missing = &#39;grey&#39;, Observed = &#39;black&#39;)) +
  labs(title = &#39;Normal Tissue Proteomics&#39;) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 90, hjust = 1, vjust = 0.5),
        axis.text.y = element_text(size = 12, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_blank(), legend.text = element_text(size = 18),
        strip.text = element_text(size = 18, face = &#39;bold&#39;),
        plot.title = element_text(size = 22, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Show availability of top important features in other datasets
# Prepare feature space of other datasets
# Klingmüller Preprocessed Combined dataset
old_proTissue_Combined_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
featSpace_old_Prepro_Combined_Klin &lt;- rownames(old_proTissue_Combined_Klin) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Klingmüller Original Combined dataset
featSpace_old_Ori_Combined_Klin &lt;- readr::read_tsv(paste0(&#39;./data/MethodDev/AG_Klingmueller/&#39;,
                                                          &#39;20230309_103228_SmartCare_TumorFree_Tumor_Yang001_Report.tsv&#39;)) %&gt;%
  dplyr::pull(PG.ProteinGroups) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Krijgsveld Preprocessed Combined dataset
proTissue_Combined_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
featSpace_Prepro_Combined_Krij &lt;- rownames(proTissue_Combined_Krij) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Krijgsveld Original Discovery dataset
featSpace_Ori_Discovery_Krij &lt;- readr::read_tsv(paste0(&#39;./data/Discovery/AG_Krijgsveld/&#39;,
                                                       &#39;20230815_MarcS_Discovery_Tissue.pg_matrix.tsv&#39;)) %&gt;%
  dplyr::pull(Protein.Group) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Krijgsveld Original Method Development dataset
featSpace_Ori_MethodDev_Krij &lt;- readr::read_delim(paste0(&#39;./data/MethodDev/AG_Krijgsveld/&#39;,
                                                         &#39;20230726_Thorax_Method_Est_tissue.txt&#39;)) %&gt;%
  dplyr::pull(Protein.Group) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Collect all feature spaces into a list
featSpaceList &lt;- list(Ori_Old_Klin = featSpace_old_Ori_Combined_Klin, Prepro_Old_Klin = featSpace_old_Prepro_Combined_Klin,
                      Ori_MDev_Krij = featSpace_Ori_MethodDev_Krij, Ori_Dis_Krij = featSpace_Ori_Discovery_Krij,
                      Prepro_Comb_Krij = featSpace_Prepro_Combined_Krij)
# Store feature availability in a matrix
featAvailMat &lt;- matrix(nrow = 5, ncol = numTopImpoFeats,
                       dimnames = list(c(&#39;Ori_Old_Klin&#39;, &#39;Prepro_Old_Klin&#39;, &#39;Ori_MDev_Krij&#39;,
                                         &#39;Ori_Dis_Krij&#39;, &#39;Prepro_Comb_Krij&#39;), topImpoFeats))
for (i in seq_len(nrow(featAvailMat))) {
  for (j in seq_len(ncol(featAvailMat))) {
    if (colnames(featAvailMat)[j] %in% featSpaceList[[rownames(featAvailMat)[i]]]) {
      featAvailMat[i, j] &lt;- &#39;Yes&#39;
    } else {
      featAvailMat[i, j] &lt;- &#39;No&#39;
    }
  }
}
# Convert matrix to long data for visualization
featAvailMat &lt;- tibble::as_tibble(featAvailMat, rownames = &#39;Dataset&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Dataset&#39;, names_to = &#39;Feature&#39;, values_to = &#39;Captured&#39;) %&gt;%
  dplyr::mutate(Dataset = factor(Dataset, levels = rev(names(featSpaceList))),
                Captured = factor(Captured, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                Feature = plyr::mapvalues(Feature, from = topImpoFeats, to = topImpoFeatAnnos),
                Feature = factor(Feature, levels = topImpoFeatAnnos))

ggplot(featAvailMat, aes(x=Feature, y=Dataset, fill=Captured)) +
  geom_tile() +
  scale_fill_manual(values = c(&#39;black&#39;, &#39;grey80&#39;)) +
  labs(title = &#39;Normal Tissue Proteomics&#39;) +
  theme(axis.ticks = element_blank(), panel.grid = element_blank(), panel.background = element_blank(),
        axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_text(size = 20), legend.text = element_text(size = 18),
        plot.title = element_text(size = 28, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-20-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="rf-1" class="section level2">
<h2>RF</h2>
<!-- <font size='5'> **Stability selection** </font> -->
<!-- **Performance statistics of 1000 models** -->
<!-- ```{r} -->
<!-- # Load imputed adjusted data -->
<!-- impuAdjProNormal_Klin <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProNormalVsn_Stage.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- proNormalRes_Klin <- readRDS('./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_Stage.rds') -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- proNormalRes_Klin$sig.feat.tab$Var1 -->
<!-- # rfRes <- iterRF(impuAdjProNormal_Klin, doImputation = F, doInitFeatSelection = T, -->
<!-- #                 sigFeatList = sigFeatList, doFeatClustering = T, iter = 1000) -->
<!-- # saveRDS(rfRes, './data/Discovery/rf/rfResSOA_Stage_impuAdjProNormal_0.9r_randSplit_1000_Klin.rds') -->
<!-- rfRes <- readRDS('./data/Discovery/rf/rfResSOA_Stage_impuAdjProNormal_0.9r_randSplit_1000_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- rfRes$rfRes$y_pred -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- **Top important features with highest median ranks** -->
<!-- ```{r} -->
<!-- # Visualize top important features -->
<!-- # Prepare feature annotation table -->
<!-- proTissue_Klin <- readRDS('./data/Discovery/AG_Klingmueller/proTissueVsn.rds') -->
<!-- featAnno <- tibble::as_tibble(rowData(proTissue_Klin), rownames = 'Feature') -->
<!-- topImpoFeatViz <- vizTopImpoFeatsRF(rfRes, featAnno = featAnno, num_p1TopFeats = 30) -->
<!-- topImpoFeatViz$rank + theme(axis.title = element_text(size = 28), -->
<!--                             axis.text = element_text(size = 16)) -->
<!-- # ggsave('./output/Discovery/group_meeting/rf_topImpo_combined_proNormal_Klin.png', -->
<!-- #        device = 'png', dpi = 400, height = 8, width = 10) -->
<!-- ``` -->
<!-- **Full feature importance table** -->
<!-- ```{r} -->
<!-- DT::datatable(topImpoFeatViz$fullImpoFeatTab) -->
<!-- ``` -->
<!-- <font size='5'> **Systematic training and evaluation** </font> -->
<!-- **Performance of models trained with different numbers of top important features** -->
<!-- ```{r} -->
<!-- # Systematically train models with different numbers of top important features -->
<!-- # sysTrainModels <- doSysTrainRF(rfRes, impuAdjProNormal_Klin, max_numTopFeats = 100) -->
<!-- # saveRDS(sysTrainModels, './data/Discovery/rf/sysTrain_rfResSOA_Stage_impuAdjProNormal_randSplit_100_Klin.rds') -->
<!-- sysTrainModels <- readRDS('./data/Discovery/rf/sysTrain_rfResSOA_Stage_impuAdjProNormal_randSplit_100_Klin.rds') -->
<!-- # Visualize systematic training -->
<!-- vizSysTrainModelPerf(sysTrainModels, numTopFeats = 40) -->
<!-- ``` -->
<!-- => Top 24, resulting in model with AUC of 0.934 [0.8,1] -->
<!-- ```{r eval=F, include=F} -->
<!-- # **Noninformative models that demonstrate effects of top important features** -->
<!-- se <- impuAdjProNormal_Klin -->
<!-- # Train models without top important features, aka noninformative models -->
<!-- numTopImpoFeats <- c(10, 50, 100, 200, 300, 400, 500) -->
<!-- # for (num in numTopImpoFeats) { -->
<!-- #   topImpoFeats <- topImpoFeatViz$fullImpoFeatTab$Feature[1:num] -->
<!-- #   # Train lasso logistic regression model -->
<!-- #   reducedData <- rfRes$featClusters$reducedData -->
<!-- #   x <- t(reducedData[!rownames(reducedData) %in% topImpoFeats,]) -->
<!-- #   y <- colData(se)$Recurrence -->
<!-- #   rfResNonInfo <- runRF(x, y, targetClass = 'Yes', iter = 100, split_method = 'random split', -->
<!-- #                         trainSet_ratio = 0.8, ntree = 10000, plot_ROC = F, save_model = T) -->
<!-- #   saveRDS(rfResNonInfo, paste0('./data/Discovery/rf/NonInfo_rmTop', num, '_rfResSOA_Stage_', -->
<!-- #                                'impuAdjProNormal_randSplit_100_Klin.rds')) -->
<!-- # } -->
<!-- # Display performance of noninformative models -->
<!-- meanAUC <- c() -->
<!-- lowerAUC <- c() -->
<!-- upperAUC <- c() -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- meanAUC <- c(meanAUC, round(mean(auc_roc), 3)) -->
<!-- ci <- calcCI(auc_roc, bootstrap = T) -->
<!-- lowerAUC <- c(lowerAUC, ci[1]) -->
<!-- upperAUC <- c(upperAUC, ci[2]) -->
<!-- for (num in numTopImpoFeats) { -->
<!--   lrResNonInfo <- readRDS(paste0('./data/Discovery/rf/NonInfo_rmTop', num, '_rfResSOA_Stage_', -->
<!--                                  'impuAdjProNormal_randSplit_100_Klin.rds')) -->
<!--   auc_roc <- lrResNonInfo$auc_roc -->
<!--   meanAUC <- c(meanAUC, round(mean(auc_roc), 3)) -->
<!--   ci <- calcCI(auc_roc, bootstrap = T) -->
<!--   lowerAUC <- c(lowerAUC, ci[1]) -->
<!--   upperAUC <- c(upperAUC, ci[2]) -->
<!-- } -->
<!-- aucStatTab <- data.frame(FeatSpace = c('Complete', paste0('-Top ', numTopImpoFeats)), -->
<!--                          MeanAUC = meanAUC, UpperAUC = upperAUC, LowerAUC = lowerAUC) %>% -->
<!--   dplyr::mutate(FeatSpace = factor(FeatSpace, levels = FeatSpace)) -->
<!-- ggplot(aucStatTab, aes(x=FeatSpace, y=MeanAUC)) + -->
<!--   geom_bar(stat = 'identity', position = position_dodge(), alpha = 0.9) + -->
<!--   geom_errorbar(aes(ymin=LowerAUC, ymax=UpperAUC), width = 0.4) + -->
<!--   labs(x = 'Feature space used to train 100 models', y = 'Mean AUC-ROC', -->
<!--        title = 'Noninformative models') + -->
<!--   th -->
<!-- ``` -->
</div>
<div id="xgboost-1" class="section level2">
<h2>XGBoost</h2>
<!-- <font size='5'> **Linear booster** </font> -->
<!-- **Models trained with core features selected by lasso logistic regression** -->
<!-- ```{r} -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProNormalVsn_Stage.rds') -->
<!-- # Load trained models -->
<!-- lrRes <- readRDS('./data/Discovery/logisR/lrResSOA_Stage_impuAdjProNormal_0.9r_randSplit_1000_5fold_auc_lmin_Klin.rds') -->
<!-- # Specify number of core features selected by lasso logistic regression -->
<!-- numTopImpoFeats <- 13 -->
<!-- ################################################################################ -->
<!-- # Retrieve core feature list -->
<!-- coreFeats <- vizTopImpoFeatsLR(lrRes)$fullImpoFeatTab$Feature[1:numTopImpoFeats] -->
<!-- # Subset core features from data -->
<!-- coreSE <- se[rownames(se) %in% coreFeats,] -->
<!-- # Train models iteratively -->
<!-- # linearBoostRes <- iterXGBoost(coreSE, booster = 'gblinear', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(linearBoostRes, './data/Discovery/xgboost/linear_xgbRes_Stage_coreImpuAdjProNormal_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- linearBoostRes <- readRDS('./data/Discovery/xgboost/linear_xgbRes_Stage_coreImpuAdjProNormal_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- linearBoostRes$xgbRes$y_pred -->
<!-- truth <- linearBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- linearBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- **Models trained with grouped recurrence-related features** -->
<!-- ```{r} -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProNormalVsn_Stage.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- soaRes <- readRDS('./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_Stage.rds') -->
<!-- ################################################################################ -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- soaRes$sig.feat.tab$Var1 -->
<!-- # linearBoostRes <- iterXGBoost(se, doImputation = F, doInitFeatSelection = T, -->
<!-- #                               sigFeatList = sigFeatList, doFeatClustering = T, -->
<!-- #                               booster = 'gblinear', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(linearBoostRes, './data/Discovery/xgboost/linear_xgbRes_Stage_impuAdjProNormal_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- linearBoostRes <- readRDS('./data/Discovery/xgboost/linear_xgbRes_Stage_impuAdjProNormal_randSplit_rmse_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- linearBoostRes$xgbRes$y_pred -->
<!-- truth <- linearBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- linearBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ```{r eval=F, include=F} -->
<!-- # <font size='5'> **Tree booster** </font> -->
<!-- #  -->
<!-- # **Models trained with core features selected by random forest** -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProNormalVsn_Stage.rds') -->
<!-- # Load trained models -->
<!-- rfRes <- readRDS('./data/Discovery/rf/rfResSOA_Stage_impuAdjProNormal_0.9r_randSplit_1000_Klin.rds') -->
<!-- # Specify number of core features selected by random forest -->
<!-- numTopImpoFeats <- 24 -->
<!-- ################################################################################ -->
<!-- # Retrieve core feature list -->
<!-- coreFeats <- vizTopImpoFeatsRF(rfRes)$fullImpoFeatTab$Feature[1:numTopImpoFeats] -->
<!-- # Subset core features from data -->
<!-- coreSE <- se[rownames(se) %in% coreFeats,] -->
<!-- # Train models iteratively -->
<!-- # treeBoostRes <- iterXGBoost(coreSE, booster = 'gbtree', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(treeBoostRes, './data/Discovery/xgboost/tree_xgbRes_Stage_coreImpuAdjProNormal_randSplit_auc_nR1000_iter100_Klin.rds') -->
<!-- treeBoostRes <- readRDS('./data/Discovery/xgboost/tree_xgbRes_Stage_coreImpuAdjProNormal_randSplit_auc_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- treeBoostRes$xgbRes$y_pred -->
<!-- truth <- treeBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- treeBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ```{r eval=F, include=F} -->
<!-- # **Models trained with grouped recurrence-related features** -->
<!-- ############### -->
<!-- # Change here # -->
<!-- ################################################################################ -->
<!-- # Load imputed adjusted data -->
<!-- se <- readRDS('./data/Discovery/AG_Klingmueller/impuAdjProNormalVsn_Stage.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- soaRes <- readRDS('./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_Stage.rds') -->
<!-- ################################################################################ -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- soaRes$sig.feat.tab$Var1 -->
<!-- # treeBoostRes <- iterXGBoost(se, doImputation = F, doInitFeatSelection = T, -->
<!-- #                             sigFeatList = sigFeatList, doFeatClustering = T, -->
<!-- #                             booster = 'gbtree', iter = 100, nrounds = 1000) -->
<!-- # saveRDS(treeBoostRes, './data/Discovery/xgboost/tree_xgbRes_Stage_impuAdjProNormal_randSplit_logloss_nR1000_iter100_Klin.rds') -->
<!-- treeBoostRes <- readRDS('./data/Discovery/xgboost/tree_xgbRes_Stage_impuAdjProNormal_randSplit_logloss_nR1000_iter100_Klin.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- treeBoostRes$xgbRes$y_pred -->
<!-- truth <- treeBoostRes$xgbRes$y_truth -->
<!-- auc_roc <- treeBoostRes$xgbRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
</div>
</div>
<div id="ttp-ag-krij." class="section level1">
<h1>TTP (AG Krij.)</h1>
<p>Tumor Tissue DIA Proteomics from AG Krijgsveld</p>
<div id="lasso-logistic-regression-2" class="section level2">
<h2>Lasso logistic regression</h2>
<p><font size='5'> <strong>Stability selection</strong> </font></p>
<p><strong>Performance statistics of 1000 models</strong></p>
<pre class="r"><code># Load preprocessed data
proTissue_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
proTumor_Krij &lt;- proTissue_Krij[, colData(proTissue_Krij)$Condition %in% &#39;Tumor&#39;]
# Load SOA results
proTumorRes_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/soaRes/combined_proTumorRes_SVsByProTissueInt.rds&#39;)
# Prepare adjusted data
adjProTumor_Krij &lt;- proTumor_Krij
assay(adjProTumor_Krij) &lt;- proTumorRes_Krij$SOA.res$dataCorrect
# Do additional feature filtering to remove features observed in less than 50% of samples
ifRmFeats &lt;- apply(assay(adjProTumor_Krij), 1, function(featVec) {
  sum(is.na(featVec)) / length(featVec) &gt; 0.5
})
adjProTumor_Krij &lt;- adjProTumor_Krij[!ifRmFeats,]

# Load imputed adjusted data
impuAdjProTumor_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/impuData/featFilt_combined_impuAdjProTumorVsn_SVsByProTissueInt.rds&#39;)

# Train models iteratively
# Prepare significant feature list
sigFeatList &lt;- proTumorRes_Krij$sig.feat.tab$Var1
# lrRes &lt;- iterLogisR(adjProTumor_Krij, doImputation = T, doInitFeatSelection = T, sigFeatList = sigFeatList,
#                     doFeatClustering = T, iter = 1000, cvFold = 5, cvMeasure = &#39;auc&#39;)
# saveRDS(lrRes, &#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProTumor_0.9r_randSplit_1000_5fold_auc_lmin_Krij.rds&#39;)
# saveRDS(lrRes$impuSE, &#39;./data/Discovery/AG_Krijgsveld/impuData/featFilt_combined_impuAdjProTumorVsn_SVsByProTissueInt.rds&#39;)
lrRes &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProTumor_0.9r_randSplit_1000_5fold_auc_lmin_Krij.rds&#39;)

# Report summarized scores
# Prepare predictions and ground truths where results from useless models are removed
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  pred &lt;- lrRes$lrRes$y_pred[-uselessModels]
  truth &lt;- lrRes$lrRes$y_truth[-uselessModels]
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  pred &lt;- lrRes$lrRes$y_pred
  truth &lt;- lrRes$lrRes$y_truth
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
suppressWarnings(summarizePredPower(pred, truth, auc_roc))</code></pre>
<pre><code>Mean AUC-ROC score: 0.942
95% CI: [0.7666667, 1]</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Top important features that are most frequently
selected</strong></p>
<pre class="r"><code># Visualize top important features
# Prepare feature annotation table
featAnno &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, Genes) %&gt;%
  dplyr::rename(Gene = Genes)
topImpoFeatViz &lt;- vizTopImpoFeatsLR(lrRes, featAnno = featAnno, num_p1TopFeats = 30)
topImpoFeatViz$pick</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Full feature importance table</strong></p>
<pre class="r"><code># Display full feature importance table
DT::datatable(topImpoFeatViz$fullImpoFeatTab)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-43456476d33848927619" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-43456476d33848927619">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136"],["Q9HD89","P20591","P31994","Q9Y4K1","P23083","P20160","Q15165","O43759","P06127","P08236","Q8TB36","P03950","Q96GA3","Q9UJ14","P13647","P29474","Q9BSD7","Q8IZ73","Q9UII2","P27169","Q96EK6","O15438","Q9NVV5","P51784","Q9BX95","P55774","Q96GM5","Q56VL3","O43708","O14495","O15056","P23141","Q8NG11","Q9UHL4","Q9NRW1","Q16527","Q02539","P08294","Q8NBM8","P21589","P01911","Q9UN76","P79483","P0C870","Q99614","P30626","P01721","Q5TBB1","P11532","Q96Q05","Q70J99","P21397","A4D1E9","P30837","P21695","P56181","Q07092","Q9BQI0","Q6BDS2","P42330","P26440","O00255","P18858","P04792","P25067","Q9NZK5","A8TX70","Q6DT37","P51688","Q9P291","P26022","Q8N755","P05164","Q0P6H9","P00395","Q969X1","Q9H6S3","Q96EV8","Q9P003","Q7Z7L8","Q9BRP4","Q5TBA9","Q9H1C4","O60941","P01893","Q4LDE5","Q96G21","P52292","Q86YQ8","P20036","P55212","Q8TEB1","Q14761","P08246","P35442","P54760","Q8N3R9","P02788","Q99572","P51157","O60825","Q8N128","Q9Y572","P06702","O14569","P61626","Q9Y2J2","A6NI28","O15247","P48200","Q96C01","Q9BVT8","P08174","P26045","Q8N2S1","P18074","P63098","Q12894","Q5H8C1","Q68DQ2","Q75T13","Q96LD4","Q9BR61","Q9BVL4","Q9NVH6","Q9Y6C9","O15213","O75832","P35555","P43403","Q16875","Q969G5","Q96KC8","Q9BVS5","Q9HCC0","Q9UKK3"],["RETN","MX1","FCGR2B","CRYBG1","IGHV1-2","AZU1","PON2","SYNGR1","CD5","GUSB","GDAP1","ANG","LTV1","GGT7","KRT5","NOS3","NTPCR","RPUSD2","ATP5IF1","PON1","GNPNAT1","ABCC3","AIG1","USP11","SGPP1","CCL18","SMARCD1","OCIAD2","GSTZ1","PLPP3","SYNJ2","CES1","TSPAN14","DPP7","RAB6B","CSRP2","H1-1","SOD3","PCYOX1L","NT5E","HLA-DRB1","SLC6A14","HLA-DRB3","JMJD7","TTC1","SRI","IGLV6-57","RNASEH2B","DMD","TRAPPC9","UNC13D","MAOA","GTPBP10","ALDH1B1","GPD1","NDUFV3","COL16A1","AIF1L","UHRF1BP1","AKR1C3","IVD","MEN1","LIG1","HSPB1","COL8A2","ADA2","COL6A5","CDC42BPG","SGSH","ARMCX1","PTX3","SLC66A3","MPO","TMEM62","MT-CO1","TMBIM1","EPS8L2","DTNBP1","CNIH4","C11orf96","PAAF1","FRY","UNC93B1","DTNB","HLA-H","SVEP1","IMP4","KPNA2","CPNE8","HLA-DPA1","CASP6","DCAF11","PTPRCAP","ELANE","THBS2","EPHB4","MPP5","LTF","P2RX7","RAB28","PFKFB2","FAM177A1","RIPK3","S100A9","CYB561D2","LYZ","EPB41L3","ARHGAP42","CLIC2","IREB2","FAM136A","TMUB1","CD55","PTPN3","LTBP4","ERCC2","PPP3R1","IFRD2","FREM1","CRYBG3","PGAP1","TRIM47","ACBD6","SELENOO","TMLHE","MTCH2","WDR46","PSMD10","FBN1","ZAP70","PFKFB3","CAVIN3","DNAJC1","TRMT61B","MCCC2","PARP4"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[0.996,0.967,0.9389999999999999,0.9350000000000001,0.887,0.878,0.797,0.774,0.76,0.739,0.734,0.728,0.716,0.676,0.65,0.644,0.631,0.599,0.596,0.571,0.532,0.522,0.501,0.483,0.476,0.463,0.458,0.457,0.453,0.449,0.428,0.411,0.395,0.393,0.388,0.385,0.37,0.351,0.343,0.304,0.301,0.291,0.29,0.232,0.222,0.209,0.204,0.197,0.173,0.165,0.163,0.16,0.157,0.152,0.134,0.121,0.118,0.107,0.105,0.08,0.07000000000000001,0.065,0.064,0.062,0.051,0.048,0.043,0.043,0.04,0.038,0.037,0.037,0.036,0.033,0.03,0.03,0.029,0.028,0.028,0.023,0.022,0.021,0.021,0.019,0.019,0.019,0.019,0.015,0.015,0.014,0.014,0.014,0.013,0.012,0.012,0.011,0.011,0.01,0.008999999999999999,0.008,0.007,0.007,0.007,0.006,0.005,0.005,0.005,0.004,0.004,0.004,0.004,0.004,0.003,0.003,0.003,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.002,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Feature<\/th>\n      <th>Annotation<\/th>\n      <th>Cluster<\/th>\n      <th>PickRate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Feature","targets":1},{"name":"Annotation","targets":2},{"name":"Cluster","targets":3},{"name":"PickRate","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><font size='5'> <strong>Systematic training and evaluation</strong>
</font></p>
<p><strong>Performance of models trained with different numbers of top
important features</strong></p>
<pre class="r"><code># Systematically train models with different numbers of top important features
# sysTrainModels &lt;- doSysTrainLogisR(lrRes, impuAdjProTumor_Krij, max_numTopFeats = 50)
# saveRDS(sysTrainModels, &#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Krij.rds&#39;)
sysTrainModels &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Krij.rds&#39;)

# Visualize systematic training
vizSysTrainModelPerf(sysTrainModels, numTopFeats = 30)</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" />
=&gt; Top 11 important features are selected, resulting in model with
AUC of 0.997 [0.98,1].<br />
=&gt; Top 18 important features are selected, resulting in model with
AUC of 0.997 [0.97,1].<br />
=&gt; Top 21 important features are selected, resulting in model with
AUC of 0.997 [0.95,1].</p>
<p><strong>Noninformative models that demonstrate effects of top
important features</strong></p>
<pre class="r"><code>se &lt;- impuAdjProTumor_Krij
# Train models without top important features, aka noninformative models
numTopImpoFeats &lt;- c(5, 10, 15, 20, 25)
# for (num in numTopImpoFeats) {
#   topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:num]
#   # Train lasso logistic regression model
#   reducedData &lt;- lrRes$featClusters$reducedData
#   x &lt;- t(reducedData[!rownames(reducedData) %in% topImpoFeats,])
#   y &lt;- colData(se)$Recurrence
#   lrResNonInfo &lt;- runLogisR(x, y, targetClass = &#39;Yes&#39;, iter = 100, regularized_method = &#39;lasso&#39;,
#                             cvFold = 5, cvMeasure = &#39;auc&#39;, used_lambda = &#39;lambda.min&#39;,
#                             trainSet_ratio = 0.8, split_method = &#39;random split&#39;,
#                             plot_ROC = F, save_model = T)
#   saveRDS(lrResNonInfo, paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
#                                &#39;_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Krij.rds&#39;))
# }

# Display performance of noninformative models
meanAUC &lt;- c()
lowerAUC &lt;- c()
upperAUC &lt;- c()
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
ci &lt;- calcCI(auc_roc, bootstrap = T)
lowerAUC &lt;- c(lowerAUC, ci[1])
upperAUC &lt;- c(upperAUC, ci[2])
for (num in numTopImpoFeats) {
  lrResNonInfo &lt;- readRDS(paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
                                 &#39;_lrResSOA_SVsByProTissueInt_impuAdjProTumor_randSplit_100_5fold_auc_lmin_Krij.rds&#39;))
  uselessModels &lt;- which(lrResNonInfo$nNonZero == 0)
  if (length(uselessModels) != 0) {
    auc_roc &lt;- lrResNonInfo$auc_roc[-uselessModels]
  } else {
    auc_roc &lt;- lrResNonInfo$auc_roc
  }
  meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
  ci &lt;- calcCI(auc_roc, bootstrap = T)
  lowerAUC &lt;- c(lowerAUC, ci[1])
  upperAUC &lt;- c(upperAUC, ci[2])
}
aucStatTab &lt;- data.frame(FeatSpace = c(&#39;Complete&#39;, paste0(&#39;-Top &#39;, numTopImpoFeats)),
                         MeanAUC = meanAUC, UpperAUC = upperAUC, LowerAUC = lowerAUC) %&gt;%
  dplyr::mutate(FeatSpace = factor(FeatSpace, levels = FeatSpace))

ggplot(aucStatTab, aes(x=FeatSpace, y=MeanAUC)) +
  geom_bar(stat = &#39;identity&#39;, position = position_dodge(), alpha = 0.9) +
  geom_errorbar(aes(ymin=LowerAUC, ymax=UpperAUC), width = 0.4) +
  labs(x = &#39;Feature space used to train 100 models&#39;, y = &#39;Mean AUC-ROC&#39;,
       title = &#39;Noninformative models&#39;) +
  th</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='5'> <strong>Investigation of core features</strong>
</font></p>
<p><font size='3'> Imputed data: </font><br />
<strong>Heatmap where samples are ordered by linear regression
scores</strong></p>
<pre class="r"><code>coreFeatPlotsImpu &lt;- vizCoreFeats(lrRes, sysTrainModels, impuAdjProTumor_Krij,
                                  heatmap_numCoreFeats = 11, colFeatAnno = &#39;Genes&#39;,
                                  scatter_twoCoreFeats = c(&#39;RETN&#39;, &#39;MX1&#39;),
                                  fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                                  cellwidth = 6, cellheight = 18)
coreFeatPlotsImpu$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Scatterplot of two core features</strong></p>
<pre class="r"><code>coreFeatPlotsImpu$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='3'> Original data: </font><br />
<strong>Heatmap</strong></p>
<pre class="r"><code>coreFeatPlots &lt;- vizCoreFeats(lrRes, sysTrainModels, adjProTumor_Krij,
                              heatmap_numCoreFeats = 11, colFeatAnno = &#39;Genes&#39;,
                              scatter_twoCoreFeats = c(&#39;RETN&#39;, &#39;MX1&#39;),
                              fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                              cellwidth = 6, cellheight = 18)
coreFeatPlots$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Scatterplot</strong></p>
<pre class="r"><code>coreFeatPlots$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Missingness and availability of core features</strong></p>
<pre class="r"><code>numTopImpoFeats &lt;- 11
# Prepare top important feature list
topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
topImpoFeatAnnos &lt;- topImpoFeatViz$fullImpoFeatTab$Annotation[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)

# Show data missingness
proTumor_Krij &lt;- proTissue_Krij[, colData(proTissue_Krij)$Condition %in% &#39;Tumor&#39;]
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Gene = stringr::str_remove(Gene, &#39;;.+&#39;))
featAvailTab &lt;- summExp2df(proTumor_Krij, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;) %&gt;%
  dplyr::select(Feature, Value, Recurrence) %&gt;%
  dplyr::mutate(Value = dplyr::case_when(!is.na(Value) ~ &#39;Observed&#39;,
                                         is.na(Value) ~ &#39;Missing&#39;),
                Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Recurrence = dplyr::case_when(Recurrence %in% &#39;Yes&#39; ~ &#39;Recurrence&#39;,
                                              Recurrence %in% &#39;No&#39; ~ &#39;Non-Recurrence&#39;)) %&gt;%
  dplyr::filter(Feature %in% topImpoFeats) %&gt;%
  dplyr::group_by(Recurrence, Feature) %&gt;%
  dplyr::mutate(SmpNumber = length(Feature)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::group_by(Recurrence, SmpNumber, Feature, Value) %&gt;%
  dplyr::summarise(Count = dplyr::n()) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = topImpoFeatAnnos),
                Recurrence = factor(Recurrence, levels = c(&#39;Recurrence&#39;, &#39;Non-Recurrence&#39;)),
                Value = factor(Value, levels = c(&#39;Missing&#39;, &#39;Observed&#39;)),
                Proportion = Count/SmpNumber)

ggplot(featAvailTab, aes(x=Gene, y=Proportion, fill=Value)) +
  geom_col(position = &#39;stack&#39;) +
  facet_wrap(vars(Recurrence), scales = &#39;free&#39;) +
  scale_fill_manual(values = c(Missing = &#39;grey&#39;, Observed = &#39;black&#39;)) +
  labs(title = &#39;Tumor Tissue Proteomics&#39;) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 90, hjust = 1, vjust = 0.5),
        axis.text.y = element_text(size = 12, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_blank(), legend.text = element_text(size = 18),
        strip.text = element_text(size = 18, face = &#39;bold&#39;),
        plot.title = element_text(size = 22, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Show availability of top important features in other datasets
# Prepare feature space of other datasets
# New Klingmüller Preprocessed Combined dataset
new_proTissue_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
featSpace_new_Prepro_Klin &lt;- rownames(new_proTissue_Klin) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# New Klingmüller Original Combined dataset
proPhosTissue_MDevDis &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/COMP_2696.SMART_Lung_MethodDiscov_ProPhos.rds&#39;)
featSpace_new_Ori_Klin &lt;- rowData(proPhosTissue_MDevDis@ExperimentList$Proteome)$UniprotID %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Klingmüller Preprocessed Combined dataset
old_proTissue_Combined_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
featSpace_old_Prepro_Combined_Klin &lt;- rownames(old_proTissue_Combined_Klin) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Klingmüller Original Combined dataset
featSpace_old_Ori_Combined_Klin &lt;- readr::read_tsv(paste0(&#39;./data/MethodDev/AG_Klingmueller/&#39;,
                                                          &#39;20230309_103228_SmartCare_TumorFree_Tumor_Yang001_Report.tsv&#39;)) %&gt;%
  dplyr::pull(PG.ProteinGroups) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Collect all feature spaces into a list
featSpaceList &lt;- list(Ori_New_Klin = featSpace_new_Ori_Klin,
                      Prepro_New_Klin = featSpace_new_Prepro_Klin,
                      Ori_Old_Klin = featSpace_old_Ori_Combined_Klin,
                      Prepro_Old_Klin = featSpace_old_Prepro_Combined_Klin)
# Store feature availability in a matrix
featAvailMat &lt;- matrix(nrow = 4, ncol = numTopImpoFeats,
                       dimnames = list(c(&#39;Ori_New_Klin&#39;, &#39;Prepro_New_Klin&#39;,
                                         &#39;Ori_Old_Klin&#39;, &#39;Prepro_Old_Klin&#39;), topImpoFeats))
for (i in seq_len(nrow(featAvailMat))) {
  for (j in seq_len(ncol(featAvailMat))) {
    if (colnames(featAvailMat)[j] %in% featSpaceList[[rownames(featAvailMat)[i]]]) {
      featAvailMat[i, j] &lt;- &#39;Yes&#39;
    } else {
      featAvailMat[i, j] &lt;- &#39;No&#39;
    }
  }
}
# Convert matrix to long data for visualization
featAvailMat &lt;- tibble::as_tibble(featAvailMat, rownames = &#39;Dataset&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Dataset&#39;, names_to = &#39;Feature&#39;, values_to = &#39;Captured&#39;) %&gt;%
  dplyr::mutate(Dataset = factor(Dataset, levels = rev(names(featSpaceList))),
                Captured = factor(Captured, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                Feature = plyr::mapvalues(Feature, from = topImpoFeats, to = topImpoFeatAnnos),
                Feature = factor(Feature, levels = topImpoFeatAnnos))

ggplot(featAvailMat, aes(x=Feature, y=Dataset, fill=Captured)) +
  geom_tile() +
  scale_fill_manual(values = c(&#39;black&#39;, &#39;grey80&#39;)) +
  labs(title = &#39;Tumor Tissue Proteomics&#39;) +
  theme(axis.ticks = element_blank(), panel.grid = element_blank(), panel.background = element_blank(),
        axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_text(size = 20), legend.text = element_text(size = 18),
        plot.title = element_text(size = 28, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-30-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="rf-2" class="section level2">
<h2>RF</h2>
<!-- <font size='5'> **Stability selection** </font> -->
<!-- **Performance statistics of 1000 models** -->
<!-- ```{r} -->
<!-- # Load imputed adjusted data -->
<!-- impuAdjProTumor_Krij <- readRDS('./data/Discovery/AG_Krijgsveld/combined_impuAdjProTumorVsn_SVs.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- proTumorRes_Krij <- readRDS('./data/Discovery/AG_Krijgsveld/soaRes/combined_proTumorRes_SVs.rds') -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- proTumorRes_Krij$sig.feat.tab$Var1 -->
<!-- # rfRes <- iterRF(impuAdjProTumor_Krij, doImputation = F, doInitFeatSelection = T, -->
<!-- #                 sigFeatList = sigFeatList, doFeatClustering = T, iter = 1000) -->
<!-- # saveRDS(rfRes, './data/Discovery/rf/rfResSOA_SVs_impuAdjProTumor_0.9r_randSplit_1000_Krij.rds') -->
<!-- rfRes <- readRDS('./data/Discovery/rf/rfResSOA_SVs_impuAdjProTumor_0.9r_randSplit_1000_Krij.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- rfRes$rfRes$y_pred -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- **Top important features with highest median ranks** -->
<!-- ```{r} -->
<!-- # Visualize top important features -->
<!-- # Prepare feature annotation table -->
<!-- proTissue_Krij <- readRDS('./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds') -->
<!-- featAnno <- tibble::as_tibble(rowData(proTissue_Krij), rownames = 'Feature') %>% -->
<!--   dplyr::select(Feature, Genes) %>% -->
<!--   dplyr::rename(Gene = Genes) -->
<!-- topImpoFeatViz <- vizTopImpoFeatsRF(rfRes, featAnno = featAnno, num_p1TopFeats = 30) -->
<!-- topImpoFeatViz$rank + theme(axis.title = element_text(size = 28), -->
<!--                             axis.text = element_text(size = 16)) -->
<!-- ``` -->
<!-- **Full feature importance table** -->
<!-- ```{r} -->
<!-- DT::datatable(topImpoFeatViz$fullImpoFeatTab) -->
<!-- ``` -->
<!-- <font size='5'> **Systematic training and evaluation** </font> -->
<!-- **Performance of models trained with different numbers of top important features** -->
<!-- ```{r} -->
<!-- # Systematically train models with different numbers of top important features -->
<!-- # sysTrainModels <- doSysTrainRF(rfRes, impuAdjProTumor_Krij, max_numTopFeats = 100) -->
<!-- # saveRDS(sysTrainModels, './data/Discovery/rf/sysTrain_rfResSOA_SVs_impuAdjProTumor_randSplit_100_Krij.rds') -->
<!-- sysTrainModels <- readRDS('./data/Discovery/rf/sysTrain_rfResSOA_SVs_impuAdjProTumor_randSplit_100_Krij.rds') -->
<!-- # Visualize systematic training -->
<!-- vizSysTrainModelPerf(sysTrainModels, numTopFeats = 30) -->
<!-- ``` -->
<!-- => Top 3, resulting in model with AUC of 0.996 [0.97,1]\ -->
<!-- => Top 5, resulting in model with AUC of 0.997 [0.97,1]\ -->
<!-- => Top 9, resulting in model with AUC of 0.998 [0.98,1] -->
</div>
</div>
<div id="ntp-ag-krij." class="section level1">
<h1>NTP (AG Krij.)</h1>
<p>Normal Tissue DIA Proteomics from AG Krijgsveld</p>
<div id="lasso-logistic-regression-3" class="section level2">
<h2>Lasso logistic regression</h2>
<p><font size='5'> <strong>Stability selection</strong> </font></p>
<p><strong>Performance statistics of 1000 models</strong></p>
<pre class="r"><code># Load preprocessed data
proTissue_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
proNormal_Krij &lt;- proTissue_Krij[, colData(proTissue_Krij)$Condition %in% &#39;Normal&#39;]
# Load SOA results
proNormalRes_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/soaRes/combined_proNormalRes_SVsByProTissueInt.rds&#39;)
# Prepare adjusted data
adjProNormal_Krij &lt;- proNormal_Krij
assay(adjProNormal_Krij) &lt;- proNormalRes_Krij$SOA.res$dataCorrect
# Do additional feature filtering to remove features observed in less than 50% of samples
ifRmFeats &lt;- apply(assay(adjProNormal_Krij), 1, function(featVec) {
  sum(is.na(featVec)) / length(featVec) &gt; 0.5
})
adjProNormal_Krij &lt;- adjProNormal_Krij[!ifRmFeats,]

# Load imputed adjusted data
impuAdjProNormal_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/impuData/featFilt_combined_impuAdjProNormalVsn_SVsByProTissueInt.rds&#39;)

# Train models iteratively
# Prepare significant feature list
sigFeatList &lt;- proNormalRes_Krij$sig.feat.tab$Var1
# lrRes &lt;- iterLogisR(adjProNormal_Krij, doImputation = T, doInitFeatSelection = T, sigFeatList = sigFeatList,
#                     doFeatClustering = T, iter = 1000, cvFold = 5, cvMeasure = &#39;auc&#39;)
# saveRDS(lrRes, &#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProNormal_0.9r_randSplit_1000_5fold_auc_lmin_Krij.rds&#39;)
# saveRDS(lrRes$impuSE, &#39;./data/Discovery/AG_Krijgsveld/impuData/featFilt_combined_impuAdjProNormalVsn_SVsByProTissueInt.rds&#39;)
lrRes &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_lrResSOA_SVsByProTissueInt_impuAdjProNormal_0.9r_randSplit_1000_5fold_auc_lmin_Krij.rds&#39;)

# Report summarized scores
# Prepare predictions and ground truths where results from useless models are removed
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  pred &lt;- lrRes$lrRes$y_pred[-uselessModels]
  truth &lt;- lrRes$lrRes$y_truth[-uselessModels]
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  pred &lt;- lrRes$lrRes$y_pred
  truth &lt;- lrRes$lrRes$y_truth
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
suppressWarnings(summarizePredPower(pred, truth, auc_roc))</code></pre>
<pre><code>Mean AUC-ROC score: 0.984
95% CI: [0.8939394, 1]</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Top important features that are most frequently
selected</strong></p>
<pre class="r"><code># Visualize top important features
# Prepare feature annotation table
featAnno &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, Genes) %&gt;%
  dplyr::rename(Gene = Genes)
topImpoFeatViz &lt;- vizTopImpoFeatsLR(lrRes, featAnno = featAnno, num_p1TopFeats = 30)
topImpoFeatViz$pick</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Full feature importance table</strong></p>
<pre class="r"><code># Display full feature importance table
DT::datatable(topImpoFeatViz$fullImpoFeatTab)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8a53d124fe189d0fbdcf" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8a53d124fe189d0fbdcf">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87"],["P55064","Q9H6Y2","Q92954","P28702","Q86TG7","Q96CT7","P23946","P51911","Q9UGM3","P05114","O95347","P47895","Q6P158","Q6PL24","P48595","P33993","Q86YL7","O14548","Q96EY8","Q8NDH2","A0A0B4J1V2","Q8NCC3","A0A0C4DH43","Q9UBK8","Q15024","Q9UII2","Q7L5Y9","Q86TX2","Q9H6T0","P06127","Q05707","Q9BSD7","O43708","Q6ZWT7","P17936","Q99973","Q17RN3","P05090","Q9Y2Q9","O43913","Q9Y6W3","P02748","P02794","Q9H1E3","P17096","P50897","P68133","Q9NNX6","Q4KMQ1","Q99715","Q9BX97","O60869","O75934","O43808","O94788","Q15651","P02788","Q05193","Q96GE9","P01599","P15428","P61916","O60487","O94851","Q32P28","Q9BZQ6","O75674","P35625","Q8N6M0","Q9H0B6","Q9H6W3","P21359","Q9BU40","O14933","O15155","Q6ZXV5","Q96HU1","O00479","P00750","P14902","P33764","Q02040","Q5JRX3","Q96SB4","Q9BY43","Q9P107","Q9UGM6"],["AQP5","WDR55","PRG4","RXRB","PEG10","CCDC124","CMA1","CNN1","DMBT1","HMGN1","SMC2","ALDH1A3","DHX57","TMED8","SERPINB10","MCM7","PDPN","COX7A2L","MMAB","CCDC168","IGHV2-26","PLA2G15","IGHV2-70D","MTRR","EXOSC7","ATP5IF1","MAEA","ACOT1","ESRP2","CD5","COL14A1","NTPCR","GSTZ1","MBOAT2","IGFBP3","TEP1","FAM98C","APOD","MRPS28","ORC5","CAPN7","C9","FTH1","NUCKS1","HMGA1","PPT1","ACTA1","CD209","TPRN","COL12A1","PLVAP","EDF1","BCAS2","SLC25A17","ALDH1A2","HMGN3","LTF","DNM1","DMAC1","IGKV1-17","HPGD","NPC2","MPZL2","MICAL2","P3H1","EDEM3","TOM1L1","TIMP3","OTUD6B","KLC2","RIOX1","NF1","CHRDL1","UBE2L6","BET1","TMTC3","SGSM3","HMGN4","PLAT","IDO1","S100A3","AKAP17A","PITRM1","SRPK1","CHMP4A","GMIP","WARS2"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Q01995/Q9NR12",null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[0.997,0.99,0.969,0.954,0.875,0.871,0.858,0.847,0.8159999999999999,0.787,0.768,0.694,0.605,0.57,0.548,0.546,0.52,0.511,0.332,0.268,0.262,0.25,0.23,0.213,0.199,0.178,0.163,0.162,0.156,0.154,0.15,0.148,0.141,0.133,0.102,0.089,0.08799999999999999,0.076,0.07199999999999999,0.064,0.06,0.046,0.04,0.039,0.036,0.035,0.027,0.025,0.023,0.023,0.013,0.012,0.011,0.01,0.01,0.01,0.008,0.008,0.008,0.007,0.006,0.006,0.005,0.005,0.005,0.005,0.004,0.004,0.004,0.004,0.004,0.003,0.003,0.002,0.002,0.002,0.002,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001,0.001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Feature<\/th>\n      <th>Annotation<\/th>\n      <th>Cluster<\/th>\n      <th>PickRate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Feature","targets":1},{"name":"Annotation","targets":2},{"name":"Cluster","targets":3},{"name":"PickRate","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><font size='5'> <strong>Systematic training and evaluation</strong>
</font></p>
<p><strong>Performance of models trained with different numbers of top
important features</strong></p>
<pre class="r"><code># Systematically train models with different numbers of top important features
# sysTrainModels &lt;- doSysTrainLogisR(lrRes, impuAdjProNormal_Krij, max_numTopFeats = 50)
# saveRDS(sysTrainModels, &#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Krij.rds&#39;)
sysTrainModels &lt;- readRDS(&#39;./data/Discovery/logisR/featFilt_sysTrain_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Krij.rds&#39;)

# Visualize systematic training
vizSysTrainModelPerf(sysTrainModels, numTopFeats = 30)</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" />
=&gt; Top 5 important features are selected, resulting in model with AUC
of 0.991 [0.95,1].<br />
=&gt; Top 8 important features are selected, resulting in model with AUC
of 0.993 [0.95,1].<br />
=&gt; Top 11 important features are selected, resulting in model with
AUC of 0.994 [0.95,1].<br />
=&gt; Top 20 important features are selected, resulting in model with
AUC of 0.994 [0.95,1].</p>
<p><strong>Noninformative models that demonstrate effects of top
important features</strong></p>
<pre class="r"><code>se &lt;- impuAdjProNormal_Krij
# Train models without top important features, aka noninformative models
numTopImpoFeats &lt;- c(5, 10, 15, 20)
# for (num in numTopImpoFeats) {
#   topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:num]
#   # Train lasso logistic regression model
#   reducedData &lt;- lrRes$featClusters$reducedData
#   x &lt;- t(reducedData[!rownames(reducedData) %in% topImpoFeats,])
#   y &lt;- colData(se)$Recurrence
#   lrResNonInfo &lt;- runLogisR(x, y, targetClass = &#39;Yes&#39;, iter = 100, regularized_method = &#39;lasso&#39;,
#                             cvFold = 5, cvMeasure = &#39;auc&#39;, used_lambda = &#39;lambda.min&#39;,
#                             trainSet_ratio = 0.8, split_method = &#39;random split&#39;,
#                             plot_ROC = F, save_model = T)
#   saveRDS(lrResNonInfo, paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
#                                &#39;_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Krij.rds&#39;))
# }

# Display performance of noninformative models
meanAUC &lt;- c()
lowerAUC &lt;- c()
upperAUC &lt;- c()
uselessModels &lt;- which(lrRes$lrRes$nNonZero == 0)
if (length(uselessModels) != 0) {
  auc_roc &lt;- lrRes$lrRes$auc_roc[-uselessModels]
} else {
  auc_roc &lt;- lrRes$lrRes$auc_roc
}
meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
ci &lt;- calcCI(auc_roc, bootstrap = T)
lowerAUC &lt;- c(lowerAUC, ci[1])
upperAUC &lt;- c(upperAUC, ci[2])
for (num in numTopImpoFeats) {
  lrResNonInfo &lt;- readRDS(paste0(&#39;./data/Discovery/logisR/noninfo/featFilt_NonInfo_rmTop&#39;, num,
                                 &#39;_lrResSOA_SVsByProTissueInt_impuAdjProNormal_randSplit_100_5fold_auc_lmin_Krij.rds&#39;))
  uselessModels &lt;- which(lrResNonInfo$nNonZero == 0)
  if (length(uselessModels) != 0) {
    auc_roc &lt;- lrResNonInfo$auc_roc[-uselessModels]
  } else {
    auc_roc &lt;- lrResNonInfo$auc_roc
  }
  meanAUC &lt;- c(meanAUC, round(mean(auc_roc), 3))
  ci &lt;- calcCI(auc_roc, bootstrap = T)
  lowerAUC &lt;- c(lowerAUC, ci[1])
  upperAUC &lt;- c(upperAUC, ci[2])
}
aucStatTab &lt;- data.frame(FeatSpace = c(&#39;Complete&#39;, paste0(&#39;-Top &#39;, numTopImpoFeats)),
                         MeanAUC = meanAUC, UpperAUC = upperAUC, LowerAUC = lowerAUC) %&gt;%
  dplyr::mutate(FeatSpace = factor(FeatSpace, levels = FeatSpace))

ggplot(aucStatTab, aes(x=FeatSpace, y=MeanAUC)) +
  geom_bar(stat = &#39;identity&#39;, position = position_dodge(), alpha = 0.9) +
  geom_errorbar(aes(ymin=LowerAUC, ymax=UpperAUC), width = 0.4) +
  labs(x = &#39;Feature space used to train 100 models&#39;, y = &#39;Mean AUC-ROC&#39;,
       title = &#39;Noninformative models&#39;) +
  th</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='5'> <strong>Investigation of core features</strong>
</font></p>
<p><font size='3'> Imputed data: </font><br />
<strong>Heatmap where samples are ordered by linear regression
scores</strong></p>
<pre class="r"><code>coreFeatPlotsImpu &lt;- vizCoreFeats(lrRes, sysTrainModels, impuAdjProNormal_Krij,
                                  heatmap_numCoreFeats = 5, colFeatAnno = &#39;Genes&#39;,
                                  scatter_twoCoreFeats = c(&#39;AQP5&#39;, &#39;WDR55&#39;),
                                  fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                                  cellwidth = 6, cellheight = 18)
coreFeatPlotsImpu$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Scatterplot of two core features</strong></p>
<pre class="r"><code>coreFeatPlotsImpu$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='3'> Original data: </font><br />
<strong>Heatmap</strong></p>
<pre class="r"><code>coreFeatPlots &lt;- vizCoreFeats(lrRes, sysTrainModels, adjProNormal_Krij,
                              heatmap_numCoreFeats = 5, colFeatAnno = &#39;Genes&#39;,
                              scatter_twoCoreFeats = c(&#39;AQP5&#39;, &#39;PRG4&#39;),
                              fontsize = 10, fontsize_col = 7, fontsize_row = 11,
                              cellwidth = 6, cellheight = 18)
coreFeatPlots$heatmap</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Scatterplot</strong></p>
<pre class="r"><code>coreFeatPlots$scatterplot</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Missingness and availability of core features</strong></p>
<pre class="r"><code>numTopImpoFeats &lt;- 5
# Prepare top important feature list
topImpoFeats &lt;- topImpoFeatViz$fullImpoFeatTab$Feature[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
topImpoFeatAnnos &lt;- topImpoFeatViz$fullImpoFeatTab$Annotation[1:numTopImpoFeats] %&gt;%
  stringr::str_remove(&#39;;.+&#39;)

# Show data missingness
proNormal_Krij &lt;- proTissue_Krij[, colData(proTissue_Krij)$Condition %in% &#39;Normal&#39;]
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  dplyr::mutate(Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Gene = stringr::str_remove(Gene, &#39;;.+&#39;))
featAvailTab &lt;- summExp2df(proNormal_Krij, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;) %&gt;%
  dplyr::select(Feature, Value, Recurrence) %&gt;%
  dplyr::mutate(Value = dplyr::case_when(!is.na(Value) ~ &#39;Observed&#39;,
                                         is.na(Value) ~ &#39;Missing&#39;),
                Feature = stringr::str_remove(Feature, &#39;;.+&#39;),
                Recurrence = dplyr::case_when(Recurrence %in% &#39;Yes&#39; ~ &#39;Recurrence&#39;,
                                              Recurrence %in% &#39;No&#39; ~ &#39;Non-Recurrence&#39;)) %&gt;%
  dplyr::filter(Feature %in% topImpoFeats) %&gt;%
  dplyr::group_by(Recurrence, Feature) %&gt;%
  dplyr::mutate(SmpNumber = length(Feature)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::group_by(Recurrence, SmpNumber, Feature, Value) %&gt;%
  dplyr::summarise(Count = dplyr::n()) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = topImpoFeatAnnos),
                Recurrence = factor(Recurrence, levels = c(&#39;Recurrence&#39;, &#39;Non-Recurrence&#39;)),
                Value = factor(Value, levels = c(&#39;Missing&#39;, &#39;Observed&#39;)),
                Proportion = Count/SmpNumber)

ggplot(featAvailTab, aes(x=Gene, y=Proportion, fill=Value)) +
  geom_col(position = &#39;stack&#39;) +
  facet_wrap(vars(Recurrence), scales = &#39;free&#39;) +
  scale_fill_manual(values = c(Missing = &#39;grey&#39;, Observed = &#39;black&#39;)) +
  labs(title = &#39;Normal Tissue Proteomics&#39;) +
  theme_bw() +
  theme(axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 90, hjust = 1, vjust = 0.5),
        axis.text.y = element_text(size = 12, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_blank(), legend.text = element_text(size = 18),
        strip.text = element_text(size = 18, face = &#39;bold&#39;),
        plot.title = element_text(size = 22, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Show availability of top important features in other datasets
# Prepare feature space of other datasets
# New Klingmüller Preprocessed Combined dataset
new_proTissue_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
featSpace_new_Prepro_Klin &lt;- rownames(new_proTissue_Klin) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# New Klingmüller Original Combined dataset
proPhosTissue_MDevDis &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/COMP_2696.SMART_Lung_MethodDiscov_ProPhos.rds&#39;)
featSpace_new_Ori_Klin &lt;- rowData(proPhosTissue_MDevDis@ExperimentList$Proteome)$UniprotID %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Klingmüller Preprocessed Combined dataset
old_proTissue_Combined_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
featSpace_old_Prepro_Combined_Klin &lt;- rownames(old_proTissue_Combined_Klin) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Klingmüller Original Combined dataset
featSpace_old_Ori_Combined_Klin &lt;- readr::read_tsv(paste0(&#39;./data/MethodDev/AG_Klingmueller/&#39;,
                                                          &#39;20230309_103228_SmartCare_TumorFree_Tumor_Yang001_Report.tsv&#39;)) %&gt;%
  dplyr::pull(PG.ProteinGroups) %&gt;%
  stringr::str_remove(&#39;;.+&#39;)
# Collect all feature spaces into a list
featSpaceList &lt;- list(Ori_New_Klin = featSpace_new_Ori_Klin,
                      Prepro_New_Klin = featSpace_new_Prepro_Klin,
                      Ori_Old_Klin = featSpace_old_Ori_Combined_Klin,
                      Prepro_Old_Klin = featSpace_old_Prepro_Combined_Klin)
# Store feature availability in a matrix
featAvailMat &lt;- matrix(nrow = 4, ncol = numTopImpoFeats,
                       dimnames = list(c(&#39;Ori_New_Klin&#39;, &#39;Prepro_New_Klin&#39;,
                                         &#39;Ori_Old_Klin&#39;, &#39;Prepro_Old_Klin&#39;), topImpoFeats))
for (i in seq_len(nrow(featAvailMat))) {
  for (j in seq_len(ncol(featAvailMat))) {
    if (colnames(featAvailMat)[j] %in% featSpaceList[[rownames(featAvailMat)[i]]]) {
      featAvailMat[i, j] &lt;- &#39;Yes&#39;
    } else {
      featAvailMat[i, j] &lt;- &#39;No&#39;
    }
  }
}
# Convert matrix to long data for visualization
featAvailMat &lt;- tibble::as_tibble(featAvailMat, rownames = &#39;Dataset&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Dataset&#39;, names_to = &#39;Feature&#39;, values_to = &#39;Captured&#39;) %&gt;%
  dplyr::mutate(Dataset = factor(Dataset, levels = rev(names(featSpaceList))),
                Captured = factor(Captured, levels = c(&#39;Yes&#39;, &#39;No&#39;)),
                Feature = plyr::mapvalues(Feature, from = topImpoFeats, to = topImpoFeatAnnos),
                Feature = factor(Feature, levels = topImpoFeatAnnos))

ggplot(featAvailMat, aes(x=Feature, y=Dataset, fill=Captured)) +
  geom_tile() +
  scale_fill_manual(values = c(&#39;black&#39;, &#39;grey80&#39;)) +
  labs(title = &#39;Normal Tissue Proteomics&#39;) +
  theme(axis.ticks = element_blank(), panel.grid = element_blank(), panel.background = element_blank(),
        axis.text.x = element_text(size = 18, face = &#39;bold&#39;, angle = 45, hjust = 1, vjust = 1),
        axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 20, face = &#39;bold&#39;),
        legend.title = element_text(size = 20), legend.text = element_text(size = 18),
        plot.title = element_text(size = 28, face = &#39;bold&#39;, hjust = 0.5, vjust = 1.5))</code></pre>
<p><img src="figure/FeatSelection_ML_FeatFilt.Rmd/unnamed-chunk-40-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="rf-3" class="section level2">
<h2>RF</h2>
<!-- <font size='5'> **Stability selection** </font> -->
<!-- **Performance statistics of 1000 models** -->
<!-- ```{r} -->
<!-- # Load imputed adjusted data -->
<!-- impuAdjProNormal_Krij <- readRDS('./data/Discovery/AG_Krijgsveld/combined_impuAdjProNormalVsn_Smoking_Gender.rds') -->
<!-- # Load SOA results for retrieving significant feature list for initial feature selection -->
<!-- proNormalRes_Krij <- readRDS('./data/Discovery/AG_Krijgsveld/soaRes/combined_proNormalRes_Smoking_Gender.rds') -->
<!-- # Train models iteratively -->
<!-- # Prepare significant feature list -->
<!-- sigFeatList <- proNormalRes_Krij$sig.feat.tab$Var1 -->
<!-- # rfRes <- iterRF(impuAdjProNormal_Krij, doImputation = F, doInitFeatSelection = T, -->
<!-- #                 sigFeatList = sigFeatList, doFeatClustering = T, iter = 1000) -->
<!-- # saveRDS(rfRes, './data/Discovery/rf/rfResSOA_Smoking_Gender_impuAdjProNormal_0.9r_randSplit_1000_Krij.rds') -->
<!-- rfRes <- readRDS('./data/Discovery/rf/rfResSOA_Smoking_Gender_impuAdjProNormal_0.9r_randSplit_1000_Krij.rds') -->
<!-- # Report summarized scores -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- rfRes$rfRes$y_pred -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- **Top important features with highest median ranks** -->
<!-- ```{r} -->
<!-- # Visualize top important features -->
<!-- # Prepare feature annotation table -->
<!-- proTissue_Krij <- readRDS('./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds') -->
<!-- featAnno <- tibble::as_tibble(rowData(proTissue_Krij), rownames = 'Feature') %>% -->
<!--   dplyr::select(Feature, Genes) %>% -->
<!--   dplyr::rename(Gene = Genes) -->
<!-- topImpoFeatViz <- vizTopImpoFeatsRF(rfRes, featAnno = featAnno, num_p1TopFeats = 30) -->
<!-- topImpoFeatViz$rank + theme(axis.title = element_text(size = 28), -->
<!--                             axis.text = element_text(size = 16)) -->
<!-- ``` -->
<!-- **Full feature importance table** -->
<!-- ```{r} -->
<!-- DT::datatable(topImpoFeatViz$fullImpoFeatTab) -->
<!-- ``` -->
<!-- <font size='5'> **Systematic training and evaluation** </font> -->
<!-- **Performance of models trained with different numbers of top important features** -->
<!-- ```{r} -->
<!-- # Systematically train models with different numbers of top important features -->
<!-- # sysTrainModels <- doSysTrainRF(rfRes, impuAdjProNormal_Krij, max_numTopFeats = 100) -->
<!-- # saveRDS(sysTrainModels, './data/Discovery/rf/sysTrain_rfResSOA_Smoking_Gender_impuAdjProNormal_randSplit_100_Krij.rds') -->
<!-- sysTrainModels <- readRDS('./data/Discovery/rf/sysTrain_rfResSOA_Smoking_Gender_impuAdjProNormal_randSplit_100_Krij.rds') -->
<!-- # Visualize systematic training -->
<!-- vizSysTrainModelPerf(sysTrainModels, numTopFeats = 40) -->
<!-- ``` -->
<!-- => Top 13, resulting in model with AUC of 0.977 [0.92,1]\ -->
<!-- => Top 23, resulting in model with AUC of 0.977 [0.91,1] -->
<!-- # Unt. Lipid. -->
<!-- Untargeted Lipidomics -->
<!-- ## RF -->
<!-- ```{r} -->
<!-- # NTL -->
<!-- # Load preprocessed data -->
<!-- # Discovery -->
<!-- # dis_lipTissue_Hopf <- readRDS('./data/Discovery/AG_Hopf/lipTissueVsn_WBC25.rds') -->
<!-- # Method development -->
<!-- mDev_lipTissue_Hopf <- readRDS('./data/MethodDev/AG_Hopf/lipTissueVsn.rds') -->
<!-- lipTissue <- mDev_lipTissue_Hopf -->
<!-- # Subset data -->
<!-- lipNormal <- lipTissue[, colData(lipTissue)$Condition == 'Normal'] -->
<!-- # Remove features observed in less than 67% of samples -->
<!-- rmFeats <- apply(assay(lipNormal), 1, function(featVec) { -->
<!--   obProp <- sum(!is.na(featVec)) / length(featVec) -->
<!--   obProp <= 0.67 -->
<!-- }) %>% -->
<!--   which() -->
<!-- if (length(rmFeats) != 0) { -->
<!--   lipNormal <- lipNormal[-rmFeats,] -->
<!-- } -->
<!-- # Impute missing values using missForest -->
<!-- dat <- t(assay(lipNormal)) -->
<!-- impuDat <- missForest(dat, maxiter = 10, verbose = T)$ximp %>% -->
<!--   t() -->
<!-- assay(lipNormal) <- impuDat -->
<!-- # saveRDS(lipNormal, './data/MethodDev/AG_Hopf/impuByMF//lipNormalVsnImpu.rds') -->
<!-- # BPL -->
<!-- # Load preprocessed data -->
<!-- # Discovery -->
<!-- # dis_lipBase_Hopf <- readRDS('./data/Discovery/AG_Hopf/lipBaseVsn_B1WBC25.rds') -->
<!-- # Method development -->
<!-- # Subset data -->
<!-- mDev_lipPlasma_Hopf <- readRDS('./data/MethodDev/AG_Hopf/lipPlasmaVsn.rds') -->
<!-- lipPlasma <- mDev_lipPlasma_Hopf -->
<!-- lipBase <- lipPlasma[, colData(lipPlasma)$TimePoint == 'Baseline'] -->
<!-- # Remove features observed in less than 67% of samples -->
<!-- rmFeats <- apply(assay(lipBase), 1, function(featVec) { -->
<!--   obProp <- sum(!is.na(featVec)) / length(featVec) -->
<!--   obProp <= 0.67 -->
<!-- }) %>% -->
<!--   which() -->
<!-- if (length(rmFeats) != 0) { -->
<!--   lipBase <- lipBase[-rmFeats,] -->
<!-- } -->
<!-- # Impute missing values using missForest -->
<!-- dat <- t(assay(lipBase)) -->
<!-- impuDat <- missForest(dat, maxiter = 10, verbose = T)$ximp %>% -->
<!--   t() -->
<!-- assay(lipBase) <- impuDat -->
<!-- # saveRDS(lipBase, './data/MethodDev/AG_Hopf/impuByMF/lipBaseVsnImpu.rds') -->
<!-- # Train RF -->
<!-- # rfRes <- iterRF(lipBase) -->
<!-- # saveRDS(rfRes, './data/Discovery/rf/untLip/rfResSig_mDev_untLipBase_randSplit_1000') -->
<!-- ``` -->
<!-- ### Dis. NTL -->
<!-- **Average model performance (Uncorrelated significant features, iter = 1000)** -->
<!-- ```{r} -->
<!-- # Load RF training results -->
<!-- rfRes <- readRDS('./data/Discovery/rf/untLip/rfResSig_dis_untLipNormal_randSplit_1000') -->
<!-- # Report summarized scores -->
<!-- trainedModels <- rfRes$rfRes$rfRes -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- as.list(rep(NA, length(trainedModels))) -->
<!-- for (i in seq_len(length(trainedModels))) { -->
<!--   pred[[i]] <- trainedModels[[i]]$test$predicted -->
<!-- } -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ### Dis. BPL -->
<!-- **Average model performance (Uncorrelated significant features, iter = 1000)** -->
<!-- ```{r} -->
<!-- # Load RF training results -->
<!-- rfRes <- readRDS('./data/Discovery/rf/untLip/rfResSig_dis_untLipBase_randSplit_1000') -->
<!-- # Report summarized scores -->
<!-- trainedModels <- rfRes$rfRes$rfRes -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- as.list(rep(NA, length(trainedModels))) -->
<!-- for (i in seq_len(length(trainedModels))) { -->
<!--   pred[[i]] <- trainedModels[[i]]$test$predicted -->
<!-- } -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ### MDev. NTL -->
<!-- **Average model performance (Uncorrelated significant features, iter = 1000)** -->
<!-- ```{r} -->
<!-- # Load RF training results -->
<!-- rfRes <- readRDS('./data/Discovery/rf/untLip/rfResSig_mDev_untLipNormal_randSplit_1000') -->
<!-- # Report summarized scores -->
<!-- trainedModels <- rfRes$rfRes$rfRes -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- as.list(rep(NA, length(trainedModels))) -->
<!-- for (i in seq_len(length(trainedModels))) { -->
<!--   pred[[i]] <- trainedModels[[i]]$test$predicted -->
<!-- } -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<!-- ### MDev. BPL -->
<!-- **Average model performance (Uncorrelated significant features, iter = 1000)** -->
<!-- ```{r} -->
<!-- # Load RF training results -->
<!-- rfRes <- readRDS('./data/Discovery/rf/untLip/rfResSig_mDev_untLipBase_randSplit_1000') -->
<!-- # Report summarized scores -->
<!-- trainedModels <- rfRes$rfRes$rfRes -->
<!-- # Prepare predictions and ground truths -->
<!-- pred <- as.list(rep(NA, length(trainedModels))) -->
<!-- for (i in seq_len(length(trainedModels))) { -->
<!--   pred[[i]] <- trainedModels[[i]]$test$predicted -->
<!-- } -->
<!-- truth <- rfRes$rfRes$y_truth -->
<!-- auc_roc <- rfRes$rfRes$auc_roc -->
<!-- summarizePredPower(pred, truth, auc_roc) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Warsaw
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] MultiAssayExperiment_1.28.0 sva_3.50.0                 
 [3] BiocParallel_1.36.0         genefilter_1.84.0          
 [5] mgcv_1.9-1                  nlme_3.1-164               
 [7] proDA_1.16.0                limma_3.58.1               
 [9] pcaMethods_1.94.0           lubridate_1.9.3            
[11] forcats_1.0.0               stringr_1.5.1              
[13] purrr_1.0.2                 readr_2.1.5                
[15] tidyr_1.3.1                 tibble_3.2.1               
[17] tidyverse_2.0.0             SummarizedExperiment_1.32.0
[19] Biobase_2.62.0              GenomicRanges_1.54.1       
[21] GenomeInfoDb_1.38.8         IRanges_2.36.0             
[23] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[25] MatrixGenerics_1.14.0       matrixStats_1.3.0          
[27] ggvenn_0.1.10               dplyr_1.1.4                
[29] glmnet_4.1-8                Matrix_1.6-5               
[31] caret_6.0-94                lattice_0.22-6             
[33] xgboost_1.7.7.1             missForest_1.5             
[35] randomForest_4.7-1.1        DT_0.33                    
[37] ggh4x_0.2.8                 ggplot2_3.5.1              
[39] spsUtil_0.2.2               workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] splines_4.3.2           later_1.3.2             bitops_1.0-7           
  [4] ggplotify_0.1.2         hardhat_1.3.1           pROC_1.18.5            
  [7] XML_3.99-0.16.1         rpart_4.1.23            lifecycle_1.0.4        
 [10] rstatix_0.7.2           edgeR_4.0.16            rprojroot_2.0.4        
 [13] vroom_1.6.5             globals_0.16.3          processx_3.8.4         
 [16] MASS_7.3-60.0.1         crosstalk_1.2.1         backports_1.4.1        
 [19] magrittr_2.0.3          sass_0.4.9              rmarkdown_2.27         
 [22] jquerylib_0.1.4         yaml_2.3.8              httpuv_1.6.15          
 [25] doRNG_1.8.6             RColorBrewer_1.1-3      DBI_1.2.2              
 [28] abind_1.4-5             zlibbioc_1.48.2         itertools_0.1-3        
 [31] RCurl_1.98-1.14         yulab.utils_0.1.4       nnet_7.3-19            
 [34] ipred_0.9-14            git2r_0.33.0            lava_1.8.0             
 [37] GenomeInfoDbData_1.2.11 listenv_0.9.1           pheatmap_1.0.12        
 [40] annotate_1.80.0         parallelly_1.37.1       codetools_0.2-20       
 [43] DelayedArray_0.28.0     tidyselect_1.2.1        shape_1.4.6.1          
 [46] farver_2.1.2            jsonlite_1.8.8          e1071_1.7-14           
 [49] survival_3.5-8          iterators_1.0.14        foreach_1.5.2          
 [52] tools_4.3.2             Rcpp_1.0.12             glue_1.7.0             
 [55] prodlim_2023.08.28      SparseArray_1.2.4       xfun_0.44              
 [58] withr_3.0.0             fastmap_1.2.0           fansi_1.0.6            
 [61] callr_3.7.6             digest_0.6.35           timechange_0.3.0       
 [64] R6_2.5.1                gridGraphics_0.5-1      colorspace_2.1-0       
 [67] RSQLite_2.3.6           utf8_1.2.4              generics_0.1.3         
 [70] data.table_1.15.4       recipes_1.0.10          class_7.3-22           
 [73] httr_1.4.7              htmlwidgets_1.6.4       S4Arrays_1.2.1         
 [76] whisker_0.4.1           ModelMetrics_1.2.2.2    pkgconfig_2.0.3        
 [79] gtable_0.3.5            timeDate_4032.109       blob_1.2.4             
 [82] XVector_0.42.0          htmltools_0.5.8.1       carData_3.0-5          
 [85] scales_1.3.0            png_0.1-8               gower_1.0.1            
 [88] knitr_1.47              rstudioapi_0.16.0       tzdb_0.4.0             
 [91] reshape2_1.4.4          proxy_0.4-27            cachem_1.1.0           
 [94] parallel_4.3.2          AnnotationDbi_1.64.1    pillar_1.9.0           
 [97] vctrs_0.6.5             promises_1.3.0          ggpubr_0.6.0           
[100] car_3.1-2               xtable_1.8-4            evaluate_0.23          
[103] cli_3.6.2               locfit_1.5-9.9          compiler_4.3.2         
[106] rlang_1.1.4             crayon_1.5.2            rngtools_1.5.2         
[109] future.apply_1.11.2     ggsignif_0.6.4          labeling_0.4.3         
[112] ps_1.7.6                getPass_0.2-4           plyr_1.8.9             
[115] fs_1.6.4                stringi_1.8.4           munsell_0.5.1          
[118] Biostrings_2.70.3       hms_1.1.3               bit64_4.0.5            
[121] future_1.33.2           KEGGREST_1.42.0         statmod_1.5.0          
[124] highr_0.11              broom_1.0.5             memoise_2.0.1          
[127] bslib_0.7.0             bit_4.0.5              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
