<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Preprocessing: DIA Proteomics of Method Dev and Discovery cohorts</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Preprocessing: DIA Proteomics of Method Dev
and Discovery cohorts</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree282daae0fdbb7078dd8d66521af4c5d654c63cfbtargetblank282daaea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/282daae0fdbb7078dd8d66521af4c5d654c63cfb" target="_blank">282daae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree282daae0fdbb7078dd8d66521af4c5d654c63cfbtargetblank282daaea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/282daae0fdbb7078dd8d66521af4c5d654c63cfb" target="_blank">282daae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Discovery/
    Ignored:    data/MethodDev/
    Ignored:    data/aliquot_metadata.xlsx
    Ignored:    data/metadata_latest/
    Ignored:    data/patient_metadata.xlsx
    Ignored:    data/sample_metadata.xlsx
    Ignored:    output/.DS_Store
    Ignored:    output/Discovery/
    Ignored:    output/MethodDev/
    Ignored:    output/SC_meeting/

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/data_availability.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R

Unstaged changes:
    Modified:   analysis/FeatSelection_ML.Rmd
    Modified:   analysis/MDev_Dis_02_soa_annotated_untargeted.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/MDev_Dis_01_preprocessing_DIA.Rmd</code>) and HTML
(<code>docs/MDev_Dis_01_preprocessing_DIA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/258edf90a9453bb3eeea0a8c7632712f3a69184f/analysis/MDev_Dis_01_preprocessing_DIA.Rmd" target="_blank">258edf9</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-29
</td>
<td>
Do protein abundance profiling of potential bad-quality tumor samples
and compare analysis results of newly and previously generated data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/40a33b213cc2f969503476a28838de6605884fd3/analysis/MDev_Dis_01_preprocessing_DIA.Rmd" target="_blank">40a33b2</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-12
</td>
<td>
Assess effects of SVs correcting data by reproducibility between new and
old proteomics from AG Klin.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/b5f8cacee8787a19248fc7f8a6fbcdecdb598b3a/analysis/MDev_Dis_01_preprocessing_DIA.Rmd" target="_blank">b5f8cac</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-05
</td>
<td>
Check data reproducibility between new and old tissue DIA proteomics
from AG Klin.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/e957730a0dc718d91bfb403da6c3e1c1f3239143/analysis/MDev_Dis_01_preprocessing_DIA.Rmd" target="_blank">e957730</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-02
</td>
<td>
Preprocess combined Method Dev and Discovery Tissue Proteomics from AG
Klin. and rename and rearrange source code and analysis results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Preprocess and combine Method
Development and Discovery Tissue DIA Proteomics generated from AG
Klingmüller and AG Krijgsveld, including data cleansing, filtering,
normalization (VSN), and data merging. All needed information was then
stored in SummarizedExperiment objects for further analyses.</p>
<p>For Klin. data: Sample <strong>Q89YQK_NG</strong>,
<strong>1TK71I_TU</strong>, <strong>NG1IWD_TU</strong>,
<strong>125CII_TU</strong>, <strong>1LJXIN_TU</strong>,
<strong>I0HVOL_TU</strong>, and <strong>7EAOX7_TU</strong> were
excluded. (Check Section ‘<strong>Show reasons for removing
samples</strong>’ for more details)</p>
<p>For Krij. data: Those 44 extra samples where Tumor samples are of low
tumor purity in Discovery cohort are not included yet. </font></p>
<p>Load libraries</p>
<pre class="r"><code>library(readr)
library(readxl)
library(vsn)
library(limma)
library(sva)
library(visdat)
library(ggrepel)
library(MultiAssayExperiment)
library(SummarizedExperiment)
library(tidyverse)
# Load user-defined functions
source(&#39;./code/analysis_pipeline.R&#39;)
source(&#39;./code/misc.R&#39;)
source(&#39;./code/comparison_funcs.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<pre class="r"><code># Load sample metadata
smpMetadat &lt;- readxl::read_excel(&#39;./data/metadata_latest/sample_metadata.xlsx&#39;)
colnames(smpMetadat) &lt;- smpMetadat[3,, drop = F]
smpMetadat &lt;- dplyr::slice(smpMetadat, -c(1:3)) %&gt;%
  dplyr::select(c(Code, Parents, Visit, `Material submitted`, `Date and time of collection or surgery`,
                  `SMART-CARE cohort identifier`)) %&gt;%
  dplyr::rename(Sample = Code, Patient = Parents, TimePoint = Visit, SmpType = `Material submitted`,
                Date = `Date and time of collection or surgery`, Cohort = `SMART-CARE cohort identifier`) %&gt;%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, &#39;^SC_T_S_|^SC_DIS_S_|^SC_VAL_S_|^SC_S_|_P&#39;),
                Patient = stringr::str_remove(Patient, &#39;^/THRX_SPACE/THRX_DB/&#39;),
                TimePoint = dplyr::case_when(TimePoint == &#39;PRETHERAPEUTIC&#39; ~ &#39;Baseline&#39;,
                                             TimePoint == &#39;FOLLOW-UP&#39; ~ &#39;Follow-up&#39;,
                                             TimePoint == &#39;RECURRENCE&#39; ~ &#39;Recurrence&#39;),
                SmpType = dplyr::case_when(SmpType == &#39;EDTA_PLASMA&#39; ~ &#39;Plasma&#39;,
                                           SmpType == &#39;FRESH_FROZEN_TISSUE&#39; ~ &#39;Tissue&#39;),
                Condition = dplyr::case_when(grepl(&#39;_TU|_TG&#39;, Sample) ~ &#39;Tumor&#39;,
                                             grepl(&#39;_NG&#39;, Sample) ~ &#39;Normal&#39;,
                                             !grepl(&#39;_TU|_TG|_NG&#39;, Sample) ~ TimePoint),
                TimePoint = dplyr::case_when(TimePoint %in% &#39;Recurrence&#39; ~ &#39;Follow-up&#39;,
                                             !TimePoint %in% &#39;Recurrence&#39; ~ TimePoint),
                Date = stringr::str_extract(Date, &#39;^\\d+-\\d+-\\d+&#39;),
                Cohort = dplyr::case_when(Cohort == &#39;DISCOVERY_COHORT&#39; ~ &#39;Discovery&#39;,
                                          Cohort == &#39;METHOD_DEVELOPMENT_COHORT&#39; ~ &#39;MethodDev&#39;,
                                          Cohort == &#39;VALIDATION_COHORT&#39; ~ &#39;Validation&#39;),
                Date = as.Date(Date, format = &#39;%Y-%m-%d&#39;))

# Load patient metadata
patientMetadat &lt;- readxl::read_excel(&#39;./data/metadata_latest/patient_metadata.xlsx&#39;)
colnames(patientMetadat) &lt;- patientMetadat[3,, drop = F]
patientMetadat &lt;- dplyr::slice(patientMetadat, -c(1:3)) %&gt;%
  dplyr::select(c(Code, Gender, `Age at diagnosis`, `Pathological stage`, `Smoking status`,
                  `Adjuvant chemotherapy`, Nutrition)) %&gt;%
  dplyr::rename(Patient = Code, Age = `Age at diagnosis`, Stage = `Pathological stage`,
                Smoking = `Smoking status`, Adjuvant = `Adjuvant chemotherapy`) %&gt;%
  dplyr::mutate(Gender = dplyr::case_when(Gender == &#39;MALE&#39; ~ &#39;Male&#39;,
                                          Gender == &#39;FEMALE&#39; ~ &#39;Female&#39;),
                Smoking = dplyr::case_when(Smoking == &#39;SMOKER&#39; ~ &#39;Smoker&#39;,
                                           Smoking == &#39;EX-SMOKER&#39; ~ &#39;Ex-smoker&#39;,
                                           Smoking == &#39;NON-SMOKER&#39; ~ &#39;Non-smoker&#39;),
                Adjuvant = dplyr::case_when(Adjuvant == &#39;true&#39; ~ &#39;True&#39;,
                                            Adjuvant == &#39;false&#39; ~ &#39;False&#39;),
                Age = as.numeric(Age),
                Nutrition = dplyr::case_when(Nutrition == &#39;WHOLE_FOOD&#39; ~ &#39;Whole_food&#39;))
# Include patient recurrence information in patient metadata
recurPats &lt;- dplyr::filter(smpMetadat, Condition == &#39;Recurrence&#39;) %&gt;%
  dplyr::pull(Patient)
patientMetadat &lt;- dplyr::mutate(patientMetadat, Recurrence = dplyr::case_when(Patient %in% recurPats ~ &#39;Yes&#39;,
                                                                              !Patient %in% recurPats ~ &#39;No&#39;))

# Load aliquot metadata
aliquotMetadat &lt;- readxl::read_excel(&#39;./data/metadata_latest/aliquot_metadata.xlsx&#39;)
colnames(aliquotMetadat) &lt;- aliquotMetadat[3,, drop = F]
aliquotMetadat &lt;- dplyr::slice(aliquotMetadat, -c(1:3)) %&gt;%
  dplyr::select(Code, Parents, `Tumor Cell Content (%)`, `Cohort Identifier`, `Submission to`,
                `Delivered?`) %&gt;%
  dplyr::rename(Aliquot = Code, Sample = Parents, TumorPurity = `Tumor Cell Content (%)`,
                Cohort = `Cohort Identifier`, To = `Submission to`, Delivered = `Delivered?`) %&gt;%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_T_S_|_P&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_DIS_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_DIS_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_VAL_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_VAL_&#39;),
                Cohort = dplyr::case_when(Cohort == &#39;method establishment&#39; ~ &#39;MethodDev&#39;,
                                          Cohort == &#39;DISCOVERY_COHORT&#39; ~ &#39;Discovery&#39;,
                                          Cohort == &#39;VALIDATION_COHORT&#39; ~ &#39;Validation&#39;),
                TumorPurity = as.numeric(TumorPurity),
                TumorPurity = dplyr::case_when(grepl(&#39;_TU|_TG&#39;, Sample) ~ TumorPurity),
                # Mislabeling??
                Sample = case_when(Sample %in% &#39;SDIR55_NG&#39; ~ &#39;PDIR55_NG&#39;,
                                   Sample %in% &#39;SDIR55_TU&#39; ~ &#39;PDIR55_TU&#39;,
                                   Sample %in% &#39;SD4ADT_NG&#39; ~ &#39;PD4ADT_NG&#39;,
                                   Sample %in% &#39;SD4ADT_TU&#39; ~ &#39;PD4ADT_TU&#39;,
                                   !Sample %in% c(&#39;SDIR55_NG&#39;, &#39;SDIR55_TU&#39;,
                                                  &#39;SD4ADT_NG&#39;, &#39;SD4ADT_TU&#39;) ~ Sample)) %&gt;%
  dplyr::filter(To %in% c(&#39;KLINGMUELLER&#39;, &#39;KRIJGSVELD&#39;))
# Prepare tumor purity information
tumorPurityInfo &lt;- dplyr::select(aliquotMetadat, Sample, TumorPurity) %&gt;%
  dplyr::filter(!duplicated(Sample))

# Combine all needed metadata
summMetadat &lt;- dplyr::select(aliquotMetadat, Aliquot, Sample, TumorPurity, To) %&gt;%
  dplyr::left_join(smpMetadat, by = &#39;Sample&#39;) %&gt;%
  dplyr::left_join(patientMetadat, by = &#39;Patient&#39;)
summMetadat4Krij &lt;- dplyr::left_join(smpMetadat, patientMetadat, by = &#39;Patient&#39;) %&gt;%
  dplyr::left_join(tumorPurityInfo, by = &#39;Sample&#39;)
# Manually add an extra sample to summarized metadata
smp &lt;- summMetadat4Krij[summMetadat4Krij$Sample == &#39;6EY5JO_TU&#39;,]
smp[1, 1] &lt;- &#39;6EY5JO_II_TU&#39;
summMetadat4Krij &lt;- rbind(summMetadat4Krij, smp)

# List sample annotations to keep in SE objects
smpAnno &lt;- c(&#39;Patient&#39;, &#39;SmpType&#39;, &#39;TimePoint&#39;, &#39;Date&#39;, &#39;Cohort&#39;, &#39;Condition&#39;,
             &#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;,
             &#39;Nutrition&#39;, &#39;TumorPurity&#39;)</code></pre>
<div id="tissue-samples" class="section level1">
<h1>Tissue samples</h1>
<div id="ag-klingmüller" class="section level2">
<h2>AG Klingmüller</h2>
<div id="proteomics" class="section level3">
<h3>Proteomics</h3>
<pre class="r"><code># Load MultiAssayExperiment object containing Proteomics, Phosphoproteomics, and
# Phospho ratio of Method Dev and Discovery cohorts summarized by Junyan
proPhosTissue_MDevDis &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/COMP_2696.SMART_Lung_MethodDiscov_ProPhos.rds&#39;)
# Convert data matrix to long data for tidying up data and conducting preprocessing
proTissue &lt;- proPhosTissue_MDevDis@ExperimentList$Proteome
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue), rownames = &#39;p&#39;) %&gt;%
  dplyr::rename(Feature = UniprotID)
summMetadat_Klin &lt;- dplyr::filter(summMetadat, To %in% &#39;KLINGMUELLER&#39;)
proTissueTab &lt;- tibble::as_tibble((assay(proTissue)), rownames = &#39;p&#39;) %&gt;%
  tidyr::pivot_longer(cols = -p, names_to = &#39;Aliquot&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Aliquot = stringr::str_remove(Aliquot, &#39;_FP$&#39;),
                Aliquot = dplyr::case_when(Aliquot %in% &#39;SC_T_A_1WFPXHTG_4&#39; ~ &#39;SC_T_A_1WFPXHTG_1&#39;,
                                           Aliquot %in% &#39;SC_T_A_7ZY8GLNG_2&#39; ~ &#39;SC_T_A_7ZY8GLNG_1&#39;,
                                           Aliquot %in% &#39;SC_DIS_A_PDIR55_TU3&#39; ~ &#39;SC_DIS_A_PDIR55_TU1&#39;,
                                           !Aliquot %in% c(&#39;SC_T_A_1WFPXHTG_4&#39;, &#39;SC_T_A_7ZY8GLNG_2&#39;,
                                                           &#39;SC_T_A_1WFPXHTG_4&#39;) ~ Aliquot)) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;p&#39;) %&gt;%
  dplyr::left_join(summMetadat_Klin, by = &#39;Aliquot&#39;) %&gt;%
  dplyr::select(-c(p, Aliquot)) %&gt;%
  # Compute sample missing levels
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(MissLevel = sum(is.na(Abundance)) / length(Abundance)) %&gt;%
  dplyr::ungroup()

# Remove tumor samples with low tumor purity
proTissue_Dis_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/proTissueVsn.rds&#39;)
goodSmps_Dis &lt;- colnames(proTissue_Dis_Krij)
allSmps_Dis &lt;- unique(proTissueTab[proTissueTab$Cohort == &#39;Discovery&#39;, &#39;Sample&#39;, drop = T])
badSmps_Dis &lt;- allSmps_Dis[!allSmps_Dis %in% goodSmps_Dis]
# Manually remove Sample &#39;1HTNWJ_TU&#39; from bad quality sample list due to its missingness
# in good quality sample list
badSmps_Dis &lt;- badSmps_Dis[-which(badSmps_Dis == &#39;1HTNWJ_TU&#39;)]
badSmps_Dis &lt;- badSmps_Dis[grepl(&#39;_TU&#39;, badSmps_Dis)]
# proTissueTab &lt;- dplyr::filter(proTissueTab, !Sample %in% badSmps_Dis)
rm(proTissue_Dis_Krij, goodSmps_Dis, allSmps_Dis)

# Remove samples with high missing level or low tumor purity and misclassified
allProTissueTab &lt;- proTissueTab
proTissueTab &lt;- dplyr::filter(proTissueTab, !Sample %in% c(&#39;Q89YQK_NG&#39;, &#39;125CII_TU&#39;,
                                                           &#39;1TK71I_TU&#39;, &#39;1LJXIN_TU&#39;,
                                                           &#39;NG1IWD_TU&#39;, &#39;I0HVOL_TU&#39;,
                                                           &#39;7EAOX7_TU&#39;))</code></pre>
<p><font size='4'> <strong>Overview original data</strong> </font><br />
6 Tumor samples and 1 Normal sample have been excluded before
preprocessing. Check <strong>Show reasons for removing samples</strong>
for more details</p>
<pre class="r"><code># Preprocess data
proTissuePrepro &lt;- doPreprocessing(proTissueTab, feat = &#39;Feature&#39;, smp = &#39;Sample&#39;,
                                   val = &#39;Abundance&#39;, smpAnno = c(smpAnno, &#39;MissLevel&#39;),
                                   featAnno = &#39;Gene&#39;, do_featFilt = T, cutoff = 0.5, viz_miss = T)
                                   # save_path = &#39;./data/Discovery/AG_Klingmueller/proTissue&#39;
# Show dimensions of data
cat(&#39;Data dimensions:&#39;, dim(proTissuePrepro$ori.data)[2], &#39;samples and&#39;,
    dim(proTissuePrepro$ori.data)[1], &#39;features&#39;)</code></pre>
<pre><code>Data dimensions: 163 samples and 10536 features</code></pre>
<pre class="r"><code># Visualize data missingness
proTissuePrepro$ori.data.miss +
  labs(title = &#39;Missingness of original data&#39;) +
  theme(axis.text.x = element_text(size = 5))</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data distribution
proTissuePrepro$ori.data.dist +
  labs(title = &#39;Distribution of original data&#39;) +
  theme(axis.text.x = element_text(size = 5))</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-2-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='4'> <strong>Overview filtered vsn normalized
data</strong> </font><br />
Features quantified in less than half of samples are removed.</p>
<pre class="r"><code># Show dimensions of data
cat(&#39;Filtered data dimensions:&#39;, dim(proTissuePrepro$filt.data)[2], &#39;samples and&#39;,
    dim(proTissuePrepro$filt.data)[1], &#39;features&#39;)</code></pre>
<pre><code>Filtered data dimensions: 163 samples and 7290 features</code></pre>
<pre class="r"><code># Visualize data missingness
proTissuePrepro$filt.data.miss +
  labs(title = &#39;Missingness of filtered data&#39;) +
  theme(axis.text.x = element_text(size = 5))</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data distribution
proTissuePrepro$vsn.data.dist +
  labs(y = &#39;Log(Abundance)&#39;, title = &#39;Distribution of filtered vsn normalized data&#39;) +
  theme(axis.text.x = element_text(size = 5))</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-3-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize mean-sd relationship
proTissuePrepro$vsn.feat.mean.var +
  labs(title = &#39;Feature mean-sd relationship after vsn normalization&#39;)</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-3-3.png" width="960" style="display: block; margin: auto;" /></p>
<p><font size='4'> <strong>Do metadata-assisted quality control: Tumor
vs Normal samples</strong> </font></p>
<pre class="r"><code># Do PCA
pcaRes &lt;- doSOA(proTissuePrepro$vsn.data, meta_var = &#39;Condition&#39;, #&#39;TumorPurity&#39;, &#39;MissLevel&#39;
                pca_method = &#39;ppca&#39;, num_PCs = 20, do_onlyPCA = T)
# Show significant associations between PCs and sample conditions
pcaRes$pcSigAssoRes</code></pre>
<pre><code>         Var1      Var2     pVal  pValAdj  Stat   Test
1 PC1 (33.8%) Condition 2.39e-71 4.77e-70 -31.8 T-test</code></pre>
<pre class="r"><code># Visualize PCs
# Prepare samples to be labeled
badSmps &lt;- gsub(&#39;_TU&#39;, &#39;_NG&#39;, badSmps_Dis)
badSmps &lt;- c(badSmps, badSmps_Dis)
pcTab &lt;- pcaRes$pcTab %&gt;%
  dplyr::mutate(Extra = dplyr::case_when(Sample %in% badSmps ~ &#39;Yes&#39;,
                                         !Sample %in% badSmps ~ &#39;No&#39;))
ggplot(pcTab, aes(x=Condition, y=`PC1 (33.8%)`, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.7, outliers = F) +
  geom_jitter(position = position_jitter(0.3), size = 4, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#   theme(axis.title = element_text(size = 22),
#         axis.text.x = element_text(size = 18),
#         axis.text.y = element_text(size = 14),
#         legend.title = element_text(size = 22),
#         legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/QC_combined_proTissue_Klin_PC1_rmSmps.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 10)

# Check quality of those extra normal samples whose tumor counterparts were removed
# ggplot(pcTab, aes(x=`PC1 (33.8%)`, y=`PC2 (14.3%)`, col=Condition, shape=Extra)) +
#   geom_point(size = 2.5) +
#   scale_color_brewer(palette = &#39;Dark2&#39;) +
#   scale_shape_manual(name = &#39;Extra samples&#39;, values = c(16, 3)) +
#   th

# Check tumor purity of tumor samples
# ggplot(pcTab, aes(x=`PC1 (33.8%)`, y=`PC2 (14.3%)`, col=TumorPurity, shape=Condition)) +
#   geom_point(size = 2.5) +
#   scale_color_continuous(name = &#39;Tumor purity (%)&#39;) +
#   th

# Check missing levels of samples
# ggplot(pcTab, aes(x=`PC1 (33.8%)`, y=`PC2 (14.3%)`, col=MissLevel*100, shape=Condition)) +
#   geom_point(size = 2.5) +
#   scale_color_continuous(name = &#39;Missing level (%)&#39;, type = &#39;viridis&#39;) +
#   th</code></pre>
<p><font size='4'> <strong>Show reasons for removing samples</strong>
</font><br />
Samples ‘1TK71I_TU’, ‘NG1IWD_TU’, ‘125CII_TU’, and ‘1LJXIN_TU’ are
removed due to low tumor purity and misclassification in
metadata-assisted quality control (Tumor vs Normal, PC1). Samples
‘I0HVOL_TU’ and ‘7EAOX7_TU’ got decent tumor purity but still got
removed due to misclassification and similar molecular signature as
normal samples (PC1, clustering). Sample ‘Q89YQK_NG’ is removed due to
high level of missingness (71%) and potential outlier (PC2).</p>
<p><strong>PCA</strong></p>
<pre class="r"><code># Prepare preprocessed data containing all samples
allProTissueVsn &lt;- doPreprocessing(allProTissueTab, feat = &#39;Feature&#39;, smp = &#39;Sample&#39;,
                                   val = &#39;Abundance&#39;, smpAnno = c(smpAnno, &#39;MissLevel&#39;),
                                   featAnno = &#39;Gene&#39;, do_featFilt = T, cutoff = 0.5)$vsn.data
# Do PCA
pcaRes &lt;- doSOA(allProTissueVsn, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;, num_PCs = 20,
                do_onlyPCA = T)
# Show significant associations between PCs and sample conditions
pcaRes$pcSigAssoRes</code></pre>
<pre><code>         Var1      Var2     pVal  pValAdj Stat   Test
1 PC1 (29.9%) Condition 1.51e-58 3.02e-57   25 T-test</code></pre>
<pre class="r"><code># Visualize PCs
pcTab &lt;- pcaRes$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;I0HVOL_NG&#39;,
                                                       &#39;7EAOX7_TU&#39;, &#39;7EAOX7_NG&#39;,
                                                       &#39;1LJXIN_TU&#39;,&#39;125CII_TU&#39;, 
                                                       &#39;1TK71I_TU&#39;, &#39;NG1IWD_TU&#39;) ~ Sample),
                Label2 = dplyr::case_when(Sample %in% &#39;Q89YQK_NG&#39; ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (29.9%)`, col=Condition, fill=Condition, label=Label)) +
  geom_boxplot(alpha = 0.7, outliers = F) +
  geom_jitter(position = position_jitter(0.3), size = 4, show.legend = F) +
  geom_text_repel(size = 6, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#   theme(axis.title = element_text(size = 22),
#         axis.text.x = element_text(size = 18),
#         axis.text.y = element_text(size = 14),
#         legend.title = element_text(size = 22),
#         legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/QC_combined_proTissue_Klin_PC1_allSmps.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 10)

# Check missing levels of samples
ggplot(pcTab, aes(x=`PC1 (29.9%)`, y=`PC2 (14%)`, col=MissLevel*100, shape=Condition, label=Label2)) +
  geom_point(size = 5) +
  geom_text_repel(size = 7, show.legend = F) +
  scale_color_continuous(name = &#39;Missing level (%)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-5-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 22),
  #       axis.text = element_text(size = 14),
  #       legend.title = element_text(size = 22),
  #       legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/QC_combined_proTissue_Klin_PC1&amp;2_allSmps.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p><strong>Hierarchical clustering</strong><br />
Top 100 significant features between Tumor and Normal samples are used,
of which Bad-quality samples were excluded for significance testing.
Features (rows) are ordered by t-statistics.</p>
<pre class="r"><code># Check if potential bad-quality Tumor samples get condition-associated molecular
# signature identified in good-quality samples
# Prepare top significant features between Tumor and Normal samples
# proTissueTab &lt;- dplyr::filter(allProTissueTab, !Sample %in% badSmps_Dis)
# proTissueVsn &lt;- doPreprocessing(proTissueTab, feat = &#39;Feature&#39;, smp = &#39;Sample&#39;,
#                                 val = &#39;Abundance&#39;, smpAnno = c(smpAnno, &#39;MissLevel&#39;),
#                                 featAnno = &#39;Gene&#39;, do_featFilt = T, cutoff = 0.5)$vsn.data
# soaRes &lt;- doSOA(proTissueVsn, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
#                 num_PCs = 20, use_proDA = T, level_metaVar = c(&#39;Tumor&#39;, &#39;Normal&#39;))
# saveRDS(soaRes, &#39;./data/Discovery/AG_Klingmueller/soaRes/proTissueRes_rmBadSmps.rds&#39;)
soaRes &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proTissueRes_rmBadSmps.rds&#39;)
topSigFeatIdx &lt;- order(abs(soaRes$featSigAssoRes$Stat), decreasing = T)[1:100]
topSigFeats &lt;- soaRes$featSigAssoRes$Var1[topSigFeatIdx]
# Order top significant features by t-statistics
topSigFeatStats &lt;- soaRes$featSigAssoRes$Stat[topSigFeatIdx]
featOrder &lt;- topSigFeats[order(topSigFeatStats, decreasing = T)]
# Prepare abundance matrix including only top significant features
datMat &lt;- dplyr::select(allProTissueTab, Feature, Sample, Abundance) %&gt;%
  tidyr::pivot_wider(names_from = &#39;Sample&#39;, values_from = &#39;Abundance&#39;) %&gt;%
  dplyr::filter(Feature %in% topSigFeats) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
# Cluster samples by sample conditions
normalSmpIdx &lt;- grep(&#39;_NG&#39; ,colnames(datMat))
smpOrder &lt;- c(colnames(datMat)[-normalSmpIdx], colnames(datMat)[normalSmpIdx])
# Prepare sample annotation table
smpAnnoTab &lt;- dplyr::select(allProTissueTab, Sample, Condition) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Color sample groups as same as in PC plots
annoCols &lt;- c(&#39;#1B9E77&#39;, &#39;#D95F02&#39;)
names(annoCols) &lt;- c(&#39;Normal&#39;, &#39;Tumor&#39;)
smpAnnoCols &lt;- list()
smpAnnoCols[[&#39;Condition&#39;]] &lt;- annoCols

# View molecular signature of all samples
datMat &lt;- datMat[featOrder, smpOrder]
pheatmap(datMat, scale = &#39;row&#39;, annotation_col = smpAnnoTab, annotation_colors = smpAnnoCols,
         color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100), clustering_method = &#39;ward.D2&#39;,
         cluster_rows = F, cluster_cols = F, show_rownames = F, fontsize_col = 5,
         main = &#39;All samples (170)&#39;)</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># View molecular signature of bad-quality samples
# rmSmps &lt;- c(&#39;NG1IWD_TU&#39;, &#39;1LJXIN_TU&#39;, &#39;1TK71I_TU&#39;, &#39;125CII_TU&#39;, &#39;7EAOX7_TU&#39;, &#39;I0HVOL_TU&#39;,
#             &#39;NG1IWD_NG&#39;, &#39;1LJXIN_NG&#39;, &#39;1TK71I_NG&#39;, &#39;125CII_NG&#39;, &#39;7EAOX7_NG&#39;, &#39;I0HVOL_NG&#39;)
badSmps &lt;- gsub(&#39;_TU&#39;, &#39;_NG&#39;, badSmps_Dis)
badSmps &lt;- c(badSmps, badSmps_Dis, c(&#39;1LJXIN_TU&#39;, &#39;7EAOX7_TU&#39;, &#39;I0HVOL_TU&#39;, &#39;1LJXIN_NG&#39;,
                                     &#39;7EAOX7_NG&#39;, &#39;I0HVOL_NG&#39;))
datMatSub &lt;- datMat[, colnames(datMat) %in% badSmps]
pheatmap(datMatSub, scale = &#39;row&#39;, annotation_col = smpAnnoTab, annotation_colors = smpAnnoCols,
         color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100), clustering_method = &#39;ward.D2&#39;,
         cluster_rows = F, show_rownames = F,
         main = &#39;Potential bad-quality samples and corresponding normal samples&#39;)</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-6-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display tumor purity and missing levels of misclassified samples and potential outlier
dplyr::distinct(allProTissueTab, Sample, .keep_all = T) %&gt;%
  dplyr::select(Sample, TumorPurity, MissLevel) %&gt;%
  dplyr::filter(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;, &#39;1LJXIN_TU&#39;, &#39;125CII_TU&#39;,
                              &#39;1TK71I_TU&#39;, &#39;NG1IWD_TU&#39;, &#39;Q89YQK_NG&#39;)) %&gt;%
  as.data.frame()</code></pre>
<pre><code>     Sample TumorPurity MissLevel
1 I0HVOL_TU          87 0.3462415
2 1LJXIN_TU          40 0.2705011
3 Q89YQK_NG          NA 0.7134586
4 7EAOX7_TU          70 0.2592065
5 125CII_TU          30 0.5099658
6 NG1IWD_TU          25 0.2764806
7 1TK71I_TU          10 0.4129651</code></pre>
<p>=&gt; Mean of sample tumor cell contents is 67%, ranging from 10% to
100%.<br />
=&gt; Mean of sample missing levels is 31%, ranging from 21% to 71%.</p>
<p></br> </br> <font size='5'> <strong>Check data reproducibility by
comparing New and Old Proteomics</strong> </font><br />
Note that only common samples between two datasets are considered.</p>
<pre class="r"><code># Prepare preprocessed data sharing same sample space
proTissue_Old &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
proTissue_New &lt;- dplyr::filter(allProTissueTab, Sample %in% colnames(proTissue_Old)) %&gt;%
  doPreprocessing(feat = &#39;Feature&#39;, smp = &#39;Sample&#39;, val = &#39;Abundance&#39;, featAnno = &#39;Gene&#39;,
                  smpAnno = smpAnno, do_featFilt = T, cutoff = 0.5)
proTissue_New &lt;- proTissue_New$vsn.data
# Prepare preprocessed data corrected for SVs
svaRes_New &lt;- doSVA(proTissue_New, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;), numSV_method = &#39;leek&#39;,
                    asso_metaVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                     &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;, &#39;TumorPurity&#39;))
svaRes_Old &lt;- doSVA(proTissue_Old, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;), numSV_method = &#39;leek&#39;,
                    asso_metaVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                     &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;, &#39;TumorPurity&#39;, &#39;Batch&#39;))
proTissueSVs_New &lt;- svaRes_New$summExp_SVs
proTissueSVs_Old &lt;- svaRes_Old$summExp_SVs</code></pre>
<p><font size='4'> <strong>Preprocessed data without any covariate
correction</strong> </font><br />
<strong>Feature correlations (data1: New data, data2: Old
data)</strong></p>
<pre class="r"><code># Compute feature correlations between two datasets
corrRes &lt;- calcFeatCorr(proTissue_New, proTissue_Old, show_nFeats = T)</code></pre>
<pre><code>Features in data1: 7408
Features in data2: 8250
Features in common: 6893</code></pre>
<pre class="r"><code># Plot feature correlations
ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-1, 1, 0.2))) +
  labs(y = &#39;Frequency&#39;, title = &#39;Feature Correlations: New vs Old Proteomics (Uncorrected)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 22),
  #       axis.text = element_text(size = 18))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/compFeatCorr_combined_proTissue_Klin_new_vs_old.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p><strong>log(FC) between Tumor and Normal samples</strong></p>
<pre class="r"><code># Compute log(FC) and t-statistics between Tumor and Normal tissue samples in two datasets
# Prepare protein-gene table for providing gene names
featInfo &lt;- tibble::as_tibble(rowData(proTissue_New), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::rename(Genes = Gene) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))
# Prepare summarized tables of comparison for visualization
# condiFeatAssoTabs &lt;- summarizeFeatAssoRes(proTissue_New, proTissue_Old, &#39;New&#39;, &#39;Old&#39;,
#                                           meta_var = &#39;Condition&#39;, subset = NULL,
#                                           method = &#39;proDA&#39;, alpha = 0.05, prot_gene_tbl = featInfo)
condiFeatAssoTabs &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/New_vs_Old_condiFeatAsso.rds&#39;)

# Plot log(FC)
ggplot(condiFeatAssoTabs$tbl_logFC, aes(x=logFC_dat1, y=logFC_dat2, col=Direction)) +
  geom_point(size = 3) +
  ggpubr::stat_cor(data = condiFeatAssoTabs$tbl_logFC,
                   aes(x=logFC_dat1, y=logFC_dat2, col=NULL,
                       label=paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;)),
                   size = 7, show.legend = F) +
  geom_abline(intercept = 0, slope = 1, col = &#39;black&#39;, linewidth = 0.8) +
  scale_color_manual(values = c(&#39;red&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;log(FC) of New data&#39;, y = &#39;log(FC) of Old data&#39;,
       title = &#39;log(FC): Tumor vs Normal (Uncorrected)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 22),
  #       axis.text = element_text(size = 14),
  #       legend.title = element_text(size = 22),
  #       legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/compLogFC_combined_proTissue_Klin_new_vs_old_condi.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p><strong>p-values (Tumor vs Normal samples, alpha = 0.05)</strong></p>
<pre class="r"><code># Plot p-values
# Compute result consistency
percent &lt;- round(sum(condiFeatAssoTabs$tbl_pVal$Significance == &#39;Sig. in Both&#39;) /
                   sum(condiFeatAssoTabs$tbl_pVal$Significance != &#39;Not Sig.&#39;), 3) * 100

ggplot(condiFeatAssoTabs$tbl_pVal, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance)) +
  geom_point(size = 2) +
  geom_abline(intercept = 0, slope = 1, col = &#39;black&#39;, linewidth = 0.8) +
  scale_color_manual(values = c(&#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;-log10(pVal) of New data&#39;, y = &#39;-log10(pVal) of Old data&#39;,
       title = &#39;p-val: Tumor vs Normal (Uncorrected)&#39;) +
  annotate(&#39;text&#39;, x = 5, y = 30, size = 4,
           label = paste0(percent, &#39;% of significant results consistent&#39;)) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<p></br> <font size='4'> <strong>Preprocessed data accounted for
SVs</strong> </font><br />
<strong>Feature correlations</strong></p>
<pre class="r"><code># Compute feature correlations between two datasets
corrRes &lt;- calcFeatCorr(svaRes_New$correctDatMat, svaRes_Old$correctDatMat, show_nFeats = F)
# Plot feature correlations
ggplot(corrRes, aes(x=Correlation)) +
  geom_histogram(binwidth = 0.1) +
  scale_x_continuous(breaks = c(seq(-1, 1, 0.1))) +
  labs(y = &#39;Frequency&#39;, title = &#39;Feature Correlations: New vs Old Proteomics (Corrected)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>log(FC) between Tumor and Normal samples</strong></p>
<pre class="r"><code># Compute log(FC) and t-statistics between Tumor and Normal tissue samples in two datasets
# Prepare summarized tables of comparison for visualization
# condiFeatAssoTabs_SVs &lt;- summarizeFeatAssoRes(proTissueSVs_New, proTissueSVs_Old,
#                                               &#39;New&#39;, &#39;Old&#39;, meta_var = &#39;Condition&#39;,
#                                               subset = NULL, method = &#39;proDA&#39;, alpha = 0.05,
#                                               unwantVar1 = &#39;SV1&#39;, unwantVar2 = &#39;SV1&#39;)
condiFeatAssoTabs_SVs &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/New_vs_Old_condiFeatAsso_SVs_leek.rds&#39;)

# Plot log(FC)
ggplot(condiFeatAssoTabs_SVs$tbl_logFC, aes(x=logFC_dat1, y=logFC_dat2, col=Direction)) +
  geom_point(size = 2) +
  ggpubr::stat_cor(data = condiFeatAssoTabs_SVs$tbl_logFC,
                   aes(x=logFC_dat1, y=logFC_dat2, col=NULL,
                       label=paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;)),
                   size = 6, show.legend = F) +
  geom_abline(intercept = 0, slope = 1, col = &#39;black&#39;, linewidth = 0.8) +
  scale_color_manual(values = c(&#39;red&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;log(FC) of New data&#39;, y = &#39;log(FC) of Old data&#39;,
       title = &#39;log(FC): Tumor vs Normal (SV-Corrected)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>p-values (Tumor vs Normal samples, alpha = 0.05)</strong></p>
<pre class="r"><code># Plot p-values
# Compute result consistency
percent &lt;- round(sum(condiFeatAssoTabs_SVs$tbl_pVal$Significance == &#39;Sig. in Both&#39;) /
                   sum(condiFeatAssoTabs_SVs$tbl_pVal$Significance != &#39;Not Sig.&#39;), 3) * 100

ggplot(condiFeatAssoTabs_SVs$tbl_pVal, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance)) +
  geom_point(size = 2) +
  geom_abline(intercept = 0, slope = 1, col = &#39;black&#39;, linewidth = 0.8) +
  scale_color_manual(values = c(&#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;-log10(pVal) of New data&#39;, y = &#39;-log10(pVal) of Old data&#39;,
       title = &#39;p-val: Tumor vs Normal (SV-Corrected)&#39;) +
  annotate(&#39;text&#39;, x = 5, y = 30, size = 4,
           label = paste0(percent, &#39;% of significant results consistent&#39;)) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" />
=&gt; SV-corrected data does not give rise to better result consistency.
</br> </br> </br> </br></p>
</div>
</div>
<div id="ag-krijgsveld" class="section level2">
<h2>AG Krijgsveld</h2>
<p>Combine Method Dev and Discovery cohorts</p>
<pre class="r"><code># Prepare long data containing original data and needed information
# Discovery
# Load original data
proTissueTab_Discovery &lt;- readr::read_tsv(&#39;./data/Discovery/AG_Krijgsveld/20230815_MarcS_Discovery_Tissue.pg_matrix.tsv&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Sample = stringr::str_remove_all(Sample, &#39;^.+SC_A_|^.+SC_DIS_A_|4\\.raw$&#39;))
# Include summarized metadata
proTissueTab_Discovery &lt;- dplyr::left_join(proTissueTab_Discovery, summMetadat4Krij, by = &#39;Sample&#39;)

# MethodDev
# Load and tidy up original data and include summarized metadata into it
summMetadat_Krij &lt;- dplyr::filter(summMetadat, To %in% &#39;KRIJGSVELD&#39;)
proTissueTab_MethodDev &lt;- readr::read_delim(&#39;./data/MethodDev/AG_Krijgsveld/20230726_Thorax_Method_Est_tissue.txt&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Aliquot&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Aliquot = stringr::str_remove_all(Aliquot, &#39;\\.raw$&#39;)) %&gt;%
  dplyr::left_join(summMetadat_Krij, by = &#39;Aliquot&#39;) %&gt;%
  dplyr::select(-Aliquot)

# Extract common features for combining data
featSpace_Discovery &lt;- unique(proTissueTab_Discovery$Protein.Group)
featSpace_MethodDev &lt;- unique(proTissueTab_MethodDev$Protein.Group)
cmnFeats &lt;- intersect(featSpace_Discovery, featSpace_MethodDev)
# Filter common features
proTissueTab_Discovery &lt;- dplyr::filter(proTissueTab_Discovery, Protein.Group %in% cmnFeats)
proTissueTab_MethodDev &lt;- dplyr::filter(proTissueTab_MethodDev, Protein.Group %in% cmnFeats)
# Combine data
proTissueTab_Combined &lt;- dplyr::bind_rows(proTissueTab_Discovery, proTissueTab_MethodDev)

# Preprocess data
proTissue_Combined &lt;- doPreprocessing(proTissueTab_Combined, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                      val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                      smpAnno = smpAnno, do_featFilt = T, cutoff = 0.5, viz_miss = T)#, save_path = &#39;./data/Discovery/AG_Krijgsveld/combined_proTissue&#39;)</code></pre>
<p>Show dimensions of original combined data (126 samples and 7091
common features)</p>
<pre class="r"><code>dim(proTissue_Combined$ori.data)</code></pre>
<pre><code>[1] 7091  126</code></pre>
<p>Show missingness of original combined data</p>
<pre class="r"><code>proTissue_Combined$ori.data.miss +
  theme(axis.text.x = element_text(size = 7))</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show distribution of original combined data</p>
<pre class="r"><code># Convert SE object into long data
longData &lt;- summExp2df(proTissue_Combined$ori.data, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;)
# Order samples based on cohorts so that same cohorts are clustered in plot
smpLevels &lt;- dplyr::arrange(longData, Cohort) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  dplyr::pull(Sample)
longData &lt;- dplyr::mutate(longData, Sample = factor(Sample, levels = smpLevels))
ggplot(longData, aes(x=Sample, y=Value, col=Cohort)) +
  geom_boxplot(outlier.size = 0.8) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_y_log10() +
  labs(y = &#39;Abundance&#39;) +
  th + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (126 samples and 6126 features)</p>
<pre class="r"><code>dim(proTissue_Combined$filt.data)</code></pre>
<pre><code>[1] 6126  126</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>proTissue_Combined$filt.data.miss +
  theme(axis.text.x = element_text(size = 7))</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code># Convert SE object into long data
longData &lt;- summExp2df(proTissue_Combined$vsn.data, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;)
# Order samples based on cohorts so that same cohorts are clustered in plot
smpLevels &lt;- dplyr::arrange(longData, Cohort) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  dplyr::pull(Sample)
longData &lt;- dplyr::mutate(longData, Sample = factor(Sample, levels = smpLevels))
ggplot(longData, aes(x=Sample, y=Value, col=Cohort)) +
  geom_boxplot(outlier.size = 0.8) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  labs(y = &#39;Abundance&#39;) +
  th + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>proTissue_Combined$vsn.feat.mean.var</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Do metadata-assisted quality control</strong><br />
Show significant associations between PCs and sample conditions and data
cohorts</p>
<pre class="r"><code># Do single-omics analysis
proTissueSOA_Combined &lt;- doSOA(proTissue_Combined$vsn.data, meta_var = c(&#39;Condition&#39;, &#39;Cohort&#39;),
                               pca_method = &#39;ppca&#39;, do_onlyPCA = T)
proTissueSOA_Combined$pcSigAssoRes</code></pre>
<pre><code>         Var1      Var2     pVal  pValAdj  Stat   Test
1   PC1 (34%)    Cohort 2.03e-80 8.11e-79 46.60 T-test
2 PC2 (24.9%) Condition 2.74e-44 5.48e-43 21.80 T-test
3 PC2 (24.9%)    Cohort 1.47e-02 1.96e-01 -2.47 T-test
4   PC1 (34%) Condition 2.91e-02 2.91e-01  2.21 T-test</code></pre>
<p>Visualize significant PCs</p>
<pre class="r"><code>pcTab &lt;- proTissueSOA_Combined$pcTab
ggplot(pcTab, aes(x=`PC1 (34%)`, y=`PC2 (24.9%)`, col=Cohort, shape=Condition)) +
  geom_point(size = 4) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_shape_manual(values = c(0, 16)) +
  th #+ theme(axis.title = element_text(size = 28),</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#              axis.text.x = element_text(size = 24), axis.text.y = element_text(size = 18),
#              legend.title = element_text(size = 26), legend.text = element_text(size = 24))
# ggsave(&#39;./output/Discovery/group_meeting/quality_control_beforeBC_combined_proTissue_Krij_condi.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 10)

# ggplot(pcTab, aes(x=Condition, y=`PC1 (34%)`, col=Condition, fill=Condition)) +
#   geom_boxplot(alpha = 0.7, outlier.shape = NA) +
#   geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
#   scale_color_brewer(palette = &#39;Dark2&#39;) +
#   scale_fill_brewer(palette = &#39;Dark2&#39;) +
#   ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
#                              method.args = list(var.equal = T),
#                              size = 6, show.legend = F) +
#   th</code></pre>
<p><strong>Do batch correction using limma</strong><br />
There is no PC significantly associated with data cohorts after batch
correction, which supports that batch effects are linear. Batch factor
(Cohort) should be included in linear model for statistical
analysis.</p>
<pre class="r"><code># Perform batch correction to see if two cohorts can be merged
# Convert SE object into wide data
datMat &lt;- SummarizedExperiment::assay(proTissue_Combined$vsn.data)
# Define design matrix to keep particular effects, e.g., treatments
smpAnnoTab &lt;- as_tibble(colData(proTissue_Combined$vsn.data), rownames = &#39;Sample&#39;)
design &lt;- model.matrix(~ smpAnnoTab$Condition + smpAnnoTab$Recurrence)
# Do batch correction
datMat_BC &lt;- limma::removeBatchEffect(datMat, batch = smpAnnoTab$Cohort, design = design)
proTissueSE_BC &lt;- proTissue_Combined$vsn.data
assay(proTissueSE_BC) &lt;- datMat_BC
# Save batch corrected data
# saveRDS(proTissueSE_BC, &#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)

# Do single-omics analysis
proTissueSOA_BC &lt;- doSOA(proTissueSE_BC, meta_var = c(&#39;Condition&#39;, &#39;Cohort&#39;),
                         pca_method = &#39;ppca&#39;, do_onlyPCA = T)
proTissueSOA_BC$pcSigAssoRes</code></pre>
<pre><code>       Var1      Var2     pVal  pValAdj  Stat   Test
1 PC1 (38%) Condition 6.55e-50 2.62e-48 -24.8 T-test</code></pre>
<p>Visualize PCs</p>
<pre class="r"><code>pcTab &lt;- proTissueSOA_BC$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;,
                                                       &#39;MJMTYR_TU&#39;, &#39;XFKHGP_TU&#39;,
                                                       &#39;XFKHGP_NG&#39;, &#39;MJMTYR_NG&#39;,
                                                       &#39;I0HVOL_NG&#39;, &#39;7EAOX7_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=`PC1 (38%)`, y=`PC2 (9.2%)`, col=Cohort, shape=Condition)) +
  geom_point(size = 4) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_shape_manual(values = c(0, 16)) +
  th #+ theme(axis.title = element_text(size = 28),</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#              axis.text.x = element_text(size = 24), axis.text.y = element_text(size = 18),
#              legend.title = element_text(size = 26), legend.text = element_text(size = 24))
# ggsave(&#39;./output/Discovery/group_meeting/quality_control_afterBC_combined_proTissue_Krij_condi.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 10)

ggplot(pcTab, aes(x=Condition, y=`PC1 (38%)`, col=Condition, fill=Condition, label = Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F, size = 5) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_01_preprocessing_DIA.Rmd/unnamed-chunk-26-2.png" width="960" style="display: block; margin: auto;" />
=&gt; Tumor purity: I0HVOL_TU - 87% / 7EAOX7_TU - 70% / MJMTYR_TU - 40%
/ XFKHGP_TU - 100%<br />
(Mean of sample tumor cell contents is 74%, ranging from 40% to
100%.)</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Warsaw
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] proDA_1.16.0                pcaMethods_1.94.0          
 [3] pheatmap_1.0.12             ggplotify_0.1.2            
 [5] lubridate_1.9.3             forcats_1.0.0              
 [7] stringr_1.5.1               dplyr_1.1.4                
 [9] purrr_1.0.2                 tidyr_1.3.1                
[11] tibble_3.2.1                tidyverse_2.0.0            
[13] MultiAssayExperiment_1.28.0 SummarizedExperiment_1.32.0
[15] GenomicRanges_1.54.1        GenomeInfoDb_1.38.8        
[17] IRanges_2.36.0              S4Vectors_0.40.2           
[19] MatrixGenerics_1.14.0       matrixStats_1.3.0          
[21] ggrepel_0.9.5               ggplot2_3.5.1              
[23] visdat_0.6.0                sva_3.50.0                 
[25] BiocParallel_1.36.0         genefilter_1.84.0          
[27] mgcv_1.9-1                  nlme_3.1-164               
[29] limma_3.58.1                vsn_3.70.0                 
[31] Biobase_2.62.0              BiocGenerics_0.48.1        
[33] readxl_1.4.3                readr_2.1.5                
[35] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      rstudioapi_0.16.0       jsonlite_1.8.8         
  [4] magrittr_2.0.3          farver_2.1.2            rmarkdown_2.27         
  [7] fs_1.6.4                zlibbioc_1.48.2         vctrs_0.6.5            
 [10] memoise_2.0.1           RCurl_1.98-1.14         rstatix_0.7.2          
 [13] htmltools_0.5.8.1       S4Arrays_1.2.1          broom_1.0.5            
 [16] cellranger_1.1.0        SparseArray_1.2.4       gridGraphics_0.5-1     
 [19] sass_0.4.9              bslib_0.7.0             cachem_1.1.0           
 [22] whisker_0.4.1           lifecycle_1.0.4         pkgconfig_2.0.3        
 [25] Matrix_1.6-5            R6_2.5.1                fastmap_1.2.0          
 [28] GenomeInfoDbData_1.2.11 digest_0.6.35           colorspace_2.1-0       
 [31] AnnotationDbi_1.64.1    ps_1.7.6                rprojroot_2.0.4        
 [34] RSQLite_2.3.6           ggpubr_0.6.0            labeling_0.4.3         
 [37] fansi_1.0.6             timechange_0.3.0        httr_1.4.7             
 [40] abind_1.4-5             compiler_4.3.2          bit64_4.0.5            
 [43] withr_3.0.0             backports_1.4.1         carData_3.0-5          
 [46] DBI_1.2.2               hexbin_1.28.3           highr_0.11             
 [49] ggsignif_0.6.4          DelayedArray_0.28.0     tools_4.3.2            
 [52] httpuv_1.6.15           glue_1.7.0              callr_3.7.6            
 [55] promises_1.3.0          grid_4.3.2              getPass_0.2-4          
 [58] generics_0.1.3          gtable_0.3.5            tzdb_0.4.0             
 [61] preprocessCore_1.64.0   hms_1.1.3               car_3.1-2              
 [64] utf8_1.2.4              XVector_0.42.0          pillar_1.9.0           
 [67] vroom_1.6.5             yulab.utils_0.1.4       later_1.3.2            
 [70] splines_4.3.2           lattice_0.22-6          survival_3.5-8         
 [73] bit_4.0.5               annotate_1.80.0         tidyselect_1.2.1       
 [76] locfit_1.5-9.9          Biostrings_2.70.3       knitr_1.47             
 [79] git2r_0.33.0            edgeR_4.0.16            xfun_0.44              
 [82] statmod_1.5.0           stringi_1.8.4           yaml_2.3.8             
 [85] evaluate_0.23           codetools_0.2-20        BiocManager_1.30.22    
 [88] cli_3.6.2               affyio_1.72.0           xtable_1.8-4           
 [91] munsell_0.5.1           processx_3.8.4          jquerylib_0.1.4        
 [94] Rcpp_1.0.12             png_0.1-8               XML_3.99-0.16.1        
 [97] parallel_4.3.2          blob_1.2.4              bitops_1.0-7           
[100] spsUtil_0.2.2           scales_1.3.0            affy_1.80.0            
[103] crayon_1.5.2            rlang_1.1.4             KEGGREST_1.42.0        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
