<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Single-omics analysis: Combined DIA Proteomics (Discovery + MethodDev)</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Single-omics analysis: Combined DIA
Proteomics (Discovery + MethodDev)</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree282daae0fdbb7078dd8d66521af4c5d654c63cfbtargetblank282daaea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/282daae0fdbb7078dd8d66521af4c5d654c63cfb" target="_blank">282daae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree282daae0fdbb7078dd8d66521af4c5d654c63cfbtargetblank282daaea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/282daae0fdbb7078dd8d66521af4c5d654c63cfb" target="_blank">282daae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/Discovery/
    Ignored:    data/MethodDev/
    Ignored:    data/aliquot_metadata.xlsx
    Ignored:    data/metadata_latest/
    Ignored:    data/patient_metadata.xlsx
    Ignored:    data/sample_metadata.xlsx
    Ignored:    output/.DS_Store
    Ignored:    output/Discovery/
    Ignored:    output/MethodDev/
    Ignored:    output/SC_meeting/

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/data_availability.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R

Unstaged changes:
    Modified:   analysis/FeatSelection_ML.Rmd
    Modified:   analysis/MDev_Dis_02_soa_annotated_untargeted.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/MDev_Dis_02_soa_DIA.Rmd</code>)
and HTML (<code>docs/MDev_Dis_02_soa_DIA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/258edf90a9453bb3eeea0a8c7632712f3a69184f/analysis/MDev_Dis_02_soa_DIA.Rmd" target="_blank">258edf9</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-29
</td>
<td>
Do protein abundance profiling of potential bad-quality tumor samples
and compare analysis results of newly and previously generated data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/8b66cfb8ac955dd6cc9e9d8b4e0ab474cebd82a5/analysis/MDev_Dis_02_soa_DIA.Rmd" target="_blank">8b66cfb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-16
</td>
<td>
Do SOA on new Klin. proteomics and combined annotated untargeted
lipidomics, recurrence vs nonrecurrence and tumor vs normal
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/e957730a0dc718d91bfb403da6c3e1c1f3239143/analysis/MDev_Dis_02_soa_DIA.Rmd" target="_blank">e957730</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-07-02
</td>
<td>
Preprocess combined Method Dev and Discovery Tissue Proteomics from AG
Klin. and rename and rearrange source code and analysis results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Conduct univariate (t-test) and
multivariate (PCA) analyses on combined Tissue DIA Proteomics from AG
Krijgsveld and AG Klingmüller to have overview of data and take initial
look at data power in terms of predicting patient cancer recurrence.
</font></p>
<p><strong>Metadata variables</strong><br />
Patients who got Tissue samples (n = 85):<br />
Recurrence -&gt; Cancer recurrences, Yes:No = 29:56<br />
Gender -&gt; Female:Male = 36:49<br />
Age -&gt; Diagnosis ages ranging from 51 to 84<br />
Smoking -&gt; Smoker:Ex-smoker:Non-Smoker = 18:56:11<br />
Stage -&gt; Pathological stages, IB:IIA:IIB = 50:17:18<br />
Adjuvant -&gt; Adjuvant chemotherapy, True:False = 26:59<br />
Samples:<br />
Condition -&gt; Tumor:Normal = 81:84</p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;pcaMethods&#39;)
library(&#39;sva&#39;)
library(&#39;limma&#39;)
library(&#39;proDA&#39;)
library(&#39;pheatmap&#39;)
library(&#39;ggplotify&#39;)
library(&#39;ggrepel&#39;)
library(&#39;ggvenn&#39;)
library(&#39;AnnotationDbi&#39;)
library(&#39;org.Hs.eg.db&#39;)
# This annotation object is accessed using AnnotationDbi:select()
hs &lt;- org.Hs.eg.db
library(&#39;msigdbr&#39;)
library(&#39;clusterProfiler&#39;)
library(spsUtil)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/analysis_pipeline.R&#39;)
source(&#39;./code/misc.R&#39;)
source(&#39;./code/comparison_funcs.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
    theme(axis.title = element_text(face = &#39;bold&#39;),
          axis.text = element_text(face = &#39;bold&#39;),
          axis.ticks = element_line(linewidth = 0.8),
          legend.text = element_text(size = 15))</code></pre>
<p>Display associations between cancer recurrence and the other patient
metadata variables to identify potential predictors (or confounders)</p>
<pre class="r"><code># Test associations between cancer recurrence and the other patient metadata variables
# Prepare patient metadata
proTissue &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
patientMetadat &lt;- tibble::as_tibble(colData(proTissue)) %&gt;%
  dplyr::filter(!duplicated(Patient)) %&gt;%
  dplyr::select(Patient, Recurrence, Gender, Age, Smoking, Stage, Adjuvant)
patientRecur &lt;- dplyr::select(patientMetadat, Patient, Recurrence)
patientMetadat &lt;- dplyr::select(patientMetadat, -Recurrence)
testAsso(patientMetadat, patientRecur, cmn_col = &#39;Patient&#39;) %&gt;%
  dplyr::select(-Stat)</code></pre>
<pre><code>      Var1       Var2       pVal   pValAdj       Test
1    Stage Recurrence 0.04766465 0.2383232 Chi-square
2  Smoking Recurrence 0.17097458 0.2470761 Chi-square
3 Adjuvant Recurrence 0.19171812 0.2470761 Chi-square
4   Gender Recurrence 0.19766086 0.2470761 Chi-square
5      Age Recurrence 0.57975337 0.5797534     T-test</code></pre>
<p>=&gt; All included patients are early-stage lung adenocarcinoma
patients.</p>
<p>Display contingency table between adjuvant chemotherapy and cancer
recurrence</p>
<pre class="r"><code>table(patientMetadat$Adjuvant, patientRecur$Recurrence)</code></pre>
<pre><code>       
        No Yes
  False 42  17
  True  14  12</code></pre>
<p>=&gt; Probability of patients receiving adjuvant chemotherapy
developing tumor recurrence is 46% in our study, and in general, 30-55%
lung cancer patients develop tumor recurrence.</p>
<div id="tissue-proteomics-ag-klingmüller" class="section level1">
<h1>Tissue Proteomics (AG Klingmüller)</h1>
<p>Discovery + MethodDev</p>
<p><font size='4'> <strong>Metadata-assisted quality control: Tumor vs
Normal samples</strong> </font></p>
<pre class="r"><code># Load preprocessed data
proTissue_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
# Perform single-omics analysis
proTissueRes_Klin &lt;- doSOA(proTissue_Klin, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
                           num_PCs = 20, do_onlyPCA = T)

# Visualize PCs
pcTab &lt;- proTissueRes_Klin$pcTab
ggplot(pcTab, aes(x=Condition, y=`PC1 (33.8%)`, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.7, outliers = F) +
  geom_jitter(position = position_jitter(0.3), size = 2, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Prepare data accounted for SVs for later visualizations
# Do SVA
proTissueSVA_Klin &lt;- doSVA(proTissue_Klin, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;),
                           numSV_method = &#39;be&#39;, asso_metaVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;,
                                                                 &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;,
                                                                 &#39;Stage&#39;, &#39;Adjuvant&#39;, &#39;TumorPurity&#39;))
# Show significant associations between SVs and sample metadata variables
# Sample condition accounts for large variance in data, so perhaps to preserve effects
# of recurrence, some variance contributed from condition is eliminated.
# proTissueSVA_Klin$sigAssoTab
# Perform single-omics analysis
proTissueSVs_Klin &lt;- proTissueSVA_Klin$summExp_SVs
SVs &lt;- grep(&#39;^SV\\d+$&#39;, colnames(colData(proTissueSVs_Klin)), value = T)
proTissueRes4Viz_Klin &lt;- doSOA(proTissueSVs_Klin, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
                               num_PCs = 20, do_onlyPCA = T, unwantVar = SVs)

# Visualize PCs
pcTab &lt;- proTissueRes4Viz_Klin$pcTab
# ggplot(pcTab, aes(x=Condition, y=`PC1 (57.8%)`, col=Condition, fill=Condition)) +
#   geom_boxplot(alpha = 0.7, outliers = F) +
#   geom_jitter(position = position_jitter(0.3), size = 2, show.legend = F) +
#   scale_color_brewer(palette = &#39;Dark2&#39;) +
#   scale_fill_brewer(palette = &#39;Dark2&#39;) +
#   th</code></pre>
<pre class="r"><code># Prepare median normalized data accounted for SVs for later computing log2(FC)
# Load preprocessed data
proTissueMedi_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueMedi.rds&#39;)
# Do SVA
proTissueMediSVA_Klin &lt;- doSVA(proTissueMedi_Klin, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;),
                               numSV_method = &#39;be&#39;, asso_metaVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;,
                                                                     &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;,
                                                                     &#39;Stage&#39;, &#39;Adjuvant&#39;, &#39;TumorPurity&#39;))
# Show significant associations between SVs and sample metadata variables
# proTissueMediSVA_Klin$sigAssoTab

# Perform single-omics analysis
proTissueMediSVs_Klin &lt;- proTissueMediSVA_Klin$summExp_SVs
SVs &lt;- grep(&#39;^SV\\d+$&#39;, colnames(colData(proTissueMediSVs_Klin)), value = T)
# proTissueMediRes_Klin &lt;- doSOA(proTissueMediSVs_Klin, meta_var = &#39;Condition&#39;,
#                                pca_method = &#39;ppca&#39;, num_PCs = 20, do_onlyPCA = T, unwantVar = SVs)
# Show significant associations between PCs and sample conditions
# proTissueMediRes_Klin
# Visualize PCs
# pcTab &lt;- proTissueMediRes_Klin$pcTab
# ggplot(pcTab, aes(x=Condition, y=`PC1 (46.4%)`, col=Condition, fill=Condition)) +
#   geom_boxplot(alpha = 0.7, outliers = F) +
#   geom_jitter(position = position_jitter(0.3), size = 2, show.legend = F) +
#   scale_color_brewer(palette = &#39;Dark2&#39;) +
#   scale_fill_brewer(palette = &#39;Dark2&#39;) +
#   th</code></pre>
<div id="normal-samples" class="section level2">
<h2>Normal samples</h2>
<pre class="r"><code># Subset certain samples
proNormal_Klin &lt;- proTissue_Klin[, which(colData(proTissue_Klin)$Condition == &#39;Normal&#39;)]
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proNormal_Klin), rownames = &#39;Protein&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
# Perform single-omics analysis
# proNormalRes_Klin &lt;- doSingleOmicsAnalysis(proNormal_Klin, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                            pca_method = &#39;ppca&#39;, num_PCs = 20, num_PCfeats = 30,
#                                            use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                            feat_conv = T, plot_title = NULL, show_rownames = F,
#                                            pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                            &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;),
#                                            # soa_unwantVar = &#39;Stage&#39;,
#                                            level_metaVar = c(&#39;Yes&#39;, &#39;No&#39;))
# fontsize = 15, fontsize_col = 14
# saveRDS(proNormalRes_Klin, &#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalRes.rds&#39;)
proNormalRes_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_Stage.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proNormalRes_Klin$data.var.source</code></pre>
<pre><code>          Var1       Var2     pVal pValAdj   Stat        Test
1   PC7 (5.5%)    Smoking 0.000181  0.0218  9.600       ANOVA
2  PC12 (2.8%)        Age 0.003310  0.1780  0.317 Correlation
3   PC7 (5.5%)   Adjuvant 0.004450  0.1780  2.930      T-test
4  PC1 (14.5%) Recurrence 0.011200  0.2820  2.600      T-test
5  PC12 (2.8%)      Stage 0.014800  0.2820  4.440       ANOVA
6   PC5 (6.7%)    Smoking 0.017000  0.2820  4.290       ANOVA
7  PC18 (1.9%)    Smoking 0.018600  0.2820  4.190       ANOVA
8   PC9 (3.8%)        Age 0.018800  0.2820  0.256 Correlation
9  PC11 (2.9%)     Gender 0.022100  0.2950  2.330      T-test
10 PC10 (3.1%) Recurrence 0.025100  0.3010 -2.280      T-test
11 PC10 (3.1%)      Stage 0.034200  0.3730  3.520       ANOVA
12 PC20 (1.6%) Recurrence 0.042000  0.3990  2.070      T-test
13 PC13 (2.6%)      Stage 0.043200  0.3990  3.270       ANOVA</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>dplyr::select(proNormalRes_Klin$sig.feat.tab, -Test)</code></pre>
<pre><code>             Var1       Var2     pVal pValAdj  Stat          Gene
1          P98179 Recurrence 0.000241   0.352 -3.84          RBM3
2          Q53GD3 Recurrence 0.000321   0.352 -3.76       SLC44A4
3          O94850 Recurrence 0.000329   0.352 -3.75           DDN
4          O43734 Recurrence 0.000494   0.352 -3.63      TRAF3IP2
5          Q13151 Recurrence 0.000596   0.352 -3.58       HNRNPA0
6          Q9NQ48 Recurrence 0.000627   0.352 -3.56        LZTFL1
7          O14949 Recurrence 0.000711   0.352 -3.52         UQCRQ
8          Q92817 Recurrence 0.000717   0.352 -3.52          EVPL
9          O00442 Recurrence 0.000772   0.352  3.50          RTCA
10         P84022 Recurrence 0.001040   0.352  3.40         SMAD3
11         Q5SRE7 Recurrence 0.001100   0.352  3.39        PHYHD1
12         Q9BUQ8 Recurrence 0.001150   0.352 -3.37         DDX23
13         Q01130 Recurrence 0.001160   0.352 -3.37         SRSF2
14         Q16537 Recurrence 0.001200   0.352 -3.36       PPP2R5E
15         Q9UBS8 Recurrence 0.001210   0.352  3.36         RNF14
16         Q9BZQ8 Recurrence 0.001270   0.352 -3.34        NIBAN1
17         Q14376 Recurrence 0.001360   0.352  3.32          GALE
18         Q14141 Recurrence 0.001430   0.352 -3.30       SEPTIN6
19         Q14203 Recurrence 0.001440   0.352 -3.30         DCTN1
20         Q14683 Recurrence 0.001540   0.352 -3.28         SMC1A
21         Q07955 Recurrence 0.001710   0.352 -3.25         SRSF1
22         P51687 Recurrence 0.001740   0.352  3.24          SUOX
23         P28070 Recurrence 0.001790   0.352  3.23         PSMB4
24         P08670 Recurrence 0.001850   0.352 -3.22           VIM
25         Q04864 Recurrence 0.001880   0.352  3.22           REL
26         Q9BUT1 Recurrence 0.001930   0.352  3.21          BDH2
27         Q96SB3 Recurrence 0.002050   0.352 -3.19       PPP1R9B
28         Q9BX66 Recurrence 0.002070   0.352 -3.18        SORBS1
29         Q13228 Recurrence 0.002110   0.352  3.18      SELENBP1
30         Q9Y3D3 Recurrence 0.002140   0.352  3.17        MRPS16
31         Q9NP97 Recurrence 0.002160   0.352 -3.17       DYNLRB1
32         P15311 Recurrence 0.002220   0.352 -3.16           EZR
33         Q01082 Recurrence 0.002300   0.352 -3.15        SPTBN1
34         Q15276 Recurrence 0.002320   0.352 -3.15        RABEP1
35         P12814 Recurrence 0.002350   0.352 -3.14         ACTN1
36         O75449 Recurrence 0.002370   0.352  3.14        KATNA1
37         Q9Y3B8 Recurrence 0.002570   0.352  3.11         REXO2
38         P35580 Recurrence 0.002590   0.352 -3.11         MYH10
39         Q9H019 Recurrence 0.002620   0.352 -3.11        MTFR1L
40         Q9NVG8 Recurrence 0.002630   0.352  3.11       TBC1D13
41         Q07866 Recurrence 0.002660   0.352 -3.10          KLC1
42         Q8TBC4 Recurrence 0.002660   0.352  3.10          UBA3
43         Q8N9T8 Recurrence 0.002720   0.352  3.09          KRI1
44         P07099 Recurrence 0.002780   0.352  3.09         EPHX1
45         P82979 Recurrence 0.002790   0.352 -3.09         SARNP
46         Q13387 Recurrence 0.002810   0.352 -3.08      MAPK8IP2
47         Q96CV9 Recurrence 0.002810   0.352 -3.08          OPTN
48         Q8N386 Recurrence 0.002890   0.352  3.07        LRRC25
49         O00187 Recurrence 0.002940   0.352 -3.07         MASP2
50         O43464 Recurrence 0.002950   0.352  3.07         HTRA2
51         Q8IUX7 Recurrence 0.003030   0.352 -3.06         AEBP1
52         Q9P281 Recurrence 0.003120   0.352  3.05        BAHCC1
53         Q16891 Recurrence 0.003150   0.352 -3.05          IMMT
54         O43639 Recurrence 0.003210   0.352 -3.04          NCK2
55         Q9UPN3 Recurrence 0.003250   0.352 -3.03         MACF1
56         Q13427 Recurrence 0.003340   0.352 -3.02          PPIG
57         P09234 Recurrence 0.003370   0.352 -3.02         SNRPC
58         O43719 Recurrence 0.003500   0.352 -3.01       HTATSF1
59         Q9UQ35 Recurrence 0.003580   0.352 -3.00         SRRM2
60         O43707 Recurrence 0.003700   0.352 -2.99         ACTN4
61         P20933 Recurrence 0.003740   0.352  2.99           AGA
62         Q8WVM7 Recurrence 0.003780   0.352 -2.98         STAG1
63         Q9Y285 Recurrence 0.003830   0.352 -2.98         FARSA
64         Q96BH1 Recurrence 0.004030   0.352 -2.96         RNF25
65         Q9BZF9 Recurrence 0.004040   0.352 -2.96          UACA
66         P12955 Recurrence 0.004050   0.352  2.96          PEPD
67         P51608 Recurrence 0.004050   0.352 -2.96         MECP2
68         P61966 Recurrence 0.004110   0.352  2.95         AP1S1
69         P56378 Recurrence 0.004130   0.352  2.95        ATP5MJ
70         O95630 Recurrence 0.004160   0.352 -2.95        STAMBP
71         O60861 Recurrence 0.004170   0.352 -2.95          GAS7
72         P12270 Recurrence 0.004170   0.352 -2.95           TPR
73         P60660 Recurrence 0.004210   0.352 -2.95          MYL6
74         P62314 Recurrence 0.004210   0.352  2.95        SNRPD1
75         P02462 Recurrence 0.004240   0.352  2.94        COL4A1
76         O00142 Recurrence 0.004260   0.352  2.94           TK2
77         Q9BRK3 Recurrence 0.004270   0.352 -2.94         MXRA8
78         Q9Y399 Recurrence 0.004300   0.352  2.94         MRPS2
79         O60437 Recurrence 0.004410   0.352 -2.93           PPL
80         P62318 Recurrence 0.004440   0.352  2.93        SNRPD3
81         P49720 Recurrence 0.004480   0.352  2.92         PSMB3
82         Q9H6S3 Recurrence 0.004560   0.352 -2.92        EPS8L2
83         Q7Z406 Recurrence 0.004590   0.352 -2.92         MYH14
84         Q93052 Recurrence 0.004600   0.352 -2.92           LPP
85         P42898 Recurrence 0.004630   0.352  2.91         MTHFR
86         P29372 Recurrence 0.004650   0.352  2.91           MPG
87         Q5UIP0 Recurrence 0.004690   0.352  2.91          RIF1
88         P0DPK5 Recurrence 0.004700   0.352  2.91          H3Y2
89         O94875 Recurrence 0.004780   0.352 -2.90        SORBS2
90         Q12888 Recurrence 0.004820   0.352 -2.90       TP53BP1
91         Q03252 Recurrence 0.004850   0.352 -2.90         LMNB2
92         P17661 Recurrence 0.004870   0.352 -2.90           DES
93         Q13200 Recurrence 0.004880   0.352 -2.90         PSMD2
94         Q9Y266 Recurrence 0.004920   0.352 -2.89          NUDC
95         Q9BUE6 Recurrence 0.004930   0.352  2.89         ISCA1
96         Q13185 Recurrence 0.004960   0.352 -2.89          CBX3
97         P23219 Recurrence 0.004970   0.352  2.89         PTGS1
98         Q9BVM4 Recurrence 0.004970   0.352  2.89         GGACT
99         Q96QE2 Recurrence 0.004990   0.352  2.89       SLC2A13
100        Q86V85 Recurrence 0.005040   0.352 -2.88        GPR180
101        Q92539 Recurrence 0.005050   0.352  2.88         LPIN2
102        P43243 Recurrence 0.005090   0.352 -2.88         MATR3
103        Q13813 Recurrence 0.005100   0.352 -2.88        SPTAN1
104        O95816 Recurrence 0.005180   0.352 -2.87          BAG2
105        Q13595 Recurrence 0.005190   0.352  2.87         TRA2A
106        Q13247 Recurrence 0.005300   0.352 -2.87         SRSF6
107        P04083 Recurrence 0.005310   0.352 -2.87         ANXA1
108        Q16539 Recurrence 0.005390   0.352  2.86        MAPK14
109        Q03001 Recurrence 0.005430   0.352 -2.86           DST
110        P35579 Recurrence 0.005550   0.352 -2.85          MYH9
111        Q15424 Recurrence 0.005630   0.352 -2.85          SAFB
112        Q9C0E2 Recurrence 0.005630   0.352  2.85          XPO4
113        Q13546 Recurrence 0.005670   0.352  2.84         RIPK1
114        P20700 Recurrence 0.005760   0.352 -2.84         LMNB1
115        P84243 Recurrence 0.005760   0.352  2.84         H3-3B
116        P31943 Recurrence 0.005770   0.352 -2.84       HNRNPH1
117        O14562 Recurrence 0.005790   0.352  2.84         UBFD1
118        Q9UPQ0 Recurrence 0.005820   0.352 -2.83        LIMCH1
119        P26038 Recurrence 0.005850   0.352 -2.83           MSN
120        O15061 Recurrence 0.005910   0.352 -2.83          SYNM
121        Q9Y5S2 Recurrence 0.005910   0.352 -2.83      CDC42BPB
122        O43374 Recurrence 0.005920   0.352  2.83         RASA4
123        P21333 Recurrence 0.005990   0.352 -2.82          FLNA
124        P53985 Recurrence 0.005990   0.352  2.82       SLC16A1
125        P13498 Recurrence 0.006130   0.353  2.82          CYBA
126        O75116 Recurrence 0.006140   0.353 -2.82         ROCK2
127        Q15287 Recurrence 0.006180   0.353 -2.81         RNPS1
128        Q5TZA2 Recurrence 0.006390   0.353 -2.80         CROCC
129        P50440 Recurrence 0.006420   0.353  2.80          GATM
130        P32856 Recurrence 0.006430   0.353  2.80          STX2
131        P09132 Recurrence 0.006510   0.353  2.79         SRP19
132        Q9Y490 Recurrence 0.006520   0.353 -2.79          TLN1
133        Q6P1M0 Recurrence 0.006570   0.353  2.79       SLC27A4
134        O15270 Recurrence 0.006770   0.353  2.78        SPTLC2
135        Q9UNE7 Recurrence 0.006810   0.353 -2.78         STUB1
136        P28068 Recurrence 0.006830   0.353  2.78       HLA-DMB
137        Q99733 Recurrence 0.006900   0.353 -2.77        NAP1L4
138        P02538 Recurrence 0.006910   0.353 -2.77         KRT6A
139        O15258 Recurrence 0.006950   0.353  2.77          RER1
140        Q9UQE7 Recurrence 0.007000   0.353 -2.77          SMC3
141        Q16629 Recurrence 0.007010   0.353 -2.77         SRSF7
142        O14979 Recurrence 0.007050   0.353 -2.77       HNRNPDL
143        Q13464 Recurrence 0.007260   0.353 -2.76         ROCK1
144        P31930 Recurrence 0.007280   0.353 -2.75        UQCRC1
145        O75410 Recurrence 0.007310   0.353 -2.75         TACC1
146        Q12874 Recurrence 0.007370   0.353 -2.75         SF3A3
147        Q6WCQ1 Recurrence 0.007390   0.353 -2.75         MPRIP
148        Q9Y2J4 Recurrence 0.007390   0.353  2.75        AMOTL2
149        P27816 Recurrence 0.007400   0.353 -2.75          MAP4
150        Q9HAU5 Recurrence 0.007410   0.353  2.75          UPF2
151        Q6ZT21 Recurrence 0.007460   0.353  2.75         TMPPE
152        Q0VF96 Recurrence 0.007480   0.353 -2.74         CGNL1
153        Q15459 Recurrence 0.007480   0.353 -2.74         SF3A1
154        P62333 Recurrence 0.007520   0.353 -2.74         PSMC6
155        Q86UP2 Recurrence 0.007550   0.353 -2.74          KTN1
156        P84103 Recurrence 0.007620   0.353 -2.74         SRSF3
157        P24844 Recurrence 0.007690   0.353 -2.73          MYL9
158        O60216 Recurrence 0.007790   0.353 -2.73         RAD21
159        Q9NYV4 Recurrence 0.007840   0.353  2.73         CDK12
160        Q14980 Recurrence 0.007910   0.353 -2.72         NUMA1
161        O60941 Recurrence 0.007950   0.353 -2.72          DTNB
162        Q9C0I1 Recurrence 0.007990   0.353  2.72        MTMR12
163        Q9Y383 Recurrence 0.008000   0.353 -2.72        LUC7L2
164        Q7Z4W1 Recurrence 0.008070   0.353  2.72          DCXR
165        P49756 Recurrence 0.008120   0.353 -2.71         RBM25
166        Q9Y2U5 Recurrence 0.008140   0.353 -2.71        MAP3K2
167        P08754 Recurrence 0.008150   0.353  2.71         GNAI3
168        O14818 Recurrence 0.008240   0.353  2.71         PSMA7
169        O95071 Recurrence 0.008240   0.353  2.71          UBR5
170        Q16718 Recurrence 0.008270   0.353 -2.71        NDUFA5
171        P61927 Recurrence 0.008280   0.353 -2.71         RPL37
172        P02545 Recurrence 0.008360   0.354 -2.70          LMNA
173        Q02338 Recurrence 0.008400   0.354  2.70          BDH1
174        Q9Y2D5 Recurrence 0.008480   0.355 -2.70         AKAP2
175        P16333 Recurrence 0.008520   0.355 -2.70          NCK1
176        Q9Y371 Recurrence 0.008710   0.359 -2.69       SH3GLB1
177        Q99729 Recurrence 0.008720   0.359 -2.69       HNRNPAB
178        P10606 Recurrence 0.008780   0.359 -2.69         COX5B
179        O43765 Recurrence 0.008830   0.359 -2.68          SGTA
180        P02511 Recurrence 0.008870   0.359 -2.68         CRYAB
181        Q96CB8 Recurrence 0.008960   0.359  2.68        INTS12
182        P30622 Recurrence 0.009000   0.359 -2.68         CLIP1
183        Q96GX9 Recurrence 0.009010   0.359  2.68          APIP
184        Q9BZ95 Recurrence 0.009220   0.360  2.67          NSD3
185        Q8NBJ7 Recurrence 0.009280   0.360 -2.67         SUMF2
186        Q92523 Recurrence 0.009330   0.360 -2.66         CPT1B
187        Q0ZGT2 Recurrence 0.009370   0.360 -2.66          NEXN
188        P13073 Recurrence 0.009480   0.360 -2.66        COX4I1
189        P49662 Recurrence 0.009510   0.360  2.66         CASP4
190        P48681 Recurrence 0.009600   0.360 -2.65           NES
191        Q9NVP2 Recurrence 0.009620   0.360 -2.65         ASF1B
192        P61018 Recurrence 0.009690   0.360  2.65         RAB4B
193        P61077 Recurrence 0.009750   0.360  2.65        UBE2D3
194        Q9Y572 Recurrence 0.009760   0.360  2.65         RIPK3
195        Q13572 Recurrence 0.009780   0.360  2.65         ITPK1
196        P08727 Recurrence 0.009820   0.360 -2.65         KRT19
197        Q16643 Recurrence 0.009820   0.360 -2.65          DBN1
198        O60760 Recurrence 0.009840   0.360  2.64         HPGDS
199        Q99570 Recurrence 0.009960   0.360  2.64        PIK3R4
200        O75054 Recurrence 0.009970   0.360  2.64         IGSF3
201        P28332 Recurrence 0.009990   0.360  2.64          ADH6
202        Q9H7B4 Recurrence 0.010000   0.360  2.64         SMYD3
203        Q9Y679 Recurrence 0.010000   0.360  2.64          AUP1
204        P08572 Recurrence 0.010200   0.362  2.63        COL4A2
205        P29692 Recurrence 0.010200   0.362 -2.63         EEF1D
206        Q16204 Recurrence 0.010200   0.362 -2.63         CCDC6
207        Q9NX63 Recurrence 0.010300   0.362 -2.63        CHCHD3
208        O75781 Recurrence 0.010400   0.363 -2.62          PALM
209        O95081 Recurrence 0.010400   0.363  2.62         AGFG2
210        O60462 Recurrence 0.010500   0.363  2.62          NRP2
211        Q8IWE2 Recurrence 0.010500   0.363 -2.62      FAM114A1
212        O15031 Recurrence 0.010600   0.365 -2.62        PLXNB2
213        O43708 Recurrence 0.010700   0.365  2.61         GSTZ1
214        P09496 Recurrence 0.010700   0.365 -2.61          CLTA
215        O75494 Recurrence 0.010800   0.365 -2.61        SRSF10
216        P28331 Recurrence 0.010800   0.365 -2.61        NDUFS1
217        P10620 Recurrence 0.010900   0.366  2.61         MGST1
218        Q6NZI2 Recurrence 0.011100   0.371 -2.60        CAVIN1
219        Q9H2D6 Recurrence 0.011100   0.371 -2.60        TRIOBP
220        Q13509 Recurrence 0.011300   0.373  2.59         TUBB3
221        Q15631 Recurrence 0.011300   0.373  2.59           TSN
222        O95760 Recurrence 0.011500   0.375 -2.59          IL33
223        Q14689 Recurrence 0.011500   0.375  2.59         DIP2A
224        Q9NQG5 Recurrence 0.011500   0.375 -2.59        RPRD1B
225        Q09028 Recurrence 0.011600   0.376 -2.58         RBBP4
226        P31942 Recurrence 0.011800   0.376 -2.58       HNRNPH3
227        O60504 Recurrence 0.011900   0.376 -2.58        SORBS3
228        P32321 Recurrence 0.011900   0.376  2.57          DCTD
229        O15173 Recurrence 0.012000   0.376 -2.57        PGRMC2
230        Q07092 Recurrence 0.012000   0.376  2.57       COL16A1
231        Q92878 Recurrence 0.012000   0.376 -2.57         RAD50
232        P37275 Recurrence 0.012100   0.376 -2.57          ZEB1
233        Q07065 Recurrence 0.012100   0.376 -2.57         CKAP4
234        Q92562 Recurrence 0.012200   0.376  2.56          FIG4
235        O95219 Recurrence 0.012300   0.376  2.56          SNX4
236        P09493 Recurrence 0.012300   0.376 -2.56          TPM1
237        Q8IX04 Recurrence 0.012300   0.376  2.56         UEVLD
238        Q9H008 Recurrence 0.012300   0.376  2.56          LHPP
239        O15126 Recurrence 0.012500   0.376 -2.55        SCAMP1
240        P08842 Recurrence 0.012500   0.376  2.56           STS
241        Q86UT6 Recurrence 0.012500   0.376  2.55         NLRX1
242        Q658P3 Recurrence 0.012600   0.376  2.55        STEAP3
243        O75190 Recurrence 0.012800   0.376 -2.55        DNAJB6
244        O75347 Recurrence 0.012800   0.376 -2.55          TBCA
245        Q70IA6 Recurrence 0.012800   0.376  2.55          MOB2
246        Q7Z7B0 Recurrence 0.012800   0.376 -2.55        FILIP1
247        Q8NFU3 Recurrence 0.012800   0.376 -2.55         TSTD1
248        Q9UK23 Recurrence 0.012800   0.376  2.55         NAGPA
249        P25787 Recurrence 0.012900   0.376  2.54         PSMA2
250        Q00536 Recurrence 0.012900   0.376  2.54         CDK16
251        P46734 Recurrence 0.013000   0.379  2.54        MAP2K3
252        Q9NRL3 Recurrence 0.013100   0.379 -2.54         STRN4
253        P08621 Recurrence 0.013200   0.379 -2.54       SNRNP70
254        Q15149 Recurrence 0.013300   0.379 -2.53          PLEC
255        Q5SZK8 Recurrence 0.013300   0.379 -2.53         FREM2
256        P46777 Recurrence 0.013400   0.379 -2.53          RPL5
257        Q9Y6X9 Recurrence 0.013400   0.379  2.53         MORC2
258        Q96HR9 Recurrence 0.013500   0.379  2.53         REEP6
259        O94804 Recurrence 0.013600   0.379 -2.52         STK10
260        O95810 Recurrence 0.013600   0.379 -2.52        CAVIN2
261        Q8WUX9 Recurrence 0.013600   0.379  2.52         CHMP7
262        P02533 Recurrence 0.013700   0.379 -2.52         KRT14
263        P27824 Recurrence 0.013700   0.379 -2.52          CANX
264        Q9Y5S9 Recurrence 0.013700   0.379 -2.52         RBM8A
265        O15155 Recurrence 0.013800   0.379  2.52          BET1
266        Q6P587 Recurrence 0.013900   0.380  2.52         FAHD1
267        Q96BP2 Recurrence 0.014000   0.381  2.51        CHCHD1
268        P46939 Recurrence 0.014100   0.384 -2.51          UTRN
269        P22415 Recurrence 0.014200   0.385  2.51          USF1
270        Q7L5N1 Recurrence 0.014200   0.385  2.51         COPS6
271        P41208 Recurrence 0.014400   0.386 -2.50         CETN2
272        Q08170 Recurrence 0.014400   0.386 -2.50         SRSF4
273        Q9Y295 Recurrence 0.014500   0.386  2.50          DRG1
274        P60900 Recurrence 0.014800   0.393  2.49         PSMA6
275        P00338 Recurrence 0.014900   0.394  2.49          LDHA
276        Q9H5X1 Recurrence 0.014900   0.394  2.49        CIAO2A
277        P62081 Recurrence 0.015100   0.397  2.48          RPS7
278 O14950;P19105 Recurrence 0.015200   0.397 -2.48 MYL12B;MYL12A
279        P57105 Recurrence 0.015200   0.397 -2.48       SYNJ2BP
280        Q8WWM9 Recurrence 0.015200   0.397  2.48          CYGB
281        P36873 Recurrence 0.015400   0.398  2.48        PPP1CC
282        P08493 Recurrence 0.015600   0.398 -2.47           MGP
283        O15021 Recurrence 0.015700   0.398 -2.47         MAST4
284        P00390 Recurrence 0.015900   0.398  2.46           GSR
285        Q53H96 Recurrence 0.015900   0.398  2.46         PYCR3
286        Q9BWJ5 Recurrence 0.015900   0.398 -2.46         SF3B5
287        O75023 Recurrence 0.016000   0.398  2.46        LILRB5
288        P15289 Recurrence 0.016000   0.398  2.46          ARSA
289        Q9Y2T2 Recurrence 0.016000   0.398  2.46         AP3M1
290        P62877 Recurrence 0.016200   0.398  2.46          RBX1
291        Q96EQ0 Recurrence 0.016200   0.398 -2.46          SGTB
292        Q96H79 Recurrence 0.016200   0.398  2.46      ZC3HAV1L
293        Q9H799 Recurrence 0.016200   0.398 -2.46       CPLANE1
294        Q9NYB9 Recurrence 0.016200   0.398 -2.46          ABI2
295        O00151 Recurrence 0.016400   0.398 -2.45        PDLIM1
296        P16930 Recurrence 0.016400   0.398  2.45           FAH
297        P50452 Recurrence 0.016400   0.398  2.45      SERPINB8
298        Q07021 Recurrence 0.016400   0.398 -2.45         C1QBP
299        Q8N0U8 Recurrence 0.016400   0.398  2.45      VKORC1L1
300        Q8WUW1 Recurrence 0.016400   0.398 -2.45          BRK1
301        Q9HAB8 Recurrence 0.016500   0.398  2.45          PPCS
302        P21810 Recurrence 0.016600   0.401 -2.45           BGN
303        Q8IWU2 Recurrence 0.016600   0.400  2.45         LMTK2
304        O75063 Recurrence 0.016800   0.402  2.44        FAM20B
305        Q9HB21 Recurrence 0.016800   0.402  2.44       PLEKHA1
306        P55209 Recurrence 0.016900   0.402 -2.44        NAP1L1
307        Q9H6Y5 Recurrence 0.016900   0.402  2.44         MAGIX
308        P61289 Recurrence 0.017000   0.403  2.44         PSME3
309        P52758 Recurrence 0.017100   0.403  2.43          RIDA
310        P52597 Recurrence 0.017200   0.403 -2.43        HNRNPF
311        Q14BN4 Recurrence 0.017400   0.403 -2.43         SLMAP
312        Q99747 Recurrence 0.017400   0.403 -2.43          NAPG
313        O14639 Recurrence 0.017500   0.403 -2.43        ABLIM1
314        P46060 Recurrence 0.017500   0.403 -2.43       RANGAP1
315        Q8TBP5 Recurrence 0.017500   0.403  2.43       FAM174A
316        Q06278 Recurrence 0.017700   0.403  2.42          AOX1
317        O14737 Recurrence 0.017800   0.403 -2.42         PDCD5
318        P24928 Recurrence 0.017800   0.403 -2.42        POLR2A
319        Q9NQG6 Recurrence 0.017800   0.403 -2.42         MIEF1
320        P07305 Recurrence 0.017900   0.403 -2.42          H1-0
321        H7BZ55 Recurrence 0.018000   0.403  2.41        CROCC2
322        P05198 Recurrence 0.018000   0.403 -2.42        EIF2S1
323        P20618 Recurrence 0.018000   0.403  2.42         PSMB1
324        Q53F19 Recurrence 0.018100   0.403 -2.41         NCBP3
325        Q86TP1 Recurrence 0.018100   0.403  2.41        PRUNE1
326        Q5MY95 Recurrence 0.018200   0.403  2.41        ENTPD8
327        Q9P2T1 Recurrence 0.018200   0.403  2.41         GMPR2
328        P08729 Recurrence 0.018300   0.403 -2.41          KRT7
329        Q6Q4G3 Recurrence 0.018300   0.403 -2.41          LVRN
330        Q8IUD2 Recurrence 0.018300   0.403 -2.41          ERC1
331        Q9UBU9 Recurrence 0.018300   0.403  2.41          NXF1
332        Q5W0Z9 Recurrence 0.018300   0.403 -2.41       ZDHHC20
333        O95248 Recurrence 0.018400   0.403  2.41          SBF1
334        Q08623 Recurrence 0.018500   0.404 -2.40          PUDP
335        Q16270 Recurrence 0.018600   0.404 -2.40        IGFBP7
336        P29558 Recurrence 0.018900   0.409 -2.40         RBMS1
337        P04921 Recurrence 0.019000   0.409 -2.39          GYPC
338        P23246 Recurrence 0.019000   0.409 -2.39          SFPQ
339        Q8N6S5 Recurrence 0.019100   0.409 -2.39       ARL6IP6
340 P14678;P63162 Recurrence 0.019200   0.409  2.39   SNRPB;SNRPN
341        Q13561 Recurrence 0.019200   0.409 -2.39         DCTN2
342        Q9BYJ9 Recurrence 0.019200   0.409  2.39        YTHDF1
343        Q9UN86 Recurrence 0.019300   0.411 -2.39         G3BP2
344        Q9P2M7 Recurrence 0.019400   0.411 -2.39           CGN
345        P61964 Recurrence 0.019500   0.411  2.38          WDR5
346        Q8WYP5 Recurrence 0.019500   0.411 -2.38        AHCTF1
347 P20231;Q15661 Recurrence 0.019800   0.414  2.38  TPSB2;TPSAB1
348        Q9Y5Z4 Recurrence 0.019800   0.414 -2.38         HEBP2
349        O94819 Recurrence 0.019900   0.414  2.38       KBTBD11
350        O95415 Recurrence 0.019900   0.414 -2.38          BRI3
351        Q5HYI8 Recurrence 0.020000   0.414  2.37         RABL3
352        O00443 Recurrence 0.020100   0.415  2.37       PIK3C2A
353        P07951 Recurrence 0.020100   0.415 -2.37          TPM2
354        Q96TA2 Recurrence 0.020200   0.415  2.37        YME1L1
355        P37837 Recurrence 0.020300   0.415  2.37        TALDO1
356        Q9UJ14 Recurrence 0.020300   0.415  2.37          GGT7
357        A8MXV4 Recurrence 0.020500   0.417  2.36        NUDT19
358        P38646 Recurrence 0.020500   0.417 -2.36         HSPA9
359        Q6PJ69 Recurrence 0.020800   0.417  2.36        TRIM65
360        Q8NDX1 Recurrence 0.020800   0.417  2.36          PSD4
361        P35609 Recurrence 0.020900   0.417 -2.36         ACTN2
362        Q96B97 Recurrence 0.020900   0.417 -2.36       SH3KBP1
363        Q9UBF8 Recurrence 0.020900   0.417  2.36         PI4KB
364        P52272 Recurrence 0.021000   0.417 -2.36        HNRNPM
365        Q9Y496 Recurrence 0.021000   0.417 -2.35         KIF3A
366        P14866 Recurrence 0.021100   0.417 -2.35        HNRNPL
367        P11172 Recurrence 0.021300   0.417  2.35          UMPS
368        Q9HBL0 Recurrence 0.021300   0.417 -2.35          TNS1
369        Q9NR99 Recurrence 0.021300   0.417  2.35         MXRA5
370        O43181 Recurrence 0.021400   0.417 -2.35        NDUFS4
371        P61221 Recurrence 0.021400   0.417  2.35         ABCE1
372        Q7Z6J4 Recurrence 0.021400   0.417  2.35          FGD2
373        Q8IXQ4 Recurrence 0.021500   0.417 -2.35       GPALPP1
374        O00625 Recurrence 0.021600   0.417 -2.34           PIR
375        Q7L7X3 Recurrence 0.021600   0.417 -2.34         TAOK1
376        Q8N4T8 Recurrence 0.021600   0.417  2.34          CBR4
377        O00499 Recurrence 0.021700   0.417 -2.34          BIN1
378        P06753 Recurrence 0.021800   0.417 -2.34          TPM3
379        P51809 Recurrence 0.021800   0.417  2.34         VAMP7
380        Q96JG6 Recurrence 0.021800   0.417  2.34         VPS50
381        Q9H9A6 Recurrence 0.021800   0.417  2.34        LRRC40
382        Q9NQE9 Recurrence 0.021800   0.417 -2.34         HINT3
383        Q96AB3 Recurrence 0.021900   0.417  2.34         ISOC2
384        Q16625 Recurrence 0.022000   0.417 -2.34          OCLN
385        Q9UKD2 Recurrence 0.022000   0.417  2.34         MRTO4
386        Q9UEG4 Recurrence 0.022100   0.417  2.33        ZNF629
387        Q5TAQ9 Recurrence 0.022200   0.417  2.33         DCAF8
388        Q8NB25 Recurrence 0.022200   0.417 -2.33       FAM184A
389        Q9BSB4 Recurrence 0.022200   0.417 -2.33        ATG101
390        O14745 Recurrence 0.022300   0.417 -2.33      SLC9A3R1
391        Q9NYL9 Recurrence 0.022500   0.419 -2.33         TMOD3
392        Q9NR12 Recurrence 0.022600   0.420 -2.33        PDLIM7
393        Q8WXH0 Recurrence 0.022700   0.420 -2.32         SYNE2
394        Q9H1D9 Recurrence 0.022800   0.420  2.32        POLR3F
395        Q7Z5N4 Recurrence 0.022900   0.420 -2.32          SDK1
396        P26583 Recurrence 0.023000   0.420 -2.32         HMGB2
397        P35749 Recurrence 0.023000   0.420 -2.32         MYH11
398        Q15031 Recurrence 0.023000   0.420  2.32         LARS2
399        Q8WWI1 Recurrence 0.023000   0.420 -2.32          LMO7
400        Q92599 Recurrence 0.023000   0.420 -2.32       SEPTIN8
401        Q9Y2W1 Recurrence 0.023100   0.420 -2.32        THRAP3
402        Q9Y3B2 Recurrence 0.023100   0.420  2.32        EXOSC1
403        P01920 Recurrence 0.023300   0.421  2.31      HLA-DQB1
404        Q6EMK4 Recurrence 0.023400   0.423 -2.31          VASN
405        P53680 Recurrence 0.023600   0.423  2.31         AP2S1
406        P25786 Recurrence 0.023700   0.423  2.31         PSMA1
407        Q4V328 Recurrence 0.023700   0.423 -2.31       GRIPAP1
408        Q9NR30 Recurrence 0.023700   0.423 -2.31         DDX21
409        P34932 Recurrence 0.023800   0.424 -2.30         HSPA4
410        Q96S66 Recurrence 0.023800   0.424 -2.30         CLCC1
411        Q969X1 Recurrence 0.024000   0.425  2.30        TMBIM1
412        Q9NX00 Recurrence 0.024100   0.426  2.30       TMEM160
413        Q13630 Recurrence 0.024300   0.426  2.30          GFUS
414        Q9Y3E0 Recurrence 0.024400   0.426  2.30        GOLT1B
415        P51970 Recurrence 0.024500   0.426 -2.29        NDUFA8
416        O14974 Recurrence 0.024600   0.426 -2.29      PPP1R12A
417        O43852 Recurrence 0.024600   0.426 -2.29          CALU
418        P38935 Recurrence 0.024600   0.426  2.29       IGHMBP2
419        P84085 Recurrence 0.024600   0.426  2.29          ARF5
420        O95817 Recurrence 0.024700   0.426 -2.29          BAG3
421        Q7LG56 Recurrence 0.024900   0.426  2.29         RRM2B
422        O75369 Recurrence 0.025000   0.426 -2.28          FLNB
423        P49721 Recurrence 0.025000   0.426  2.28         PSMB2
424        O00291 Recurrence 0.025100   0.426 -2.28          HIP1
425        Q92504 Recurrence 0.025200   0.426  2.28       SLC39A7
426        Q2M3G0 Recurrence 0.025300   0.426 -2.28         ABCB5
427        Q6NUS6 Recurrence 0.025300   0.426 -2.28         TCTN3
428        P25789 Recurrence 0.025400   0.426  2.28         PSMA4
429        Q9UH99 Recurrence 0.025400   0.426 -2.28          SUN2
430        Q7Z7L9 Recurrence 0.025500   0.426  2.28        ZSCAN2
431        Q13867 Recurrence 0.025600   0.426  2.28          BLMH
432        Q96EK6 Recurrence 0.025600   0.426  2.27       GNPNAT1
433        Q9H7X7 Recurrence 0.025600   0.426 -2.27         IFT22
434        Q9Y2H1 Recurrence 0.025600   0.426  2.27        STK38L
435        Q9Y333 Recurrence 0.025600   0.426  2.28          LSM2
436        Q86WJ1 Recurrence 0.025700   0.426 -2.27         CHD1L
437        O14817 Recurrence 0.025800   0.426 -2.27        TSPAN4
438        Q8WUN7 Recurrence 0.025800   0.426  2.27         UBTD2
439        P35241 Recurrence 0.025900   0.426 -2.27           RDX
440        Q07157 Recurrence 0.025900   0.426 -2.27          TJP1
441        Q8IWZ3 Recurrence 0.025900   0.426  2.27        ANKHD1
442        P11387 Recurrence 0.026000   0.426 -2.27          TOP1
443        Q8N118 Recurrence 0.026000   0.426 -2.27        CYP4X1
444        Q8WUD4 Recurrence 0.026100   0.426  2.27        CCDC12
445        Q99943 Recurrence 0.026100   0.426  2.27        AGPAT1
446        Q14119 Recurrence 0.026200   0.426  2.27         VEZF1
447        Q9Y4Z0 Recurrence 0.026200   0.426  2.27          LSM4
448    A0A0B4J1U3 Recurrence 0.026200   0.426 -2.26      IGLV1-36
449        P38159 Recurrence 0.026300   0.426 -2.26          RBMX
450        Q9UMS6 Recurrence 0.026300   0.426 -2.26        SYNPO2
451        Q8NC51 Recurrence 0.026600   0.429 -2.26        SERBP1
452        Q96G23 Recurrence 0.026600   0.429  2.26         CERS2
453        Q9UJY1 Recurrence 0.026600   0.429 -2.26         HSPB8
454        P10155 Recurrence 0.026700   0.429  2.26          RO60
455        O43264 Recurrence 0.026900   0.430 -2.25          ZW10
456        P36551 Recurrence 0.026900   0.430  2.25          CPOX
457        Q04637 Recurrence 0.027000   0.430 -2.25        EIF4G1
458        Q8NB15 Recurrence 0.027600   0.438 -2.24        ZNF511
459        Q9NR00 Recurrence 0.027600   0.438 -2.24          TCIM
460        Q9UFN0 Recurrence 0.027600   0.438  2.24     NIPSNAP3A
461        P11279 Recurrence 0.027900   0.438  2.24         LAMP1
462        P51159 Recurrence 0.027900   0.438  2.24        RAB27A
463        Q8TB72 Recurrence 0.027900   0.438  2.24          PUM2
464        Q9BQ13 Recurrence 0.027900   0.438  2.24        KCTD14
465        Q86V81 Recurrence 0.028000   0.439 -2.24        ALYREF
466        Q05682 Recurrence 0.028200   0.439 -2.24         CALD1
467        Q15628 Recurrence 0.028200   0.439 -2.24         TRADD
468        Q969H8 Recurrence 0.028300   0.439  2.23         MYDGF
469        Q96KR1 Recurrence 0.028300   0.439 -2.23           ZFR
470        P18583 Recurrence 0.028400   0.439 -2.23           SON
471        Q9H0W8 Recurrence 0.028400   0.439  2.23          SMG9
472        O60716 Recurrence 0.028500   0.439 -2.23        CTNND1
473        P02458 Recurrence 0.028600   0.439  2.23        COL2A1
474        P05154 Recurrence 0.028700   0.439  2.23      SERPINA5
475        P28072 Recurrence 0.028700   0.439  2.23         PSMB6
476        Q9Y613 Recurrence 0.028700   0.439 -2.23         FHOD1
477        P08962 Recurrence 0.028800   0.439  2.23          CD63
478        Q9NYR9 Recurrence 0.028800   0.439 -2.23       NKIRAS2
479        P09668 Recurrence 0.028900   0.439  2.23          CTSH
480        Q7Z4V5 Recurrence 0.029000   0.441 -2.22        HDGFL2
481        Q15118 Recurrence 0.029100   0.441  2.22          PDK1
482        Q13627 Recurrence 0.029100   0.441 -2.22        DYRK1A
483        O00170 Recurrence 0.029300   0.441 -2.22           AIP
484        Q13496 Recurrence 0.029300   0.441  2.22          MTM1
485        P09417 Recurrence 0.029500   0.443  2.22          QDPR
486        Q13907 Recurrence 0.029600   0.443  2.22          IDI1
487        O15382 Recurrence 0.029700   0.443  2.21         BCAT2
488        P62995 Recurrence 0.029700   0.443 -2.21         TRA2B
489 P0CG29;P0CG30 Recurrence 0.029800   0.444  2.21  GSTT2;GSTT2B
490        Q9NUQ2 Recurrence 0.029900   0.445 -2.21        AGPAT5
491        P25685 Recurrence 0.030100   0.446 -2.21        DNAJB1
492        Q9Y3E1 Recurrence 0.030100   0.446 -2.21        HDGFL3
493        P08294 Recurrence 0.030300   0.448 -2.21          SOD3
494        Q96S99 Recurrence 0.030400   0.449  2.20       PLEKHF1
495        Q14151 Recurrence 0.030500   0.449 -2.20         SAFB2
496        Q9ULA0 Recurrence 0.030500   0.449  2.20         DNPEP
497        P49754 Recurrence 0.030600   0.449  2.20         VPS41
498        O15120 Recurrence 0.030800   0.449  2.20        AGPAT2
499        Q8IV50 Recurrence 0.030800   0.449  2.20        LYSMD2
500        Q14257 Recurrence 0.030900   0.449 -2.20          RCN2
501        Q6NUK4 Recurrence 0.030900   0.449  2.20         REEP3
502        Q9H444 Recurrence 0.030900   0.449 -2.20        CHMP4B
503        P67936 Recurrence 0.031000   0.449 -2.20          TPM4
504        P53365 Recurrence 0.031100   0.449  2.19        ARFIP2
505        P55287 Recurrence 0.031100   0.449 -2.19         CDH11
506        Q9GZT3 Recurrence 0.031100   0.449 -2.19         SLIRP
507        O95881 Recurrence 0.031300   0.450 -2.19       TXNDC12
508        Q9HC84 Recurrence 0.031400   0.450  2.19         MUC5B
509        Q9NWY4 Recurrence 0.031400   0.450  2.19          HPF1
510        Q9Y3D6 Recurrence 0.031700   0.453  2.19          FIS1
511        P25686 Recurrence 0.032000   0.456 -2.18        DNAJB2
512        Q9C0J8 Recurrence 0.032000   0.456 -2.18         WDR33
513        Q15599 Recurrence 0.032200   0.456 -2.18      SLC9A3R2
514        Q16775 Recurrence 0.032200   0.456  2.18          HAGH
515        Q9P0K7 Recurrence 0.032200   0.456 -2.18         RAI14
516        Q9NXS2 Recurrence 0.032300   0.457  2.18         QPCTL
517        O75746 Recurrence 0.032800   0.461 -2.17      SLC25A12
518        Q9HCD5 Recurrence 0.032800   0.461 -2.17         NCOA5
519        P15954 Recurrence 0.032900   0.461 -2.17         COX7C
520        Q7Z7A4 Recurrence 0.032900   0.461  2.17           PXK
521        Q01813 Recurrence 0.033000   0.462  2.17          PFKP
522        P25445 Recurrence 0.033200   0.462 -2.17           FAS
523        P18206 Recurrence 0.033400   0.462 -2.17           VCL
524        Q12774 Recurrence 0.033400   0.462  2.16       ARHGEF5
525        Q9H0X9 Recurrence 0.033400   0.462 -2.16        OSBPL5
526        Q9Y3C8 Recurrence 0.033400   0.462 -2.16          UFC1
527        Q9NQ29 Recurrence 0.033500   0.462 -2.16         LUC7L
528        O75940 Recurrence 0.033600   0.462 -2.16        SMNDC1
529        Q96CP6 Recurrence 0.033600   0.462  2.16       GRAMD1A
530        Q9H8Y5 Recurrence 0.033600   0.462  2.16        ANKZF1
531        Q9Y2V2 Recurrence 0.033600   0.462 -2.16       CARHSP1
532        P46109 Recurrence 0.033900   0.463 -2.16          CRKL
533        P30533 Recurrence 0.034000   0.463 -2.16        LRPAP1
534        P50579 Recurrence 0.034000   0.463  2.16        METAP2
535        Q9BRQ6 Recurrence 0.034000   0.463 -2.16        CHCHD6
536        O96000 Recurrence 0.034100   0.463 -2.16       NDUFB10
537        P51888 Recurrence 0.034100   0.463 -2.16         PRELP
538        O95833 Recurrence 0.034200   0.464 -2.15         CLIC3
539        Q14232 Recurrence 0.034300   0.464  2.15        EIF2B1
540        Q9H0N5 Recurrence 0.034300   0.464  2.15         PCBD2
541        Q8WV41 Recurrence 0.034500   0.465  2.15         SNX33
542        Q5TBB1 Recurrence 0.034600   0.466  2.15      RNASEH2B
543        P10909 Recurrence 0.034700   0.466 -2.15           CLU
544        Q9P021 Recurrence 0.034800   0.466 -2.15         CRIPT
545        Q9UIG0 Recurrence 0.035000   0.467 -2.14         BAZ1B
546        Q14690 Recurrence 0.035100   0.467  2.14        PDCD11
547        P50895 Recurrence 0.035200   0.467 -2.14          BCAM
548        Q9NRY4 Recurrence 0.035200   0.467  2.14      ARHGAP35
549        Q9Y5Y2 Recurrence 0.035200   0.467  2.14         NUBP2
550        P07737 Recurrence 0.035400   0.467  2.14          PFN1
551        O75947 Recurrence 0.035500   0.467 -2.14        ATP5PD
552        Q13131 Recurrence 0.035500   0.467  2.14        PRKAA1
553        Q15493 Recurrence 0.035500   0.467  2.14           RGN
554        Q9H9Q4 Recurrence 0.035500   0.467  2.14         NHEJ1
555        Q9Y5P6 Recurrence 0.035600   0.467  2.14         GMPPB
556        P55058 Recurrence 0.035700   0.468  2.14          PLTP
557        P30626 Recurrence 0.035800   0.468 -2.14           SRI
558        Q8WYR1 Recurrence 0.035900   0.468 -2.13        PIK3R5
559        Q04837 Recurrence 0.036000   0.468  2.13         SSBP1
560        P33992 Recurrence 0.036100   0.468 -2.13          MCM5
561        Q12983 Recurrence 0.036300   0.468  2.13         BNIP3
562        Q9Y6X8 Recurrence 0.036300   0.468 -2.13          ZHX2
563        Q13435 Recurrence 0.036400   0.468 -2.13         SF3B2
564        P98170 Recurrence 0.036500   0.468  2.13          XIAP
565        Q15075 Recurrence 0.036500   0.468 -2.13          EEA1
566        O75533 Recurrence 0.036600   0.468 -2.13         SF3B1
567        P50402 Recurrence 0.036600   0.468 -2.13           EMD
568        P62330 Recurrence 0.036600   0.468  2.13          ARF6
569        Q9BY89 Recurrence 0.036600   0.468  2.13      KIAA1671
570        Q8N138 Recurrence 0.036700   0.468  2.12        ORMDL3
571        Q12789 Recurrence 0.036800   0.468 -2.12        GTF3C1
572        Q9BX95 Recurrence 0.036800   0.468  2.12         SGPP1
573        Q9UQV4 Recurrence 0.036800   0.468  2.12         LAMP3
574        P16402 Recurrence 0.037000   0.468 -2.12          H1-3
575        P17900 Recurrence 0.037100   0.468  2.12          GM2A
576        P30041 Recurrence 0.037200   0.468  2.12         PRDX6
577        Q9UPA5 Recurrence 0.037200   0.468  2.12           BSN
578        P60903 Recurrence 0.037300   0.468 -2.12       S100A10
579        P30519 Recurrence 0.037400   0.468 -2.12         HMOX2
580        Q92945 Recurrence 0.037400   0.468 -2.12         KHSRP
581        P83436 Recurrence 0.037500   0.468  2.12          COG7
582        Q659C4 Recurrence 0.037500   0.468  2.12        LARP1B
583        Q9H410 Recurrence 0.037500   0.468 -2.12          DSN1
584        Q9UDY2 Recurrence 0.037500   0.468 -2.12          TJP2
585        Q96EL3 Recurrence 0.037600   0.469 -2.11        MRPL53
586        Q9NVS9 Recurrence 0.037700   0.470 -2.11          PNPO
587        P53602 Recurrence 0.037900   0.470  2.11           MVD
588        Q96FJ2 Recurrence 0.037900   0.470  2.11        DYNLL2
589        P60709 Recurrence 0.038000   0.470 -2.11          ACTB
590        Q8TB37 Recurrence 0.038000   0.470 -2.11         NUBPL
591        P46379 Recurrence 0.038400   0.470 -2.11          BAG6
592        Q9UBC2 Recurrence 0.038400   0.470 -2.11       EPS15L1
593        P30049 Recurrence 0.038500   0.470 -2.10       ATP5F1D
594        Q9Y3A5 Recurrence 0.038500   0.470  2.10          SBDS
595        P55210 Recurrence 0.038700   0.470  2.10         CASP7
596        A1Z1Q3 Recurrence 0.038800   0.470  2.10       MACROD2
597        Q9UNZ2 Recurrence 0.038800   0.470 -2.10        NSFL1C
598        Q9H3H3 Recurrence 0.038900   0.470  2.10      C11orf68
599        O43815 Recurrence 0.039000   0.470 -2.10          STRN
600        O75934 Recurrence 0.039000   0.470 -2.10         BCAS2
601        P06733 Recurrence 0.039000   0.470  2.10          ENO1
602        Q96A72 Recurrence 0.039000   0.470  2.10        MAGOHB
603        P24821 Recurrence 0.039100   0.470 -2.10           TNC
604        Q05315 Recurrence 0.039100   0.470  2.10           CLC
605        Q7Z7H5 Recurrence 0.039100   0.470  2.10         TMED4
606        Q9H3G5 Recurrence 0.039100   0.470  2.10          CPVL
607        P33240 Recurrence 0.039400   0.471 -2.09         CSTF2
608        Q7LGA3 Recurrence 0.039600   0.471  2.09        HS2ST1
609        Q9UPU5 Recurrence 0.039600   0.471  2.09         USP24
610        P62745 Recurrence 0.039700   0.471 -2.09          RHOB
611        Q4V9L6 Recurrence 0.039700   0.471 -2.09       TMEM119
612        Q96EM0 Recurrence 0.039800   0.471  2.09       L3HYPDH
613        O15287 Recurrence 0.039800   0.471 -2.09         FANCG
614        P54709 Recurrence 0.040000   0.471 -2.09        ATP1B3
615        O75594 Recurrence 0.040100   0.471  2.09       PGLYRP1
616        Q12906 Recurrence 0.040200   0.471 -2.09          ILF3
617        Q86UL3 Recurrence 0.040200   0.471  2.09         GPAT4
618        Q8NEZ4 Recurrence 0.040200   0.471  2.09         KMT2C
619        Q9UHK6 Recurrence 0.040200   0.471  2.09         AMACR
620        O60237 Recurrence 0.040300   0.471 -2.08      PPP1R12B
621        O95171 Recurrence 0.040300   0.471 -2.08          SCEL
622        P24534 Recurrence 0.040300   0.471 -2.08        EEF1B2
623        Q9BV81 Recurrence 0.040400   0.471 -2.08          EMC6
624        P12277 Recurrence 0.040600   0.471 -2.08           CKB
625        P11274 Recurrence 0.040700   0.471 -2.08           BCR
626        Q9NS15 Recurrence 0.040700   0.471  2.08         LTBP3
627        Q13615 Recurrence 0.040800   0.471 -2.08         MTMR3
628        Q8NC96 Recurrence 0.040900   0.471  2.08        NECAP1
629        Q9NP80 Recurrence 0.040900   0.471  2.08        PNPLA8
630        Q6F5E8 Recurrence 0.041000   0.471 -2.08       CARMIL2
631        Q8NHU6 Recurrence 0.041000   0.471  2.08         TDRD7
632        Q86V48 Recurrence 0.041200   0.471 -2.08         LUZP1
633        P07988 Recurrence 0.041300   0.471  2.07         SFTPB
634        Q27J81 Recurrence 0.041300   0.471 -2.07          INF2
635        Q13232 Recurrence 0.041400   0.471  2.07          NME3
636        O14776 Recurrence 0.041500   0.471 -2.07        TCERG1
637        P24941 Recurrence 0.041500   0.471  2.07          CDK2
638        P52943 Recurrence 0.041500   0.471 -2.07         CRIP2
639        Q6DKK2 Recurrence 0.041500   0.471  2.07         TTC19
640        O75635 Recurrence 0.041600   0.471  2.07      SERPINB7
641        Q93034 Recurrence 0.041600   0.471  2.07          CUL5
642        O00483 Recurrence 0.041700   0.471 -2.07        NDUFA4
643        P02461 Recurrence 0.041800   0.471  2.07        COL3A1
644        P35221 Recurrence 0.041800   0.471 -2.07        CTNNA1
645        Q15811 Recurrence 0.041800   0.471 -2.07         ITSN1
646        P14618 Recurrence 0.041900   0.471  2.07           PKM
647        Q92743 Recurrence 0.041900   0.471 -2.07         HTRA1
648        Q99961 Recurrence 0.041900   0.471 -2.07        SH3GL1
649        Q9P2E9 Recurrence 0.041900   0.471 -2.07         RRBP1
650        Q5HYK7 Recurrence 0.042100   0.471 -2.07        SH3D19
651        Q8N511 Recurrence 0.042100   0.471  2.07       TMEM199
652        Q9H936 Recurrence 0.042100   0.471  2.07      SLC25A22
653        Q9BYX2 Recurrence 0.042200   0.471 -2.06        TBC1D2
654        Q8WX92 Recurrence 0.042300   0.471  2.06         NELFB
655        O00422 Recurrence 0.042500   0.472 -2.06         SAP18
656        P49796 Recurrence 0.042600   0.472  2.06          RGS3
657        O60268 Recurrence 0.042600   0.472 -2.06      KIAA0513
658        O95613 Recurrence 0.042700   0.472  2.06          PCNT
659        Q96DE0 Recurrence 0.042700   0.472  2.06        NUDT16
660        O43556 Recurrence 0.042800   0.472  2.06          SGCE
661        Q13242 Recurrence 0.042800   0.472 -2.06         SRSF9
662        O75420 Recurrence 0.043100   0.473  2.06        GIGYF1
663        O95182 Recurrence 0.043100   0.473 -2.06        NDUFA7
664        Q08752 Recurrence 0.043100   0.473 -2.06          PPID
665        Q92733 Recurrence 0.043100   0.473  2.06          PRCC
666        Q7Z4Q2 Recurrence 0.043300   0.474  2.05        HEATR3
667        Q9NPF5 Recurrence 0.043400   0.474 -2.05         DMAP1
668        P55196 Recurrence 0.043500   0.474 -2.05          AFDN
669        O43157 Recurrence 0.043500   0.474 -2.05        PLXNB1
670        Q9C035 Recurrence 0.043700   0.475 -2.05         TRIM5
671        Q9NZV8 Recurrence 0.044000   0.475 -2.05         KCND2
672        P42126 Recurrence 0.044100   0.475  2.05          ECI1
673        Q06481 Recurrence 0.044100   0.475 -2.05         APLP2
674        Q8NBQ5 Recurrence 0.044100   0.475  2.05      HSD17B11
675        O00481 Recurrence 0.044100   0.475 -2.05        BTN3A1
676        O94979 Recurrence 0.044200   0.475  2.04        SEC31A
677        P11021 Recurrence 0.044200   0.475 -2.04         HSPA5
678        Q9P2Y4 Recurrence 0.044300   0.475  2.04        ZNF219
679        Q7Z2Z1 Recurrence 0.044400   0.475 -2.04         TICRR
680        Q92621 Recurrence 0.044400   0.475  2.04        NUP205
681        Q96IZ0 Recurrence 0.044400   0.475 -2.04          PAWR
682        Q9NP61 Recurrence 0.044400   0.475  2.04       ARFGAP3
683        Q10469 Recurrence 0.044500   0.475  2.04         MGAT2
684        P84077 Recurrence 0.044600   0.475  2.04          ARF1
685        P33151 Recurrence 0.044700   0.475 -2.04          CDH5
686        P53582 Recurrence 0.044800   0.475 -2.04        METAP1
687        O00264 Recurrence 0.044900   0.475 -2.04        PGRMC1
688        O14879 Recurrence 0.044900   0.475 -2.04         IFIT3
689        P42677 Recurrence 0.044900   0.475  2.04         RPS27
690        Q15084 Recurrence 0.045000   0.475 -2.04         PDIA6
691        Q96MU7 Recurrence 0.045100   0.476 -2.04        YTHDC1
692        P35914 Recurrence 0.045300   0.476  2.03         HMGCL
693        P58546 Recurrence 0.045300   0.476 -2.03          MTPN
694        Q9P0L0 Recurrence 0.045400   0.476 -2.03          VAPA
695        O75352 Recurrence 0.045500   0.477  2.03         MPDU1
696        Q9BQS8 Recurrence 0.045500   0.477 -2.03         FYCO1
697        P22626 Recurrence 0.045600   0.477 -2.03     HNRNPA2B1
698        O75607 Recurrence 0.045800   0.478 -2.03          NPM3
699        P56962 Recurrence 0.045900   0.478  2.03         STX17
700        Q01415 Recurrence 0.045900   0.478  2.03         GALK2
701        Q13214 Recurrence 0.045900   0.478 -2.03        SEMA3B
702        Q14684 Recurrence 0.046100   0.479  2.03         RRP1B
703        P08397 Recurrence 0.046200   0.479  2.02          HMBS
704        Q14562 Recurrence 0.046300   0.479 -2.02          DHX8
705        Q53GL0 Recurrence 0.046300   0.479  2.02       PLEKHO1
706        Q14161 Recurrence 0.046500   0.480  2.02          GIT2
707        Q9BYD3 Recurrence 0.046500   0.480  2.02         MRPL4
708        O95428 Recurrence 0.046800   0.482 -2.02         PAPLN
709        O60518 Recurrence 0.047200   0.484  2.02        RANBP6
710        Q9BZL4 Recurrence 0.047200   0.484 -2.02      PPP1R12C
711        Q9UHB6 Recurrence 0.047200   0.484 -2.02         LIMA1
712        P29590 Recurrence 0.047300   0.484 -2.01           PML
713        Q9HBL7 Recurrence 0.047800   0.488  2.01        PLGRKT
714        Q6PJT7 Recurrence 0.047900   0.489 -2.01        ZC3H14
715        P39656 Recurrence 0.048100   0.490  2.01         DDOST
716        Q96A19 Recurrence 0.048300   0.490 -2.01      CCDC102A
717        Q9BQ69 Recurrence 0.048300   0.490  2.01       MACROD1
718        Q15365 Recurrence 0.048400   0.490  2.00         PCBP1
719        Q6UWV6 Recurrence 0.048400   0.490  2.00         ENPP7
720        Q9Y2Q0 Recurrence 0.048400   0.490  2.00        ATP8A1
721        Q8IW45 Recurrence 0.048700   0.492  2.00          NAXD
722        O95807 Recurrence 0.048800   0.492  2.00       TMEM50A
723        Q08378 Recurrence 0.048900   0.492 -2.00        GOLGA3
724        P23469 Recurrence 0.049000   0.492  2.00         PTPRE
725        P83881 Recurrence 0.049000   0.492 -2.00        RPL36A
726        P55001 Recurrence 0.049100   0.492  2.00         MFAP2
727        Q06124 Recurrence 0.049200   0.492 -2.00        PTPN11
728        Q13126 Recurrence 0.049200   0.492  2.00          MTAP
729        P05067 Recurrence 0.049300   0.492 -2.00           APP
730        Q9H0N0 Recurrence 0.049300   0.492  2.00         RAB6C
731        P62310 Recurrence 0.049500   0.493 -1.99          LSM3
732        Q6DN90 Recurrence 0.049500   0.493  1.99        IQSEC1
733        O60271 Recurrence 0.049700   0.494 -1.99         SPAG9
734        O15231 Recurrence 0.049800   0.494 -1.99        ZNF185
735        Q9H1C4 Recurrence 0.049900   0.495  1.99       UNC93B1
736        Q96S44 Recurrence 0.050000   0.495  1.99        TP53RK</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proNormalRes_Klin$sig.feat.heat</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Normal samples (ns:
p &gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01, <span
class="math inline">\(***\)</span>: p &lt;= 0.001)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
# datMat &lt;- proTissueRes_Klin$data
datMat &lt;- doBatchCorrection(proTissue_Klin, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;),
                            unwantedVar = &#39;Stage&#39;)
smpAnnoTab &lt;- proTissueRes_Klin$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proNormalRes_Klin$sig.feat.tab
num_sigFeats &lt;- 6
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = Gene))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Condition = factor(Condition, levels = c(&#39;Normal&#39;, &#39;Tumor&#39;)),
                Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
# Visualize top significant features
ggplot(topSigFeatDat, aes(x=Condition, y=Abundance, fill=Recurrence)) +
  geom_boxplot(alpha = 1, linewidth = 1, outlier.color = &#39;grey50&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             label = &#39;p.signif&#39;, size = 7, vjust = 0.5) +
  labs(x = &#39;Sample condition&#39;, y = &#39;log(Abundance)&#39;) +
  scale_fill_manual(values = c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  geom_vline(xintercept = 1.5, size = 0.3) +
  th +
  theme(strip.text = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 22),
        axis.text.x = element_text(size = 18),
        legend.title = element_text(size = 24), legend.text = element_text(size = 22))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#   theme(strip.text = element_text(size = 20, face = &#39;bold&#39;),
#         axis.title = element_text(size = 22),
#         axis.text.x = element_text(size = 20), axis.text.y = element_text(size = 14),
#         legend.title = element_text(size = 22), legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/topSigFeats_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proNormalRes_Klin$pval.hist +
  theme(axis.title = element_text(size = 22), axis.text = element_text(size = 18))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/25.07.24_LGM/pValHist_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 14)</code></pre>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proNormalRes_Klin$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2    pVal pValAdj Stat   Test
1 PC10 (3.6%) Recurrence 0.00113  0.0206 3.38 T-test
2 PC1 (14.3%) Recurrence 0.00206  0.0206 3.18 T-test</code></pre>
<p>Visualize significant PCs</p>
<pre class="r"><code>p &lt;- proNormalRes_Klin$sig.pc.dist$PC10
p$layers[[2]]$aes_params$size &lt;- 4
p$layers[[3]]$aes_params$size &lt;- 8
p #+ theme(axis.title = element_text(size = 22),</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>          # axis.text = element_text(size = 14),
          # legend.title = element_text(size = 22),
          # legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/sigPC10_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)

p &lt;- proNormalRes_Klin$sig.pc.dist$PC1
p$layers[[2]]$aes_params$size &lt;- 4
p$layers[[3]]$aes_params$size &lt;- 8
p #+ theme(axis.title = element_text(size = 22),</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-13-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>          # axis.text = element_text(size = 14),
          # legend.title = element_text(size = 22),
          # legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/sigPC1_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p>Visualize feature significance through volcano plot</p>
<pre class="r"><code>#### CHANGE HERE ####
soaRes &lt;- proNormalRes_Klin
seMedi &lt;- proTissueMediSVs_Klin[, which(colData(proTissueMediSVs_Klin)$Condition == &#39;Normal&#39;)]
fcCutoff &lt;- 0.5
#####################

# Compute log2(FC) using proDA
# Do SOA to view data structure and p-value histogram
# Gender is significantly associated with PC1 and PC11 that is also associated with Recurrence
# soaMediRes &lt;- doSingleOmicsAnalysis(seMedi, soa_metaVar = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
#                                     num_PCs = 20, use_proDA = T, soa_unwantVar = &#39;Gender&#39;,
#                                     pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                     &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;))
# soaMediRes$pval.hist
# soaMediRes$data.var.source

# Prepare data matrix and sample metadata table for creating model matrices
datMat &lt;- assay(seMedi)
metadatTab &lt;- tibble::as_tibble(colData(seMedi), rownames = &#39;Sample&#39;) %&gt;%
  # Make comparison Recurrence vs Nonrecurrence
  dplyr::mutate(Recurrence = factor(Recurrence, levels = c(&#39;No&#39;, &#39;Yes&#39;)))
design &lt;- model.matrix(~Recurrence+Gender, data = metadatTab)

# fit &lt;- proDA(datMat, design = design)
# diffTerm &lt;- proDA::result_names(fit)[2]
# proDARes &lt;- proDA::test_diff(fit, contrast = diffTerm)
# saveRDS(proDARes, &#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalLog2FC_Gender.rds&#39;)
proDARes &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalLog2FC_Gender.rds&#39;)

# Make volcano plot
# Prepare needed information
pValTab &lt;- soaRes$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, pVal, pValAdj) %&gt;%
  dplyr::rename(Feature = Var1)
log2FCTab &lt;- dplyr::select(proDARes, name, diff) %&gt;%
  dplyr::rename(Feature = name, log2FC = diff)
featAnnoTab &lt;- tibble::as_tibble(rowData(seMedi), rownames = &#39;Feature&#39;)
# Combine all needed information
volcanoTab &lt;- dplyr::left_join(pValTab, log2FCTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  # Pinpoint significant features
  dplyr::mutate(DiffExp = dplyr::case_when(pVal &lt;= 0.05 &amp; log2FC &gt;= fcCutoff ~ &#39;Recurrence&#39;,
                                           pVal &lt;= 0.05 &amp; log2FC &lt;= -fcCutoff ~ &#39;Nonrecurrence&#39;),
                DiffExp = dplyr::case_when(is.na(DiffExp) ~ &#39;Not significant&#39;,
                                           !is.na(DiffExp) ~ DiffExp),
                DiffExp = factor(DiffExp, levels = c(&#39;Recurrence&#39;, &#39;Nonrecurrence&#39;, &#39;Not significant&#39;)),
                # Label = dplyr::case_when(!DiffExp %in% &#39;Not significant&#39; ~ Gene),
                Label = dplyr::case_when(pVal &lt;= 0.01 &amp; log2FC &gt;= fcCutoff ~ Gene,
                                         pVal &lt;= 0.01 &amp; log2FC &lt;= -fcCutoff ~ Gene))

ggplot(volcanoTab, aes(x=log2FC, y=-log10(pVal), col=DiffExp, label=Label)) +
  geom_point(size = 2, alpha = 0.9) +
  geom_hline(yintercept = -log10(0.05), linewidth = 0.2) +
  geom_vline(xintercept = c(-fcCutoff, fcCutoff), linewidth = 0.2) +
  geom_text_repel(size = 4, show.legend = F) +
  scale_color_manual(name = &#39;&#39;, values = c(&#39;red&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;log2(FC)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#   theme(axis.title = element_text(size = 22),
#         axis.text = element_text(size = 18),
#         legend.title = element_text(size = 22),
#         legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/volcano_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 13)</code></pre>
<p>Enrichment analysis<br />
<strong>t-values</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proNormalRes_Klin$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, Stat) %&gt;%
  tibble::column_to_rownames(&#39;Var1&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)

# Do GSEA
# Hallmark gene sets
gseaRes_H &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 20, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaRes_H$plotTopSigGS +
  labs(title = &#39;Hallmark gene sets&#39;) +
  theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 18),
        axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
        legend.title = element_text(size = 20),
        legend.text = element_text(size = 16),
        panel.grid.minor = element_blank(),
        plot.title = element_text(size = 14, hjust = 0.5, vjust = 2, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/25.07.24_LGM/gsea_tStat_overrepGS_combined_proNormal_Klin_H.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)

# GO
gseaRes_C5 &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;C5&#39;, numSigGS = 10)
# Curated gene sets
gseaRes_C2 &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;C2&#39;, numSigGS = 10)</code></pre>
<p>Tree plot clustering associated enriched terms to identify functional
modules</p>
<pre class="r"><code># Compute pairwise similarity of enriched terms (Jaccard&#39;s index)
gseaRes &lt;- enrichplot::pairwise_termsim(gseaRes_H$gseaRes)
# Make tree plot
enrichplot::treeplot(gseaRes, nCluster = 3, nWords = 0, hclust_method = &#39;ward.D2&#39;)</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Make enrichment map
# clusterProfiler::emapplot(gseaRes, showCategory = 20, node_label = &#39;category&#39;, repel = T,
#                           cex.params = list(category_node = 1.2, category_label = 0.8),
#                           highlight.params = list())</code></pre>
<p>Heatmap-like functional classification</p>
<pre class="r"><code>#### CHANGE HERE ####
interestGS &lt;- c(&#39;HALLMARK_GLYCOLYSIS&#39;, &#39;HALLMARK_PEROXISOME&#39;, &#39;HALLMARK_FATTY_ACID_METABOLISM&#39;,
                &#39;HALLMARK_MTORC1_SIGNALING&#39;)
soaRes &lt;- proNormalRes_Klin
rankedList &lt;- rankedGeneList_Klin
seMedi &lt;- proTissueMediSVs_Klin[, which(colData(proTissueMediSVs_Klin)$Condition == &#39;Normal&#39;)]
#####################

# Prepare core enrichment list retrieved from EA result
coreEnrichmentTab &lt;- gseaRes_H$gseaRes@result %&gt;%
  dplyr::select(Description, core_enrichment) %&gt;%
  dplyr::filter(Description %in% interestGS)
coreEnrichmentList &lt;- lapply(seq_len(nrow(coreEnrichmentTab)), function(i) {
  stringr::str_split_1(coreEnrichmentTab$core_enrichment[i], &#39;/&#39;)
})
names(coreEnrichmentList) &lt;- coreEnrichmentTab$Description
# Prepare feature list of interest
coreFeats &lt;- unique(unlist(coreEnrichmentList))
# Retrieve significant feature list from SOA result
sigFeats &lt;- stringr::str_remove(soaRes$sig.feat.tab$Gene, &#39;;.+$&#39;)
# Check if there is any duplicated gene
# any(duplicated(sigFeats))

# Prepare feature list of interest ordered by feature significance
# interestFeats &lt;- intersect(sigFeats, coreFeats)
# # Remove overrepresented gene sets that do not include any feature of interest
# keptGS &lt;- sapply(seq_len(length(coreEnrichmentList)), function(i) {
#   any(coreEnrichmentList[[i]] %in% interestFeats)
# })
# coreEnrichmentList &lt;- coreEnrichmentList[keptGS]
# 
# # Plot heatmap based on t-statistics
# # Create binary matrix indicating presence of features in overrepresented gene sets
# binaryMat &lt;- matrix(nrow = length(coreEnrichmentList), ncol = length(interestFeats),
#                     dimnames = list(names(coreEnrichmentList), interestFeats))
# for (i in seq_len(nrow(binaryMat))) {
#   for (j in seq_len(ncol(binaryMat))) {
#     if (colnames(binaryMat)[j] %in% coreEnrichmentList[[rownames(binaryMat)[i]]]) {
#       binaryMat[i, j] &lt;- T
#     } else {
#       binaryMat[i, j] &lt;- F
#     }
#   }
# }
# # Create heatmap matrix based on binary matrix and statistics results
# gsOverrepHeatmap &lt;- tibble::as_tibble(binaryMat, rownames = &#39;Pathway&#39;) %&gt;%
#   tidyr::pivot_longer(cols = -&#39;Pathway&#39;, names_to = &#39;Gene&#39;, values_to = &#39;Presence&#39;) %&gt;%
#   dplyr::mutate(tStat = dplyr::case_when(Presence ~ rankedList[Gene],
#                                          !Presence ~ 0),
#                 Gene = factor(Gene, levels = interestFeats),
#                 Pathway = factor(Pathway, levels = rev(interestGS)))
# ggplot(gsOverrepHeatmap, aes(x=Gene, y=Pathway, fill=tStat)) +
#   geom_tile() +
#   scale_fill_gradient2(high = &#39;darkred&#39;, low = &#39;darkblue&#39;, mid = &#39;white&#39;, midpoint = 0) +
#   scale_x_discrete(expand = c(0, 0)) +
#   scale_y_discrete(expand = c(0, 0)) +
#   labs(title = &#39;Hallmark: t-statistics&#39;) +
#   theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))


# Prepare log2(FC) table
proDARes &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalLog2FC_Gender.rds&#39;)
featAnnoTab &lt;- tibble::as_tibble(rowData(seMedi), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = stringr::str_remove(Gene, &#39;;.+$&#39;)) %&gt;%
  dplyr::filter(Gene %in% coreFeats)
log2FCTab &lt;- dplyr::select(proDARes, name, diff) %&gt;%
  dplyr::rename(Feature = name, log2FC = diff) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::filter(!is.na(Gene)) %&gt;%
  dplyr::select(-Feature) %&gt;%
  tibble::column_to_rownames(&#39;Gene&#39;)
coreFeatsOrderByT &lt;- soaRes$SOA.res$featAssoRes %&gt;%
  dplyr::rename(Feature = Var1) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::filter(!is.na(Gene)) %&gt;%
  dplyr::mutate(Gene = dplyr::case_when(Gene %in% sigFeats ~ paste0(&#39;*&#39;, Gene),
                                        !Gene %in% sigFeats ~ Gene)) %&gt;%
  dplyr::arrange(desc(Stat)) %&gt;%
  dplyr::pull(Gene)

# Plot heatmap based on log2(FC)
# Create binary matrix indicating presence of features in overrepresented gene sets
binaryMat &lt;- matrix(nrow = length(coreEnrichmentList), ncol = length(coreFeats),
                    dimnames = list(names(coreEnrichmentList), coreFeats))
for (i in seq_len(nrow(binaryMat))) {
  for (j in seq_len(ncol(binaryMat))) {
    if (colnames(binaryMat)[j] %in% coreEnrichmentList[[rownames(binaryMat)[i]]]) {
      binaryMat[i, j] &lt;- T
    } else {
      binaryMat[i, j] &lt;- F
    }
  }
}
# Create heatmap matrix based on binary matrix and statistics results
gsOverrepHeatmap &lt;- tibble::as_tibble(binaryMat, rownames = &#39;Pathway&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Pathway&#39;, names_to = &#39;Gene&#39;, values_to = &#39;Presence&#39;) %&gt;%
  dplyr::mutate(log2FC = dplyr::case_when(Presence ~ log2FCTab[Gene,],
                                          !Presence ~ 0),
                Gene = dplyr::case_when(Gene %in% sigFeats ~ paste0(&#39;*&#39;, Gene),
                                        !Gene %in% sigFeats ~ Gene),
                Gene = factor(Gene, levels = rev(coreFeatsOrderByT)),
                Pathway = stringr::str_remove(Pathway, &#39;^HALLMARK_&#39;),
                Pathway = factor(Pathway, levels = rev(str_remove(interestGS, &#39;^HALLMARK_&#39;))))
ggplot(gsOverrepHeatmap, aes(x=Pathway, y=Gene, fill=log2FC)) +
  geom_tile() +
  scale_fill_gradient2(high = &#39;darkred&#39;, low = &#39;darkblue&#39;, mid = &#39;white&#39;, midpoint = 0,
                       name = &#39;log2(FC)&#39;) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) +
  labs(title = &#39;Hallmark: log2(FC)&#39;) +
  theme(axis.text.y = element_text(size = 10))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggplot(gsOverrepHeatmap, aes(x=Gene, y=Pathway, fill=log2FC)) +
#   geom_tile() +
#   scale_fill_gradient2(high = &#39;darkred&#39;, low = &#39;darkblue&#39;, mid = &#39;white&#39;, midpoint = 0,
#                        name = &#39;log2(FC)&#39;) +
#   scale_x_discrete(expand = c(0, 0)) +
#   scale_y_discrete(expand = c(0, 0)) +
#   theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
#         axis.text.x = element_text(size = 14, angle = 90, vjust = 0.5, hjust = 1, face = &#39;bold&#39;),
#         axis.text.y = element_text(size = 12, angle = 90, vjust = 1, hjust = 0.5, face = &#39;bold&#39;),
#         axis.ticks.y = element_blank(),
#         legend.title = element_text(size = 16),
#         legend.text = element_text(size = 14))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/gsea_log2FC_heatmap_combined_proNormal_Klin_H.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 20)</code></pre>
<p>UpSet plot showing gene overlaps among enriched terms and their fold
changes</p>
<pre class="r"><code>enrichplot::upsetplot(gseaRes_H$gseaRes, n = 20)</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Intersection between Hallmark founder gene sets and enriched GO
terms, etc. for narrowing down interesting biological concepts</p>
<pre class="r"><code># Prepare founder gene sets of Hallmark gene sets
fgsFAM &lt;- read.gmt(paste0(&#39;./data/Discovery/MSigDB/proNormal_Klin/HALLMARK_&#39;,
                          &#39;FATTY_ACID_METABOLISM_FOUNDERS.v2023.2.Hs.gmt&#39;)) %&gt;%
  dplyr::pull(term) %&gt;%
  unique()
fgsPERO &lt;- read.gmt(paste0(&#39;./data/Discovery/MSigDB/proNormal_Klin/HALLMARK_&#39;,
                           &#39;PEROXISOME_FOUNDERS.v2023.2.Hs.gmt&#39;)) %&gt;%
  dplyr::pull(term) %&gt;%
  unique()
fgsGLY &lt;- read.gmt(paste0(&#39;./data/Discovery/MSigDB/proNormal_Klin/HALLMARK_&#39;,
                          &#39;GLYCOLYSIS_FOUNDERS.v2023.2.Hs.gmt&#39;)) %&gt;%
  dplyr::pull(term) %&gt;%
  unique()
# Prepare overrepresented gene sets
gsOverrep_C5 &lt;- gseaRes_C5$gseaRes@result$ID
gsOverrep_C2 &lt;- gseaRes_C2$gseaRes@result$ID

# Display overrepresented gene sets that are founders of Hallmark gene sets
cat(&#39;FA Metabolism:\n&#39;, paste0(intersect(fgsFAM, gsOverrep_C5), collapse = &#39;\n&#39;), sep = &#39;&#39;)</code></pre>
<pre><code>FA Metabolism:
GOMF_CARBON_OXYGEN_LYASE_ACTIVITY
GOMF_HYDRO_LYASE_ACTIVITY
GOMF_LYASE_ACTIVITY
GOMF_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_CH_OH_GROUP_OF_DONORS</code></pre>
<pre class="r"><code>cat(&#39;\n&#39;)</code></pre>
<pre class="r"><code>cat(&#39;Peroxisome:\n&#39;, paste0(intersect(fgsPERO, gsOverrep_C2), collapse = &#39;\n&#39;), sep = &#39;&#39;)</code></pre>
<pre><code>Peroxisome:
REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT</code></pre>
<pre class="r"><code>cat(&#39;\n&#39;)</code></pre>
<pre class="r"><code>cat(&#39;Glycolysis:\n&#39;, paste0(intersect(fgsGLY, gsOverrep_C5), collapse = &#39;\n&#39;), &#39;\n&#39;,
    paste0(intersect(fgsGLY, gsOverrep_C2), collapse = &#39;\n&#39;), sep = &#39;&#39;)</code></pre>
<pre><code>Glycolysis:
GOBP_ALCOHOL_METABOLIC_PROCESS
GOBP_CARBOHYDRATE_BIOSYNTHETIC_PROCESS
GOBP_CARBOHYDRATE_CATABOLIC_PROCESS
GOBP_CARBOHYDRATE_METABOLIC_PROCESS
GOBP_CELLULAR_CARBOHYDRATE_METABOLIC_PROCESS
GOBP_GLUCOSE_METABOLIC_PROCESS
GOBP_SULFUR_COMPOUND_BIOSYNTHETIC_PROCESS
GOBP_SULFUR_COMPOUND_METABOLIC_PROCESS
GROSS_HIF1A_TARGETS_DN
KEGG_AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM
KEGG_GLYCOLYSIS_GLUCONEOGENESIS
KEGG_PENTOSE_PHOSPHATE_PATHWAY
KEGG_PYRUVATE_METABOLISM
MOOTHA_GLUCONEOGENESIS
PID_HIF1_TFPATHWAY
REACTOME_METABOLISM_OF_CARBOHYDRATES
SEMENZA_HIF1_TARGETS</code></pre>
<p><strong>PC loadings</strong><br />
<strong>PC10</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proNormalRes_Klin$SOA.res$pcaRes@loadings %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, PC10) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)

# Do GSEA
# Hallmark gene sets
gseaResH &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 10, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaResH$plotTopSigGS +
  labs(title = &#39;Hallmark gene sets&#39;)</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p></br> </br> <font size='5'> <strong>Check result consistency: New vs
Old</strong> </font><br />
<strong>log(FC) between Recurrence and Non-Recurrence patient
groups</strong></p>
<pre class="r"><code># Compute log(FC) and t-statistics
# Load preprocessed data
proTissue_Old &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
proTissue_New &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/proTissueVsn.rds&#39;)
# Prepare protein-gene table for providing gene names
featInfo &lt;- tibble::as_tibble(rowData(proTissue_New), rownames = &#39;Proteins&#39;) %&gt;%
  dplyr::rename(Genes = Gene) %&gt;%
  dplyr::mutate(Proteins = stringr::str_remove(Proteins, &#39;;.*&#39;),
                Genes = stringr::str_remove(Genes, &#39;;.*&#39;))
# Prepare summarized tables of comparison for visualization
# featAssoTabs &lt;- summarizeFeatAssoRes(proTissue_New, proTissue_Old, &#39;New&#39;, &#39;Old&#39;,
#                                      meta_var = &#39;Recurrence&#39;, subset = c(&#39;Condition&#39;, &#39;Normal&#39;),
#                                      method = &#39;proDA&#39;, alpha = 0.05, prot_gene_tbl = featInfo,
#                                      unwantVar1 = &#39;Stage&#39;, unwantVar2 = &#39;Stage&#39;)
# saveRDS(featAssoTabs, &#39;./data/Discovery/AG_Klingmueller/soaRes/New_vs_Old_recurFeatAsso_Stage.rds&#39;)
featAssoTabs &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/New_vs_Old_recurFeatAsso_Stage.rds&#39;)

# Plot log(FC)
ggplot(featAssoTabs$tbl_logFC, aes(x=logFC_dat1, y=logFC_dat2, col=Direction)) +
  geom_point(size = 3) +
  ggpubr::stat_cor(data = featAssoTabs$tbl_logFC,
                   aes(x=logFC_dat1, y=logFC_dat2, col=NULL,
                       label=paste(..rr.label.., ..p.label.., sep = &quot;~`,`~&quot;)),
                   size = 7, show.legend = F) +
  geom_abline(intercept = 0, slope = 1, col = &#39;black&#39;, linewidth = 0.8) +
  scale_color_manual(values = c(&#39;red&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;log(FC) of New data&#39;, y = &#39;log(FC) of Old data&#39;,
       title = &#39;log(FC): Recurrence vs Non-Recurrence&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 22),
  #       axis.text = element_text(size = 14),
  #       legend.title = element_text(size = 22),
  #       legend.text = element_text(size = 20))</code></pre>
<p><strong>p-values (Recurrence vs Non-Recurrence, alpha =
0.05)</strong></p>
<pre class="r"><code># Plot p-values
# Compute result consistency
percent &lt;- round(sum(featAssoTabs$tbl_pVal$Significance == &#39;Sig. in Both&#39;) /
                   sum(featAssoTabs$tbl_pVal$Significance != &#39;Not Sig.&#39;), 3) * 100

ggplot(featAssoTabs$tbl_pVal, aes(x=-log10(pVal_dat1), y=-log10(pVal_dat2), col=Significance)) +
  geom_point(size = 2) +
  geom_abline(intercept = 0, slope = 1, col = &#39;black&#39;, linewidth = 0.8) +
  scale_color_manual(values = c(&#39;red&#39;, &#39;darkgreen&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;-log10(pVal) of New data&#39;, y = &#39;-log10(pVal) of Old data&#39;,
       title = &#39;p-val: Recurrence vs Non-Recurrence&#39;) +
  annotate(&#39;text&#39;, x = 1, y = 5, size = 4,
           label = paste0(percent, &#39;% of significant results consistent&#39;)) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="tumor-samples" class="section level2">
<h2>Tumor samples</h2>
<pre class="r"><code># Subset certain samples
proTumor_Klin &lt;- proTissueSVs_Klin[, which(colData(proTissueSVs_Klin)$Condition == &#39;Tumor&#39;)]
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proTumor_Klin), rownames = &#39;Protein&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
# Perform single-omics analysis
# proTumorRes_Klin &lt;- doSingleOmicsAnalysis(proTumor_Klin, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                           pca_method = &#39;ppca&#39;, num_PCs = 20, num_PCfeats = 30,
#                                           use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                           feat_conv = T, plot_title = NULL, show_rownames = F,
#                                           pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                           &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;, &#39;TumorPurity&#39;),
#                                           soa_unwantVar = grep(&#39;^SV\\d+$&#39;, colnames(colData(proTumor_Klin)),
#                                                                value = T),
#                                           level_metaVar = c(&#39;Yes&#39;, &#39;No&#39;))
# saveRDS(proTumorRes_Klin, &#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs.rds&#39;)
proTumorRes_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proTumorRes_Klin$data.var.source</code></pre>
<pre><code>          Var1        Var2     pVal  pValAdj   Stat        Test
1  PC2 (11.9%) TumorPurity 4.74e-06 0.000664  0.489 Correlation
2   PC6 (5.1%) TumorPurity 1.16e-05 0.000810  0.471 Correlation
3   PC7 (4.9%)  Recurrence 6.16e-03 0.288000  2.820      T-test
4    PC12 (3%)  Recurrence 2.09e-02 0.458000 -2.360      T-test
5  PC13 (2.7%)     Smoking 2.17e-02 0.458000  4.030       ANOVA
6   PC8 (3.9%)         Age 2.44e-02 0.458000  0.253 Correlation
7   PC8 (3.9%) TumorPurity 2.53e-02 0.458000 -0.252 Correlation
8   PC7 (4.9%)         Age 2.74e-02 0.458000 -0.248 Correlation
9   PC7 (4.9%) TumorPurity 2.96e-02 0.458000  0.245 Correlation
10 PC17 (2.2%)         Age 3.55e-02 0.458000 -0.237 Correlation
11  PC6 (5.1%)     Smoking 3.78e-02 0.458000  3.420       ANOVA
12 PC17 (2.2%)       Stage 4.09e-02 0.458000  3.330       ANOVA
13  PC5 (6.3%) TumorPurity 4.27e-02 0.458000  0.229 Correlation</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>dplyr::select(proTumorRes_Klin$sig.feat.tab, -Test)</code></pre>
<pre><code>          Var1       Var2     pVal pValAdj  Stat       Gene
1       Q8WXG6 Recurrence 0.000159   0.580  4.04       MADD
2       P51808 Recurrence 0.000166   0.580 -4.03     DYNLT3
3       Q9UDX4 Recurrence 0.000348   0.580  3.80    SEC14L3
4       P07737 Recurrence 0.000426   0.580  3.74       PFN1
5       O75923 Recurrence 0.000614   0.580 -3.62       DYSF
6       Q5T6X4 Recurrence 0.000660   0.580 -3.60    FAM162B
7       P10515 Recurrence 0.000694   0.580  3.58       DLAT
8       Q14CN2 Recurrence 0.000706   0.580  3.58      CLCA4
9       P51531 Recurrence 0.000716   0.580 -3.57    SMARCA2
10      Q9BYD1 Recurrence 0.000876   0.593  3.51     MRPL13
11      P07711 Recurrence 0.000964   0.593 -3.48       CTSL
12      Q8WVM7 Recurrence 0.000977   0.593 -3.47      STAG1
13      O43734 Recurrence 0.001430   0.595 -3.35   TRAF3IP2
14      O75475 Recurrence 0.001620   0.595 -3.31      PSIP1
15      Q8IYB5 Recurrence 0.001750   0.595  3.28      SMAP1
16      P41250 Recurrence 0.001880   0.595  3.26      GARS1
17      Q8NEZ4 Recurrence 0.001900   0.595  3.25      KMT2C
18      Q15386 Recurrence 0.002100   0.595  3.22      UBE3C
19      Q92526 Recurrence 0.002120   0.595  3.22      CCT6B
20      O94804 Recurrence 0.002260   0.595 -3.20      STK10
21      P22105 Recurrence 0.002260   0.595 -3.20       TNXB
22      Q9UPN3 Recurrence 0.002270   0.595 -3.19      MACF1
23      Q96AB3 Recurrence 0.002450   0.595  3.17      ISOC2
24      P35442 Recurrence 0.002470   0.595  3.17      THBS2
25      P49589 Recurrence 0.002480   0.595  3.16      CARS1
26      Q4G0N4 Recurrence 0.002490   0.595  3.16      NADK2
27      P61019 Recurrence 0.002530   0.595  3.16      RAB2A
28      P33151 Recurrence 0.002590   0.595 -3.15       CDH5
29      Q9BVG9 Recurrence 0.002810   0.595  3.12     PTDSS2
30      Q15170 Recurrence 0.002820   0.595  3.12     TCEAL1
31  A0A0G2JS06 Recurrence 0.002830   0.595 -3.12   IGLV5-39
32      P49959 Recurrence 0.002900   0.595  3.11      MRE11
33      Q9BYT8 Recurrence 0.003040   0.595  3.09        NLN
34      Q15172 Recurrence 0.003170   0.595 -3.08    PPP2R5A
35      Q14520 Recurrence 0.003190   0.595 -3.08      HABP2
36      Q8IWV8 Recurrence 0.003250   0.595  3.07       UBR2
37      O95081 Recurrence 0.003290   0.595 -3.07      AGFG2
38      P23368 Recurrence 0.003290   0.595  3.07        ME2
39      Q13822 Recurrence 0.003370   0.595  3.06      ENPP2
40      Q32P28 Recurrence 0.003580   0.595  3.04       P3H1
41      Q92604 Recurrence 0.003620   0.595  3.03     LPGAT1
42      Q58FG1 Recurrence 0.003680   0.595  3.03  HSP90AA4P
43      O75112 Recurrence 0.003710   0.595 -3.02       LDB3
44      Q9NPG8 Recurrence 0.003790   0.595 -3.02     ZDHHC4
45      O75531 Recurrence 0.003830   0.595  3.01      BANF1
46      O00442 Recurrence 0.004100   0.595  2.99       RTCA
47      P14854 Recurrence 0.004100   0.595 -2.99     COX6B1
48      Q8TE77 Recurrence 0.004150   0.595  2.98       SSH3
49      Q9C086 Recurrence 0.004180   0.595  2.98     INO80B
50      Q8IZL8 Recurrence 0.004230   0.595  2.98      PELP1
51      Q9BVM4 Recurrence 0.004240   0.595  2.98      GGACT
52      P62877 Recurrence 0.004350   0.595  2.97       RBX1
53      Q7Z7L9 Recurrence 0.004360   0.595  2.97     ZSCAN2
54      O00483 Recurrence 0.004410   0.595 -2.96     NDUFA4
55      Q8N699 Recurrence 0.004570   0.606 -2.95      MYCT1
56      P02461 Recurrence 0.004990   0.617  2.92     COL3A1
57      P08236 Recurrence 0.005020   0.617 -2.92       GUSB
58      O76024 Recurrence 0.005050   0.617 -2.92       WFS1
59      P57087 Recurrence 0.005130   0.617 -2.91       JAM2
60      Q08426 Recurrence 0.005130   0.617 -2.91     EHHADH
61      P13611 Recurrence 0.005220   0.617  2.90       VCAN
62      P50238 Recurrence 0.005250   0.617 -2.90      CRIP1
63      P56385 Recurrence 0.005440   0.621  2.89     ATP5ME
64      Q8N3V7 Recurrence 0.005570   0.621 -2.88      SYNPO
65      P61916 Recurrence 0.005600   0.621 -2.88       NPC2
66      P26038 Recurrence 0.005700   0.621 -2.87        MSN
67      P41226 Recurrence 0.005850   0.621 -2.86       UBA7
68      O75718 Recurrence 0.005880   0.621  2.86      CRTAP
69      P01903 Recurrence 0.005960   0.621 -2.86    HLA-DRA
70      O14498 Recurrence 0.006000   0.621  2.85       ISLR
71      P20742 Recurrence 0.006080   0.621  2.85        PZP
72      Q9UKZ4 Recurrence 0.006240   0.621  2.84      TENM1
73      Q9H082 Recurrence 0.006290   0.621 -2.84     RAB33B
74      Q9H267 Recurrence 0.006300   0.621  2.83     VPS33B
75      Q8NB90 Recurrence 0.006460   0.628 -2.83     SPATA5
76      Q8TCE6 Recurrence 0.007040   0.636 -2.79    DENND10
77      Q9Y2J2 Recurrence 0.007160   0.636 -2.79    EPB41L3
78      O75691 Recurrence 0.007170   0.636  2.79      UTP20
79      O95208 Recurrence 0.007190   0.636 -2.79       EPN2
80      Q9H3H3 Recurrence 0.007200   0.636  2.79   C11orf68
81      Q9H488 Recurrence 0.007280   0.636  2.78     POFUT1
82      Q86YP4 Recurrence 0.007310   0.636 -2.78    GATAD2A
83      Q9H0N0 Recurrence 0.007340   0.636  2.78      RAB6C
84      Q8WUA7 Recurrence 0.007350   0.636  2.78   TBC1D22A
85      Q9Y680 Recurrence 0.007450   0.636  2.77      FKBP7
86      Q9P0M6 Recurrence 0.007580   0.636 -2.77  MACROH2A2
87      Q13509 Recurrence 0.007610   0.636  2.77      TUBB3
88      Q6IA69 Recurrence 0.007700   0.636  2.76    NADSYN1
89      Q9Y296 Recurrence 0.007780   0.636  2.76    TRAPPC4
90      Q9NR99 Recurrence 0.007850   0.636  2.75      MXRA5
91      Q15063 Recurrence 0.008050   0.645  2.74      POSTN
92      P52565 Recurrence 0.008170   0.647  2.74    ARHGDIA
93      O60941 Recurrence 0.008460   0.660 -2.73       DTNB
94      Q14203 Recurrence 0.008640   0.660 -2.72      DCTN1
95      P28906 Recurrence 0.008740   0.660 -2.71       CD34
96      P36955 Recurrence 0.008800   0.660  2.71   SERPINF1
97      P49585 Recurrence 0.008860   0.660 -2.71     PCYT1A
98      O75390 Recurrence 0.008870   0.660  2.71         CS
99      Q8IZ73 Recurrence 0.008960   0.660 -2.70     RPUSD2
100     Q96JG6 Recurrence 0.009260   0.668  2.69      VPS50
101     Q8TEB9 Recurrence 0.009360   0.668 -2.69     RHBDD1
102     P54619 Recurrence 0.009600   0.668  2.68     PRKAG1
103     Q96I59 Recurrence 0.009880   0.668  2.67      NARS2
104     P27701 Recurrence 0.010100   0.668 -2.66       CD82
105     P68402 Recurrence 0.010100   0.668  2.66   PAFAH1B2
106     Q92608 Recurrence 0.010100   0.668 -2.66      DOCK2
107     Q9NY33 Recurrence 0.010100   0.668  2.66       DPP3
108     P55212 Recurrence 0.010200   0.668  2.65      CASP6
109     Q8N2G8 Recurrence 0.010200   0.668 -2.66       GHDC
110     P02452 Recurrence 0.010300   0.668  2.65     COL1A1
111     P63104 Recurrence 0.010300   0.668  2.65      YWHAZ
112     P15090 Recurrence 0.010600   0.668 -2.64      FABP4
113     Q9UBI9 Recurrence 0.010800   0.668  2.63       HECA
114     Q9Y625 Recurrence 0.010800   0.668  2.63       GPC6
115     Q8IV50 Recurrence 0.010900   0.668  2.63     LYSMD2
116     Q9UK76 Recurrence 0.010900   0.668  2.63       JPT1
117     P62314 Recurrence 0.011100   0.668  2.62     SNRPD1
118     Q96RT7 Recurrence 0.011200   0.668 -2.62    TUBGCP6
119     Q6UVK1 Recurrence 0.011300   0.668 -2.62      CSPG4
120     O75683 Recurrence 0.011400   0.668 -2.61      SURF6
121     Q6NZI2 Recurrence 0.011400   0.668 -2.61     CAVIN1
122     P23458 Recurrence 0.011500   0.668 -2.61       JAK1
123     Q01082 Recurrence 0.011500   0.668 -2.61     SPTBN1
124     O15530 Recurrence 0.011700   0.668  2.60      PDPK1
125     P08123 Recurrence 0.011700   0.668  2.60     COL1A2
126     Q07960 Recurrence 0.011700   0.668  2.60    ARHGAP1
127     P62847 Recurrence 0.011800   0.668  2.60      RPS24
128     Q96FN9 Recurrence 0.011800   0.668  2.60       DTD2
129     O95630 Recurrence 0.011900   0.668 -2.60     STAMBP
130     Q8IYL3 Recurrence 0.011900   0.668 -2.60   C1orf174
131     O15247 Recurrence 0.012300   0.678 -2.58      CLIC2
132     Q9NT62 Recurrence 0.012300   0.678 -2.58       ATG3
133     Q53HV7 Recurrence 0.012400   0.678  2.58      SMUG1
134     P53420 Recurrence 0.012600   0.678  2.58     COL4A4
135     Q96PZ0 Recurrence 0.012600   0.678  2.58       PUS7
136     Q9BXA6 Recurrence 0.012800   0.678  2.57      TSSK6
137     Q9NXW2 Recurrence 0.012800   0.678 -2.57    DNAJB12
138     Q9UGM5 Recurrence 0.012800   0.678  2.57      FETUB
139     Q8TBM8 Recurrence 0.013000   0.678  2.56    DNAJB14
140     Q9NZQ3 Recurrence 0.013000   0.678 -2.56    NCKIPSD
141     Q9H9C1 Recurrence 0.013100   0.679  2.56    VIPAS39
142     P61077 Recurrence 0.013400   0.682  2.55     UBE2D3
143     P21397 Recurrence 0.013600   0.682 -2.55       MAOA
144     P53677 Recurrence 0.013600   0.682  2.54      AP3M2
145     Q8NF91 Recurrence 0.013600   0.682 -2.55      SYNE1
146     Q8NDH3 Recurrence 0.013700   0.682  2.54     NPEPL1
147     Q03167 Recurrence 0.013800   0.687 -2.54     TGFBR3
148     Q9UIG8 Recurrence 0.014300   0.701 -2.53    SLCO3A1
149     O15116 Recurrence 0.014600   0.701  2.52       LSM1
150     P26440 Recurrence 0.014600   0.701 -2.52        IVD
151     Q96FX7 Recurrence 0.014700   0.701  2.51    TRMT61A
152     P04233 Recurrence 0.014900   0.701 -2.51       CD74
153     O00193 Recurrence 0.015000   0.701  2.51       SMAP
154     P63096 Recurrence 0.015000   0.701 -2.51      GNAI1
155     Q13136 Recurrence 0.015000   0.701  2.51     PPFIA1
156     Q7Z7H5 Recurrence 0.015000   0.701  2.51      TMED4
157     Q5T1M5 Recurrence 0.015400   0.704 -2.50     FKBP15
158     Q8IUZ5 Recurrence 0.015400   0.704 -2.50     PHYKPL
159     P20908 Recurrence 0.015500   0.704  2.49     COL5A1
160     P51784 Recurrence 0.015600   0.704  2.49      USP11
161     Q8NBJ9 Recurrence 0.015600   0.704  2.49      SIDT2
162     Q9Y3C4 Recurrence 0.015700   0.704  2.49      TPRKB
163     P43235 Recurrence 0.015700   0.704  2.49       CTSK
164     Q14119 Recurrence 0.015900   0.707 -2.48      VEZF1
165     Q8NAP3 Recurrence 0.016200   0.717 -2.48     ZBTB38
166     O60238 Recurrence 0.016500   0.722  2.47     BNIP3L
167     O15021 Recurrence 0.016600   0.722 -2.47      MAST4
168     O60879 Recurrence 0.016700   0.722 -2.46     DIAPH2
169     Q9NPB8 Recurrence 0.016800   0.722  2.46     GPCPD1
170     P01906 Recurrence 0.017100   0.722 -2.45   HLA-DQA2
171     Q5T5C0 Recurrence 0.017100   0.722  2.46     STXBP5
172     Q9Y3B8 Recurrence 0.017200   0.722  2.45      REXO2
173     P49450 Recurrence 0.017300   0.722 -2.45      CENPA
174     Q9H3Q1 Recurrence 0.017400   0.722 -2.45   CDC42EP4
175     Q9Y508 Recurrence 0.017500   0.722  2.45     RNF114
176     Q13576 Recurrence 0.017600   0.722 -2.44     IQGAP2
177     Q6ZNW5 Recurrence 0.017600   0.722  2.44     GDPGP1
178     Q8NC24 Recurrence 0.017600   0.722  2.44      RELL2
179     Q93052 Recurrence 0.017800   0.722 -2.44        LPP
180     Q7Z5V6 Recurrence 0.017800   0.722  2.44    PPP1R32
181     Q9P0L0 Recurrence 0.018300   0.736 -2.43       VAPA
182     P05997 Recurrence 0.018500   0.738  2.42     COL5A2
183     Q99538 Recurrence 0.018500   0.738  2.42       LGMN
184     Q6ZMQ8 Recurrence 0.018600   0.738  2.42       AATK
185     Q9BQ04 Recurrence 0.018800   0.738  2.42      RBM4B
186     P00568 Recurrence 0.019000   0.738 -2.41        AK1
187     P05154 Recurrence 0.019000   0.738  2.41   SERPINA5
188     P25098 Recurrence 0.019100   0.738  2.41       GRK2
189     Q02543 Recurrence 0.019100   0.738  2.41     RPL18A
190     A5YM69 Recurrence 0.019700   0.738  2.40   ARHGEF35
191     P00750 Recurrence 0.019700   0.738 -2.40       PLAT
192     P50454 Recurrence 0.019800   0.738  2.40   SERPINH1
193     O75880 Recurrence 0.019900   0.738 -2.39       SCO1
194     Q8WUA2 Recurrence 0.019900   0.738 -2.39      PPIL4
195     P04216 Recurrence 0.020000   0.738  2.39       THY1
196     P24468 Recurrence 0.020000   0.738 -2.39      NR2F2
197     Q8NHG8 Recurrence 0.020200   0.738  2.39      ZNRF2
198     P61421 Recurrence 0.020400   0.738 -2.38   ATP6V0D1
199     Q9H3U5 Recurrence 0.020400   0.738 -2.38      MFSD1
200     Q13451 Recurrence 0.020500   0.738 -2.38      FKBP5
201     Q15283 Recurrence 0.020500   0.738  2.38      RASA2
202     Q7LGA3 Recurrence 0.020600   0.738  2.38     HS2ST1
203     Q9BQ95 Recurrence 0.020800   0.738  2.38      ECSIT
204     Q9NTX5 Recurrence 0.021400   0.738 -2.36     ECHDC1
205     Q8N1F7 Recurrence 0.021500   0.738 -2.36      NUP93
206     Q5HYJ3 Recurrence 0.021700   0.738  2.36     FAM76B
207     P54709 Recurrence 0.021800   0.738 -2.36     ATP1B3
208     P20701 Recurrence 0.021900   0.738 -2.36      ITGAL
209     Q96EB6 Recurrence 0.021900   0.738 -2.35      SIRT1
210     Q99439 Recurrence 0.022100   0.738  2.35       CNN2
211     Q96EV2 Recurrence 0.022200   0.738  2.35      RBM33
212     Q16658 Recurrence 0.022400   0.738  2.35      FSCN1
213     Q9NUQ3 Recurrence 0.022400   0.738 -2.35      TXLNG
214     P61225 Recurrence 0.022500   0.738 -2.34      RAP2B
215     Q9H3N8 Recurrence 0.022500   0.738  2.34       HRH4
216     Q9H8L6 Recurrence 0.022500   0.738 -2.34      MMRN2
217     Q6P1L5 Recurrence 0.022500   0.738 -2.34    FAM117B
218     P55196 Recurrence 0.022600   0.738 -2.34       AFDN
219     Q9NWT6 Recurrence 0.022800   0.738 -2.34     HIF1AN
220     P49770 Recurrence 0.022900   0.738 -2.34     EIF2B2
221     P05186 Recurrence 0.023100   0.738 -2.33       ALPL
222     P0DOX3 Recurrence 0.023300   0.738  2.33           
223     Q6KB66 Recurrence 0.023300   0.738  2.33      KRT80
224     O60826 Recurrence 0.023500   0.738 -2.33     CCDC22
225     Q5U651 Recurrence 0.023500   0.738 -2.33     RASIP1
226     Q7Z4H8 Recurrence 0.023600   0.738  2.32    POGLUT3
227     Q96JP5 Recurrence 0.023600   0.738  2.32      ZFP91
228     Q8WTS6 Recurrence 0.023900   0.738 -2.32      SETD7
229     Q9Y3B2 Recurrence 0.024000   0.738 -2.32     EXOSC1
230     P20036 Recurrence 0.024100   0.738 -2.32   HLA-DPA1
231     Q6PIU2 Recurrence 0.024100   0.738 -2.32      NCEH1
232     Q8WWP7 Recurrence 0.024100   0.738 -2.32     GIMAP1
233     P49189 Recurrence 0.024300   0.738 -2.31    ALDH9A1
234     O14494 Recurrence 0.024400   0.738  2.31      PLPP1
235     O15014 Recurrence 0.024500   0.738  2.31     ZNF609
236     O94813 Recurrence 0.024500   0.738 -2.31      SLIT2
237     Q9Y2T2 Recurrence 0.024600   0.738  2.31      AP3M1
238     Q15124 Recurrence 0.024700   0.738 -2.31       PGM5
239     Q99943 Recurrence 0.024800   0.738  2.30     AGPAT1
240     P50914 Recurrence 0.024900   0.738  2.30      RPL14
241     Q12874 Recurrence 0.024900   0.738 -2.30      SF3A3
242     Q96D46 Recurrence 0.024900   0.738 -2.30       NMD3
243     Q99836 Recurrence 0.024900   0.738  2.30      MYD88
244     Q9UBT7 Recurrence 0.024900   0.738 -2.30    CTNNAL1
245     P00403 Recurrence 0.025000   0.738 -2.30     MT-CO2
246     Q03591 Recurrence 0.025000   0.738 -2.30      CFHR1
247     Q8TEW0 Recurrence 0.025000   0.738 -2.30      PARD3
248     P06310 Recurrence 0.025300   0.740  2.30   IGKV2-30
249     Q5XKP0 Recurrence 0.025300   0.740  2.30    MICOS13
250     O75152 Recurrence 0.025400   0.741 -2.29    ZC3H11A
251     P08473 Recurrence 0.025600   0.741 -2.29        MME
252     Q9UN37 Recurrence 0.025600   0.741 -2.29      VPS4A
253     Q92508 Recurrence 0.025900   0.745 -2.29     PIEZO1
254     P35573 Recurrence 0.026000   0.745  2.29        AGL
255     O43314 Recurrence 0.026100   0.745  2.28    PPIP5K2
256     Q96AJ9 Recurrence 0.026500   0.745 -2.28      VTI1A
257     Q9BRQ8 Recurrence 0.026500   0.745 -2.28      AIFM2
258     Q86VP3 Recurrence 0.026700   0.745 -2.27      PACS2
259     Q96TC7 Recurrence 0.026700   0.745 -2.27      RMDN3
260     Q9Y3Z3 Recurrence 0.026700   0.745 -2.27     SAMHD1
261     Q96LZ7 Recurrence 0.026800   0.745 -2.27      RMDN2
262     O75071 Recurrence 0.026900   0.745 -2.27    EFCAB14
263     Q9Y276 Recurrence 0.026900   0.745  2.27      BCS1L
264     P16070 Recurrence 0.027100   0.746 -2.27       CD44
265     Q9NZN4 Recurrence 0.027100   0.746 -2.27       EHD2
266     Q92522 Recurrence 0.027200   0.746 -2.27      H1-10
267     Q9UGU5 Recurrence 0.027300   0.746  2.26     HMGXB4
268     Q8IXQ6 Recurrence 0.027700   0.753 -2.26      PARP9
269     Q13541 Recurrence 0.027800   0.753  2.26   EIF4EBP1
270     P00739 Recurrence 0.028000   0.753 -2.25        HPR
271     Q9ULV4 Recurrence 0.028000   0.753 -2.25     CORO1C
272     Q8NDV1 Recurrence 0.028200   0.754  2.25 ST6GALNAC3
273     P24311 Recurrence 0.028300   0.754 -2.25      COX7B
274     Q9NVU7 Recurrence 0.028300   0.754 -2.25      SDAD1
275     P78527 Recurrence 0.028800   0.762  2.24      PRKDC
276     Q9Y653 Recurrence 0.029200   0.769  2.24     ADGRG1
277     Q8NAA6 Recurrence 0.029400   0.769 -2.23  LINC02694
278     Q86T13 Recurrence 0.029600   0.769 -2.23    CLEC14A
279     Q9BUK6 Recurrence 0.029600   0.769 -2.23      MSTO1
280     Q9NTZ6 Recurrence 0.029700   0.769  2.23      RBM12
281     Q9UEG4 Recurrence 0.029700   0.769 -2.23     ZNF629
282     Q9Y5U2 Recurrence 0.029700   0.769  2.23      TSSC4
283     Q8IVH2 Recurrence 0.030000   0.772  2.23      FOXP4
284     P16930 Recurrence 0.030400   0.774 -2.22        FAH
285     Q9HD42 Recurrence 0.030400   0.774 -2.22     CHMP1A
286     O75128 Recurrence 0.030500   0.774  2.22       COBL
287     P49326 Recurrence 0.030600   0.774 -2.22       FMO5
288     P50281 Recurrence 0.030600   0.774  2.22      MMP14
289     Q15599 Recurrence 0.030900   0.779 -2.21   SLC9A3R2
290     Q9NRN7 Recurrence 0.031100   0.783  2.21   AASDHPPT
291     O94762 Recurrence 0.031500   0.787 -2.20     RECQL5
292     Q9P031 Recurrence 0.031500   0.787 -2.20     CCDC59
293     Q9Y4X5 Recurrence 0.031600   0.787  2.20      ARIH1
294     O60884 Recurrence 0.032100   0.791 -2.20     DNAJA2
295     P21860 Recurrence 0.032100   0.791  2.20      ERBB3
296     Q5VST9 Recurrence 0.032100   0.791  2.20      OBSCN
297     Q99611 Recurrence 0.032400   0.791  2.19     SEPHS2
298     P00747 Recurrence 0.032500   0.791 -2.19        PLG
299     Q14247 Recurrence 0.032500   0.791  2.19       CTTN
300     Q9H492 Recurrence 0.032600   0.791  2.19   MAP1LC3A
301     A0M8Q6 Recurrence 0.032800   0.793 -2.19      IGLC7
302     Q13404 Recurrence 0.032900   0.793  2.19     UBE2V1
303     P34741 Recurrence 0.033000   0.794  2.18       SDC2
304     Q13488 Recurrence 0.033200   0.794 -2.18     TCIRG1
305     Q15113 Recurrence 0.033600   0.794  2.18     PCOLCE
306     O95785 Recurrence 0.033700   0.794 -2.18        WIZ
307     Q9NWV4 Recurrence 0.033900   0.794 -2.17       CZIB
308     Q06278 Recurrence 0.034000   0.794 -2.17       AOX1
309     Q9NQC8 Recurrence 0.034200   0.794 -2.17      IFT46
310     Q5TAQ9 Recurrence 0.034300   0.794 -2.17      DCAF8
311     Q12955 Recurrence 0.034400   0.794 -2.17       ANK3
312     Q9UHL4 Recurrence 0.034500   0.794 -2.17       DPP7
313     O14841 Recurrence 0.034700   0.794  2.16      OPLAH
314     Q15293 Recurrence 0.034700   0.794  2.16       RCN1
315     Q9Y2Z4 Recurrence 0.034900   0.794  2.16      YARS2
316     Q15843 Recurrence 0.035100   0.794  2.16      NEDD8
317     Q16822 Recurrence 0.035100   0.794  2.16       PCK2
318     P62826 Recurrence 0.035200   0.794  2.16        RAN
319     Q9BWM7 Recurrence 0.035200   0.794 -2.16      SFXN3
320     Q9UDY2 Recurrence 0.035200   0.794 -2.16       TJP2
321     Q96S44 Recurrence 0.035600   0.794  2.15     TP53RK
322     O60220 Recurrence 0.035700   0.794  2.15     TIMM8A
323     Q3SY69 Recurrence 0.036100   0.794  2.15    ALDH1L2
324     Q9UPY3 Recurrence 0.036200   0.794  2.14     DICER1
325     Q01484 Recurrence 0.036300   0.794 -2.14       ANK2
326     Q8WV93 Recurrence 0.036500   0.794  2.14      AFG1L
327     Q8WVK2 Recurrence 0.036500   0.794 -2.14    SNRNP27
328     O95436 Recurrence 0.036600   0.794 -2.14    SLC34A2
329     P62854 Recurrence 0.036600   0.794  2.14      RPS26
330     Q9Y613 Recurrence 0.036600   0.794 -2.14      FHOD1
331     P08754 Recurrence 0.036700   0.794  2.14      GNAI3
332     P57053 Recurrence 0.036700   0.794 -2.14    H2BC12L
333     P49591 Recurrence 0.036900   0.794  2.14      SARS1
334     P61158 Recurrence 0.036900   0.794  2.14      ACTR3
335     Q16512 Recurrence 0.037000   0.794  2.14       PKN1
336     P35556 Recurrence 0.037100   0.794  2.13       FBN2
337     P80188 Recurrence 0.037100   0.794  2.13       LCN2
338     Q9NUY8 Recurrence 0.037100   0.794 -2.13    TBC1D23
339     P48147 Recurrence 0.037200   0.794  2.13       PREP
340     Q6ZNL6 Recurrence 0.037300   0.794  2.13       FGD5
341     Q86TP1 Recurrence 0.037300   0.794  2.13     PRUNE1
342     P51692 Recurrence 0.037400   0.794  2.13     STAT5B
343     P49755 Recurrence 0.037500   0.794  2.13     TMED10
344     Q9UHR6 Recurrence 0.037500   0.794  2.13     ZNHIT2
345     Q969X1 Recurrence 0.037600   0.794  2.13     TMBIM1
346     Q9NWY4 Recurrence 0.037800   0.794  2.13       HPF1
347     Q9Y4G8 Recurrence 0.037800   0.794  2.13    RAPGEF2
348     Q92796 Recurrence 0.038100   0.798 -2.12       DLG3
349     P01599 Recurrence 0.038500   0.799 -2.12   IGKV1-17
350     Q7Z4I7 Recurrence 0.038600   0.799 -2.12      LIMS2
351     P21246 Recurrence 0.038700   0.799 -2.12        PTN
352     P62306 Recurrence 0.038700   0.799  2.12      SNRPF
353     Q16627 Recurrence 0.038800   0.799 -2.11      CCL14
354     Q9H0F6 Recurrence 0.038800   0.799  2.11    SHARPIN
355     Q12802 Recurrence 0.039000   0.800 -2.11     AKAP13
356     P10746 Recurrence 0.039200   0.800 -2.11       UROS
357     Q96D15 Recurrence 0.039300   0.800  2.11       RCN3
358     Q9UJS0 Recurrence 0.039300   0.800  2.11   SLC25A13
359     Q6PD62 Recurrence 0.039400   0.800  2.11       CTR9
360     O43318 Recurrence 0.039600   0.800  2.11     MAP3K7
361     Q13813 Recurrence 0.039600   0.800 -2.11     SPTAN1
362     Q6FIF0 Recurrence 0.039900   0.802 -2.10     ZFAND6
363     Q9BXB4 Recurrence 0.040000   0.802 -2.10    OSBPL11
364     Q06265 Recurrence 0.040100   0.802 -2.10     EXOSC9
365     P08631 Recurrence 0.040400   0.804 -2.10        HCK
366     Q99685 Recurrence 0.040400   0.804 -2.10       MGLL
367     Q15819 Recurrence 0.040600   0.806  2.09     UBE2V2
368     O95470 Recurrence 0.040700   0.806  2.09      SGPL1
369     P37235 Recurrence 0.041100   0.810 -2.09     HPCAL1
370     Q4G176 Recurrence 0.041100   0.810 -2.09      ACSF3
371     Q13433 Recurrence 0.041800   0.811  2.08    SLC39A6
372     Q96EK6 Recurrence 0.042100   0.811  2.08    GNPNAT1
373     P48739 Recurrence 0.042500   0.811  2.07     PITPNB
374     P13073 Recurrence 0.042600   0.811 -2.07     COX4I1
375     P17813 Recurrence 0.042600   0.811 -2.07        ENG
376     Q7RTP6 Recurrence 0.042600   0.811 -2.07     MICAL3
377     Q16647 Recurrence 0.042700   0.811  2.07      PTGIS
378     Q9UK41 Recurrence 0.042700   0.811  2.07      VPS28
379     O75380 Recurrence 0.042800   0.811 -2.07     NDUFS6
380     Q12765 Recurrence 0.042800   0.811 -2.07      SCRN1
381     Q8IUE6 Recurrence 0.042800   0.811  2.07     H2AC21
382     Q8WXF1 Recurrence 0.042800   0.811  2.07      PSPC1
383     P29466 Recurrence 0.042900   0.811 -2.07      CASP1
384     Q9UKY3 Recurrence 0.043000   0.811 -2.07     CES1P1
385     Q5JPH6 Recurrence 0.043200   0.811  2.07      EARS2
386     Q96AT1 Recurrence 0.043400   0.811 -2.06   KIAA1143
387     Q9H223 Recurrence 0.043500   0.811 -2.06       EHD4
388     O94956 Recurrence 0.043800   0.811 -2.06    SLCO2B1
389     O14530 Recurrence 0.043900   0.811  2.06     TXNDC9
390     O15143 Recurrence 0.043900   0.811  2.06     ARPC1B
391     Q96SL4 Recurrence 0.043900   0.811  2.06       GPX7
392     Q9BXR0 Recurrence 0.043900   0.811  2.06      QTRT1
393     Q9HCC0 Recurrence 0.044000   0.811  2.06      MCCC2
394     P01742 Recurrence 0.044100   0.811 -2.06   IGHV1-69
395     P05546 Recurrence 0.044400   0.811 -2.05   SERPIND1
396     P13796 Recurrence 0.044400   0.811 -2.06       LCP1
397     P10606 Recurrence 0.044500   0.811 -2.05      COX5B
398     Q96CD0 Recurrence 0.044500   0.811  2.05      FBXL8
399     Q9NWU5 Recurrence 0.044500   0.811  2.05     MRPL22
400     Q99758 Recurrence 0.044800   0.811 -2.05      ABCA3
401     P23528 Recurrence 0.044900   0.811  2.05       CFL1
402     Q9UPA5 Recurrence 0.044900   0.811  2.05        BSN
403     Q13137 Recurrence 0.045100   0.811 -2.05   CALCOCO2
404     Q92930 Recurrence 0.045300   0.811 -2.05      RAB8B
405     Q9H1E3 Recurrence 0.045500   0.811  2.04     NUCKS1
406     Q9NVS9 Recurrence 0.045500   0.811 -2.04       PNPO
407     Q8WUH6 Recurrence 0.045600   0.811  2.04    TMEM263
408     Q9UPQ0 Recurrence 0.045700   0.811 -2.04     LIMCH1
409     Q7L804 Recurrence 0.045800   0.811  2.04  RAB11FIP2
410     O95340 Recurrence 0.045900   0.811 -2.04     PAPSS2
411     P67812 Recurrence 0.046100   0.811  2.04     SEC11A
412     Q567U6 Recurrence 0.046100   0.811 -2.04     CCDC93
413     Q6ZSY5 Recurrence 0.046200   0.811  2.04    PPP1R3F
414     P16333 Recurrence 0.046400   0.811 -2.04       NCK1
415     Q13907 Recurrence 0.046400   0.811  2.04       IDI1
416     P61313 Recurrence 0.046600   0.811  2.03      RPL15
417     Q9C037 Recurrence 0.046600   0.811 -2.03      TRIM4
418     Q9UIV1 Recurrence 0.046800   0.811  2.03      CNOT7
419     Q96EY4 Recurrence 0.046900   0.811 -2.03      TMA16
420     Q9P1F3 Recurrence 0.046900   0.811  2.03     ABRACL
421     Q08174 Recurrence 0.047000   0.811 -2.03      PCDH1
422     Q15493 Recurrence 0.047000   0.811 -2.03        RGN
423     O00418 Recurrence 0.047200   0.811 -2.03      EEF2K
424     Q9Y606 Recurrence 0.047200   0.811  2.03       PUS1
425     Q9UKK9 Recurrence 0.047300   0.811 -2.03      NUDT5
426     Q99988 Recurrence 0.047400   0.811  2.03      GDF15
427     P0C870 Recurrence 0.047600   0.813 -2.02      JMJD7
428     P61956 Recurrence 0.047800   0.814  2.02      SUMO2
429     Q96GX9 Recurrence 0.048100   0.817  2.02       APIP
430     O75367 Recurrence 0.048300   0.817 -2.02  MACROH2A1
431     Q9Y5X1 Recurrence 0.048300   0.817  2.02       SNX9
432     P41218 Recurrence 0.048600   0.821 -2.01       MNDA
433     Q9UIA9 Recurrence 0.049200   0.828  2.01       XPO7
434     O43157 Recurrence 0.049300   0.828  2.01     PLXNB1
435     Q2M385 Recurrence 0.049400   0.828  2.01      MPEG1
436     P50591 Recurrence 0.049500   0.828 -2.01    TNFSF10
437     Q9HBL0 Recurrence 0.049600   0.828 -2.01       TNS1
438     O95810 Recurrence 0.050000   0.829 -2.00     CAVIN2</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proTumorRes_Klin$sig.feat.heat</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Normal samples (ns:
p &gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01, <span
class="math inline">\(***\)</span>: p &lt;= 0.001)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes4Viz_Klin$dataCorrect
smpAnnoTab &lt;- proTissueRes4Viz_Klin$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proTumorRes_Klin$sig.feat.tab
num_sigFeats &lt;- 6
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = Gene))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Condition = factor(Condition, levels = c(&#39;Tumor&#39;, &#39;Normal&#39;)),
                Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
# Visualize top significant features
ggplot(topSigFeatDat, aes(x=Condition, y=Abundance, fill=Recurrence)) +
  geom_boxplot(alpha = 1, linewidth = 1, outlier.color = &#39;grey50&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             label = &#39;p.signif&#39;, size = 7, vjust = 0.5) +
  labs(x = &#39;Sample condition&#39;, y = &#39;log(Abundance)&#39;) +
  scale_fill_manual(values = c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  geom_vline(xintercept = 1.5, size = 0.3) +
  th +
  theme(strip.text = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 22),
        axis.text.x = element_text(size = 18),
        legend.title = element_text(size = 24), legend.text = element_text(size = 22))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proTumorRes_Klin$pval.hist +
  theme(axis.title = element_text(size = 22), axis.text = element_text(size = 18))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/25.07.24_LGM/pValHist_combined_proTumor_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 14)</code></pre>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2     pVal  pValAdj  Stat   Test
1 PC1 (54.3%) Recurrence 6.43e-06 0.000129  4.84 T-test
2  PC5 (1.8%) Recurrence 1.68e-05 0.000168  4.59 T-test
3 PC2 (21.3%) Recurrence 2.69e-05 0.000179  4.47 T-test
4  PC9 (1.3%) Recurrence 2.67e-02 0.134000 -2.26 T-test</code></pre>
<p>Visualize significant PCs</p>
<pre class="r"><code>p &lt;- proTumorRes_Klin$sig.pc.dist$PC1
p$layers[[2]]$aes_params$size &lt;- 4
p$layers[[3]]$aes_params$size &lt;- 8
p #+ theme(axis.title = element_text(size = 22),</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#           axis.text = element_text(size = 14),
#           legend.title = element_text(size = 22),
#           legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/sigPC1_combined_proTumor_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)

proTumorRes_Klin$sig.pc.dist$PC5</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-30-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.dist$PC2</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-30-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.dist$PC9</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-30-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># pcTab &lt;- proTumorRes_Klin$SOA.res$pcTab %&gt;%
#   dplyr::mutate(Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
# ggplot(pcTab, aes(x=`PC1 (54.3%)`, y=`PC2 (21.3%)`, col=Recurrence)) +
#   geom_point(size = 5) +
#   th +
#   theme(axis.title = element_text(size = 22),
#         axis.text = element_text(size = 14),
#         legend.title = element_text(size = 22),
#         legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/sigPC1&amp;2_combined_proTumor_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 11)</code></pre>
<p>Visualize feature significance through volcano plot</p>
<pre class="r"><code>#### CHANGE HERE ####
soaRes &lt;- proTumorRes_Klin
seMedi &lt;- proTissueMediSVs_Klin[, which(colData(proTissueMediSVs_Klin)$Condition == &#39;Tumor&#39;)]
fcCutoff &lt;- 0.5
#####################

# Compute log2(FC) using proDA
# Do SOA to view data structure and p-value histogram
# soaMediRes &lt;- doSingleOmicsAnalysis(seMedi, soa_metaVar = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
#                                     num_PCs = 20, use_proDA = T,
#                                     pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                     &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;),
#                                     soa_unwantVar = grep(&#39;^SV\\d+$&#39;, colnames(colData(seMedi)),
#                                                          value = T))
# soaMediRes$pval.hist
# soaMediRes$data.var.source

# Prepare data matrix and sample metadata table for creating model matrices
datMat &lt;- assay(seMedi)
metadatTab &lt;- tibble::as_tibble(colData(seMedi), rownames = &#39;Sample&#39;) %&gt;%
  # Make comparison Recurrence vs Nonrecurrence
  dplyr::mutate(Recurrence = factor(Recurrence, levels = c(&#39;No&#39;, &#39;Yes&#39;)))
formu &lt;- paste0(&#39;~Recurrence+&#39;, paste0(grep(&#39;^SV\\d+$&#39;, colnames(metadatTab),
                                            value = T), collapse = &#39;+&#39;)) %&gt;%
  as.formula()
design &lt;- model.matrix(formu, data = metadatTab)

# fit &lt;- proDA(datMat, design = design)
# diffTerm &lt;- proDA::result_names(fit)[2]
# proDARes &lt;- proDA::test_diff(fit, contrast = diffTerm)
# saveRDS(proDARes, &#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorLog2FC_SVs.rds&#39;)
proDARes &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorLog2FC_SVs.rds&#39;)

# Make volcano plot
# Prepare needed information
pValTab &lt;- soaRes$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, pVal, pValAdj) %&gt;%
  dplyr::rename(Feature = Var1)
log2FCTab &lt;- dplyr::select(proDARes, name, diff) %&gt;%
  dplyr::rename(Feature = name, log2FC = diff)
featAnnoTab &lt;- tibble::as_tibble(rowData(seMedi), rownames = &#39;Feature&#39;)
# Combine all needed information
volcanoTab &lt;- dplyr::left_join(pValTab, log2FCTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  # Pinpoint significant features
  dplyr::mutate(DiffExp = dplyr::case_when(pVal &lt;= 0.05 &amp; log2FC &gt;= fcCutoff ~ &#39;Recurrence&#39;,
                                           pVal &lt;= 0.05 &amp; log2FC &lt;= -fcCutoff ~ &#39;Nonrecurrence&#39;),
                DiffExp = dplyr::case_when(is.na(DiffExp) ~ &#39;Not significant&#39;,
                                           !is.na(DiffExp) ~ DiffExp),
                DiffExp = factor(DiffExp, levels = c(&#39;Recurrence&#39;, &#39;Nonrecurrence&#39;, &#39;Not significant&#39;)),
                # Label = dplyr::case_when(!DiffExp %in% &#39;Not significant&#39; ~ Gene),
                Label = dplyr::case_when(pVal &lt;= 0.01 &amp; log2FC &gt;= fcCutoff ~ Gene,
                                         pVal &lt;= 0.01 &amp; log2FC &lt;= -fcCutoff ~ Gene))

ggplot(volcanoTab, aes(x=log2FC, y=-log10(pVal), col=DiffExp, label=Label)) +
  geom_point(size = 2, alpha = 0.9) +
  geom_hline(yintercept = -log10(0.05), linewidth = 0.2) +
  geom_vline(xintercept = c(-fcCutoff, fcCutoff), linewidth = 0.2) +
  geom_text_repel(size = 4, show.legend = F) +
  scale_color_manual(name = &#39;&#39;, values = c(&#39;red&#39;, &#39;navy&#39;, &#39;grey&#39;)) +
  labs(x = &#39;log2(FC)&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#   theme(axis.title = element_text(size = 22),
#         axis.text = element_text(size = 18),
#         legend.title = element_text(size = 22),
#         legend.text = element_text(size = 20))
# ggsave(&#39;./output/Discovery/25.07.24_LGM/volcano_combined_proTumor_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 8, width = 13)</code></pre>
<p>Enrichment analysis<br />
Reference: Hallmark gene sets<br />
<strong>t-values</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proTumorRes_Klin$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, Stat) %&gt;%
  tibble::column_to_rownames(&#39;Var1&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)

# Do GSEA
# Hallmark gene sets
gseaRes_H &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 10, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaRes_H$plotTopSigGS +
  labs(title = &#39;Hallmark gene sets&#39;) +
  theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 18),
        axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
        legend.title = element_text(size = 20),
        legend.text = element_text(size = 16),
        panel.grid.minor = element_blank(),
        plot.title = element_text(size = 14, hjust = 0.5, vjust = 2, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize functional classification through heatmap</p>
<pre class="r"><code>#### CHANGE HERE ####
interestGS &lt;- c(&#39;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION&#39;, &#39;HALLMARK_KRAS_SIGNALING_UP&#39;)
soaRes &lt;- proTumorRes_Klin
seMedi &lt;- proTissueMediSVs_Klin[, which(colData(proTissueMediSVs_Klin)$Condition == &#39;Tumor&#39;)]
#####################

# Prepare core enrichment list retrieved from EA result
coreEnrichmentTab &lt;- gseaRes_H$gseaRes@result %&gt;%
  dplyr::select(Description, core_enrichment) %&gt;%
  dplyr::filter(Description %in% interestGS)
coreEnrichmentList &lt;- lapply(seq_len(nrow(coreEnrichmentTab)), function(i) {
  stringr::str_split_1(coreEnrichmentTab$core_enrichment[i], &#39;/&#39;)
})
names(coreEnrichmentList) &lt;- coreEnrichmentTab$Description
# Prepare feature list of interest
coreFeats &lt;- unique(unlist(coreEnrichmentList))
# Retrieve significant feature list from SOA result
sigFeats &lt;- stringr::str_remove(soaRes$sig.feat.tab$Gene, &#39;;.+$&#39;)
# Check if there is any duplicated gene
# any(duplicated(sigFeats))

# Prepare log2(FC) table
proDARes &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorLog2FC_SVs.rds&#39;)
featAnnoTab &lt;- tibble::as_tibble(rowData(seMedi), rownames = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = stringr::str_remove(Gene, &#39;;.+$&#39;)) %&gt;%
  dplyr::filter(Gene %in% coreFeats)
log2FCTab &lt;- dplyr::select(proDARes, name, diff) %&gt;%
  dplyr::rename(Feature = name, log2FC = diff) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::filter(!is.na(Gene)) %&gt;%
  dplyr::select(-Feature) %&gt;%
  tibble::column_to_rownames(&#39;Gene&#39;)
coreFeatsOrderByT &lt;- soaRes$SOA.res$featAssoRes %&gt;%
  dplyr::rename(Feature = Var1) %&gt;%
  dplyr::left_join(featAnnoTab, by = &#39;Feature&#39;) %&gt;%
  dplyr::filter(!is.na(Gene)) %&gt;%
  dplyr::mutate(Gene = dplyr::case_when(Gene %in% sigFeats ~ paste0(&#39;*&#39;, Gene),
                                        !Gene %in% sigFeats ~ Gene)) %&gt;%
  dplyr::arrange(desc(Stat)) %&gt;%
  dplyr::pull(Gene)

# Plot heatmap
# Create binary matrix indicating presence of features in overrepresented gene sets
binaryMat &lt;- matrix(nrow = length(coreEnrichmentList), ncol = length(coreFeats),
                    dimnames = list(names(coreEnrichmentList), coreFeats))
for (i in seq_len(nrow(binaryMat))) {
  for (j in seq_len(ncol(binaryMat))) {
    if (colnames(binaryMat)[j] %in% coreEnrichmentList[[rownames(binaryMat)[i]]]) {
      binaryMat[i, j] &lt;- T
    } else {
      binaryMat[i, j] &lt;- F
    }
  }
}
# Create heatmap matrix based on binary matrix and statistics results
gsOverrepHeatmap &lt;- tibble::as_tibble(binaryMat, rownames = &#39;Pathway&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Pathway&#39;, names_to = &#39;Gene&#39;, values_to = &#39;Presence&#39;) %&gt;%
  dplyr::mutate(log2FC = dplyr::case_when(Presence ~ log2FCTab[Gene,],
                                          !Presence ~ 0),
                Gene = dplyr::case_when(Gene %in% sigFeats ~ paste0(&#39;*&#39;, Gene),
                                        !Gene %in% sigFeats ~ Gene),
                Gene = factor(Gene, levels = rev(coreFeatsOrderByT)),
                Pathway = stringr::str_remove(Pathway, &#39;^HALLMARK_&#39;),
                Pathway = factor(Pathway, levels = str_remove(interestGS, &#39;^HALLMARK_&#39;)))
ggplot(gsOverrepHeatmap, aes(x=Pathway, y=Gene, fill=log2FC)) +
  geom_tile() +
  scale_fill_gradient2(high = &#39;darkred&#39;, low = &#39;darkblue&#39;, mid = &#39;white&#39;, midpoint = 0,
                       name = &#39;log2(FC)&#39;) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) +
  labs(title = &#39;Hallmark: log2(FC)&#39;) +
  theme(axis.text.y = element_text(size = 10))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>PC loadings</strong><br />
<strong>PC5</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proTumorRes_Klin$SOA.res$pcaRes@loadings %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, PC5) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)

# Do GSEA
# Hallmark gene sets
gseaRes_H &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 10, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaRes_H$plotTopSigGS</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>PC9</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proTumorRes_Klin$SOA.res$pcaRes@loadings %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, PC9) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)
rankedGeneList_Klin &lt;- sort(-rankedGeneList_Klin, decreasing = T)

# Do GSEA
# Hallmark gene sets
gseaRes_H &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 10, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaRes_H$plotTopSigGS</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tissue-proteomics-ag-krijgsveld" class="section level1">
<h1>Tissue Proteomics (AG Krijgsveld)</h1>
<p>Discovery + MethodDev</p>
<pre class="r"><code># Load preprocessed data
proTissue_Krij &lt;- readRDS(&#39;./data/Discovery/AG_Krijgsveld/combined_proTissueVsnBC.rds&#39;)
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::select(Protein, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)

# Perform single-omics analysis
proTissueRes_Krij &lt;- doSOA(proTissue_Krij, meta_var = c(&#39;Condition&#39;, &#39;Cohort&#39;), pca_method = &#39;ppca&#39;, do_onlyPCA = T)</code></pre>
<p><strong>Do metadata-assisted quality control</strong><br />
After testing associations between PCs and tissue sample conditions, we
found that PC1 significantly separates Tumor and Normal samples, which
indicates decent data quality. Yet, there are 4 Tumor samples
misclassified, reported as follows:<br />
=&gt; Tumor purity: I0HVOL_TU - 87% / 7EAOX7_TU - 70% / MJMTYR_TU - 40%
/ XFKHGP_TU - 100%<br />
(Mean of sample tumor cell contents is 74%, ranging from 40% to
100%.)</p>
<pre class="r"><code>pcTab &lt;- proTissueRes_Krij$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;,
                                                       &#39;MJMTYR_TU&#39;, &#39;XFKHGP_TU&#39;,
                                                       &#39;XFKHGP_NG&#39;, &#39;MJMTYR_NG&#39;,
                                                       &#39;I0HVOL_NG&#39;, &#39;7EAOX7_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (38%)`, color=Condition, fill=Condition, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="tumor-samples-1" class="section level2">
<h2>Tumor samples</h2>
<p>Display significant associations between SVs and sample metadata
variables. SVs (surrogate variables) will be included in linear model
for differential analysis</p>
<pre class="r"><code># Subset certain samples
proTumor_Krij &lt;- proTissue_Krij[, which(colData(proTissue_Krij)$Condition == &#39;Tumor&#39;)]

# Do SVA to identify latent factors that explain unknown variance
# Prepare data matrix and sample metadata table for creating model matrices
datMat &lt;- assay(proTumor_Krij)
# Keep only complete features
keptFeats &lt;- apply(datMat, 1, function(featVec) {
  !any(is.na(featVec))
})
datMat &lt;- datMat[keptFeats,]
metadatTab &lt;- colData(proTumor_Krij) %&gt;%
  tibble::as_tibble(rownames = &#39;Sample&#39;)
# Create full model matrix including both adjustment variables and variables of interest
modelMat &lt;- model.matrix(~ Recurrence, data = metadatTab)
# Create null model matrix including only adjustment variables
modelMat0 &lt;- model.matrix(~ 1, data = metadatTab)

# Perform SVA
# Identify number of latent factors that need to be estimated
numSV = sva::num.sv(datMat, modelMat, method = &#39;be&#39;)
svObj = sva(datMat, modelMat, modelMat0, n.sv = numSV)</code></pre>
<pre><code>Number of significant surrogate variables is:  11 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code># Do association tests between SVs and patient metadata
# Prepare surrogate variable table
svTab &lt;- svObj$sv
rownames(svTab) &lt;- metadatTab$Sample
colnames(svTab) &lt;- paste0(&#39;SV&#39;, seq_len(ncol(svTab)))
svTab &lt;- tibble::as_tibble(svTab, rownames = &#39;Sample&#39;)
# Prepare patient metadata table
metaTab &lt;- dplyr::select(metadatTab, Sample, Recurrence, Gender, Age, Smoking, Stage,
                         Adjuvant, TumorPurity)
# Do association tests
testAsso(svTab, metaTab, cmn_col = &#39;Sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05)</code></pre>
<pre><code>  Var1        Var2         pVal      pValAdj       Stat        Test
1  SV4 TumorPurity 1.213136e-06 9.341146e-05 -0.5678292 Correlation
2  SV1 TumorPurity 2.813245e-04 8.122635e-03 -0.4425680 Correlation
3  SV6      Gender 3.164663e-04 8.122635e-03 -3.8186621      T-test
4  SV9    Adjuvant 5.595505e-03 1.077135e-01  2.8724503      T-test
5  SV6 TumorPurity 1.426434e-02 2.196709e-01  0.3073793 Correlation
6  SV1       Stage 2.534211e-02 2.848100e-01  3.9098990       ANOVA
7  SV9     Smoking 2.589182e-02 2.848100e-01  3.8856512       ANOVA
8  SV5       Stage 3.288883e-02 3.165550e-01  3.6165372       ANOVA</code></pre>
<pre class="r"><code># Include surrogate variables into SE object
proTumorTab &lt;- summExp2df(proTumor_Krij, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;)
proTumorTab &lt;- dplyr::left_join(proTumorTab, svTab, by = &#39;Sample&#39;)
proTumor_Krij &lt;- df2SummExp(proTumorTab, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;,
                            values = &#39;Value&#39;, row_anno = colnames(rowData(proTumor_Krij)),
                            col_anno = c(colnames(colData(proTumor_Krij)),
                                         paste0(&#39;SV&#39;, seq_len(ncol(svTab)-1))))
assay(proTumor_Krij) &lt;- as.matrix(assay(proTumor_Krij))</code></pre>
<pre class="r"><code># Perform single-omics analysis
# proTumorRes_Krij &lt;- doSingleOmicsAnalysis(proTumor_Krij, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                           pca_method = &#39;ppca&#39;, num_PCs = 40, num_PCfeats = 30,
#                                           use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                           feat_conv = T, plot_title = NULL, show_rownames = F,
#                                           pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                           &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;,
#                                                           &#39;TumorPurity&#39;, &#39;Cohort&#39;),
#                                           soa_unwantVar = paste0(&#39;SV&#39;, seq_len(11)))
proTumorRes_Krij &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proTumorRes_Krij.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proTumorRes_Krij$data.var.source</code></pre>
<pre><code>          Var1        Var2     pVal pValAdj   Stat        Test
1   PC4 (5.3%) TumorPurity 0.000029  0.0093  0.501 Correlation
2   PC2 (7.7%) TumorPurity 0.003280  0.3900  0.365 Correlation
3   PC5 (4.9%)         Age 0.006250  0.3900 -0.341 Correlation
4  PC35 (0.9%)      Cohort 0.007340  0.3900  2.770      T-test
5  PC10 (3.4%)       Stage 0.008400  0.3900  5.180       ANOVA
6  PC21 (1.8%)    Adjuvant 0.008750  0.3900  2.710      T-test
7   PC9 (3.6%)       Stage 0.009720  0.3900  5.010       ANOVA
8   PC1 (9.1%)       Stage 0.009750  0.3900  5.010       ANOVA
9  PC14 (2.6%)      Cohort 0.011800  0.4050  2.600      T-test
10 PC11 (3.1%)    Adjuvant 0.013800  0.4050 -2.540      T-test
11 PC26 (1.4%)    Adjuvant 0.013900  0.4050  2.530      T-test
12 PC23 (1.5%)      Gender 0.015400  0.4110  2.490      T-test
13   PC33 (1%)  Recurrence 0.016800  0.4130  2.460      T-test
14 PC11 (3.1%)  Recurrence 0.018200  0.4130 -2.430      T-test
15 PC29 (1.2%)      Cohort 0.020400  0.4130  2.380      T-test
16 PC12 (2.8%)      Gender 0.020700  0.4130 -2.380      T-test
17  PC1 (9.1%) TumorPurity 0.022600  0.4210 -0.287 Correlation
18 PC21 (1.8%)         Age 0.024300  0.4210 -0.284 Correlation
19  PC7 (4.2%)    Adjuvant 0.025000  0.4210 -2.300      T-test
20 PC27 (1.3%)         Age 0.032400  0.5190 -0.270 Correlation
21 PC28 (1.3%)         Age 0.036300  0.5300  0.264 Correlation
22  PC8 (3.8%)       Stage 0.036500  0.5300  3.500       ANOVA
23 PC17 (2.1%)     Smoking 0.040800  0.5680  3.380       ANOVA
24 PC23 (1.5%)       Stage 0.046400  0.6070  3.230       ANOVA
25 PC31 (1.1%)         Age 0.049200  0.6070  0.249 Correlation</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2). Owing to missing values, there may be situation that
only few observations in sample groups, resulting in unreliable
t-statistics. To this end, we used probabilistic dropout model (proDA)
to account for data missingness.</p>
<pre class="r"><code>proTumorRes_Krij$sig.feat.tab</code></pre>
<pre><code>      Var1       Var2     pVal pValAdj  Stat           Test       Gene
1   Q9Y6C9 Recurrence 0.000208   0.921  4.00 T-test (proDA)      MTCH2
2   Q96A58 Recurrence 0.000387   0.921 -3.81 T-test (proDA)       RERG
3   Q8WXF1 Recurrence 0.000473   0.921  3.74 T-test (proDA)      PSPC1
4   P15169 Recurrence 0.001040   0.921 -3.48 T-test (proDA)       CPN1
5   Q00613 Recurrence 0.001260   0.921  3.42 T-test (proDA)       HSF1
6   P46776 Recurrence 0.001840   0.921 -3.29 T-test (proDA)     RPL27A
7   P05164 Recurrence 0.001910   0.921 -3.28 T-test (proDA)        MPO
8   P03950 Recurrence 0.002540   0.921 -3.18 T-test (proDA)        ANG
9   Q9UHL4 Recurrence 0.002600   0.921 -3.17 T-test (proDA)       DPP7
10  O60934 Recurrence 0.003360   0.921  3.08 T-test (proDA)        NBN
11  P18074 Recurrence 0.003380   0.921  3.08 T-test (proDA)      ERCC2
12  Q8NCM8 Recurrence 0.003700   0.921 -3.05 T-test (proDA)    DYNC2H1
13  O14569 Recurrence 0.003890   0.921 -3.03 T-test (proDA)   CYB561D2
14  P10515 Recurrence 0.004010   0.921  3.02 T-test (proDA)       DLAT
15  P49589 Recurrence 0.004060   0.921  3.01 T-test (proDA)      CARS1
16  Q14671 Recurrence 0.004160   0.921  3.00 T-test (proDA)       PUM1
17  Q92878 Recurrence 0.004190   0.921  3.00 T-test (proDA)      RAD50
18  P23368 Recurrence 0.004200   0.921  3.00 T-test (proDA)        ME2
19  Q53H12 Recurrence 0.004340   0.921  2.99 T-test (proDA)        AGK
20  Q9BX95 Recurrence 0.004410   0.921 -2.98 T-test (proDA)      SGPP1
21  Q9UPY8 Recurrence 0.004410   0.921 -2.98 T-test (proDA)     MAPRE3
22  P20160 Recurrence 0.004450   0.921 -2.98 T-test (proDA)       AZU1
23  P61513 Recurrence 0.004490   0.921 -2.98 T-test (proDA)     RPL37A
24  P27169 Recurrence 0.004690   0.921 -2.96 T-test (proDA)       PON1
25  Q8IZL8 Recurrence 0.004690   0.921  2.96 T-test (proDA)      PELP1
26  O15211 Recurrence 0.004710   0.921 -2.96 T-test (proDA)       RGL2
27  Q8TDR0 Recurrence 0.005230   0.921 -2.92 T-test (proDA)   TRAF3IP1
28  Q9HC78 Recurrence 0.005230   0.921 -2.92 T-test (proDA)     ZBTB20
29  O14920 Recurrence 0.005250   0.921  2.92 T-test (proDA)      IKBKB
30  Q9BVT8 Recurrence 0.005250   0.921  2.92 T-test (proDA)      TMUB1
31  Q9Y2J2 Recurrence 0.005300   0.921 -2.92 T-test (proDA)    EPB41L3
32  P42696 Recurrence 0.005360   0.921 -2.91 T-test (proDA)      RBM34
33  Q6P1M3 Recurrence 0.005360   0.921  2.91 T-test (proDA)      LLGL2
34  O75348 Recurrence 0.005440   0.921  2.91 T-test (proDA)   ATP6V1G1
35  P26022 Recurrence 0.005590   0.921 -2.90 T-test (proDA)       PTX3
36  P30536 Recurrence 0.005620   0.921  2.89 T-test (proDA)       TSPO
37  O14495 Recurrence 0.005820   0.921 -2.88 T-test (proDA)      PLPP3
38  Q8NBM8 Recurrence 0.006050   0.921 -2.87 T-test (proDA)    PCYOX1L
39  P61158 Recurrence 0.006140   0.921  2.86 T-test (proDA)      ACTR3
40  P20702 Recurrence 0.006390   0.921  2.85 T-test (proDA)      ITGAX
41  P00739 Recurrence 0.006480   0.921 -2.84 T-test (proDA)        HPR
42  Q8NF91 Recurrence 0.006750   0.921 -2.83 T-test (proDA)      SYNE1
43  O60268 Recurrence 0.006990   0.921 -2.81 T-test (proDA)   KIAA0513
44  A6NDU8 Recurrence 0.007090   0.921 -2.81 T-test (proDA)    C5orf51
45  Q14318 Recurrence 0.007100   0.921 -2.81 T-test (proDA)      FKBP8
46  Q14165 Recurrence 0.007170   0.921 -2.80 T-test (proDA)       MLEC
47  Q8TD16 Recurrence 0.007260   0.921 -2.80 T-test (proDA)      BICD2
48  P09622 Recurrence 0.007430   0.921  2.79 T-test (proDA)        DLD
49  P08651 Recurrence 0.007480   0.921 -2.79 T-test (proDA)       NFIC
50  O43493 Recurrence 0.007760   0.921 -2.77 T-test (proDA)     TGOLN2
51  P08246 Recurrence 0.007810   0.921 -2.77 T-test (proDA)      ELANE
52  P08236 Recurrence 0.007820   0.921 -2.77 T-test (proDA)       GUSB
53  Q53LP3 Recurrence 0.008090   0.921  2.76 T-test (proDA)     SOWAHC
54  Q16822 Recurrence 0.008120   0.921  2.76 T-test (proDA)       PCK2
55  Q9ULC3 Recurrence 0.008500   0.946 -2.74 T-test (proDA)      RAB23
56  Q5HYK7 Recurrence 0.008720   0.953  2.73 T-test (proDA)     SH3D19
57  P21283 Recurrence 0.008970   0.953  2.72 T-test (proDA)   ATP6V1C1
58  Q9ULR0 Recurrence 0.009030   0.953 -2.72 T-test (proDA)       ISY1
59  O00330 Recurrence 0.009400   0.962  2.70 T-test (proDA)       PDHX
60  Q14160 Recurrence 0.009730   0.962  2.69 T-test (proDA)      SCRIB
61  Q9C0B1 Recurrence 0.009740   0.962 -2.69 T-test (proDA)        FTO
62  Q96EC8 Recurrence 0.009790   0.962 -2.69 T-test (proDA)      YIPF6
63  Q6P1N0 Recurrence 0.010000   0.962 -2.68 T-test (proDA)     CC2D1A
64  P49959 Recurrence 0.010500   0.962  2.66 T-test (proDA)      MRE11
65  Q9P003 Recurrence 0.010700   0.962 -2.65 T-test (proDA)      CNIH4
66  P61601 Recurrence 0.011100   0.962  2.64 T-test (proDA)      NCALD
67  Q9Y4E6 Recurrence 0.011100   0.962  2.64 T-test (proDA)       WDR7
68  Q6UWE0 Recurrence 0.011300   0.962  2.63 T-test (proDA)     LRSAM1
69  Q9HCC0 Recurrence 0.011300   0.962  2.63 T-test (proDA)      MCCC2
70  Q8N6S5 Recurrence 0.011600   0.962 -2.62 T-test (proDA)    ARL6IP6
71  P56181 Recurrence 0.011700   0.962  2.62 T-test (proDA)     NDUFV3
72  Q16647 Recurrence 0.012200   0.962  2.60 T-test (proDA)      PTGIS
73  O75051 Recurrence 0.012300   0.962  2.60 T-test (proDA)     PLXNA2
74  Q99538 Recurrence 0.012300   0.962  2.60 T-test (proDA)       LGMN
75  Q9NRD1 Recurrence 0.012300   0.962  2.60 T-test (proDA)      FBXO6
76  P06702 Recurrence 0.012600   0.962 -2.59 T-test (proDA)     S100A9
77  Q12768 Recurrence 0.012600   0.962  2.59 T-test (proDA)     WASHC5
78  Q96Q05 Recurrence 0.012800   0.962 -2.58 T-test (proDA)    TRAPPC9
79  Q9H977 Recurrence 0.012800   0.962 -2.58 T-test (proDA)      WDR54
80  P62888 Recurrence 0.012900   0.962 -2.58 T-test (proDA)      RPL30
81  Q68EM7 Recurrence 0.013200   0.962  2.57 T-test (proDA)   ARHGAP17
82  P23497 Recurrence 0.013500   0.962  2.56 T-test (proDA)      SP100
83  Q5HYI8 Recurrence 0.013500   0.962 -2.56 T-test (proDA)      RABL3
84  Q969X1 Recurrence 0.013600   0.962  2.56 T-test (proDA)     TMBIM1
85  Q96HS1 Recurrence 0.013600   0.962  2.56 T-test (proDA)      PGAM5
86  Q96C01 Recurrence 0.013800   0.962  2.55 T-test (proDA)    FAM136A
87  Q8IY21 Recurrence 0.014000   0.962  2.55 T-test (proDA)      DDX60
88  Q15024 Recurrence 0.014100   0.962 -2.54 T-test (proDA)     EXOSC7
89  Q9UBS4 Recurrence 0.014100   0.962 -2.54 T-test (proDA)    DNAJB11
90  Q8TB24 Recurrence 0.014300   0.962  2.54 T-test (proDA)       RIN3
91  Q6IC98 Recurrence 0.014600   0.962  2.53 T-test (proDA)     GRAMD4
92  P01903 Recurrence 0.014700   0.962 -2.53 T-test (proDA)    HLA-DRA
93  Q9H4B0 Recurrence 0.014700   0.962  2.53 T-test (proDA)    OSGEPL1
94  Q9BRX2 Recurrence 0.014900   0.962 -2.52 T-test (proDA)       PELO
95  P57678 Recurrence 0.015100   0.962  2.52 T-test (proDA)     GEMIN4
96  P98082 Recurrence 0.015100   0.962  2.52 T-test (proDA)       DAB2
97  Q93034 Recurrence 0.015900   0.962  2.50 T-test (proDA)       CUL5
98  O75923 Recurrence 0.016000   0.962 -2.49 T-test (proDA)       DYSF
99  Q8TCS8 Recurrence 0.016100   0.962  2.49 T-test (proDA)      PNPT1
100 Q9BVH7 Recurrence 0.016200   0.962  2.49 T-test (proDA) ST6GALNAC5
101 P63218 Recurrence 0.016300   0.962  2.49 T-test (proDA)       GNG5
102 Q14534 Recurrence 0.016300   0.962  2.49 T-test (proDA)       SQLE
103 Q8NB49 Recurrence 0.016300   0.962  2.48 T-test (proDA)     ATP11C
104 Q9Y6D6 Recurrence 0.016400   0.962  2.48 T-test (proDA)    ARFGEF1
105 Q6H8Q1 Recurrence 0.016500   0.962  2.48 T-test (proDA)     ABLIM2
106 Q8WTV0 Recurrence 0.016800   0.962  2.47 T-test (proDA)     SCARB1
107 O75390 Recurrence 0.016900   0.962  2.47 T-test (proDA)         CS
108 P30626 Recurrence 0.017000   0.962  2.47 T-test (proDA)        SRI
109 P08195 Recurrence 0.017500   0.984  2.46 T-test (proDA)     SLC3A2
110 P83731 Recurrence 0.018900   1.000 -2.43 T-test (proDA)      RPL24
111 P15586 Recurrence 0.019000   1.000 -2.42 T-test (proDA)        GNS
112 P01911 Recurrence 0.019200   1.000 -2.42 T-test (proDA)   HLA-DRB1
113 Q9H299 Recurrence 0.019200   1.000  2.42 T-test (proDA)   SH3BGRL3
114 Q01433 Recurrence 0.019600   1.000  2.41 T-test (proDA)      AMPD2
115 Q9BR61 Recurrence 0.019600   1.000 -2.41 T-test (proDA)      ACBD6
116 O75306 Recurrence 0.019700   1.000  2.41 T-test (proDA)     NDUFS2
117 P22894 Recurrence 0.020000   1.000 -2.40 T-test (proDA)       MMP8
118 P78383 Recurrence 0.020300   1.000 -2.40 T-test (proDA)    SLC35B1
119 Q12907 Recurrence 0.020600   1.000 -2.39 T-test (proDA)      LMAN2
120 O94762 Recurrence 0.020800   1.000 -2.39 T-test (proDA)     RECQL5
121 Q15269 Recurrence 0.021000   1.000  2.38 T-test (proDA)       PWP2
122 P61604 Recurrence 0.021100   1.000  2.38 T-test (proDA)      HSPE1
123 Q92598 Recurrence 0.021300   1.000  2.38 T-test (proDA)      HSPH1
124 P24752 Recurrence 0.021400   1.000  2.38 T-test (proDA)      ACAT1
125 P17612 Recurrence 0.021600   1.000 -2.37 T-test (proDA)     PRKACA
126 Q9NVV5 Recurrence 0.021700   1.000 -2.37 T-test (proDA)       AIG1
127 Q9NXR1 Recurrence 0.021700   1.000  2.37 T-test (proDA)       NDE1
128 Q14192 Recurrence 0.021800   1.000 -2.37 T-test (proDA)       FHL2
129 Q9P2P5 Recurrence 0.022200   1.000  2.36 T-test (proDA)      HECW2
130 P17213 Recurrence 0.022500   1.000 -2.35 T-test (proDA)        BPI
131 O95202 Recurrence 0.022800   1.000  2.35 T-test (proDA)      LETM1
132 O00168 Recurrence 0.022900   1.000 -2.35 T-test (proDA)      FXYD1
133 O95573 Recurrence 0.022900   1.000  2.35 T-test (proDA)      ACSL3
134 O75251 Recurrence 0.023300   1.000  2.34 T-test (proDA)     NDUFS7
135 Q9Y572 Recurrence 0.023300   1.000  2.34 T-test (proDA)      RIPK3
136 Q14789 Recurrence 0.023900   1.000 -2.33 T-test (proDA)     GOLGB1
137 Q9BRP4 Recurrence 0.024300   1.000 -2.32 T-test (proDA)      PAAF1
138 Q9UII2 Recurrence 0.024300   1.000  2.32 T-test (proDA)    ATP5IF1
139 Q15493 Recurrence 0.024700   1.000 -2.32 T-test (proDA)        RGN
140 Q16527 Recurrence 0.024800   1.000 -2.32 T-test (proDA)      CSRP2
141 Q96QD9 Recurrence 0.025100   1.000 -2.31 T-test (proDA)     FYTTD1
142 Q96HL8 Recurrence 0.025300   1.000 -2.31 T-test (proDA)     SH3YL1
143 O14653 Recurrence 0.025500   1.000 -2.30 T-test (proDA)      GOSR2
144 Q9NRX4 Recurrence 0.025500   1.000 -2.30 T-test (proDA)      PHPT1
145 P58546 Recurrence 0.025600   1.000  2.30 T-test (proDA)       MTPN
146 Q02083 Recurrence 0.025600   1.000 -2.30 T-test (proDA)       NAAA
147 Q9HCK8 Recurrence 0.025600   1.000  2.30 T-test (proDA)       CHD8
148 P21757 Recurrence 0.026000   1.000  2.29 T-test (proDA)       MSR1
149 Q13201 Recurrence 0.026000   1.000  2.29 T-test (proDA)      MMRN1
150 O15247 Recurrence 0.026100   1.000 -2.29 T-test (proDA)      CLIC2
151 Q9UHY7 Recurrence 0.026400   1.000  2.29 T-test (proDA)     ENOPH1
152 Q9H257 Recurrence 0.026500   1.000  2.29 T-test (proDA)      CARD9
153 P24390 Recurrence 0.026700   1.000 -2.28 T-test (proDA)     KDELR1
154 Q9H936 Recurrence 0.026800   1.000  2.28 T-test (proDA)   SLC25A22
155 Q9BQS2 Recurrence 0.026900   1.000  2.28 T-test (proDA)      SYT15
156 P10114 Recurrence 0.027300   1.000  2.27 T-test (proDA)      RAP2A
157 P08133 Recurrence 0.027500   1.000 -2.27 T-test (proDA)      ANXA6
158 O43617 Recurrence 0.027600   1.000 -2.27 T-test (proDA)    TRAPPC3
159 P61254 Recurrence 0.027600   1.000 -2.27 T-test (proDA)      RPL26
160 Q96AQ6 Recurrence 0.027800   1.000 -2.27 T-test (proDA)     PBXIP1
161 Q9BSD7 Recurrence 0.027800   1.000 -2.27 T-test (proDA)      NTPCR
162 P05109 Recurrence 0.027900   1.000 -2.27 T-test (proDA)     S100A8
163 Q9NQX3 Recurrence 0.027900   1.000  2.26 T-test (proDA)       GPHN
164 Q9BVS5 Recurrence 0.028200   1.000 -2.26 T-test (proDA)    TRMT61B
165 Q9Y2Q3 Recurrence 0.028300   1.000  2.26 T-test (proDA)      GSTK1
166 Q15819 Recurrence 0.028400   1.000  2.26 T-test (proDA)     UBE2V2
167 Q99614 Recurrence 0.028500   1.000 -2.26 T-test (proDA)       TTC1
168 Q6P1X6 Recurrence 0.028600   1.000  2.25 T-test (proDA)    C8orf82
169 Q86YQ8 Recurrence 0.028600   1.000  2.25 T-test (proDA)      CPNE8
170 Q9NW97 Recurrence 0.028800   1.000  2.25 T-test (proDA)     TMEM51
171 Q9NY33 Recurrence 0.028800   1.000  2.25 T-test (proDA)       DPP3
172 Q15653 Recurrence 0.028900   1.000  2.25 T-test (proDA)     NFKBIB
173 Q9UNF0 Recurrence 0.029000   1.000  2.25 T-test (proDA)    PACSIN2
174 Q9HAV0 Recurrence 0.029500   1.000  2.24 T-test (proDA)       GNB4
175 Q07960 Recurrence 0.029600   1.000  2.24 T-test (proDA)    ARHGAP1
176 Q8IVV8 Recurrence 0.029900   1.000 -2.24 T-test (proDA)     NKAIN4
177 P17050 Recurrence 0.030000   1.000 -2.23 T-test (proDA)       NAGA
178 P08294 Recurrence 0.030200   1.000 -2.23 T-test (proDA)       SOD3
179 Q14240 Recurrence 0.030200   1.000  2.23 T-test (proDA)     EIF4A2
180 Q9UJS0 Recurrence 0.030200   1.000  2.23 T-test (proDA)   SLC25A13
181 Q12913 Recurrence 0.030300   1.000  2.23 T-test (proDA)      PTPRJ
182 Q9BXB4 Recurrence 0.030400   1.000 -2.23 T-test (proDA)    OSBPL11
183 P13489 Recurrence 0.030700   1.000  2.22 T-test (proDA)       RNH1
184 P55011 Recurrence 0.030900   1.000  2.22 T-test (proDA)    SLC12A2
185 Q8N556 Recurrence 0.031200   1.000  2.22 T-test (proDA)      AFAP1
186 Q9Y5Q8 Recurrence 0.031300   1.000 -2.22 T-test (proDA)     GTF3C5
187 P49184 Recurrence 0.031400   1.000  2.21 T-test (proDA)   DNASE1L1
188 P12109 Recurrence 0.031700   1.000 -2.21 T-test (proDA)     COL6A1
189 Q9Y450 Recurrence 0.031900   1.000 -2.21 T-test (proDA)      HBS1L
190 O00443 Recurrence 0.032100   1.000  2.20 T-test (proDA)    PIK3C2A
191 Q9UIG8 Recurrence 0.032800   1.000 -2.20 T-test (proDA)    SLCO3A1
192 P84098 Recurrence 0.032900   1.000 -2.19 T-test (proDA)      RPL19
193 O95445 Recurrence 0.033000   1.000 -2.19 T-test (proDA)       APOM
194 Q9BTZ2 Recurrence 0.033000   1.000  2.19 T-test (proDA)      DHRS4
195 O00429 Recurrence 0.033900   1.000  2.18 T-test (proDA)      DNM1L
196 O00442 Recurrence 0.033900   1.000  2.18 T-test (proDA)       RTCA
197 O75832 Recurrence 0.033900   1.000 -2.18 T-test (proDA)     PSMD10
198 Q7Z434 Recurrence 0.033900   1.000  2.18 T-test (proDA)       MAVS
199 P36957 Recurrence 0.034000   1.000  2.18 T-test (proDA)       DLST
200 Q15386 Recurrence 0.034100   1.000  2.18 T-test (proDA)      UBE3C
201 O00182 Recurrence 0.034200   1.000  2.18 T-test (proDA)     LGALS9
202 Q9H3H3 Recurrence 0.034300   1.000  2.18 T-test (proDA)   C11orf68
203 P11182 Recurrence 0.034400   1.000  2.18 T-test (proDA)        DBT
204 Q9H1C4 Recurrence 0.034700   1.000  2.17 T-test (proDA)    UNC93B1
205 O60828 Recurrence 0.034900   1.000  2.17 T-test (proDA)      PQBP1
206 Q6UXH1 Recurrence 0.035400   1.000 -2.16 T-test (proDA)     CRELD2
207 Q9HCN8 Recurrence 0.035400   1.000 -2.16 T-test (proDA)     SDF2L1
208 P62266 Recurrence 0.035700   1.000 -2.16 T-test (proDA)      RPS23
209 Q9UIA9 Recurrence 0.036000   1.000  2.15 T-test (proDA)       XPO7
210 P62314 Recurrence 0.036300   1.000 -2.15 T-test (proDA)     SNRPD1
211 P41218 Recurrence 0.037200   1.000 -2.14 T-test (proDA)       MNDA
212 Q13425 Recurrence 0.037300   1.000 -2.14 T-test (proDA)      SNTB2
213 Q8TB36 Recurrence 0.037500   1.000  2.14 T-test (proDA)      GDAP1
214 P01033 Recurrence 0.037700   1.000  2.13 T-test (proDA)      TIMP1
215 P51784 Recurrence 0.037700   1.000  2.14 T-test (proDA)      USP11
216 Q8N755 Recurrence 0.038000   1.000  2.13 T-test (proDA)    SLC66A3
217 Q9NRN7 Recurrence 0.038200   1.000  2.13 T-test (proDA)   AASDHPPT
218 Q01196 Recurrence 0.038300   1.000  2.13 T-test (proDA)      RUNX1
219 Q6PI78 Recurrence 0.038300   1.000  2.13 T-test (proDA)     TMEM65
220 Q9BV38 Recurrence 0.038500   1.000  2.13 T-test (proDA)      WDR18
221 Q05397 Recurrence 0.039000   1.000  2.12 T-test (proDA)       PTK2
222 Q9Y3F4 Recurrence 0.039200   1.000  2.12 T-test (proDA)      STRAP
223 Q6NXG1 Recurrence 0.039700   1.000  2.11 T-test (proDA)      ESRP1
224 O76024 Recurrence 0.039800   1.000 -2.11 T-test (proDA)       WFS1
225 P48634 Recurrence 0.040100   1.000  2.11 T-test (proDA)     PRRC2A
226 O00423 Recurrence 0.040200   1.000 -2.11 T-test (proDA)       EML1
227 Q13287 Recurrence 0.040200   1.000  2.11 T-test (proDA)        NMI
228 Q9Y277 Recurrence 0.040200   1.000  2.11 T-test (proDA)      VDAC3
229 O43824 Recurrence 0.040300   1.000  2.11 T-test (proDA)     GTPBP6
230 Q6GMV2 Recurrence 0.040400   1.000 -2.10 T-test (proDA)      SMYD5
231 Q9P206 Recurrence 0.040400   1.000 -2.10 T-test (proDA)   KIAA1522
232 Q92973 Recurrence 0.040600   1.000  2.10 T-test (proDA)      TNPO1
233 P19174 Recurrence 0.040700   1.000 -2.10 T-test (proDA)      PLCG1
234 Q9H974 Recurrence 0.040700   1.000  2.10 T-test (proDA)      QTRT2
235 Q9UHI6 Recurrence 0.040900   1.000  2.10 T-test (proDA)      DDX20
236 P61225 Recurrence 0.041300   1.000  2.09 T-test (proDA)      RAP2B
237 Q8NCC3 Recurrence 0.041300   1.000  2.10 T-test (proDA)    PLA2G15
238 Q96JQ2 Recurrence 0.041300   1.000  2.09 T-test (proDA)       CLMN
239 O60832 Recurrence 0.041400   1.000  2.09 T-test (proDA)       DKC1
240 P51911 Recurrence 0.041700   1.000 -2.09 T-test (proDA)       CNN1
241 Q5TEC6 Recurrence 0.041700   1.000 -2.09 T-test (proDA)       H3-2
242 Q8N5M1 Recurrence 0.041700   1.000  2.09 T-test (proDA)     ATPAF2
243 Q9H444 Recurrence 0.042000   1.000  2.09 T-test (proDA)     CHMP4B
244 P51688 Recurrence 0.042300   1.000 -2.08 T-test (proDA)       SGSH
245 Q92796 Recurrence 0.043100   1.000 -2.08 T-test (proDA)       DLG3
246 Q96G97 Recurrence 0.043400   1.000 -2.07 T-test (proDA)      BSCL2
247 O60238 Recurrence 0.043500   1.000  2.07 T-test (proDA)     BNIP3L
248 P06127 Recurrence 0.043500   1.000 -2.07 T-test (proDA)        CD5
249 Q15417 Recurrence 0.043600   1.000 -2.07 T-test (proDA)       CNN3
250 Q6ICG6 Recurrence 0.043600   1.000  2.07 T-test (proDA)   KIAA0930
251 O75475 Recurrence 0.043800   1.000 -2.07 T-test (proDA)      PSIP1
252 Q9NQT8 Recurrence 0.043900   1.000  2.07 T-test (proDA)     KIF13B
253 Q9BY42 Recurrence 0.044000   1.000 -2.07 T-test (proDA)       RTF2
254 Q9ULH7 Recurrence 0.044600   1.000 -2.06 T-test (proDA)      MRTFB
255 P19367 Recurrence 0.044800   1.000  2.06 T-test (proDA)        HK1
256 P61916 Recurrence 0.044800   1.000 -2.06 T-test (proDA)       NPC2
257 O14617 Recurrence 0.045100   1.000  2.06 T-test (proDA)      AP3D1
258 P05161 Recurrence 0.045100   1.000  2.06 T-test (proDA)      ISG15
259 P02788 Recurrence 0.045400   1.000 -2.05 T-test (proDA)        LTF
260 Q86U38 Recurrence 0.045700   1.000 -2.05 T-test (proDA)       NOP9
261 P21399 Recurrence 0.045900   1.000 -2.05 T-test (proDA)       ACO1
262 P07602 Recurrence 0.046200   1.000  2.04 T-test (proDA)       PSAP
263 Q9BVL4 Recurrence 0.046300   1.000 -2.04 T-test (proDA)    SELENOO
264 P49908 Recurrence 0.046400   1.000 -2.04 T-test (proDA)    SELENOP
265 O43765 Recurrence 0.046800   1.000  2.04 T-test (proDA)       SGTA
266 Q9NX47 Recurrence 0.046900   1.000 -2.04 T-test (proDA)    MARCHF5
267 O43237 Recurrence 0.047200   1.000 -2.03 T-test (proDA)   DYNC1LI2
268 O75935 Recurrence 0.047200   1.000 -2.03 T-test (proDA)      DCTN3
269 Q12923 Recurrence 0.047400   1.000  2.03 T-test (proDA)     PTPN13
270 Q8N163 Recurrence 0.047400   1.000  2.03 T-test (proDA)      CCAR2
271 Q6P4A8 Recurrence 0.048100   1.000 -2.03 T-test (proDA)      PLBD1
272 Q96SB3 Recurrence 0.048100   1.000  2.03 T-test (proDA)    PPP1R9B
273 P10809 Recurrence 0.048400   1.000  2.02 T-test (proDA)      HSPD1
274 P20036 Recurrence 0.048900   1.000 -2.02 T-test (proDA)   HLA-DPA1
275 P20591 Recurrence 0.048900   1.000  2.02 T-test (proDA)        MX1
276 O75347 Recurrence 0.049000   1.000  2.02 T-test (proDA)       TBCA
277 P40925 Recurrence 0.049600   1.000 -2.01 T-test (proDA)       MDH1
278 Q96I59 Recurrence 0.049800   1.000  2.01 T-test (proDA)      NARS2
279 P20585 Recurrence 0.050000   1.000  2.01 T-test (proDA)       MSH3
280 Q13123 Recurrence 0.050000   1.000 -2.01 T-test (proDA)         IK
281 Q92633 Recurrence 0.050000   1.000 -2.01 T-test (proDA)      LPAR1</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proTumorRes_Krij$sig.feat.heat</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Tumor samples (ns: p
&gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Krij$data
smpAnnoTab &lt;- proTissueRes_Krij$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proTumorRes_Krij$sig.feat.tab
num_sigFeats &lt;- 6
gpLevel = c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;, &#39;Yes_Normal&#39;, &#39;No_Normal&#39;)
condiCol = c(&#39;grey50&#39;, &#39;firebrick&#39;) #c(Normal, Tumor)
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(newVar1 = paste0(Var1, &#39; (&#39;, Gene, &#39;)&#39;),
                newVar1 = factor(newVar1, levels = unique(newVar1)))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Recur_Condi = paste0(Recurrence, &#39;_&#39;, Condition),
                Recur_Condi = factor(Recur_Condi, levels = gpLevel))
ggplot(topSigFeatDat, aes(x=Recur_Condi, y=Abundance, col=Condition, fill=Recurrence)) +
  geom_boxplot(alpha = 1, outlier.shape = NA, linewidth = 1) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             comparisons = list(c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;), c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;)),
                             label = &#39;p.signif&#39;, tip.length = 0.015, bracket.size = 0.7, size = 4) +
  labs(x = &#39;Recurrence&#39;) +
  scale_color_manual(values = condiCol) +
  scale_fill_manual(values = c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(newVar1), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;),
             axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proTumorRes_Krij$pval.hist</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proTumorRes_Krij$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2   pVal pValAdj  Stat   Test
1   PC33 (1%) Recurrence 0.0168   0.363  2.46 T-test
2 PC11 (3.1%) Recurrence 0.0182   0.363 -2.43 T-test</code></pre>
<p>Visualize significant PCs and display their top features with
greatest absolute loadings</p>
<pre class="r"><code>proTumorRes_Krij$sig.pc.dist$PC33</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-47-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Krij$pc.top.feat.tab$PC33</code></pre>
<pre><code>   Feature Loading    Gene
1   Q12891 -0.1810   HYAL2
2   P30838  0.1140 ALDH3A1
3   Q99453  0.1140  PHOX2B
4   P02008 -0.1110     HBZ
5   P46439 -0.1110   GSTM5
6   P11684 -0.0932 SCGB1A1
7   Q9BYB0 -0.0907  SHANK3
8   P56856 -0.0895  CLDN18
9   Q13421  0.0879    MSLN
10  P05976  0.0849    MYL1
11  O14756 -0.0839 HSD17B6
12  P04271 -0.0804   S100B
13  Q8TDL5 -0.0791  BPIFB1
14  Q9Y342 -0.0783    PLLP
15  P29972 -0.0741    AQP1
16  Q5TBA9 -0.0731     FRY
17  O14578 -0.0726     CIT
18  Q92859 -0.0707    NEO1
19  O00767  0.0683     SCD
20  P13942 -0.0683 COL11A2
21  Q14642 -0.0673  INPP5A
22  P55064  0.0668    AQP5
23  Q92954  0.0660    PRG4
24  P04278  0.0647    SHBG
25  Q86TI0 -0.0646  TBC1D1
26  Q13938 -0.0633    CAPS
27  Q8TC71 -0.0625 SPATA18
28  Q9BW30 -0.0624   TPPP3
29  P07451 -0.0617     CA3
30  O94911  0.0609   ABCA8</code></pre>
<pre class="r"><code>proTumorRes_Krij$sig.pc.dist$PC11</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-47-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Krij$pc.top.feat.tab$PC11</code></pre>
<pre><code>      Feature Loading     Gene
1      P30838 -0.1680  ALDH3A1
2      Q9UK22 -0.1280    FBXO2
3      P11684 -0.1130  SCGB1A1
4  A0A0C4DH31  0.0970 IGHV1-18
5      P13929  0.0951     ENO3
6      P11686 -0.0900    SFTPC
7      P21695  0.0891     GPD1
8      P31327 -0.0842     CPS1
9      Q9NVM6  0.0802  DNAJC17
10     P56470  0.0793   LGALS4
11     O00626  0.0771    CCL22
12     Q8TC26 -0.0766  TMEM163
13     Q86VY4 -0.0764   TSPYL5
14     O94811 -0.0761     TPPP
15     Q5H8C1  0.0746    FREM1
16     Q99453 -0.0737   PHOX2B
17     O43927  0.0726   CXCL13
18     Q13075 -0.0718     NAIP
19     Q9Y6R1 -0.0715   SLC4A4
20     P58166  0.0712    INHBE
21     P13726 -0.0698       F3
22     Q96T58  0.0695     SPEN
23     O15056  0.0671    SYNJ2
24     P01743  0.0669 IGHV1-46
25     Q9NVV5 -0.0667     AIG1
26     P78524 -0.0654  DENND2B
27     Q8N135  0.0653     LGI4
28     Q9UPW5 -0.0643  AGTPBP1
29     O43310 -0.0635     CTIF
30     Q9UHP3  0.0630    USP25</code></pre>
</div>
<div id="normal-samples-1" class="section level2">
<h2>Normal samples</h2>
<p>Display significant associations between SVs and sample metadata
variables. SVs (surrogate variables) will be included in linear model
for differential analysis</p>
<pre class="r"><code># Subset certain samples
proNormal_Krij &lt;- proTissue_Krij[, which(colData(proTissue_Krij)$Condition == &#39;Normal&#39;)]

# Do SVA to identify latent factors that explain unknown variance
# Prepare data matrix and sample metadata table for creating model matrices
datMat &lt;- assay(proNormal_Krij)
# Keep only complete features
keptFeats &lt;- apply(datMat, 1, function(featVec) {
  !any(is.na(featVec))
})
datMat &lt;- datMat[keptFeats,]
metadatTab &lt;- colData(proNormal_Krij) %&gt;%
  tibble::as_tibble(rownames = &#39;Sample&#39;)
# Create full model matrix including both adjustment variables and variables of interest
modelMat &lt;- model.matrix(~ Recurrence, data = metadatTab)
# Create null model matrix including only adjustment variables
modelMat0 &lt;- model.matrix(~ 1, data = metadatTab)

# Perform SVA
# Identify number of latent factors that need to be estimated
numSV = sva::num.sv(datMat, modelMat, method = &#39;be&#39;)
svObj = sva(datMat, modelMat, modelMat0, n.sv = numSV)</code></pre>
<pre><code>Number of significant surrogate variables is:  8 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code># Do association tests between SVs and patient metadata
# Prepare surrogate variable table
svTab &lt;- svObj$sv
rownames(svTab) &lt;- metadatTab$Sample
colnames(svTab) &lt;- paste0(&#39;SV&#39;, seq_len(ncol(svTab)))
svTab &lt;- tibble::as_tibble(svTab, rownames = &#39;Sample&#39;)
# Prepare patient metadata table
metaTab &lt;- dplyr::select(metadatTab, Sample, Recurrence, Gender, Age, Smoking, Stage,
                         Adjuvant)
# Do association tests
testAsso(svTab, metaTab, cmn_col = &#39;Sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05)</code></pre>
<pre><code>  Var1    Var2        pVal    pValAdj      Stat        Test
1  SV1 Smoking 0.001295091 0.06216437 7.4436271       ANOVA
2  SV8   Stage 0.003625945 0.08702267 6.1804431       ANOVA
3  SV7     Age 0.005514949 0.08823919 0.3457265 Correlation
4  SV7   Stage 0.013370455 0.16044546 4.6404120       ANOVA
5  SV3  Gender 0.020034795 0.19233403 2.3883472      T-test
6  SV3   Stage 0.027425489 0.21940391 3.8207143       ANOVA</code></pre>
<pre class="r"><code># Include surrogate variables into SE object
proNormalTab &lt;- summExp2df(proNormal_Krij, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;)
proNormalTab &lt;- dplyr::left_join(proNormalTab, svTab, by = &#39;Sample&#39;)
proNormal_Krij &lt;- df2SummExp(proNormalTab, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;,
                             values = &#39;Value&#39;, row_anno = colnames(rowData(proNormal_Krij)),
                             col_anno = c(colnames(colData(proNormal_Krij)),
                                          paste0(&#39;SV&#39;, seq_len(ncol(svTab)-1))))
assay(proNormal_Krij) &lt;- as.matrix(assay(proNormal_Krij))</code></pre>
<pre class="r"><code># Perform single-omics analysis
# proNormalRes_Krij &lt;- doSingleOmicsAnalysis(proNormal_Krij, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                            pca_method = &#39;ppca&#39;, num_PCs = 20, num_PCfeats = 30,
#                                            use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                            feat_conv = T, plot_title = NULL, show_rownames = F,
#                                            pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                            &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;),
#                                            soa_unwantVar = paste0(&#39;SV&#39;, seq_len(8)))
proNormalRes_Krij &lt;- readRDS(&#39;./data/Discovery/potential_proteins/Combined_proNormalRes_Krij_20PCs.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proNormalRes_Krij$data.var.source</code></pre>
<pre><code>         Var1       Var2    pVal pValAdj   Stat        Test
1 PC18 (1.4%)      Stage 0.00131  0.0925  7.430       ANOVA
2  PC6 (6.6%)     Gender 0.00177  0.0925 -3.270      T-test
3 PC12 (2.6%)        Age 0.00231  0.0925  0.377 Correlation
4  PC8 (4.2%) Recurrence 0.01190  0.3560  2.590      T-test
5 PC15 (1.8%)        Age 0.02270  0.5070  0.287 Correlation
6  PC5 (6.6%)     Gender 0.02530  0.5070  2.290      T-test
7 PC10 (3.3%)   Adjuvant 0.03730  0.5570  2.130      T-test
8  PC4 (8.1%)     Gender 0.04380  0.5570 -2.060      T-test
9  PC6 (6.6%)    Smoking 0.04860  0.5570  3.180       ANOVA</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>dplyr::select(proNormalRes_Krij$sig.feat.tab, -Test)</code></pre>
<pre><code>                    Var1       Var2     pVal pValAdj  Stat
1                 Q9UH17 Recurrence 1.84e-06  0.0113 -5.36
2                 O14548 Recurrence 8.25e-05  0.1660 -4.27
3                 Q86YB8 Recurrence 1.06e-04  0.1660  4.19
4                 P18887 Recurrence 1.08e-04  0.1660  4.18
5                 Q02040 Recurrence 2.09e-04  0.2560 -3.98
6                 P56589 Recurrence 3.08e-04  0.2880 -3.86
7                 Q562R1 Recurrence 3.65e-04  0.2880  3.81
8                 P63241 Recurrence 3.76e-04  0.2880 -3.80
9                 O60869 Recurrence 4.54e-04  0.3090  3.74
10                Q9NUQ8 Recurrence 5.24e-04  0.3210  3.69
11                Q9Y2Q9 Recurrence 6.47e-04  0.3320 -3.63
12                Q96IZ7 Recurrence 6.50e-04  0.3320 -3.62
13                Q9NX05 Recurrence 1.03e-03  0.4840 -3.47
14                Q9NZI7 Recurrence 1.16e-03  0.4950  3.43
15                P09429 Recurrence 1.21e-03  0.4950  3.42
16                Q7L4I2 Recurrence 1.55e-03  0.5420 -3.34
17                Q9UK53 Recurrence 1.87e-03  0.5420 -3.27
18                Q6ZVM7 Recurrence 1.95e-03  0.5420 -3.26
19                P48059 Recurrence 2.05e-03  0.5420 -3.24
20                P49184 Recurrence 2.06e-03  0.5420  3.24
21                Q12872 Recurrence 2.16e-03  0.5420 -3.22
22                Q01658 Recurrence 2.33e-03  0.5420 -3.20
23                O75173 Recurrence 2.37e-03  0.5420  3.19
24                Q14847 Recurrence 2.46e-03  0.5420  3.18
25                P31949 Recurrence 2.60e-03  0.5420 -3.16
26                P09651 Recurrence 2.76e-03  0.5420  3.14
27                P47895 Recurrence 2.79e-03  0.5420  3.14
28                P04040 Recurrence 2.90e-03  0.5420  3.12
29                Q15811 Recurrence 2.91e-03  0.5420  3.12
30                Q99542 Recurrence 2.92e-03  0.5420  3.12
31                Q96N67 Recurrence 2.94e-03  0.5420  3.12
32                Q9BUJ2 Recurrence 3.21e-03  0.5420  3.09
33                O43237 Recurrence 3.59e-03  0.5420 -3.05
34                O60264 Recurrence 3.59e-03  0.5420  3.05
35                P82930 Recurrence 3.64e-03  0.5420 -3.04
36                O00192 Recurrence 3.71e-03  0.5420  3.04
37                O00499 Recurrence 3.72e-03  0.5420  3.04
38                P37235 Recurrence 3.81e-03  0.5420 -3.03
39                P23921 Recurrence 3.83e-03  0.5420 -3.02
40                P46782 Recurrence 3.83e-03  0.5420 -3.02
41                Q9BXJ9 Recurrence 3.99e-03  0.5420 -3.01
42                Q14011 Recurrence 4.06e-03  0.5420  3.00
43                Q6ZWT7 Recurrence 4.10e-03  0.5420 -3.00
44                Q14938 Recurrence 4.13e-03  0.5420  3.00
45                Q9Y2R0 Recurrence 4.15e-03  0.5420 -3.00
46                P30048 Recurrence 4.16e-03  0.5420 -3.00
47                O43760 Recurrence 4.34e-03  0.5420 -2.98
48                P48634 Recurrence 4.77e-03  0.5420  2.95
49                O43759 Recurrence 4.88e-03  0.5420  2.94
50                Q9UNE7 Recurrence 4.88e-03  0.5420 -2.94
51                O43708 Recurrence 4.89e-03  0.5420  2.94
52                P78559 Recurrence 4.92e-03  0.5420 -2.94
53                P01599 Recurrence 5.16e-03  0.5420 -2.92
54                O43427 Recurrence 5.28e-03  0.5420  2.91
55                Q8TBQ9 Recurrence 5.29e-03  0.5420 -2.91
56                Q969G3 Recurrence 5.29e-03  0.5420  2.91
57                O75695 Recurrence 5.38e-03  0.5420 -2.90
58                Q9BUT1 Recurrence 5.39e-03  0.5420  2.90
59                P16615 Recurrence 5.65e-03  0.5420 -2.89
60                Q8WYP5 Recurrence 5.67e-03  0.5420  2.88
61                Q9ULA0 Recurrence 5.70e-03  0.5420  2.88
62                Q9Y2I1 Recurrence 5.88e-03  0.5420  2.87
63                O43182 Recurrence 6.05e-03  0.5420  2.86
64                P37108 Recurrence 6.08e-03  0.5420  2.86
65                P13798 Recurrence 6.09e-03  0.5420  2.86
66                P49773 Recurrence 6.25e-03  0.5420 -2.85
67                Q96LD4 Recurrence 6.26e-03  0.5420  2.85
68                O60828 Recurrence 6.46e-03  0.5420  2.84
69                P67809 Recurrence 6.46e-03  0.5420  2.84
70                P28702 Recurrence 6.50e-03  0.5420  2.83
71                Q92759 Recurrence 6.52e-03  0.5420  2.83
72                Q9H9J2 Recurrence 6.53e-03  0.5420 -2.83
73                Q5SRE7 Recurrence 6.68e-03  0.5420  2.82
74                O43772 Recurrence 6.72e-03  0.5420 -2.82
75                Q9Y4K4 Recurrence 6.72e-03  0.5420 -2.82
76                Q9Y305 Recurrence 6.75e-03  0.5420 -2.82
77                Q04917 Recurrence 6.82e-03  0.5420 -2.82
78                Q0VDG4 Recurrence 6.90e-03  0.5420  2.81
79                Q96FW1 Recurrence 7.03e-03  0.5450  2.80
80                Q96B23 Recurrence 7.16e-03  0.5480 -2.80
81                Q9BY43 Recurrence 7.26e-03  0.5490  2.79
82                P10599 Recurrence 7.50e-03  0.5540 -2.78
83                P50213 Recurrence 7.52e-03  0.5540 -2.78
84                Q14314 Recurrence 7.64e-03  0.5540 -2.77
85                P05114 Recurrence 7.68e-03  0.5540  2.77
86                Q9UHB6 Recurrence 7.88e-03  0.5570  2.76
87                Q8N5K1 Recurrence 7.99e-03  0.5570 -2.76
88                O14933 Recurrence 8.17e-03  0.5570  2.75
89                Q8NC51 Recurrence 8.20e-03  0.5570  2.75
90                O95167 Recurrence 8.24e-03  0.5570  2.75
91                Q03701 Recurrence 8.28e-03  0.5570  2.74
92                P49006 Recurrence 8.69e-03  0.5590  2.72
93                P53985 Recurrence 8.89e-03  0.5590  2.72
94                Q96BP3 Recurrence 8.90e-03  0.5590  2.72
95                Q86TX2 Recurrence 9.10e-03  0.5590 -2.71
96                Q96JK2 Recurrence 9.11e-03  0.5590 -2.71
97                Q9H488 Recurrence 9.11e-03  0.5590  2.71
98                Q96AP7 Recurrence 9.30e-03  0.5590  2.70
99                Q96FJ2 Recurrence 9.43e-03  0.5590 -2.69
100               O94811 Recurrence 9.46e-03  0.5590 -2.69
101               P11274 Recurrence 9.48e-03  0.5590  2.69
102               Q8IYS1 Recurrence 9.50e-03  0.5590 -2.69
103               Q9BV38 Recurrence 9.67e-03  0.5590  2.68
104               Q32P28 Recurrence 1.00e-02  0.5590  2.67
105               Q9P2Y5 Recurrence 1.00e-02  0.5590 -2.67
106               O43924 Recurrence 1.01e-02  0.5590  2.67
107               O15379 Recurrence 1.02e-02  0.5590 -2.67
108               Q03188 Recurrence 1.02e-02  0.5590  2.66
109               Q9BZE9 Recurrence 1.02e-02  0.5590  2.66
110               O60333 Recurrence 1.03e-02  0.5590 -2.66
111               O76094 Recurrence 1.03e-02  0.5590  2.66
112               Q9NPD3 Recurrence 1.03e-02  0.5590  2.66
113               Q9Y5L4 Recurrence 1.03e-02  0.5590 -2.66
114               P61923 Recurrence 1.05e-02  0.5590  2.65
115               Q15642 Recurrence 1.05e-02  0.5590  2.65
116               Q9H6K4 Recurrence 1.06e-02  0.5590 -2.65
117               P12236 Recurrence 1.08e-02  0.5650 -2.64
118               Q9UJU6 Recurrence 1.11e-02  0.5740  2.63
119               Q96CT7 Recurrence 1.13e-02  0.5740  2.62
120               P29317 Recurrence 1.14e-02  0.5740 -2.62
121               P61160 Recurrence 1.14e-02  0.5740 -2.62
122               P82979 Recurrence 1.14e-02  0.5740  2.62
123               O76003 Recurrence 1.17e-02  0.5800 -2.61
124               P12955 Recurrence 1.17e-02  0.5800  2.61
125               Q86UK7 Recurrence 1.21e-02  0.5940  2.60
126               P53384 Recurrence 1.22e-02  0.5950 -2.59
127               P11177 Recurrence 1.26e-02  0.6100 -2.58
128               P23526 Recurrence 1.29e-02  0.6110  2.57
129               P51687 Recurrence 1.29e-02  0.6110  2.57
130               Q86YL7 Recurrence 1.32e-02  0.6210 -2.57
131               Q01433 Recurrence 1.34e-02  0.6240  2.56
132               P43405 Recurrence 1.35e-02  0.6240  2.55
133               Q00325 Recurrence 1.36e-02  0.6240 -2.55
134               P27658 Recurrence 1.37e-02  0.6240  2.55
135               Q6P3W7 Recurrence 1.38e-02  0.6240  2.55
136               Q6NWY9 Recurrence 1.39e-02  0.6240 -2.55
137               Q13442 Recurrence 1.41e-02  0.6280  2.54
138               Q86U44 Recurrence 1.42e-02  0.6280  2.54
139               Q4L180 Recurrence 1.44e-02  0.6280 -2.53
140               Q96HY7 Recurrence 1.44e-02  0.6280  2.53
141               Q9UBS8 Recurrence 1.45e-02  0.6280 -2.53
142               Q99417 Recurrence 1.46e-02  0.6290 -2.53
143               P12268 Recurrence 1.47e-02  0.6310  2.52
144               O00635 Recurrence 1.48e-02  0.6310 -2.52
145               P17706 Recurrence 1.51e-02  0.6380 -2.51
146               P01042 Recurrence 1.54e-02  0.6440  2.50
147               P39656 Recurrence 1.61e-02  0.6440  2.49
148               Q9BRT2 Recurrence 1.61e-02  0.6440 -2.49
149               Q9UHD2 Recurrence 1.61e-02  0.6440  2.49
150               Q9Y287 Recurrence 1.61e-02  0.6440  2.49
151               P24043 Recurrence 1.62e-02  0.6440  2.48
152               Q5TA45 Recurrence 1.63e-02  0.6440  2.48
153               Q9NP97 Recurrence 1.64e-02  0.6440 -2.48
154               P49356 Recurrence 1.65e-02  0.6440  2.48
155               Q8N139 Recurrence 1.65e-02  0.6440  2.48
156               Q8WXA3 Recurrence 1.65e-02  0.6440  2.48
157               Q96RF0 Recurrence 1.67e-02  0.6440  2.47
158               Q14152 Recurrence 1.68e-02  0.6440  2.47
159               P40429 Recurrence 1.69e-02  0.6440 -2.47
160               Q8IVM0 Recurrence 1.69e-02  0.6440  2.47
161               P41732 Recurrence 1.70e-02  0.6440 -2.47
162               Q9Y2Q0 Recurrence 1.71e-02  0.6440  2.46
163               Q15024 Recurrence 1.74e-02  0.6440  2.45
164               O60784 Recurrence 1.75e-02  0.6440  2.45
165               Q92888 Recurrence 1.75e-02  0.6440  2.45
166               Q96FZ2 Recurrence 1.75e-02  0.6440 -2.45
167               P08559 Recurrence 1.76e-02  0.6440 -2.45
168               Q8N4Q0 Recurrence 1.78e-02  0.6490  2.45
169               P24534 Recurrence 1.79e-02  0.6500 -2.44
170               O94973 Recurrence 1.85e-02  0.6610  2.43
171               O43819 Recurrence 1.87e-02  0.6610  2.43
172               P62750 Recurrence 1.87e-02  0.6610  2.43
173               Q9UKX2 Recurrence 1.87e-02  0.6610 -2.43
174               P60981 Recurrence 1.88e-02  0.6610 -2.42
175               Q8IZ07 Recurrence 1.89e-02  0.6610  2.42
176               Q5VW36 Recurrence 1.91e-02  0.6640  2.42
177               P61956 Recurrence 1.93e-02  0.6670  2.41
178               O14976 Recurrence 1.95e-02  0.6700  2.41
179               P09543 Recurrence 1.98e-02  0.6700  2.40
180               P30626 Recurrence 1.98e-02  0.6700 -2.40
181               Q9BX97 Recurrence 1.99e-02  0.6700  2.40
182               Q5TAQ9 Recurrence 2.00e-02  0.6700  2.40
183               Q92900 Recurrence 2.02e-02  0.6700  2.39
184               Q8WUF8 Recurrence 2.03e-02  0.6700  2.39
185               Q9NUU6 Recurrence 2.03e-02  0.6700 -2.39
186               Q96SB4 Recurrence 2.04e-02  0.6700 -2.39
187               Q15843 Recurrence 2.06e-02  0.6700 -2.39
188               Q96I15 Recurrence 2.08e-02  0.6700  2.38
189               P19784 Recurrence 2.10e-02  0.6700  2.38
190               P33527 Recurrence 2.10e-02  0.6700  2.38
191               Q96J84 Recurrence 2.11e-02  0.6700  2.38
192               O94804 Recurrence 2.13e-02  0.6700  2.37
193               P23458 Recurrence 2.14e-02  0.6700  2.37
194               P35606 Recurrence 2.16e-02  0.6700  2.37
195               Q9NW08 Recurrence 2.16e-02  0.6700  2.37
196               Q9NQ66 Recurrence 2.17e-02  0.6700  2.36
197               Q14156 Recurrence 2.18e-02  0.6700  2.36
198               Q7Z7K6 Recurrence 2.18e-02  0.6700  2.36
199               Q9BWF3 Recurrence 2.19e-02  0.6700  2.36
200               Q9NVA2 Recurrence 2.19e-02  0.6700  2.36
201               P61916 Recurrence 2.22e-02  0.6750 -2.36
202               P10809 Recurrence 2.24e-02  0.6760 -2.35
203               P51784 Recurrence 2.24e-02  0.6760  2.35
204               Q13724 Recurrence 2.28e-02  0.6850  2.34
205               Q9UII2 Recurrence 2.30e-02  0.6860  2.34
206               O75054 Recurrence 2.34e-02  0.6920 -2.33
207               P38919 Recurrence 2.34e-02  0.6920  2.33
208               Q53FZ2 Recurrence 2.36e-02  0.6920  2.33
209               Q8TDQ7 Recurrence 2.36e-02  0.6920  2.33
210               O15439 Recurrence 2.38e-02  0.6920  2.33
211               Q14112 Recurrence 2.38e-02  0.6920  2.33
212               Q9NXE8 Recurrence 2.44e-02  0.7030  2.32
213               Q7KZF4 Recurrence 2.48e-02  0.7030  2.31
214               Q7Z4W1 Recurrence 2.48e-02  0.7030  2.31
215               P55769 Recurrence 2.51e-02  0.7030 -2.31
216               Q9BUR5 Recurrence 2.55e-02  0.7030 -2.30
217               Q9Y3F4 Recurrence 2.55e-02  0.7030  2.30
218               Q8N4P3 Recurrence 2.56e-02  0.7030  2.30
219               Q9NVT9 Recurrence 2.59e-02  0.7030 -2.29
220               A0PJZ3 Recurrence 2.61e-02  0.7030  2.29
221               Q9Y315 Recurrence 2.61e-02  0.7030  2.29
222               Q9GZT3 Recurrence 2.66e-02  0.7030 -2.28
223               Q5T653 Recurrence 2.67e-02  0.7030 -2.28
224               Q9UHD8 Recurrence 2.67e-02  0.7030  2.28
225               P17252 Recurrence 2.68e-02  0.7030  2.28
226               P26358 Recurrence 2.68e-02  0.7030  2.28
227               P60866 Recurrence 2.68e-02  0.7030  2.28
228               Q15061 Recurrence 2.69e-02  0.7030  2.28
229               Q14676 Recurrence 2.70e-02  0.7030  2.28
230               P09619 Recurrence 2.71e-02  0.7030  2.27
231               P28066 Recurrence 2.71e-02  0.7030  2.27
232               P05386 Recurrence 2.72e-02  0.7030 -2.27
233               P23142 Recurrence 2.72e-02  0.7030  2.27
234               P49755 Recurrence 2.72e-02  0.7030  2.27
235               P63173 Recurrence 2.72e-02  0.7030  2.27
236               Q96PD5 Recurrence 2.72e-02  0.7030  2.27
237               O14880 Recurrence 2.73e-02  0.7030 -2.27
238               P17936 Recurrence 2.73e-02  0.7030  2.27
239               P51911 Recurrence 2.80e-02  0.7150 -2.26
240               Q96AA3 Recurrence 2.80e-02  0.7150 -2.26
241               P42226 Recurrence 2.82e-02  0.7150  2.26
242               Q86TM6 Recurrence 2.83e-02  0.7150 -2.25
243               P26038 Recurrence 2.85e-02  0.7150  2.25
244               Q9H444 Recurrence 2.85e-02  0.7150  2.25
245               Q00688 Recurrence 2.88e-02  0.7170  2.25
246               Q9NWW5 Recurrence 2.88e-02  0.7170  2.25
247               P49406 Recurrence 2.91e-02  0.7190  2.24
248               Q96JM3 Recurrence 2.91e-02  0.7190 -2.24
249               Q8IZP0 Recurrence 2.94e-02  0.7220  2.24
250               Q7Z4Q2 Recurrence 2.95e-02  0.7230  2.24
251               Q5JU69 Recurrence 2.96e-02  0.7230 -2.24
252               P30536 Recurrence 2.97e-02  0.7230 -2.23
253               P63010 Recurrence 3.00e-02  0.7240  2.23
254               Q96NC0 Recurrence 3.00e-02  0.7240  2.23
255               Q6PEZ8 Recurrence 3.02e-02  0.7240 -2.23
256               P51991 Recurrence 3.03e-02  0.7240  2.23
257               Q6BCY4 Recurrence 3.07e-02  0.7240  2.22
258               P68402 Recurrence 3.08e-02  0.7240 -2.22
259        Q5T1J5;Q9Y6H1 Recurrence 3.09e-02  0.7240 -2.22
260               P42704 Recurrence 3.10e-02  0.7240 -2.22
261               P37802 Recurrence 3.11e-02  0.7240 -2.22
262               P52735 Recurrence 3.11e-02  0.7240  2.21
263               Q7L0Y3 Recurrence 3.11e-02  0.7240 -2.21
264               Q13683 Recurrence 3.12e-02  0.7240 -2.21
265               Q9H6Z4 Recurrence 3.16e-02  0.7290  2.21
266               P33992 Recurrence 3.17e-02  0.7290  2.21
267               O75448 Recurrence 3.19e-02  0.7290  2.20
268               Q9UBG0 Recurrence 3.21e-02  0.7290  2.20
269               P06748 Recurrence 3.24e-02  0.7290  2.20
270               P21912 Recurrence 3.24e-02  0.7290 -2.20
271               Q96KR1 Recurrence 3.24e-02  0.7290  2.20
272               Q9P107 Recurrence 3.24e-02  0.7290  2.20
273               Q9Y5B0 Recurrence 3.29e-02  0.7360  2.19
274               P48147 Recurrence 3.30e-02  0.7360  2.19
275               P50148 Recurrence 3.30e-02  0.7360  2.19
276               O43752 Recurrence 3.33e-02  0.7360  2.18
277               Q9Y6C9 Recurrence 3.35e-02  0.7360 -2.18
278               Q7Z417 Recurrence 3.37e-02  0.7360 -2.18
279               O95563 Recurrence 3.38e-02  0.7360 -2.18
280               P48595 Recurrence 3.38e-02  0.7360 -2.18
281               P52434 Recurrence 3.40e-02  0.7360  2.18
282               Q86YN1 Recurrence 3.40e-02  0.7360 -2.18
283               Q14767 Recurrence 3.42e-02  0.7360  2.17
284               P78347 Recurrence 3.47e-02  0.7360  2.17
285               Q6GMV3 Recurrence 3.47e-02  0.7360 -2.17
286               Q15056 Recurrence 3.49e-02  0.7360 -2.17
287               P06858 Recurrence 3.50e-02  0.7360  2.16
288               Q86V81 Recurrence 3.52e-02  0.7360  2.16
289               Q9Y2L1 Recurrence 3.52e-02  0.7360  2.16
290               O94905 Recurrence 3.54e-02  0.7360 -2.16
291               Q9H1Z4 Recurrence 3.54e-02  0.7360  2.16
292               P00491 Recurrence 3.58e-02  0.7360  2.15
293               P60604 Recurrence 3.58e-02  0.7360 -2.15
294               Q5JSH3 Recurrence 3.58e-02  0.7360  2.15
295               P24941 Recurrence 3.60e-02  0.7360  2.15
296               Q02978 Recurrence 3.60e-02  0.7360 -2.15
297               P49959 Recurrence 3.62e-02  0.7360  2.15
298               P30044 Recurrence 3.66e-02  0.7360 -2.15
299               P24844 Recurrence 3.67e-02  0.7360 -2.14
300               O15460 Recurrence 3.69e-02  0.7360  2.14
301               P50151 Recurrence 3.69e-02  0.7360 -2.14
302               Q14554 Recurrence 3.69e-02  0.7360  2.14
303               Q6PI48 Recurrence 3.69e-02  0.7360 -2.14
304               Q9H7P6 Recurrence 3.69e-02  0.7360 -2.14
305               Q9Y2S6 Recurrence 3.69e-02  0.7360  2.14
306               P55287 Recurrence 3.72e-02  0.7360  2.14
307               Q15629 Recurrence 3.74e-02  0.7360 -2.14
308               P40938 Recurrence 3.75e-02  0.7360  2.13
309               P55789 Recurrence 3.77e-02  0.7360 -2.13
310               P22059 Recurrence 3.78e-02  0.7360  2.13
311               P49459 Recurrence 3.78e-02  0.7360 -2.13
312               P27144 Recurrence 3.81e-02  0.7360 -2.13
313               Q01650 Recurrence 3.81e-02  0.7360 -2.13
314               Q9UKY7 Recurrence 3.81e-02  0.7360  2.13
315               Q7Z392 Recurrence 3.82e-02  0.7360  2.13
316               Q8NHP8 Recurrence 3.82e-02  0.7360 -2.13
317               Q13347 Recurrence 3.83e-02  0.7360  2.12
318               O94979 Recurrence 3.84e-02  0.7360  2.12
319               Q15404 Recurrence 3.84e-02  0.7360 -2.12
320               Q96FS4 Recurrence 3.84e-02  0.7360  2.12
321               P21291 Recurrence 3.86e-02  0.7360  2.12
322               O95772 Recurrence 3.89e-02  0.7390  2.12
323               P33993 Recurrence 3.91e-02  0.7390 -2.12
324               P55263 Recurrence 3.91e-02  0.7390  2.12
325               Q53SF7 Recurrence 3.94e-02  0.7390  2.11
326               Q9NVZ3 Recurrence 3.94e-02  0.7390  2.11
327               Q15654 Recurrence 3.96e-02  0.7390  2.11
328               Q5JVF3 Recurrence 3.97e-02  0.7390  2.11
329               P46926 Recurrence 3.98e-02  0.7390  2.11
330               Q8IZ83 Recurrence 3.98e-02  0.7390  2.11
331               P09913 Recurrence 4.00e-02  0.7390 -2.11
332               Q07817 Recurrence 4.01e-02  0.7390  2.10
333               Q8N2S1 Recurrence 4.03e-02  0.7410  2.10
334           A0A0C4DH43 Recurrence 4.07e-02  0.7410 -2.10
335               P30566 Recurrence 4.07e-02  0.7410  2.10
336               O95136 Recurrence 4.11e-02  0.7410  2.09
337 P52435;Q9GZM3;Q9H1A7 Recurrence 4.11e-02  0.7410 -2.09
338               P61247 Recurrence 4.11e-02  0.7410  2.09
339               P42771 Recurrence 4.12e-02  0.7410  2.09
340               Q13263 Recurrence 4.12e-02  0.7410  2.09
341               Q9UPU7 Recurrence 4.13e-02  0.7410  2.09
342        O60739;P41567 Recurrence 4.15e-02  0.7440 -2.09
343               A4D1P6 Recurrence 4.17e-02  0.7440  2.09
344               Q9HCG8 Recurrence 4.19e-02  0.7440 -2.09
345               Q15651 Recurrence 4.22e-02  0.7440  2.08
346               Q9UI10 Recurrence 4.24e-02  0.7440  2.08
347               P62857 Recurrence 4.25e-02  0.7440  2.08
348               Q8TF01 Recurrence 4.25e-02  0.7440 -2.08
349               P42229 Recurrence 4.26e-02  0.7440 -2.08
350               Q96PU8 Recurrence 4.26e-02  0.7440 -2.08
351               Q8N335 Recurrence 4.27e-02  0.7440 -2.08
352               Q86VN1 Recurrence 4.28e-02  0.7440  2.08
353               Q5T9L3 Recurrence 4.29e-02  0.7440  2.07
354               P50336 Recurrence 4.30e-02  0.7440 -2.07
355               Q8TEA8 Recurrence 4.31e-02  0.7440 -2.07
356               Q5T2E6 Recurrence 4.35e-02  0.7480  2.07
357               P19623 Recurrence 4.36e-02  0.7480  2.07
358               Q92508 Recurrence 4.39e-02  0.7490 -2.06
359               Q9NRY5 Recurrence 4.39e-02  0.7490  2.06
360               Q9ULQ1 Recurrence 4.41e-02  0.7490 -2.06
361               Q8IWT0 Recurrence 4.42e-02  0.7490  2.06
362               P62937 Recurrence 4.43e-02  0.7490 -2.06
363               P42126 Recurrence 4.50e-02  0.7530  2.05
364               Q5JRX3 Recurrence 4.51e-02  0.7530  2.05
365               O00584 Recurrence 4.52e-02  0.7530 -2.05
366               P54577 Recurrence 4.52e-02  0.7530  2.05
367               Q96C11 Recurrence 4.52e-02  0.7530  2.05
368               Q96H20 Recurrence 4.52e-02  0.7530 -2.05
369               Q9NZC3 Recurrence 4.55e-02  0.7540  2.05
370               P08670 Recurrence 4.56e-02  0.7540  2.05
371               Q5W0V3 Recurrence 4.57e-02  0.7540  2.05
372               Q9Y676 Recurrence 4.61e-02  0.7580 -2.04
373               Q8WWY3 Recurrence 4.62e-02  0.7580  2.04
374               Q15542 Recurrence 4.65e-02  0.7590  2.04
375               O75907 Recurrence 4.66e-02  0.7590  2.04
376               P42677 Recurrence 4.66e-02  0.7590  2.04
377               Q9P2E3 Recurrence 4.68e-02  0.7600  2.04
378               O60613 Recurrence 4.73e-02  0.7610 -2.03
379               Q14197 Recurrence 4.73e-02  0.7610  2.03
380               Q8N755 Recurrence 4.73e-02  0.7610  2.03
381               P17096 Recurrence 4.74e-02  0.7610  2.03
382               Q92572 Recurrence 4.74e-02  0.7610  2.03
383               Q05513 Recurrence 4.76e-02  0.7620 -2.03
384               P07947 Recurrence 4.78e-02  0.7620  2.03
385               Q8N1F7 Recurrence 4.79e-02  0.7620  2.02
386               Q6EMK4 Recurrence 4.80e-02  0.7620  2.02
387               Q99798 Recurrence 4.81e-02  0.7620 -2.02
388               Q9UJ68 Recurrence 4.84e-02  0.7640  2.02
389               Q12894 Recurrence 4.86e-02  0.7650  2.02
390               P23634 Recurrence 4.88e-02  0.7650 -2.02
391               Q9UK45 Recurrence 4.88e-02  0.7650 -2.02
392               P26641 Recurrence 4.92e-02  0.7680  2.01
393               P61088 Recurrence 4.93e-02  0.7680 -2.01
394               O00461 Recurrence 4.94e-02  0.7680  2.01
395               P46734 Recurrence 4.96e-02  0.7690  2.01
                      Gene
1                 APOBEC3B
2                  COX7A2L
3                    ERO1B
4                    XRCC1
5                  AKAP17A
6                     PEX3
7                   ACTBL2
8                    EIF5A
9                     EDF1
10                   ABCF3
11                  MRPS28
12                   RSRC1
13                 FAM120C
14                    UBP1
15                   HMGB1
16                   RSRC2
17                    ING1
18                  TOM1L2
19                   LIMS1
20                DNASE1L1
21                  SFSWAP
22                     DR1
23                 ADAMTS4
24                   LASP1
25                 S100A11
26                 HNRNPA1
27                 ALDH1A3
28                     CAT
29                   ITSN1
30                   MMP19
31                   DOCK7
32                HNRNPUL1
33                DYNC1LI2
34                 SMARCA5
35                  MRPS34
36                   ARVCF
37                    BIN1
38                  HPCAL1
39                    RRM1
40                    RPS5
41                   NAA15
42                   CIRBP
43                  MBOAT2
44                    NFIX
45                    COA3
46                   PRDX3
47                  SYNGR2
48                  PRRC2A
49                  SYNGR1
50                   STUB1
51                   GSTZ1
52                   MAP1A
53                IGKV1-17
54                    FIBP
55                TMEM167A
56                 SMARCE1
57                     RP2
58                    BDH2
59                  ATP2A2
60                  AHCTF1
61                   DNPEP
62                   NISCH
63                 ARHGAP6
64                   SRP14
65                    APEH
66                   HINT1
67                  TRIM47
68                   PQBP1
69                    YBX1
70                    RXRB
71                  GTF2H4
72                  MRPL44
73                  PHYHD1
74                SLC25A20
75                  MAP4K5
76                   ACOT9
77                   YWHAH
78                   SCRN3
79                   OTUB1
80                C18orf25
81                  CHMP4A
82                     TXN
83                   IDH3A
84                    FGL2
85                   HMGN1
86                   LIMA1
87                   CISD2
88                  UBE2L6
89                  SERBP1
90                  NDUFA3
91                   CEBPZ
92                MARCKSL1
93                 SLC16A1
94                   PPWD1
95                   ACOT1
96                   DCAF5
97                  POFUT1
98                    ESAM
99                  DYNLL2
100                   TPPP
101                    BCR
102                 PM20D2
103                  WDR18
104                   P3H1
105                  UVRAG
106                  PDE6D
107                  HDAC3
108                  CENPC
109                ASPSCR1
110                  KIF1B
111                  SRP72
112                 EXOSC4
113                 TIMM13
114                  COPZ1
115                 TRIP10
116                   OPA3
117                SLC25A6
118                   DBNL
119                CCDC124
120                  EPHA2
121                  ACTR2
122                  SARNP
123                  GLRX3
124                   PEPD
125                 ZNF598
126                  NUBP1
127                   PDHB
128                   AHCY
129                   SUOX
130                   PDPN
131                  AMPD2
132                    SYK
133                SLC25A3
134                 COL8A1
135                  SCYL2
136                PRPF40B
137                  PDAP1
138                 METTL3
139                FILIP1L
140                 DHTKD1
141                  RNF14
142                  MYCBP
143                 IMPDH2
144                 TRIM38
145                  PTPN2
146                   KNG1
147                  DDOST
148                  UQCC2
149                   TBK1
150                  ITM2B
151                  LAMA2
152                 INTS11
153                DYNLRB1
154                   FNTB
155                  ABCA6
156                  RUFY2
157                  SNX18
158                  EIF3A
159                 RPL13A
160                 CCDC50
161                 TSPAN7
162                 ATP8A1
163                 EXOSC7
164                   TOM1
165                ARHGEF1
166                  HMCES
167                  PDHA1
168                  ZADH2
169                 EEF1B2
170                  AP2A2
171                   SCO2
172                 RPL23A
173                   MYH2
174                   DSTN
175               ANKRD13A
176                  FOCAD
177                  SUMO2
178                    GAK
179                    CNP
180                    SRI
181                  PLVAP
182                  DCAF8
183                   UPF1
184                FAM172A
185                OTULINL
186                  SRPK1
187                  NEDD8
188                   SCLY
189                CSNK2A2
190                  ABCC1
191                KIRREL1
192                  STK10
193                   JAK1
194                  COPB2
195                 POLR3B
196                  PLCB1
197                  EFR3A
198                  CENPV
199                   RBM4
200               SEPTIN11
201                   NPC2
202                  HSPD1
203                  USP11
204                   MOGS
205                ATP5IF1
206                  IGSF3
207                 EIF4A3
208                  ACSM3
209                 GNPDA2
210                  ABCC4
211                   NID2
212                  CWC25
213                   SND1
214                   DCXR
215                  SNU13
216                   APOO
217                  STRAP
218                  HDDC3
219                  ARMC1
220                 GXYLT2
221                   DERA
222                  SLIRP
223                  MRPL2
224                SEPTIN9
225                  PRKCA
226                  DNMT1
227                  RPS20
228                  WDR43
229                   MDC1
230                 PDGFRB
231                  PSMA5
232                  RPLP1
233                  FBLN1
234                 TMED10
235                  RPL38
236                PGLYRP2
237                  MGST3
238                 IGFBP3
239                   CNN1
240                   RFT1
241                  STAT6
242                  SYVN1
243                    MSN
244                 CHMP4B
245                  FKBP3
246                   CLN6
247                 MRPL19
248                 CHAMP1
249                   ABI1
250                 HEATR3
251                  TOR2A
252                   TSPO
253                  AP2B1
254                  ZMAT2
255                 PODNL1
256                HNRNPA3
257                 CYB5R2
258               PAFAH1B2
259        CHCHD2;CHCHD2P9
260                 LRPPRC
261                 TAGLN2
262                   VAV2
263                TRMT10C
264                  ITGA7
265                 RANBP3
266                   MCM5
267                  MED24
268                   MRC2
269                   NPM1
270                   SDHB
271                    ZFR
272                   GMIP
273                  CTDP1
274                   PREP
275                   GNAQ
276                   STX6
277                  MTCH2
278                 NUFIP2
279                   MPC2
280              SERPINB10
281                 POLR2H
282                 DOLPP1
283                  LTBP2
284                  GTF2I
285                 PTRHD1
286                  EIF4H
287                    LPL
288                 ALYREF
289                   DIS3
290                 ERLIN2
291                  WDR13
292                    PNP
293                 UBE2G2
294                  WDR44
295                   CDK2
296               SLC25A11
297                  MRE11
298                  PRDX5
299                   MYL9
300                  P4HA2
301                  GNG10
302                  PDIA5
303                  DARS2
304                 MVB12B
305                   TMA7
306                  CDH11
307                  TRAM1
308                   RFC3
309                   GFER
310                   OSBP
311                  UBE2A
312                    AK4
313                 SLC7A5
314                   CDV3
315               TRAPPC11
316                  PLBD2
317                  EIF3I
318                 SEC31A
319                   RSU1
320                  SIPA1
321                  CSRP1
322               STARD3NL
323                   MCM7
324                    ADK
325                 COBLL1
326                 NECAP2
327                  TRIP6
328                  PCID2
329                 GNPDA1
330               ALDH16A1
331                  IFIT2
332                 BCL2L1
333                  LTBP4
334              IGHV2-70D
335                   ADSL
336                  S1PR2
337 POLR2J;POLR2J2;POLR2J3
338                  RPS3A
339                 CDKN2A
340                 TRIM28
341                TBC1D2B
342             EIF1;EIF1B
343                  WDR91
344                  CWC22
345                  HMGN3
346                 EIF2B4
347                  RPS28
348                  PNISR
349                 STAT5A
350                    QKI
351                  GPD1L
352                  VPS36
353                    WLS
354                   PPOX
355                   DTD1
356                  ARMH3
357                    SRM
358                 PIEZO1
359               FAM114A2
360                  TPCN1
361                ZBTB8OS
362                   PPIA
363                   ECI1
364                 PITRM1
365                RNASET2
366                  YARS1
367                   FGGY
368                   SNF8
369                   GDE1
370                    VIM
371               FAM160B1
372                MRPS18B
373                 PRPF31
374                   TAF5
375                  DGAT1
376                  RPS27
377                  ZNFX1
378                SELENOF
379                 MRPL58
380                SLC66A3
381                  HMGA1
382                  AP3S1
383                  PRKCZ
384                   YES1
385                  NUP93
386                   VASN
387                   ACO2
388                   MSRA
389                  IFRD2
390                 ATP2B4
391                   LSM7
392                  EEF1G
393                  UBE2N
394                 GOLIM4
395                 MAP2K3</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proNormalRes_Krij$sig.feat.heat</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-52-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Normal samples (ns:
p &gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01, <span
class="math inline">\(***\)</span>: p &lt;= 0.001)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Krij$data
smpAnnoTab &lt;- proTissueRes_Krij$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proNormalRes_Krij$sig.feat.tab
num_sigFeats &lt;- 6
gpLevel = c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;, &#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;)
condiCol = c(&#39;firebrick&#39;, &#39;grey50&#39;) #c(Normal, Tumor)
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(newVar1 = paste0(Var1, &#39; (&#39;, Gene, &#39;)&#39;),
                newVar1 = factor(newVar1, levels = unique(newVar1)))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Recur_Condi = paste0(Recurrence, &#39;_&#39;, Condition),
                Recur_Condi = factor(Recur_Condi, levels = gpLevel))
ggplot(topSigFeatDat, aes(x=Recur_Condi, y=Abundance, col=Condition, fill=Recurrence)) +
  geom_boxplot(alpha = 1, outlier.shape = NA, linewidth = 1) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             comparisons = list(c(&#39;Yes_Normal&#39;, &#39;No_Normal&#39;), c(&#39;Yes_Tumor&#39;, &#39;No_Tumor&#39;)),
                             label = &#39;p.signif&#39;, tip.length = 0.015, bracket.size = 0.7, size = 4) +
  labs(x = &#39;Recurrence&#39;) +
  scale_color_manual(values = condiCol) +
  scale_fill_manual(values = c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(newVar1), scales = &#39;free&#39;) +
  th +
  theme(strip.text = element_text(size = 13, face = &#39;bold&#39;),
             axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-53-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># theme(strip.text = element_text(size = 14, face = &#39;bold&#39;),
#       axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 20),
#       axis.title = element_text(size = 26),
#       legend.title = element_text(size = 24), legend.text = element_text(size = 22))</code></pre>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proNormalRes_Krij$pval.hist +
  labs(title = &#39;Krij. Combined Normal Tissue DIA Proteomics&#39;) +
  theme(axis.title = element_text(size = 28),
        axis.text = element_text(size = 16))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-54-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/group_meeting/pVals_histogram_combined_proNormal_Krij.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 6, width = 11)</code></pre>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proNormalRes_Krij$sig.pc.tab</code></pre>
<pre><code>        Var1       Var2   pVal pValAdj Stat   Test
1 PC8 (4.2%) Recurrence 0.0119   0.237 2.59 T-test</code></pre>
<p>Visualize significant PCs and display their top features with
greatest absolute loadings</p>
<pre class="r"><code>proNormalRes_Krij$sig.pc.dist$PC8</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-56-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proNormalRes_Krij$pc.top.feat.tab$PC8</code></pre>
<pre><code>   Feature Loading     Gene
1   Q6JQN1 -0.1430   ACAD10
2   Q96RN5 -0.1420    MED15
3   Q5VV41  0.0968 ARHGEF16
4   Q8TCC3  0.0865   MRPL30
5   Q08499  0.0826    PDE4D
6   P03956  0.0818     MMP1
7   Q7RTS9  0.0710      DYM
8   Q9Y653 -0.0710   ADGRG1
9   Q9HC36  0.0708     MRM3
10  P02538 -0.0704    KRT6A
11  P42771 -0.0700   CDKN2A
12  Q13361  0.0699    MFAP5
13  Q14667 -0.0698 KIAA0100
14  P02100 -0.0697     HBE1
15  Q6UX53 -0.0672  METTL7B
16  Q7L8L6  0.0671  FASTKD5
17  O15213  0.0670    WDR46
18  Q9HA82  0.0660    CERS4
19  Q8N1Q1 -0.0645     CA13
20  Q9NYU1 -0.0639    UGGT2
21  Q86W34 -0.0626     AMZ2
22  O43927 -0.0626   CXCL13
23  P02008 -0.0623      HBZ
24  Q9H0L4  0.0623   CSTF2T
25  O94985 -0.0616   CLSTN1
26  Q6PEZ8 -0.0608   PODNL1
27  P35908 -0.0598     KRT2
28  Q8WVQ1  0.0590    CANT1
29  Q13509  0.0588    TUBB3
30  Q16678  0.0573   CYP1B1</code></pre>
<p>Enrichment analysis<br />
Reference: Hallmark gene sets<br />
<strong>t-values</strong></p>
<pre class="r"><code># Prepare ranked gene list
sigRes &lt;- proNormalRes_Krij$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, Stat) %&gt;%
  tibble::column_to_rownames(&#39;Var1&#39;)
featAnnoTab &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::select(Protein, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)

rankedGeneList &lt;- prepRankedFeatList(statRes = sigRes, featAnno = featAnnoTab, to_gene = T)</code></pre>
<pre class="r"><code># Hallmark gene sets
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)

# Run GSEA
gseaResH &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabH,
                                  minGSSize = 10, maxGSSize = 500,
                                  pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                  by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.6, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-58-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>PC8 loadings</strong><br />
There is no any overrepresented biological process.</p>
<pre class="r"><code># Prepare ranked gene list
statRes_Krij &lt;- proNormalRes_Krij$SOA.res$pcaRes@loadings %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, PC8) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
featAnno_Krij &lt;- tibble::as_tibble(rowData(proTissue_Krij), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::select(Protein, Genes) %&gt;%
  dplyr::rename(Gene = Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
rankedGeneList_Krij &lt;- prepRankedFeatList(statRes = statRes_Krij, featAnno = featAnno_Krij, to_gene = T)

# Hallmark gene sets
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)
# Run GSEA
gseaResH_pc8_Krij &lt;- clusterProfiler::GSEA(geneList = rankedGeneList_Krij, TERM2GENE = msigTabH,
                                           minGSSize = 10, maxGSSize = 500,
                                           pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                           by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
# gseaResH_pc8_Krij@result</code></pre>
</div>
</div>
<div id="tissue-proteomics-incomplete-ag-klingmüller"
class="section level1">
<h1>Tissue Proteomics (Incomplete, AG Klingmüller)</h1>
<p>Discovery (incomplete) + MethodDev</p>
<p>Display associations between cancer recurrence and the other patient
metadata variables to identify potential predictors (or confounders)</p>
<pre class="r"><code># Test associations between cancer recurrence and the other patient metadata variables
# Prepare patient metadata
proTissue &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
patientMetadat &lt;- tibble::as_tibble(colData(proTissue)) %&gt;%
  dplyr::filter(!duplicated(Patient)) %&gt;%
  dplyr::select(Patient, Recurrence, Gender, Age, Smoking, Stage) #Adjuvant
patientRecur &lt;- dplyr::select(patientMetadat, Patient, Recurrence)
patientMetadat &lt;- dplyr::select(patientMetadat, -Recurrence)
testAsso(patientMetadat, patientRecur, cmn_col = &#39;Patient&#39;) %&gt;%
  dplyr::select(-Stat)</code></pre>
<pre><code>     Var1       Var2       pVal   pValAdj       Test
1   Stage Recurrence 0.06792046 0.1792319 Chi-square
2 Smoking Recurrence 0.08961593 0.1792319 Chi-square
3  Gender Recurrence 0.15599559 0.2079941 Chi-square
4     Age Recurrence 0.91146280 0.9114628     T-test</code></pre>
<p><strong>metadata-assisted quality control: Tumor vs Normal
samples</strong></p>
<pre class="r"><code># Load preprocessed data
proTissue_Klin &lt;- readRDS(&#39;./data/MethodDev/AG_Klingmueller/proTissueVsn.rds&#39;)
# Perform single-omics analysis
proTissueRes_Klin &lt;- doSOA(proTissue_Klin, meta_var = c(&#39;Condition&#39;, &#39;Cohort&#39;),
                           pca_method = &#39;ppca&#39;, num_PCs = 20, do_onlyPCA = T)

# Visualize PCs
pcTab &lt;- proTissueRes_Klin$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;I0HVOL_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (31.1%)`, color=Condition, fill=Condition, label=Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 20),
        legend.title = element_text(size = 28), legend.text = element_text(size = 26))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-61-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/boxplot_quality_control_combined_proTissue_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 9, width = 11)</code></pre>
<pre class="r"><code># Prepare data accounted for SVs for later visualizations
# Do SVA
proTissueSVA_Klin &lt;- doSVA(proTissue_Klin, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;),
                           numSV_method = &#39;be&#39;, asso_metaVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;,
                                                                 &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;,
                                                                 &#39;Stage&#39;, &#39;TumorPurity&#39;))
# Show significant associations between SVs and sample metadata variables
# Sample condition accounts for large variance in data, so perhaps to preserve effects
# of recurrence, some variance contributed from condition is eliminated.
# proTissueSVA_Klin$sigAssoTab
# Perform single-omics analysis
proTissueSVs_Klin &lt;- proTissueSVA_Klin$summExp_SVs
SVs &lt;- grep(&#39;^SV\\d+$&#39;, colnames(colData(proTissueSVs_Klin)), value = T)
proTissueRes4Viz_Klin &lt;- doSOA(proTissueSVs_Klin, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
                               num_PCs = 20, do_onlyPCA = T, unwantVar = SVs)

# Visualize PCs
pcTab &lt;- proTissueRes4Viz_Klin$pcTab
# ggplot(pcTab, aes(x=Condition, y=`PC1 (48%)`, col=Condition, fill=Condition)) +
#   geom_boxplot(alpha = 0.7, outliers = F) +
#   geom_jitter(position = position_jitter(0.3), size = 2, show.legend = F) +
#   scale_color_brewer(palette = &#39;Dark2&#39;) +
#   scale_fill_brewer(palette = &#39;Dark2&#39;) +
#   th</code></pre>
<div id="tumor-samples-2" class="section level2">
<h2>Tumor samples</h2>
<pre class="r"><code># Subset certain samples
proTumor_Klin &lt;- proTissueSVs_Klin[, which(colData(proTissueSVs_Klin)$Condition == &#39;Tumor&#39;)]
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proTumor_Klin), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::rename(Gene = PG.Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
# Perform single-omics analysis
# proTumorRes_Klin &lt;- doSingleOmicsAnalysis(proTumor_Klin, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                           pca_method = &#39;ppca&#39;, num_PCs = 20, num_PCfeats = 30,
#                                           use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                           feat_conv = T, plot_title = NULL, show_rownames = F,
#                                           pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                           &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Batch&#39;, &#39;TumorPurity&#39;),
#                                           soa_unwantVar = grep(&#39;^SV\\d+$&#39;, colnames(colData(proTumor_Klin)),
#                                                                value = T),
#                                           level_metaVar = c(&#39;Yes&#39;, &#39;No&#39;))
# saveRDS(proTumorRes_Klin, &#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs_Old.rds&#39;)
proTumorRes_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proTumorRes_SVs_Old.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proTumorRes_Klin$data.var.source</code></pre>
<pre><code>          Var1        Var2    pVal pValAdj   Stat        Test
1   PC8 (4.7%)       Batch 0.00663   0.427  2.880      T-test
2  PC13 (3.2%)      Gender 0.00840   0.427 -2.780      T-test
3  PC14 (3.1%)         Age 0.01940   0.427  0.373 Correlation
4   PC9 (4.4%)       Stage 0.01960   0.427  4.400       ANOVA
5  PC1 (13.5%)       Batch 0.02010   0.427  2.430      T-test
6  PC16 (2.8%)     Smoking 0.02190   0.427  4.260       ANOVA
7  PC2 (10.4%)       Batch 0.02320   0.427  2.370      T-test
8  PC19 (2.2%) TumorPurity 0.02440   0.427  0.360 Correlation
9  PC1 (13.5%)         Age 0.03170   0.493 -0.345 Correlation
10 PC12 (3.7%) TumorPurity 0.03780   0.530  0.334 Correlation
11  PC3 (9.4%)       Stage 0.04340   0.553  3.430       ANOVA
12  PC9 (4.4%)         Age 0.04790   0.559 -0.319 Correlation</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2). Owing to missing values, there may be situation that
only few observations in sample groups, resulting in unreliable
t-statistics. To this end, we used probabilistic dropout model (proDA)
to account for data missingness.</p>
<pre class="r"><code>dplyr::select(proTumorRes_Klin$sig.feat.tab, -Test)</code></pre>
<pre><code>          Var1       Var2     pVal pValAdj  Stat      Gene
1       Q96G91 Recurrence 2.61e-05   0.215  5.29    P2RY11
2       Q15828 Recurrence 2.63e-04   1.000  4.34      CST6
3       Q9UHY7 Recurrence 6.15e-04   1.000  3.99    ENOPH1
4       P00739 Recurrence 7.19e-04   1.000 -3.93       HPR
5       O95445 Recurrence 7.21e-04   1.000 -3.93      APOM
6       Q13637 Recurrence 8.52e-04   1.000  3.86     RAB32
7       O94811 Recurrence 1.45e-03   1.000 -3.64      TPPP
8       A5D8V6 Recurrence 1.51e-03   1.000  3.62    VPS37C
9       P0C0L5 Recurrence 1.56e-03   1.000 -3.61     C4B_2
10      Q330K2 Recurrence 1.71e-03   1.000  3.57   NDUFAF6
11  A0A0C4DH36 Recurrence 1.79e-03   1.000 -3.55  IGHV3-38
12      P48595 Recurrence 1.85e-03   1.000 -3.54 SERPINB10
13      Q13153 Recurrence 1.95e-03   1.000  3.52      PAK1
14      Q8N4T8 Recurrence 2.01e-03   1.000  3.50      CBR4
15      P48449 Recurrence 2.41e-03   1.000  3.43       LSS
16      O00330 Recurrence 2.45e-03   1.000  3.42      PDHX
17      P30626 Recurrence 3.09e-03   1.000  3.32       SRI
18      O75475 Recurrence 3.22e-03   1.000 -3.31     PSIP1
19      P0C0L4 Recurrence 3.41e-03   1.000 -3.28       C4A
20      Q13541 Recurrence 3.41e-03   1.000  3.28  EIF4EBP1
21      P27169 Recurrence 3.51e-03   1.000 -3.27      PON1
22      P13489 Recurrence 3.62e-03   1.000  3.26      RNH1
23      A7KAX9 Recurrence 3.64e-03   1.000  3.25  ARHGAP32
24      O75531 Recurrence 3.71e-03   1.000  3.25     BANF1
25      Q03468 Recurrence 3.76e-03   1.000 -3.24     ERCC6
26      Q9Y3B8 Recurrence 4.00e-03   1.000  3.21     REXO2
27      Q96I59 Recurrence 4.08e-03   1.000  3.21     NARS2
28      P10515 Recurrence 4.20e-03   1.000  3.19      DLAT
29      P49326 Recurrence 4.44e-03   1.000 -3.17      FMO5
30      P0DPD7 Recurrence 4.78e-03   1.000 -3.14 EEF1AKMT4
31      Q9UBF6 Recurrence 4.88e-03   1.000  3.13      RNF7
32      Q14247 Recurrence 4.94e-03   1.000  3.12      CTTN
33      Q9NP99 Recurrence 5.16e-03   1.000  3.11     TREM1
34      P10451 Recurrence 5.19e-03   1.000  3.10      SPP1
35      Q99988 Recurrence 5.32e-03   1.000  3.09     GDF15
36      P04114 Recurrence 5.33e-03   1.000 -3.09      APOB
37      P0C870 Recurrence 5.54e-03   1.000 -3.07     JMJD7
38      Q14624 Recurrence 5.77e-03   1.000 -3.06     ITIH4
39      Q6IA69 Recurrence 5.82e-03   1.000  3.05   NADSYN1
40      A1L4H1 Recurrence 6.13e-03   1.000 -3.03     SSC5D
41      Q13790 Recurrence 6.16e-03   1.000 -3.03      APOF
42      P28065 Recurrence 6.20e-03   1.000 -3.03     PSMB9
43      Q92974 Recurrence 6.28e-03   1.000 -3.02   ARHGEF2
44      Q14CN2 Recurrence 6.42e-03   1.000 -3.01     CLCA4
45      Q9UK76 Recurrence 6.53e-03   1.000  3.00      JPT1
46      Q06278 Recurrence 6.54e-03   1.000 -3.00      AOX1
47      Q9UHD1 Recurrence 6.54e-03   1.000  3.00   CHORDC1
48      Q5T3J3 Recurrence 6.57e-03   1.000 -3.00     LRIF1
49      Q6JBY9 Recurrence 6.90e-03   1.000 -2.98     RCSD1
50      Q96RS6 Recurrence 6.90e-03   1.000  2.98    NUDCD1
51      Q8N9N8 Recurrence 7.14e-03   1.000  2.97    EIF1AD
52      Q8WUA2 Recurrence 7.20e-03   1.000 -2.96     PPIL4
53      P21246 Recurrence 7.36e-03   1.000 -2.95       PTN
54      Q9HC16 Recurrence 7.60e-03   1.000 -2.94  APOBEC3G
55      Q6RW13 Recurrence 7.74e-03   1.000 -2.93    AGTRAP
56      P29317 Recurrence 8.19e-03   1.000 -2.91     EPHA2
57      P02654 Recurrence 8.39e-03   1.000 -2.90     APOC1
58      Q9Y570 Recurrence 8.59e-03   1.000  2.89     PPME1
59      P30043 Recurrence 8.61e-03   1.000 -2.88     BLVRB
60      Q9C002 Recurrence 8.64e-03   1.000  2.88     NMES1
61      Q8N6H7 Recurrence 8.84e-03   1.000  2.87   ARFGAP2
62      P15169 Recurrence 9.00e-03   1.000 -2.86      CPN1
63      O15228 Recurrence 9.05e-03   1.000  2.86     GNPAT
64      P23508 Recurrence 9.37e-03   1.000 -2.85       MCC
65      Q9BZW5 Recurrence 9.51e-03   1.000  2.84    TM6SF1
66      O00237 Recurrence 9.54e-03   1.000 -2.84    RNF103
67      Q9NUL5 Recurrence 1.02e-02   1.000  2.81      SHFL
68      O14791 Recurrence 1.06e-02   1.000 -2.79     APOL1
69      P06748 Recurrence 1.08e-02   1.000 -2.78      NPM1
70      Q8NCE2 Recurrence 1.08e-02   1.000  2.78    MTMR14
71      Q9BXN1 Recurrence 1.08e-02   1.000  2.78      ASPN
72      Q5TH69 Recurrence 1.12e-02   1.000 -2.77   ARFGEF3
73      Q13835 Recurrence 1.13e-02   1.000 -2.77      PKP1
74      Q7Z4H8 Recurrence 1.13e-02   1.000  2.77   POGLUT3
75      Q92954 Recurrence 1.15e-02   1.000 -2.76      PRG4
76      Q9HBU6 Recurrence 1.15e-02   1.000  2.76     ETNK1
77      Q9C0B1 Recurrence 1.16e-02   1.000 -2.76       FTO
78      Q96RF0 Recurrence 1.18e-02   1.000  2.75     SNX18
79      Q96MP8 Recurrence 1.19e-02   1.000  2.74     KCTD7
80      Q9UNX4 Recurrence 1.20e-02   1.000  2.74      WDR3
81      P06276 Recurrence 1.21e-02   1.000 -2.73      BCHE
82      Q8TE77 Recurrence 1.21e-02   1.000  2.74      SSH3
83      P35813 Recurrence 1.23e-02   1.000  2.73     PPM1A
84      O60934 Recurrence 1.24e-02   1.000  2.72       NBN
85      Q9ULR0 Recurrence 1.26e-02   1.000 -2.71      ISY1
86      Q8N1F8 Recurrence 1.27e-02   1.000  2.71   STK11IP
87      Q6DHV7 Recurrence 1.29e-02   1.000 -2.71      ADAL
88      Q9NS73 Recurrence 1.29e-02   1.000  2.71      MBIP
89      P05546 Recurrence 1.30e-02   1.000 -2.70  SERPIND1
90      P80108 Recurrence 1.34e-02   1.000 -2.69     GPLD1
91      Q5T5P2 Recurrence 1.35e-02   1.000  2.68  KIAA1217
92      Q96AY4 Recurrence 1.35e-02   1.000  2.69     TTC28
93      P08236 Recurrence 1.36e-02   1.000 -2.68      GUSB
94      Q14012 Recurrence 1.39e-02   1.000  2.67     CAMK1
95      O95487 Recurrence 1.41e-02   1.000  2.67    SEC24B
96      P51965 Recurrence 1.43e-02   1.000  2.66    UBE2E1
97      Q6BCY4 Recurrence 1.43e-02   1.000  2.66    CYB5R2
98      Q9NTI5 Recurrence 1.44e-02   1.000 -2.66     PDS5B
99      Q15417 Recurrence 1.47e-02   1.000  2.65      CNN3
100     P67870 Recurrence 1.48e-02   1.000  2.65    CSNK2B
101     Q12857 Recurrence 1.48e-02   1.000 -2.64      NFIA
102     Q13126 Recurrence 1.48e-02   1.000 -2.64      MTAP
103     Q96FW1 Recurrence 1.49e-02   1.000  2.64     OTUB1
104     Q8WUY3 Recurrence 1.53e-02   1.000 -2.63    PRUNE2
105     O14939 Recurrence 1.54e-02   1.000 -2.63      PLD2
106     Q9HCK4 Recurrence 1.54e-02   1.000 -2.63     ROBO2
107     Q96HC4 Recurrence 1.55e-02   1.000  2.62    PDLIM5
108     O15234 Recurrence 1.58e-02   1.000 -2.61     CASC3
109     Q08174 Recurrence 1.58e-02   1.000 -2.62     PCDH1
110     P04180 Recurrence 1.60e-02   1.000 -2.61      LCAT
111     Q86SQ0 Recurrence 1.61e-02   1.000 -2.61    PHLDB2
112     Q9UJY1 Recurrence 1.63e-02   1.000  2.60     HSPB8
113     P01024 Recurrence 1.64e-02   1.000 -2.60        C3
114     Q7LDG7 Recurrence 1.69e-02   1.000 -2.59   RASGRP2
115     Q92506 Recurrence 1.69e-02   1.000  2.58   HSD17B8
116     A6NDB9 Recurrence 1.70e-02   1.000  2.58     PALM3
117     Q5MNZ9 Recurrence 1.71e-02   1.000 -2.58     WIPI1
118     P02647 Recurrence 1.72e-02   1.000 -2.58     APOA1
119     Q6P1X6 Recurrence 1.73e-02   1.000  2.58   C8orf82
120     Q8N7R7 Recurrence 1.74e-02   1.000  2.57    CCNYL1
121     Q96C57 Recurrence 1.74e-02   1.000  2.57    CUSTOS
122     Q9Y5X1 Recurrence 1.76e-02   1.000  2.57      SNX9
123     Q9NPJ8 Recurrence 1.77e-02   1.000  2.56      NXT2
124     P12829 Recurrence 1.78e-02   1.000 -2.56      MYL4
125     Q14331 Recurrence 1.81e-02   1.000 -2.55      FRG1
126     Q9Y2D2 Recurrence 1.81e-02   1.000  2.55   SLC35A3
127     O43866 Recurrence 1.83e-02   1.000 -2.55      CD5L
128     P01903 Recurrence 1.85e-02   1.000 -2.54   HLA-DRA
129     P54578 Recurrence 1.86e-02   1.000 -2.54     USP14
130     Q15493 Recurrence 1.87e-02   1.000 -2.54       RGN
131     P27824 Recurrence 1.89e-02   1.000 -2.53      CANX
132     Q3SY69 Recurrence 1.92e-02   1.000  2.53   ALDH1L2
133     Q9UM54 Recurrence 1.92e-02   1.000  2.53      MYO6
134     P58546 Recurrence 1.93e-02   1.000  2.52      MTPN
135     Q9BV73 Recurrence 1.95e-02   1.000 -2.52    CEP250
136     O14684 Recurrence 1.97e-02   1.000  2.52     PTGES
137     Q9UPR0 Recurrence 1.99e-02   1.000 -2.51     PLCL2
138     P30679 Recurrence 2.02e-02   1.000  2.50     GNA15
139     O14975 Recurrence 2.08e-02   1.000 -2.49   SLC27A2
140     Q9H0X4 Recurrence 2.08e-02   1.000 -2.49   FAM234A
141     Q6WKZ4 Recurrence 2.10e-02   1.000  2.49 RAB11FIP1
142     Q8NB37 Recurrence 2.13e-02   1.000  2.48     GATD1
143     P25391 Recurrence 2.15e-02   1.000 -2.47     LAMA1
144     P01601 Recurrence 2.18e-02   1.000 -2.47 IGKV1D-16
145     Q8N465 Recurrence 2.18e-02   1.000  2.47    D2HGDH
146     Q7Z434 Recurrence 2.19e-02   1.000  2.47      MAVS
147     P20036 Recurrence 2.21e-02   1.000 -2.46  HLA-DPA1
148     P40189 Recurrence 2.21e-02   1.000 -2.46     IL6ST
149     Q92692 Recurrence 2.21e-02   1.000 -2.46   NECTIN2
150     Q14678 Recurrence 2.23e-02   1.000 -2.46     KANK1
151     Q9UIA9 Recurrence 2.23e-02   1.000  2.46      XPO7
152     Q9H307 Recurrence 2.25e-02   1.000 -2.45       PNN
153     P12259 Recurrence 2.27e-02   1.000 -2.45        F5
154     Q56VL3 Recurrence 2.27e-02   1.000  2.45    OCIAD2
155     P02656 Recurrence 2.28e-02   1.000 -2.45     APOC3
156     Q14139 Recurrence 2.28e-02   1.000  2.45     UBE4A
157     Q8TF01 Recurrence 2.28e-02   1.000 -2.45     PNISR
158     P17066 Recurrence 2.29e-02   1.000 -2.45     HSPA6
159     Q15904 Recurrence 2.30e-02   1.000 -2.44   ATP6AP1
160     P62714 Recurrence 2.35e-02   1.000  2.43    PPP2CB
161     Q02985 Recurrence 2.36e-02   1.000 -2.43     CFHR3
162     Q6P179 Recurrence 2.36e-02   1.000 -2.43     ERAP2
163     P30511 Recurrence 2.38e-02   1.000 -2.43     HLA-F
164     Q9Y365 Recurrence 2.40e-02   1.000  2.42   STARD10
165     Q86UU0 Recurrence 2.42e-02   1.000  2.42     BCL9L
166     O75896 Recurrence 2.44e-02   1.000  2.42     TUSC2
167     P43250 Recurrence 2.45e-02   1.000 -2.42      GRK6
168     Q99614 Recurrence 2.47e-02   1.000 -2.41      TTC1
169     P62191 Recurrence 2.48e-02   1.000 -2.41     PSMC1
170     P49589 Recurrence 2.50e-02   1.000  2.41     CARS1
171     P25788 Recurrence 2.52e-02   1.000 -2.40     PSMA3
172     Q7Z591 Recurrence 2.52e-02   1.000 -2.40      AKNA
173     Q70J99 Recurrence 2.53e-02   1.000  2.40    UNC13D
174     O43731 Recurrence 2.56e-02   1.000  2.39    KDELR3
175     Q16822 Recurrence 2.58e-02   1.000  2.39      PCK2
176     Q7Z3D4 Recurrence 2.60e-02   1.000  2.39    LYSMD3
177     O95696 Recurrence 2.64e-02   1.000 -2.38      BRD1
178     P55212 Recurrence 2.67e-02   1.000  2.37     CASP6
179     Q16625 Recurrence 2.67e-02   1.000  2.37      OCLN
180     Q53ET0 Recurrence 2.69e-02   1.000  2.37     CRTC2
181     Q96NT5 Recurrence 2.69e-02   1.000 -2.37   SLC46A1
182     P40394 Recurrence 2.71e-02   1.000 -2.37      ADH7
183     Q9H936 Recurrence 2.72e-02   1.000  2.37  SLC25A22
184     Q9Y250 Recurrence 2.72e-02   1.000 -2.37     LZTS1
185     P01023 Recurrence 2.75e-02   1.000 -2.36       A2M
186     Q16769 Recurrence 2.76e-02   1.000  2.36      QPCT
187     Q8NEV8 Recurrence 2.76e-02   1.000  2.36     EXPH5
188     Q96FJ0 Recurrence 2.77e-02   1.000 -2.36  STAMBPL1
189     P16930 Recurrence 2.78e-02   1.000 -2.36       FAH
190     P52788 Recurrence 2.79e-02   1.000  2.36       SMS
191     P61086 Recurrence 2.79e-02   1.000  2.35     UBE2K
192     Q9UNN5 Recurrence 2.79e-02   1.000 -2.35      FAF1
193     Q8NBJ7 Recurrence 2.80e-02   1.000 -2.35     SUMF2
194     P28070 Recurrence 2.81e-02   1.000 -2.35     PSMB4
195     P02655 Recurrence 2.82e-02   1.000 -2.35     APOC2
196     Q13336 Recurrence 2.82e-02   1.000  2.35   SLC14A1
197     Q8N0Z3 Recurrence 2.82e-02   1.000 -2.35    SPICE1
198     Q9NUQ6 Recurrence 2.83e-02   1.000  2.35   SPATS2L
199     O43586 Recurrence 2.89e-02   1.000 -2.34   PSTPIP1
200     P19397 Recurrence 2.89e-02   1.000 -2.34      CD53
201     Q04837 Recurrence 2.89e-02   1.000  2.34     SSBP1
202     Q9UKU7 Recurrence 2.90e-02   1.000  2.34     ACAD8
203     Q8TF66 Recurrence 2.91e-02   1.000  2.33    LRRC15
204     Q9BXI6 Recurrence 2.91e-02   1.000  2.34  TBC1D10A
205     P29034 Recurrence 2.93e-02   1.000  2.33    S100A2
206     P08514 Recurrence 2.94e-02   1.000 -2.33    ITGA2B
207     P28067 Recurrence 2.96e-02   1.000 -2.33   HLA-DMA
208     Q13057 Recurrence 2.96e-02   1.000 -2.33     COASY
209     Q96P16 Recurrence 2.96e-02   1.000 -2.33    RPRD1A
210     P56589 Recurrence 2.97e-02   1.000  2.33      PEX3
211     Q9NNW5 Recurrence 2.98e-02   1.000 -2.32      WDR6
212     Q9H0E2 Recurrence 3.00e-02   1.000  2.32    TOLLIP
213     Q12981 Recurrence 3.04e-02   1.000 -2.31     BNIP1
214     Q8NF50 Recurrence 3.07e-02   1.000 -2.31     DOCK8
215     Q8NC42 Recurrence 3.08e-02   1.000  2.31    RNF149
216     P10114 Recurrence 3.11e-02   1.000  2.30     RAP2A
217     P04233 Recurrence 3.12e-02   1.000 -2.30      CD74
218     Q9Y6C9 Recurrence 3.13e-02   1.000  2.30     MTCH2
219     Q9Y5J9 Recurrence 3.15e-02   1.000  2.30    TIMM8B
220     Q8N511 Recurrence 3.17e-02   1.000 -2.29   TMEM199
221     Q13201 Recurrence 3.18e-02   1.000 -2.29     MMRN1
222     Q14919 Recurrence 3.18e-02   1.000  2.29     DRAP1
223     Q9P270 Recurrence 3.19e-02   1.000  2.29    SLAIN2
224     P40306 Recurrence 3.20e-02   1.000 -2.29    PSMB10
225     Q5SYC1 Recurrence 3.22e-02   1.000 -2.29     CLVS2
226     Q8TER0 Recurrence 3.23e-02   1.000  2.28     SNED1
227     Q96EY4 Recurrence 3.23e-02   1.000 -2.29     TMA16
228     P55287 Recurrence 3.25e-02   1.000  2.28     CDH11
229     Q4G0F5 Recurrence 3.27e-02   1.000  2.28    VPS26B
230     Q8N0U8 Recurrence 3.27e-02   1.000  2.28  VKORC1L1
231     Q15063 Recurrence 3.30e-02   1.000  2.27     POSTN
232     Q14202 Recurrence 3.32e-02   1.000 -2.27     ZMYM3
233     P61160 Recurrence 3.34e-02   1.000  2.27     ACTR2
234     Q9BZX2 Recurrence 3.36e-02   1.000 -2.27      UCK2
235     Q9UHR6 Recurrence 3.37e-02   1.000 -2.27    ZNHIT2
236     P05186 Recurrence 3.39e-02   1.000 -2.26      ALPL
237     P12544 Recurrence 3.41e-02   1.000 -2.26      GZMA
238     P24557 Recurrence 3.41e-02   1.000  2.26    TBXAS1
239     O95807 Recurrence 3.43e-02   1.000  2.26   TMEM50A
240     Q6H8Q1 Recurrence 3.43e-02   1.000 -2.26    ABLIM2
241     Q6ZV73 Recurrence 3.43e-02   1.000 -2.26      FGD6
242     P00749 Recurrence 3.44e-02   1.000  2.25      PLAU
243     Q8IZ52 Recurrence 3.44e-02   1.000  2.26      CHPF
244     Q8NBI5 Recurrence 3.44e-02   1.000  2.25   SLC43A3
245     Q8TES7 Recurrence 3.44e-02   1.000 -2.26      FBF1
246     P01033 Recurrence 3.48e-02   1.000  2.25     TIMP1
247 A0A075B6R2 Recurrence 3.49e-02   1.000 -2.25   IGHV4-4
248 A0A075B6K4 Recurrence 3.52e-02   1.000 -2.24  IGLV3-10
249     Q99538 Recurrence 3.53e-02   1.000  2.24      LGMN
250     P01011 Recurrence 3.55e-02   1.000 -2.24  SERPINA3
251     Q4G0X4 Recurrence 3.56e-02   1.000  2.24    KCTD21
252     O75608 Recurrence 3.64e-02   1.000  2.23    LYPLA1
253     P60891 Recurrence 3.65e-02   1.000 -2.23     PRPS1
254     O00221 Recurrence 3.66e-02   1.000  2.23    NFKBIE
255     Q9C040 Recurrence 3.66e-02   1.000  2.23     TRIM2
256     Q53H12 Recurrence 3.71e-02   1.000  2.22       AGK
257     Q9Y305 Recurrence 3.71e-02   1.000  2.22     ACOT9
258     Q9H1A4 Recurrence 3.72e-02   1.000 -2.22    ANAPC1
259     O14493 Recurrence 3.74e-02   1.000  2.22     CLDN4
260     P33908 Recurrence 3.75e-02   1.000  2.21    MAN1A1
261     Q16643 Recurrence 3.75e-02   1.000 -2.21      DBN1
262     O75976 Recurrence 3.77e-02   1.000  2.21       CPD
263     P20851 Recurrence 3.78e-02   1.000 -2.21     C4BPB
264     Q86TJ2 Recurrence 3.79e-02   1.000 -2.21    TADA2B
265     Q9Y4I1 Recurrence 3.79e-02   1.000  2.21     MYO5A
266     P00734 Recurrence 3.85e-02   1.000 -2.20        F2
267     O75182 Recurrence 3.89e-02   1.000 -2.20     SIN3B
268     Q99611 Recurrence 3.89e-02   1.000  2.20    SEPHS2
269     Q9BVV7 Recurrence 3.89e-02   1.000 -2.20    TIMM21
270     P0C626 Recurrence 3.90e-02   1.000  2.20     OR5G3
271     Q9BUK6 Recurrence 3.94e-02   1.000 -2.19     MSTO1
272     Q9Y2G2 Recurrence 3.94e-02   1.000 -2.19     CARD8
273 A0A075B6H7 Recurrence 3.96e-02   1.000 -2.19   IGKV3-7
274     Q9BXT4 Recurrence 3.97e-02   1.000 -2.19     TDRD1
275     Q3KP31 Recurrence 3.98e-02   1.000  2.18    ZNF791
276     Q5XPI4 Recurrence 3.98e-02   1.000 -2.18    RNF123
277     Q9P0L0 Recurrence 3.98e-02   1.000 -2.18      VAPA
278     P25787 Recurrence 4.00e-02   1.000 -2.18     PSMA2
279     Q01650 Recurrence 4.03e-02   1.000  2.18    SLC7A5
280     Q96PY6 Recurrence 4.03e-02   1.000 -2.18      NEK1
281     Q8WVV9 Recurrence 4.04e-02   1.000  2.18   HNRNPLL
282     Q9P1F3 Recurrence 4.07e-02   1.000  2.17    ABRACL
283     P15907 Recurrence 4.09e-02   1.000 -2.17   ST6GAL1
284     P24844 Recurrence 4.11e-02   1.000  2.17      MYL9
285     P39880 Recurrence 4.11e-02   1.000  2.17      CUX1
286     Q96KP4 Recurrence 4.12e-02   1.000 -2.17     CNDP2
287     P06703 Recurrence 4.15e-02   1.000  2.17    S100A6
288     Q9BRP1 Recurrence 4.18e-02   1.000  2.16    PDCD2L
289     O60306 Recurrence 4.19e-02   1.000 -2.16       AQR
290     Q96GA7 Recurrence 4.19e-02   1.000  2.16      SDSL
291     Q08477 Recurrence 4.20e-02   1.000 -2.16    CYP4F3
292     P55196 Recurrence 4.21e-02   1.000 -2.16      AFDN
293     P62877 Recurrence 4.22e-02   1.000  2.16      RBX1
294     P22061 Recurrence 4.25e-02   1.000 -2.15     PCMT1
295     Q02224 Recurrence 4.25e-02   1.000  2.15     CENPE
296     P02652 Recurrence 4.27e-02   1.000 -2.15     APOA2
297     Q07960 Recurrence 4.28e-02   1.000  2.15   ARHGAP1
298     Q32P44 Recurrence 4.28e-02   1.000  2.15      EML3
299     P80188 Recurrence 4.29e-02   1.000  2.15      LCN2
300     Q9UHL4 Recurrence 4.29e-02   1.000 -2.15      DPP7
301     P14317 Recurrence 4.30e-02   1.000  2.15     HCLS1
302     Q9Y3Q0 Recurrence 4.30e-02   1.000  2.15   NAALAD2
303     Q9Y286 Recurrence 4.31e-02   1.000  2.15   SIGLEC7
304     P06280 Recurrence 4.33e-02   1.000  2.14       GLA
305     Q9Y646 Recurrence 4.33e-02   1.000 -2.14       CPQ
306     Q9NRY5 Recurrence 4.34e-02   1.000 -2.14  FAM114A2
307     Q8WUN7 Recurrence 4.38e-02   1.000 -2.14     UBTD2
308     P51809 Recurrence 4.40e-02   1.000  2.14     VAMP7
309     Q8WV74 Recurrence 4.41e-02   1.000  2.14     NUDT8
310     P48454 Recurrence 4.42e-02   1.000  2.13    PPP3CC
311     P20718 Recurrence 4.43e-02   1.000 -2.13      GZMH
312     Q9BSH5 Recurrence 4.44e-02   1.000  2.13     HDHD3
313     P08195 Recurrence 4.46e-02   1.000  2.13    SLC3A2
314     Q92843 Recurrence 4.46e-02   1.000  2.13    BCL2L2
315     O94887 Recurrence 4.47e-02   1.000  2.13     FARP2
316     Q99828 Recurrence 4.47e-02   1.000  2.13      CIB1
317     Q9BQE3 Recurrence 4.47e-02   1.000  2.13    TUBA1C
318     P07225 Recurrence 4.49e-02   1.000 -2.13     PROS1
319     Q7L591 Recurrence 4.50e-02   1.000  2.13      DOK3
320     O15327 Recurrence 4.51e-02   1.000  2.12    INPP4B
321     Q7LBR1 Recurrence 4.51e-02   1.000 -2.12    CHMP1B
322     P02794 Recurrence 4.54e-02   1.000  2.12      FTH1
323     Q16549 Recurrence 4.55e-02   1.000 -2.12     PCSK7
324     Q96C90 Recurrence 4.56e-02   1.000  2.12  PPP1R14B
325     Q96QH2 Recurrence 4.56e-02   1.000  2.12     PRAM1
326     P25786 Recurrence 4.57e-02   1.000 -2.12     PSMA1
327     Q17R89 Recurrence 4.59e-02   1.000 -2.12  ARHGAP44
328     Q86YT9 Recurrence 4.60e-02   1.000  2.12      JAML
329     Q9BXB4 Recurrence 4.61e-02   1.000 -2.11   OSBPL11
330     P35542 Recurrence 4.62e-02   1.000 -2.11      SAA4
331     Q96BR1 Recurrence 4.62e-02   1.000  2.11      SGK3
332     Q9UIG8 Recurrence 4.62e-02   1.000 -2.11   SLCO3A1
333     Q8NB90 Recurrence 4.67e-02   1.000 -2.11    SPATA5
334     Q96MM7 Recurrence 4.67e-02   1.000  2.11    HS6ST2
335     Q969G6 Recurrence 4.72e-02   1.000 -2.10       RFK
336     P08697 Recurrence 4.73e-02   1.000 -2.10  SERPINF2
337     Q92604 Recurrence 4.74e-02   1.000  2.10    LPGAT1
338     O43166 Recurrence 4.79e-02   1.000  2.09   SIPA1L1
339     O43768 Recurrence 4.79e-02   1.000  2.09      ENSA
340     P00414 Recurrence 4.79e-02   1.000 -2.10    MT-CO3
341     P13995 Recurrence 4.80e-02   1.000 -2.09    MTHFD2
342     Q6P2E9 Recurrence 4.82e-02   1.000 -2.09      EDC4
343     Q5T5C0 Recurrence 4.83e-02   1.000  2.09    STXBP5
344     P23443 Recurrence 4.84e-02   1.000  2.09   RPS6KB1
345     P19838 Recurrence 4.85e-02   1.000  2.09     NFKB1
346     Q6PI78 Recurrence 4.90e-02   1.000  2.08    TMEM65
347     Q68D91 Recurrence 4.91e-02   1.000 -2.08    MBLAC2
348     O15504 Recurrence 4.93e-02   1.000 -2.08     NUP42
349     P51153 Recurrence 4.94e-02   1.000 -2.08     RAB13
350     Q9BWT7 Recurrence 4.96e-02   1.000  2.08    CARD10
351 A0A075B6J9 Recurrence 4.98e-02   1.000 -2.08  IGLV2-18
352     P20618 Recurrence 4.98e-02   1.000 -2.08     PSMB1
353     Q96NA2 Recurrence 4.98e-02   1.000 -2.08      RILP
354     Q9UK55 Recurrence 4.99e-02   1.000 -2.08 SERPINA10</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proTumorRes_Klin$sig.feat.heat</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-66-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize top significant features identified in Tumor samples (ns: p
&gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes4Viz_Klin$dataCorrect
smpAnnoTab &lt;- proTissueRes4Viz_Klin$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proTumorRes_Klin$sig.feat.tab
num_sigFeats &lt;- 6
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = Gene))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Condition = factor(Condition, levels = c(&#39;Tumor&#39;, &#39;Normal&#39;)),
                Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
# Visualize top significant features
ggplot(topSigFeatDat, aes(x=Condition, y=Abundance, fill=Recurrence)) +
  geom_boxplot(alpha = 1, linewidth = 1, outlier.color = &#39;grey50&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             label = &#39;p.signif&#39;, size = 7, vjust = 0.5) +
  labs(x = &#39;Sample condition&#39;, y = &#39;log(Abundance)&#39;) +
  scale_fill_manual(values = c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  geom_vline(xintercept = 1.5, size = 0.3) +
  th</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-67-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proTumorRes_Klin$pval.hist</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-68-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2     pVal  pValAdj  Stat   Test
1 PC1 (77.7%) Recurrence 3.78e-10 7.56e-09  8.44 T-test
2  PC2 (8.3%) Recurrence 1.91e-02 1.91e-01 -2.45 T-test</code></pre>
<p>Visualize significant PCs and display their top features with
greatest absolute loadings</p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.dist$PC1</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-70-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proTumorRes_Klin$sig.pc.dist$PC2</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-70-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="normal-samples-2" class="section level2">
<h2>Normal samples</h2>
<pre class="r"><code># Subset certain samples
proNormal_Klin &lt;- proTissue_Klin[, which(colData(proTissue_Klin)$Condition == &#39;Normal&#39;)]
# Prepare feature annotation table
featAnnoTab &lt;- tibble::as_tibble(rowData(proNormal_Klin), rownames = &#39;Protein&#39;) %&gt;%
  dplyr::rename(Gene = PG.Genes) %&gt;%
  tibble::column_to_rownames(&#39;Protein&#39;)
# Perform single-omics analysis
# proNormalRes_Klin &lt;- doSingleOmicsAnalysis(proNormal_Klin, soa_metaVar = &#39;Recurrence&#39;, num_sigFeats = 6,
#                                            pca_method = &#39;ppca&#39;, num_PCs = 20, num_PCfeats = 30,
#                                            use_limma = F, use_proDA = T, feat_anno = featAnnoTab,
#                                            feat_conv = T, plot_title = NULL, show_rownames = F,
#                                            pca_metaVar = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
#                                                            &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Batch&#39;),
#                                            soa_unwantVar = &#39;Stage&#39;,
#                                            level_metaVar = c(&#39;Yes&#39;, &#39;No&#39;))
# saveRDS(proNormalRes_Klin, &#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_Stage_Old.rds&#39;)
proNormalRes_Klin &lt;- readRDS(&#39;./data/Discovery/AG_Klingmueller/soaRes/proNormalRes_Stage_Old.rds&#39;)</code></pre>
<p>Display significant associations between PCs (Var1) and sample
metadata variables (Var2) to overview sources of variance in data</p>
<pre class="r"><code>proNormalRes_Klin$data.var.source</code></pre>
<pre><code>          Var1       Var2    pVal pValAdj   Stat        Test
1  PC14 (2.9%)      Batch 0.00740   0.386 -2.830      T-test
2   PC8 (5.4%)     Gender 0.00945   0.386 -2.740      T-test
3  PC20 (1.8%)      Stage 0.00982   0.386  5.270       ANOVA
4  PC11 (3.6%)        Age 0.01500   0.386  0.387 Correlation
5   PC7 (5.7%)     Gender 0.01860   0.386  2.460      T-test
6   PC5 (6.9%) Recurrence 0.01930   0.386 -2.450      T-test
7  PC20 (1.8%)      Batch 0.02800   0.437 -2.290      T-test
8  PC17 (2.3%)    Smoking 0.02920   0.437  3.910       ANOVA
9  PC2 (10.6%)      Stage 0.03790   0.450  3.590       ANOVA
10  PC7 (5.7%)        Age 0.04430   0.450  0.324 Correlation
11 PC16 (2.4%)     Gender 0.04490   0.450 -2.080      T-test
12 PC14 (2.9%) Recurrence 0.04500   0.450  2.080      T-test</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>dplyr::select(proNormalRes_Klin$sig.feat.tab, -Test)</code></pre>
<pre><code>                  Var1       Var2     pVal pValAdj  Stat               Gene
1               Q9Y6Y8 Recurrence 1.73e-05   0.101  4.98            SEC23IP
2               P63104 Recurrence 2.46e-05   0.101 -4.86              YWHAZ
3               O95865 Recurrence 6.79e-05   0.101 -4.52              DDAH2
4               Q9NVZ3 Recurrence 7.26e-05   0.101  4.50             NECAP2
5               P07951 Recurrence 7.57e-05   0.101 -4.48               TPM2
6               P31946 Recurrence 8.14e-05   0.101 -4.46              YWHAB
7               Q9NSC5 Recurrence 9.45e-05   0.101 -4.41             HOMER3
8               P09493 Recurrence 9.84e-05   0.101 -4.39               TPM1
9               Q96LZ7 Recurrence 1.22e-04   0.111  4.32              RMDN2
10              P41208 Recurrence 1.52e-04   0.125 -4.25              CETN2
11              Q8NI36 Recurrence 1.71e-04   0.128  4.21              WDR36
12              Q9Y512 Recurrence 2.08e-04   0.134  4.14             SAMM50
13              Q5S007 Recurrence 2.11e-04   0.134  4.13              LRRK2
14              Q13190 Recurrence 2.93e-04   0.144  4.02               STX5
15              Q96E40 Recurrence 2.99e-04   0.144 -4.01             SPACA9
16              P84243 Recurrence 3.17e-04   0.144 -3.99              H3-3B
17              Q9Y5X2 Recurrence 3.24e-04   0.144  3.99               SNX8
18              O75818 Recurrence 3.60e-04   0.144  3.95              RPP40
19              Q14137 Recurrence 3.66e-04   0.144  3.94               BOP1
20              Q12923 Recurrence 3.79e-04   0.144  3.93             PTPN13
21              P08294 Recurrence 3.81e-04   0.144 -3.93               SOD3
22              Q16637 Recurrence 4.15e-04   0.144  3.90               SMN2
23              P21810 Recurrence 4.23e-04   0.144 -3.89                BGN
24              Q96C36 Recurrence 4.23e-04   0.144  3.89              PYCR2
25              Q8IUX7 Recurrence 4.64e-04   0.144 -3.86              AEBP1
26              Q08554 Recurrence 4.68e-04   0.144 -3.86               DSC1
27              P62736 Recurrence 4.92e-04   0.144 -3.84              ACTA2
28              P11441 Recurrence 5.06e-04   0.144 -3.83              UBL4A
29              O94855 Recurrence 5.20e-04   0.144  3.82             SEC24D
30              Q10570 Recurrence 5.25e-04   0.144  3.82              CPSF1
31              Q9H936 Recurrence 5.99e-04   0.155  3.77           SLC25A22
32              Q9BZE4 Recurrence 6.42e-04   0.155  3.75             GTPBP4
33              Q5SY16 Recurrence 6.61e-04   0.155  3.74               NOL9
34              P02649 Recurrence 6.66e-04   0.155 -3.74               APOE
35              Q13464 Recurrence 6.79e-04   0.155 -3.73              ROCK1
36              P17302 Recurrence 6.93e-04   0.155 -3.72               GJA1
37              P02538 Recurrence 7.10e-04   0.155 -3.71              KRT6A
38              P02533 Recurrence 7.22e-04   0.155 -3.71              KRT14
39              P07205 Recurrence 7.32e-04   0.155  3.70               PGK2
40              Q8TDB4 Recurrence 8.29e-04   0.160 -3.66              MGARP
41              P55036 Recurrence 8.37e-04   0.160 -3.65              PSMD4
42              Q8NBQ5 Recurrence 8.47e-04   0.160  3.65           HSD17B11
43              Q05823 Recurrence 8.51e-04   0.160  3.65             RNASEL
44              Q9UJ83 Recurrence 8.79e-04   0.160  3.64              HACL1
45              Q92851 Recurrence 8.86e-04   0.160  3.63             CASP10
46              Q05682 Recurrence 8.93e-04   0.160 -3.63              CALD1
47              Q15036 Recurrence 9.44e-04   0.166  3.61              SNX17
48              Q9H7X7 Recurrence 1.03e-03   0.168 -3.58              IFT22
49              Q9Y2T2 Recurrence 1.03e-03   0.168  3.58              AP3M1
50              Q14BN4 Recurrence 1.07e-03   0.168 -3.57              SLMAP
51              Q9Y679 Recurrence 1.07e-03   0.168  3.57               AUP1
52              Q9UH62 Recurrence 1.15e-03   0.168  3.54             ARMCX3
53              Q9Y6D6 Recurrence 1.15e-03   0.168  3.54            ARFGEF1
54              Q6KB66 Recurrence 1.16e-03   0.168 -3.54              KRT80
55              Q9UNE7 Recurrence 1.16e-03   0.168 -3.54              STUB1
56              Q6P9B9 Recurrence 1.21e-03   0.168  3.52              INTS5
57              Q99497 Recurrence 1.22e-03   0.168 -3.52              PARK7
58              Q9H9H4 Recurrence 1.24e-03   0.168 -3.51             VPS37B
59              Q6P3W7 Recurrence 1.28e-03   0.168  3.50              SCYL2
60              Q8IXQ6 Recurrence 1.30e-03   0.168  3.50              PARP9
61              Q6IAN0 Recurrence 1.35e-03   0.168  3.48             DHRS7B
62              Q6P4A7 Recurrence 1.35e-03   0.168  3.48              SFXN4
63              P49810 Recurrence 1.37e-03   0.168  3.48              PSEN2
64              P09917 Recurrence 1.40e-03   0.168  3.47              ALOX5
65              Q5TZA2 Recurrence 1.42e-03   0.168 -3.46              CROCC
66              P52789 Recurrence 1.45e-03   0.168  3.46                HK2
67              P63173 Recurrence 1.46e-03   0.168  3.46              RPL38
68              Q8IX04 Recurrence 1.46e-03   0.168  3.45              UEVLD
69              A0AV96 Recurrence 1.47e-03   0.168  3.45              RBM47
70              P01700 Recurrence 1.49e-03   0.168 -3.45           IGLV1-47
71              O14972 Recurrence 1.53e-03   0.168  3.44             VPS26C
72              Q9UGP8 Recurrence 1.58e-03   0.168  3.43              SEC63
73              P19878 Recurrence 1.59e-03   0.168  3.42               NCF2
74              P24844 Recurrence 1.60e-03   0.168 -3.42               MYL9
75              Q9Y4C5 Recurrence 1.67e-03   0.168 -3.41              CHST2
76              Q9Y490 Recurrence 1.70e-03   0.168 -3.40               TLN1
77              Q0ZGT2 Recurrence 1.72e-03   0.168 -3.39               NEXN
78              P10909 Recurrence 1.74e-03   0.168 -3.39                CLU
79              Q99816 Recurrence 1.76e-03   0.168 -3.39             TSG101
80              Q9Y3A3 Recurrence 1.76e-03   0.168  3.39               MOB4
81              O60825 Recurrence 1.81e-03   0.168  3.38             PFKFB2
82              Q96DI7 Recurrence 1.81e-03   0.168  3.38            SNRNP40
83              Q9H4M9 Recurrence 1.82e-03   0.168 -3.38               EHD1
84              Q12955 Recurrence 1.83e-03   0.168  3.37               ANK3
85              P08240 Recurrence 1.84e-03   0.168  3.37              SRPRA
86              Q14203 Recurrence 1.84e-03   0.168 -3.37              DCTN1
87              Q03426 Recurrence 1.85e-03   0.168  3.37                MVK
88              Q9BQ13 Recurrence 1.85e-03   0.168  3.37             KCTD14
89              P53992 Recurrence 1.89e-03   0.168  3.36             SEC24C
90              Q9NXN4 Recurrence 1.89e-03   0.168  3.36              GDAP2
91              O94979 Recurrence 1.91e-03   0.168  3.36             SEC31A
92              O60343 Recurrence 1.92e-03   0.168  3.35             TBC1D4
93              P61966 Recurrence 1.92e-03   0.168  3.35              AP1S1
94              Q9Y5U9 Recurrence 1.93e-03   0.168  3.35            IER3IP1
95              P15311 Recurrence 1.94e-03   0.168 -3.35                EZR
96              Q9UMY4 Recurrence 1.97e-03   0.169 -3.35              SNX12
97              Q96KG9 Recurrence 2.01e-03   0.171  3.34              SCYL1
98              Q92621 Recurrence 2.08e-03   0.172  3.33             NUP205
99              Q6UWP8 Recurrence 2.09e-03   0.172 -3.32               SBSN
100             Q12996 Recurrence 2.10e-03   0.172  3.32              CSTF3
101             O14656 Recurrence 2.14e-03   0.172  3.31              TOR1A
102             P53618 Recurrence 2.15e-03   0.172  3.31              COPB1
103             O14874 Recurrence 2.18e-03   0.172  3.31              BCKDK
104             O96008 Recurrence 2.19e-03   0.172  3.31             TOMM40
105             Q8WVV4 Recurrence 2.19e-03   0.172 -3.31              POF1B
106             O15382 Recurrence 2.24e-03   0.174  3.30              BCAT2
107             P04259 Recurrence 2.28e-03   0.176 -3.29              KRT6B
108             O76024 Recurrence 2.36e-03   0.179 -3.28               WFS1
109             Q14118 Recurrence 2.38e-03   0.179 -3.28               DAG1
110             P18206 Recurrence 2.41e-03   0.179 -3.27                VCL
111             Q96CV9 Recurrence 2.41e-03   0.179 -3.27               OPTN
112             Q9UG63 Recurrence 2.49e-03   0.180  3.26              ABCF2
113             Q6ZT21 Recurrence 2.52e-03   0.180  3.25              TMPPE
114             O60603 Recurrence 2.56e-03   0.180  3.25               TLR2
115             P09382 Recurrence 2.59e-03   0.180 -3.25             LGALS1
116             P10620 Recurrence 2.64e-03   0.180  3.24              MGST1
117             Q9H6T0 Recurrence 2.65e-03   0.180  3.24              ESRP2
118             Q07617 Recurrence 2.67e-03   0.180 -3.23              SPAG1
119             Q96EQ0 Recurrence 2.74e-03   0.180 -3.22               SGTB
120             P62875 Recurrence 2.78e-03   0.180  3.22             POLR2L
121             P17980 Recurrence 2.81e-03   0.180 -3.21              PSMC3
122             Q01995 Recurrence 2.83e-03   0.180 -3.21              TAGLN
123             Q96A00 Recurrence 2.84e-03   0.180 -3.21           PPP1R14A
124             A6NI72 Recurrence 2.85e-03   0.180  3.21              NCF1B
125             Q9NWV4 Recurrence 2.86e-03   0.180 -3.21               CZIB
126             Q9BXP2 Recurrence 2.87e-03   0.180  3.21            SLC12A9
127             Q96GY0 Recurrence 2.88e-03   0.180 -3.21            ZC2HC1A
128             P21333 Recurrence 2.91e-03   0.180 -3.20               FLNA
129             P53609 Recurrence 2.92e-03   0.180 -3.20             PGGT1B
130             Q92629 Recurrence 2.93e-03   0.180 -3.20               SGCD
131             O95639 Recurrence 2.94e-03   0.180  3.20              CPSF4
132             P05141 Recurrence 2.94e-03   0.180  3.20            SLC25A5
133             Q69YQ0 Recurrence 2.94e-03   0.180 -3.20            SPECC1L
134             Q969V3 Recurrence 2.94e-03   0.180  3.20               NCLN
135             Q99733 Recurrence 2.94e-03   0.180 -3.20             NAP1L4
136             Q9UKD2 Recurrence 2.99e-03   0.181  3.19              MRTO4
137             Q13561 Recurrence 3.04e-03   0.181 -3.18              DCTN2
138             Q8IYU8 Recurrence 3.06e-03   0.181  3.18              MICU2
139             P02511 Recurrence 3.11e-03   0.181 -3.18              CRYAB
140             P17661 Recurrence 3.15e-03   0.181 -3.17                DES
141             P50502 Recurrence 3.16e-03   0.181 -3.17               ST13
142             O60762 Recurrence 3.18e-03   0.181  3.17               DPM1
143             Q9UBX3 Recurrence 3.22e-03   0.181  3.16           SLC25A10
144             A1A4S6 Recurrence 3.30e-03   0.181  3.15           ARHGAP10
145             O00233 Recurrence 3.32e-03   0.181 -3.15              PSMD9
146             P55160 Recurrence 3.34e-03   0.181  3.15            NCKAP1L
147             P14649 Recurrence 3.35e-03   0.181 -3.15              MYL6B
148             Q13409 Recurrence 3.35e-03   0.181 -3.15            DYNC1I2
149             Q9Y487 Recurrence 3.35e-03   0.181  3.15           ATP6V0A2
150             P23469 Recurrence 3.36e-03   0.181  3.15              PTPRE
151             P51911 Recurrence 3.38e-03   0.181 -3.14               CNN1
152             Q9UPU5 Recurrence 3.41e-03   0.181  3.14              USP24
153             Q9H267 Recurrence 3.43e-03   0.181  3.14             VPS33B
154             P62258 Recurrence 3.44e-03   0.181 -3.14              YWHAE
155             Q6NXT1 Recurrence 3.45e-03   0.181 -3.14            ANKRD54
156             Q8N1F8 Recurrence 3.48e-03   0.181  3.13            STK11IP
157             Q14515 Recurrence 3.49e-03   0.181 -3.13            SPARCL1
158             Q9NXJ5 Recurrence 3.50e-03   0.181  3.13             PGPEP1
159             P17066 Recurrence 3.54e-03   0.181  3.13              HSPA6
160             P62937 Recurrence 3.56e-03   0.181 -3.13               PPIA
161             O60941 Recurrence 3.61e-03   0.181 -3.12               DTNB
162             Q14258 Recurrence 3.61e-03   0.181  3.12             TRIM25
163             Q9NXC5 Recurrence 3.61e-03   0.181  3.12               MIOS
164             Q15437 Recurrence 3.64e-03   0.181  3.12             SEC23B
165             Q8N5G0 Recurrence 3.67e-03   0.181  3.11             SMIM20
166             Q53S58 Recurrence 3.69e-03   0.181  3.11            TMEM177
167             Q9Y6I3 Recurrence 3.69e-03   0.181 -3.11               EPN1
168             Q14376 Recurrence 3.71e-03   0.181  3.11               GALE
169             P15924 Recurrence 3.72e-03   0.181 -3.11                DSP
170             Q96P48 Recurrence 3.77e-03   0.181  3.10              ARAP1
171             Q9Y5Z4 Recurrence 3.79e-03   0.181 -3.10              HEBP2
172             O60264 Recurrence 3.80e-03   0.181  3.10            SMARCA5
173             Q68DK7 Recurrence 3.80e-03   0.181 -3.10               MSL1
174             Q9Y2V7 Recurrence 3.84e-03   0.182  3.10               COG6
175             Q9P2D1 Recurrence 3.88e-03   0.182 -3.09               CHD7
176             Q5TH69 Recurrence 3.89e-03   0.182  3.09            ARFGEF3
177             Q7Z3U7 Recurrence 3.92e-03   0.182  3.09               MON2
178             Q7L014 Recurrence 3.94e-03   0.182  3.09              DDX46
179             O60341 Recurrence 3.96e-03   0.182  3.08              KDM1A
180             Q99417 Recurrence 3.98e-03   0.182 -3.08              MYCBP
181             Q9NVM9 Recurrence 4.01e-03   0.182  3.08             INTS13
182             Q8N131 Recurrence 4.04e-03   0.182 -3.08            TMEM123
183             Q96BR1 Recurrence 4.04e-03   0.182  3.08               SGK3
184             Q9NZT1 Recurrence 4.13e-03   0.185 -3.07             CALML5
185             Q8N766 Recurrence 4.16e-03   0.185  3.07               EMC1
186             O15254 Recurrence 4.25e-03   0.187  3.06              ACOX3
187             Q06828 Recurrence 4.26e-03   0.187 -3.06               FMOD
188             O94925 Recurrence 4.27e-03   0.187  3.06                GLS
189             Q96JB2 Recurrence 4.29e-03   0.187  3.05               COG3
190             Q7Z794 Recurrence 4.42e-03   0.192 -3.04              KRT77
191             P31483 Recurrence 4.46e-03   0.192  3.04               TIA1
192             P78559 Recurrence 4.46e-03   0.192 -3.04              MAP1A
193             Q9Y696 Recurrence 4.51e-03   0.192 -3.04              CLIC4
194             Q8IY95 Recurrence 4.53e-03   0.192  3.03            TMEM192
195             O00267 Recurrence 4.59e-03   0.194  3.03             SUPT5H
196             P55209 Recurrence 4.65e-03   0.195 -3.02             NAP1L1
197             P30837 Recurrence 4.67e-03   0.195 -3.02            ALDH1B1
198             Q9Y3A4 Recurrence 4.67e-03   0.195  3.02              RRP7A
199             P42167 Recurrence 4.76e-03   0.196 -3.02               TMPO
200             Q9Y6A9 Recurrence 4.77e-03   0.196  3.01              SPCS1
201             P35052 Recurrence 4.81e-03   0.196 -3.01               GPC1
202             P62333 Recurrence 4.87e-03   0.196 -3.01              PSMC6
203             O75190 Recurrence 4.88e-03   0.196 -3.01             DNAJB6
204             Q9HCE6 Recurrence 4.88e-03   0.196  3.01          ARHGEF10L
205             O43556 Recurrence 4.91e-03   0.196 -3.00               SGCE
206         A0A0B4J1U3 Recurrence 4.93e-03   0.196 -3.00           IGLV1-36
207             Q86WT1 Recurrence 4.96e-03   0.196 -3.00             TTC30A
208             P04632 Recurrence 4.98e-03   0.196 -3.00             CAPNS1
209             Q7Z6M1 Recurrence 5.01e-03   0.196 -3.00             RABEPK
210             P49754 Recurrence 5.03e-03   0.196  2.99              VPS41
211             P62191 Recurrence 5.04e-03   0.196 -2.99              PSMC1
212             Q8TC05 Recurrence 5.04e-03   0.196 -2.99               MDM1
213             Q6S8J3 Recurrence 5.26e-03   0.201 -2.98              POTEE
214             Q92600 Recurrence 5.27e-03   0.201  2.98              CNOT9
215             Q9HD20 Recurrence 5.27e-03   0.201  2.98            ATP13A1
216             Q9UPN3 Recurrence 5.28e-03   0.201 -2.97              MACF1
217             P01701 Recurrence 5.31e-03   0.201 -2.97           IGLV1-51
218             P13797 Recurrence 5.35e-03   0.201 -2.97               PLS3
219             P58340 Recurrence 5.36e-03   0.201 -2.97               MLF1
220             Q9BUN8 Recurrence 5.38e-03   0.201  2.97              DERL1
221             Q9HAN9 Recurrence 5.39e-03   0.201  2.97             NMNAT1
222             Q16822 Recurrence 5.43e-03   0.201  2.96               PCK2
223             Q9Y5Q8 Recurrence 5.43e-03   0.201  2.96             GTF3C5
224             Q16825 Recurrence 5.46e-03   0.201 -2.96             PTPN21
225      P0DMV8;P0DMV9 Recurrence 5.52e-03   0.202 -2.96      HSPA1A;HSPA1B
226             Q9NQP4 Recurrence 5.63e-03   0.203 -2.95              PFDN4
227             P21953 Recurrence 5.65e-03   0.203  2.95             BCKDHB
228             O60237 Recurrence 5.66e-03   0.203 -2.95           PPP1R12B
229             Q8NBJ5 Recurrence 5.68e-03   0.203  2.95           COLGALT1
230             Q9NQ48 Recurrence 5.80e-03   0.203 -2.94             LZTFL1
231             Q8TEB9 Recurrence 5.82e-03   0.203 -2.94             RHBDD1
232             P22897 Recurrence 5.83e-03   0.203  2.94               MRC1
233             P55212 Recurrence 5.85e-03   0.203  2.94              CASP6
234             P33121 Recurrence 5.87e-03   0.203  2.93              ACSL1
235             O75368 Recurrence 5.89e-03   0.203 -2.93            SH3BGRL
236             B0I1T2 Recurrence 5.90e-03   0.203  2.93              MYO1G
237             P22061 Recurrence 5.92e-03   0.203 -2.93              PCMT1
238             Q9NTZ6 Recurrence 5.92e-03   0.203  2.93              RBM12
239             P12277 Recurrence 5.97e-03   0.203 -2.93                CKB
240             P51659 Recurrence 5.98e-03   0.203  2.93            HSD17B4
241             Q9NP74 Recurrence 5.98e-03   0.203 -2.93              PALMD
242             O95678 Recurrence 6.01e-03   0.203 -2.93              KRT75
243             Q49A26 Recurrence 6.05e-03   0.203  2.92              GLYR1
244             Q9NYH9 Recurrence 6.05e-03   0.203  2.92               UTP6
245             P32321 Recurrence 6.06e-03   0.203  2.92               DCTD
246             P18085 Recurrence 6.07e-03   0.203  2.92               ARF4
247             Q9Y6I8 Recurrence 6.09e-03   0.203  2.92              PXMP4
248             P30086 Recurrence 6.17e-03   0.204 -2.91              PEBP1
249             Q9Y608 Recurrence 6.21e-03   0.204 -2.91            LRRFIP2
250             Q12929 Recurrence 6.22e-03   0.204  2.91               EPS8
251             Q2TAA2 Recurrence 6.28e-03   0.204 -2.91               IAH1
252             P04406 Recurrence 6.32e-03   0.204 -2.91              GAPDH
253             Q9H8L6 Recurrence 6.33e-03   0.204 -2.90              MMRN2
254             Q86T03 Recurrence 6.37e-03   0.204  2.90             PIP4P1
255             P06727 Recurrence 6.39e-03   0.204 -2.90              APOA4
256             Q9BUK0 Recurrence 6.41e-03   0.204  2.90             CHCHD7
257             A6NJ78 Recurrence 6.42e-03   0.204  2.90            METTL15
258             P48729 Recurrence 6.44e-03   0.204  2.90            CSNK1A1
259             Q6NY19 Recurrence 6.46e-03   0.204 -2.90              KANK3
260             Q9BQI0 Recurrence 6.46e-03   0.204 -2.90              AIF1L
261             P30101 Recurrence 6.48e-03   0.204 -2.90              PDIA3
262             Q86U38 Recurrence 6.52e-03   0.204  2.89               NOP9
263             P13647 Recurrence 6.56e-03   0.204 -2.89               KRT5
264             P60660 Recurrence 6.56e-03   0.204 -2.89               MYL6
265             P54886 Recurrence 6.57e-03   0.204  2.89           ALDH18A1
266             Q8WXX5 Recurrence 6.57e-03   0.204 -2.89             DNAJC9
267             P11532 Recurrence 6.64e-03   0.204 -2.89                DMD
268             Q9NVP1 Recurrence 6.64e-03   0.204  2.89              DDX18
269             Q9NX20 Recurrence 6.67e-03   0.204  2.88             MRPL16
270             P19447 Recurrence 6.69e-03   0.204  2.88              ERCC3
271             Q5SNT2 Recurrence 6.71e-03   0.204  2.88            TMEM201
272             Q9Y277 Recurrence 6.77e-03   0.205  2.88              VDAC3
273             Q6KCM7 Recurrence 6.86e-03   0.207  2.87           SLC25A25
274             O00533 Recurrence 6.87e-03   0.207 -2.87               CHL1
275             Q9Y2X3 Recurrence 6.98e-03   0.208  2.87              NOP58
276             O75879 Recurrence 7.03e-03   0.208 -2.86               GATB
277             Q8IZ81 Recurrence 7.04e-03   0.208  2.86             ELMOD2
278             P06753 Recurrence 7.06e-03   0.208 -2.86               TPM3
279             Q9UPN9 Recurrence 7.07e-03   0.208  2.86             TRIM33
280             P01703 Recurrence 7.11e-03   0.208 -2.86           IGLV1-40
281             Q9Y266 Recurrence 7.12e-03   0.208 -2.86               NUDC
282             P67936 Recurrence 7.13e-03   0.208 -2.86               TPM4
283             P98194 Recurrence 7.13e-03   0.208  2.86             ATP2C1
284             Q8NBJ7 Recurrence 7.23e-03   0.209 -2.85              SUMF2
285             P61326 Recurrence 7.26e-03   0.209 -2.85              MAGOH
286             Q9NRG9 Recurrence 7.26e-03   0.209  2.85               AAAS
287             Q9BW83 Recurrence 7.28e-03   0.209 -2.85              IFT27
288             O60760 Recurrence 7.32e-03   0.210  2.85              HPGDS
289             O15258 Recurrence 7.34e-03   0.210  2.85               RER1
290             Q9Y6R1 Recurrence 7.50e-03   0.211  2.84             SLC4A4
291             P55290 Recurrence 7.51e-03   0.211 -2.84              CDH13
292             Q7Z3J2 Recurrence 7.52e-03   0.211  2.84             VPS35L
293             Q8WTW3 Recurrence 7.54e-03   0.211  2.84               COG1
294             Q15629 Recurrence 7.55e-03   0.211  2.84              TRAM1
295             Q9NPH3 Recurrence 7.56e-03   0.211 -2.84             IL1RAP
296             O14832 Recurrence 7.61e-03   0.211  2.83               PHYH
297             P02647 Recurrence 7.61e-03   0.211 -2.83              APOA1
298             O14683 Recurrence 7.63e-03   0.211  2.83            TP53I11
299      O14950;P19105 Recurrence 7.78e-03   0.215 -2.82      MYL12B;MYL12A
300             P17544 Recurrence 7.81e-03   0.215  2.82               ATF7
301             P0DTE1 Recurrence 7.90e-03   0.215 -2.82         IGHV3-38-3
302             O75436 Recurrence 7.94e-03   0.215  2.82             VPS26A
303             Q587I9 Recurrence 7.95e-03   0.215  2.82             SFT2D3
304             P07332 Recurrence 8.00e-03   0.215  2.81                FES
305             Q96P11 Recurrence 8.00e-03   0.215  2.81              NSUN5
306             P30876 Recurrence 8.04e-03   0.215  2.81             POLR2B
307             Q13813 Recurrence 8.05e-03   0.215 -2.81             SPTAN1
308             Q96PE3 Recurrence 8.06e-03   0.215  2.81             INPP4A
309             Q9Y281 Recurrence 8.07e-03   0.215 -2.81               CFL2
310             O60711 Recurrence 8.09e-03   0.215  2.81               LPXN
311             P35241 Recurrence 8.11e-03   0.215 -2.81                RDX
312             Q9Y606 Recurrence 8.15e-03   0.215  2.81               PUS1
313             P42892 Recurrence 8.24e-03   0.217 -2.80               ECE1
314             Q15436 Recurrence 8.30e-03   0.218  2.80             SEC23A
315             Q3KQZ1 Recurrence 8.33e-03   0.218  2.80           SLC25A35
316             Q9H7B2 Recurrence 8.38e-03   0.218  2.79               RPF2
317             Q6P996 Recurrence 8.39e-03   0.218  2.79             PDXDC1
318             Q8IWP9 Recurrence 8.39e-03   0.218 -2.79            CCDC28A
319             P50570 Recurrence 8.43e-03   0.218  2.79               DNM2
320             Q9NRP2 Recurrence 8.54e-03   0.218  2.79               CMC2
321             Q496M5 Recurrence 8.56e-03   0.218 -2.79               PLK5
322             Q15404 Recurrence 8.58e-03   0.218 -2.79               RSU1
323             O14526 Recurrence 8.59e-03   0.218 -2.78              FCHO1
324             Q8NI60 Recurrence 8.60e-03   0.218  2.78              COQ8A
325             Q9BR39 Recurrence 8.60e-03   0.218 -2.78               JPH2
326             O43865 Recurrence 8.63e-03   0.218  2.78             AHCYL1
327             Q9Y6D5 Recurrence 8.69e-03   0.219  2.78            ARFGEF2
328             P54289 Recurrence 8.81e-03   0.222 -2.77           CACNA2D1
329             Q8TDR0 Recurrence 8.85e-03   0.222 -2.77           TRAF3IP1
330             P08590 Recurrence 8.90e-03   0.223 -2.77               MYL3
331             Q92900 Recurrence 8.96e-03   0.223  2.77               UPF1
332             O00443 Recurrence 8.97e-03   0.223  2.77            PIK3C2A
333             Q9BZF1 Recurrence 8.99e-03   0.223  2.77             OSBPL8
334             Q96RU3 Recurrence 9.06e-03   0.223 -2.76              FNBP1
335             Q9BZF9 Recurrence 9.06e-03   0.223 -2.76               UACA
336             Q5D862 Recurrence 9.18e-03   0.223 -2.76               FLG2
337             Q9NZB2 Recurrence 9.23e-03   0.223  2.76            FAM120A
338             Q9NRA2 Recurrence 9.27e-03   0.223  2.75            SLC17A5
339             Q5M775 Recurrence 9.36e-03   0.223 -2.75             SPECC1
340             Q93050 Recurrence 9.37e-03   0.223  2.75           ATP6V0A1
341             Q96CM8 Recurrence 9.39e-03   0.223  2.75              ACSF2
342             Q96GC9 Recurrence 9.41e-03   0.223  2.75               VMP1
343             P49406 Recurrence 9.47e-03   0.223  2.75             MRPL19
344             Q8WZA0 Recurrence 9.47e-03   0.223 -2.75               LZIC
345             O95487 Recurrence 9.48e-03   0.223  2.75             SEC24B
346             O95833 Recurrence 9.48e-03   0.223 -2.75              CLIC3
347             P43686 Recurrence 9.48e-03   0.223 -2.75              PSMC4
348             P38159 Recurrence 9.49e-03   0.223 -2.74               RBMX
349             Q96RQ1 Recurrence 9.50e-03   0.223  2.74             ERGIC2
350             Q92616 Recurrence 9.51e-03   0.223  2.74               GCN1
351             Q9Y2Q0 Recurrence 9.52e-03   0.223  2.74             ATP8A1
352             Q96L92 Recurrence 9.56e-03   0.223  2.74              SNX27
353             O43707 Recurrence 9.57e-03   0.223 -2.74              ACTN4
354             Q16651 Recurrence 9.58e-03   0.223 -2.74              PRSS8
355             P04844 Recurrence 9.61e-03   0.223  2.74               RPN2
356             P30622 Recurrence 9.66e-03   0.223 -2.74              CLIP1
357             P36955 Recurrence 9.68e-03   0.223 -2.74           SERPINF1
358             Q6NSJ5 Recurrence 9.70e-03   0.223  2.74             LRRC8E
359             Q6UVK1 Recurrence 9.79e-03   0.224 -2.73              CSPG4
360             Q9H9B1 Recurrence 9.83e-03   0.224  2.73              EHMT1
361             Q9BQG0 Recurrence 9.86e-03   0.224  2.73            MYBBP1A
362             P39656 Recurrence 9.89e-03   0.224  2.73              DDOST
363             Q9UN36 Recurrence 9.90e-03   0.224 -2.73              NDRG2
364             A0PJW6 Recurrence 9.92e-03   0.224  2.73            TMEM223
365             P12814 Recurrence 9.92e-03   0.224 -2.73              ACTN1
366             Q9H2V7 Recurrence 9.92e-03   0.224  2.73              SPNS1
367             Q9UNZ2 Recurrence 1.00e-02   0.225 -2.72             NSFL1C
368             P12429 Recurrence 1.01e-02   0.226 -2.72              ANXA3
369             Q7KZF4 Recurrence 1.01e-02   0.225  2.72               SND1
370             Q96RQ3 Recurrence 1.01e-02   0.225  2.72              MCCC1
371             Q9NWS8 Recurrence 1.01e-02   0.225  2.72              RMND1
372             Q96BJ3 Recurrence 1.02e-02   0.227 -2.71               AIDA
373             Q6NUK4 Recurrence 1.03e-02   0.227  2.71              REEP3
374             Q9Y3E5 Recurrence 1.03e-02   0.227  2.71              PTRH2
375             P30085 Recurrence 1.04e-02   0.227 -2.71              CMPK1
376             Q01780 Recurrence 1.04e-02   0.227  2.71            EXOSC10
377             Q8IY17 Recurrence 1.04e-02   0.227  2.71             PNPLA6
378             P53680 Recurrence 1.06e-02   0.231  2.70              AP2S1
379             Q9UMS6 Recurrence 1.06e-02   0.231 -2.70             SYNPO2
380             Q5VW38 Recurrence 1.07e-02   0.233  2.70             GPR107
381             Q01082 Recurrence 1.08e-02   0.235 -2.69             SPTBN1
382             Q5VWJ9 Recurrence 1.09e-02   0.235  2.69              SNX30
383             P13498 Recurrence 1.10e-02   0.236  2.68               CYBA
384             Q9H2K8 Recurrence 1.10e-02   0.236  2.69              TAOK3
385             Q9H3E2 Recurrence 1.10e-02   0.236  2.69              SNX25
386             Q9UN79 Recurrence 1.10e-02   0.236 -2.68              SOX13
387             O95816 Recurrence 1.11e-02   0.236 -2.68               BAG2
388             P05067 Recurrence 1.11e-02   0.236 -2.68                APP
389             P54920 Recurrence 1.12e-02   0.237 -2.68               NAPA
390             O00203 Recurrence 1.13e-02   0.239  2.67              AP3B1
391             Q5TD94 Recurrence 1.13e-02   0.238 -2.67             RSPH4A
392             Q6UXB2 Recurrence 1.13e-02   0.238  2.68             CXCL17
393             Q8NI27 Recurrence 1.14e-02   0.240  2.67              THOC2
394             O14974 Recurrence 1.15e-02   0.240 -2.67           PPP1R12A
395             Q9ULX6 Recurrence 1.16e-02   0.242  2.66             AKAP8L
396             P26038 Recurrence 1.17e-02   0.242 -2.66                MSN
397             P50454 Recurrence 1.17e-02   0.242  2.66           SERPINH1
398             Q86Y07 Recurrence 1.17e-02   0.242  2.66               VRK2
399             Q8WXF7 Recurrence 1.17e-02   0.242  2.66               ATL1
400             Q969Q5 Recurrence 1.17e-02   0.242  2.66              RAB24
401             O75781 Recurrence 1.18e-02   0.242 -2.66               PALM
402             Q8TEA8 Recurrence 1.19e-02   0.243 -2.65               DTD1
403             O14569 Recurrence 1.20e-02   0.243  2.65           CYB561D2
404             P04083 Recurrence 1.20e-02   0.243 -2.65              ANXA1
405             P08648 Recurrence 1.20e-02   0.243 -2.65              ITGA5
406             Q96AT9 Recurrence 1.20e-02   0.243 -2.65                RPE
407             Q9HB90 Recurrence 1.20e-02   0.243  2.65              RRAGC
408             Q2M2I3 Recurrence 1.21e-02   0.244  2.65             FAM83E
409             Q6DN90 Recurrence 1.21e-02   0.244  2.65             IQSEC1
410             Q9Y624 Recurrence 1.21e-02   0.244 -2.65               F11R
411             A5YKK6 Recurrence 1.22e-02   0.244  2.64              CNOT1
412             P29728 Recurrence 1.22e-02   0.244  2.64               OAS2
413             Q7Z4W1 Recurrence 1.22e-02   0.244  2.64               DCXR
414             Q99487 Recurrence 1.23e-02   0.245  2.64             PAFAH2
415             P55287 Recurrence 1.24e-02   0.245 -2.64              CDH11
416             Q13601 Recurrence 1.24e-02   0.245  2.64               KRR1
417             Q96QA5 Recurrence 1.24e-02   0.245 -2.64              GSDMA
418             Q9H5N1 Recurrence 1.24e-02   0.245 -2.64             RABEP2
419             P28288 Recurrence 1.25e-02   0.245  2.63              ABCD3
420             Q15080 Recurrence 1.25e-02   0.245  2.63               NCF4
421             Q8NBJ9 Recurrence 1.25e-02   0.245  2.63              SIDT2
422             Q9H270 Recurrence 1.25e-02   0.245  2.63              VPS11
423             O43464 Recurrence 1.26e-02   0.245  2.63              HTRA2
424             Q96IW7 Recurrence 1.26e-02   0.245  2.63             SEC22A
425             O43427 Recurrence 1.27e-02   0.246  2.63               FIBP
426             P41222 Recurrence 1.27e-02   0.246 -2.63              PTGDS
427             Q9NZC7 Recurrence 1.27e-02   0.246 -2.63               WWOX
428             Q13367 Recurrence 1.28e-02   0.246  2.62              AP3B2
429             Q8IXB1 Recurrence 1.28e-02   0.246  2.62            DNAJC10
430             P08729 Recurrence 1.29e-02   0.247 -2.62               KRT7
431             Q02880 Recurrence 1.29e-02   0.247  2.62              TOP2B
432             Q6P2E9 Recurrence 1.30e-02   0.247  2.62               EDC4
433             Q9BZ95 Recurrence 1.30e-02   0.247  2.62               NSD3
434             O15173 Recurrence 1.31e-02   0.247 -2.61             PGRMC2
435             P43405 Recurrence 1.31e-02   0.247  2.61                SYK
436             Q00341 Recurrence 1.31e-02   0.247  2.61              HDLBP
437             P21730 Recurrence 1.32e-02   0.247  2.61              C5AR1
438             P78527 Recurrence 1.32e-02   0.247  2.61              PRKDC
439             Q14435 Recurrence 1.32e-02   0.247  2.61             GALNT3
440             O75312 Recurrence 1.33e-02   0.247 -2.61               ZPR1
441             Q15370 Recurrence 1.33e-02   0.247 -2.61               ELOB
442             Q86Y79 Recurrence 1.33e-02   0.247  2.61              PTRH1
443             Q9Y3E7 Recurrence 1.33e-02   0.247 -2.61              CHMP3
444             Q9Y6E2 Recurrence 1.33e-02   0.247  2.61               BZW2
445             P04843 Recurrence 1.34e-02   0.247  2.60               RPN1
446             P55285 Recurrence 1.34e-02   0.247 -2.61               CDH6
447             Q9BZE1 Recurrence 1.34e-02   0.247  2.60             MRPL37
448             Q9UHB9 Recurrence 1.34e-02   0.247  2.60              SRP68
449             O00154 Recurrence 1.35e-02   0.247  2.60              ACOT7
450             O00750 Recurrence 1.35e-02   0.247 -2.60            PIK3C2B
451             O43765 Recurrence 1.35e-02   0.247 -2.60               SGTA
452             P54198 Recurrence 1.35e-02   0.247  2.60               HIRA
453             P49792 Recurrence 1.36e-02   0.247  2.60             RANBP2
454             Q96JC1 Recurrence 1.36e-02   0.247  2.60              VPS39
455             Q13084 Recurrence 1.37e-02   0.247  2.60             MRPL28
456             Q96LR5 Recurrence 1.37e-02   0.247  2.60             UBE2E2
457             P40123 Recurrence 1.38e-02   0.248 -2.59               CAP2
458             O60684 Recurrence 1.39e-02   0.248  2.59              KPNA6
459             P08134 Recurrence 1.39e-02   0.248 -2.59               RHOC
460             Q96AA3 Recurrence 1.39e-02   0.248  2.59               RFT1
461             Q9H063 Recurrence 1.39e-02   0.248 -2.59               MAF1
462             Q9UBY9 Recurrence 1.39e-02   0.248 -2.59              HSPB7
463             Q9H5Q4 Recurrence 1.40e-02   0.248 -2.59              TFB2M
464             Q9HC38 Recurrence 1.40e-02   0.248 -2.59              GLOD4
465             Q9Y312 Recurrence 1.40e-02   0.248  2.59               AAR2
466             P53602 Recurrence 1.41e-02   0.248  2.59                MVD
467             Q13107 Recurrence 1.41e-02   0.249  2.58               USP4
468             Q15642 Recurrence 1.41e-02   0.248 -2.59             TRIP10
469             Q15848 Recurrence 1.42e-02   0.249 -2.58             ADIPOQ
470             Q96AE7 Recurrence 1.42e-02   0.249 -2.58              TTC17
471             P42858 Recurrence 1.43e-02   0.250  2.58                HTT
472             Q9Y2S7 Recurrence 1.43e-02   0.250  2.58            POLDIP2
473             O75427 Recurrence 1.44e-02   0.251  2.58              LRCH4
474             P13473 Recurrence 1.46e-02   0.251  2.57              LAMP2
475             Q709C8 Recurrence 1.46e-02   0.251  2.57             VPS13C
476             Q8NB16 Recurrence 1.46e-02   0.251  2.57               MLKL
477             Q8NE86 Recurrence 1.46e-02   0.251  2.57                MCU
478             Q96M27 Recurrence 1.46e-02   0.251  2.57              PRRC1
479             Q99571 Recurrence 1.46e-02   0.251  2.57              P2RX4
480             O75367 Recurrence 1.47e-02   0.251  2.57          MACROH2A1
481             P63165 Recurrence 1.47e-02   0.251 -2.57              SUMO1
482             Q13488 Recurrence 1.47e-02   0.251  2.57             TCIRG1
483             Q99732 Recurrence 1.47e-02   0.251 -2.57              LITAF
484             Q9UKM7 Recurrence 1.47e-02   0.251  2.57             MAN1B1
485             Q8TC71 Recurrence 1.48e-02   0.252 -2.56            SPATA18
486             Q12899 Recurrence 1.49e-02   0.253  2.56             TRIM26
487             O15357 Recurrence 1.50e-02   0.253  2.56             INPPL1
488             P04156 Recurrence 1.50e-02   0.253 -2.56               PRNP
489             P16949 Recurrence 1.50e-02   0.253 -2.56              STMN1
490             P61586 Recurrence 1.50e-02   0.253 -2.56               RHOA
491             P01111 Recurrence 1.52e-02   0.254 -2.55               NRAS
492             Q96GQ5 Recurrence 1.52e-02   0.254  2.55              RUSF1
493             Q9BV10 Recurrence 1.52e-02   0.254  2.55              ALG12
494             O95219 Recurrence 1.53e-02   0.254  2.55               SNX4
495             P21796 Recurrence 1.53e-02   0.254  2.55              VDAC1
496             P58546 Recurrence 1.53e-02   0.254 -2.55               MTPN
497             Q8N1F7 Recurrence 1.53e-02   0.254  2.55              NUP93
498             Q96HV5 Recurrence 1.53e-02   0.254  2.55            TMEM41A
499             P57740 Recurrence 1.54e-02   0.254  2.55             NUP107
500             P68133 Recurrence 1.54e-02   0.254 -2.55              ACTA1
501             Q8N3C0 Recurrence 1.54e-02   0.254  2.55              ASCC3
502             P51884 Recurrence 1.55e-02   0.255 -2.54                LUM
503             Q9BZL4 Recurrence 1.55e-02   0.255 -2.55           PPP1R12C
504             P33151 Recurrence 1.56e-02   0.255 -2.54               CDH5
505             P51888 Recurrence 1.56e-02   0.255 -2.54              PRELP
506             Q9UDT6 Recurrence 1.57e-02   0.255 -2.54              CLIP2
507             Q9Y365 Recurrence 1.57e-02   0.255  2.54            STARD10
508             Q13085 Recurrence 1.58e-02   0.256  2.54              ACACA
509             Q5T7N3 Recurrence 1.58e-02   0.256 -2.54              KANK4
510             Q99747 Recurrence 1.58e-02   0.256 -2.54               NAPG
511             P51812 Recurrence 1.59e-02   0.256  2.53            RPS6KA3
512             P16118 Recurrence 1.60e-02   0.256  2.53             PFKFB1
513             Q9UBC5 Recurrence 1.60e-02   0.256 -2.53              MYO1A
514             Q76N32 Recurrence 1.61e-02   0.256 -2.53              CEP68
515             Q8NES3 Recurrence 1.61e-02   0.256 -2.53               LFNG
516             Q9UP83 Recurrence 1.61e-02   0.256  2.53               COG5
517             O94776 Recurrence 1.62e-02   0.256  2.53               MTA2
518             P17405 Recurrence 1.62e-02   0.256  2.53              SMPD1
519             P46977 Recurrence 1.62e-02   0.256  2.53              STT3A
520             Q8TBX8 Recurrence 1.62e-02   0.256  2.53            PIP4K2C
521             Q9BWS9 Recurrence 1.62e-02   0.256  2.53              CHID1
522             O75817 Recurrence 1.63e-02   0.256  2.52               POP7
523             Q14257 Recurrence 1.63e-02   0.256 -2.52               RCN2
524             Q15365 Recurrence 1.63e-02   0.256  2.52              PCBP1
525             Q16795 Recurrence 1.63e-02   0.256  2.52             NDUFA9
526             Q53GG5 Recurrence 1.63e-02   0.256 -2.52             PDLIM3
527             Q5RHP9 Recurrence 1.63e-02   0.256 -2.52             ERICH3
528             P46087 Recurrence 1.65e-02   0.257  2.52               NOP2
529             Q99611 Recurrence 1.66e-02   0.258  2.52             SEPHS2
530             Q9H1A3 Recurrence 1.66e-02   0.258  2.52             METTL9
531             Q9NXS2 Recurrence 1.66e-02   0.258  2.52              QPCTL
532             P62745 Recurrence 1.67e-02   0.258 -2.51               RHOB
533             Q8TCJ2 Recurrence 1.67e-02   0.258  2.51              STT3B
534             Q96NB2 Recurrence 1.68e-02   0.259  2.51              SFXN2
535             O60832 Recurrence 1.69e-02   0.259  2.51               DKC1
536             Q6JBY9 Recurrence 1.69e-02   0.259 -2.51              RCSD1
537             Q9Y4W2 Recurrence 1.69e-02   0.259  2.51              LAS1L
538             Q56VL3 Recurrence 1.70e-02   0.259  2.51             OCIAD2
539             Q8NDH2 Recurrence 1.70e-02   0.259 -2.51            CCDC168
540             Q8WUH2 Recurrence 1.70e-02   0.259  2.51           TGFBRAP1
541             Q9BV38 Recurrence 1.70e-02   0.259  2.51              WDR18
542             Q9Y5Y2 Recurrence 1.70e-02   0.259  2.51              NUBP2
543             Q92947 Recurrence 1.71e-02   0.260  2.50               GCDH
544             O94808 Recurrence 1.72e-02   0.261  2.50              GFPT2
545             Q16643 Recurrence 1.72e-02   0.261 -2.50               DBN1
546             Q08174 Recurrence 1.73e-02   0.261 -2.50              PCDH1
547             Q96G23 Recurrence 1.73e-02   0.261  2.50              CERS2
548             P98170 Recurrence 1.74e-02   0.261  2.50               XIAP
549             Q00059 Recurrence 1.74e-02   0.261 -2.50               TFAM
550             Q10471 Recurrence 1.74e-02   0.261  2.50             GALNT2
551             Q99961 Recurrence 1.74e-02   0.261 -2.50             SH3GL1
552             P62487 Recurrence 1.75e-02   0.261  2.49             POLR2G
553             Q9H2A7 Recurrence 1.75e-02   0.261  2.49             CXCL16
554             P13747 Recurrence 1.76e-02   0.261 -2.49              HLA-E
555             P82930 Recurrence 1.76e-02   0.261  2.49             MRPS34
556             O95436 Recurrence 1.77e-02   0.261  2.49            SLC34A2
557             Q15819 Recurrence 1.77e-02   0.261 -2.49             UBE2V2
558             Q16831 Recurrence 1.77e-02   0.261  2.49               UPP1
559             Q99426 Recurrence 1.77e-02   0.261 -2.49               TBCB
560             Q15329 Recurrence 1.78e-02   0.261  2.49               E2F5
561             Q93033 Recurrence 1.78e-02   0.261  2.49              CD101
562             Q9Y5J5 Recurrence 1.78e-02   0.261  2.49             PHLDA3
563             O75165 Recurrence 1.79e-02   0.261  2.48            DNAJC13
564             P31948 Recurrence 1.79e-02   0.261 -2.48              STIP1
565             P61981 Recurrence 1.79e-02   0.261 -2.48              YWHAG
566             Q9Y5B9 Recurrence 1.79e-02   0.261  2.49            SUPT16H
567             O95104 Recurrence 1.80e-02   0.262  2.48              SCAF4
568             Q02413 Recurrence 1.80e-02   0.261 -2.48               DSG1
569             Q8N6R0 Recurrence 1.80e-02   0.261  2.48            METTL13
570             P35908 Recurrence 1.81e-02   0.262 -2.48               KRT2
571             Q9Y2G2 Recurrence 1.81e-02   0.262 -2.48              CARD8
572             H7BZ55 Recurrence 1.82e-02   0.262 -2.48             CROCC2
573             P01599 Recurrence 1.82e-02   0.262 -2.48           IGKV1-17
574             O76021 Recurrence 1.83e-02   0.263  2.47             RSL1D1
575             Q14642 Recurrence 1.83e-02   0.263 -2.48             INPP5A
576             Q9Y3B2 Recurrence 1.83e-02   0.263  2.47             EXOSC1
577             Q13404 Recurrence 1.84e-02   0.263 -2.47             UBE2V1
578             P28068 Recurrence 1.85e-02   0.263  2.47            HLA-DMB
579             P35580 Recurrence 1.85e-02   0.263 -2.47              MYH10
580             Q6UWE0 Recurrence 1.86e-02   0.264  2.47             LRSAM1
581             Q9H2D6 Recurrence 1.86e-02   0.264 -2.47             TRIOBP
582             Q9Y678 Recurrence 1.86e-02   0.264  2.47              COPG1
583             O75616 Recurrence 1.87e-02   0.264  2.47              ERAL1
584             P51452 Recurrence 1.88e-02   0.265 -2.46              DUSP3
585             Q86UP2 Recurrence 1.88e-02   0.265 -2.46               KTN1
586             Q8IWB9 Recurrence 1.88e-02   0.265  2.46               TEX2
587             O15347 Recurrence 1.89e-02   0.265 -2.46              HMGB3
588             P06310 Recurrence 1.89e-02   0.265 -2.46           IGKV2-30
589             P30511 Recurrence 1.89e-02   0.265 -2.46              HLA-F
590             O15120 Recurrence 1.90e-02   0.266  2.46             AGPAT2
591             P61952 Recurrence 1.90e-02   0.265 -2.46              GNG11
592             Q8TB40 Recurrence 1.91e-02   0.266  2.46              ABHD4
593             Q99567 Recurrence 1.91e-02   0.266  2.46              NUP88
594             O15018 Recurrence 1.92e-02   0.266 -2.46              PDZD2
595             Q5JVS0 Recurrence 1.92e-02   0.266  2.46              HABP4
596             P13727 Recurrence 1.93e-02   0.267  2.45               PRG2
597             P54652 Recurrence 1.93e-02   0.267 -2.45              HSPA2
598             P54619 Recurrence 1.96e-02   0.270  2.45             PRKAG1
599             Q9H7B4 Recurrence 1.96e-02   0.270  2.45              SMYD3
600             P20839 Recurrence 1.97e-02   0.270  2.44             IMPDH1
601             Q9NQ89 Recurrence 1.97e-02   0.270 -2.44            C12orf4
602             Q9P2G1 Recurrence 1.97e-02   0.270 -2.44             ANKIB1
603             Q86UT6 Recurrence 1.98e-02   0.271  2.44              NLRX1
604             P13646 Recurrence 2.00e-02   0.273 -2.44              KRT13
605             Q330K2 Recurrence 2.01e-02   0.274  2.44            NDUFAF6
606             Q71UM5 Recurrence 2.01e-02   0.274  2.44             RPS27L
607             O75347 Recurrence 2.02e-02   0.274 -2.43               TBCA
608             O96005 Recurrence 2.02e-02   0.274  2.43             CLPTM1
609             O95994 Recurrence 2.03e-02   0.274 -2.43               AGR2
610             P30049 Recurrence 2.03e-02   0.274 -2.43            ATP5F1D
611             P49815 Recurrence 2.03e-02   0.274  2.43               TSC2
612             Q13740 Recurrence 2.03e-02   0.274 -2.43              ALCAM
613             O00148 Recurrence 2.04e-02   0.275  2.43             DDX39A
614             Q9NZM1 Recurrence 2.04e-02   0.275  2.43               MYOF
615             Q9Y6N9 Recurrence 2.05e-02   0.275 -2.43              USH1C
616             O14924 Recurrence 2.06e-02   0.275  2.43              RGS12
617             O14976 Recurrence 2.06e-02   0.275  2.43                GAK
618             Q96QE2 Recurrence 2.06e-02   0.275  2.43            SLC2A13
619             Q9UJY4 Recurrence 2.06e-02   0.275  2.42               GGA2
620             O43633 Recurrence 2.07e-02   0.275 -2.42             CHMP2A
621             Q00325 Recurrence 2.07e-02   0.275  2.42            SLC25A3
622             Q9UH65 Recurrence 2.07e-02   0.275 -2.42             SWAP70
623             Q9Y4D1 Recurrence 2.07e-02   0.275 -2.42              DAAM1
624             Q9Y6K5 Recurrence 2.08e-02   0.275  2.42               OAS3
625             P43699 Recurrence 2.09e-02   0.276  2.42             NKX2-1
626             Q5VWC8 Recurrence 2.10e-02   0.276  2.42              HACD4
627             Q92608 Recurrence 2.10e-02   0.276  2.42              DOCK2
628      B7ZAQ6;P0CG08 Recurrence 2.12e-02   0.276  2.41      GPR89A;GPR89B
629             Q9H0R4 Recurrence 2.12e-02   0.276 -2.41              HDHD2
630             Q6IA69 Recurrence 2.13e-02   0.276  2.41            NADSYN1
631             Q86VS8 Recurrence 2.13e-02   0.276 -2.41              HOOK3
632             Q8IXL7 Recurrence 2.13e-02   0.276 -2.41              MSRB3
633             O00244 Recurrence 2.14e-02   0.276 -2.41              ATOX1
634             O75896 Recurrence 2.14e-02   0.276 -2.41              TUSC2
635             P51398 Recurrence 2.14e-02   0.276  2.41               DAP3
636             Q9HC35 Recurrence 2.14e-02   0.276  2.41               EML4
637             P33240 Recurrence 2.15e-02   0.276 -2.41              CSTF2
638             Q13546 Recurrence 2.15e-02   0.276  2.41              RIPK1
639             Q7L2H7 Recurrence 2.15e-02   0.276  2.41              EIF3M
640             Q8IWX8 Recurrence 2.15e-02   0.276  2.41              CHERP
641             P12236 Recurrence 2.16e-02   0.276  2.40            SLC25A6
642             Q6P4A8 Recurrence 2.16e-02   0.276  2.41              PLBD1
643             Q7Z3B4 Recurrence 2.16e-02   0.276  2.41              NUP54
644             Q6EMK4 Recurrence 2.17e-02   0.276 -2.40               VASN
645             Q96DX5 Recurrence 2.17e-02   0.276 -2.40               ASB9
646             Q96T76 Recurrence 2.17e-02   0.276  2.40              MMS19
647             Q9NPA0 Recurrence 2.17e-02   0.276  2.40               EMC7
648             Q9NZU5 Recurrence 2.17e-02   0.276 -2.40              LMCD1
649             Q9P1Y5 Recurrence 2.17e-02   0.276  2.40            CAMSAP3
650             A4D1P6 Recurrence 2.19e-02   0.278  2.40              WDR91
651             Q9HBL7 Recurrence 2.19e-02   0.278  2.40             PLGRKT
652             P53708 Recurrence 2.20e-02   0.278 -2.40              ITGA8
653             P60983 Recurrence 2.20e-02   0.278 -2.40               GMFB
654             Q8N2K0 Recurrence 2.20e-02   0.278  2.40             ABHD12
655             Q8IUR7 Recurrence 2.21e-02   0.278  2.40              ARMC8
656             Q9ULP9 Recurrence 2.21e-02   0.279  2.39            TBC1D24
657             O75694 Recurrence 2.22e-02   0.279  2.39             NUP155
658             Q9BYV8 Recurrence 2.22e-02   0.279 -2.39              CEP41
659             P57088 Recurrence 2.23e-02   0.279  2.39             TMEM33
660             Q8N4T8 Recurrence 2.23e-02   0.279  2.39               CBR4
661             Q9Y4C8 Recurrence 2.23e-02   0.279  2.39              RBM19
662             O00116 Recurrence 2.24e-02   0.279  2.39               AGPS
663             Q92604 Recurrence 2.24e-02   0.279  2.39             LPGAT1
664             O95486 Recurrence 2.25e-02   0.280  2.39             SEC24A
665             Q9GZL7 Recurrence 2.26e-02   0.281  2.38              WDR12
666             P05556 Recurrence 2.28e-02   0.282 -2.38              ITGB1
667             P26583 Recurrence 2.29e-02   0.283 -2.38              HMGB2
668             Q13905 Recurrence 2.30e-02   0.285  2.38            RAPGEF1
669             P78345 Recurrence 2.31e-02   0.285  2.38              RPP38
670             Q99828 Recurrence 2.33e-02   0.287 -2.37               CIB1
671             P13693 Recurrence 2.34e-02   0.287 -2.37               TPT1
672             Q6UXY8 Recurrence 2.34e-02   0.287 -2.37               TMC5
673             Q5T4S7 Recurrence 2.35e-02   0.288  2.37               UBR4
674             Q96J01 Recurrence 2.35e-02   0.287  2.37              THOC3
675             Q6ZV73 Recurrence 2.36e-02   0.288 -2.37               FGD6
676             Q9P253 Recurrence 2.36e-02   0.288  2.37              VPS18
677             Q4ZHG4 Recurrence 2.37e-02   0.288 -2.36              FNDC1
678             Q96MH2 Recurrence 2.37e-02   0.288 -2.36             HEXIM2
679             Q9H6F5 Recurrence 2.37e-02   0.288 -2.36             CCDC86
680             A1X283 Recurrence 2.39e-02   0.290  2.36           SH3PXD2B
681             Q9P0J0 Recurrence 2.40e-02   0.290  2.36            NDUFA13
682             O00182 Recurrence 2.41e-02   0.290  2.36             LGALS9
683             O75044 Recurrence 2.41e-02   0.290 -2.36             SRGAP2
684             Q8N5D0 Recurrence 2.41e-02   0.290 -2.36              WDTC1
685             Q9Y2G9 Recurrence 2.41e-02   0.290  2.36              SBNO2
686             Q86WU2 Recurrence 2.43e-02   0.292  2.35               LDHD
687             P38571 Recurrence 2.44e-02   0.292  2.35               LIPA
688             P60174 Recurrence 2.44e-02   0.292 -2.35               TPI1
689             P28289 Recurrence 2.45e-02   0.293 -2.35              TMOD1
690             O00308 Recurrence 2.46e-02   0.294 -2.35               WWP2
691             P78347 Recurrence 2.46e-02   0.294  2.35              GTF2I
692             Q9NR50 Recurrence 2.46e-02   0.294  2.35             EIF2B3
693             P25686 Recurrence 2.47e-02   0.294 -2.35             DNAJB2
694             O94880 Recurrence 2.48e-02   0.294  2.35              PHF14
695             Q6ZSS7 Recurrence 2.48e-02   0.294 -2.35              MFSD6
696             Q7L591 Recurrence 2.49e-02   0.295  2.34               DOK3
697             Q8TDP1 Recurrence 2.50e-02   0.295 -2.34           RNASEH2C
698             Q92979 Recurrence 2.51e-02   0.296  2.34               EMG1
699             Q9ULW3 Recurrence 2.51e-02   0.296  2.34               ABT1
700             Q9UNI6 Recurrence 2.51e-02   0.296 -2.34             DUSP12
701             P43121 Recurrence 2.53e-02   0.297 -2.34               MCAM
702             Q03001 Recurrence 2.53e-02   0.297 -2.34                DST
703             P55211 Recurrence 2.54e-02   0.298 -2.33              CASP9
704             P81605 Recurrence 2.54e-02   0.298 -2.33                DCD
705             Q9H553 Recurrence 2.54e-02   0.298  2.33               ALG2
706             Q9P0M6 Recurrence 2.55e-02   0.298  2.33          MACROH2A2
707             P04049 Recurrence 2.56e-02   0.299  2.33               RAF1
708             Q8IY31 Recurrence 2.57e-02   0.299 -2.33              IFT20
709             P62942 Recurrence 2.58e-02   0.299 -2.33             FKBP1A
710             Q9BUH6 Recurrence 2.58e-02   0.299 -2.33               PAXX
711             Q9Y223 Recurrence 2.58e-02   0.299  2.33                GNE
712             Q16540 Recurrence 2.59e-02   0.299  2.33             MRPL23
713             O95292 Recurrence 2.60e-02   0.299 -2.33               VAPB
714             P53621 Recurrence 2.60e-02   0.299  2.32               COPA
715             P55072 Recurrence 2.60e-02   0.299 -2.32                VCP
716             P61619 Recurrence 2.60e-02   0.299  2.32            SEC61A1
717             Q13310 Recurrence 2.60e-02   0.299  2.32             PABPC4
718             Q9NWW5 Recurrence 2.60e-02   0.299  2.33               CLN6
719             O95163 Recurrence 2.61e-02   0.299  2.32               ELP1
720             Q8IXS6 Recurrence 2.61e-02   0.299 -2.32              PALM2
721             Q8TDD1 Recurrence 2.61e-02   0.299  2.32              DDX54
722             Q9H078 Recurrence 2.62e-02   0.299  2.32               CLPB
723             O60279 Recurrence 2.63e-02   0.299 -2.32              SUSD5
724             P42356 Recurrence 2.63e-02   0.299  2.32              PI4KA
725             Q53FV1 Recurrence 2.63e-02   0.299  2.32             ORMDL2
726             Q8N5K1 Recurrence 2.64e-02   0.300  2.32              CISD2
727             P10398 Recurrence 2.65e-02   0.300  2.32               ARAF
728             P37802 Recurrence 2.65e-02   0.300 -2.32             TAGLN2
729             P10301 Recurrence 2.66e-02   0.301 -2.31               RRAS
730             P35221 Recurrence 2.66e-02   0.301 -2.31             CTNNA1
731             Q92538 Recurrence 2.67e-02   0.301  2.31               GBF1
732             Q9NRS6 Recurrence 2.67e-02   0.301 -2.31              SNX15
733             P19388 Recurrence 2.68e-02   0.301  2.31             POLR2E
734             P50851 Recurrence 2.68e-02   0.301  2.31               LRBA
735             Q6WCQ1 Recurrence 2.69e-02   0.301 -2.31              MPRIP
736             O76094 Recurrence 2.70e-02   0.301  2.31              SRP72
737             P16035 Recurrence 2.70e-02   0.301 -2.31              TIMP2
738             Q03135 Recurrence 2.70e-02   0.301 -2.31               CAV1
739             Q9UEW3 Recurrence 2.70e-02   0.301  2.31              MARCO
740             O14684 Recurrence 2.71e-02   0.301  2.31              PTGES
741             O95445 Recurrence 2.71e-02   0.301 -2.31               APOM
742             Q96N66 Recurrence 2.71e-02   0.301  2.31             MBOAT7
743             Q9UKF6 Recurrence 2.72e-02   0.302  2.31              CPSF3
744             Q86X02 Recurrence 2.73e-02   0.302 -2.30              CDR2L
745             Q8WV41 Recurrence 2.73e-02   0.302  2.30              SNX33
746             Q9UPQ0 Recurrence 2.73e-02   0.302 -2.30             LIMCH1
747             P22307 Recurrence 2.74e-02   0.302  2.30               SCP2
748             Q9H7D7 Recurrence 2.74e-02   0.302  2.30              WDR26
749             Q9Y6Q5 Recurrence 2.75e-02   0.303  2.30              AP1M2
750             Q8N3P4 Recurrence 2.76e-02   0.304  2.30               VPS8
751             Q02809 Recurrence 2.77e-02   0.304  2.30              PLOD1
752             Q8NF50 Recurrence 2.77e-02   0.304  2.30              DOCK8
753             O14734 Recurrence 2.78e-02   0.304  2.30              ACOT8
754             O75352 Recurrence 2.78e-02   0.304  2.30              MPDU1
755             P17568 Recurrence 2.79e-02   0.305  2.29             NDUFB7
756             O00560 Recurrence 2.80e-02   0.306  2.29              SDCBP
757             Q15773 Recurrence 2.81e-02   0.306 -2.29               MLF2
758             O95347 Recurrence 2.82e-02   0.307 -2.29               SMC2
759             P50402 Recurrence 2.84e-02   0.308 -2.29                EMD
760             O15247 Recurrence 2.85e-02   0.308 -2.29              CLIC2
761             Q5VTL8 Recurrence 2.85e-02   0.308  2.28            PRPF38B
762             Q8WX92 Recurrence 2.85e-02   0.308  2.28              NELFB
763      P14678;P63162 Recurrence 2.86e-02   0.309  2.28        SNRPB;SNRPN
764             Q5TEJ8 Recurrence 2.86e-02   0.309  2.28            THEMIS2
765             O14920 Recurrence 2.87e-02   0.309  2.28              IKBKB
766             Q9NVD7 Recurrence 2.87e-02   0.309 -2.28              PARVA
767             O00142 Recurrence 2.88e-02   0.309  2.28                TK2
768             Q96HY6 Recurrence 2.88e-02   0.309 -2.28             DDRGK1
769             Q5TBB1 Recurrence 2.89e-02   0.310  2.28           RNASEH2B
770             Q96QK1 Recurrence 2.89e-02   0.310  2.28              VPS35
771             O75306 Recurrence 2.90e-02   0.311  2.28             NDUFS2
772             P11310 Recurrence 2.92e-02   0.311  2.27              ACADM
773             P54252 Recurrence 2.92e-02   0.311 -2.27              ATXN3
774             Q8IZ69 Recurrence 2.92e-02   0.311  2.27             TRMT2A
775             Q96CD0 Recurrence 2.92e-02   0.311  2.27              FBXL8
776             Q6UW02 Recurrence 2.93e-02   0.312  2.27            CYP20A1
777             O75828 Recurrence 2.94e-02   0.313 -2.27               CBR3
778             P61978 Recurrence 2.96e-02   0.313 -2.27             HNRNPK
779             Q8N9B5 Recurrence 2.96e-02   0.313 -2.27                JMY
780             Q8IZV5 Recurrence 2.97e-02   0.314  2.27              RDH10
781             O60271 Recurrence 2.98e-02   0.314 -2.27              SPAG9
782             Q969G5 Recurrence 2.98e-02   0.314 -2.27             CAVIN3
783             Q96SK2 Recurrence 2.98e-02   0.314  2.26            TMEM209
784             O15228 Recurrence 2.99e-02   0.314  2.26              GNPAT
785             Q9H4Z3 Recurrence 2.99e-02   0.314  2.26              PCIF1
786             P26196 Recurrence 3.01e-02   0.316  2.26               DDX6
787             O15264 Recurrence 3.02e-02   0.316  2.26             MAPK13
788             O75475 Recurrence 3.02e-02   0.316 -2.26              PSIP1
789      P04908;Q7L7L0 Recurrence 3.03e-02   0.316 -2.26       H2AC8;H2AC25
790             Q04695 Recurrence 3.03e-02   0.316 -2.26              KRT17
791             Q14571 Recurrence 3.04e-02   0.316  2.26              ITPR2
792             Q5VZ89 Recurrence 3.04e-02   0.316  2.26            DENND4C
793             Q7Z4S6 Recurrence 3.04e-02   0.316 -2.26             KIF21A
794             Q96S44 Recurrence 3.04e-02   0.316  2.26             TP53RK
795             Q8IVS2 Recurrence 3.05e-02   0.317 -2.25               MCAT
796             O95749 Recurrence 3.06e-02   0.317 -2.25              GGPS1
797             Q08211 Recurrence 3.06e-02   0.317  2.25               DHX9
798             Q86UW6 Recurrence 3.06e-02   0.317 -2.25              N4BP2
799             O00567 Recurrence 3.08e-02   0.317  2.25              NOP56
800             Q9NYL9 Recurrence 3.08e-02   0.317 -2.25              TMOD3
801             Q9Y2Z4 Recurrence 3.08e-02   0.317  2.25              YARS2
802             P11021 Recurrence 3.09e-02   0.317 -2.25              HSPA5
803             P16870 Recurrence 3.09e-02   0.317 -2.25                CPE
804             P61956 Recurrence 3.09e-02   0.317 -2.25              SUMO2
805             P12235 Recurrence 3.10e-02   0.317  2.25            SLC25A4
806             Q2M3G4 Recurrence 3.10e-02   0.317 -2.25            SHROOM1
807             Q8N5C6 Recurrence 3.10e-02   0.317  2.25              SRBD1
808             O95049 Recurrence 3.11e-02   0.317  2.25               TJP3
809             P16403 Recurrence 3.11e-02   0.317 -2.25               H1-2
810             Q9Y3A6 Recurrence 3.11e-02   0.317  2.25              TMED5
811             P11234 Recurrence 3.12e-02   0.317 -2.24               RALB
812             Q8N159 Recurrence 3.12e-02   0.317 -2.24               NAGS
813             P51689 Recurrence 3.13e-02   0.318 -2.24               ARSD
814             P0DOX8 Recurrence 3.14e-02   0.318 -2.24               &lt;NA&gt;
815             Q9BQ52 Recurrence 3.14e-02   0.318  2.24              ELAC2
816             Q9UBQ0 Recurrence 3.14e-02   0.318  2.24              VPS29
817             Q9H0J9 Recurrence 3.15e-02   0.318  2.24             PARP12
818             O94829 Recurrence 3.16e-02   0.318 -2.24              IPO13
819             P35606 Recurrence 3.16e-02   0.318  2.24              COPB2
820             Q96A72 Recurrence 3.16e-02   0.318  2.24             MAGOHB
821             Q9Y2D5 Recurrence 3.16e-02   0.318 -2.24              AKAP2
822             P39687 Recurrence 3.17e-02   0.318 -2.24             ANP32A
823             Q96DR5 Recurrence 3.18e-02   0.318  2.24             BPIFA2
824             Q9UL12 Recurrence 3.18e-02   0.318 -2.24              SARDH
825             P55263 Recurrence 3.19e-02   0.319  2.23                ADK
826             Q7L0Y3 Recurrence 3.19e-02   0.319  2.24            TRMT10C
827             P63261 Recurrence 3.20e-02   0.319 -2.23              ACTG1
828             Q9UBI6 Recurrence 3.20e-02   0.319 -2.23              GNG12
829             O14939 Recurrence 3.23e-02   0.320  2.23               PLD2
830             P61960 Recurrence 3.23e-02   0.320 -2.23               UFM1
831             Q15061 Recurrence 3.23e-02   0.320  2.23              WDR43
832             Q15542 Recurrence 3.23e-02   0.320  2.23               TAF5
833             Q96EP5 Recurrence 3.23e-02   0.320  2.23             DAZAP1
834             Q9NY47 Recurrence 3.23e-02   0.320 -2.23           CACNA2D2
835             Q9H5Z1 Recurrence 3.24e-02   0.320  2.23              DHX35
836             Q9NV88 Recurrence 3.24e-02   0.320  2.23              INTS9
837             O43920 Recurrence 3.26e-02   0.321  2.23             NDUFS5
838             P09960 Recurrence 3.26e-02   0.321  2.23              LTA4H
839             P43897 Recurrence 3.27e-02   0.321  2.22               TSFM
840             O94875 Recurrence 3.28e-02   0.321 -2.22             SORBS2
841             Q8NCE2 Recurrence 3.28e-02   0.321  2.22             MTMR14
842             Q9BQ95 Recurrence 3.28e-02   0.321  2.22              ECSIT
843             Q9NWS0 Recurrence 3.28e-02   0.321 -2.22             PIH1D1
844             P48553 Recurrence 3.29e-02   0.321  2.22           TRAPPC10
845             Q4G0F5 Recurrence 3.29e-02   0.321  2.22             VPS26B
846             Q8WTS6 Recurrence 3.29e-02   0.321 -2.22              SETD7
847             P19838 Recurrence 3.30e-02   0.321  2.22              NFKB1
848             Q5R3I4 Recurrence 3.30e-02   0.321 -2.22              TTC38
849             Q6NUK1 Recurrence 3.30e-02   0.321  2.22           SLC25A24
850             Q9UGQ3 Recurrence 3.30e-02   0.321  2.22             SLC2A6
851             P55196 Recurrence 3.31e-02   0.321 -2.22               AFDN
852             Q12846 Recurrence 3.32e-02   0.321 -2.22               STX4
853             P24928 Recurrence 3.33e-02   0.321  2.22             POLR2A
854             Q13206 Recurrence 3.33e-02   0.321 -2.22              DDX10
855             Q8N9V3 Recurrence 3.33e-02   0.321  2.22             WDSUB1
856             Q9UBB9 Recurrence 3.33e-02   0.321  2.22             TFIP11
857             O00273 Recurrence 3.34e-02   0.321 -2.21               DFFA
858             Q86SQ0 Recurrence 3.34e-02   0.321 -2.21             PHLDB2
859             O00291 Recurrence 3.35e-02   0.322 -2.21               HIP1
860             Q7Z4Q2 Recurrence 3.37e-02   0.323  2.21             HEATR3
861             Q9ULC3 Recurrence 3.37e-02   0.323 -2.21              RAB23
862             P28347 Recurrence 3.38e-02   0.323 -2.21              TEAD1
863             P48681 Recurrence 3.38e-02   0.323 -2.21                NES
864             Q5JTH9 Recurrence 3.38e-02   0.323  2.21              RRP12
865             O75116 Recurrence 3.39e-02   0.323 -2.21              ROCK2
866             Q8TB96 Recurrence 3.39e-02   0.323 -2.21              ITFG1
867             Q96PU5 Recurrence 3.40e-02   0.323 -2.21             NEDD4L
868             Q8NI37 Recurrence 3.41e-02   0.324  2.21              PPTC7
869             P58107 Recurrence 3.42e-02   0.324  2.20              EPPK1
870             Q13938 Recurrence 3.42e-02   0.324 -2.20               CAPS
871             Q5JPH6 Recurrence 3.42e-02   0.324  2.20              EARS2
872             Q8IX12 Recurrence 3.42e-02   0.324  2.20              CCAR1
873             Q9P2I0 Recurrence 3.44e-02   0.325  2.20              CPSF2
874             Q7Z478 Recurrence 3.45e-02   0.326  2.20              DHX29
875             P83916 Recurrence 3.47e-02   0.327 -2.20               CBX1
876             Q8TDY2 Recurrence 3.47e-02   0.327 -2.20             RB1CC1
877             Q13888 Recurrence 3.48e-02   0.327  2.20             GTF2H2
878             Q13526 Recurrence 3.49e-02   0.327 -2.20               PIN1
879             Q9H0A0 Recurrence 3.49e-02   0.327  2.19              NAT10
880             Q9Y279 Recurrence 3.49e-02   0.327  2.19              VSIG4
881         A0A0B4J1V0 Recurrence 3.50e-02   0.327 -2.19           IGHV3-15
882             Q5VYK3 Recurrence 3.51e-02   0.327  2.19              ECPAS
883             O60507 Recurrence 3.52e-02   0.327  2.19              TPST1
884             O94956 Recurrence 3.52e-02   0.327  2.19            SLCO2B1
885             P11217 Recurrence 3.52e-02   0.327 -2.19               PYGM
886             P16066 Recurrence 3.52e-02   0.327  2.19               NPR1
887             Q8TBP6 Recurrence 3.52e-02   0.327  2.19           SLC25A40
888             O75822 Recurrence 3.53e-02   0.327 -2.19              EIF3J
889             P11362 Recurrence 3.53e-02   0.327 -2.19              FGFR1
890             Q96BY6 Recurrence 3.53e-02   0.327  2.19             DOCK10
891             P05026 Recurrence 3.54e-02   0.327 -2.19             ATP1B1
892             Q9NVS9 Recurrence 3.54e-02   0.327 -2.19               PNPO
893             P32246 Recurrence 3.55e-02   0.327  2.19               CCR1
894             Q96MD2 Recurrence 3.55e-02   0.327 -2.19              KICS2
895             Q9NX74 Recurrence 3.55e-02   0.327 -2.19               DUS2
896             P20827 Recurrence 3.56e-02   0.327 -2.19              EFNA1
897             P48444 Recurrence 3.56e-02   0.327  2.19              ARCN1
898             Q01433 Recurrence 3.56e-02   0.327  2.19              AMPD2
899             Q13242 Recurrence 3.59e-02   0.330 -2.18              SRSF9
900             Q70CQ2 Recurrence 3.60e-02   0.330 -2.18              USP34
901             P00973 Recurrence 3.62e-02   0.331  2.18               OAS1
902             Q3SY84 Recurrence 3.63e-02   0.331 -2.18              KRT71
903             Q8IWA4 Recurrence 3.63e-02   0.331  2.18               MFN1
904             Q8TCF1 Recurrence 3.63e-02   0.331  2.18             ZFAND1
905             Q12849 Recurrence 3.64e-02   0.331 -2.18              GRSF1
906             Q99547 Recurrence 3.65e-02   0.331 -2.17           MPHOSPH6
907             Q9BYD3 Recurrence 3.65e-02   0.331  2.18              MRPL4
908             P53582 Recurrence 3.66e-02   0.331  2.17             METAP1
909             P62829 Recurrence 3.66e-02   0.331  2.17              RPL23
910             Q16630 Recurrence 3.66e-02   0.331  2.17              CPSF6
911             Q92843 Recurrence 3.66e-02   0.331 -2.17             BCL2L2
912             Q9UNN5 Recurrence 3.66e-02   0.331 -2.17               FAF1
913             Q8IWL1 Recurrence 3.67e-02   0.331 -2.17             SFTPA2
914             Q8NBM4 Recurrence 3.67e-02   0.331  2.17              UBAC2
915             Q8WZA9 Recurrence 3.67e-02   0.331 -2.17               IRGQ
916             O95477 Recurrence 3.68e-02   0.331 -2.17              ABCA1
917             P27348 Recurrence 3.68e-02   0.331 -2.17              YWHAQ
918             P35680 Recurrence 3.68e-02   0.331 -2.17              HNF1B
919             Q53H12 Recurrence 3.69e-02   0.331  2.17                AGK
920             O00625 Recurrence 3.70e-02   0.331 -2.17                PIR
921             P56557 Recurrence 3.70e-02   0.331  2.17            TMEM50B
922             P29536 Recurrence 3.71e-02   0.331 -2.17              LMOD1
923             Q9Y485 Recurrence 3.71e-02   0.331  2.17              DMXL1
924             Q6V0I7 Recurrence 3.72e-02   0.331 -2.17               FAT4
925             Q7Z4F1 Recurrence 3.72e-02   0.331  2.17              LRP10
926             O00187 Recurrence 3.73e-02   0.331 -2.16              MASP2
927             Q6ICB0 Recurrence 3.73e-02   0.331 -2.16              DESI1
928             Q8NBI5 Recurrence 3.73e-02   0.331  2.16            SLC43A3
929             O43150 Recurrence 3.74e-02   0.331 -2.16              ASAP2
930             Q9UHB7 Recurrence 3.74e-02   0.331  2.16               AFF4
931             Q14728 Recurrence 3.75e-02   0.331  2.16             MFSD10
932             Q86YM7 Recurrence 3.75e-02   0.331 -2.16             HOMER1
933             O14653 Recurrence 3.76e-02   0.331  2.16              GOSR2
934             Q86SX6 Recurrence 3.76e-02   0.331 -2.16              GLRX5
935             Q9NR56 Recurrence 3.76e-02   0.331  2.16              MBNL1
936             O43143 Recurrence 3.77e-02   0.331  2.16              DHX15
937             O75643 Recurrence 3.77e-02   0.331  2.16           SNRNP200
938             P52948 Recurrence 3.77e-02   0.331  2.16              NUP98
939             P01009 Recurrence 3.79e-02   0.333 -2.16           SERPINA1
940             Q2VPB7 Recurrence 3.80e-02   0.334  2.16              AP5B1
941             P49736 Recurrence 3.82e-02   0.334 -2.15               MCM2
942             Q99459 Recurrence 3.82e-02   0.334 -2.15              CDC5L
943             Q9NRH2 Recurrence 3.82e-02   0.334 -2.15               SNRK
944             O75954 Recurrence 3.83e-02   0.334 -2.15             TSPAN9
945             P08727 Recurrence 3.83e-02   0.334 -2.15              KRT19
946             Q9UQ16 Recurrence 3.83e-02   0.334  2.15               DNM3
947             Q460N5 Recurrence 3.84e-02   0.334  2.15             PARP14
948             Q9Y584 Recurrence 3.84e-02   0.334  2.15             TIMM22
949             Q9H1Y0 Recurrence 3.85e-02   0.334  2.15               ATG5
950             Q9Y4Y9 Recurrence 3.85e-02   0.334 -2.15               LSM5
951             O15061 Recurrence 3.87e-02   0.336 -2.15               SYNM
952             O60568 Recurrence 3.88e-02   0.336  2.15              PLOD3
953             Q8WTT2 Recurrence 3.89e-02   0.336  2.15              NOC3L
954             Q9UM54 Recurrence 3.89e-02   0.336  2.15               MYO6
955             P22314 Recurrence 3.90e-02   0.336 -2.14               UBA1
956             Q5HYK3 Recurrence 3.90e-02   0.336  2.14               COQ5
957             Q8N4S9 Recurrence 3.90e-02   0.336 -2.15           MARVELD2
958             P78344 Recurrence 3.91e-02   0.337  2.14             EIF4G2
959             Q6UX71 Recurrence 3.91e-02   0.337 -2.14             PLXDC2
960             Q07065 Recurrence 3.92e-02   0.337 -2.14              CKAP4
961             O75648 Recurrence 3.94e-02   0.338 -2.14               TRMU
962             P16278 Recurrence 3.95e-02   0.338  2.14               GLB1
963             P04792 Recurrence 3.96e-02   0.339 -2.14              HSPB1
964             Q9UHR4 Recurrence 3.96e-02   0.339 -2.14           BAIAP2L1
965             P31942 Recurrence 3.97e-02   0.339 -2.14            HNRNPH3
966             Q03701 Recurrence 3.98e-02   0.340  2.14              CEBPZ
967             Q9Y3Q8 Recurrence 3.98e-02   0.340 -2.14            TSC22D4
968             P00519 Recurrence 4.00e-02   0.341 -2.13               ABL1
969             Q02978 Recurrence 4.00e-02   0.341  2.13           SLC25A11
970             O43772 Recurrence 4.02e-02   0.342  2.13           SLC25A20
971             Q9NVH0 Recurrence 4.03e-02   0.342  2.13               EXD2
972             Q9UI10 Recurrence 4.03e-02   0.342  2.13             EIF2B4
973             A4QPH2 Recurrence 4.05e-02   0.343  2.13            PI4KAP2
974             O94819 Recurrence 4.07e-02   0.344  2.13            KBTBD11
975             P30626 Recurrence 4.07e-02   0.344 -2.12                SRI
976             Q9UBK8 Recurrence 4.07e-02   0.344  2.12               MTRR
977             O95630 Recurrence 4.08e-02   0.344 -2.12             STAMBP
978             Q9NYB0 Recurrence 4.08e-02   0.344 -2.12            TERF2IP
979             Q9Y385 Recurrence 4.08e-02   0.344  2.12             UBE2J1
980             Q969M3 Recurrence 4.09e-02   0.344  2.12              YIPF5
981             Q6P1X6 Recurrence 4.10e-02   0.344  2.12            C8orf82
982             Q6ZVM7 Recurrence 4.10e-02   0.344 -2.12             TOM1L2
983             Q9UKK3 Recurrence 4.10e-02   0.344  2.12              PARP4
984             P53677 Recurrence 4.11e-02   0.344  2.12              AP3M2
985             Q8TBQ9 Recurrence 4.11e-02   0.344  2.12           TMEM167A
986             Q99805 Recurrence 4.11e-02   0.344  2.12             TM9SF2
987             Q96DT6 Recurrence 4.12e-02   0.344  2.12              ATG4C
988             Q8IZQ1 Recurrence 4.13e-02   0.344 -2.12              WDFY3
989             Q9NUI1 Recurrence 4.13e-02   0.344  2.12              DECR2
990             P35579 Recurrence 4.14e-02   0.344 -2.12               MYH9
991             P49589 Recurrence 4.14e-02   0.344 -2.12              CARS1
992             P22695 Recurrence 4.15e-02   0.344  2.12             UQCRC2
993             Q8NAP3 Recurrence 4.15e-02   0.344 -2.12             ZBTB38
994             Q9BST9 Recurrence 4.15e-02   0.344 -2.12               RTKN
995             Q9Y3B7 Recurrence 4.15e-02   0.344  2.12             MRPL11
996             P28799 Recurrence 4.16e-02   0.344  2.11                GRN
997             P56199 Recurrence 4.16e-02   0.344 -2.12              ITGA1
998             Q6P2Q9 Recurrence 4.17e-02   0.344  2.11              PRPF8
999             Q9BTU6 Recurrence 4.17e-02   0.344  2.11             PI4K2A
1000            O15400 Recurrence 4.18e-02   0.344 -2.11               STX7
1001 A0A075B6S2;A2NJV5 Recurrence 4.19e-02   0.344 -2.11 IGKV2D-29;IGKV2-29
1002            P49411 Recurrence 4.19e-02   0.344  2.11               TUFM
1003            Q96BP2 Recurrence 4.19e-02   0.344  2.11             CHCHD1
1004            Q96ER3 Recurrence 4.19e-02   0.344  2.11              SAAL1
1005            Q12913 Recurrence 4.20e-02   0.345 -2.11              PTPRJ
1006            O43809 Recurrence 4.21e-02   0.345  2.11             NUDT21
1007            P29972 Recurrence 4.22e-02   0.345 -2.11               AQP1
1008            Q9BTT0 Recurrence 4.22e-02   0.345 -2.11             ANP32E
1009            Q9Y371 Recurrence 4.22e-02   0.345 -2.11            SH3GLB1
1010            Q12933 Recurrence 4.23e-02   0.345 -2.11              TRAF2
1011            P61769 Recurrence 4.24e-02   0.346 -2.11                B2M
1012            Q9BSF4 Recurrence 4.25e-02   0.346  2.11             TIMM29
1013            P30039 Recurrence 4.28e-02   0.348 -2.10               PBLD
1014            P33176 Recurrence 4.28e-02   0.348 -2.10              KIF5B
1015            Q8NE71 Recurrence 4.29e-02   0.348  2.10              ABCF1
1016            O75122 Recurrence 4.30e-02   0.349 -2.10             CLASP2
1017            O00499 Recurrence 4.32e-02   0.350 -2.10               BIN1
1018            Q14012 Recurrence 4.32e-02   0.350 -2.10              CAMK1
1019            Q9BX66 Recurrence 4.32e-02   0.350 -2.10             SORBS1
1020            P18669 Recurrence 4.33e-02   0.350 -2.10              PGAM1
1021            Q12768 Recurrence 4.33e-02   0.350  2.10             WASHC5
1022            Q10589 Recurrence 4.35e-02   0.351 -2.09               BST2
1023            Q8TBP5 Recurrence 4.35e-02   0.351 -2.09            FAM174A
1024            P52799 Recurrence 4.37e-02   0.352 -2.09              EFNB2
1025            P57772 Recurrence 4.37e-02   0.352  2.09             EEFSEC
1026            Q7Z3B1 Recurrence 4.37e-02   0.352 -2.09              NEGR1
1027     A8MWD9;P62308 Recurrence 4.39e-02   0.352  2.09     SNRPGP15;SNRPG
1028            Q15147 Recurrence 4.39e-02   0.352 -2.09              PLCB4
1029            Q9Y6X3 Recurrence 4.39e-02   0.352  2.09               MAU2
1030            Q8N4P3 Recurrence 4.40e-02   0.352 -2.09              HDDC3
1031            O60925 Recurrence 4.42e-02   0.353 -2.09              PFDN1
1032            P00568 Recurrence 4.42e-02   0.353 -2.09                AK1
1033            O95976 Recurrence 4.43e-02   0.354 -2.09              IGSF6
1034            P46976 Recurrence 4.43e-02   0.354 -2.09               GYG1
1035            Q12769 Recurrence 4.45e-02   0.354  2.08             NUP160
1036            Q8TB61 Recurrence 4.45e-02   0.354  2.08            SLC35B2
1037            O96011 Recurrence 4.48e-02   0.356  2.08             PEX11B
1038            P43353 Recurrence 4.48e-02   0.356 -2.08            ALDH3B1
1039            Q6PI48 Recurrence 4.48e-02   0.356  2.08              DARS2
1040            O60812 Recurrence 4.49e-02   0.356  2.08           HNRNPCL1
1041            Q9H9A5 Recurrence 4.49e-02   0.356  2.08             CNOT10
1042            Q8IWL3 Recurrence 4.51e-02   0.356 -2.08               HSCB
1043            Q06187 Recurrence 4.52e-02   0.356  2.08                BTK
1044            Q99570 Recurrence 4.52e-02   0.356  2.08             PIK3R4
1045            O14980 Recurrence 4.53e-02   0.356  2.08               XPO1
1046            Q07817 Recurrence 4.53e-02   0.356 -2.08             BCL2L1
1047            Q9HCD5 Recurrence 4.53e-02   0.356  2.08              NCOA5
1048            P01709 Recurrence 4.54e-02   0.356 -2.08            IGLV2-8
1049            P35609 Recurrence 4.54e-02   0.356 -2.07              ACTN2
1050            P53367 Recurrence 4.54e-02   0.356 -2.07             ARFIP1
1051            P61421 Recurrence 4.55e-02   0.356  2.07           ATP6V0D1
1052            P62857 Recurrence 4.56e-02   0.356 -2.07              RPS28
1053            Q8TCD5 Recurrence 4.56e-02   0.356 -2.07               NT5C
1054            Q9C0K1 Recurrence 4.56e-02   0.356  2.07            SLC39A8
1055            Q9H6Q4 Recurrence 4.56e-02   0.356  2.07              CIAO3
1056            O00592 Recurrence 4.57e-02   0.356 -2.07              PODXL
1057            Q06265 Recurrence 4.57e-02   0.356  2.07             EXOSC9
1058            Q16706 Recurrence 4.57e-02   0.356  2.07             MAN2A1
1059        A0A075B6K4 Recurrence 4.58e-02   0.356 -2.07           IGLV3-10
1060            O43837 Recurrence 4.58e-02   0.356  2.07              IDH3B
1061            Q8N3U4 Recurrence 4.58e-02   0.356  2.07              STAG2
1062            Q9BTC8 Recurrence 4.58e-02   0.356  2.07               MTA3
1063            Q86VX2 Recurrence 4.59e-02   0.356  2.07             COMMD7
1064            P00813 Recurrence 4.60e-02   0.356 -2.07                ADA
1065        A0A075B6H9 Recurrence 4.61e-02   0.357 -2.07           IGLV4-69
1066            Q9P035 Recurrence 4.62e-02   0.357  2.07              HACD3
1067            O75449 Recurrence 4.63e-02   0.358 -2.07             KATNA1
1068            P20930 Recurrence 4.64e-02   0.358 -2.06                FLG
1069            O60907 Recurrence 4.65e-02   0.358  2.06              TBL1X
1070            P31937 Recurrence 4.65e-02   0.358 -2.06             HIBADH
1071            Q58WW2 Recurrence 4.65e-02   0.358 -2.06              DCAF6
1072            O00160 Recurrence 4.68e-02   0.360  2.06              MYO1F
1073            Q96EE3 Recurrence 4.68e-02   0.360  2.06              SEH1L
1074            Q9P2B4 Recurrence 4.68e-02   0.360 -2.06          CTTNBP2NL
1075            Q58FG1 Recurrence 4.69e-02   0.360  2.06          HSP90AA4P
1076            Q9BZJ0 Recurrence 4.69e-02   0.360  2.06             CRNKL1
1077            P27816 Recurrence 4.70e-02   0.360 -2.06               MAP4
1078            Q9Y6N5 Recurrence 4.70e-02   0.360  2.06               SQOR
1079            Q6BCY4 Recurrence 4.72e-02   0.361  2.06             CYB5R2
1080            Q96MW5 Recurrence 4.72e-02   0.361  2.06               COG8
1081            Q8WYA0 Recurrence 4.73e-02   0.361 -2.06              IFT81
1082            Q6UXH9 Recurrence 4.74e-02   0.361 -2.05              PAMR1
1083            Q9H0V1 Recurrence 4.74e-02   0.361  2.06            TMEM168
1084            Q08AM6 Recurrence 4.75e-02   0.361  2.05              VAC14
1085            Q8TDM6 Recurrence 4.76e-02   0.362  2.05               DLG5
1086            P39060 Recurrence 4.77e-02   0.362 -2.05            COL18A1
1087            Q969Q0 Recurrence 4.77e-02   0.362 -2.05            RPL36AL
1088            P13928 Recurrence 4.78e-02   0.362 -2.05              ANXA8
1089            Q12981 Recurrence 4.78e-02   0.362  2.05              BNIP1
1090            Q8IYS2 Recurrence 4.78e-02   0.362  2.05           KIAA2013
1091            Q8TB24 Recurrence 4.78e-02   0.362  2.05               RIN3
1092            P11049 Recurrence 4.80e-02   0.362  2.05               CD37
1093            Q8IWE5 Recurrence 4.80e-02   0.362  2.05            PLEKHM2
1094            O00423 Recurrence 4.81e-02   0.363 -2.05               EML1
1095            P56556 Recurrence 4.82e-02   0.363  2.05             NDUFA6
1096            Q9P2J5 Recurrence 4.82e-02   0.363  2.05              LARS1
1097            Q9Y4P3 Recurrence 4.82e-02   0.363  2.05               TBL2
1098            O75251 Recurrence 4.83e-02   0.363  2.05             NDUFS7
1099            P63244 Recurrence 4.86e-02   0.365  2.04              RACK1
1100            O76070 Recurrence 4.87e-02   0.365 -2.04               SNCG
1101            P11940 Recurrence 4.87e-02   0.365  2.04             PABPC1
1102            P78367 Recurrence 4.87e-02   0.365 -2.04             NKX3-2
1103            Q86Y82 Recurrence 4.88e-02   0.365 -2.04              STX12
1104            Q9H9J2 Recurrence 4.88e-02   0.365  2.04             MRPL44
1105            Q5RI15 Recurrence 4.89e-02   0.365  2.04              COX20
1106            Q96Q42 Recurrence 4.89e-02   0.365  2.04               ALS2
1107            Q9Y6U3 Recurrence 4.90e-02   0.365  2.04               SCIN
1108            Q7Z406 Recurrence 4.91e-02   0.365 -2.04              MYH14
1109            O75976 Recurrence 4.92e-02   0.365 -2.04                CPD
1110            Q29RF7 Recurrence 4.92e-02   0.365  2.04              PDS5A
1111            Q6NUQ4 Recurrence 4.92e-02   0.365  2.04            TMEM214
1112            Q9BQA9 Recurrence 4.92e-02   0.365  2.04              CYBC1
1113            P08670 Recurrence 4.93e-02   0.365 -2.04                VIM
1114            Q96SB3 Recurrence 4.93e-02   0.365 -2.04            PPP1R9B
1115            O75947 Recurrence 4.94e-02   0.366 -2.04             ATP5PD
1116            P00441 Recurrence 4.95e-02   0.366 -2.03               SOD1
1117            Q99624 Recurrence 4.95e-02   0.366 -2.03            SLC38A3
1118            P46108 Recurrence 4.96e-02   0.366 -2.03                CRK
1119            P42285 Recurrence 4.97e-02   0.367  2.03              MTREX
1120            P18510 Recurrence 4.98e-02   0.367 -2.03              IL1RN
1121            P11233 Recurrence 4.99e-02   0.367 -2.03               RALA
1122            Q96KA5 Recurrence 4.99e-02   0.367  2.03            CLPTM1L
1123            P54727 Recurrence 5.00e-02   0.367 -2.03             RAD23B
1124            Q567U6 Recurrence 5.00e-02   0.367  2.03             CCDC93</code></pre>
<p>Visualize molecular signatures of significant features in data. Note
that features in rows are ordered by t-statistics.</p>
<pre class="r"><code>proNormalRes_Klin$sig.feat.heat</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-74-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/heatmap_sigFeats_combined_proNormal_Klin.png&#39;, a,
#        device = &#39;png&#39;, dpi = 400, height = 10, width = 12)</code></pre>
<p>Visualize top significant features identified in Normal samples (ns:
p &gt; 0.05, <span class="math inline">\(*\)</span>: p &lt;= 0.05, <span
class="math inline">\(**\)</span>: p &lt;= 0.01, <span
class="math inline">\(***\)</span>: p &lt;= 0.001)</p>
<pre class="r"><code># Prepare data matrix and metadata including both Tumor and Normal samples
datMat &lt;- proTissueRes_Klin$data
# datMat &lt;- doBatchCorrection(proTissue_Klin, wantedVar = c(&#39;Condition&#39;, &#39;Recurrence&#39;),
#                             unwantedVar = &#39;Stage&#39;)
smpAnnoTab &lt;- proTissueRes_Klin$smpMetadata
# Prepare significant feature table
featSigAssoTab &lt;- proNormalRes_Klin$sig.feat.tab
num_sigFeats &lt;- 6
# Extract top significant features and prepare needed information
topSigFeats &lt;- featSigAssoTab[1:num_sigFeats,] %&gt;%
  dplyr::select(Var1, Gene) %&gt;%
  dplyr::mutate(Gene = factor(Gene, levels = Gene))
topSigFeatDat &lt;- tibble::as_tibble(datMat[topSigFeats$Var1,], rownames = &#39;Var1&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Var1&#39;, names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Var1&#39;) %&gt;%
  dplyr::left_join(smpAnnoTab, by = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Condition = factor(Condition, levels = c(&#39;Normal&#39;, &#39;Tumor&#39;)),
                Recurrence = factor(Recurrence, levels = c(&#39;Yes&#39;, &#39;No&#39;)))
# Visualize top significant features
ggplot(topSigFeatDat, aes(x=Condition, y=Abundance, fill=Recurrence)) +
  geom_boxplot(alpha = 1, linewidth = 1, outlier.color = &#39;grey50&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, method.args = list(var.equal = T),
                             label = &#39;p.signif&#39;, size = 7, vjust = 0.5) +
  labs(x = &#39;Sample condition&#39;, y = &#39;log(Abundance)&#39;) +
  scale_fill_manual(values = c(&#39;#F8766D&#39;, &#39;#00BFC4&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  geom_vline(xintercept = 1.5, size = 0.3) +
  th +
  theme(strip.text = element_text(size = 18, face = &#39;bold&#39;),
        axis.title = element_text(size = 22),
        axis.text.x = element_text(size = 18),
        legend.title = element_text(size = 24), legend.text = element_text(size = 22))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-75-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>proNormalRes_Klin$pval.hist +
  labs(title = &#39;Klin. Combined Normal Tissue DIA Proteomics&#39;) +
  theme(axis.title = element_text(size = 30), axis.text = element_text(size = 18))</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-76-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Discovery/2nd_TAC/histogram_pVals_combined_proNormal_Klin.png&#39;,
#        device = &#39;png&#39;, dpi = 400, height = 9, width = 11)</code></pre>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code>proNormalRes_Klin$sig.pc.tab</code></pre>
<pre><code>         Var1       Var2   pVal pValAdj  Stat   Test
1   PC13 (3%) Recurrence 0.0101   0.202  2.71 T-test
2 PC15 (2.7%) Recurrence 0.0364   0.301  2.17 T-test
3  PC5 (7.1%) Recurrence 0.0451   0.301 -2.07 T-test</code></pre>
<p>Visualize significant PCs</p>
<pre class="r"><code>proNormalRes_Klin$sig.pc.dist$PC5</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-78-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proNormalRes_Klin$sig.pc.dist$PC13</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-78-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proNormalRes_Klin$sig.pc.dist$PC15</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-78-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Compute ROC-AUC score
# response &lt;- ifelse(pcTab$Recurrence == &#39;Yes&#39;, yes = &#39;1&#39;, no = &#39;0&#39;) %&gt;%
#   factor(levels = c(&#39;1&#39;, &#39;0&#39;))
# predictor &lt;- pcTab$`PC1 (11.1%)`
# rocRes &lt;- pROC::roc(response = response, predictor = predictor, plot = T,
#                     legacy.axes = T, print.auc = T, print.auc.x = 0.4,
#                     xlab = substitute(paste(bold(&#39;False positive rate&#39;))),
#                     ylab = substitute(paste(bold(&#39;True positive rate&#39;))),
#                     main = &#39;ROC Curve for PC1&#39;,
#                     cex.lab = 1.8, cex.axis = 1.2, cex.main = 1.8, col = &#39;#377eb8&#39;,
#                     lwd = 4, direction = &#39;&lt;&#39;, quiet = T)
# pROC::ci.auc(rocRes)</code></pre>
<p>Enrichment analysis<br />
Reference: Hallmark gene sets<br />
<strong>t-values</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proNormalRes_Klin$SOA.res$featAssoRes %&gt;%
  dplyr::select(Var1, Stat) %&gt;%
  tibble::column_to_rownames(&#39;Var1&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)

# Do GSEA
# Hallmark gene sets
gseaRes_H &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 10, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaRes_H$plotTopSigGS</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-79-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>PC5 loadings</strong></p>
<pre class="r"><code># Prepare ranked gene list
statRes_Klin &lt;- proNormalRes_Klin$SOA.res$pcaRes@loadings %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  dplyr::select(Feature, PC5) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
featAnno_Klin &lt;- as.data.frame(rowData(proTissue_Klin))
rankedGeneList_Klin &lt;- prepRankedFeatList(statRes = statRes_Klin, featAnno = featAnno_Klin, to_gene = T)
rankedGeneList_Klin &lt;- sort(-rankedGeneList_Klin, decreasing = T)

# Do GSEA
# Hallmark gene sets
gseaResH &lt;- doEA(rankedGeneList_Klin, categoryDB = &#39;H&#39;, numSigGS = 10, rmPrefixGS = &#39;^HALLMARK_&#39;)
gseaResH$plotTopSigGS</code></pre>
<p><img src="figure/MDev_Dis_02_soa_DIA.Rmd/unnamed-chunk-80-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Warsaw
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] visdat_0.6.0                vsn_3.70.0                 
 [3] lubridate_1.9.3             forcats_1.0.0              
 [5] stringr_1.5.1               purrr_1.0.2                
 [7] readr_2.1.5                 tidyr_1.3.1                
 [9] tibble_3.2.1                tidyverse_2.0.0            
[11] SummarizedExperiment_1.32.0 GenomicRanges_1.54.1       
[13] GenomeInfoDb_1.38.8         MatrixGenerics_1.14.0      
[15] matrixStats_1.3.0           spsUtil_0.2.2              
[17] clusterProfiler_4.10.1      msigdbr_7.5.1              
[19] org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.1       
[21] IRanges_2.36.0              S4Vectors_0.40.2           
[23] ggvenn_0.1.10               dplyr_1.1.4                
[25] ggrepel_0.9.5               ggplot2_3.5.1              
[27] ggplotify_0.1.2             pheatmap_1.0.12            
[29] proDA_1.16.0                limma_3.58.1               
[31] sva_3.50.0                  BiocParallel_1.36.0        
[33] genefilter_1.84.0           mgcv_1.9-1                 
[35] nlme_3.1-164                pcaMethods_1.94.0          
[37] Biobase_2.62.0              BiocGenerics_0.48.1        
[39] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] splines_4.3.2           later_1.3.2             bitops_1.0-7           
  [4] polyclip_1.10-6         preprocessCore_1.64.0   XML_3.99-0.16.1        
  [7] lifecycle_1.0.4         rstatix_0.7.2           edgeR_4.0.16           
 [10] rprojroot_2.0.4         processx_3.8.4          lattice_0.22-6         
 [13] MASS_7.3-60.0.1         backports_1.4.1         magrittr_2.0.3         
 [16] sass_0.4.9              rmarkdown_2.27          jquerylib_0.1.4        
 [19] yaml_2.3.8              httpuv_1.6.15           cowplot_1.1.3          
 [22] DBI_1.2.2               RColorBrewer_1.1-3      abind_1.4-5            
 [25] zlibbioc_1.48.2         ggraph_2.2.1            RCurl_1.98-1.14        
 [28] yulab.utils_0.1.4       tweenr_2.0.3            git2r_0.33.0           
 [31] GenomeInfoDbData_1.2.11 enrichplot_1.22.0       tidytree_0.4.6         
 [34] annotate_1.80.0         codetools_0.2-20        DelayedArray_0.28.0    
 [37] DOSE_3.28.2             ggforce_0.4.2           tidyselect_1.2.1       
 [40] aplot_0.2.2             farver_2.1.2            viridis_0.6.5          
 [43] jsonlite_1.8.8          tidygraph_1.3.1         survival_3.5-8         
 [46] ggnewscale_0.4.10       tools_4.3.2             treeio_1.26.0          
 [49] Rcpp_1.0.12             glue_1.7.0              gridExtra_2.3          
 [52] SparseArray_1.2.4       xfun_0.44               qvalue_2.34.0          
 [55] withr_3.0.0             BiocManager_1.30.22     fastmap_1.2.0          
 [58] fansi_1.0.6             callr_3.7.6             digest_0.6.35          
 [61] timechange_0.3.0        R6_2.5.1                gridGraphics_0.5-1     
 [64] colorspace_2.1-0        GO.db_3.18.0            RSQLite_2.3.6          
 [67] utf8_1.2.4              generics_0.1.3          data.table_1.15.4      
 [70] graphlayouts_1.1.1      httr_1.4.7              S4Arrays_1.2.1         
 [73] scatterpie_0.2.2        whisker_0.4.1           pkgconfig_2.0.3        
 [76] gtable_0.3.5            blob_1.2.4              XVector_0.42.0         
 [79] shadowtext_0.1.3        htmltools_0.5.8.1       carData_3.0-5          
 [82] fgsea_1.28.0            ggupset_0.3.0           scales_1.3.0           
 [85] png_0.1-8               ggfun_0.1.4             knitr_1.47             
 [88] rstudioapi_0.16.0       tzdb_0.4.0              reshape2_1.4.4         
 [91] cachem_1.1.0            parallel_4.3.2          HDO.db_0.99.1          
 [94] pillar_1.9.0            vctrs_0.6.5             ggpubr_0.6.0           
 [97] promises_1.3.0          car_3.1-2               xtable_1.8-4           
[100] evaluate_0.23           cli_3.6.2               locfit_1.5-9.9         
[103] compiler_4.3.2          rlang_1.1.4             crayon_1.5.2           
[106] ggsignif_0.6.4          labeling_0.4.3          ps_1.7.6               
[109] affy_1.80.0             getPass_0.2-4           plyr_1.8.9             
[112] fs_1.6.4                stringi_1.8.4           viridisLite_0.4.2      
[115] babelgene_22.9          munsell_0.5.1           Biostrings_2.70.3      
[118] lazyeval_0.2.2          GOSemSim_2.28.1         Matrix_1.6-5           
[121] hms_1.1.3               patchwork_1.2.0         bit64_4.0.5            
[124] KEGGREST_1.42.0         statmod_1.5.0           highr_0.11             
[127] broom_1.0.5             igraph_2.0.3            memoise_2.0.1          
[130] affyio_1.72.0           bslib_0.7.0             ggtree_3.10.1          
[133] fastmatch_1.1-4         bit_4.0.5               ape_5.8                
[136] gson_0.1.0             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
