<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Preprocessing: DIA Proteomics of Method Development cohort</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Preprocessing: DIA Proteomics of Method
Development cohort</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMART-CARE_LungCancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree9a5191e865f9c1d40a49ea46a7c1c020e986c53etargetblank9a5191ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/9a5191e865f9c1d40a49ea46a7c1c020e986c53e" target="_blank">9a5191e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancertree9a5191e865f9c1d40a49ea46a7c1c020e986c53etargetblank9a5191ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/tree/9a5191e865f9c1d40a49ea46a7c1c020e986c53e" target="_blank">9a5191e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/CrossCohort_investigation_EXP.Rmd
    Untracked:  code/archive/
    Untracked:  code/comparison_funcs.R
    Untracked:  code/dataset_list.R
    Untracked:  code/workflowr_commands.R
    Untracked:  data/Discovery/
    Untracked:  data/MethodDev/
    Untracked:  data/aliquot_metadata.xlsx
    Untracked:  data/patient_metadata.xlsx
    Untracked:  data/sample_metadata.xlsx
    Untracked:  output/Discovery/
    Untracked:  output/MethodDev/
    Untracked:  output/SC_meeting/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/MethodDev_01_preprocessing_DIA.Rmd</code>) and HTML
(<code>docs/MethodDev_01_preprocessing_DIA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/b83f6736534c624c72cb444130bf22766780a846/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">b83f673</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-05-15
</td>
<td>
Just for quick update of analysis report webpage
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/df818f2365f9e09d63fc82c5fb78ed7a7c78749c/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">df818f2</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-04-21
</td>
<td>
Redo investigation of impacts of sample missing levels on data and
retrain MOFA models using only common samples
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/fddf694f5651a6def9cf517cfc70d65591bbf119/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">fddf694</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-04-04
</td>
<td>
Redo investigation of association between data missingness and processed
data using PCA, instead of PPCA, and improve some parts in untargeted
lipidomics analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/5c2e33e66899c6f03e720181a0a4100db5e8a483/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">5c2e33e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-03-22
</td>
<td>
Fix patient recurrence metadata and check associations between DIA
proteomics and sample median expressions and missing levels
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/75eb25420814d279cfe825d9e110660f475a6408/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">75eb254</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2024-03-18
</td>
<td>
Correct sample time points and conditions in metadata
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer/8f6fa9088e1c6ead26dbd5616cc5b50ce73922b8/docs/MethodDev_01_preprocessing_DIA.html" target="_blank">8f6fa90</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/4e9331f246f3cf1a92c7e2dbd73963ddc36d8d02/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">4e9331f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-12
</td>
<td>
Combine discovery and method development data and do batch correction
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/MethodDev_01_preprocessing_DIA.html" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/e9ae79526a098e2bdd9b4446f1489f6d5dc45642/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">e9ae795</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-05
</td>
<td>
Update metadata
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/bba10115517c46c67704f4f5a2d707d3172ef1a5/analysis/MethodDev_01_preprocessing_DIA.Rmd" target="_blank">bba1011</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-11-17
</td>
<td>
Rearrange file locations and redo preprocessing of all method
development datasets
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Preprocess Tissue and Plasma DIA
Proteomics generated by Karim Aljakouch from AG Krijgsveld and Tissue
DIA Proteomics from AG Klingmüller, including data cleansing, filtering,
and normalization (VSN). All needed information was then stored in
SummarizedExperiment objects for further analyses. Note that there are
two Tissue and two Plasma DIA Proteomics generated by AG Krijgsveld, and
concentration is suggested on new ones owing to some technical problems
and missing samples with old ones. </font></p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;readr&#39;)
library(&#39;readxl&#39;)
library(&#39;vsn&#39;)
library(&#39;visdat&#39;)
library(&#39;ggrepel&#39;)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/analysis_pipeline.R&#39;)
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<pre class="r"><code># Load sample metadata
smpMetadat &lt;- readxl::read_excel(&#39;./data/sample_metadata.xlsx&#39;)
colnames(smpMetadat) &lt;- smpMetadat[3,, drop = F]
smpMetadat &lt;- dplyr::slice(smpMetadat, -c(1:3)) %&gt;%
  dplyr::select(c(Code, Parents, Visit, `Material submitted`, `Date and time of collection or surgery`,
                  `SMART-CARE cohort identifier`)) %&gt;%
  dplyr::rename(Sample = Code, Patient = Parents, TimePoint = Visit, SmpType = `Material submitted`,
                Date = `Date and time of collection or surgery`, Cohort = `SMART-CARE cohort identifier`) %&gt;%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, &#39;^SC_T_S_|^SC_DIS_S_|^SC_S_|_P&#39;),
                Patient = stringr::str_remove(Patient, &#39;^/THRX_SPACE/THRX_DB/&#39;),
                TimePoint = dplyr::case_when(TimePoint == &#39;PRETHERAPEUTIC&#39; ~ &#39;Baseline&#39;,
                                             TimePoint == &#39;FOLLOW-UP&#39; ~ &#39;Follow-up&#39;,
                                             TimePoint == &#39;RECURRENCE&#39; ~ &#39;Recurrence&#39;),
                SmpType = dplyr::case_when(SmpType == &#39;EDTA_PLASMA&#39; ~ &#39;Plasma&#39;,
                                           SmpType == &#39;FRESH_FROZEN_TISSUE&#39; ~ &#39;Tissue&#39;),
                Condition = dplyr::case_when(grepl(&#39;_TU|_TG&#39;, Sample) ~ &#39;Tumor&#39;,
                                             grepl(&#39;_NG&#39;, Sample) ~ &#39;Normal&#39;,
                                             !grepl(&#39;_TU|_TG|_NG&#39;, Sample) ~ TimePoint),
                TimePoint = dplyr::case_when(TimePoint %in% &#39;Recurrence&#39; ~ &#39;Follow-up&#39;,
                                             !TimePoint %in% &#39;Recurrence&#39; ~ TimePoint),
                Date = stringr::str_extract(Date, &#39;^\\d+-\\d+-\\d+&#39;),
                Cohort = dplyr::case_when(Cohort == &#39;DISCOVERY_COHORT&#39; ~ &#39;Discovery&#39;,
                                          Cohort == &#39;METHOD_DEVELOPMENT_COHORT&#39; ~ &#39;MethodDev&#39;),
                Date = as.Date(Date, format = &#39;%Y-%m-%d&#39;))

# Load patient metadata
patientMetadat &lt;- readxl::read_excel(&#39;./data/patient_metadata.xlsx&#39;)
colnames(patientMetadat) &lt;- patientMetadat[3,, drop = F]
patientMetadat &lt;- dplyr::slice(patientMetadat, -c(1:3)) %&gt;%
  dplyr::select(c(Code, Gender, `Age at diagnosis`, `Pathological stage`, `Smoking status`,
                  `Adjuvant chemotherapy`,)) %&gt;%
  dplyr::rename(Patient = Code, Age = `Age at diagnosis`, Stage = `Pathological stage`,
                Smoking = `Smoking status`, Adjuvant = `Adjuvant chemotherapy`) %&gt;%
  dplyr::mutate(Gender = dplyr::case_when(Gender == &#39;MALE&#39; ~ &#39;Male&#39;,
                                          Gender == &#39;FEMALE&#39; ~ &#39;Female&#39;),
                Smoking = dplyr::case_when(Smoking == &#39;SMOKER&#39; ~ &#39;Smoker&#39;,
                                           Smoking == &#39;EX-SMOKER&#39; ~ &#39;Ex-smoker&#39;,
                                           Smoking == &#39;NON-SMOKER&#39; ~ &#39;Non-smoker&#39;),
                Adjuvant = dplyr::case_when(Adjuvant == &#39;true&#39; ~ &#39;True&#39;,
                                            Adjuvant == &#39;false&#39; ~ &#39;False&#39;),
                Age = as.numeric(Age))
# Include patient recurrence information in patient metadata
recurPats &lt;- dplyr::filter(smpMetadat, Condition == &#39;Recurrence&#39;) %&gt;%
  dplyr::pull(Patient)
patientMetadat &lt;- dplyr::mutate(patientMetadat, Recurrence = dplyr::case_when(Patient %in% recurPats ~ &#39;Yes&#39;,
                                                                              !Patient %in% recurPats ~ &#39;No&#39;))

# Load aliquot metadata
aliquotMetadat &lt;- readxl::read_excel(&#39;./data/aliquot_metadata.xlsx&#39;)
colnames(aliquotMetadat) &lt;- aliquotMetadat[3,, drop = F]
aliquotMetadat &lt;- dplyr::slice(aliquotMetadat, -c(1:3)) %&gt;%
  dplyr::select(Code, Parents, `Tumor Cell Content (%)`, `Cohort Identifier`, `Submission to`,
                `Delivered?`) %&gt;%
  dplyr::rename(Aliquot = Code, Sample = Parents, TumorPurity = `Tumor Cell Content (%)`,
                Cohort = `Cohort Identifier`, To = `Submission to`, Delivered = `Delivered?`) %&gt;%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_T_S_|_P&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_DIS_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_S_&#39;),
                Sample = stringr::str_remove(Sample, &#39;^/THRX_SPACE/THRX_DB/SC_DIS_&#39;),
                Cohort = dplyr::case_when(Cohort == &#39;method establishment&#39; ~ &#39;MethodDev&#39;,
                                          Cohort == &#39;DISCOVERY_COHORT&#39; ~ &#39;Discovery&#39;),
                TumorPurity = as.numeric(TumorPurity)) %&gt;%
  dplyr::filter(To %in% c(&#39;KRIJGSVELD&#39;, &#39;KLINGMUELLER&#39;))
# Prepare aliquot metadata containing tumor purity information
tumorPurityInfo &lt;- dplyr::select(aliquotMetadat, Aliquot, Sample, TumorPurity)

# Combine all needed metadata
summMetadat &lt;- dplyr::left_join(tumorPurityInfo, smpMetadat, by = &#39;Sample&#39;) %&gt;%
  dplyr::left_join(patientMetadat, by = &#39;Patient&#39;)

# List sample annotations to keep in SE objects
smpAnno &lt;- c(&#39;Patient&#39;, &#39;SmpType&#39;, &#39;TimePoint&#39;, &#39;Date&#39;, &#39;Cohort&#39;, &#39;Condition&#39;,
             &#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Adjuvant&#39;)</code></pre>
<div id="tissue-proteomics-ag-krijgsveld" class="section level1">
<h1>Tissue Proteomics (AG Krijgsveld)</h1>
<div id="new" class="section level2">
<h2>New</h2>
<p>Old DIA Tissue Proteomics generated by AG Krijgsveld appears not to
have good quality because sample conditions are not major variance in
data, i.e., Tumor and Normal samples are not nicely separated. Check
visualization down below in ‘Old’ part</p>
<pre class="r"><code># Load and tidy up unpreprocessed data and include summarized metadata into it
new_proTissueTab &lt;- readr::read_delim(&#39;./data/MethodDev/AG_Krijgsveld/20230726_Thorax_Method_Est_tissue.txt&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Aliquot&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Aliquot = stringr::str_remove_all(Aliquot, &#39;\\.raw$&#39;)) %&gt;%
  dplyr::left_join(summMetadat, by = &#39;Aliquot&#39;)</code></pre>
<p>Show dimensions of original data (40 samples and 7537 features)</p>
<pre class="r"><code># Preprocess data
new_proTissue &lt;- doPreprocessing(new_proTissueTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                 val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                 smpAnno = c(smpAnno, &#39;TumorPurity&#39;), do_featFilt = T,
                                 cutoff = 0.5, viz_miss = T)#, save_path = &#39;./data/MethodDev/AG_Krijgsveld/new_proTissue&#39;)
dim(new_proTissue$ori.data)</code></pre>
<pre><code>[1] 7537   40</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code>new_proTissue$ori.data.miss</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-3-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Show distribution of original data</p>
<pre class="r"><code>new_proTissue$ori.data.dist</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-4-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (40 samples and 6614 features)</p>
<pre class="r"><code>dim(new_proTissue$filt.data)</code></pre>
<pre><code>[1] 6614   40</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>new_proTissue$filt.data.miss</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-6-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code>new_proTissue$vsn.data.dist</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-7-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>new_proTissue$vsn.feat.mean.var</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-8-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Do metadata-assisted quality control</strong><br />
PC1 significantly separates Tumor and Normal samples, which implies fine
data quality.</p>
<pre class="r"><code># Do single-omics analysis
new_proTissueSOA &lt;- doSOA(new_proTissue$vsn.data, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
                          do_onlyPCA = T)
pcTab &lt;- new_proTissueSOA$pcTab
ggplot(pcTab, aes(x=Condition, y=`PC1 (24.2%)`, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-9-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Investigate impacts of sample missing levels on observed
features</strong><br />
Display relationship between sample missing levels and median
expressions computed by fully observed features</p>
<pre class="r"><code># Normalize unfiltered data
proTissueUnfiltVsn &lt;- doPreprocessing(new_proTissueTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                      val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                      smpAnno = c(smpAnno, &#39;TumorPurity&#39;), do_featFilt = F)$vsn.data
# Prepare long data for computing sample missing levels and median expressions
proTissueUnfiltVsnTab &lt;- summExp2df(proTissueUnfiltVsn, assay = &#39;Abundance&#39;,
                                    row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;) %&gt;%
  # Prepare information of sample missing levels
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(OriMissLevel = sum(is.na(Value)) / length(Value)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Prepare information of sample median expressions computed by fully observed features
  dplyr::group_by(Protein.Group) %&gt;%
  dplyr::mutate(Keep = dplyr::case_when(!any(is.na(Value)) ~ &#39;Yes&#39;,
                                        any(is.na(Value)) ~ &#39;No&#39;)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::filter(Keep %in% &#39;Yes&#39;) %&gt;%
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(MedianVsnAbun = median(Value)) %&gt;%
  dplyr::ungroup()
# Create SE object containing information of sample missing levels and median expressions
colAnno &lt;- dplyr::select(proTissueUnfiltVsnTab, Sample, OriMissLevel, MedianVsnAbun) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
proTissueUnfiltVsn &lt;- SummarizedExperiment(assays = list(Abundance = assay(proTissueUnfiltVsn)),
                                           colData = colAnno)

# Visualize relationship between sample missing levels and median expressions
ggplot(proTissueUnfiltVsnTab, aes(x=OriMissLevel, y=MedianVsnAbun)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(x = &#39;Missing level&#39;, y = &#39;Median expression&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show significant associations between data and sample missing
levels</p>
<pre class="r"><code># Do PCA
pcaRes &lt;- doSOA(proTissueUnfiltVsn, meta_var = &#39;OriMissLevel&#39;, pca_method = &#39;pca&#39;,
                num_PCs = 20, do_onlyPCA = T)
# Show significant associations between PCs and sample missing levels
pcaRes$pcSigAssoRes %&gt;%
  dplyr::filter(grepl(&#39;PC[1-9] \\(&#39;, Var1))</code></pre>
<pre><code>         Var1         Var2    pVal pValAdj   Stat        Test
1 PC1 (45.8%) OriMissLevel 0.00066  0.0264 -0.516 Correlation
2  PC7 (2.8%) OriMissLevel 0.00235  0.0471  0.467 Correlation
3  PC3 (5.9%) OriMissLevel 0.00413  0.0551  0.444 Correlation</code></pre>
<pre class="r"><code># Visualize relationship between PCs and sample missing levels
pcTab &lt;- pcaRes$pcTab
ggplot(pcTab, aes(x=`PC1 (45.8%)`, y=OriMissLevel)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(y = &#39;Missing level&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="old" class="section level2">
<h2>Old</h2>
<p>Q: Samples from Patient ‘P3457YLC’, ‘93417JCT’, and ‘G33TNUQD’ are
missing? A: They were dropped out from study.</p>
<pre class="r"><code># Load and tidy up unpreprocessed data and include summarized metadata into it
proTissueTab &lt;- readr::read_tsv(&#39;./data/MethodDev/AG_Krijgsveld/20230104_MarcS_Method_Estab_Tissue.pg_matrix.tsv&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Aliquot&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Aliquot = stringr::str_remove_all(Aliquot, &#39;D:.*oecf4.*Exploris_|\\.raw&#39;)) %&gt;%
  dplyr::left_join(summMetadat, by = &#39;Aliquot&#39;)
# Preprocess data
proTissue &lt;- doPreprocessing(proTissueTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                             val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                             smpAnno = c(smpAnno, &#39;TumorPurity&#39;), do_featFilt = T,
                             cutoff = 0.5, viz_miss = T)</code></pre>
<p>Show dimensions of normalized filtered data (34 samples and 5746
features)</p>
<pre class="r"><code>dim(proTissue$vsn.data)</code></pre>
<pre><code>[1] 5746   34</code></pre>
<p><strong>Do metadata-assisted quality control</strong><br />
Visualize significant PCs associated with sample conditions</p>
<pre class="r"><code># Do single-omics analysis
proTissueSOA &lt;- doSOA(proTissue$vsn.data, meta_var = &#39;Condition&#39;, pca_method = &#39;ppca&#39;,
                      do_onlyPCA = T)
proTissueSOA$pcSigAssoRes</code></pre>
<pre><code>         Var1      Var2   pVal pValAdj  Stat   Test
1 PC19 (1.6%) Condition 0.0112   0.208 -2.69 T-test
2  PC6 (5.2%) Condition 0.0208   0.208  2.43 T-test</code></pre>
<pre class="r"><code>pcTab &lt;- proTissueSOA$pcTab
ggplot(pcTab, aes(x=Condition, y=`PC19 (1.6%)`, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Condition, y=`PC6 (5.2%)`, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-14-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plasma-proteomics-ag-krijgsveld" class="section level1">
<h1>Plasma Proteomics (AG Krijgsveld)</h1>
<div id="new-1" class="section level2">
<h2>New</h2>
<pre class="r"><code># Load unpreprocessed data
new_proPlasmaTab &lt;- readr::read_delim(&#39;./data/MethodDev/AG_Krijgsveld/20230726_Thorax_Method_Est_plasma.txt&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Aliquot&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Aliquot = stringr::str_remove_all(Aliquot, &#39;\\.raw$&#39;)) %&gt;%
  dplyr::left_join(summMetadat, by = &#39;Aliquot&#39;)</code></pre>
<p>Show dimensions of original data (40 samples and 1097 features)</p>
<pre class="r"><code># Preprocess data
new_proPlasma &lt;- doPreprocessing(new_proPlasmaTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                 val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                 smpAnno = smpAnno, do_featFilt = T, cutoff = 0.5,
                                 viz_miss = T, bins = 30)#, save_path = &#39;./data/MethodDev/AG_Krijgsveld/new_proPlasma&#39;)
dim(new_proPlasma$ori.data)</code></pre>
<pre><code>[1] 1097   40</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code>new_proPlasma$ori.data.miss</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show distribution of original data</p>
<pre class="r"><code>new_proPlasma$ori.data.dist</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-18-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (40 samples and 640 features)</p>
<pre class="r"><code>dim(new_proPlasma$filt.data)</code></pre>
<pre><code>[1] 640  40</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>new_proPlasma$filt.data.miss</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-20-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code>new_proPlasma$vsn.data.dist</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>new_proPlasma$vsn.feat.mean.var</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Investigate impacts of sample missing levels on observed
features</strong><br />
Display relationship between sample missing levels and median
expressions computed by fully observed features</p>
<pre class="r"><code># Normalize unfiltered data
proPlasmaUnfiltVsn &lt;- doPreprocessing(new_proPlasmaTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                                      val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                                      smpAnno = smpAnno, do_featFilt = F)$vsn.data
# Prepare long data for computing sample missing levels and median expressions
proPlasmaUnfiltVsnTab &lt;- summExp2df(proPlasmaUnfiltVsn, assay = &#39;Abundance&#39;,
                                    row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;) %&gt;%
  # Prepare information of sample missing levels
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(OriMissLevel = sum(is.na(Value)) / length(Value)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Prepare information of sample median expressions computed by fully observed features
  dplyr::group_by(Protein.Group) %&gt;%
  dplyr::mutate(Keep = dplyr::case_when(!any(is.na(Value)) ~ &#39;Yes&#39;,
                                        any(is.na(Value)) ~ &#39;No&#39;)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::filter(Keep %in% &#39;Yes&#39;) %&gt;%
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(MedianVsnAbun = median(Value)) %&gt;%
  dplyr::ungroup()
# Create SE object containing information of sample missing levels and median expressions
colAnno &lt;- dplyr::select(proPlasmaUnfiltVsnTab, Sample, OriMissLevel, MedianVsnAbun) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
proPlasmaUnfiltVsn &lt;- SummarizedExperiment(assays = list(Abundance = assay(proPlasmaUnfiltVsn)),
                                           colData = colAnno)

# Visualize relationship between sample missing levels and median expressions
ggplot(proPlasmaUnfiltVsnTab, aes(x=OriMissLevel, y=MedianVsnAbun)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(x = &#39;Missing level&#39;, y = &#39;Median expression&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-23-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Show significant associations between data and sample missing
levels</p>
<pre class="r"><code># Do PCA
pcaRes &lt;- doSOA(proPlasmaUnfiltVsn, meta_var = &#39;OriMissLevel&#39;, pca_method = &#39;pca&#39;,
                num_PCs = 20, do_onlyPCA = T)
# Show significant associations between PCs and sample missing levels
pcaRes$pcSigAssoRes %&gt;%
  dplyr::filter(grepl(&#39;PC[1-9] \\(&#39;, Var1))</code></pre>
<pre><code>         Var1         Var2     pVal  pValAdj   Stat        Test
1 PC2 (13.6%) OriMissLevel 7.13e-06 0.000285  0.645 Correlation
2  PC6 (3.6%) OriMissLevel 6.88e-04 0.013800 -0.514 Correlation</code></pre>
<pre class="r"><code># Visualize relationship between PCs and sample missing levels
pcTab &lt;- pcaRes$pcTab
ggplot(pcTab, aes(x=`PC2 (13.6%)`, y=OriMissLevel)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(y = &#39;Missing level&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-24-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="old-1" class="section level2">
<h2>Old</h2>
<p>Samples from Patient ‘P3457YLC’, ‘93417JCT’, and ‘G33TNUQD’ are
missing.</p>
<pre class="r"><code># Load and tidy up unpreprocessed data and include summarized metadata into it
proPlasmaTab &lt;- readr::read_tsv(&#39;./data/MethodDev/AG_Krijgsveld/20230104_MarcS_Method_Estab_Plasma.pg_matrix.tsv&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Aliquot&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Aliquot = stringr::str_remove_all(Aliquot, &#39;D:.*oecf4.*Exploris_|\\.raw&#39;)) %&gt;%
  dplyr::left_join(summMetadat, by = &#39;Aliquot&#39;)</code></pre>
<p>Show dimensions of original data (34 samples and 505 features)</p>
<pre class="r"><code># Preprocess data
proPlasma &lt;- doPreprocessing(proPlasmaTab, feat = &#39;Protein.Group&#39;, smp = &#39;Sample&#39;,
                             val = &#39;Abundance&#39;, featAnno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                             smpAnno = smpAnno, do_featFilt = T, cutoff = 0.5,
                             viz_miss = T, bins = 30)#, save_path = &#39;./data/MethodDev/AG_Krijgsveld/proPlasma&#39;)
dim(proPlasma$ori.data)</code></pre>
<pre><code>[1] 505  34</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code>proPlasma$ori.data.miss</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-27-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Show distribution of original data</p>
<pre class="r"><code>proPlasma$ori.data.dist</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-28-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (34 samples and 446 features)</p>
<pre class="r"><code>dim(proPlasma$filt.data)</code></pre>
<pre><code>[1] 446  34</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>proPlasma$filt.data.miss</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code>proPlasma$vsn.data.dist</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-31-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>proPlasma$vsn.feat.mean.var</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-32-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tissue-proteomics-ag-klingmüller" class="section level1">
<h1>Tissue Proteomics (AG Klingmüller)</h1>
<p>Q: Samples from Method Development Patient ‘M2J6VHZ3’ and ‘1HU6PFFQ’
are missing?<br />
Samples from 21 Discovery Patients were measured.</p>
<pre class="r"><code># Load and tidy unpreprocessed data
proTissueTab_Klin &lt;- readr::read_tsv(
  &#39;./data/MethodDev/AG_Klingmueller/20230309_103228_SmartCare_TumorFree_Tumor_Yang001_Report.tsv&#39;
) %&gt;%
  dplyr::select(-c(PG.ProteinDescriptions, contains(&#39;PG.NrOfPrecursorsIdentified&#39;),
                   contains(&#39;PG.IBAQ&#39;), contains(&#39;PG.IsIdentified&#39;))) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;PG.ProteinGroups&#39;, &#39;PG.Genes&#39;),
                      names_to = &#39;Experiment_ID&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Experiment_ID = stringr::str_remove_all(Experiment_ID, &#39;\\[.*\\] |\\.PG\\.Quantity&#39;),
                Abundance = gsub(&#39;,&#39;, &#39;\\.&#39;, Abundance),
                Abundance = as.numeric(Abundance))
# Convert NaN to NA for vsn
proTissueTab_Klin$Abundance[is.nan(proTissueTab_Klin$Abundance)] &lt;- NA

# Load experimental ID annotations
idAnnoTab &lt;- readxl::read_excel(&#39;./data/MethodDev/AG_Klingmueller/230620_SMART-CARE_overview_OpenBIS_BH.xlsx&#39;) %&gt;%
  dplyr::select(`Experimental ID`, `Aliquot ID`, Batch) %&gt;%
  dplyr::rename(Aliquot = `Aliquot ID`, Experiment_ID = `Experimental ID`) %&gt;%
  dplyr::mutate(Batch = dplyr::case_when(Batch == &#39;first&#39; ~ &#39;1&#39;, Batch == &#39;second&#39; ~ &#39;2&#39;))
# Manually correct typo
idAnnoTab[which(idAnnoTab$Experiment_ID == &#39;Recurrence _TumorFree_P21&#39;), &#39;Experiment_ID&#39;] &lt;- &#39;Recurrence_TumorFree_P21&#39;
idAnnoTab[which(idAnnoTab$Experiment_ID == &#39;Recurrence _Tumor_P21&#39;), &#39;Experiment_ID&#39;] &lt;- &#39;Recurrence_Tumor_P21&#39;

# Include summarized metadata
proTissueTab_Klin &lt;- dplyr::left_join(proTissueTab_Klin, idAnnoTab, by = &#39;Experiment_ID&#39;) %&gt;%
  dplyr::left_join(summMetadat, by = &#39;Aliquot&#39;)</code></pre>
<p>Show dimensions of original data (78 samples and 9659 features)</p>
<pre class="r"><code># Preprocess data
proTissue_Klin &lt;- doPreprocessing(proTissueTab_Klin, feat = &#39;PG.ProteinGroups&#39;,
                                  smp = &#39;Sample&#39;, val = &#39;Abundance&#39;, featAnno = &#39;PG.Genes&#39;,
                                  smpAnno = c(smpAnno, &#39;TumorPurity&#39;, &#39;Batch&#39;),
                                  do_featFilt = T, cutoff = 0.5, viz_miss = T)
# save_path = &#39;./data/MethodDev/AG_Klingmueller/proTissue&#39;)
dim(proTissue_Klin$ori.data)</code></pre>
<pre><code>[1] 9659   78</code></pre>
<p>Show missingness of original data</p>
<pre class="r"><code>proTissue_Klin$ori.data.miss +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-35-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Show distribution of original data</p>
<pre class="r"><code>proTissue_Klin$ori.data.dist +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-36-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Remove features quantified in less than 1/2 of
samples</strong><br />
Show dimensions of filtered data (78 samples and 8250 features)</p>
<pre class="r"><code>dim(proTissue_Klin$filt.data)</code></pre>
<pre><code>[1] 8250   78</code></pre>
<p>Show missingness of filtered data</p>
<pre class="r"><code>proTissue_Klin$filt.data.miss +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/8f6fa9088e1c6ead26dbd5616cc5b50ce73922b8/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-38-1.png" target="_blank">8f6fa90</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer/blob/d71a18f5e66ebfd60aab06bf6e44877fcd01d701/docs/figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-38-1.png" target="_blank">d71a18f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Normalize filtered data by VSN</strong><br />
Show distribution of vsn normalized data</p>
<pre class="r"><code>proTissue_Klin$vsn.data.dist +
  theme(axis.text.x = element_text(size = 10))</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show feature mean-variance relationship of vsn normalized data</p>
<pre class="r"><code>proTissue_Klin$vsn.feat.mean.var</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Do metadata-assisted quality control</strong><br />
Visualize significant PCs associated with sample conditions and batches
(1 is Method Development cohort and 2 is Discovery cohort)</p>
<pre class="r"><code># Do single-omics analysis
proTissueSOA_Klin &lt;- doSOA(proTissue_Klin$vsn.data, meta_var = c(&#39;Condition&#39;, &#39;Batch&#39;),
                           pca_method = &#39;ppca&#39;, do_onlyPCA = T)
proTissueSOA_Klin$pcSigAssoRes</code></pre>
<pre><code>         Var1      Var2     pVal  pValAdj  Stat   Test
1 PC1 (31.1%) Condition 4.33e-32 1.73e-30 20.10 T-test
2  PC8 (3.5%)     Batch 7.35e-03 1.47e-01  2.75 T-test
3 PC12 (2.6%)     Batch 1.10e-02 1.47e-01 -2.61 T-test
4 PC14 (2.4%)     Batch 1.91e-02 1.91e-01  2.39 T-test
5 PC2 (11.3%)     Batch 3.03e-02 2.32e-01  2.21 T-test
6 PC17 (1.9%)     Batch 3.48e-02 2.32e-01  2.15 T-test
7  PC5 (4.5%)     Batch 4.49e-02 2.57e-01 -2.04 T-test</code></pre>
<pre class="r"><code># Prepare labels for samples of interest
pcTab &lt;- proTissueSOA_Klin$pcTab %&gt;%
  dplyr::mutate(Label = dplyr::case_when(Sample %in% c(&#39;I0HVOL_TU&#39;, &#39;7EAOX7_TU&#39;,
                                                       &#39;MJMTYR_TU&#39;, &#39;XFKHGP_TU&#39;,
                                                       &#39;XFKHGP_NG&#39;, &#39;MJMTYR_NG&#39;,
                                                       &#39;I0HVOL_NG&#39;, &#39;7EAOX7_NG&#39;) ~ Sample))
ggplot(pcTab, aes(x=Condition, y=`PC1 (31.1%)`, col=Condition, fill=Condition, label = Label)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  geom_text_repel(show.legend = F, size = 5) +
  scale_color_brewer(palette = &#39;Dark2&#39;) +
  scale_fill_brewer(palette = &#39;Dark2&#39;) +
  labs(x = &#39;Sample condition&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-41-1.png" width="960" style="display: block; margin: auto;" />
=&gt; Tumor purity: I0HVOL_TU - 87% (Mean of sample tumor cell contents
is 93%, ranging from 73% to 100%.)</p>
<p><strong>Hierarchical clustering</strong><br />
Check if there exists batch effect</p>
<pre class="r"><code># Compute sample euclidean distances
d &lt;- dist(t(assay(proTissue_Klin$vsn.data)), method = &#39;euclidean&#39;)

# Visualize sample distances by heatmap with clustering
batchAnno &lt;- tibble::as_tibble(colData(proTissue_Klin$vsn.data), rownames = &#39;Sample&#39;) %&gt;%
  dplyr::select(Sample, Batch) %&gt;%
  dplyr::mutate(Batch = dplyr::case_when(Batch == &#39;1&#39; ~ &#39;MethodDev&#39;,
                                         Batch == &#39;2&#39; ~ &#39;Discovery&#39;)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap(as.matrix(d), annotation_col = batchAnno, scale = &#39;row&#39;, #row scaling is across columns
         color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
         cluster_cols = T, cluster_rows = F, show_rownames = F,
         clustering_method = &#39;ward.D2&#39;, fontsize_col = 6,
         annotation_colors = list(Batch = c(MethodDev = &#39;grey30&#39;, Discovery = &#39;grey60&#39;)))</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-42-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Do hierarchical clustering
# hc &lt;- hclust(d, method = &#39;ward.D2&#39;)
# # Plot clustering result
# plot(hc, hang = 0.04, cex = 0.5, frame.plot = T, main = NULL)</code></pre>
<p><strong>PCA</strong><br />
Check if there exists potential batch effect</p>
<pre class="r"><code>ggplot(pcTab, aes(x=`PC1 (31.1%)`, y=`PC2 (11.3%)`, col=Batch, shape=Condition)) +
  geom_point(size = 4) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_shape_manual(values = c(0, 16)) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Batch, y=`PC8 (3.5%)`, col=Batch, fill=Batch)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_fill_brewer(palette = &#39;Set2&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, show.legend = F) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-43-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Batch, y=`PC12 (2.6%)`, col=Batch, fill=Batch)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_fill_brewer(palette = &#39;Set2&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, show.legend = F) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-43-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Batch, y=`PC5 (4.5%)`, col=Batch, fill=Batch)) +
  geom_boxplot(alpha = 0.7, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.3), size = 3, show.legend = F) +
  scale_color_brewer(palette = &#39;Set2&#39;) +
  scale_fill_brewer(palette = &#39;Set2&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, show.legend = F) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-43-4.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Do batch correction using limma</strong><br />
There is no PC significantly associated with batches after batch
correction, which supports that batch effects are linear. Batch factor
(Batch or Cohort) could be included in linear model for statistical
analysis.</p>
<pre class="r"><code># Perform batch correction to see if two batches can be combined even better
# Convert SE object into wide data
datMat &lt;- SummarizedExperiment::assay(proTissue_Klin$vsn.data)
# Define design matrix to keep particular effects, e.g., treatments
smpAnnoTab &lt;- as_tibble(colData(proTissue_Klin$vsn.data), rownames = &#39;Sample&#39;)
design &lt;- model.matrix(~ smpAnnoTab$Condition + smpAnnoTab$Recurrence)
# Do batch correction
datMat_BC &lt;- limma::removeBatchEffect(datMat, batch = smpAnnoTab$Batch, design = design)
proTissueSE_BC &lt;- proTissue_Klin$vsn.data
assay(proTissueSE_BC) &lt;- datMat_BC
# Save batch corrected data
# saveRDS(proTissueSE_BC, &#39;./data/MethodDev/AG_Klingmueller/proTissueVsnBC.rds&#39;)

# Do single-omics analysis
proTissueSOA_BC &lt;- doSOA(proTissueSE_BC, meta_var = c(&#39;Condition&#39;, &#39;Batch&#39;),
                         pca_method = &#39;ppca&#39;, do_onlyPCA = T)
proTissueSOA_BC$pcSigAssoRes</code></pre>
<pre><code>         Var1      Var2     pVal  pValAdj Stat   Test
1 PC1 (31.8%) Condition 1.88e-32 7.51e-31 20.3 T-test</code></pre>
<p><strong>Investigate impacts of sample missing levels on observed
features</strong><br />
Display relationship between sample missing levels and median
expressions computed by fully observed features</p>
<pre class="r"><code># Normalize unfiltered data
proTissueUnfiltVsn &lt;- doPreprocessing(proTissueTab_Klin, feat = &#39;PG.ProteinGroups&#39;,
                                      smp = &#39;Sample&#39;, val = &#39;Abundance&#39;, featAnno = &#39;PG.Genes&#39;,
                                      smpAnno = c(smpAnno, &#39;TumorPurity&#39;, &#39;Batch&#39;),
                                      do_featFilt = F)$vsn.data
# Prepare long data for computing sample missing levels and median expressions
proTissueUnfiltVsnTab &lt;- summExp2df(proTissueUnfiltVsn, assay = &#39;Abundance&#39;,
                                    row_id = &#39;PG.ProteinGroups&#39;, col_id = &#39;Sample&#39;) %&gt;%
  # Prepare information of sample missing levels
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(OriMissLevel = sum(is.na(Value)) / length(Value)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Prepare information of sample median expressions computed by fully observed features
  dplyr::group_by(PG.ProteinGroups) %&gt;%
  dplyr::mutate(Keep = dplyr::case_when(!any(is.na(Value)) ~ &#39;Yes&#39;,
                                        any(is.na(Value)) ~ &#39;No&#39;)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::filter(Keep %in% &#39;Yes&#39;) %&gt;%
  dplyr::group_by(Sample) %&gt;%
  dplyr::mutate(MedianVsnAbun = median(Value)) %&gt;%
  dplyr::ungroup()
# Create SE object containing information of sample missing levels and median expressions
colAnno &lt;- dplyr::select(proTissueUnfiltVsnTab, Sample, OriMissLevel, MedianVsnAbun) %&gt;%
  dplyr::filter(!duplicated(Sample)) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
proTissueUnfiltVsn &lt;- SummarizedExperiment(assays = list(Abundance = assay(proTissueUnfiltVsn)),
                                           colData = colAnno)

# Visualize relationship between sample missing levels and median expressions
ggplot(proTissueUnfiltVsnTab, aes(x=OriMissLevel, y=MedianVsnAbun)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(x = &#39;Missing level&#39;, y = &#39;Median expression&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show significant associations between data and sample missing
levels</p>
<pre class="r"><code># Do PCA
pcaRes &lt;- doSOA(proTissueUnfiltVsn, meta_var = &#39;OriMissLevel&#39;, pca_method = &#39;pca&#39;,
                num_PCs = 20, do_onlyPCA = T)
# Show significant associations between PCs and sample missing levels
pcaRes$pcSigAssoRes %&gt;%
  dplyr::filter(grepl(&#39;PC[1-9] \\(&#39;, Var1))</code></pre>
<pre><code>         Var1         Var2     pVal pValAdj   Stat        Test
1 PC1 (34.3%) OriMissLevel 2.35e-05 0.00112 -0.459 Correlation
2  PC3 (7.5%) OriMissLevel 2.86e-05 0.00112  0.455 Correlation
3  PC2 (9.2%) OriMissLevel 1.82e-03 0.04740 -0.348 Correlation</code></pre>
<pre class="r"><code># Visualize relationship between PCs and sample missing levels
pcTab &lt;- pcaRes$pcTab
ggplot(pcTab, aes(x=`PC1 (34.3%)`, y=OriMissLevel)) +
  geom_point(size = 3.5) +
  ggpubr::stat_cor(aes(label=after_stat(r.label)), method = &#39;pearson&#39;, size = 6) +
  labs(y = &#39;Missing level&#39;) +
  th</code></pre>
<p><img src="figure/MethodDev_01_preprocessing_DIA.Rmd/unnamed-chunk-46-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] proDA_1.16.0                pcaMethods_1.94.0          
 [3] pheatmap_1.0.12             ggplotify_0.1.2            
 [5] limma_3.58.1                lubridate_1.9.3            
 [7] forcats_1.0.0               stringr_1.5.1              
 [9] dplyr_1.1.4                 purrr_1.0.2                
[11] tidyr_1.3.1                 tibble_3.2.1               
[13] tidyverse_2.0.0             SummarizedExperiment_1.32.0
[15] GenomicRanges_1.54.1        GenomeInfoDb_1.38.8        
[17] IRanges_2.36.0              S4Vectors_0.40.2           
[19] MatrixGenerics_1.14.0       matrixStats_1.3.0          
[21] ggrepel_0.9.5               ggplot2_3.5.1              
[23] visdat_0.6.0                vsn_3.70.0                 
[25] Biobase_2.62.0              BiocGenerics_0.48.1        
[27] readxl_1.4.3                readr_2.1.5                
[29] workflowr_1.7.1            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7            rlang_1.1.3             magrittr_2.0.3         
 [4] git2r_0.33.0            compiler_4.3.2          getPass_0.2-4          
 [7] callr_3.7.6             vctrs_0.6.5             pkgconfig_2.0.3        
[10] crayon_1.5.2            fastmap_1.1.1           backports_1.4.1        
[13] XVector_0.42.0          labeling_0.4.3          utf8_1.2.4             
[16] promises_1.3.0          rmarkdown_2.26          tzdb_0.4.0             
[19] ps_1.7.6                preprocessCore_1.64.0   bit_4.0.5              
[22] xfun_0.43               zlibbioc_1.48.2         cachem_1.0.8           
[25] jsonlite_1.8.8          highr_0.10              later_1.3.2            
[28] DelayedArray_0.28.0     broom_1.0.5             parallel_4.3.2         
[31] R6_2.5.1                bslib_0.7.0             stringi_1.8.3          
[34] RColorBrewer_1.1-3      car_3.1-2               jquerylib_0.1.4        
[37] cellranger_1.1.0        Rcpp_1.0.12             knitr_1.46             
[40] httpuv_1.6.15           Matrix_1.6-5            timechange_0.3.0       
[43] tidyselect_1.2.1        rstudioapi_0.16.0       abind_1.4-5            
[46] yaml_2.3.8              affy_1.80.0             processx_3.8.4         
[49] lattice_0.22-6          withr_3.0.0             evaluate_0.23          
[52] gridGraphics_0.5-1      ggpubr_0.6.0            pillar_1.9.0           
[55] affyio_1.72.0           BiocManager_1.30.22     carData_3.0-5          
[58] whisker_0.4.1           generics_0.1.3          vroom_1.6.5            
[61] rprojroot_2.0.4         RCurl_1.98-1.14         hms_1.1.3              
[64] munsell_0.5.1           scales_1.3.0            glue_1.7.0             
[67] tools_4.3.2             hexbin_1.28.3           ggsignif_0.6.4         
[70] fs_1.6.3                grid_4.3.2              colorspace_2.1-0       
[73] GenomeInfoDbData_1.2.11 cli_3.6.2               fansi_1.0.6            
[76] S4Arrays_1.2.1          gtable_0.3.5            rstatix_0.7.2          
[79] yulab.utils_0.1.4       sass_0.4.9              digest_0.6.35          
[82] SparseArray_1.2.4       farver_2.1.1            memoise_2.0.1          
[85] htmltools_0.5.8.1       lifecycle_1.0.4         httr_1.4.7             
[88] statmod_1.5.0           bit64_4.0.5            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
