<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>MOFA: Downstream analysis</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MOFA: Downstream analysis</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-09-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree16c8851769227ae8172f4e0537d6f1730a283dcftargetblank16c8851a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/16c8851769227ae8172f4e0537d6f1730a283dcf" target="_blank">16c8851</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree16c8851769227ae8172f4e0537d6f1730a283dcftargetblank16c8851a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/16c8851769227ae8172f4e0537d6f1730a283dcf" target="_blank">16c8851</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/feature_selection.Rmd
    Untracked:  code/Junyan/
    Untracked:  code/about.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/feature_selection_old.Rmd
    Untracked:  code/license.Rmd
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/mofa_varied_omics_combinations.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Klingmuller/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/MethodDev_LungCancer_Datasets.RData
    Untracked:  data/mofa/
    Untracked:  data/patient_metadata.tsv
    Untracked:  data/rf/
    Untracked:  data/stats/
    Untracked:  output/Latest/
    Untracked:  output/TAC_2023/
    Untracked:  output/YIG_seminar/
    Untracked:  output/preliminary_analysis_untargeted.html

Unstaged changes:
    Deleted:    analysis/mofa_varied_omics_combinations.Rmd
    Modified:   code/misc.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/mofa_downstream_analysis.Rmd</code>) and HTML
(<code>docs/mofa_downstream_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/7a352f974d64a04bea99c8d84c26045b997248ce/analysis/mofa_downstream_analysis.Rmd" target="_blank">7a352f9</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-08-29
</td>
<td>
Do enrichment analysis of trained MOFA model
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Do enrichment analysis of MOFA model
trained on DIA Plasma Proteomics (DIAPP), Untargeted Plasma Metabolomics
(Unt. PM), Untargeted Plasma Lipidomics (Unt. PL), and DIA Normal Tissue
Proteomics (DIANP) for biological interpretations. Note that DIAPP in
this script indicates latest data from AG Krijgsveld and DIANP indicates
AG Klingmüller one. </font><br />
</p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;MOFA2&#39;)
library(&#39;reticulate&#39;)
# use_python(&#39;/Users/qianwu/opt/anaconda3/bin/python&#39;)
library(&#39;AnnotationDbi&#39;)
library(&#39;org.Hs.eg.db&#39;)
# This annotation object is accessed using AnnotationDbi:select()
hs &lt;- org.Hs.eg.db
library(&#39;msigdbr&#39;)
library(&#39;clusterProfiler&#39;)
library(&#39;ggrepel&#39;)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)
source(&#39;./code/mofa_funcs.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<div id="dia-pp-unt.-pm-unt.-pl-dianp" class="section level1">
<h1>DIA PP + Unt. PM + Unt. PL + DIANP</h1>
<p>Overview model setup</p>
<pre class="r"><code># Load trained MOFA model
mofaObj &lt;- readRDS(&#39;./data/mofa/newDIAPP_UntPM_UntPL_DIANP_9Fac.rds&#39;)
# Overview model
MOFA2::plot_data_overview(mofaObj)</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show data variance explained by factors</p>
<pre class="r"><code># Show data variance explained by learned factors
MOFA2::plot_variance_explained(mofaObj, max_r2 = 10) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, vjust = 1, size = 13),
        axis.text.y = element_text(size = 15, face = &#39;bold&#39;),
        strip.text = element_text(size = 15),
        legend.title = element_text(size = 15), legend.text = element_text(size = 11))</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Latest/mofa_PP+UntPM_PL+NP_varExplained.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p>Show significant factor</p>
<pre class="r"><code># Show significant recurrence-related factors
sigFactor &lt;- mofa_vizSigFactor(mofaObj, smpGroup = &#39;Baseline Group&#39;)</code></pre>
<pre><code>     Var1       Var2       pVal   pValAdj     Stat   Test
1 Factor5 Recurrence 0.01939999 0.1745999 2.566932 T-test</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Show features with highest weights, which is most associated with
recurrence patients</p>
<pre class="r"><code># Show top features of significant factor with highest absolute weights
# Prepare feature annotation table
# DIA Tissue Proteomics (Klingmüller)
pro_gene_DIA_Klin &lt;- rowData(readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)) %&gt;%
  tibble::as_tibble(rownames = &#39;Proteins&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Proteins&#39;)
# Modify feature names
tmp_mofaObj &lt;- mofa_rmFeatSuffix(mofaObj, view = &#39;DIA Normal Proteomics&#39;,
                                 feat_anno = list(`DIA Normal Proteomics` = pro_gene_DIA_Klin))

MOFA2::plot_top_weights(tmp_mofaObj, view = &#39;DIA Normal Proteomics&#39;, factor = 5,
                        nfeatures = 10, scale = T, sign = &#39;negative&#39;) +
  labs(title = &#39;DIA Normal Tissue Proteomics&#39;) +
  theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 16), axis.text.y = element_text(size = 20, face = &#39;bold&#39;),
        strip.text = element_text(size = 20, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Latest/mofa_PP+UntPM_PL+NP_topWeightedFeats_NP.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 5)


# Untargeted Plasma Lipidomics
mzrt &lt;- rowData(readRDS(&#39;./data/AG_Hopf/lipPlasmaVsn.rds&#39;)) %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Feature&#39;)
# Modify feature names
tmp_mofaObj &lt;- mofa_rmFeatSuffix(mofaObj, view = &#39;Untargeted Plasma Lipidomics&#39;,
                                 feat_anno = list(`Untargeted Plasma Lipidomics` = mzrt))

MOFA2::plot_top_weights(tmp_mofaObj, view = &#39;Untargeted Plasma Lipidomics&#39;, factor = 5,
                        nfeatures = 10, scale = T, sign = &#39;negative&#39;) +
  labs(title = &#39;Untargeted Plasma Lipidomics&#39;) +
  theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 16), axis.text.y = element_text(size = 20, face = &#39;bold&#39;),
        strip.text = element_text(size = 20, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-5-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&#39;./output/Latest/mofa_PP+UntPM_PL+NP_topWeightedFeats_PL.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 5)</code></pre>
<div id="enrichment-analysis" class="section level2">
<h2>Enrichment analysis</h2>
<pre class="r"><code># Tidy up weight matrix and feature metadata to reduce information loss
# Prepare feature annotation table to extract proteins&#39; gene annotations
# DIA Tissue Proteomics (Klingmüller)
pro_gene_DIA_Klin &lt;- rowData(readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)) %&gt;%
  tibble::as_tibble(rownames = &#39;Proteins&#39;) %&gt;%
  tibble::column_to_rownames(&#39;Proteins&#39;)
proNormalWeight &lt;- mofa_keepUniFeats(mofaObj, factor = 5, view = &#39;DIA Normal Proteomics&#39;,
                                     feat_anno = pro_gene_DIA_Klin, to_genes = T)

# Rank feature weights
rankedGeneList &lt;- mofa_rankFeatList(proNormalWeight, factor = 5)
# Make Recurrence group positive weights
rankedGeneList &lt;- sort(-rankedGeneList, decreasing = T)</code></pre>
<div id="curated-gene-sets" class="section level3">
<h3>Curated gene sets</h3>
<p>MSigDB - C2</p>
<pre class="r"><code># Curated gene sets - DIA Normal Proteomics
# View supported species and collections in MSigDB
# msigdbr_species()
# msigdbr_collections()
# Retrieve annotated gene sets
msigTabC2 &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;C2&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)

# Run GSEA
gseaResC2 &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabC2,
                                   minGSSize = 10, maxGSSize = 500,
                                   pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                   by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResC2@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                # Description = stringr::str_remove(Description, &#39;^REACTOME_&#39;),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.7, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
  #       axis.text.x = element_text(size = 18),
  #       axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
  #       legend.title = element_text(size = 20),
  #       legend.text = element_text(size = 16),
  #       panel.grid.minor = element_blank())
# ggsave(&#39;./output/Latest/mofa_PP+UntPM_PL+NP_EA_C2.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 13)

# ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust)) +
#   geom_point(aes(size = gene_ratio), shape = 21) +
#   scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;,
#                        midpoint = midGrad, name = &#39;Adjusted p-value&#39;) +
#   scale_size_continuous(name = &#39;Core gene ratio&#39;) +
#   geom_vline(xintercept = 0) +
#   labs(x = &#39;Normalized Enrichment Score (NES)&#39;, y = &#39;Pathway&#39;) +
#   theme_minimal()

# enrichplot::dotplot(gseaResC2, x = &#39;GeneRatio&#39;, color = &#39;p.adjust&#39;, showCategory = 10,
#                     font.size = 12, title = &#39;DIA Normal Proteomics (Curated)&#39;)</code></pre>
<p>=&gt; Increased uptake, synthesis, and storage of lipids has been
demonstrated to contribute to growth of many cancer types, including
lung cancer.</p>
</div>
<div id="oncogenic-signature-gene-sets" class="section level3">
<h3>Oncogenic signature gene sets</h3>
<p>MSigDB - C6</p>
<pre class="r"><code># Oncogenic signatures gene sets - DIA Normal Proteomics
# Retrieve annotated gene sets
msigTabC6 &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;C6&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)
# Retrieve descriptions of gene sets for interpretation
descripC6 &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;C6&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, gs_description) %&gt;%
  dplyr::filter(!duplicated(gs_name))

# Run GSEA
gseaResC6 &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabC6,
                                   minGSSize = 10, maxGSSize = 500,
                                   pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                   by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResC6@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.6, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(axis.title = element_text(size = 20, face = &#39;bold&#39;),
  #       axis.text.x = element_text(size = 18),
  #       axis.text.y = element_text(size = 18, face = &#39;bold&#39;),
  #       legend.title = element_text(size = 20),
  #       legend.text = element_text(size = 16),
  #       panel.grid.minor = element_blank())
# ggsave(&#39;./output/Latest/mofa_PP+UntPM_PL+NP_EA_C6.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 10)</code></pre>
<p>=&gt; Aberrant activation of KRAS and mTOR and suppression of p53 has
been reported to be associated with lung cancer. mTOR signaling pathway
regulates many fundamental cellular activities, including lipid
metabolism.</p>
</div>
<div id="hallmark-gene-sets" class="section level3">
<h3>Hallmark gene sets</h3>
<p>MSigDB - H</p>
<pre class="r"><code># Hallmark gene sets - DIA Normal Proteomics
# Retrieve annotated gene sets
msigTabH &lt;- msigdbr(species = &#39;Homo sapiens&#39;, category = &#39;H&#39;, subcategory = NULL) %&gt;%
  dplyr::select(gs_name, human_gene_symbol)

# Run GSEA
gseaResH &lt;- clusterProfiler::GSEA(geneList = rankedGeneList, TERM2GENE = msigTabH,
                                  minGSSize = 10, maxGSSize = 500,
                                  pvalueCutoff = 0.05, pAdjustMethod = &#39;BH&#39;,
                                  by = &#39;fgsea&#39;, eps = 0)

# Plot enrichment analysis results
gseaRes4Plot &lt;- gseaResH@result %&gt;%
  dplyr::select(Description, setSize, NES, pvalue, p.adjust, leading_edge) %&gt;%
  dplyr::arrange(p.adjust) %&gt;%
  dplyr::slice_head(n = 8) %&gt;%
  dplyr::mutate(leading_edge = stringr::str_extract(leading_edge, &#39;\\d\\d&#39;),
                leading_edge = as.numeric(leading_edge),
                Recurrence = ifelse(test = NES &gt; 0, yes = &#39;Yes&#39;, no = &#39;No&#39;)) %&gt;%
  dplyr::rename(gene_ratio = leading_edge)
midGrad &lt;- mean(gseaRes4Plot$p.adjust)

ggplot(gseaRes4Plot, aes(x=NES, y=reorder(Description, NES), fill=p.adjust, col=Recurrence)) +
  geom_bar(stat = &#39;identity&#39;, width = 0.6, size = 2) +
  scale_fill_gradient2(low=&#39;#5B1C00&#39;, high=&#39;#FFFCA5&#39;, mid = &#39;#E86900&#39;, midpoint = midGrad, name = &#39;Adjusted pVal&#39;) +
  scale_color_manual(values = c(Yes = &#39;red&#39;, No = &#39;darkblue&#39;)) +
  labs(x = &#39;Normalized Enrichment Score&#39;, y = &#39;Pathway&#39;) +
  theme_minimal() +
  theme(axis.title = element_text(size = 16, face = &#39;bold&#39;),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 10, face = &#39;bold&#39;),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_downstream_analysis.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.2                
 [5] purrr_1.0.1                 readr_2.1.4                
 [7] tidyr_1.3.0                 tibble_3.2.1               
 [9] tidyverse_2.0.0             SummarizedExperiment_1.30.2
[11] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[13] MatrixGenerics_1.12.2       matrixStats_1.0.0          
[15] ggrepel_0.9.3               ggplot2_3.4.2              
[17] clusterProfiler_4.8.1       msigdbr_7.5.1              
[19] org.Hs.eg.db_3.17.0         AnnotationDbi_1.62.2       
[21] IRanges_2.34.1              S4Vectors_0.38.1           
[23] Biobase_2.60.0              BiocGenerics_0.46.0        
[25] reticulate_1.30             MOFA2_1.10.0               
[27] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] splines_4.3.0           later_1.3.1             bitops_1.0-7           
  [4] ggplotify_0.1.1         filelock_1.0.2          polyclip_1.10-4        
  [7] basilisk.utils_1.12.1   lifecycle_1.0.3         rstatix_0.7.2          
 [10] rprojroot_2.0.3         processx_3.8.2          lattice_0.21-8         
 [13] MASS_7.3-60             backports_1.4.1         magrittr_2.0.3         
 [16] sass_0.4.7              rmarkdown_2.23          jquerylib_0.1.4        
 [19] yaml_2.3.7              httpuv_1.6.11           cowplot_1.1.1          
 [22] DBI_1.1.3               RColorBrewer_1.1-3      abind_1.4-5            
 [25] zlibbioc_1.46.0         Rtsne_0.16              ggraph_2.1.0           
 [28] RCurl_1.98-1.12         yulab.utils_0.0.6       tweenr_2.0.2           
 [31] git2r_0.32.0            GenomeInfoDbData_1.2.10 enrichplot_1.20.0      
 [34] tidytree_0.4.4          pheatmap_1.0.12         codetools_0.2-19       
 [37] DelayedArray_0.26.6     DOSE_3.26.1             ggforce_0.4.1          
 [40] tidyselect_1.2.0        aplot_0.1.10            farver_2.1.1           
 [43] viridis_0.6.3           jsonlite_1.8.7          tidygraph_1.2.3        
 [46] tools_4.3.0             treeio_1.24.1           Rcpp_1.0.11            
 [49] glue_1.6.2              gridExtra_2.3           xfun_0.39              
 [52] qvalue_2.32.0           HDF5Array_1.28.1        withr_2.5.0            
 [55] fastmap_1.1.1           basilisk_1.11.2         rhdf5filters_1.12.1    
 [58] fansi_1.0.4             callr_3.7.3             digest_0.6.33          
 [61] timechange_0.2.0        R6_2.5.1                gridGraphics_0.5-1     
 [64] colorspace_2.1-0        GO.db_3.17.0            RSQLite_2.3.1          
 [67] utf8_1.2.3              generics_0.1.3          data.table_1.14.8      
 [70] graphlayouts_1.0.0      httr_1.4.6              S4Arrays_1.0.4         
 [73] scatterpie_0.2.1        whisker_0.4.1           uwot_0.1.16            
 [76] pkgconfig_2.0.3         gtable_0.3.3            blob_1.2.4             
 [79] XVector_0.40.0          shadowtext_0.1.2        htmltools_0.5.5        
 [82] carData_3.0-5           fgsea_1.26.0            scales_1.2.1           
 [85] png_0.1-8               corrplot_0.92           ggfun_0.1.1            
 [88] knitr_1.43              rstudioapi_0.15.0       tzdb_0.4.0             
 [91] reshape2_1.4.4          nlme_3.1-162            cachem_1.0.8           
 [94] rhdf5_2.44.0            parallel_4.3.0          HDO.db_0.99.1          
 [97] pillar_1.9.0            grid_4.3.0              vctrs_0.6.3            
[100] ggpubr_0.6.0            promises_1.2.0.1        car_3.1-2              
[103] evaluate_0.21           cli_3.6.1               compiler_4.3.0         
[106] rlang_1.1.1             crayon_1.5.2            ggsignif_0.6.4         
[109] labeling_0.4.2          ps_1.7.5                getPass_0.2-2          
[112] plyr_1.8.8              fs_1.6.2                stringi_1.7.12         
[115] viridisLite_0.4.2       BiocParallel_1.34.2     babelgene_22.9         
[118] munsell_0.5.0           Biostrings_2.68.1       lazyeval_0.2.2         
[121] GOSemSim_2.26.0         Matrix_1.6-0            dir.expiry_1.8.0       
[124] hms_1.1.3               patchwork_1.1.2         bit64_4.0.5            
[127] Rhdf5lib_1.22.0         KEGGREST_1.40.0         highr_0.10             
[130] broom_1.0.5             igraph_1.5.0            memoise_2.0.1          
[133] bslib_0.5.0             ggtree_3.8.0            fastmatch_1.1-3        
[136] bit_4.0.5               downloader_0.4          ape_5.7-1              
[139] gson_0.1.0             </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
