<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>MOFA: Training models using varied omics combinations</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MOFA: Training models using varied omics
combinations</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree80c5e2dc52362aacd28d1d54cc0cd2600a1f7110targetblank80c5e2da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/80c5e2dc52362aacd28d1d54cc0cd2600a1f7110" target="_blank">80c5e2d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree80c5e2dc52362aacd28d1d54cc0cd2600a1f7110targetblank80c5e2da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/80c5e2dc52362aacd28d1d54cc0cd2600a1f7110" target="_blank">80c5e2d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  code/about.Rmd
    Untracked:  code/feature_selection.Rmd
    Untracked:  code/license.Rmd
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/corrMetaPlasmaRes_Hopf_Hell.rds
    Untracked:  data/corrMetaTissueRes_Hopf_Hell.rds
    Untracked:  data/mofa/
    Untracked:  output/AG_Hopf/
    Untracked:  output/network_tar_untar_PM.png
    Untracked:  output/percentage_sigFeats.png
    Untracked:  output/percentage_sigFeats2.png

Unstaged changes:
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd
    Modified:   analysis/preliminary_analysis_DIA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/mofa_varied_omics_combinations.Rmd</code>) and HTML
(<code>docs/mofa_varied_omics_combinations.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/70459c11aabe575c447b2ecba6cd3cf3914d5b0a/analysis/mofa_varied_omics_combinations.Rmd" target="_blank">70459c1</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-07
</td>
<td>
use new untargeted features filtered by retention time to rerun
comparisons_targeted_untargeted.Rmd and
analysis/mofa_varied_omics_combinations.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/60e2be1baf6c417a1fac3ca5b43772501dff8cbb/docs/mofa_varied_omics_combinations.html" target="_blank">60e2be1</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/mofa_varied_omics_combinations.html" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/e54886fedde7f02bc105d3e5a84a3a7f63455652/analysis/mofa_varied_omics_combinations.Rmd" target="_blank">e54886f</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
<td>
Finalize scripts about data preprocessing, preliminary analysis,
comparisons between targeted and untargeted
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Train MOFA models using varied omics
data combinations and observe training results to gain some insights in
terms of predicting patient cancer recurrences, thereby focusing on
Baseline sample group. Criteria for comparisons among models are (1)
filtering out trained models without any learned Recurrence-related
factor, and then (2) comparing t-statistics (p-values) between those
significant factors. Since all models share same sample space (Method
Development cohort), it allows us to compare results directly by
p-values. All omics datasets were generated from the following three
research groups:<br />
AG Hell: Targeted (1) Plasma Metabolomics and (2) Tissue
Metabolomics<br />
AG Krijgsveld: DDA (1) Plasma Proteomics and (2) Tissue Proteomics<br />
AG Hopf: Untargeted (1) Plasma Metabolomics, (2) Plasma Lipidomics, (3)
Tissue Metabolomics, and (4) Tissue Lipidomics<br />
Among them, Plasma (and Tissue) Metabolomics from AG Hell and AG Hopf
was generated using different methods (Targeted vs Untargeted).
Therefore, comparisons are mainly between models trained with
combinations containing either of these two datasets to see if
Untargeted datasets can potentially provide more information. However,
later, we found that correlations between Targeted and Untargeted
metabolite features are overall not that high. Therefore, we also
trained models with omics data including both Targeted and Untargeted
datasets. Lastly, we observed that there exists this outlier Patient
‘H03Y07RT’ in most of trained models, otherwise without this patient,
many models can be quite comparable. Hence, we excluded this patient and
redid association tests between Recurrence-related factors and cancer
recurrence. Generally, model trained on Untargeted Plasma Metabolomics +
Plasma Lipidomics + Plasma Proteomics best performs on predicting
patient cancer recurrences. This result is exciting! </font><br />
<strong>Note that new datasets Untargeted Tissue Metabolomics and
Lipidomics are not included in this script.</strong></p>
<p>Load libraries</p>
<pre class="r"><code>library(MOFA2)
library(reticulate)
use_python(&#39;/Users/qianwu/opt/anaconda3/bin/python&#39;)
library(ggrepel)
library(SummarizedExperiment)
library(tidyverse)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)</code></pre>
<pre class="r"><code># Load normalized datasets
# Tissue proteomics
proTissue &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm.rds&#39;)
# Tissue metabolomics
metaTissue &lt;- readRDS(&#39;./data/AG_Hell/metaTissueNorm.rds&#39;)
# Plasma proteomics
proPlasma &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm.rds&#39;)
# Plasma metabolomics (Hell)
metaPlasma_Hell &lt;- readRDS(&#39;./data/AG_Hell/metaPlasmaNorm.rds&#39;)
# Plasma metabolomics (Hopf)
metaPlasma_Hopf &lt;- readRDS(&#39;./data/AG_Hopf/metaPlasmaVsn.rds&#39;)
# Plasma lipidomics
lipPlasma &lt;- readRDS(&#39;./data/AG_Hopf/lipPlasmaVsn.rds&#39;)

# Change column names to interpretable ones and manually modify potential mislabeling
# Tissue proteomics
colnames(proTissue) &lt;- colData(proTissue)$Sample
colnames(proTissue)[22] &lt;- &#39;F04ERF3M_TG&#39;
colData(proTissue)$Sample &lt;- colnames(proTissue)
colData(proTissue)$Patient[22] &lt;- &#39;F04ERF3M&#39;
colData(proTissue)$Recurrence[22] &lt;- &#39;Yes&#39;
# Plasma proteomics
colnames(proPlasma) &lt;- colData(proPlasma)$Sample

# Retrieve recurrence annotations of Baseline samples for heatmap
recurBaseAnno &lt;- tibble::as_tibble(colData(proPlasma)) %&gt;%
  dplyr::filter(Condition == &#39;Baseline&#39;) %&gt;%
  dplyr::select(Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)</code></pre>
<p>Display dimensions of each dataset (Feature X Sample)</p>
<pre class="r"><code># Overview dimensionality of each dataset
lungCancerData &lt;- list(Tissue_Proteomics = assay(proTissue),
                       Tissue_Metabolomics = assay(metaTissue),
                       Plasma_Proteomics = assay(proPlasma),
                       Plasma_Metabolomics_Hell = assay(metaPlasma_Hell),
                       Plasma_Metabolomics_Hopf = assay(metaPlasma_Hopf),
                       Plasma_Lipidomics = assay(lipPlasma))
lapply(lungCancerData, dim)</code></pre>
<pre><code>$Tissue_Proteomics
[1] 5193   40

$Tissue_Metabolomics
[1] 630  40

$Plasma_Proteomics
[1] 435  40

$Plasma_Metabolomics_Hell
[1] 630  40

$Plasma_Metabolomics_Hopf
[1] 974  40

$Plasma_Lipidomics
[1] 2456   40</code></pre>
<pre class="r"><code># lapply(lungCancerData, colnames)</code></pre>
<pre class="r"><code># Convert SE objects to long data for creating MOFA object through
# &#39;create_mofa_from_df&#39; to include metadata
# Tissue proteomics
proTissueTab &lt;- summExp2df(proTissue, row_id = &#39;feature&#39;, col_id = &#39;sample&#39;) %&gt;%
  dplyr::select(-c(PG.Genes, Sample_IDs, Sample)) %&gt;%
  dplyr::rename(value = Value) %&gt;%
  dplyr::mutate(group = &#39;Tissue&#39;, view = &#39;Proteomics&#39;)
# Tissue metabolomics
metaTissueTab &lt;- summExp2df(metaTissue, row_id = &#39;feature&#39;, col_id = &#39;sample&#39;) %&gt;%
  dplyr::rename(value = Value) %&gt;%
  dplyr::mutate(group = &#39;Tissue&#39;, view = &#39;Metabolomics&#39;)
# Plasma proteomics
proPlasmaTab &lt;- summExp2df(proPlasma, row_id = &#39;feature&#39;, col_id = &#39;sample&#39;) %&gt;%
  dplyr::select(-c(PG.Genes, Sample_IDs, Sample)) %&gt;%
  dplyr::rename(value = Value) %&gt;%
  dplyr::mutate(group = &#39;Plasma&#39;, view = &#39;Proteomics&#39;)
# Plasma metabolomics (Hell)
metaPlasmaTab_Hell &lt;- summExp2df(metaPlasma_Hell, row_id = &#39;feature&#39;,
                                 col_id = &#39;sample&#39;) %&gt;%
  dplyr::rename(value = Value) %&gt;%
  dplyr::mutate(group = &#39;Plasma&#39;, view = &#39;Metabolomics&#39;)
# Plasma metabolomics (Hopf)
metaPlasmaTab_Hopf &lt;- summExp2df(metaPlasma_Hopf, row_id = &#39;feature&#39;,
                                 col_id = &#39;sample&#39;) %&gt;%
  dplyr::select(-m.z_RT) %&gt;%
  dplyr::rename(value = Value) %&gt;%
  dplyr::mutate(group = &#39;Plasma&#39;, view = &#39;Metabolomics&#39;)
# Plasma lipidomics
lipPlasmaTab &lt;- summExp2df(lipPlasma, row_id = &#39;feature&#39;, col_id = &#39;sample&#39;) %&gt;%
  dplyr::select(-m.z_RT) %&gt;%
  dplyr::rename(value = Value) %&gt;%
  dplyr::mutate(group = &#39;Plasma&#39;, view = &#39;Lipidomics&#39;)

# Differential tissue proteomics
# proTissueDiffTab &lt;- summExp2df(proTissueDiff, row_id = &#39;feature&#39;,
#                                col_id = &#39;sample&#39;) %&gt;%
#   dplyr::select(-PG.Genes) %&gt;%
#   dplyr::rename(value = Value) %&gt;%
#   dplyr::mutate(group = &#39;Tissue&#39;, view = &#39;Proteomics Diff&#39;)</code></pre>
<div id="pm-hell-pl-pp" class="section level1">
<h1>PM (Hell) + PL + PP</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics</strong><br />
In clinical situations, it would be much easier to obtain Plasma samples
compared to obtaining Tissue samples, so firstly only and all Plasma
omics was used to see model performance.<br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proPlasmaTab3V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(group = dplyr::case_when(
    Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
    Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
    ))

metaPlasmaTab3V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = dplyr::case_when(
    Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
    Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
    ))

lipPlasmaTab3V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = dplyr::case_when(
    Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
    Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
    ),
    sample = dplyr::case_when(
      Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
      Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
    ))

# Create MOFA object
# Concatenate long data
summTab3V2G &lt;- dplyr::bind_rows(proPlasmaTab3V2G,
                                metaPlasmaTab3V2G,
                                lipPlasmaTab3V2G)
mofaObject3V2G &lt;- MOFA2::create_mofa_from_df(summTab3V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject3V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_3V2G_10Fac.rds&#39;)

# Overview data in MOFA object
MOFA2::plot_data_overview(mofaObject3V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-1-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;3V2G_data_overview.png&#39;), device = &#39;png&#39;, dpi = 400)


# Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject3V2G)
# model_opts &lt;- get_default_model_options(mofaObject3V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject3V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject3V2G &lt;- prepare_mofa(mofaObject3V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject3V2G &lt;- run_mofa(mofaObject3V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_3V2G_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject3V2G, &#39;./data/mofa/Hell_mofa_3V2G_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# MOFA2::plot_factor_cor(mofaObject3V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject3V2G, max_r2 = 15)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-2-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/Hell_3V2G_var_explained.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2). Basically t-test was performed to test
significance of difference between recurrence and non-recurrence patient
groups.<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Baseline Plasma&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(Condition == &#39;Baseline&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2       pVal      tStats   pValAdj
1   Factor1 Recurrence 0.07444049 -1.89378369 0.4372505
2   Factor8 Recurrence 0.08745011 -1.80735569 0.4372505
3  Factor10 Recurrence 0.14358357  1.52926024 0.4786119
4   Factor9 Recurrence 0.24810427 -1.19366502 0.6202607
5   Factor5 Recurrence 0.58821019  0.55129511 0.9434001
6   Factor2 Recurrence 0.63037186  0.48954016 0.9434001
7   Factor3 Recurrence 0.81958053  0.23144392 0.9434001
8   Factor4 Recurrence 0.82152515  0.22890265 0.9434001
9   Factor7 Recurrence 0.87541896 -0.15902618 0.9434001
10  Factor6 Recurrence 0.94340009 -0.07199466 0.9434001</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Follow-up Plasma&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(Condition != &#39;Baseline&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor8 Recurrence 0.04070646 -2.204931 0.3035924</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
</p>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
MOFA2::plot_factor(mofaObject3V2G, factors = 8, color_by = &#39;Recurrence&#39;,
                   add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-7-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hopf-pl-pp" class="section level1">
<h1>PM (Hopf) + PL + PP</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
metaPlasmaTab3V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = dplyr::case_when(
    Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
    Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
    ),
    sample = dplyr::case_when(
      Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
      Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
    ))

# Create MOFA object
# Concatenate long data
summTab3V2G &lt;- dplyr::bind_rows(proPlasmaTab3V2G,
                                metaPlasmaTab3V2G,
                                lipPlasmaTab3V2G)
mofaObject3V2G &lt;- create_mofa_from_df(summTab3V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject3V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_3V2G_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject3V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-8-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject3V2G)
# model_opts &lt;- get_default_model_options(mofaObject3V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject3V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject3V2G &lt;- prepare_mofa(mofaObject3V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject3V2G &lt;- run_mofa(mofaObject3V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_3V2G_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject3V2G, &#39;./data/mofa/Hopf_mofa_3V2G_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject3V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject3V2G, max_r2 = 15)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-9-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Baseline Plasma&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(Condition == &#39;Baseline&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal   tStats   pValAdj
1 Factor8 Recurrence 0.01409272 2.718299 0.1409272
2 Factor6 Recurrence 0.02949090 2.364564 0.1474545</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Follow-up Plasma&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(Condition != &#39;Baseline&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor1 Recurrence 0.03230461 -2.319811 0.2528873</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject3V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-12-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;Hopf_3V2G_baseline_factor8.png&#39;), device = &#39;png&#39;, dpi = 400)

plot_factor(mofaObject3V2G, factors = 6, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  # geom_label_repel(aes(label=sample), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-12-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-12-2.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;Hopf_3V2G_baseline_factor6.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<pre class="r"><code># Display molecular signatures captured by Factor 6 in data through heatmap

# Molecular signatures in data
data_heatmap(mofaObject3V2G, factor = 6, view = &#39;Proteomics&#39;,
             group = &#39;Baseline Plasma&#39;, features = 30,
             smp_anno = recurBaseAnno, smp_cluster = &#39;Recurrence&#39;,
             cluster_rows = T, cluster_cols = T,
             show_rownames = F, show_colnames = T,
             color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
             scale = &#39;row&#39;, main = &#39;Baseline Plasma Proteomics - Fac6&#39;)
data_heatmap(mofaObject3V2G, factor = 6, view = &#39;Lipidomics&#39;,
             group = &#39;Baseline Plasma&#39;, features = 30,
             smp_anno = recurBaseAnno, smp_cluster = &#39;Recurrence&#39;,
             cluster_rows = T, cluster_cols = T,
             show_rownames = T, show_colnames = T,
             color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
             scale = &#39;row&#39;, main = &#39;Baseline Plasma Lipidomics - Fac6&#39;)</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject3V2G, factors = 1, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-14-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hell-pl-pp-ttp-ntp" class="section level1">
<h1>PM (Hell) + PL + PP + TTP + NTP</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Tumor Tissue</strong> <strong>Proteomics + Normal Tissue
Proteomics</strong><br />
The idea is to see if adding Tissue Proteomics can improve model
performance.<br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab5V2G &lt;- proTissueTab %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = dplyr::case_when(
                  Condition == &#39;Tumor&#39; ~ paste(&#39;Tumor&#39;, view),
                  Condition == &#39;Normal&#39; ~ paste(&#39;Normal&#39;, view)
                  ),
                sample = stringr::str_replace(sample, &#39;_TG|_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

proPlasmaTab5V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                view = &#39;Plasma Proteomics&#39;)%&gt;%
  dplyr::select(-Condition)

metaPlasmaTab5V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab5V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab5V2G &lt;- dplyr::bind_rows(proTissueTab5V2G,
                                proPlasmaTab5V2G,
                                metaPlasmaTab5V2G,
                                lipPlasmaTab5V2G)
mofaObject5V2G &lt;- create_mofa_from_df(summTab5V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject5V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_5V2G_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject5V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-15-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;5V2G_data_overview.png&#39;), device = &#39;png&#39;, dpi = 400)


# Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject5V2G)
# model_opts &lt;- get_default_model_options(mofaObject5V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject5V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject5V2G &lt;- prepare_mofa(mofaObject5V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject5V2G &lt;- run_mofa(mofaObject5V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_5V2G_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject5V2G, &#39;./data/mofa/Hell_mofa_5V2G_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject5V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject5V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-16-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2). Basically t-test was performed to test
significance of difference between recurrence and non-recurrence patient
groups.<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2       pVal     tStats   pValAdj
1   Factor8 Recurrence 0.09973083  1.7355495 0.6249813
2   Factor1 Recurrence 0.12499627  1.6090817 0.6249813
3   Factor9 Recurrence 0.36464583 -0.9300611 0.8925392
4   Factor3 Recurrence 0.50926573  0.6733710 0.8925392
5   Factor4 Recurrence 0.54542382 -0.6162825 0.8925392
6   Factor6 Recurrence 0.60373700 -0.5283108 0.8925392
7   Factor2 Recurrence 0.65087528 -0.4602113 0.8925392
8   Factor5 Recurrence 0.71887420  0.3656688 0.8925392
9   Factor7 Recurrence 0.80328525 -0.2528026 0.8925392
10 Factor10 Recurrence 0.95261447  0.0602576 0.9526145</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal   tStats   pValAdj
1 Factor8 Recurrence 0.03942931 2.220894 0.2808324</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Follow-up group</strong></p>
<pre class="r"><code># View factor values for samples
# Follow-up
plot_factor(mofaObject5V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-19-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hopf-pl-pp-ttp-ntp" class="section level1">
<h1>PM (Hopf) + PL + PP + TTP + NTP</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Tumor</strong> <strong>Tissue Proteomics + Normal Tissue
Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
metaPlasmaTab5V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab5V2G &lt;- dplyr::bind_rows(proTissueTab5V2G,
                                proPlasmaTab5V2G,
                                metaPlasmaTab5V2G,
                                lipPlasmaTab5V2G)
mofaObject5V2G &lt;- create_mofa_from_df(summTab5V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject5V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_5V2G_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject5V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-20-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject5V2G)
# model_opts &lt;- get_default_model_options(mofaObject5V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject5V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject5V2G &lt;- prepare_mofa(mofaObject5V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject5V2G &lt;- run_mofa(mofaObject5V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_5V2G_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject5V2G, &#39;./data/mofa/Hopf_mofa_5V2G_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject5V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject5V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-21-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor8 Recurrence 0.02426571 -2.459432 0.2426571</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor5 Recurrence 0.04811215 -2.120531 0.3000412</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject5V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  # geom_label_repel(aes(label=sample), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-24-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;Hopf_5V2G_baseline_factor10.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<pre class="r"><code># Molecular signatures in data
data_heatmap(mofaObject5V2G, factor = 8, view = &#39;Plasma Proteomics&#39;,
             group = &#39;Baseline Group&#39;, features = 30,
             smp_anno = recurBaseAnno, smp_cluster = &#39;Recurrence&#39;,
             cluster_rows = T, cluster_cols = T,
             show_rownames = F, show_colnames = T,
             color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
             scale = &#39;row&#39;, main = &#39;Baseline Plasma Proteomics - Fac10&#39;)
data_heatmap(mofaObject5V2G, factor = 8, view = &#39;Lipidomics&#39;,
             group = &#39;Baseline Group&#39;, features = 30,
             smp_anno = recurBaseAnno, smp_cluster = &#39;Recurrence&#39;,
             cluster_rows = T, cluster_cols = T,
             show_rownames = T, show_colnames = T,
             color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
             scale = &#39;row&#39;, main = &#39;Baseline Plasma Lipidomics - Fac10&#39;)</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject5V2G, factors = 5, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-26-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hell-pl-ttp-ntp" class="section level1">
<h1>PM (Hell) + PL + TTP + NTP</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Tumor
Tissue Proteomics + Normal</strong> <strong>Tissue
Proteomics</strong><br />
The idea is to observe changes in model performance if Plasma Proteomics
(that might provide signal or introduce noise) is removed.<br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab4V2G &lt;- proTissueTab %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = dplyr::case_when(
                  Condition == &#39;Tumor&#39; ~ paste(&#39;Tumor&#39;, view),
                  Condition == &#39;Normal&#39; ~ paste(&#39;Normal&#39;, view)
                  ),
                sample = stringr::str_replace(sample, &#39;_TG|_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

metaPlasmaTab4V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab4V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proTissueTab4V2G,
                                metaPlasmaTab4V2G,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_4V2G-PP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-27-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_4V2G-PP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/Hell_mofa_4V2G-PP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-28-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2). Basically t-test was performed to test
significance of difference between recurrence and non-recurrence patient
groups.<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2       pVal      tStats   pValAdj
1  Factor10 Recurrence 0.07159436  1.91445891 0.7159436
2   Factor4 Recurrence 0.26171705  1.15870886 0.8582685
3   Factor8 Recurrence 0.36963480 -0.92020392 0.8582685
4   Factor1 Recurrence 0.40084049  0.86048107 0.8582685
5   Factor3 Recurrence 0.47488059  0.72983861 0.8582685
6   Factor7 Recurrence 0.51496109  0.66423235 0.8582685
7   Factor9 Recurrence 0.79824269 -0.25943617 0.9550910
8   Factor2 Recurrence 0.79877805 -0.25873132 0.9550910
9   Factor5 Recurrence 0.85958189 -0.17945914 0.9550910
10  Factor6 Recurrence 0.98810421 -0.01511818 0.9881042</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>      Var1       Var2       pVal tStats   pValAdj
1 Factor10 Recurrence 0.01526409 2.6807 0.1526409</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Follow-up group</strong></p>
<pre class="r"><code># View factor values for samples
# Follow-up
plot_factor(mofaObject4V2G, factors = 10, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-31-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hopf-pl-ttp-ntp" class="section level1">
<h1>PM (Hopf) + PL + TTP + NTP</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Tumor
Tissue Proteomics +</strong> <strong>Normal Tissue
Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
metaPlasmaTab4V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proTissueTab4V2G,
                                metaPlasmaTab4V2G,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_4V2G-PP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-32-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_4V2G-PP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/Hopf_mofa_4V2G-PP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-33-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>      Var1       Var2       pVal   tStats   pValAdj
1 Factor10 Recurrence 0.04478669 2.156844 0.4143585</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor7 Recurrence 0.01441642 -2.707619 0.1441642</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject4V2G, factors = 10, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-36-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject4V2G, factors = 7, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-37-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hell-pl-pp-ttp" class="section level1">
<h1>PM (Hell) + PL + PP + TTP</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Tumor</strong> <strong>Tissue Proteomics</strong><br />
The idea is to observe changes in model performance if Normal Tissue
Proteomics is removed.<br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab4V2G &lt;- proTissueTab %&gt;%
  dplyr::filter(Condition == &#39;Tumor&#39;) %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = paste(&#39;Tumor&#39;, view),
                sample = stringr::str_replace(sample, &#39;_TG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

proPlasmaTab4V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                view = &#39;Plasma Proteomics&#39;)%&gt;%
  dplyr::select(-Condition)

metaPlasmaTab4V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab4V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proTissueTab4V2G,
                                proPlasmaTab4V2G,
                                metaPlasmaTab4V2G,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_4V2G-NTP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-38-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_4V2G-NTP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/Hell_mofa_4V2G-NTP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-39-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal  tStats  pValAdj
1 Factor1 Recurrence 0.04992286 2.10171 0.488045</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal   tStats   pValAdj
1 Factor4 Recurrence 0.01627488 2.650407 0.1544435
2 Factor8 Recurrence 0.03088870 2.341857 0.1544435</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject4V2G, factors = 1, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-42-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-42-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># View factor values for samples
# Follow-up
plot_factor(mofaObject4V2G, factors = 4, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-43-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_factor(mofaObject4V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-43-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pm-hopf-pl-pp-ttp" class="section level1">
<h1>PM (Hopf) + PL + PP + TTP</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Tumor</strong> <strong>Tissue Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab4V2G &lt;- proTissueTab %&gt;%
  dplyr::filter(Condition == &#39;Tumor&#39;) %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = paste(&#39;Tumor&#39;, view),
                sample = stringr::str_replace(sample, &#39;_TG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

proPlasmaTab4V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                view = &#39;Plasma Proteomics&#39;)%&gt;%
  dplyr::select(-Condition)

metaPlasmaTab4V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab4V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proTissueTab4V2G,
                                proPlasmaTab4V2G,
                                metaPlasmaTab4V2G,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_4V2G-NTP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-44-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_4V2G-NTP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/Hopf_mofa_4V2G-NTP_10Fac.rds&#39;)

# Iterate model training to see differences in results
# n_inits &lt;- 5
# for (i in seq_len(n_inits)) {
#   # Create MOFA object
#   mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)
#
#   # Train MOFA model
#   data_opts &lt;- get_default_data_options(mofaObject4V2G)
#   model_opts &lt;- get_default_model_options(mofaObject4V2G)
#   model_opts$num_factors &lt;- 10
#   train_opts &lt;- get_default_training_options(mofaObject4V2G)
#   train_opts$convergence_mode &lt;- &#39;slow&#39;
#   train_opts$seed &lt;- sample(seq_len(1000), 1)
#
#   mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                  data_options = data_opts,
#                                  model_options = model_opts,
#                                  training_options = train_opts)
#
#   mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                              outfile = paste0(project_path,
#                                               &#39;data/mofa_model/Hopf_mofa_4V2G-NTP_10Fac&#39;,
#                                               &#39;_&#39;, i, &#39;.hdf5&#39;),
#                              use_basilisk = F)
#   saveRDS(mofaObject4V2G, paste0(project_path,
#                                  &#39;data/mofa_model/Hopf_mofa_4V2G-NTP_10Fac&#39;,
#                                  &#39;_&#39;, i, &#39;.rds&#39;))
# }</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2        pVal    tStats    pValAdj
1 Factor9 Recurrence 0.006135715  3.103237 0.06135715
2 Factor7 Recurrence 0.021934121 -2.508128 0.10967060</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor6 Recurrence 0.01639459  2.646939 0.1530585
2 Factor1 Recurrence 0.03061170 -2.346280 0.1530585</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject4V2G, factors = 9, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  # geom_label_repel(aes(label=sample), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-48-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-48-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_factor(mofaObject4V2G, factors = 7, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-48-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-2">
Past versions of unnamed-chunk-48-2.png
</button>
</p>
<div id="fig-unnamed-chunk-48-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-48-2.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject4V2G, factors = 6, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-49-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-49-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(project_path, &#39;results/Hopf_mofa_4V2G_fac7.png&#39;), device = &#39;png&#39;, dpi = 400)

plot_factor(mofaObject4V2G, factors = 1, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-49-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(project_path, &#39;results/Hopf_mofa_4V2G_fac1.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
</div>
<div id="pm-hell-pl-ttp" class="section level1">
<h1>PM (Hell) + PL + TTP</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Tumor
Tissue Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab3V2G &lt;- proTissueTab %&gt;%
  dplyr::filter(Condition == &#39;Tumor&#39;) %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = paste(&#39;Tumor&#39;, view),
                sample = stringr::str_replace(sample, &#39;_TG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

metaPlasmaTab3V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab3V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab3V2G &lt;- dplyr::bind_rows(proTissueTab3V2G,
                                metaPlasmaTab3V2G,
                                lipPlasmaTab3V2G)
mofaObject3V2G &lt;- create_mofa_from_df(summTab3V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject3V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_3V2G-NTP&amp;-PP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject3V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-50-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-50-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject3V2G)
# model_opts &lt;- get_default_model_options(mofaObject3V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject3V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject3V2G &lt;- prepare_mofa(mofaObject3V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject3V2G &lt;- run_mofa(mofaObject3V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_3V2G-NTP&amp;-PP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject3V2G, &#39;./data/mofa/Hell_mofa_3V2G-NTP&amp;-PP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject3V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject3V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-51-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2       pVal      tStats   pValAdj
1   Factor1 Recurrence 0.05796784  2.02493180 0.4767743
2   Factor8 Recurrence 0.09535485 -1.76020675 0.4767743
3  Factor10 Recurrence 0.15837070  1.47169778 0.5279023
4   Factor9 Recurrence 0.26797088 -1.14310646 0.6699272
5   Factor4 Recurrence 0.38084916 -0.89836897 0.7616983
6   Factor7 Recurrence 0.53414098  0.63386620 0.8902350
7   Factor2 Recurrence 0.67318616  0.42875866 0.9323366
8   Factor3 Recurrence 0.78882887  0.27185270 0.9323366
9   Factor5 Recurrence 0.83910293 -0.20599997 0.9323366
10  Factor6 Recurrence 0.95341599 -0.05923708 0.9534160</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor8 Recurrence 0.03752482 -2.245606 0.2935925</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Follow-up group</strong></p>
<pre class="r"><code># View factor values for samples
# Follow-up
plot_factor(mofaObject3V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
 ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-54-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-54-1">
Past versions of unnamed-chunk-54-1.png
</button>
</p>
<div id="fig-unnamed-chunk-54-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-54-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hopf-pl-ttp" class="section level1">
<h1>PM (Hopf) + PL + TTP</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Tumor
Tissue Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
metaPlasmaTab3V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab3V2G &lt;- dplyr::bind_rows(proTissueTab3V2G,
                                metaPlasmaTab3V2G,
                                lipPlasmaTab3V2G)
mofaObject3V2G &lt;- create_mofa_from_df(summTab3V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject3V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_3V2G-NTP&amp;-PP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject3V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-55-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-55-1">
Past versions of unnamed-chunk-55-1.png
</button>
</p>
<div id="fig-unnamed-chunk-55-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-55-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject3V2G)
# model_opts &lt;- get_default_model_options(mofaObject3V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject3V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject3V2G &lt;- prepare_mofa(mofaObject3V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject3V2G &lt;- run_mofa(mofaObject3V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_3V2G-NTP&amp;-PP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject3V2G, &#39;./data/mofa/Hopf_mofa_3V2G-NTP&amp;-PP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject3V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject3V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-56-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2        pVal   tStats    pValAdj
1 Factor8 Recurrence 0.009903568 2.882934 0.09903568</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject3V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject3V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor6 Recurrence 0.01485321 -2.693565 0.1485321
2 Factor2 Recurrence 0.03113526 -2.337952 0.1556763</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject3V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-59-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-59-1">
Past versions of unnamed-chunk-59-1.png
</button>
</p>
<div id="fig-unnamed-chunk-59-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-59-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject3V2G, factors = 6, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-60-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-60-1">
Past versions of unnamed-chunk-60-1.png
</button>
</p>
<div id="fig-unnamed-chunk-60-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-60-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_factor(mofaObject3V2G, factors = 2, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-60-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pm-hell-pl-pp-ntp" class="section level1">
<h1>PM (Hell) + PL + PP + NTP</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Normal</strong> <strong>Tissue Proteomics</strong><br />
Normal Tissue Proteomics might be able to form tumor microenvironment
and guide model training.<br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab4V2G &lt;- proTissueTab %&gt;%
  dplyr::filter(Condition == &#39;Normal&#39;) %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = paste(&#39;Normal&#39;, view),
                sample = stringr::str_replace(sample, &#39;_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

proPlasmaTab4V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                view = &#39;Plasma Proteomics&#39;)%&gt;%
  dplyr::select(-Condition)

metaPlasmaTab4V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab4V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proTissueTab4V2G,
                                proPlasmaTab4V2G,
                                metaPlasmaTab4V2G,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_4V2G-TTP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-61-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-61-1">
Past versions of unnamed-chunk-61-1.png
</button>
</p>
<div id="fig-unnamed-chunk-61-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-61-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_4V2G-TTP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/Hell_mofa_4V2G-TTP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-62-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal   tStats   pValAdj
1 Factor1 Recurrence 0.04191539 2.190239 0.3391734</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2        pVal   tStats    pValAdj
1 Factor9 Recurrence 0.005818565 -3.12748 0.05818565</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject4V2G, factors = 1, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-65-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-65-1">
Past versions of unnamed-chunk-65-1.png
</button>
</p>
<div id="fig-unnamed-chunk-65-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-65-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(paste0(result_path, &#39;Hell_4V2G_normal_baseline_factor2.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject4V2G, factors = 9, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-66-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-66-1">
Past versions of unnamed-chunk-66-1.png
</button>
</p>
<div id="fig-unnamed-chunk-66-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-66-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hopf-pl-pp-ntp" class="section level1">
<h1>PM (Hopf) + PL + PP + NTP</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Normal</strong> <strong>Tissue Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
metaPlasmaTab4V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proTissueTab4V2G,
                                proPlasmaTab4V2G,
                                metaPlasmaTab4V2G,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_4V2G-TTP_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-67-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-67-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_4V2G-TTP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/Hopf_mofa_4V2G-TTP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-68-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor9 Recurrence 0.02044541 -2.541841 0.2044541</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats  pValAdj
1 Factor3 Recurrence 0.04197005 -2.189584 0.174247
2 Factor1 Recurrence 0.04936806 -2.107410 0.174247</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject4V2G, factors = 9, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  # geom_label_repel(aes(label = sample), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65) +
  theme(strip.text = element_blank(), axis.ticks.x = element_blank(),
        axis.text.x = element_text(size = 16),
        axis.title.y = element_text(size = 16),
        legend.text = element_text(size = 14),
        legend.title = element_text(size = 12))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-71-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;Hopf_4V2G_normal_baseline_factor9.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<pre class="r"><code># Molecular signatures in data
data_heatmap(mofaObject4V2G, factor = 9, view = &#39;Plasma Proteomics&#39;,
             group = &#39;Baseline Group&#39;, features = 30,
             smp_anno = recurBaseAnno, smp_cluster = &#39;Recurrence&#39;,
             cluster_rows = T, cluster_cols = T,
             show_rownames = F, show_colnames = T,
             color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
             scale = &#39;row&#39;, main = &#39;Baseline Plasma Proteomics - Fac9&#39;)
data_heatmap(mofaObject4V2G, factor = 9, view = &#39;Lipidomics&#39;,
             group = &#39;Baseline Group&#39;, features = 30,
             smp_anno = recurBaseAnno, smp_cluster = &#39;Recurrence&#39;,
             cluster_rows = T, cluster_cols = T,
             show_rownames = T, show_colnames = T,
             color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
             scale = &#39;row&#39;, main = &#39;Baseline Plasma Lipidomics - Fac9&#39;)</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Follow-up
plot_factor(mofaObject4V2G, factors = 3, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-73-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-73-1">
Past versions of unnamed-chunk-73-1.png
</button>
</p>
<div id="fig-unnamed-chunk-73-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-73-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_factor(mofaObject4V2G, factors = 1, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-73-2.png" width="960" style="display: block; margin: auto;" /></p>
<!-- # PM (Hell) + PL + PP + DTP -->
<!-- **Targeted Plasma Metabolomics + Plasma Lipidomics + Plasma Proteomics + Differential** -->
<!-- **Tissue Proteomics**\ -->
<!-- Instead of using only Tumor or Normal Tissue Proteomics, differential Tissue Proteomics -->
<!-- (Tumor - Normal) was used.\ -->
<!-- <br/> -->
<!-- Visualize model setup -->
<!-- ```{r} -->
<!-- # Load trained MOFA object -->
<!-- mofaObject4V2G <- readRDS('./data/mofa/Hell_mofa_4V2G_Diff_10Fac.rds') -->
<!-- # Redefine model setup -->
<!-- # proTissueTab4V2G <- proTissueDiffTab %>% -->
<!-- #   dplyr::mutate(group = 'Baseline Group', -->
<!-- #                 view = 'Differential Tissue Proteomics', -->
<!-- #                 sample = paste0(sample, '_P_B')) -->
<!-- # -->
<!-- # proPlasmaTab4V2G <- proPlasmaTab %>% -->
<!-- #   dplyr::mutate(group = 'Group', -->
<!-- #                 group = dplyr::case_when( -->
<!-- #                   Condition == 'Baseline' ~ paste('Baseline', group), -->
<!-- #                   Condition != 'Baseline' ~ paste('Follow-up', group) -->
<!-- #                   ), -->
<!-- #                 view = 'Plasma Proteomics')%>% -->
<!-- #   dplyr::select(-Condition) -->
<!-- # -->
<!-- # metaPlasmaTab4V2G <- metaPlasmaTab_Hell %>% -->
<!-- #   dplyr::mutate(group = 'Group', -->
<!-- #                 group = dplyr::case_when( -->
<!-- #                   Condition == 'Baseline' ~ paste('Baseline', group), -->
<!-- #                   Condition != 'Baseline' ~ paste('Follow-up', group) -->
<!-- #                   )) %>% -->
<!-- #   dplyr::select(-Condition) -->
<!-- # -->
<!-- # lipPlasmaTab4V2G <- lipPlasmaTab %>% -->
<!-- #   dplyr::mutate(group = 'Group', -->
<!-- #                 group = dplyr::case_when( -->
<!-- #                   Condition == 'Baseline' ~ paste('Baseline', group), -->
<!-- #                   Condition != 'Baseline' ~ paste('Follow-up', group) -->
<!-- #                   ), -->
<!-- #                 sample = dplyr::case_when( -->
<!-- #                   Condition == 'Baseline' ~ paste0(Patient, '_P_B'), -->
<!-- #                   Condition != 'Baseline' ~ paste0(Patient, '_P_R') -->
<!-- #                   )) %>% -->
<!-- #   dplyr::select(-Condition) -->
<!-- # Create MOFA object -->
<!-- # Concatenate long data -->
<!-- # summTab4V2G <- dplyr::bind_rows(proTissueTab4V2G, -->
<!-- #                                 proPlasmaTab4V2G, -->
<!-- #                                 metaPlasmaTab4V2G, -->
<!-- #                                 lipPlasmaTab4V2G) -->
<!-- # mofaObject4V2G <- create_mofa_from_df(summTab4V2G, extract_metadata = T) -->
<!-- # Overview data in MOFA object -->
<!-- plot_data_overview(mofaObject4V2G) -->
<!-- # Train MOFA model -->
<!-- # data_opts <- get_default_data_options(mofaObject4V2G) -->
<!-- # model_opts <- get_default_model_options(mofaObject4V2G) -->
<!-- # model_opts$num_factors <- 10 -->
<!-- # train_opts <- get_default_training_options(mofaObject4V2G) -->
<!-- # -->
<!-- # mofaObject4V2G <- prepare_mofa(mofaObject4V2G, -->
<!-- #                                data_options = data_opts, -->
<!-- #                                model_options = model_opts, -->
<!-- #                                training_options = train_opts) -->
<!-- # -->
<!-- # mofaObject4V2G <- run_mofa(mofaObject4V2G, -->
<!-- #                            outfile = paste0(project_path, -->
<!-- #                                             'data/mofa_model/Hell_mofa_4V2G_Diff_10Fac.hdf5'), -->
<!-- #                            use_basilisk = F) -->
<!-- # saveRDS(mofaObject4V2G, paste0(project_path, 'data/mofa_model/Hell_mofa_4V2G_Diff_10Fac.rds')) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # Do sanity check for factor correlations -->
<!-- # plot_factor_cor(mofaObject4V2G) -->
<!-- # Perform variance dedecomposition analysis (coefficient of determination) -->
<!-- plot_variance_explained(mofaObject4V2G, max_r2 = 15) + -->
<!--   theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) -->
<!-- ``` -->
<!-- **Baseline group** -->
<!-- ```{r} -->
<!-- # Conduct association test between learnt factors and cancer recurrence -->
<!-- # Group: Baseline -->
<!-- facTab <- tibble::as_tibble(get_factors(mofaObject4V2G)[['Baseline Group']], -->
<!--                             rownames = 'sample') -->
<!-- metaTab <- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %>% -->
<!--   dplyr::filter(group == 'Baseline Group') %>% -->
<!--   dplyr::select(sample, Recurrence) -->
<!-- tBaseRes <- tTestIter(facTab, metaTab, cmn_col = 'sample') %>% -->
<!--   # dplyr::filter(pVal <= 0.05) %>% -->
<!--   dplyr::select(-c('Group1', 'Group2', 'Mu1', 'Mu2')) -->
<!-- tBaseRes -->
<!-- ``` -->
<!-- **Follow-up group** -->
<!-- ```{r} -->
<!-- # Group: Follow-up -->
<!-- facTab <- tibble::as_tibble(get_factors(mofaObject4V2G)[['Follow-up Group']], -->
<!--                             rownames = 'sample') -->
<!-- metaTab <- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %>% -->
<!--   dplyr::filter(group == 'Follow-up Group') %>% -->
<!--   dplyr::select(sample, Recurrence) -->
<!-- tFoloRes <- tTestIter(facTab, metaTab, cmn_col = 'sample') %>% -->
<!--   dplyr::filter(pVal <= 0.05) %>% -->
<!--   dplyr::select(-c('Group1', 'Group2', 'Mu1', 'Mu2')) -->
<!-- tFoloRes -->
<!-- ``` -->
<!-- <br/> -->
<!-- **Follow-up group** -->
<!-- ```{r} -->
<!-- # View factor values for samples -->
<!-- # Follow-up -->
<!-- plot_factor(mofaObject4V2G, factors = 1, color_by = 'Recurrence', -->
<!--             add_violin = T, dodge = T) + -->
<!--   ggpubr::stat_compare_means(method = 't.test', paired = F, -->
<!--                              method.args = list(var.equal = T)) -->
<!-- # cat('Effect size (Follow-up):\n', -->
<!-- #     effSize(mofaObject4V2G, gp = 'Follow-up Group', fac = 1)) -->
<!-- ``` -->
<!-- # PM (Hopf) + PL + PP + DTP -->
<!-- **Untargeted Plasma Metabolomics + Plasma Lipidomics + Plasma Proteomics + Differential** -->
<!-- **Tissue Proteomics**\ -->
<!-- <br/> -->
<!-- Visualize model setup -->
<!-- ```{r} -->
<!-- # Load trained MOFA object -->
<!-- mofaObject4V2G <- readRDS('./data/mofa/Hopf_mofa_4V2G_Diff_10Fac.rds') -->
<!-- # Redefine model setup -->
<!-- # metaPlasmaTab4V2G <- metaPlasmaTab_Hopf %>% -->
<!-- #   dplyr::mutate(group = 'Group', -->
<!-- #                 group = dplyr::case_when( -->
<!-- #                   Condition == 'Baseline' ~ paste('Baseline', group), -->
<!-- #                   Condition != 'Baseline' ~ paste('Follow-up', group) -->
<!-- #                   ), -->
<!-- #                 sample = dplyr::case_when( -->
<!-- #                   Condition == 'Baseline' ~ paste0(Patient, '_P_B'), -->
<!-- #                   Condition != 'Baseline' ~ paste0(Patient, '_P_R') -->
<!-- #                   )) %>% -->
<!-- #   dplyr::select(-Condition) -->
<!-- # Create MOFA object -->
<!-- # Concatenate long data -->
<!-- # summTab4V2G <- dplyr::bind_rows(proTissueTab4V2G, -->
<!-- #                                 proPlasmaTab4V2G, -->
<!-- #                                 metaPlasmaTab4V2G, -->
<!-- #                                 lipPlasmaTab4V2G) -->
<!-- # mofaObject4V2G <- create_mofa_from_df(summTab4V2G, extract_metadata = T) -->
<!-- # Overview data in MOFA object -->
<!-- plot_data_overview(mofaObject4V2G) -->
<!-- # Train MOFA model -->
<!-- # data_opts <- get_default_data_options(mofaObject4V2G) -->
<!-- # model_opts <- get_default_model_options(mofaObject4V2G) -->
<!-- # model_opts$num_factors <- 10 -->
<!-- # train_opts <- get_default_training_options(mofaObject4V2G) -->
<!-- # -->
<!-- # mofaObject4V2G <- prepare_mofa(mofaObject4V2G, -->
<!-- #                                data_options = data_opts, -->
<!-- #                                model_options = model_opts, -->
<!-- #                                training_options = train_opts) -->
<!-- # -->
<!-- # mofaObject4V2G <- run_mofa(mofaObject4V2G, -->
<!-- #                            outfile = paste0(project_path, -->
<!-- #                                             'data/mofa_model/Hopf_mofa_4V2G_Diff_10Fac.hdf5'), -->
<!-- #                            use_basilisk = F) -->
<!-- # saveRDS(mofaObject4V2G, paste0(project_path, 'data/mofa_model/Hopf_mofa_4V2G_Diff_10Fac.rds')) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # Do sanity check for factor correlations -->
<!-- # plot_factor_cor(mofaObject4V2G) -->
<!-- # Perform variance dedecomposition analysis (coefficient of determination) -->
<!-- plot_variance_explained(mofaObject4V2G, max_r2 = 15) + -->
<!--   theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) -->
<!-- ``` -->
<!-- **Baseline group** -->
<!-- ```{r} -->
<!-- # Conduct association test between learnt factors and cancer recurrence -->
<!-- # Group: Baseline -->
<!-- facTab <- tibble::as_tibble(get_factors(mofaObject4V2G)[['Baseline Group']], -->
<!--                             rownames = 'sample') -->
<!-- metaTab <- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %>% -->
<!--   dplyr::filter(group == 'Baseline Group') %>% -->
<!--   dplyr::select(sample, Recurrence) -->
<!-- tBaseRes <- tTestIter(facTab, metaTab, cmn_col = 'sample') %>% -->
<!--   dplyr::filter(pVal <= 0.05) %>% -->
<!--   dplyr::select(-c('Group1', 'Group2', 'Mu1', 'Mu2')) -->
<!-- tBaseRes -->
<!-- ``` -->
<!-- **Follow-up group** -->
<!-- ```{r} -->
<!-- # Group: Follow-up -->
<!-- facTab <- tibble::as_tibble(get_factors(mofaObject4V2G)[['Follow-up Group']], -->
<!--                             rownames = 'sample') -->
<!-- metaTab <- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %>% -->
<!--   dplyr::filter(group == 'Follow-up Group') %>% -->
<!--   dplyr::select(sample, Recurrence) -->
<!-- tFoloRes <- tTestIter(facTab, metaTab, cmn_col = 'sample') %>% -->
<!--   dplyr::filter(pVal <= 0.05) %>% -->
<!--   dplyr::select(-c('Group1', 'Group2', 'Mu1', 'Mu2')) -->
<!-- tFoloRes -->
<!-- ``` -->
<!-- <br/> -->
<!-- **Baseline group** -->
<!-- ```{r} -->
<!-- # View factor values for samples -->
<!-- # Baseline -->
<!-- plot_factor(mofaObject4V2G, factors = 8, color_by = 'Recurrence', -->
<!--             add_violin = T, dodge = T) + -->
<!--   ggpubr::stat_compare_means(method = 't.test', paired = F, -->
<!--                              method.args = list(var.equal = T), -->
<!--                              size = 6, hjust = 0.65) + -->
<!--   theme(strip.text = element_blank(), axis.ticks.x = element_blank(), -->
<!--         axis.text.x = element_text(size = 16), -->
<!--         axis.title.y = element_text(size = 16), -->
<!--         legend.text = element_text(size = 14), -->
<!--         legend.title = element_text(size = 12)) -->
<!-- # ggsave(paste0(result_path, 'Hopf_4V2G_diff_baseline_factor8.png'), device = 'png', dpi = 400) -->
<!-- # cat('Effect size (Baseline):\n', -->
<!-- #     effSize(mofaObject4V2G, gp = 'Baseline Group', fac = 8)) -->
<!-- ``` -->
<!-- **Follow-up group** -->
<!-- ```{r} -->
<!-- # Follow-up -->
<!-- plot_factor(mofaObject4V2G, factors = 1, color_by = 'Recurrence', -->
<!--             add_violin = T, dodge = T) + -->
<!--   ggpubr::stat_compare_means(method = 't.test', paired = F, -->
<!--                              method.args = list(var.equal = T)) -->
<!-- # cat('Effect size (Follow-up):\n', -->
<!-- #     effSize(mofaObject4V2G, gp = 'Follow-up Group', fac = 1)) -->
<!-- ``` -->
</div>
<div id="pm-hell-pl-pp-ttp-ntp-ttm-ntm" class="section level1">
<h1>PM (Hell) + PL + PP + TTP + NTP + TTM + NTM</h1>
<p><strong>Targeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Tumor Tissue</strong> <strong>Proteomics + Normal Tissue
Proteomics + Tumor Tissue Metabolomics + Normal Tissue
Metabolomics</strong><br />
All datasets were used to train a model.<br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proTissueTab7V2G &lt;- proTissueTab %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = dplyr::case_when(
                  Condition == &#39;Tumor&#39; ~ paste(&#39;Tumor&#39;, view),
                  Condition == &#39;Normal&#39; ~ paste(&#39;Normal&#39;, view)
                  ),
                sample = stringr::str_replace(sample, &#39;_TG|_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

metaTissueTab7V2G &lt;- metaTissueTab %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = dplyr::case_when(
                  Condition == &#39;Tumor&#39; ~ paste(&#39;Tumor&#39;, view),
                  Condition == &#39;Normal&#39; ~ paste(&#39;Normal&#39;, view)
                  ),
                sample = stringr::str_replace(sample, &#39;_TG|_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

proPlasmaTab7V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                view = &#39;Plasma Proteomics&#39;)%&gt;%
  dplyr::select(-Condition)

metaPlasmaTab7V2G &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  )) %&gt;%
  dplyr::select(-Condition)

lipPlasmaTab7V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                  )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab7V2G &lt;- dplyr::bind_rows(proTissueTab7V2G,
                                metaTissueTab7V2G,
                                proPlasmaTab7V2G,
                                metaPlasmaTab7V2G,
                                lipPlasmaTab7V2G)
mofaObject7V2G &lt;- create_mofa_from_df(summTab7V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject7V2G &lt;- readRDS(&#39;./data/mofa/Hell_mofa_7V2G_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject7V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-74-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-74-1">
Past versions of unnamed-chunk-74-1.png
</button>
</p>
<div id="fig-unnamed-chunk-74-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-74-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject7V2G)
# model_opts &lt;- get_default_model_options(mofaObject7V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject7V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject7V2G &lt;- prepare_mofa(mofaObject7V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject7V2G &lt;- run_mofa(mofaObject7V2G,
#                            outfile = &#39;./data/mofa/Hell_mofa_7V2G_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject7V2G, &#39;./data/mofa/Hell_mofa_7V2G_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject7V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject7V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-75-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject7V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject7V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2      pVal      tStats   pValAdj
1   Factor9 Recurrence 0.0891263  1.79705584 0.7656674
2   Factor1 Recurrence 0.1531335  1.49155658 0.7656674
3   Factor5 Recurrence 0.3805455 -0.89895452 0.8735248
4  Factor10 Recurrence 0.5152961 -0.66369662 0.8735248
5   Factor6 Recurrence 0.5584395  0.59623980 0.8735248
6   Factor8 Recurrence 0.6757446 -0.42518073 0.8735248
7   Factor2 Recurrence 0.6942412 -0.39947800 0.8735248
8   Factor3 Recurrence 0.7001890  0.39127218 0.8735248
9   Factor7 Recurrence 0.7861723  0.27536454 0.8735248
10  Factor4 Recurrence 0.9716374  0.03605233 0.9716374</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject7V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject7V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2        pVal   tStats    pValAdj
1 Factor9 Recurrence 0.007408144 3.016888 0.07408144</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Follow-up group</strong></p>
<pre class="r"><code># View factor values for samples
# Follow-up
plot_factor(mofaObject7V2G, factors = 9, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-78-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-78-1">
Past versions of unnamed-chunk-78-1.png
</button>
</p>
<div id="fig-unnamed-chunk-78-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-78-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pm-hopf-pl-pp-ttp-ntp-ttm-ntm" class="section level1">
<h1>PM (Hopf) + PL + PP + TTP + NTP + TTM + NTM</h1>
<p><strong>Untargeted Plasma Metabolomics + Plasma Lipidomics + Plasma
Proteomics + Tumor Tissue</strong> <strong>Proteomics + Normal Tissue
Proteomics + Tumor Tissue Metabolomics + Normal Tissue
Metabolomics</strong><br />
</p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
metaPlasmaTab7V2G &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(group = &#39;Group&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste(&#39;Baseline&#39;, group),
                  Condition != &#39;Baseline&#39; ~ paste(&#39;Follow-up&#39;, group)
                  ),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)
                )) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab7V2G &lt;- dplyr::bind_rows(proTissueTab7V2G,
                                metaTissueTab7V2G,
                                proPlasmaTab7V2G,
                                metaPlasmaTab7V2G,
                                lipPlasmaTab7V2G)
mofaObject7V2G &lt;- create_mofa_from_df(summTab7V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject7V2G &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_7V2G_10Fac.rds&#39;)

# Overview data in MOFA object
plot_data_overview(mofaObject7V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-79-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-79-1">
Past versions of unnamed-chunk-79-1.png
</button>
</p>
<div id="fig-unnamed-chunk-79-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-79-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject7V2G)
# model_opts &lt;- get_default_model_options(mofaObject7V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject7V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject7V2G &lt;- prepare_mofa(mofaObject7V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject7V2G &lt;- run_mofa(mofaObject7V2G,
#                            outfile = &#39;./data/mofa/Hopf_mofa_7V2G_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject7V2G, &#39;./data/mofa/Hopf_mofa_7V2G_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# plot_factor_cor(mofaObject7V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
plot_variance_explained(mofaObject7V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-80-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject7V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject7V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal   tStats   pValAdj
1 Factor8 Recurrence 0.04899802 2.111244 0.4899802</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject7V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject7V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2         pVal    tStats     pValAdj
1 Factor8 Recurrence 0.0008467691  3.996305 0.008467691
2 Factor5 Recurrence 0.0140075266 -2.721149 0.070037633</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
plot_factor(mofaObject7V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-83-1.png" width="960" style="display: block; margin: auto;" />
<strong>Follow-up group</strong></p>
<pre class="r"><code># View factor values for samples
# Follow-up
plot_factor(mofaObject7V2G, factors = 8, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-84-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-84-1">
Past versions of unnamed-chunk-84-1.png
</button>
</p>
<div id="fig-unnamed-chunk-84-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-84-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_factor(mofaObject7V2G, factors = 5, color_by = &#39;Recurrence&#39;,
            add_violin = T, dodge = T) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-84-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 2. Plasma Proteomics holds Recurrence-related information. Besides, variation in
# Plasma Proteomics and Plasma Lipidomics is always explained by Recurrence-related
# factors.\
# 3. There exists this outlier Patient &#39;H03Y07RT&#39; in most of trained models, except
# a model trained on PM (Hopf) + PL + PP + TTP. Consequently, this model outperforms
# the others in view of p-values. If this outlier is excluded or we look results
# concerning classification accuracy, many models can be pretty comparable, even better.
# It might be that (1) Patient &#39;H03Y07RT&#39; is false positive, i.e., non-recurrence,
# (2) technical or biological noise is in Plasma Proteomics or Plasma Lipidomics
# datasets since they are common point across trained models, or (3) model trained
# on PM (Hopf) + PL + PP + TTP can really somehow classify Patient &#39;H03Y07RT&#39; accurately
# (Factor 10 and 9!).</code></pre>
<p>Some findings:<br />
1. Models trained on omics data combinations containing Untargeted
Plasma Metabolomics perform better than those using Targeted Plasma
Metabolomics in terms of predicting patient cancer recurrences.
Evaluation includes learning Recurrence-related factors and separating
recurrence and non-recurrence patient groups in Baseline group. This is
sort of expected since Untargeted Plasma Metabolomics captures more
features, which potentially provides more information.<br />
2. There exists this outlier Patient ‘H03Y07RT’ in most of trained
models. If we look results in view of classification accuracy, many
models can perform quite well. In next section, we will exclude Patient
‘H03Y07RT’ and recalculate t-statistics.</p>
</div>
<div id="outlier-h03y07rt-removal" class="section level1">
<h1>Outlier ‘H03Y07RT’ removal</h1>
<p>We observed that Sample ‘H03Y07RT_P_B’ is usually an extreme outlier
in recurrence patient group in most of trained models, otherwise without
this sample, many models can be quite comparable. Plus, current sample
size is still small, which makes t-test very sensitive to outliers.
Therefore, we removed this outlier when performing t-tests for model
selection.</p>
<pre class="r"><code># Load trained MOFA object
# 3 views 2 groups: Plasma Proteomics, Metabolomics, Lipidomics
# Plasma Metabolomics from AG Hell
mofa3V2G_1 &lt;- readRDS(&#39;./data/mofa/Hell_mofa_3V2G_10Fac.rds&#39;)
# Plasma Metabolomics from AG Hopf
mofa3V2G_2 &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_3V2G_10Fac.rds&#39;)

# 5V2G: Plasma Proteomics, Metabolomics, Lipidomics, Tissue Proteomics
mofa5V2G_1 &lt;- readRDS(&#39;./data/mofa/Hell_mofa_5V2G_10Fac.rds&#39;)
mofa5V2G_2 &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_5V2G_10Fac.rds&#39;)

# 4V2G: Plasma Proteomics, Metabolomics, Lipidomics, Tissue Proteomics (Tumor)
mofa4V2G_Tumor_1 &lt;- readRDS(&#39;./data/mofa/Hell_mofa_4V2G-NTP_10Fac.rds&#39;)
mofa4V2G_Tumor_2 &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_4V2G-NTP_10Fac.rds&#39;)

# 4V2G: Plasma Proteomics, Metabolomics, Lipidomics, Tissue Proteomics (Normal)
mofa4V2G_Normal_1 &lt;- readRDS(&#39;./data/mofa/Hell_mofa_4V2G-TTP_10Fac.rds&#39;)
mofa4V2G_Normal_2 &lt;- readRDS(&#39;./data/mofa/Hopf_mofa_4V2G-TTP_10Fac.rds&#39;)</code></pre>
<p>Display p-values from t-test results of Recurrence-related factors,
where Patient ‘H03Y07RT’ is excluded</p>
<pre class="r"><code># Prepare tables containing promising and highly correlated factors learned from
# different models
# 3V2G: PP, PM, PL
sigFac3V2G_1 &lt;- MOFA2::get_factors(mofa3V2G_1)[[&#39;Baseline Plasma&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor1) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hell_3V_Factor1 = Factor1)
sigFac3V2G_2 &lt;- MOFA2::get_factors(mofa3V2G_2)[[&#39;Baseline Plasma&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor8) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hopf_3V_Factor8 = Factor8)

# 5V2G: PP, PM, PL, TP (Both Tumor and Normal)
# sigFac5V2G_1 &lt;- MOFA2::get_factors(mofa5V2G_1)[[&#39;Baseline Group&#39;]] %&gt;%
#   tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
#   dplyr::select(sample, Factor8) %&gt;%
#   dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
#   dplyr::rename(Hell_5V_Factor8 = Factor8)
sigFac5V2G_2 &lt;- MOFA2::get_factors(mofa5V2G_2)[[&#39;Baseline Group&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor8) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hopf_5V_Factor8 = Factor8)

# 4V2G: PP, PM, PL, TP (Only Tumor)
sigFac4V2G_Tumor_1 &lt;- MOFA2::get_factors(mofa4V2G_Tumor_1)[[&#39;Baseline Group&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor1) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hell_4V_Tumor_Factor1 = Factor1)
sigFac4V2G_Tumor_2 &lt;- MOFA2::get_factors(mofa4V2G_Tumor_2)[[&#39;Baseline Group&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor9) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hopf_4V_Tumor_Factor9 = Factor9)

# 4V2G: PP, PM, PL, TP (Only Normal)
sigFac4V2G_Normal_1 &lt;- MOFA2::get_factors(mofa4V2G_Normal_1)[[&#39;Baseline Group&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor1) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hell_4V_Normal_Factor1 = Factor1)
sigFac4V2G_Normal_2 &lt;- MOFA2::get_factors(mofa4V2G_Normal_2)[[&#39;Baseline Group&#39;]] %&gt;%
  tibble::as_tibble(rownames = &#39;sample&#39;) %&gt;%
  dplyr::select(sample, Factor9) %&gt;%
  dplyr::filter(sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::rename(Hopf_4V_Normal_Factor9 = Factor9)

# Combine all information
tabList &lt;- list(sigFac3V2G_1, sigFac3V2G_2, sigFac5V2G_2,
                sigFac4V2G_Tumor_1, sigFac4V2G_Tumor_2,
                sigFac4V2G_Normal_1, sigFac4V2G_Normal_2)
combinedTab &lt;- dplyr::left_join(tabList[[1]], tabList[[2]], by = &#39;sample&#39;)
for (i in 3:length(tabList)) {
  combinedTab &lt;- dplyr::left_join(combinedTab, tabList[[i]], by = &#39;sample&#39;)
}

# Prepare table containing sample metadata
metaTab &lt;- MOFA2::samples_metadata(mofa3V2G_1) %&gt;%
  dplyr::filter(group == &#39;Baseline Plasma&#39;, sample != &#39;H03Y07RT_P_B&#39;) %&gt;%
  dplyr::select(sample, Recurrence)

# Systematically perform t-tests
tRes &lt;- tTestIter(combinedTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::select(Var1, pVal) %&gt;%
  dplyr::rename(Factor = Var1) %&gt;%
  dplyr::mutate(Factor = factor(Factor, levels = Factor))

# Visualize t-test results
ggplot(tRes, aes(x=Factor, y=pVal)) +
  geom_bar(stat=&#39;identity&#39;) +
  geom_text(aes(label=round(pVal, 5)), vjust = -1, size=3.5) +
  labs(x = &#39;Recurrence-related factor&#39;, y = &#39;P-value&#39;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-86-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-86-1">
Past versions of unnamed-chunk-86-1.png
</button>
</p>
<div id="fig-unnamed-chunk-86-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-86-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Hopf_3V_Factor8 -&gt; PM (Hopf) + PL + PP<br />
Hopf_4V_Tumor_Factor9 -&gt; PM (Hopf) + PL + PP + TTP<br />
Hopf_5V_Factor8 -&gt; PM (Hopf) + PL + PP + TTP + NTP<br />
Hell_4V_Tumor_Factor1 -&gt; PM (Hell) + PL + PP + TTP<br />
Hell_4V_Normal_Factor1 -&gt; PM (Hell) + PL + PP + NTP<br />
Hell_3V_Factor1 -&gt; PM (Hell) + PL + PP<br />
Hopf_4V_Normal_Factor9 -&gt; PM (Hopf) + PL + PP + NTP<br />
<br/></p>
<p>Visualize top best Recurrence-related factors</p>
<pre class="r"><code>tab4Boxplot &lt;- dplyr::left_join(combinedTab, metaTab, by = &#39;sample&#39;)

# 3V2G: PP, PM (Hopf), PL
ggplot(tab4Boxplot, aes(x=Recurrence, y=Hopf_3V_Factor8,
                        col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;PM (Hopf) + PL + PP&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-87-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-87-1">
Past versions of unnamed-chunk-87-1.png
</button>
</p>
<div id="fig-unnamed-chunk-87-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-87-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 4V2G: PP, PM (Hopf), PL, TP (Only Tumor)
ggplot(tab4Boxplot, aes(x=Recurrence, y=Hopf_4V_Tumor_Factor9,
                        col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;PM (Hopf) + PL + PP + TTP&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-87-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 5V2G: PP, PM (Hopf), PL, TP (Both Tumor and Normal)
ggplot(tab4Boxplot, aes(x=Recurrence, y=Hopf_5V_Factor8,
                        col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;PM (Hopf) + PL + PP + TTP + NTP&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-87-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 4V2G: PP, PM (Hell), PL, TP (Only Tumor)
ggplot(tab4Boxplot, aes(x=Recurrence, y=Hell_4V_Tumor_Factor1,
                        col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;PM (Hell) + PL + PP + TTP&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-87-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 3V2G: PP, PM (Hell), PL
ggplot(tab4Boxplot, aes(x=Recurrence, y=Hopf_4V_Normal_Factor9,
                        col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;PM (Hell) + PL + PP&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-87-5.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="using-both-pm-hell-hopf" class="section level1">
<h1>Using both PM (Hell &amp; Hopf)</h1>
<p>We found that correlations between Targeted and Untargeted metabolite
features are overall not that high (all below 0.9). It might be that
these two datasets provide different information that can be
complementary to each other as Targeted method captured a lot of e.g.,
DG (Diglyceride) or TG (Triglyceride) derivatives. To this end, we
trained models with omics data including both Targeted and Untargeted
Plasma Metabolomics. The idea remains same that, firstly, only and all
Plasma samples are used and then Tissue Proteomics is added to see
changes in model performance.<br />
(It might also be that Untargeted method is still immature, which
introduces technical or biological noise, leading to inaccurate
data?)</p>
<div id="pm-pl-pp" class="section level2">
<h2>PM + PL + PP</h2>
<p><strong>Targeted and Untargeted Plasma Metabolomics + Plasma
Lipidomics + Plasma Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proPlasmaTab4V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Proteomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab4V2G_Hell &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hell)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab4V2G_Hopf &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hopf)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

lipPlasmaTab4V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Lipidomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

# Create MOFA object
# Concatenate long data
summTab4V2G &lt;- dplyr::bind_rows(proPlasmaTab4V2G,
                                metaPlasmaTab4V2G_Hell,
                                metaPlasmaTab4V2G_Hopf,
                                lipPlasmaTab4V2G)
mofaObject4V2G &lt;- MOFA2::create_mofa_from_df(summTab4V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject4V2G &lt;- readRDS(&#39;./data/mofa/mofa_4V2G_Both_10Fac.rds&#39;)

# Overview data in MOFA object
MOFA2::plot_data_overview(mofaObject4V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-88-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject4V2G)
# model_opts &lt;- get_default_model_options(mofaObject4V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject4V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject4V2G &lt;- prepare_mofa(mofaObject4V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject4V2G &lt;- run_mofa(mofaObject4V2G,
#                            outfile = &#39;./data/mofa/mofa_4V2G_Both_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject4V2G, &#39;./data/mofa/mofa_4V2G_Both_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# MOFA2::plot_factor_cor(mofaObject4V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject4V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-89-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(Condition == &#39;Baseline&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2       pVal      tStats   pValAdj
1   Factor1 Recurrence 0.05076938  2.09312309 0.4479855
2   Factor6 Recurrence 0.08959709  1.79419337 0.4479855
3   Factor9 Recurrence 0.21527791 -1.28442735 0.6860026
4  Factor10 Recurrence 0.27440105 -1.12734462 0.6860026
5   Factor4 Recurrence 0.58050736 -0.56281002 0.9549761
6   Factor2 Recurrence 0.82739158 -0.22124566 0.9549761
7   Factor5 Recurrence 0.84348271  0.20031157 0.9549761
8   Factor8 Recurrence 0.86871557 -0.16766582 0.9549761
9   Factor3 Recurrence 0.90029579  0.12706739 0.9549761
10  Factor7 Recurrence 0.95497608 -0.05725091 0.9549761</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject4V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject4V2G)) %&gt;%
  dplyr::filter(Condition != &#39;Baseline&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2      pVal   tStats   pValAdj
1 Factor6 Recurrence 0.0304821 2.348362 0.2636503</code></pre>
</div>
<div id="pm-pl-pp-ttp-ntp" class="section level2">
<h2>PM + PL + PP + TTP + NTP</h2>
<p><strong>Targeted and Untargeted Plasma Metabolomics + Plasma
Lipidomics + Plasma Proteomics</strong> <strong>+ Tumor Tissue
Proteomics + Normal Tissue Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proPlasmaTab6V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Proteomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab6V2G_Hell &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hell)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab6V2G_Hopf &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hopf)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

lipPlasmaTab6V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Lipidomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

proTissueTab6V2G &lt;- proTissueTab %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = dplyr::case_when(
                  Condition == &#39;Tumor&#39; ~ paste(&#39;Tumor Tissue&#39;, view),
                  Condition == &#39;Normal&#39; ~ paste(&#39;Normal Tissue&#39;, view)),
                sample = stringr::str_replace(sample, &#39;_TG|_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab6V2G &lt;- dplyr::bind_rows(proPlasmaTab6V2G,
                                metaPlasmaTab6V2G_Hell,
                                metaPlasmaTab6V2G_Hopf,
                                lipPlasmaTab6V2G,
                                proTissueTab6V2G)
mofaObject6V2G &lt;- MOFA2::create_mofa_from_df(summTab6V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject6V2G &lt;- readRDS(&#39;./data/mofa/mofa_6V2G_Both_10Fac.rds&#39;)

# Overview data in MOFA object
MOFA2::plot_data_overview(mofaObject6V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-92-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-92-1">
Past versions of unnamed-chunk-92-1.png
</button>
</p>
<div id="fig-unnamed-chunk-92-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-92-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject6V2G)
# model_opts &lt;- get_default_model_options(mofaObject6V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject6V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject6V2G &lt;- prepare_mofa(mofaObject6V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject6V2G &lt;- run_mofa(mofaObject6V2G,
#                            outfile = &#39;./data/mofa/mofa_6V2G_Both_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject6V2G, &#39;./data/mofa/mofa_6V2G_Both_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# MOFA2::plot_factor_cor(mofaObject6V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject6V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-93-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Factor 7,8,9 are highly correlated with all features in specific individual data
# modalities, which indicates these factors very probably capture technical noise,
# since, theoretically, biological variation should be co-observed in different data
# modalities.</code></pre>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject6V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject6V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2       pVal     tStats   pValAdj
1   Factor7 Recurrence 0.08288583  1.8363050 0.6181635
2   Factor1 Recurrence 0.12363270  1.6153226 0.6181635
3  Factor10 Recurrence 0.34551397 -0.9687371 0.8930324
4   Factor3 Recurrence 0.41225312 -0.8394034 0.8930324
5   Factor9 Recurrence 0.54171634 -0.6220385 0.8930324
6   Factor5 Recurrence 0.55552263  0.6007096 0.8930324
7   Factor6 Recurrence 0.64718722  0.4654558 0.8930324
8   Factor8 Recurrence 0.78291116  0.2796805 0.8930324
9   Factor4 Recurrence 0.88970737  0.1406514 0.8930324
10  Factor2 Recurrence 0.89303243 -0.1363828 0.8930324</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject6V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject6V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2        pVal   tStats    pValAdj
1 Factor7 Recurrence 0.006512511 3.075975 0.06512511</code></pre>
</div>
<div id="pm-pl-pp-ttp" class="section level2">
<h2>PM + PL + PP + TTP</h2>
<p><strong>Targeted and Untargeted Plasma Metabolomics + Plasma
Lipidomics + Plasma Proteomics</strong> <strong>+ Tumor Tissue
Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proPlasmaTab5V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Proteomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab5V2G_Hell &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hell)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab5V2G_Hopf &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hopf)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

lipPlasmaTab5V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Lipidomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

proTissueTab5V2G &lt;- proTissueTab %&gt;%
  dplyr::filter(Condition == &#39;Tumor&#39;) %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = paste(&#39;Tumor Tissue&#39;, view),
                sample = stringr::str_replace(sample, &#39;_TG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab5V2G &lt;- dplyr::bind_rows(proPlasmaTab5V2G,
                                metaPlasmaTab5V2G_Hell,
                                metaPlasmaTab5V2G_Hopf,
                                lipPlasmaTab5V2G,
                                proTissueTab5V2G)
mofaObject5V2G &lt;- MOFA2::create_mofa_from_df(summTab5V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject5V2G &lt;- readRDS(&#39;./data/mofa/mofa_5V2G_Both-NTP_10Fac.rds&#39;)

# Overview data in MOFA object
MOFA2::plot_data_overview(mofaObject5V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-96-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-96-1">
Past versions of unnamed-chunk-96-1.png
</button>
</p>
<div id="fig-unnamed-chunk-96-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-96-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject5V2G)
# model_opts &lt;- get_default_model_options(mofaObject5V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject5V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# 
# mofaObject5V2G &lt;- prepare_mofa(mofaObject5V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject5V2G &lt;- run_mofa(mofaObject5V2G,
#                            outfile = &#39;./data/mofa/mofa_5V2G_Both-NTP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject5V2G, &#39;./data/mofa/mofa_5V2G_Both-NTP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# MOFA2::plot_factor_cor(mofaObject5V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject5V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-97-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-97-1">
Past versions of unnamed-chunk-97-1.png
</button>
</p>
<div id="fig-unnamed-chunk-97-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-97-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  # dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>       Var1       Var2      pVal      tStats   pValAdj
1   Factor6 Recurrence 0.0658456  1.95855159 0.6584560
2  Factor10 Recurrence 0.3052146 -1.05537399 0.7837472
3   Factor1 Recurrence 0.3276006  1.00631626 0.7837472
4   Factor7 Recurrence 0.3619812 -0.93536326 0.7837472
5   Factor8 Recurrence 0.4616554 -0.75219876 0.7837472
6   Factor3 Recurrence 0.4702483  0.73762748 0.7837472
7   Factor5 Recurrence 0.8345587  0.21190934 0.9351945
8   Factor4 Recurrence 0.8546108  0.18588870 0.9351945
9   Factor2 Recurrence 0.8731385  0.16196390 0.9351945
10  Factor9 Recurrence 0.9351945 -0.08245546 0.9351945</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal   tStats   pValAdj
1 Factor6 Recurrence 0.01338921 2.742344 0.1338921
2 Factor1 Recurrence 0.03766780 2.243711 0.1883390</code></pre>
</div>
<div id="pm-pl-pp-ntp" class="section level2">
<h2>PM + PL + PP + NTP</h2>
<p><strong>Targeted and Untargeted Plasma Metabolomics + Plasma
Lipidomics + Plasma Proteomics</strong> <strong>+ Normal Tissue
Proteomics</strong><br />
<br/></p>
<p>Visualize model setup</p>
<pre class="r"><code># Redefine model setup
proPlasmaTab5V2G &lt;- proPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Proteomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab5V2G_Hell &lt;- metaPlasmaTab_Hell %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hell)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;))

metaPlasmaTab5V2G_Hopf &lt;- metaPlasmaTab_Hopf %&gt;%
  dplyr::mutate(view = &#39;Plasma Metabolomics (AG Hopf)&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

lipPlasmaTab5V2G &lt;- lipPlasmaTab %&gt;%
  dplyr::mutate(view = &#39;Plasma Lipidomics&#39;,
                group = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ &#39;Baseline Group&#39;,
                  Condition != &#39;Baseline&#39; ~ &#39;Follow-up Group&#39;),
                sample = dplyr::case_when(
                  Condition == &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_B&#39;),
                  Condition != &#39;Baseline&#39; ~ paste0(Patient, &#39;_P_R&#39;)))

proTissueTab5V2G &lt;- proTissueTab %&gt;%
  dplyr::filter(Condition == &#39;Normal&#39;) %&gt;%
  dplyr::mutate(group = &#39;Baseline Group&#39;,
                view = paste(&#39;Normal Tissue&#39;, view),
                sample = stringr::str_replace(sample, &#39;_NG&#39;, &#39;_P_B&#39;)) %&gt;%
  dplyr::select(-Condition)

# Create MOFA object
# Concatenate long data
summTab5V2G &lt;- dplyr::bind_rows(proPlasmaTab5V2G,
                                metaPlasmaTab5V2G_Hell,
                                metaPlasmaTab5V2G_Hopf,
                                lipPlasmaTab5V2G,
                                proTissueTab5V2G)
mofaObject5V2G &lt;- MOFA2::create_mofa_from_df(summTab5V2G, extract_metadata = T)


# Load trained MOFA object
mofaObject5V2G &lt;- readRDS(&#39;./data/mofa/mofa_5V2G_Both-TTP_10Fac.rds&#39;)

# Overview data in MOFA object
MOFA2::plot_data_overview(mofaObject5V2G)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-100-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-100-1">
Past versions of unnamed-chunk-100-1.png
</button>
</p>
<div id="fig-unnamed-chunk-100-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-100-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Train MOFA model
# data_opts &lt;- get_default_data_options(mofaObject5V2G)
# model_opts &lt;- get_default_model_options(mofaObject5V2G)
# model_opts$num_factors &lt;- 10
# train_opts &lt;- get_default_training_options(mofaObject5V2G)
# train_opts$convergence_mode &lt;- &#39;slow&#39;
# train_opts$maxiter &lt;- 1200
# 
# mofaObject5V2G &lt;- prepare_mofa(mofaObject5V2G,
#                                data_options = data_opts,
#                                model_options = model_opts,
#                                training_options = train_opts)
# 
# mofaObject5V2G &lt;- run_mofa(mofaObject5V2G,
#                            outfile = &#39;./data/mofa/mofa_5V2G_Both-TTP_10Fac.hdf5&#39;,
#                            use_basilisk = F)
# saveRDS(mofaObject5V2G, &#39;./data/mofa/mofa_5V2G_Both-TTP_10Fac.rds&#39;)</code></pre>
<p>Quantify amount of variance explained by each factor in each data
modality</p>
<pre class="r"><code># Do sanity check for factor correlations
# MOFA2::plot_factor_cor(mofaObject5V2G)

# Perform variance dedecomposition analysis (coefficient of determination)
MOFA2::plot_variance_explained(mofaObject5V2G, max_r2 = 15) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-101-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-101-1">
Past versions of unnamed-chunk-101-1.png
</button>
</p>
<div id="fig-unnamed-chunk-101-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/0bbd8ebdea3501fa5b70010b29acf61a4c41d27b/docs/figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-101-1.png" target="_blank">0bbd8eb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between learned factors (Var1) and
cancer recurrence (Var2)<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># Conduct association test between learnt factors and cancer recurrence
# Group: Baseline
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Baseline Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Baseline Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tBaseRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tBaseRes</code></pre>
<pre><code>     Var1       Var2       pVal    tStats   pValAdj
1 Factor2 Recurrence 0.03403724  2.294023 0.2455105
2 Factor7 Recurrence 0.04910211 -2.110163 0.2455105</code></pre>
<p><strong>Follow-up group</strong></p>
<pre class="r"><code># Group: Follow-up
facTab &lt;- tibble::as_tibble(get_factors(mofaObject5V2G)[[&#39;Follow-up Group&#39;]],
                            rownames = &#39;sample&#39;)
metaTab &lt;- tibble::as_tibble(samples_metadata(mofaObject5V2G)) %&gt;%
  dplyr::filter(group == &#39;Follow-up Group&#39;) %&gt;%
  dplyr::select(sample, Recurrence)
tFoloRes &lt;- tTestIter(facTab, metaTab, cmn_col = &#39;sample&#39;) %&gt;%
  dplyr::filter(pVal &lt;= 0.05) %&gt;%
  dplyr::select(-c(&#39;Group1&#39;, &#39;Group2&#39;, &#39;Mu1&#39;, &#39;Mu2&#39;))
tFoloRes</code></pre>
<pre><code>     Var1       Var2       pVal  tStats   pValAdj
1 Factor5 Recurrence 0.01188985 2.79791 0.1188985
2 Factor1 Recurrence 0.04777844 2.12407 0.2388922</code></pre>
<p><br/></p>
<p>Visualize Recurrence-related factors<br />
<strong>Baseline group</strong></p>
<pre class="r"><code># View factor values for samples
# Baseline
MOFA2::plot_factor(mofaObject5V2G, factors = 2, color_by = &#39;Recurrence&#39;,
                   add_violin = T, dodge = T) +
  # geom_label_repel(aes(label=sample), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-104-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>MOFA2::plot_factor(mofaObject5V2G, factors = 7, color_by = &#39;Recurrence&#39;,
                   add_violin = T, dodge = T) +
  # geom_label_repel(aes(label=sample), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             size = 6, hjust = 0.65)</code></pre>
<p><img src="figure/mofa_varied_omics_combinations.Rmd/unnamed-chunk-104-2.png" width="960" style="display: block; margin: auto;" />
=&gt; Most models trained on omics data combinations containing both
Targeted and Untargeted Plasma Metabolomics cannot even capture good
Recurrence-related factors. Recurrence-related factors captured by model
trained on PM + PL + PP + NTP do not perform better either.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.2.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.1                
 [5] purrr_1.0.1                 readr_2.1.4                
 [7] tidyr_1.3.0                 tibble_3.2.1               
 [9] tidyverse_2.0.0             SummarizedExperiment_1.26.1
[11] Biobase_2.56.0              GenomicRanges_1.48.0       
[13] GenomeInfoDb_1.32.4         IRanges_2.30.1             
[15] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[17] MatrixGenerics_1.8.1        matrixStats_0.63.0         
[19] ggrepel_0.9.3               ggplot2_3.4.1              
[21] reticulate_1.28             MOFA2_1.6.0                
[23] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.6.1               filelock_1.0.2        
 [4] RColorBrewer_1.1-3     httr_1.4.5             rprojroot_2.0.3       
 [7] backports_1.4.1        tools_4.2.1            bslib_0.4.2           
[10] utf8_1.2.3             R6_2.5.1               HDF5Array_1.24.2      
[13] uwot_0.1.14            DBI_1.1.3              colorspace_2.1-0      
[16] rhdf5filters_1.8.0     withr_2.5.0            tidyselect_1.2.0      
[19] processx_3.8.0         compiler_4.2.1         git2r_0.31.0          
[22] cli_3.6.1              basilisk.utils_1.8.0   DelayedArray_0.22.0   
[25] labeling_0.4.2         sass_0.4.5             scales_1.2.1          
[28] callr_3.7.3            digest_0.6.31          rmarkdown_2.21        
[31] basilisk_1.8.1         XVector_0.36.0         pkgconfig_2.0.3       
[34] htmltools_0.5.5        fastmap_1.1.1          highr_0.10            
[37] rlang_1.1.0            rstudioapi_0.14        farver_2.1.1          
[40] jquerylib_0.1.4        generics_0.1.3         jsonlite_1.8.4        
[43] car_3.1-1              RCurl_1.98-1.10        magrittr_2.0.3        
[46] GenomeInfoDbData_1.2.8 Matrix_1.5-3           Rcpp_1.0.10           
[49] munsell_0.5.0          Rhdf5lib_1.18.2        fansi_1.0.4           
[52] abind_1.4-5            lifecycle_1.0.3        stringi_1.7.12        
[55] whisker_0.4.1          yaml_2.3.7             carData_3.0-5         
[58] zlibbioc_1.42.0        rhdf5_2.40.0           Rtsne_0.16            
[61] plyr_1.8.8             grid_4.2.1             parallel_4.2.1        
[64] promises_1.2.0.1       dir.expiry_1.4.0       lattice_0.20-45       
[67] cowplot_1.1.1          hms_1.1.3              knitr_1.42            
[70] ps_1.7.3               pillar_1.9.0           ggpubr_0.6.0          
[73] ggsignif_0.6.4         reshape2_1.4.4         glue_1.6.2            
[76] evaluate_0.20          getPass_0.2-2          tzdb_0.3.0            
[79] png_0.1-8              vctrs_0.6.1            httpuv_1.6.9          
[82] gtable_0.3.3           cachem_1.0.7           xfun_0.38             
[85] broom_1.0.4            rstatix_0.7.2          later_1.3.0           
[88] pheatmap_1.0.12        corrplot_0.92          timechange_0.2.0      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
