<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Single-omics analysis: Method development lung cancer patient cohort</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Single-omics analysis: Method development
lung cancer patient cohort</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree717cb2321703f047d6659449429f744bc5e74644targetblank717cb23a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/717cb2321703f047d6659449429f744bc5e74644" target="_blank">717cb23</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree717cb2321703f047d6659449429f744bc5e74644targetblank717cb23a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/717cb2321703f047d6659449429f744bc5e74644" target="_blank">717cb23</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/feature_selection.Rmd
    Untracked:  analysis/mofa_downstream_analysis.Rmd
    Untracked:  code/Junyan/
    Untracked:  code/about.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/feature_selection_old.Rmd
    Untracked:  code/license.Rmd
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/mofa_varied_omics_combinations.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Klingmuller/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/MethodDev_LungCancer_Datasets.RData
    Untracked:  data/mofa/
    Untracked:  data/patient_metadata.tsv
    Untracked:  data/rf/
    Untracked:  data/stats/
    Untracked:  output/TAC_2023/
    Untracked:  output/YIG_seminar/
    Untracked:  output/preliminary_analysis_untargeted.html

Unstaged changes:
    Deleted:    analysis/mofa_varied_omics_combinations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/preliminary_analysis_DIA.Rmd</code>) and HTML
(<code>docs/preliminary_analysis_DIA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/9a0840b76aceb404bc0147e8caca1cd3a5a42beb/analysis/preliminary_analysis_DIA.Rmd" target="_blank">9a0840b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-08-04
</td>
<td>
Redo feature filtering for DIA proteomics and rerun subsequent analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/83a849a244396dfe32e55476a8e333eebc667fb4/analysis/preliminary_analysis_DIA.Rmd" target="_blank">83a849a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-31
</td>
<td>
Do single-omics analysis of new DIA plasma and tissue proteomics from AG
Krijgsveld and compare it with other proteomics we have
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/128dd12ec84dc9456c33f1481db143f6784d230d/docs/preliminary_analysis_DIA.html" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/f8e22ca7461cbc4d794b9383ab6485721287bacf/analysis/preliminary_analysis_DIA.Rmd" target="_blank">f8e22ca</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
<td>
Update untargeted metabolomics and lipidomics analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/4d5db4a46e8ac975848741857077394c93902c56/analysis/preliminary_analysis_DIA.Rmd" target="_blank">4d5db4a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
<td>
Reanalyze filtered untargeted metabolomics and lipidomics with PPCA and
additional patient metadata
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/preliminary_analysis_DIA.html" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/b0836405d1d46e75d85da360f9aa744d48e4ae9c/analysis/preliminary_analysis_DIA.Rmd" target="_blank">b083640</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-27
</td>
<td>
Preprocess DIA tissue proteomics from AG Klingmüller and compare it with
DIA and DDA data from AG Krijgsveld
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/preliminary_analysis_DIA.html" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/analysis/preliminary_analysis_DIA.Rmd" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Filter out features with massive missingness and run PPCA instead of
conventional PCA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/3959bda2b761e6042f9d33c2e48b406617714006/docs/preliminary_analysis_DIA.html" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
<td>
Filter out features with massive missingness and run PPCA instead of
conventional PCA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/preliminary_analysis_DIA.html" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/067f268d4dd15d5996a358a29212c8cdd93fcf2e/analysis/preliminary_analysis_DIA.Rmd" target="_blank">067f268</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
<td>
Publish initial DIA proteomics preprocessing and single-omics analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/120c24c4b83817c76e7d9d480612e93e706ac572/analysis/preliminary_analysis_DIA.Rmd" target="_blank">120c24c</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-04
</td>
<td>
Perform preliminary analysis of DIA proteomics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Conduct univariate (association test)
and multivariate (PCA) analyses on individual datasets, DIA Plasma and
Tissue Proteomics from AG Krijgsveld and DIA Tissue Proteomics generated
by AG Klingmüller (3 datasets), to take an initial look at power of each
dataset in terms of predicting patient cancer recurrences and also
potentially to provide evidence for latter multi-omics analyses.
Association test between each feature (protein level) and cancer
recurrence was performed to capture significant features that can
separate recurrence and non-recurrence patients. PCA was performed to
see if there is any significant PC that can separate recurrence and
non-recurrence patients. At the end of this report, summary of all
Method Development datasets we currently have is also displayed.<br />
</p>
<p>DIA Tissue Proteomics from AG Krijgsveld seems to have bad quality
due to weak separation of Normal and Tumor tissues? </font></p>
<p><strong>Metadata variables</strong><br />
Patient:<br />
Gender -&gt; Male:Female = 13:7<br />
Age -&gt; Diagnosis ages ranging from 56 to 84<br />
Smoking -&gt; Smoker:Ex-smoker:Non-Smoker = 3:14:3<br />
Recurrence -&gt; Cancer recurrences, Yes:No = 1:1<br />
Stage -&gt; Pathological stages, IB:IIA:IIB = 16:1:3<br />
Metastasis -&gt; Derived from stages, whether cancer cells spread to
lymph nodes, Yes:No = 3:17<br />
Sample:<br />
Condition -&gt; Baseline:Follow-up:Recurrence = 2:1:1, Tumor:Normal =
1:1<br />
TimePoint -&gt; Time points that samples were collected at, Baseline:2
years later = 1:1</p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;SummarizedExperiment&#39;)
library(&#39;proDA&#39;)
library(&#39;ggrepel&#39;)
library(&#39;ggfortify&#39;)
library(&#39;openxlsx&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))</code></pre>
<div id="plasma-proteomics-ag-krijgsveld" class="section level1">
<h1>Plasma Proteomics (AG Krijgsveld)</h1>
<pre class="r"><code># Load normalized data
proPlasma &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm_DIA.rds&#39;)
# Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(proPlasma), rownames = &#39;Protein.Groups&#39;)
# Manually modify metadata information (time point, metastasis)
tmp_metadata &lt;- as.data.frame(colData(proPlasma)) %&gt;%
  dplyr::mutate(
    TimePoint = dplyr::case_when(
      Condition == &#39;Baseline&#39; ~ &#39;Baseline&#39;,
      Condition != &#39;Baseline&#39; ~ &#39;2 years later&#39;
      ),
    Metastasis = case_when(
      Stage == &#39;IIB&#39; ~ &#39;Yes&#39;,
      Stage != &#39;IIB&#39; ~ &#39;No&#39;
    ))
colData(proPlasma)[&#39;TimePoint&#39;] &lt;- tmp_metadata$TimePoint
colData(proPlasma)[&#39;Metastasis&#39;] &lt;- tmp_metadata$Metastasis

# Perform analysis
proPlasmaRes &lt;- doSOA(proPlasma, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                              &#39;Condition&#39;, &#39;TimePoint&#39;, &#39;Smoking&#39;,
                                              &#39;Stage&#39;, &#39;Metastasis&#39;),
                      do_onlyPCA = T, pca_method = &#39;ppca&#39;)
tPCASigRes &lt;- proPlasmaRes$tPCASigRes
pcaRes &lt;- proPlasmaRes$pcaRes
pcTab &lt;- proPlasmaRes$pcTab</code></pre>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2). Note that each patient has two samples that
have same patient annotations, so association test results may have
additive effects. Besides, some patient annotations are only meaningful
at Baseline time point, such as diagnosis ages, pathological stages,
etc.</p>
<pre class="r"><code># Display PCs of interest
tPCASigRes</code></pre>
<pre><code>   Var1       Var2         pVal    pValAdj       Stat        Test
1  PC11    Smoking 0.0002017478 0.03227966 11.3368244       ANOVA
2  PC11 Recurrence 0.0007783770 0.06227016 -3.7130878      T-test
3  PC17        Age 0.0027338898 0.14580746 -0.4978633 Correlation
4  PC11  Condition 0.0045067519 0.18027008  6.4631857       ANOVA
5  PC15      Stage 0.0073403717 0.23489189  5.7827286       ANOVA
6  PC14      Stage 0.0118389649 0.28167068  5.1364048       ANOVA
7  PC11 Metastasis 0.0143414382 0.28167068 -2.5896442      T-test
8   PC9     Gender 0.0154894879 0.28167068 -2.5573377      T-test
9   PC2     Gender 0.0158439755 0.28167068 -2.5478079      T-test
10 PC16  Condition 0.0272296120 0.37510588  4.0567537       ANOVA
11 PC14 Metastasis 0.0272766301 0.37510588 -2.3134682      T-test
12 PC11      Stage 0.0281329407 0.37510588  4.0156192       ANOVA
13 PC13  Condition 0.0335990417 0.38175670  3.7933384       ANOVA
14  PC1 Metastasis 0.0379643078 0.38175670  2.1648603      T-test
15 PC12     Gender 0.0421971826 0.38175670 -2.1162327      T-test
16  PC1      Stage 0.0427577862 0.38175670  3.4956123       ANOVA
17 PC19        Age 0.0431089825 0.38175670  0.3489449 Correlation
18  PC5 Metastasis 0.0432858170 0.38175670 -2.1044282      T-test
19 PC11     Gender 0.0453336080 0.38175670 -2.0829174      T-test</code></pre>
<p>Visualize some significant PCs</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2, col=Gender, shape=Metastasis, group=Patient)) +
  geom_point(size = 4) +
  geom_line(col = &#39;grey50&#39;, linetype = &#39;dashed&#39;) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-3-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-3-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pcTab, aes(x=Recurrence, y=PC11, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-3-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-3-2.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-3-2.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display variance explained by each PC</p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-4-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-4-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>=&gt; We attempt to identify biomarkers that can predict whether
patients WILL suffer from cancer recurrence or not. Therefore, we
decided to analyze Baseline and Follow-up samples separately.</p>
<div id="baseline-samples" class="section level2">
<h2>Baseline samples</h2>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2)</p>
<pre class="r"><code># Subset baseline samples
smpBaseIdx &lt;- which(colData(proPlasma)$Condition == &#39;Baseline&#39;)
proBase &lt;- proPlasma[, smpBaseIdx]

# Display significant associations between PCs and metadata variables
proBaseAssoRes &lt;- doSOA(proBase, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                              &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Metastasis&#39;),
                        do_onlyPCA = T, pca_method = &#39;ppca&#39;, num_PCs = 15)
proBaseAssoRes$tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal   pValAdj       Stat        Test
1  PC8      Stage 0.002764126 0.2487714  9.2401400       ANOVA
2  PC1 Metastasis 0.023951427 0.5308680 -2.5115870      T-test
3  PC3        Age 0.032532574 0.5308680 -0.5196195 Correlation
4  PC1      Stage 0.033845860 0.5308680  4.3545173       ANOVA
5  PC5 Recurrence 0.034005385 0.5308680  2.3326423      T-test
6 PC10     Gender 0.035391199 0.5308680 -2.3120295      T-test</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Perform analysis
proBaseRes &lt;- doSOA(proBase, meta_var = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
                    num_PCs = 15, num_PCfeats = 30)
datMat &lt;- proBaseRes$data
smpAnno &lt;- proBaseRes$smpMetadata
tFeatSigRes &lt;- proBaseRes$tFeatSigRes
tPCASigRes &lt;- proBaseRes$tPCASigRes
pcaRes &lt;- proBaseRes$pcaRes
pcTab &lt;- proBaseRes$pcTab
pcTopFeatTab &lt;- proBaseRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
dplyr::mutate(tFeatSigRes, Gene = plyr::mapvalues(Var1,
                                                  from = featInfo$Protein.Groups,
                                                  to = featInfo$Genes,
                                                  warn_missing = F))</code></pre>
<pre><code>         Var1       Var2        pVal   pValAdj      Stat   Test       Gene
1  A0A0B4J1U3 Recurrence 0.004934386 0.6766383  3.332335 T-test   IGLV1-36
2      Q08380 Recurrence 0.005037103 0.6766383 -3.282432 T-test   LGALS3BP
3      P49913 Recurrence 0.005499750 0.6766383  3.239556 T-test       CAMP
4      P80723 Recurrence 0.006616116 0.6766383  3.226888 T-test      BASP1
5      P02538 Recurrence 0.008036085 0.6766383 -3.125892 T-test      KRT6A
6      P01833 Recurrence 0.009885926 0.6936624 -2.952358 T-test       PIGR
7      P68371 Recurrence 0.012585380 0.6942499 -2.930795 T-test     TUBB4B
8      P27169 Recurrence 0.015310568 0.6942499  2.736021 T-test       PON1
9      P04439 Recurrence 0.016533143 0.6942499 -2.697752 T-test      HLA-A
10     P61026 Recurrence 0.017649317 0.6942499 -2.715934 T-test      RAB10
11     P07339 Recurrence 0.020786945 0.6942499 -2.583082 T-test       CTSD
12     P02745 Recurrence 0.020879498 0.6942499 -2.580847 T-test       C1QA
13     Q9ULV4 Recurrence 0.024803222 0.6942499 -2.564327 T-test     CORO1C
14     O95810 Recurrence 0.026850291 0.6942499 -2.494809 T-test     CAVIN2
15     P05141 Recurrence 0.029527685 0.6942499 -2.444297 T-test    SLC25A5
16     P06312 Recurrence 0.029900907 0.6942499  2.398698 T-test    IGKV4-1
17 A0A0C4DH31 Recurrence 0.030182265 0.6942499  2.393905 T-test   IGHV1-18
18 A0A0J9YXX1 Recurrence 0.030560832 0.6942499 -2.387521 T-test IGHV5-10-1
19     P05106 Recurrence 0.031331945 0.6942499 -2.412682 T-test      ITGB3
20     Q9UK55 Recurrence 0.034562931 0.7230051  2.324256 T-test  SERPINA10
21     P14625 Recurrence 0.036064386 0.7230051 -2.302289 T-test    HSP90B1
22 A0A075B6Q5 Recurrence 0.040989688 0.7765296 -2.235804 T-test   IGHV3-64
23     Q05682 Recurrence 0.042423231 0.7765296 -2.294732 T-test      CALD1
24 A0A087WSZ0 Recurrence 0.048190999 0.8290172  2.199373 T-test   IGKV1D-8
25     P01599 Recurrence 0.049229051 0.8290172  2.139665 T-test   IGKV1-17</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(Stat))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, #row scaling is across columns
                   main = &#39;Baseline plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-7-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-7-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-7-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Owing to missing values, there might be a situation that only few
observations detected in one sample group, resulting in unreliable
t-statistics. We used probabilistic dropout model (proDA R package) to
account for data missingness and identify differentially abundant
entities that are significantly associated with cancer recurrence. Note
that features in heatmap are ordered by their t-statistics.</p>
<pre class="r"><code># Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
# fit

# Check coefficient names for Parameter &#39;contrast&#39; in proDA::test_diff()
# proDA::result_names(fit)
# Identify significant differentially abundant entities
all_proDAProBaseRes &lt;- proDA::test_diff(fit,
                                        contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                        sort_by = &#39;pval&#39;)
proDAProBaseRes &lt;- dplyr::select(all_proDAProBaseRes, -c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::filter(pval &lt; 0.05) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(name,
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F))
proDAProBaseRes</code></pre>
<pre><code>         name        pval  adj_pval t_statistic       Gene
1  A0A0B4J1U3 0.002811053 0.7199974   -3.566735   IGLV1-36
2      P49913 0.005865462 0.7199974   -3.208124       CAMP
3      P80723 0.006044850 0.7199974   -3.193410      BASP1
4      P01833 0.009400173 0.7199974    2.977135       PIGR
5      Q08380 0.012356629 0.7199974    2.842333   LGALS3BP
6      P04439 0.013423765 0.7199974    2.801326      HLA-A
7      P02538 0.013840188 0.7199974    2.786179      KRT6A
8      Q9ULV4 0.015834347 0.7199974    2.719276     CORO1C
9      P61026 0.016348112 0.7199974    2.703364      RAB10
10     P27169 0.019361404 0.7199974   -2.618768       PON1
11     Q05682 0.022902510 0.7199974    2.534229      CALD1
12     P68371 0.024499739 0.7199974    2.500125     TUBB4B
13     P05141 0.024810621 0.7199974    2.493735    SLC25A5
14 A0A0C4DH31 0.025790473 0.7199974   -2.474082   IGHV1-18
15     O95810 0.027608367 0.7199974    2.439431     CAVIN2
16 A0A0J9YXX1 0.029228451 0.7199974    2.410331 IGHV5-10-1
17     P07339 0.029582016 0.7199974    2.404183       CTSD
18     P05106 0.031798051 0.7199974    2.367167      ITGB3
19 A0A075B6Q5 0.033449139 0.7199974    2.341139   IGHV3-64
20     P02745 0.034204152 0.7199974    2.329638       C1QA
21     P01599 0.045659881 0.9153719   -2.179323   IGKV1-17
22     P06312 0.047968563 0.9179439   -2.153357    IGKV4-1</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(proDAProBaseRes, dplyr::desc(t_statistic))$name
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-8-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-8-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Visualize data of top 6 significant features based on proDA
result</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- proDAProBaseRes[1:6,] %&gt;%
  dplyr::select(name, pval, Gene) %&gt;%
  dplyr::rename(Feature = name, pVal = pval)
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats$Feature,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = paste0(Gene, &#39;\n(p = &#39;, round(pVal, 4), &#39;)&#39;),
                Gene = factor(Gene, levels = unique(Gene))) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-9-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-9-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-9-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>p &lt;- all_proDAProBaseRes$pval
hist(p, xlab = &#39;P-value&#39;, main = &#39;Baseline Plasma Proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-10-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-10-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal   pValAdj     Stat   Test
1  PC5 Recurrence 0.03400539 0.5100808 2.332642 T-test</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Baseline plasma proteomics&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-11-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-11-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-11-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PC5 potentially separates recurrence and non-recurrence patients.</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC5, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Baseline plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-12-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-12-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-12-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot molecular signatures captured by PC5 in input data. We took a
look at top 20 features with highest loadings and observed protein
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings. There is no single feature that
can decently hint if patients will suffer cancer recurrence or not,
which means each of these top features contributes a bit to explaining
variation between recurrence and non-recurrence patients.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC5$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
# Change protein names to gene names
datMatSub &lt;- datMat[topFeatsIdx, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma proteomics - PC5&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-13-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-13-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-13-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display top features that build the PC5</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab &lt;- dplyr::mutate(pcTopFeatTab$PC5, Gene = plyr::mapvalues(Feature,
                                                                       from = featInfo$Protein.Groups,
                                                                       to = featInfo$Genes,
                                                                       warn_missing = F))
pcTopFeatTab</code></pre>
<pre><code>      Feature    Loading       Gene
1      P05109 0.30933336     S100A8
2      P06702 0.27450558     S100A9
3  A0A0J9YXX1 0.26164364 IGHV5-10-1
4      P02730 0.18484733     SLC4A1
5      P02545 0.16972269       LMNA
6      Q08830 0.16644718       FGL1
7  A0A075B6I1 0.16039633   IGLV4-60
8      P04275 0.15778177        VWF
9      Q13790 0.13720214       APOF
10     P62328 0.12494514     TMSB4X
11     P02751 0.11689121        FN1
12     P20742 0.10534742        PZP
13     P05141 0.10345438    SLC25A5
14     P23284 0.10186125       PPIB
15     Q05682 0.09938294      CALD1
16     P30740 0.09690732   SERPINB1
17     P02538 0.09629934      KRT6A
18     Q06033 0.09448717      ITIH3
19     P02656 0.08934928      APOC3
20     Q86YZ3 0.08710937       HRNR
21 A0A0C4DH31 0.08594824   IGHV1-18
22     P02775 0.08483874       PPBP
23     Q06830 0.08438013      PRDX1
24     P0DJI8 0.08355375       SAA1
25     P01714 0.08271785   IGLV3-19
26     P07737 0.08120008       PFN1
27 A0A075B6Q5 0.08117949   IGHV3-64
28     P13645 0.08088325      KRT10
29     P29401 0.08072294        TKT
30     P62937 0.07973489       PPIA</code></pre>
</div>
<div id="follow-up-samples" class="section level2">
<h2>Follow-up samples</h2>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2)</p>
<pre class="r"><code># Subset follow-up samples
smpFoloIdx &lt;- which(colData(proPlasma)$Condition != &#39;Baseline&#39;)
proFolo &lt;- proPlasma[, smpFoloIdx]

# Display significant associations between PCs and metadata variables
proFoloAssoRes &lt;- doSOA(proFolo, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;),
                        do_onlyPCA = T, pca_method = &#39;ppca&#39;, num_PCs = 15)
proFoloAssoRes$tPCASigRes</code></pre>
<pre><code>  Var1       Var2      pVal   pValAdj     Stat   Test
1  PC7 Recurrence 0.0472877 0.4794623 2.160892 T-test</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Perform analysis
proFoloRes &lt;- doSOA(proFolo, meta_var = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
                    num_PCs = 15, num_PCfeats = 30)
datMat &lt;- proFoloRes$data
smpAnno &lt;- proFoloRes$smpMetadata
tFeatSigRes &lt;- proFoloRes$tFeatSigRes
tPCASigRes &lt;- proFoloRes$tPCASigRes
pcaRes &lt;- proFoloRes$pcaRes
pcTab &lt;- proFoloRes$pcTab
pcTopFeatTab &lt;- proFoloRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
dplyr::mutate(tFeatSigRes, Gene = plyr::mapvalues(Var1,
                                                  from = featInfo$Protein.Groups,
                                                  to = featInfo$Genes,
                                                  warn_missing = F)) %&gt;%
  dplyr::select(-Test)</code></pre>
<pre><code>           Var1       Var2       pVal   pValAdj      Stat              Gene
1    A0A075B6J1 Recurrence 0.02003575 0.9996994  2.717080          IGLV5-37
2        Q13822 Recurrence 0.02497644 0.9996994  2.533137             ENPP2
3 P01782;P0DP04 Recurrence 0.02528369 0.9996994  2.484156 IGHV3-43D;IGHV3-9
4        P02649 Recurrence 0.02788789 0.9996994  2.434297              APOE
5        P02748 Recurrence 0.03282869 0.9996994 -2.350775                C9
6        Q92954 Recurrence 0.03848447 0.9996994  2.268626              PRG4
7        P01817 Recurrence 0.03969102 0.9996994  2.252577           IGHV2-5
8    A0A0J9YXX1 Recurrence 0.04422811 0.9996994 -2.210412        IGHV5-10-1
9        P23528 Recurrence 0.04946733 0.9996994  2.184793              CFL1</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(Stat))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Follow-up plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-16-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-16-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-16-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Use probabilistic dropout model to account for data missingness and
identify differentially abundant entities that are significantly
associated with cancer recurrence. Note that features in heatmap are
ordered by their t-statistics.</p>
<pre class="r"><code># Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
# fit

# Check coefficient names for Parameter &#39;contrast&#39; in proDA::test_diff()
# proDA::result_names(fit)
# Identify significant differentially abundant entities
all_proDAProFoloRes &lt;- proDA::test_diff(fit,
                                        contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                        sort_by = &#39;pval&#39;)
proDAProFoloRes &lt;- dplyr::select(all_proDAProFoloRes, -c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::filter(pval &lt; 0.05) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(name,
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F))
proDAProFoloRes</code></pre>
<pre><code>           name       pval  adj_pval t_statistic              Gene
1    A0A075B6J1 0.01586506 0.9973257   -2.718311          IGLV5-37
2 P01782;P0DP04 0.02417340 0.9973257   -2.506917 IGHV3-43D;IGHV3-9
3    A0A0J9YXX1 0.03169641 0.9973257    2.368811        IGHV5-10-1
4        P02748 0.03534464 0.9973257    2.312710                C9
5        P01817 0.04237103 0.9973257   -2.218499           IGHV2-5
6        Q13822 0.04278423 0.9973257   -2.213425             ENPP2
7        Q92954 0.04608097 0.9973257   -2.174497              PRG4
8        P02649 0.04669447 0.9973257   -2.167540              APOE
9    A0A075B6S5 0.04883658 0.9973257    2.143893          IGKV1-27</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(proDAProFoloRes, dplyr::desc(t_statistic))$name
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Follow-up plasma proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-17-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Visualize data of top 6 significant features based on proDA
result</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- proDAProFoloRes[1:6,] %&gt;%
  dplyr::select(name, pval, Gene) %&gt;%
  dplyr::rename(Feature = name, pVal = pval)
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats$Feature,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = paste0(stringr::str_remove(Gene, &#39;;.*&#39;),
                              &#39;\n(p = &#39;, round(pVal, 4), &#39;)&#39;),
                Gene = factor(Gene, levels = unique(Gene))) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-18-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-18-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-18-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>p &lt;- all_proDAProFoloRes$pval
hist(p, xlab = &#39;P-value&#39;, main = &#39;Follow-up Plasma Proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-19-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2      pVal   pValAdj     Stat   Test
1  PC7 Recurrence 0.0472877 0.3196416 2.160892 T-test</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Follow-up plasma proteomics&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-20-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-20-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PC7 potentially separates recurrence and non-recurrence patients.</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC7, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Follow-up plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-21-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot molecular signatures captured by PC7 in input data. We took a
look at top 20 features with highest loadings and observed protein
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC7$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
# Change protein names to gene names
datMatSub &lt;- datMat[topFeatsIdx, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Follow-up plasma proteomics - PC7&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-22-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-22-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display top features that build the PC7</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab &lt;- dplyr::mutate(pcTopFeatTab$PC7, Gene = plyr::mapvalues(Feature,
                                                                       from = featInfo$Protein.Groups,
                                                                       to = featInfo$Genes,
                                                                       warn_missing = F))
pcTopFeatTab</code></pre>
<pre><code>             Feature    Loading               Gene
1             P01704 0.25054546           IGLV2-14
2             P02533 0.23932767              KRT14
3             P05109 0.21743773             S100A8
4             P09172 0.21688073                DBH
5             P02538 0.21320094              KRT6A
6             P13647 0.20614320               KRT5
7             P11226 0.18976469               MBL2
8             P06702 0.18832615             S100A9
9             P02545 0.18193739               LMNA
10            P01880 0.16973256               IGHD
11        A0A0G2JMI3 0.14761474         IGHV1-69-2
12        A0A0B4J1U3 0.13612220           IGLV1-36
13        A0A0J9YXX1 0.13325981         IGHV5-10-1
14        A0A0G2JS06 0.13195416           IGLV5-39
15            P07333 0.11745863              CSF1R
16            P0DJI9 0.10758455               SAA2
17            P20742 0.10600724                PZP
18     P11169;Q8TDB8 0.10514740     SLC2A14;SLC2A3
19 A0A0B4J2D9;P0DP09 0.10032402 IGKV1-13;IGKV1D-13
20            Q86YZ3 0.09747706               HRNR
21            A0M8Q6 0.09516126              IGLC7
22            P14151 0.09286508               SELL
23        A0A075B6I1 0.08721417           IGLV4-60
24            P08519 0.08638822                LPA
25            P55056 0.08548302              APOC4
26            P00915 0.08521883                CA1
27            P01861 0.08401075              IGHG4
28            P01706 0.08298383           IGLV2-11
29        A0A0C4DH31 0.08260859           IGHV1-18
30            P01023 0.08215703                A2M</code></pre>
</div>
<div id="changes-in-time-course" class="section level2">
<h2>Changes in time course</h2>
<p>Differences between Follow-up and Baseline samples (Follow-up matrix
- baseline matrix):<br />
</p>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2)</p>
<pre class="r"><code># Subset baseline samples
smpBaseIdx &lt;- which(colData(proPlasma)$Condition == &#39;Baseline&#39;)
proBase &lt;- proPlasma[, smpBaseIdx]
# Subset follow-up samples
smpFoloIdx &lt;- which(colData(proPlasma)$Condition != &#39;Baseline&#39;)
proFolo &lt;- proPlasma[, smpFoloIdx]
# Modify column names for matching two matrices
colnames(proBase) &lt;- stringr::str_remove(colnames(proBase), &#39;_P_B&#39;)
colnames(proFolo) &lt;- stringr::str_remove(colnames(proFolo), &#39;_P_R&#39;)
# Retrieve data matrix
proBaseMat &lt;- SummarizedExperiment::assay(proBase)
proFoloMat &lt;- SummarizedExperiment::assay(proFolo)
# Subtract follow-up matrix by baseline matrix
if (identical(colnames(proBaseMat), colnames(proFoloMat)) &amp;
    identical(rownames(proBaseMat), rownames(proFoloMat))) {
  proPlasmaDiffMat &lt;- proFoloMat - proBaseMat
} else {
  print(&#39;Align samples or features first.&#39;)
}
# Retrieve samples metadata for creating SE object
colAnno &lt;- as.data.frame(colData(proBase))
proPlasmaDiff &lt;- SummarizedExperiment(assays = proPlasmaDiffMat, colData = colAnno)

# Display significant associations between PCs and metadata variables
proPlasmaDiffAssoRes &lt;- doSOA(proPlasmaDiff, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                          &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Metastasis&#39;),
                              do_onlyPCA = T, pca_method = &#39;ppca&#39;, num_PCs = 15)
proPlasmaDiffAssoRes$tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal   pValAdj     Stat   Test
1  PC7    Smoking 0.01698656 0.6707131 5.529684  ANOVA
2  PC7     Gender 0.02411575 0.6707131 2.508126 T-test
3  PC8 Metastasis 0.04536010 0.6707131 2.182783 T-test
4 PC12    Smoking 0.04702411 0.6707131 3.833444  ANOVA</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Perform analysis
proPlasmaDiffRes &lt;- doSOA(proPlasmaDiff, meta_var = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
                          num_PCs = 15, num_PCfeats = 30)
datMat &lt;- proPlasmaDiffRes$data
smpAnno &lt;- proPlasmaDiffRes$smpMetadata
tFeatSigRes &lt;- proPlasmaDiffRes$tFeatSigRes
tPCASigRes &lt;- proPlasmaDiffRes$tPCASigRes

# Display features that can significantly separate recurrence and non-recurrence
# patients
dplyr::mutate(tFeatSigRes, Gene = plyr::mapvalues(Var1,
                                                  from = featInfo$Protein.Groups,
                                                  to = featInfo$Genes,
                                                  warn_missing = F))</code></pre>
<pre><code>         Var1       Var2         pVal   pValAdj      Stat   Test     Gene
1      P02745 Recurrence 0.0009437409 0.3973149  4.101314 T-test     C1QA
2      P61769 Recurrence 0.0200113816 0.9822648  2.602195 T-test      B2M
3  A0A0C4DH29 Recurrence 0.0214478795 0.9822648  2.613534 T-test  IGHV1-3
4  A0A075B6J1 Recurrence 0.0224212067 0.9822648  2.697191 T-test IGLV5-37
5      P01742 Recurrence 0.0253562680 0.9822648 -2.482702 T-test IGHV1-69
6      P01871 Recurrence 0.0264572031 0.9822648  2.461111 T-test     IGHM
7      P01591 Recurrence 0.0315869119 0.9822648  2.370587 T-test   JCHAIN
8      P04070 Recurrence 0.0388732092 0.9822648 -2.263404 T-test     PROC
9  A0A075B6S5 Recurrence 0.0400679579 0.9822648 -2.247656 T-test IGKV1-27
10     Q9ULV4 Recurrence 0.0438176099 0.9822648  2.390559 T-test   CORO1C
11     P04439 Recurrence 0.0440744212 0.9822648  2.229095 T-test    HLA-A</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(Stat))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Differential plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-25-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-25-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-25-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Use probabilistic dropout model to account for data missingness and
identify differentially abundant entity differences that are
significantly associated with cancer recurrence. Note that there is no
differentially abundant protein.</p>
<pre class="r"><code># Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
# fit

# Check coefficient names for Parameter &#39;contrast&#39; in proDA::test_diff()
# proDA::result_names(fit)
# Identify significant differentially abundant entities
proDAProPlasmaDiffRes &lt;- proDA::test_diff(fit,
                                          contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                          sort_by = &#39;pval&#39;, n_max = 20) %&gt;%
  dplyr::select(-c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(name,
                                       from = featInfo$Protein.Groups,
                                       to = featInfo$Genes,
                                       warn_missing = F))
proDAProPlasmaDiffRes</code></pre>
<pre><code>      name      pval  adj_pval t_statistic     Gene
140 P02745 0.5657252 0.9998735  -0.5873203     C1QA
119 P01871 0.6073851 0.9998735  -0.5248103     IGHM
90  P01591 0.6451897 0.9998735  -0.4698935   JCHAIN
155 P02786 0.6596865 0.9998735   0.4492326     TFRC
365 Q12805 0.6635088 0.9998735  -0.4438189   EFEMP1
48  B9A064 0.6779719 0.9998735   0.4234558    IGLL5
336 P61769 0.6859097 0.9998735  -0.4123585      B2M
184 P05362 0.6894870 0.9998735  -0.4073751    ICAM1
103 P01717 0.6926200 0.9998735   0.4030191 IGLV3-25
165 P04114 0.7003480 0.9998735  -0.3923089     APOB
260 P14151 0.7041439 0.9998735   0.3870656     SELL
278 P22352 0.7050977 0.9998735   0.3857499     GPX3
139 P02743 0.7124810 0.9998735   0.3755886     APCS
66  P00488 0.7174944 0.9998735  -0.3687123    F13A1
398 Q92954 0.7188293 0.9998735  -0.3668845     PRG4
105 P01742 0.7226774 0.9998735   0.3616226 IGHV1-69
163 P04070 0.7264944 0.9998735   0.3564138     PROC
326 P55072 0.7291577 0.9998735  -0.3527854      VCP
157 P02790 0.7297740 0.9998735   0.3519463      HPX
183 P05160 0.7310756 0.9998735  -0.3501755     F13B</code></pre>
<p>Visualize data of top 6 significant features based on regular t-test
result</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = plyr::mapvalues(Feature,
                                          from = featInfo$Protein.Groups,
                                          to = featInfo$Genes,
                                          warn_missing = F),
                Feature = factor(Feature, levels = plyr::mapvalues(topSigFeats,
                                                                   from = featInfo$Protein.Groups,
                                                                   to = featInfo$Genes,
                                                                   warn_missing = F))) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-27-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-27-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-27-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There is no significant association between PCs (Var1) and cancer
recurrence (Var2).</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>[1] Var1    Var2    pVal    pValAdj Stat    Test   
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<div id="tissue-proteomics-ag-krijgsveld" class="section level1">
<h1>Tissue Proteomics (AG Krijgsveld)</h1>
<p><font size = 4> <strong>Tumor and Normal Tissue sample groups cannot
be easily separated, bad data quality?</strong> </font></p>
<pre class="r"><code># Load normalized data
proTissue &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm_DIA.rds&#39;)
# Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(proTissue), rownames = &#39;Protein.Groups&#39;)
# Manually modify metadata information (metastasis)
tmp_metadata &lt;- as.data.frame(colData(proTissue)) %&gt;%
  dplyr::mutate(
    Metastasis = case_when(
      Stage == &#39;IIB&#39; ~ &#39;Yes&#39;,
      Stage != &#39;IIB&#39; ~ &#39;No&#39;
    ))
colData(proTissue)[&#39;Metastasis&#39;] &lt;- tmp_metadata$Metastasis

# Perform analysis
proTissueRes &lt;- doSOA(proTissue, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                              &#39;Condition&#39;, &#39;Smoking&#39;, &#39;Stage&#39;,
                                              &#39;Metastasis&#39;),
                      do_onlyPCA = T, pca_method = &#39;ppca&#39;)
tPCASigRes &lt;- proTissueRes$tPCASigRes
pcaRes &lt;- proTissueRes$pcaRes
pcTab &lt;- proTissueRes$pcTab</code></pre>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2). Note that each patient has two samples that
have same patient annotations, so association test results may have
additive effects.</p>
<pre class="r"><code># Display PCs of interest
tPCASigRes</code></pre>
<pre><code>  Var1       Var2         pVal     pValAdj       Stat        Test
1  PC6      Stage 3.034066e-05 0.004247693 14.8259177       ANOVA
2 PC13 Metastasis 1.202939e-02 0.691217409 -2.6626942      T-test
3  PC1      Stage 1.481180e-02 0.691217409  4.8402832       ANOVA
4 PC13      Stage 3.205828e-02 0.718171937  3.8518572       ANOVA
5  PC6        Age 3.807032e-02 0.718171937  0.3572340 Correlation
6 PC10        Age 3.863527e-02 0.718171937 -0.3562619 Correlation
7 PC15  Condition 4.395972e-02 0.718171937 -2.0972518      T-test
8 PC16  Condition 4.528282e-02 0.718171937  2.0834404      T-test
9 PC10 Recurrence 4.823825e-02 0.718171937 -2.0538320      T-test</code></pre>
<p>Visualize some significant PCs</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC15, y=PC16, col=Condition, group=Patient)) +
  geom_point(size = 4) +
  geom_line(col = &#39;grey50&#39;, linetype = &#39;dashed&#39;) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-31-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-31-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-31-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pcTab, aes(x=Condition, y=PC15, color=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-31-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Condition, y=PC16, color=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-31-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display variance explained by each PC</p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-33-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-33-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-33-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tissue-proteomics-ag-klingmüller" class="section level1">
<h1>Tissue Proteomics (AG Klingmüller)</h1>
<p>Based on PCA result obtained by analyzing whole Tissue Proteomics
dataset that includes Tumor and Normal samples, we found that PCA mainly
captures source of variation in tissue conditions, which indicates good
data quality. Tumor and Normal samples are characteristically different
and might be noise to each other. Therefore, we decided to analyze Tumor
and Normal samples separately.</p>
<pre class="r"><code># Load normalized data
proTissue &lt;- readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)
# Load another Method Development datasets for retrieving MD samples
metaTissue &lt;- readRDS(&#39;./data/AG_Hell/metaTissueNorm.rds&#39;)
methodDevSmps &lt;- colnames(metaTissue)
proTissue &lt;- proTissue[, colnames(proTissue) %in% methodDevSmps]
# Prepare protein-gene table for changing protein names to gene names
featInfo &lt;- tibble::as_tibble(rowData(proTissue), rownames = &#39;PG.Groups&#39;)
# Manually modify metadata information (metastasis)
tmp_metadata &lt;- as.data.frame(colData(proTissue)) %&gt;%
  dplyr::mutate(
    Metastasis = case_when(
      Stage == &#39;IIB&#39; ~ &#39;Yes&#39;,
      Stage != &#39;IIB&#39; ~ &#39;No&#39;
    ))
colData(proTissue)[&#39;Metastasis&#39;] &lt;- tmp_metadata$Metastasis

# Perform analysis
proTissueRes &lt;- doSOA(proTissue, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                              &#39;Condition&#39;, &#39;Smoking&#39;, &#39;Stage&#39;,
                                              &#39;Metastasis&#39;),
                      do_onlyPCA = T, pca_method = &#39;ppca&#39;)
tPCASigRes &lt;- proTissueRes$tPCASigRes
pcaRes &lt;- proTissueRes$pcaRes
pcTab &lt;- proTissueRes$pcTab</code></pre>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2). Note that each patient has two samples that
have same patient annotations, so association test results may have
additive effects.</p>
<pre class="r"><code># Display PCs of interest
tPCASigRes</code></pre>
<pre><code>   Var1       Var2         pVal      pValAdj        Stat        Test
1   PC1  Condition 2.305324e-19 3.227454e-17 -18.5573405      T-test
2   PC7 Metastasis 4.008766e-04 2.806136e-02  -3.9253429      T-test
3  PC13     Gender 8.191656e-04 3.822773e-02  -3.6723130      T-test
4   PC7      Stage 1.767187e-03 6.185155e-02   7.7278740       ANOVA
5   PC7    Smoking 3.269450e-03 9.154459e-02   6.8411281       ANOVA
6   PC8 Metastasis 6.994154e-03 1.598143e-01   2.8709707      T-test
7  PC13 Recurrence 8.845123e-03 1.598143e-01  -2.7777172      T-test
8  PC19      Stage 9.132244e-03 1.598143e-01   5.4323570       ANOVA
9  PC14    Smoking 2.386533e-02 3.498316e-01   4.1919333       ANOVA
10  PC8      Stage 2.498797e-02 3.498316e-01   4.1343674       ANOVA
11  PC3     Gender 3.508904e-02 4.465877e-01  -2.1950137      T-test
12  PC4 Metastasis 3.892564e-02 4.524819e-01   2.1480060      T-test
13 PC13        Age 4.201618e-02 4.524819e-01  -0.3407027 Correlation</code></pre>
<p>PC1 can significantly separate Normal and Tumor tissues</p>
<pre class="r"><code>ggplot(pcTab, aes(x=Condition, y=PC1, color=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-36-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-36-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-36-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pcTab, aes(x=Recurrence, y=PC13, color=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-36-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display variance explained by each PC</p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-37-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-37-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="tumor-samples" class="section level2">
<h2>Tumor samples</h2>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2)</p>
<pre class="r"><code># Subset tumor samples
smpTumorIdx &lt;- which(colData(proTissue)$Condition == &#39;Tumor&#39;)
proTumor &lt;- proTissue[, smpTumorIdx]

# Display significant associations between PCs and metadata variables
proTumorAssoRes &lt;- doSOA(proTumor, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Metastasis&#39;),
                         do_onlyPCA = T, pca_method = &#39;ppca&#39;, num_PCs = 16)
proTumorAssoRes$tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal   pValAdj       Stat        Test
1  PC4 Metastasis 0.005067460 0.2760318  3.2456238      T-test
2 PC11     Gender 0.005750662 0.2760318  3.1854658      T-test
3  PC4      Stage 0.022509439 0.7203021  4.9378688       ANOVA
4 PC11        Age 0.043662352 0.7866260  0.4802791 Correlation
5 PC15 Recurrence 0.046109343 0.7866260 -2.1620480      T-test
6 PC11 Recurrence 0.049164127 0.7866260  2.1286985      T-test</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Perform analysis
proTumorRes &lt;- doSOA(proTumor, meta_var = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
                     num_PCs = 16, num_PCfeats = 30)
datMat &lt;- proTumorRes$data
smpAnno &lt;- proTumorRes$smpMetadata
tFeatSigRes &lt;- proTumorRes$tFeatSigRes
tPCASigRes &lt;- proTumorRes$tPCASigRes
pcaRes &lt;- proTumorRes$pcaRes
pcTab &lt;- proTumorRes$pcTab
pcTopFeatTab &lt;- proTumorRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes &lt;- dplyr::mutate(tFeatSigRes, Gene = plyr::mapvalues(Var1,
                                                                 from = featInfo$PG.Groups,
                                                                 to = featInfo$PG.Genes,
                                                                 warn_missing = F)) %&gt;%
  dplyr::select(-Test)
tFeatSigRes</code></pre>
<pre><code>      Var1       Var2         pVal   pValAdj      Stat     Gene
1   Q9Y3Q0 Recurrence 0.0002162419 0.9998225 -6.368684  NAALAD2
2   O75475 Recurrence 0.0015505714 0.9998225  3.806726    PSIP1
3   Q9BSE4 Recurrence 0.0015874444 0.9998225 -3.795587  HERPUD2
4   Q9Y320 Recurrence 0.0036405987 0.9998225 -3.402540     TMX2
5   Q93063 Recurrence 0.0039264147 0.9998225  3.366715     EXT2
6   Q9UN76 Recurrence 0.0040069545 0.9998225  3.622776  SLC6A14
7   Q9NRN7 Recurrence 0.0043139205 0.9998225 -3.322074 AASDHPPT
8   Q9H0X9 Recurrence 0.0044422450 0.9998225 -3.343715   OSBPL5
9   Q86Y79 Recurrence 0.0045517676 0.9998225 -3.331841    PTRH1
10  Q9BQ95 Recurrence 0.0046274655 0.9998225 -3.288771    ECSIT
11  Q8NFI3 Recurrence 0.0046658796 0.9998225 -3.319768   ENGASE
12  Q9Y3B8 Recurrence 0.0046913182 0.9998225 -3.282264    REXO2
13  Q9BX69 Recurrence 0.0065514385 0.9998225  3.189851    CARD6
14  Q93091 Recurrence 0.0076621889 0.9998225 -3.048535   RNASE6
15  Q6IA69 Recurrence 0.0082420774 0.9998225 -3.013610  NADSYN1
16  P01718 Recurrence 0.0082663724 0.9998225 -3.157026 IGLV3-27
17  Q92843 Recurrence 0.0095588002 0.9998225 -2.999615   BCL2L2
18  Q96M96 Recurrence 0.0098111004 0.9998225  2.929958     FGD4
19  Q68CR1 Recurrence 0.0108460926 0.9998225 -2.906709   SEL1L3
20  P41217 Recurrence 0.0138861908 0.9998225 -2.841316    CD200
21  Q99633 Recurrence 0.0141616847 0.9998225  2.752480   PRPF18
22  Q8NBI5 Recurrence 0.0147734560 0.9998225 -2.731895  SLC43A3
23  Q16647 Recurrence 0.0153821934 0.9998225 -2.712213    PTGIS
24  P35659 Recurrence 0.0156679296 0.9998225  2.703232      DEK
25  Q8WVM0 Recurrence 0.0161520124 0.9998225 -2.688370    TFB1M
26  Q14318 Recurrence 0.0169998533 0.9998225  2.663343    FKBP8
27  Q92620 Recurrence 0.0170865405 0.9998225  2.660852    DHX38
28  Q9NSU2 Recurrence 0.0172861681 0.9998225 -2.726828    TREX1
29  Q9BZK7 Recurrence 0.0176218258 0.9998225  2.645735  TBL1XR1
30  Q6PL18 Recurrence 0.0177307840 0.9998225  2.662828    ATAD2
31  Q66PJ3 Recurrence 0.0177872526 0.9998225  2.641152  ARL6IP4
32  Q99519 Recurrence 0.0181251970 0.9998225 -2.631915     NEU1
33  Q5THJ4 Recurrence 0.0189519232 0.9998225 -2.609996   VPS13D
34  L0R8F8 Recurrence 0.0191046325 0.9998225 -2.606047    MIEF1
35  Q6PJG2 Recurrence 0.0195376766 0.9998225  2.595013   MIDEAS
36  Q9Y3B7 Recurrence 0.0197085579 0.9998225 -2.590723   MRPL11
37  P08134 Recurrence 0.0202119268 0.9998225 -2.578289     RHOC
38  Q9NTI5 Recurrence 0.0203075863 0.9998225  2.575959    PDS5B
39  P06280 Recurrence 0.0205158783 0.9998225 -2.570923      GLA
40  Q07960 Recurrence 0.0208607045 0.9998225 -2.562692  ARHGAP1
41  O00330 Recurrence 0.0210622900 0.9998225 -2.557940     PDHX
42  Q9BTC0 Recurrence 0.0219279335 0.9998225  2.538015    DIDO1
43  Q9H061 Recurrence 0.0228804765 0.9998225 -2.516936 TMEM126A
44  P10515 Recurrence 0.0231782089 0.9998225 -2.510519     DLAT
45  Q9BVG9 Recurrence 0.0233197279 0.9998225 -2.507496   PTDSS2
46  O15047 Recurrence 0.0240825376 0.9998225  2.491497   SETD1A
47  Q969V5 Recurrence 0.0241813654 0.9998225 -2.526778     MUL1
48  Q8NCG7 Recurrence 0.0245271119 0.9998225 -2.482392    DAGLB
49  Q99542 Recurrence 0.0254067452 0.9998225  2.675160    MMP19
50  P50151 Recurrence 0.0256672149 0.9998225  2.476515    GNG10
51  Q9NWW5 Recurrence 0.0260967372 0.9998225 -2.468085     CLN6
52  O96019 Recurrence 0.0261455493 0.9998225  2.450520   ACTL6A
53  Q96AQ6 Recurrence 0.0262941792 0.9998225  2.447687   PBXIP1
54  Q9GZU7 Recurrence 0.0263195668 0.9998225 -2.482937   CTDSP1
55  P51858 Recurrence 0.0263512093 0.9998225  2.446604     HDGF
56  Q6P4Q7 Recurrence 0.0268946662 0.9998225 -2.436394    CNNM4
57  Q7Z3E5 Recurrence 0.0274197518 0.9998225 -2.426713    ARMC9
58  Q8IVS2 Recurrence 0.0276591128 0.9998225 -2.422358     MCAT
59  P16220 Recurrence 0.0277127800 0.9998225  2.421387    CREB1
60  Q9H8Y5 Recurrence 0.0282076717 0.9998225  2.412510   ANKZF1
61  Q8WXI9 Recurrence 0.0282404427 0.9998225  2.411927  GATAD2B
62  Q01628 Recurrence 0.0284947901 0.9998225 -2.407427   IFITM3
63  Q15063 Recurrence 0.0290619833 0.9998225 -2.397526    POSTN
64  P54098 Recurrence 0.0290869036 0.9998225 -2.397096     POLG
65  Q96S55 Recurrence 0.0293075535 0.9998225  2.393296   WRNIP1
66  Q8WVK2 Recurrence 0.0297779276 0.9998225  2.400807  SNRNP27
67  Q99549 Recurrence 0.0300369579 0.9998225  2.380925 MPHOSPH8
68  Q4VC05 Recurrence 0.0306468616 0.9998225  2.370796    BCL7A
69  Q15046 Recurrence 0.0307015903 0.9998225  2.369897    KARS1
70  Q9BW71 Recurrence 0.0307988129 0.9998225  2.368302   HIRIP3
71  Q9NPI1 Recurrence 0.0308552430 0.9998225  2.382608     BRD7
72  P99999 Recurrence 0.0310388316 0.9998225 -2.364386     CYCS
73  Q9P0U4 Recurrence 0.0310857543 0.9998225  2.363624    CXXC1
74  Q9BWQ6 Recurrence 0.0327404517 0.9998225 -2.337412    YIPF2
75  Q14644 Recurrence 0.0329099375 0.9998225 -2.349503    RASA3
76  Q8IWL3 Recurrence 0.0332185159 0.9998225 -2.330070     HSCB
77  Q9NZQ3 Recurrence 0.0337614960 0.9998225  2.321850  NCKIPSD
78  Q9BQL6 Recurrence 0.0340660781 0.9998225 -2.317293   FERMT1
79  Q9NSY2 Recurrence 0.0341794208 0.9998225 -2.315607   STARD5
80  Q04637 Recurrence 0.0342686932 0.9998225  2.314283   EIF4G1
81  Q9UDW1 Recurrence 0.0342793642 0.9998225 -2.314125   UQCR10
82  O43837 Recurrence 0.0346903885 0.9998225 -2.308071    IDH3B
83  Q96I59 Recurrence 0.0348238359 0.9998225 -2.306120    NARS2
84  Q16718 Recurrence 0.0357347966 0.9998225 -2.292984   NDUFA5
85  O95070 Recurrence 0.0358386370 0.9998225 -2.291507    YIF1A
86  P54687 Recurrence 0.0364595643 0.9998225  2.282755    BCAT1
87  Q9UBX1 Recurrence 0.0365808338 0.9998225 -2.294931     CTSF
88  O75110 Recurrence 0.0367881670 0.9998225 -2.292004    ATP9A
89  Q11206 Recurrence 0.0372990885 0.9998225 -2.284858  ST3GAL4
90  P35250 Recurrence 0.0374066131 0.9998225  2.269670     RFC2
91  Q9HB63 Recurrence 0.0374789629 0.9998225 -2.561327     NTN4
92  O60907 Recurrence 0.0380662140 0.9998225  2.260738    TBL1X
93  Q9NUL5 Recurrence 0.0388056129 0.9998225 -2.250896     SHFL
94  Q86VP1 Recurrence 0.0389634676 0.9998225 -2.248817  TAX1BP1
95  Q9ULU4 Recurrence 0.0391164746 0.9998225  2.246809   ZMYND8
96  Q14789 Recurrence 0.0393852070 0.9998225  2.243301   GOLGB1
97  Q9Y5Y7 Recurrence 0.0395729433 0.9998225 -2.269463    LYVE1
98  Q9BV20 Recurrence 0.0398294076 0.9998225 -2.237551     MRI1
99  Q8TES7 Recurrence 0.0409601764 0.9998225  2.236180     FBF1
100 Q7L775 Recurrence 0.0413245677 0.9998225  2.218624 EPM2AIP1
101 Q9HAV7 Recurrence 0.0413908059 0.9998225 -2.217800   GRPEL1
102 P51809 Recurrence 0.0418363130 0.9998225 -2.212291    VAMP7
103 Q6P2E9 Recurrence 0.0422175259 0.9998225  2.207621     EDC4
104 Q6P6C2 Recurrence 0.0425648428 0.9998225  2.203399   ALKBH5
105 P01033 Recurrence 0.0425726044 0.9998225 -2.203305    TIMP1
106 P06731 Recurrence 0.0426912443 0.9998225 -2.201870  CEACAM5
107 P21246 Recurrence 0.0429181865 0.9998225  2.288135      PTN
108 P12270 Recurrence 0.0430018218 0.9998225  2.198133      TPR
109 Q9NPI6 Recurrence 0.0433030921 0.9998225  2.194531    DCP1A
110 Q8N6H7 Recurrence 0.0436384934 0.9998225 -2.190548  ARFGAP2
111 Q9Y5J9 Recurrence 0.0436477376 0.9998225 -2.190438   TIMM8B
112 Q99538 Recurrence 0.0451917668 0.9998225 -2.172463     LGMN
113 P52565 Recurrence 0.0452685845 0.9998225 -2.171584  ARHGDIA
114 P21741 Recurrence 0.0457473480 0.9998225  2.166134      MDK
115 Q9Y6N5 Recurrence 0.0458893826 0.9998225 -2.164527     SQOR
116 Q8NC42 Recurrence 0.0459926884 0.9998225 -2.175506   RNF149
117 P49454 Recurrence 0.0462863586 0.9998225  2.160061    CENPF
118 Q9H147 Recurrence 0.0465264511 0.9998225  2.157377  DNTTIP1
119 P20340 Recurrence 0.0470124055 0.9998225 -2.151982    RAB6A
120 Q13084 Recurrence 0.0471922992 0.9998225 -2.149999   MRPL28
121 Q9NY33 Recurrence 0.0476569682 0.9998225 -2.144906     DPP3
122 Q9P2K3 Recurrence 0.0487591847 0.9998225  2.133008    RCOR3
123 Q13405 Recurrence 0.0488231859 0.9998225 -2.132325   MRPL49
124 Q8N9F7 Recurrence 0.0489556750 0.9998225 -2.130913    GDPD1
125 Q13868 Recurrence 0.0495441478 0.9998225  2.124684   EXOSC2
126 Q96NC0 Recurrence 0.0496387338 0.9998225  2.123689    ZMAT2</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(Stat))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, #row scaling is across columns
                   main = &#39;Tumor Tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-39-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-39-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Owing to missing values, there might be a situation that only few
observations in sample groups, resulting in unreliable t-statistics. To
this end, we used probabilistic dropout model (proDA R package) to
account for data missingness and identify differentially abundant
entities that are significantly associated with cancer recurrence. Note
that features in heatmap are ordered by their t-statistics.</p>
<pre class="r"><code># Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
# fit

# Check coefficient names for Parameter &#39;contrast&#39; in proDA::test_diff()
# proDA::result_names(fit)
# Identify significant differentially abundant entities
all_proDAProTumorRes &lt;- proDA::test_diff(fit,
                                         contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                         sort_by = &#39;pval&#39;)
proDAProTumorRes &lt;- dplyr::select(all_proDAProTumorRes, -c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::filter(pval &lt; 0.05) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(name,
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F))
proDAProTumorRes</code></pre>
<pre><code>     name         pval  adj_pval t_statistic     Gene
1  Q9Y3Q0 5.888143e-05 0.4529749    5.400688  NAALAD2
2  O75475 4.459482e-03 0.9998238   -3.306325    PSIP1
3  Q9Y3B8 5.683614e-03 0.9998238    3.191048    REXO2
4  Q9UN76 7.175442e-03 0.9998238   -3.079911  SLC6A14
5  Q8NFI3 7.368343e-03 0.9998238    3.067234   ENGASE
6  Q9BQ95 7.763728e-03 0.9998238    3.042236    ECSIT
7  Q86Y79 8.941100e-03 0.9998238    2.974574    PTRH1
8  Q93091 9.282011e-03 0.9998238    2.956606   RNASE6
9  P01718 9.621556e-03 0.9998238    2.939340 IGLV3-27
10 Q92843 1.083418e-02 0.9998238    2.882180   BCL2L2
11 Q6IA69 1.131683e-02 0.9998238    2.861146  NADSYN1
12 Q9H0X9 1.134970e-02 0.9998238    2.859745   OSBPL5
13 Q68CR1 1.466738e-02 0.9998238    2.735405   SEL1L3
14 Q9BSE4 1.505756e-02 0.9998238    2.722614  HERPUD2
15 Q16647 1.585219e-02 0.9998238    2.697524    PTGIS
16 Q93063 1.607123e-02 0.9998238   -2.690820     EXT2
17 Q8NBI5 1.701997e-02 0.9998238    2.662764  SLC43A3
18 Q9NRN7 1.894614e-02 0.9998238    2.610146 AASDHPPT
19 Q99519 1.948209e-02 0.9998238    2.596416     NEU1
20 P50151 2.016530e-02 0.9998238   -2.579428    GNG10
21 Q9Y320 2.179613e-02 0.9998238    2.541000     TMX2
22 P06280 2.493818e-02 0.9998238    2.474112      GLA
23 Q96M96 2.495174e-02 0.9998238   -2.473841     FGD4
24 Q969V5 2.635470e-02 0.9998238    2.446538     MUL1
25 Q8WVM0 2.650737e-02 0.9998238    2.443650    TFB1M
26 O00330 2.652843e-02 0.9998238    2.443253     PDHX
27 Q99633 2.701122e-02 0.9998238   -2.434230   PRPF18
28 P41217 2.765922e-02 0.9998238    2.422356    CD200
29 Q96AQ6 2.836786e-02 0.9998238   -2.409668   PBXIP1
30 P10515 2.858679e-02 0.9998238    2.405809     DLAT
31 Q9NSU2 2.869737e-02 0.9998238    2.403870    TREX1
32 Q9H061 2.882351e-02 0.9998238    2.401667 TMEM126A
33 L0R8F8 2.887880e-02 0.9998238    2.400704    MIEF1
34 Q9BVG9 2.949997e-02 0.9998238    2.390005   PTDSS2
35 Q7Z3E5 2.962270e-02 0.9998238    2.387916    ARMC9
36 Q07960 2.968047e-02 0.9998238    2.386936  ARHGAP1
37 Q8NCG7 3.000302e-02 0.9998238    2.381495    DAGLB
38 Q9UBX1 3.073700e-02 0.9998238    2.369315     CTSF
39 P16220 3.147604e-02 0.9998238   -2.357325    CREB1
40 Q6PL18 3.174027e-02 0.9998238   -2.353103    ATAD2
41 Q8IVS2 3.188153e-02 0.9998238    2.350859     MCAT
42 Q66PJ3 3.247959e-02 0.9998238   -2.341460  ARL6IP4
43 Q96S55 3.329166e-02 0.9998238   -2.328955   WRNIP1
44 Q8TES7 3.337118e-02 0.9998238   -2.327746     FBF1
45 Q01628 3.501355e-02 0.9998238    2.303358   IFITM3
46 Q9H8Y5 3.542544e-02 0.9998238   -2.297409   ANKZF1
47 Q6PJG2 3.585064e-02 0.9998238   -2.291336   MIDEAS
48 Q8WXI9 3.622500e-02 0.9998238   -2.286045  GATAD2B
49 Q8NC42 3.696853e-02 0.9998238    2.275684   RNF149
50 Q6P4Q7 3.698629e-02 0.9998238    2.275439    CNNM4
51 Q8WVK2 3.731436e-02 0.9998238   -2.270931  SNRNP27
52 P54098 3.750016e-02 0.9998238    2.268395     POLG
53 Q8IWL3 3.795199e-02 0.9998238    2.262275     HSCB
54 Q92620 3.880509e-02 0.9998238   -2.250903    DHX38
55 Q9NWW5 3.895478e-02 0.9998238    2.248931     CLN6
56 P99999 3.901895e-02 0.9998238    2.248088     CYCS
57 Q4VC05 3.903653e-02 0.9998238   -2.247857    BCL7A
58 Q9BZK7 3.943036e-02 0.9998238   -2.242714  TBL1XR1
59 P54687 3.951774e-02 0.9998238   -2.241579    BCAT1
60 Q9NTI5 3.965980e-02 0.9998238   -2.239740    PDS5B
61 Q9BQL6 3.993052e-02 0.9998238    2.236250   FERMT1
62 P21246 4.076449e-02 0.9998238   -2.225638      PTN
63 Q9NPI1 4.078127e-02 0.9998238   -2.225427     BRD7
64 Q9UDW1 4.087477e-02 0.9998238    2.224250   UQCR10
65 Q9NSY2 4.113650e-02 0.9998238    2.220969   STARD5
66 Q9P0U4 4.119304e-02 0.9998238   -2.220263    CXXC1
67 Q9NUL5 4.175835e-02 0.9998238    2.213251     SHFL
68 Q9BW71 4.214125e-02 0.9998238   -2.208552   HIRIP3
69 Q5THJ4 4.219450e-02 0.9998238    2.207902   VPS13D
70 Q15063 4.243932e-02 0.9998238    2.204921    POSTN
71 P35250 4.278695e-02 0.9998238   -2.200716     RFC2
72 Q96I59 4.358345e-02 0.9998238    2.191199    NARS2
73 Q9Y5J9 4.367252e-02 0.9998238    2.190145   TIMM8B
74 Q9Y3B7 4.467663e-02 0.9998238    2.178396   MRPL11
75 Q9BTC0 4.589353e-02 0.9998238   -2.164480    DIDO1
76 P51858 4.629748e-02 0.9998238   -2.159936     HDGF
77 Q9HAV7 4.653072e-02 0.9998238    2.157329   GRPEL1
78 Q7L775 4.690375e-02 0.9998238   -2.153184 EPM2AIP1
79 Q14318 4.721834e-02 0.9998238   -2.149712    FKBP8
80 P01033 4.753146e-02 0.9998238    2.146277    TIMP1
81 Q11206 4.818593e-02 0.9998238    2.139165  ST3GAL4
82 Q9Y6N5 4.832216e-02 0.9998238    2.137696     SQOR
83 Q99538 4.902606e-02 0.9998238    2.130164     LGMN
84 Q9ULU4 4.994550e-02 0.9998238   -2.120475   ZMYND8
85 P01130 4.995954e-02 0.9998238    2.120328     LDLR</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(proDAProTumorRes, dplyr::desc(t_statistic))$name
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Tumor Tissue proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-40-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-40-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>p &lt;- all_proDAProTumorRes$pval
hist(p, xlab = &#39;P-value&#39;, main = &#39;Tumor Tissue Proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-41-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-41-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-41-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-41-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-41-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Visualize data of top 6 significant features based on proDA
result</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- proDAProTumorRes[1:6,] %&gt;%
  dplyr::select(name, pval, Gene) %&gt;%
  dplyr::rename(Feature = name, pVal = pval)
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats$Feature,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = paste0(Gene, &#39;\n(p = &#39;, round(pVal, 4), &#39;)&#39;),
                Gene = factor(Gene, levels = unique(Gene))) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-42-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-42-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-42-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-42-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-42-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal  pValAdj      Stat   Test
1 PC15 Recurrence 0.04610934 0.393313 -2.162048 T-test
2 PC11 Recurrence 0.04916413 0.393313  2.128698 T-test</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Tumor tissue proteomics&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-43-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-43-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-43-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-43-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PC15 potentially separates recurrence and non-recurrence
patients.</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC15, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tumor Tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-44-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-44-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-44-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot molecular signatures captured by PC15 in input data. We took a
look at top 20 features with highest loadings and observed protein
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings. There is no single feature that
can decently hint if patients will suffer cancer recurrence or not,
which means each of these top features contributes a bit to explaining
variation between recurrence and non-recurrence patients.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC15$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
# Change protein names to gene names
datMatSub &lt;- datMat[topFeatsIdx, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Tumor Tissue proteomics - PC15&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-45-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-45-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Display top features that build the PC15</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
dplyr::mutate(pcTopFeatTab$PC15, Gene = plyr::mapvalues(Feature,
                                                        from = featInfo$PG.Groups,
                                                        to = featInfo$PG.Genes,
                                                        warn_missing = F))</code></pre>
<pre><code>                    Feature    Loading                Gene
1                    Q9H799 0.13994562             CPLANE1
2                    Q6W4X9 0.10958915                MUC6
3                    Q9Y2Y1 0.09097460              POLR3K
4                    P25815 0.07706168               S100P
5                    P35247 0.07381273               SFTPD
6                    P01127 0.07375632               PDGFB
7                    Q13336 0.07309935             SLC14A1
8                    Q9H1X3 0.07016832             DNAJC25
9                    Q9NSB2 0.06766856               KRT84
10                   Q8WXU2 0.06679488              DNAAF4
11                   Q8WVV4 0.06201437               POF1B
12                   Q8NFL0 0.06108079              B3GNT7
13                   O95994 0.05923125                AGR2
14                   P50226 0.05913443             SULT1A2
15                   P01705 0.05790487            IGLV2-23
16                   Q9NXE8 0.05725224               CWC25
17                   P26992 0.05667827               CNTFR
18                   Q03013 0.05343148               GSTM4
19                   P20160 0.05212271                AZU1
20                   P07476 0.05157655                 IVL
21                   Q9NSC7 0.05092696          ST6GALNAC1
22                   P29762 0.05082542              CRABP1
23                   Q00537 0.05032055               CDK17
24                   Q02539 0.04975801                H1-1
25                   P07996 0.04849028               THBS1
26                   Q8TD06 0.04817355                AGR3
27                   Q9NQW6 0.04770211                ANLN
28                   Q9Y2K1 0.04696283               ZBTB1
29 A0A0A6YYL3;A0JP26;H3BUK9 0.04614433 POTEB;POTEB3;POTEB2
30                   P52740 0.04593619              ZNF132</code></pre>
<p>Another significant PC11</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC11, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tumor Tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-47-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-47-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-47-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/3959bda2b761e6042f9d33c2e48b406617714006/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-47-1.png" target="_blank">3959bda</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/2f62f8a2900c25f1a598fc01530afdbf41f7d6bb/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-47-1.png" target="_blank">2f62f8a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC11$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
# Change protein names to gene names
datMatSub &lt;- datMat[topFeatsIdx, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Tumor Tissue proteomics - PC11&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-47-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
dplyr::mutate(pcTopFeatTab$PC11, Gene = plyr::mapvalues(Feature,
                                                        from = featInfo$PG.Groups,
                                                        to = featInfo$PG.Genes,
                                                        warn_missing = F))</code></pre>
<pre><code>      Feature    Loading     Gene
1      Q6W4X9 0.12136336     MUC6
2      O75635 0.09567258 SERPINB7
3      Q9NXE8 0.09533713    CWC25
4      Q13424 0.08744031    SNTA1
5      Q9H1X3 0.07770459  DNAJC25
6      Q9NQW6 0.07728492     ANLN
7      O60218 0.07314394  AKR1B10
8      P30838 0.06985575  ALDH3A1
9      Q00537 0.06886527    CDK17
10 A0A075B6J9 0.06821237 IGLV2-18
11     Q04695 0.06455621    KRT17
12     P15328 0.06359702    FOLR1
13     P01127 0.06357283    PDGFB
14     Q13938 0.05949660     CAPS
15     P56470 0.05857279   LGALS4
16 A0A0B4J1Y8 0.05740466 IGLV9-49
17     Q6P179 0.05590139    ERAP2
18     Q8TF66 0.05564934   LRRC15
19     Q8WYL5 0.05440480     SSH1
20     P01893 0.05387943    HLA-H
21     Q93099 0.05336492      HGD
22     Q9NTW7 0.05148200    ZFP64
23     P27487 0.05081520     DPP4
24     P11509 0.04995874   CYP2A6
25     Q96NY7 0.04987538    CLIC6
26     P31327 0.04940874     CPS1
27     P01344 0.04856748     IGF2
28     Q15847 0.04754163    ADIRF
29     P09327 0.04748133     VIL1
30     P50895 0.04635453     BCAM</code></pre>
</div>
<div id="normal-samples" class="section level2">
<h2>Normal samples</h2>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2)</p>
<pre class="r"><code># Subset normal samples
smpNormalIdx &lt;- which(colData(proTissue)$Condition == &#39;Normal&#39;)
proNormal &lt;- proTissue[, smpNormalIdx]

# Display significant associations between PCs and metadata variables
proNormalAssoRes &lt;- doSOA(proNormal, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                  &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Metastasis&#39;),
                          do_onlyPCA = T, pca_method = &#39;ppca&#39;, num_PCs = 16)
proNormalAssoRes$tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal   pValAdj       Stat        Test
1 PC11      Stage 0.001205676 0.1157449 10.8751339       ANOVA
2  PC4 Metastasis 0.005708825 0.2740236  3.1889416      T-test
3  PC1 Recurrence 0.013344932 0.4270378  2.7813417      T-test
4  PC7 Recurrence 0.024533427 0.4793635 -2.4822641      T-test
5  PC4      Stage 0.024966851 0.4793635  4.7672183       ANOVA
6  PC9        Age 0.030540961 0.4886554  0.5101478 Correlation</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Perform analysis
proNormalRes &lt;- doSOA(proNormal, meta_var = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
                      num_PCs = 16, num_PCfeats = 30)
datMat &lt;- proNormalRes$data
smpAnno &lt;- proNormalRes$smpMetadata
tFeatSigRes &lt;- proNormalRes$tFeatSigRes
tPCASigRes &lt;- proNormalRes$tPCASigRes
pcaRes &lt;- proNormalRes$pcaRes
pcTab &lt;- proNormalRes$pcTab
pcTopFeatTab &lt;- proNormalRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes &lt;- dplyr::mutate(tFeatSigRes, Gene = plyr::mapvalues(Var1,
                                                                 from = featInfo$PG.Groups,
                                                                 to = featInfo$PG.Genes,
                                                                 warn_missing = F)) %&gt;%
  dplyr::select(-Test)
# Manually shorten names
tFeatSigRes$Var1[413] &lt;- &#39;P0DP23;P0DP24&#39;
tFeatSigRes$Gene[413] &lt;- &#39;CALM1;CALM2&#39;
tFeatSigRes</code></pre>
<pre><code>             Var1       Var2         pVal   pValAdj      Stat             Gene
1          Q9UKK3 Recurrence 2.958086e-05 0.2275656 -5.754439            PARP4
2          Q17RN3 Recurrence 6.546680e-05 0.2518180  5.346930           FAM98C
3          Q96CV9 Recurrence 1.564367e-04 0.2553000  4.911732             OPTN
4          O96008 Recurrence 1.747290e-04 0.2553000 -4.857249           TOMM40
5          Q8NI37 Recurrence 2.043799e-04 0.2553000 -4.780292            PPTC7
6          Q6NXT1 Recurrence 2.119962e-04 0.2553000  4.953667          ANKRD54
7          O15382 Recurrence 2.640334e-04 0.2553000 -4.655196            BCAT2
8          Q6IPR1 Recurrence 3.242456e-04 0.2553000 -4.634209           ETFRF1
9          Q8N9L9 Recurrence 3.287668e-04 0.2553000  5.129108            ACOT4
10         Q96C36 Recurrence 3.318601e-04 0.2553000 -4.544135            PYCR2
11         Q9NSC5 Recurrence 3.952296e-04 0.2718684  4.459616           HOMER3
12         A6NKN8 Recurrence 4.348654e-04 0.2718684  4.572228           PCP4L1
13         P35914 Recurrence 5.330873e-04 0.2718684 -4.315567            HMGCL
14         Q92851 Recurrence 5.439219e-04 0.2718684 -4.305909           CASP10
15         A4D1E1 Recurrence 5.957514e-04 0.2718684  4.407846          ZNF804B
16         Q9Y277 Recurrence 6.231952e-04 0.2718684 -4.240685            VDAC3
17         Q9NPH3 Recurrence 7.172775e-04 0.2718684  4.173428           IL1RAP
18         Q14515 Recurrence 7.750714e-04 0.2718684  4.136419          SPARCL1
19         Q9Y512 Recurrence 7.791566e-04 0.2718684 -4.133910           SAMM50
20         Q96E40 Recurrence 8.249492e-04 0.2718684  4.324525           SPACA9
21         P61326 Recurrence 9.727650e-04 0.2718684  4.235673            MAGOH
22         P13647 Recurrence 1.003520e-03 0.2718684  4.013324             KRT5
23         P78346 Recurrence 1.027989e-03 0.2718684 -4.001864            RPP30
24         O15195 Recurrence 1.086690e-03 0.2718684  4.176222             VILL
25         Q96EQ0 Recurrence 1.091549e-03 0.2718684  3.973336             SGTB
26         O60825 Recurrence 1.120444e-03 0.2718684 -3.960917           PFKFB2
27         P62979 Recurrence 1.155060e-03 0.2718684  3.946459           RPS27A
28         P43897 Recurrence 1.218928e-03 0.2718684 -3.920895             TSFM
29         P21953 Recurrence 1.224794e-03 0.2718684 -3.972999           BCKDHB
30         Q9ULP9 Recurrence 1.236563e-03 0.2718684 -3.914075          TBC1D24
31         Q9NVZ3 Recurrence 1.248885e-03 0.2718684 -3.909367           NECAP2
32         Q15006 Recurrence 1.302689e-03 0.2718684 -3.889345             EMC2
33         Q13790 Recurrence 1.325481e-03 0.2718684  3.934210             APOF
34         P51649 Recurrence 1.347849e-03 0.2718684 -3.873173          ALDH5A1
35         P98194 Recurrence 1.410269e-03 0.2718684 -3.851697           ATP2C1
36         Q14376 Recurrence 1.411976e-03 0.2718684 -3.851124             GALE
37         P52789 Recurrence 1.498808e-03 0.2718684 -3.822823              HK2
38         Q99805 Recurrence 1.577312e-03 0.2718684 -3.798622           TM9SF2
39         Q56VL3 Recurrence 1.588911e-03 0.2718684 -3.795149           OCIAD2
40         Q5VWJ9 Recurrence 1.631082e-03 0.2718684 -3.782735            SNX30
41         Q9Y5U9 Recurrence 1.686114e-03 0.2718684 -3.767012          IER3IP1
42         Q16540 Recurrence 1.722361e-03 0.2718684 -3.805891           MRPL23
43         Q5M775 Recurrence 1.730889e-03 0.2718684  3.754595           SPECC1
44         Q05823 Recurrence 1.731694e-03 0.2718684 -3.754375           RNASEL
45         Q8TDB4 Recurrence 1.753449e-03 0.2718684  4.226511            MGARP
46         P84243 Recurrence 1.773134e-03 0.2718684  3.791681            H3-3B
47         P02533 Recurrence 1.800642e-03 0.2718684  3.735880            KRT14
48         P09668 Recurrence 1.813114e-03 0.2718684 -3.732611             CTSH
49         Q96P48 Recurrence 1.816190e-03 0.2718684 -3.731808            ARAP1
50         Q9BXB4 Recurrence 1.824282e-03 0.2718684 -3.777776          OSBPL11
51         O60341 Recurrence 1.861296e-03 0.2718684 -3.720190            KDM1A
52         Q8IY95 Recurrence 1.864609e-03 0.2718684 -3.719347          TMEM192
53         Q9NVJ2 Recurrence 1.873004e-03 0.2718684 -3.717220            ARL8B
54         Q9H9P8 Recurrence 2.023438e-03 0.2818414 -3.680639           L2HGDH
55         P21796 Recurrence 2.079407e-03 0.2818414 -3.667721            VDAC1
56         Q8N4T8 Recurrence 2.088632e-03 0.2818414 -3.665625             CBR4
57         P52849 Recurrence 2.113866e-03 0.2818414 -3.705792            NDST2
58         Q96BM9 Recurrence 2.147324e-03 0.2818414 -3.652505            ARL8A
59         P23919 Recurrence 2.203733e-03 0.2818414 -3.640230            DTYMK
60         P22735 Recurrence 2.223248e-03 0.2818414  3.733857             TGM1
61         Q96M27 Recurrence 2.324202e-03 0.2818414 -3.615035            PRRC1
62         P11310 Recurrence 2.325939e-03 0.2818414 -3.614682            ACADM
63         O15258 Recurrence 2.337642e-03 0.2818414 -3.612306             RER1
64         P61956 Recurrence 2.344710e-03 0.2818414  3.610877            SUMO2
65         P24821 Recurrence 2.389669e-03 0.2825078  3.601887              TNC
66         Q02413 Recurrence 2.423699e-03 0.2825078  3.595194             DSG1
67         Q96F07 Recurrence 2.492505e-03 0.2861916 -3.581943           CYFIP2
68         Q5JPH6 Recurrence 2.564787e-03 0.2901604 -3.568411            EARS2
69         P09917 Recurrence 2.625794e-03 0.2927570 -3.557284            ALOX5
70         Q0VDG4 Recurrence 2.705952e-03 0.2973841 -3.543049            SCRN3
71         O00142 Recurrence 2.770294e-03 0.2980094 -3.531925              TK2
72         O94808 Recurrence 2.812619e-03 0.2980094 -3.524747            GFPT2
73         Q5T200 Recurrence 2.833793e-03 0.2980094  3.521196           ZC3H13
74         O15344 Recurrence 2.869116e-03 0.2980094  3.515332             MID1
75         A0AV96 Recurrence 2.918184e-03 0.2980094 -3.507303            RBM47
76         Q7Z794 Recurrence 2.976669e-03 0.2980094  3.497908            KRT77
77         Q8TCS8 Recurrence 3.024862e-03 0.2980094 -3.490304            PNPT1
78         Q86YM7 Recurrence 3.054186e-03 0.2980094  3.485736           HOMER1
79         O00116 Recurrence 3.060281e-03 0.2980094 -3.484792             AGPS
80         Q96E11 Recurrence 3.143177e-03 0.3002753 -3.472135             MRRF
81         O14578 Recurrence 3.161615e-03 0.3002753  3.556223              CIT
82         Q9UG63 Recurrence 3.203968e-03 0.3005869 -3.463064            ABCF2
83         P02538 Recurrence 3.374363e-03 0.3127587  3.438521            KRT6A
84         P99999 Recurrence 3.438881e-03 0.3149441 -3.429549             CYCS
85         P49642 Recurrence 3.496158e-03 0.3164229 -4.316263            PRIM1
86         Q9NP74 Recurrence 3.589074e-03 0.3210551  3.409295            PALMD
87         Q96QE2 Recurrence 3.719249e-03 0.3287763 -3.392411          SLC2A13
88         Q9UKU7 Recurrence 3.760862e-03 0.3287763 -3.387137            ACAD8
89         Q7Z3J2 Recurrence 3.861219e-03 0.3337568 -3.374654           VPS35L
90         P07910 Recurrence 3.931509e-03 0.3360567  3.366101           HNRNPC
91         Q9Y678 Recurrence 3.983174e-03 0.3367314 -3.359910            COPG1
92         Q8NB16 Recurrence 4.117218e-03 0.3389594 -3.344213             MLKL
93         Q9H936 Recurrence 4.251453e-03 0.3389594 -3.328994         SLC25A22
94         P53602 Recurrence 4.278764e-03 0.3389594 -3.325956              MVD
95         P48681 Recurrence 4.299495e-03 0.3389594  3.323663              NES
96         Q8TDQ1 Recurrence 4.303105e-03 0.3389594 -3.401142          CD300LF
97         P43405 Recurrence 4.305928e-03 0.3389594 -3.322953              SYK
98         P50502 Recurrence 4.391654e-03 0.3389594  3.313599             ST13
99         Q6KB66 Recurrence 4.412285e-03 0.3389594  3.311375            KRT80
100        P10620 Recurrence 4.670225e-03 0.3389594 -3.284404            MGST1
101        Q8IZQ1 Recurrence 4.682331e-03 0.3389594  3.283175            WDFY3
102        Q5S007 Recurrence 4.780701e-03 0.3389594 -3.273301            LRRK2
103        Q96LZ7 Recurrence 4.867427e-03 0.3389594 -3.264761            RMDN2
104        O75832 Recurrence 4.899345e-03 0.3389594 -3.261656           PSMD10
105        Q969E8 Recurrence 4.955213e-03 0.3389594  3.256268             TSR2
106        Q96EX1 Recurrence 4.979025e-03 0.3389594 -3.288089           SMIM12
107        P07205 Recurrence 5.010083e-03 0.3389594 -3.251036             PGK2
108        Q96IQ7 Recurrence 5.027168e-03 0.3389594  3.249418            VSIG2
109        Q9H4M9 Recurrence 5.043300e-03 0.3389594  3.247895             EHD1
110        P07951 Recurrence 5.053848e-03 0.3389594  3.246902             TPM2
111        O15231 Recurrence 5.074439e-03 0.3389594  3.244970           ZNF185
112        O60828 Recurrence 5.120884e-03 0.3389594  3.274385            PQBP1
113        P51149 Recurrence 5.180644e-03 0.3389594 -3.235124            RAB7A
114        Q96RQ3 Recurrence 5.189382e-03 0.3389594 -3.234323            MCCC1
115        P67936 Recurrence 5.198563e-03 0.3389594  3.233482             TPM4
116        P54252 Recurrence 5.239549e-03 0.3389594  3.229748            ATXN3
117        Q8N2K0 Recurrence 5.261938e-03 0.3389594 -3.227721           ABHD12
118        Q5XXA6 Recurrence 5.284229e-03 0.3389594  3.225711             ANO1
119        Q8N131 Recurrence 5.301511e-03 0.3389594  3.224158          TMEM123
120        O15372 Recurrence 5.360904e-03 0.3389594 -3.218860            EIF3H
121        Q8NBQ5 Recurrence 5.368086e-03 0.3389594 -3.218223         HSD17B11
122        P15924 Recurrence 5.375412e-03 0.3389594  3.217575              DSP
123        P49006 Recurrence 5.445820e-03 0.3406073  3.211385         MARCKSL1
124        P33121 Recurrence 5.678397e-03 0.3411207 -3.191485            ACSL1
125        Q9H7X7 Recurrence 5.743326e-03 0.3411207  3.186073            IFT22
126        P17405 Recurrence 5.759955e-03 0.3411207 -3.254582            SMPD1
127        Q6IAN0 Recurrence 5.787063e-03 0.3411207 -3.182462           DHRS7B
128        Q08554 Recurrence 5.838529e-03 0.3411207  3.247772             DSC1
129        Q96EH3 Recurrence 5.841776e-03 0.3411207 -3.210100           MALSU1
130        Q6P4A7 Recurrence 5.842680e-03 0.3411207 -3.291483            SFXN4
131        P54886 Recurrence 5.878902e-03 0.3411207 -3.174965         ALDH18A1
132        Q9Y6Y8 Recurrence 5.904474e-03 0.3411207 -3.172898          SEC23IP
133        Q9UNZ2 Recurrence 6.007024e-03 0.3411207  3.164697           NSFL1C
134        Q99816 Recurrence 6.119595e-03 0.3411207  3.155851           TSG101
135        O75436 Recurrence 6.155841e-03 0.3411207 -3.153037           VPS26A
136        P04843 Recurrence 6.187509e-03 0.3411207 -3.150592             RPN1
137        Q14103 Recurrence 6.283513e-03 0.3411207  3.143254           HNRNPD
138        P39656 Recurrence 6.322576e-03 0.3411207 -3.140299            DDOST
139        Q7Z4W1 Recurrence 6.323517e-03 0.3411207 -3.140228             DCXR
140        P09493 Recurrence 6.368182e-03 0.3411207  3.136873             TPM1
141        P26038 Recurrence 6.413090e-03 0.3411207  3.133522              MSN
142        Q9H2M9 Recurrence 6.414738e-03 0.3411207 -3.133400         RAB3GAP2
143        Q15424 Recurrence 6.441395e-03 0.3411207  3.131422             SAFB
144        O95678 Recurrence 6.456128e-03 0.3411207  3.130333            KRT75
145        Q96RQ1 Recurrence 6.477512e-03 0.3411207 -3.128756           ERGIC2
146        Q9Y5G0 Recurrence 6.512130e-03 0.3411207  3.157024          PCDHGB5
147        Q8N3P4 Recurrence 6.554752e-03 0.3411207 -3.123102             VPS8
148        O95436 Recurrence 6.562572e-03 0.3411207 -3.122533          SLC34A2
149        Q9UKU9 Recurrence 6.619587e-03 0.3417751  3.118406          ANGPTL2
150        Q9H8L6 Recurrence 6.743836e-03 0.3453951  3.109533            MMRN2
151        Q5MIZ7 Recurrence 6.793536e-03 0.3453951  3.106029          PPP4R3B
152        P45880 Recurrence 6.824392e-03 0.3453951 -3.103866            VDAC2
153        Q86SQ0 Recurrence 6.949367e-03 0.3494214  3.095202           PHLDB2
154        O43291 Recurrence 7.023334e-03 0.3507045  3.090146           SPINT2
155        Q9UIG8 Recurrence 7.083272e-03 0.3507045  3.086087          SLCO3A1
156        Q13825 Recurrence 7.137984e-03 0.3507045 -3.082412              AUH
157        Q9NPF4 Recurrence 7.157234e-03 0.3507045 -3.081125            OSGEP
158        Q76N32 Recurrence 7.309436e-03 0.3547835  3.100522            CEP68
159        Q13464 Recurrence 7.332715e-03 0.3547835  3.069551            ROCK1
160        P32321 Recurrence 7.429214e-03 0.3564941 -3.063301             DCTD
161        P01127 Recurrence 7.519741e-03 0.3564941  3.208023            PDGFB
162        Q6NY19 Recurrence 7.533332e-03 0.3564941  3.056647            KANK3
163        Q6DT37 Recurrence 7.567737e-03 0.3564941  3.054467         CDC42BPG
164        Q5T7V8 Recurrence 7.728432e-03 0.3564941  3.530316            GORAB
165        Q14435 Recurrence 7.758714e-03 0.3564941 -3.042545           GALNT3
166        Q9Y2R9 Recurrence 7.777377e-03 0.3564941 -3.041396            MRPS7
167        Q9H7D7 Recurrence 7.823664e-03 0.3564941 -3.038556            WDR26
168        Q14137 Recurrence 7.831081e-03 0.3564941 -3.038103             BOP1
169        Q96BJ3 Recurrence 7.861745e-03 0.3564941  3.036232             AIDA
170        O94923 Recurrence 7.929801e-03 0.3564941 -3.093777             GLCE
171        P06753 Recurrence 7.951332e-03 0.3564941  3.030809             TPM3
172        Q9Y2S7 Recurrence 7.979683e-03 0.3564941 -3.029105          POLDIP2
173        Q8IVM0 Recurrence 8.060676e-03 0.3564941  3.024269           CCDC50
174        P41208 Recurrence 8.108263e-03 0.3564941  3.021450            CETN2
175        P05388 Recurrence 8.133146e-03 0.3564941 -3.019983            RPLP0
176        Q92506 Recurrence 8.155851e-03 0.3564941 -3.018648          HSD17B8
177        P55263 Recurrence 8.253265e-03 0.3587140 -3.012960              ADK
178        Q9BZL4 Recurrence 8.380703e-03 0.3593390  3.005617         PPP1R12C
179        Q9NPG4 Recurrence 8.417623e-03 0.3593390  3.063709           PCDH12
180        Q9HAN9 Recurrence 8.472460e-03 0.3593390 -3.000398           NMNAT1
181        P19878 Recurrence 8.489861e-03 0.3593390 -2.999415             NCF2
182        P30622 Recurrence 8.528295e-03 0.3593390  2.997249            CLIP1
183        Q7L2H7 Recurrence 8.561893e-03 0.3593390 -2.995364            EIF3M
184        P49327 Recurrence 8.651562e-03 0.3593390 -2.990368             FASN
185        P17302 Recurrence 8.695581e-03 0.3593390  2.987934             GJA1
186        O96005 Recurrence 8.781258e-03 0.3593390 -2.983230           CLPTM1
187        P29374 Recurrence 8.856480e-03 0.3593390  2.979138           ARID4A
188        Q8TF09 Recurrence 9.041818e-03 0.3593390  2.969197          DYNLRB2
189        Q9P2N2 Recurrence 9.063214e-03 0.3593390  3.026466         ARHGAP28
190        Q3KQU3 Recurrence 9.141061e-03 0.3593390  2.963956           MAP7D1
191        Q96CM8 Recurrence 9.156474e-03 0.3593390 -2.963147            ACSF2
192        Q92600 Recurrence 9.211306e-03 0.3593390 -2.960279            CNOT9
193        O75781 Recurrence 9.316383e-03 0.3593390  2.954831             PALM
194        Q16643 Recurrence 9.341709e-03 0.3593390  2.953526             DBN1
195        P07741 Recurrence 9.371859e-03 0.3593390 -2.951978             APRT
196        Q03426 Recurrence 9.387540e-03 0.3593390 -2.951175              MVK
197        Q9UBY9 Recurrence 9.417219e-03 0.3593390  2.976245            HSPB7
198        Q8IU68 Recurrence 9.424969e-03 0.3593390  2.975841             TMC8
199        Q13563 Recurrence 9.503837e-03 0.3593390  2.971744             PKD2
200        Q8WZA9 Recurrence 9.513717e-03 0.3593390  2.944758             IRGQ
201        Q96IZ0 Recurrence 9.519961e-03 0.3593390  2.944443             PAWR
202        Q6P4E1 Recurrence 9.562722e-03 0.3593390  2.942289            GOLM2
203        Q16825 Recurrence 9.664302e-03 0.3593390  2.937208           PTPN21
204        P08590 Recurrence 9.712085e-03 0.3593390  2.934836             MYL3
205        O14683 Recurrence 9.756063e-03 0.3593390 -2.932664          TP53I11
206        Q969Q5 Recurrence 9.764794e-03 0.3593390 -2.958423            RAB24
207        Q8IV50 Recurrence 9.795559e-03 0.3593390  2.930720           LYSMD2
208        Q9BTZ2 Recurrence 9.829415e-03 0.3593390 -2.929061            DHRS4
209        Q9NUJ1 Recurrence 9.930953e-03 0.3593390 -2.924116           ABHD10
210        Q9NPJ3 Recurrence 9.943415e-03 0.3593390 -2.923512           ACOT13
211        O75602 Recurrence 9.993522e-03 0.3593390  2.977170            SPAG6
212        Q5JTZ9 Recurrence 9.999904e-03 0.3593390 -2.920786            AARS2
213        Q9NR45 Recurrence 1.001667e-02 0.3593390 -2.919980             NANS
214        Q14166 Recurrence 1.005835e-02 0.3593390 -2.917981           TTLL12
215        Q96GS4 Recurrence 1.021188e-02 0.3593390  2.910688           BORCS6
216        Q99653 Recurrence 1.025098e-02 0.3593390  2.908848             CHP1
217        P16278 Recurrence 1.036243e-02 0.3593390 -2.903639             GLB1
218        Q86VS8 Recurrence 1.039983e-02 0.3593390  2.901904            HOOK3
219        Q8IWU2 Recurrence 1.052828e-02 0.3593390  2.921365            LMTK2
220        P61218 Recurrence 1.055491e-02 0.3593390  2.894771           POLR2F
221        Q9NYB0 Recurrence 1.055858e-02 0.3593390  2.894604          TERF2IP
222        Q9H2U2 Recurrence 1.057951e-02 0.3593390 -2.893649             PPA2
223        P02647 Recurrence 1.059661e-02 0.3593390  2.892871            APOA1
224        Q96C11 Recurrence 1.060336e-02 0.3593390 -3.023014             FGGY
225        Q9NNW7 Recurrence 1.067125e-02 0.3593390 -2.889487           TXNRD2
226        Q96EP5 Recurrence 1.067975e-02 0.3593390 -2.889103           DAZAP1
227        P42702 Recurrence 1.069131e-02 0.3593390  2.888582             LIFR
228        Q8WW59 Recurrence 1.069516e-02 0.3593390 -2.888408           SPRYD4
229        Q96DT6 Recurrence 1.073776e-02 0.3593390 -3.016236            ATG4C
230        Q8IY34 Recurrence 1.074327e-02 0.3593390 -2.886244          SLC15A3
231    A0A075B6H9 Recurrence 1.082836e-02 0.3606172  2.882439         IGLV4-69
232        O43837 Recurrence 1.092351e-02 0.3622180 -2.878219            IDH3B
233        O75629 Recurrence 1.111731e-02 0.3670621 -2.869733            CREG1
234        Q9ULX6 Recurrence 1.136790e-02 0.3730912 -2.858971           AKAP8L
235        Q8IY63 Recurrence 1.144945e-02 0.3730912  2.855519           AMOTL1
236        O14656 Recurrence 1.161522e-02 0.3730912 -2.848573            TOR1A
237        Q05682 Recurrence 1.163015e-02 0.3730912  2.847952            CALD1
238        Q92947 Recurrence 1.166432e-02 0.3730912 -2.846534             GCDH
239        Q9UJC5 Recurrence 1.168014e-02 0.3730912 -2.845879         SH3BGRL2
240        Q9NZK5 Recurrence 1.168106e-02 0.3730912 -2.845841             ADA2
241        Q93050 Recurrence 1.173327e-02 0.3730912 -2.843686         ATP6V0A1
242        P30084 Recurrence 1.173639e-02 0.3730912 -2.843557            ECHS1
243        O15254 Recurrence 1.188427e-02 0.3762199 -2.837503            ACOX3
244        P10909 Recurrence 1.193262e-02 0.3762199  2.835539              CLU
245        Q9NTJ4 Recurrence 1.214597e-02 0.3774072 -2.826965           MAN2C1
246        Q0VF96 Recurrence 1.214655e-02 0.3774072  2.826942            CGNL1
247        P63010 Recurrence 1.244075e-02 0.3774072 -2.815357            AP2B1
248        O00308 Recurrence 1.253684e-02 0.3774072  2.862503             WWP2
249        P04150 Recurrence 1.267327e-02 0.3774072  2.806386            NR3C1
250        Q8NCJ5 Recurrence 1.284846e-02 0.3774072  2.799732           SPRYD3
251        Q7Z3B1 Recurrence 1.289155e-02 0.3774072  2.798109            NEGR1
252        Q15404 Recurrence 1.290343e-02 0.3774072  2.797663             RSU1
253        Q6PI78 Recurrence 1.298271e-02 0.3774072  2.794693           TMEM65
254        Q9UQN3 Recurrence 1.304129e-02 0.3774072  2.792509           CHMP2B
255        P34896 Recurrence 1.304427e-02 0.3774072 -2.792398            SHMT1
256        Q9Y679 Recurrence 1.311173e-02 0.3774072 -2.789896             AUP1
257        Q2TB90 Recurrence 1.314139e-02 0.3774072 -2.811859            HKDC1
258        Q8N4H5 Recurrence 1.315017e-02 0.3774072 -2.869720            TOMM5
259        Q8TB24 Recurrence 1.321109e-02 0.3774072 -2.786233             RIN3
260        O14974 Recurrence 1.322010e-02 0.3774072  2.785903         PPP1R12A
261        Q07617 Recurrence 1.325774e-02 0.3774072  2.834155            SPAG1
262        P34741 Recurrence 1.337269e-02 0.3774072  2.780333             SDC2
263        P48444 Recurrence 1.338967e-02 0.3774072 -2.779717            ARCN1
264        Q7Z406 Recurrence 1.344351e-02 0.3774072  2.777769            MYH14
265        Q7L5L3 Recurrence 1.349061e-02 0.3774072  2.776071            GDPD3
266        Q16822 Recurrence 1.354793e-02 0.3774072 -2.774012             PCK2
267        Q96C90 Recurrence 1.361171e-02 0.3774072  2.771731         PPP1R14B
268        Q01804 Recurrence 1.364330e-02 0.3774072  2.770605            OTUD4
269        Q7Z4V5 Recurrence 1.368860e-02 0.3774072  2.768995           HDGFL2
270        Q9NWB7 Recurrence 1.368964e-02 0.3774072  2.768958            IFT57
271        P14923 Recurrence 1.369454e-02 0.3774072  2.768784              JUP
272        Q10570 Recurrence 1.370419e-02 0.3774072 -2.768442            CPSF1
273        Q9Y3B7 Recurrence 1.379317e-02 0.3774072 -2.765298           MRPL11
274        P23588 Recurrence 1.381349e-02 0.3774072  2.764582            EIF4B
275        Q99983 Recurrence 1.383529e-02 0.3774072  3.138559              OMD
276        Q9P0K7 Recurrence 1.390279e-02 0.3774072  2.761451            RAI14
277        Q99714 Recurrence 1.391332e-02 0.3774072 -2.761082         HSD17B10
278        P35606 Recurrence 1.391336e-02 0.3774072 -2.761081            COPB2
279        O43765 Recurrence 1.395873e-02 0.3774072  2.759498             SGTA
280        P08729 Recurrence 1.397279e-02 0.3774072  2.759009             KRT7
281        Q03001 Recurrence 1.408321e-02 0.3774072  2.755182              DST
282        P06727 Recurrence 1.410766e-02 0.3774072  2.754338            APOA4
283        Q6IA17 Recurrence 1.414891e-02 0.3774072  2.752919           SIGIRR
284        O60711 Recurrence 1.429327e-02 0.3774072 -2.747981             LPXN
285        P01700 Recurrence 1.434948e-02 0.3774072  2.746071         IGLV1-47
286        Q7L5D6 Recurrence 1.435937e-02 0.3774072 -2.745736             GET4
287        Q92828 Recurrence 1.440571e-02 0.3774072 -2.744168           CORO2A
288        Q9H6F5 Recurrence 1.444733e-02 0.3774072  3.109916           CCDC86
289        Q9BUK0 Recurrence 1.459163e-02 0.3774072 -2.737926           CHCHD7
290        Q12913 Recurrence 1.463188e-02 0.3774072  2.736585            PTPRJ
291        Q8N766 Recurrence 1.468618e-02 0.3774072 -2.734781             EMC1
292        Q8IXS6 Recurrence 1.470223e-02 0.3774072  2.734249            PALM2
293        Q9Y4D1 Recurrence 1.472574e-02 0.3774072  2.733471            DAAM1
294        Q86W92 Recurrence 1.474823e-02 0.3774072  2.732728          PPFIBP1
295        P23786 Recurrence 1.478314e-02 0.3774072 -2.731576             CPT2
296        P22695 Recurrence 1.479641e-02 0.3774072 -2.731139           UQCRC2
297        P15170 Recurrence 1.483126e-02 0.3774072 -2.729993            GSPT1
298        Q9NQG5 Recurrence 1.486287e-02 0.3774072  2.728956           RPRD1B
299        Q13287 Recurrence 1.497293e-02 0.3774072 -2.725361              NMI
300        Q9BRZ2 Recurrence 1.502175e-02 0.3774072  2.723775           TRIM56
301        Q04695 Recurrence 1.504071e-02 0.3774072  2.723160            KRT17
302        Q96P11 Recurrence 1.508722e-02 0.3774072 -2.721655            NSUN5
303        Q92870 Recurrence 1.519222e-02 0.3774072  2.739882            APBB2
304        P49321 Recurrence 1.519943e-02 0.3774072  2.718042             NASP
305        Q9H425 Recurrence 1.521161e-02 0.3774072  2.717652         C1orf198
306        Q9H7Z7 Recurrence 1.521236e-02 0.3774072 -2.717628           PTGES2
307        Q5RI15 Recurrence 1.525360e-02 0.3774072 -2.716308            COX20
308        O94925 Recurrence 1.529613e-02 0.3774072 -2.714950              GLS
309        Q9H2D6 Recurrence 1.537589e-02 0.3774072  2.712413           TRIOBP
310        O60936 Recurrence 1.542682e-02 0.3774072  2.710800             NOL3
311        O75439 Recurrence 1.548127e-02 0.3774072 -2.709081            PMPCB
312        Q86W42 Recurrence 1.558948e-02 0.3774072 -2.705682            THOC6
313        Q9NXN4 Recurrence 1.559247e-02 0.3774072 -2.705588            GDAP2
314        Q9H2J4 Recurrence 1.560093e-02 0.3774072 -2.705324            PDCL3
315        Q8WVV4 Recurrence 1.576966e-02 0.3774072  2.700073            POF1B
316        O15173 Recurrence 1.579139e-02 0.3774072  2.699400           PGRMC2
317        Q9BWH6 Recurrence 1.579985e-02 0.3774072 -2.808276            RPAP1
318        P62736 Recurrence 1.580450e-02 0.3774072  2.698995            ACTA2
319        Q8ND56 Recurrence 1.582712e-02 0.3774072  2.698296           LSM14A
320        P18206 Recurrence 1.591190e-02 0.3774072  2.695688              VCL
321        Q8N5G0 Recurrence 1.592595e-02 0.3774072 -2.695257           SMIM20
322        P26885 Recurrence 1.597260e-02 0.3774072 -2.693828            FKBP2
323        P04632 Recurrence 1.599580e-02 0.3774072  2.693119           CAPNS1
324        Q99547 Recurrence 1.604859e-02 0.3774072  2.691509         MPHOSPH6
325        P26022 Recurrence 1.612005e-02 0.3774072  2.689338             PTX3
326        O14874 Recurrence 1.613500e-02 0.3774072 -2.688885            BCKDK
327        H7BZ55 Recurrence 1.619494e-02 0.3774072  2.687073           CROCC2
328        P61952 Recurrence 1.626241e-02 0.3774072  2.685041            GNG11
329        Q5T1C6 Recurrence 1.630487e-02 0.3774072 -2.683766            THEM4
330        P08727 Recurrence 1.630923e-02 0.3774072  2.683635            KRT19
331        Q9UHX1 Recurrence 1.639886e-02 0.3774072 -2.680956            PUF60
332        Q9Y3E0 Recurrence 1.643015e-02 0.3774072 -2.680024           GOLT1B
333        P61966 Recurrence 1.643395e-02 0.3774072 -2.679911            AP1S1
334        Q9H0V9 Recurrence 1.647336e-02 0.3774072 -2.678739           LMAN2L
335        O60941 Recurrence 1.655091e-02 0.3774072  2.676442             DTNB
336        Q9UKX5 Recurrence 1.658339e-02 0.3774072  2.675483           ITGA11
337        Q96GC9 Recurrence 1.659129e-02 0.3774072 -2.822568             VMP1
338        Q8N999 Recurrence 1.667642e-02 0.3774072  2.672746         C12orf29
339        P09960 Recurrence 1.676074e-02 0.3774072 -2.670278            LTA4H
340        Q9NX74 Recurrence 1.679874e-02 0.3774072  2.741807             DUS2
341        O00264 Recurrence 1.692564e-02 0.3774072  2.665485           PGRMC1
342        P36551 Recurrence 1.693072e-02 0.3774072 -2.665338             CPOX
343        Q8IWE5 Recurrence 1.695749e-02 0.3774072 -2.736882          PLEKHM2
344        P38571 Recurrence 1.697996e-02 0.3774072 -2.663917             LIPA
345        Q15365 Recurrence 1.703042e-02 0.3774072 -2.662464            PCBP1
346        O75394 Recurrence 1.703081e-02 0.3774072 -2.706726           MRPL33
347        P08842 Recurrence 1.706796e-02 0.3774072  2.661385              STS
348        Q6YN16 Recurrence 1.716858e-02 0.3774072 -2.658506            HSDL2
349        Q5D862 Recurrence 1.719321e-02 0.3774072  2.701881             FLG2
350        Q969E4 Recurrence 1.731876e-02 0.3774072  2.654239           TCEAL3
351        P26440 Recurrence 1.733369e-02 0.3774072 -2.653817              IVD
352        Q99797 Recurrence 1.735799e-02 0.3774072 -2.653130            MIPEP
353        Q9Y6A9 Recurrence 1.743311e-02 0.3774072 -2.651014            SPCS1
354        Q15027 Recurrence 1.746536e-02 0.3774072  2.650108            ACAP1
355        P05165 Recurrence 1.746761e-02 0.3774072 -2.650044             PCCA
356        O95273 Recurrence 1.746977e-02 0.3774072  2.721292          CCNDBP1
357        Q8TBX8 Recurrence 1.757306e-02 0.3774072 -2.647094          PIP4K2C
358        Q9NSE4 Recurrence 1.760529e-02 0.3774072 -2.646195            IARS2
359        Q9H5N1 Recurrence 1.769397e-02 0.3774072  2.643731           RABEP2
360        Q9Y3A6 Recurrence 1.774861e-02 0.3774072 -2.642218            TMED5
361        Q86T03 Recurrence 1.776710e-02 0.3774072 -2.641708           PIP4P1
362        Q9NWW5 Recurrence 1.778928e-02 0.3774072 -2.661181             CLN6
363    A0A0G2JMI3 Recurrence 1.796748e-02 0.3774072  2.679375       IGHV1-69-2
364        Q66LE6 Recurrence 1.819814e-02 0.3774072 -2.629943          PPP2R2D
365        Q02252 Recurrence 1.825902e-02 0.3774072 -2.628303          ALDH6A1
366        Q8N4A0 Recurrence 1.840855e-02 0.3774072  2.764457           GALNT4
367        P63208 Recurrence 1.843301e-02 0.3774072  2.623644             SKP1
368        Q99417 Recurrence 1.843879e-02 0.3774072  2.623490            MYCBP
369        O00115 Recurrence 1.849461e-02 0.3774072 -2.622005           DNASE2
370        Q15532 Recurrence 1.851303e-02 0.3774072  2.664076             SS18
371        O75190 Recurrence 1.852754e-02 0.3774072  2.621130           DNAJB6
372        O60830 Recurrence 1.872233e-02 0.3774072 -2.635594          TIMM17B
373        P09382 Recurrence 1.873641e-02 0.3774072  2.615620           LGALS1
374        Q6NZ67 Recurrence 1.876408e-02 0.3774072  2.614895            MZT2B
375        Q9NRA2 Recurrence 1.876800e-02 0.3774072 -2.614792          SLC17A5
376        Q9H1E3 Recurrence 1.879276e-02 0.3774072  2.614144           NUCKS1
377        P80723 Recurrence 1.893445e-02 0.3774072  2.610450            BASP1
378        O75182 Recurrence 1.895265e-02 0.3774072  2.678564            SIN3B
379        Q9NVD7 Recurrence 1.896180e-02 0.3774072  2.609740            PARVA
380        O96018 Recurrence 1.901603e-02 0.3774072  2.608335            APBA3
381        Q14498 Recurrence 1.902542e-02 0.3774072 -2.608092            RBM39
382        P07711 Recurrence 1.906102e-02 0.3774072 -2.607172             CTSL
383        Q99571 Recurrence 1.912751e-02 0.3774072 -2.605458            P2RX4
384        Q9Y2G9 Recurrence 1.913815e-02 0.3774072 -2.848430            SBNO2
385        Q9C0C2 Recurrence 1.918899e-02 0.3774072  2.603879         TNKS1BP1
386        Q709C8 Recurrence 1.920896e-02 0.3774072 -2.603367           VPS13C
387        P06454 Recurrence 1.923848e-02 0.3774072  2.602611             PTMA
388        Q9BQB6 Recurrence 1.924948e-02 0.3774072 -2.739476           VKORC1
389        Q14151 Recurrence 1.925494e-02 0.3774072  2.602190            SAFB2
390        Q8N5K1 Recurrence 1.927164e-02 0.3774072 -2.601764            CISD2
391        O95347 Recurrence 1.929423e-02 0.3774072  2.620512             SMC2
392        Q9Y2Z4 Recurrence 1.933779e-02 0.3774072 -2.600077            YARS2
393        Q16651 Recurrence 1.938556e-02 0.3774072  2.598862            PRSS8
394        P05141 Recurrence 1.940097e-02 0.3774072 -2.598471          SLC25A5
395        O15226 Recurrence 1.945565e-02 0.3774072  2.664809             NKRF
396        P38606 Recurrence 1.955942e-02 0.3774072 -2.594465          ATP6V1A
397        O60760 Recurrence 1.959381e-02 0.3774072 -2.593600            HPGDS
398        Q86UL3 Recurrence 1.960091e-02 0.3774072 -2.612600            GPAT4
399        O95865 Recurrence 1.968016e-02 0.3774072  2.591433            DDAH2
400        Q09666 Recurrence 1.977816e-02 0.3774072  2.588986            AHNAK
401        P54198 Recurrence 1.984970e-02 0.3774072 -2.654274             HIRA
402        P32019 Recurrence 1.986375e-02 0.3774072  2.586857           INPP5B
403        P15311 Recurrence 1.992183e-02 0.3774072  2.585418              EZR
404        Q13136 Recurrence 1.993618e-02 0.3774072  2.585063           PPFIA1
405        Q13107 Recurrence 1.993634e-02 0.3774072 -2.585059             USP4
406        Q5SWX8 Recurrence 1.999412e-02 0.3774072 -2.583632             ODR4
407        P09132 Recurrence 2.003557e-02 0.3774072 -2.582611            SRP19
408        P23368 Recurrence 2.010998e-02 0.3774072 -2.580783              ME2
409        Q86UP2 Recurrence 2.018174e-02 0.3774072  2.579026             KTN1
410        P56945 Recurrence 2.023792e-02 0.3774072  2.577655            BCAR1
411        A1A4S6 Recurrence 2.029181e-02 0.3774072 -2.576343         ARHGAP10
412        Q9H799 Recurrence 2.032959e-02 0.3774072  2.575425          CPLANE1
413 P0DP23;P0DP24 Recurrence 2.033771e-02 0.3774072  2.575228      CALM1;CALM2
414        P51784 Recurrence 2.037069e-02 0.3774072 -2.574428            USP11
415        Q13137 Recurrence 2.039337e-02 0.3774072  2.573879         CALCOCO2
416        Q01082 Recurrence 2.042014e-02 0.3774072  2.573232           SPTBN1
417        Q00722 Recurrence 2.052922e-02 0.3774072 -2.570602            PLCB2
418        P98170 Recurrence 2.060189e-02 0.3774072 -2.568858             XIAP
419        Q14764 Recurrence 2.062040e-02 0.3774072 -2.568414              MVP
420        P05362 Recurrence 2.098740e-02 0.3774072  2.559700            ICAM1
421        Q07075 Recurrence 2.101330e-02 0.3774072  2.559091            ENPEP
422        O75410 Recurrence 2.106391e-02 0.3774072  2.557902            TACC1
423        Q7LGA3 Recurrence 2.110635e-02 0.3774072 -2.556907           HS2ST1
424        Q12905 Recurrence 2.116053e-02 0.3774072 -2.555640             ILF2
425        Q9NZI7 Recurrence 2.116802e-02 0.3774072 -2.573940             UBP1
426        P22897 Recurrence 2.119491e-02 0.3774072 -2.554837             MRC1
427        Q14165 Recurrence 2.137744e-02 0.3774072 -2.550597             MLEC
428        Q5VWQ8 Recurrence 2.140891e-02 0.3774072  2.549869           DAB2IP
429        Q96CU9 Recurrence 2.143712e-02 0.3774072 -2.588856          FOXRED1
430        P16949 Recurrence 2.150606e-02 0.3774072  2.547630            STMN1
431        Q9H845 Recurrence 2.158895e-02 0.3774072 -2.545726            ACAD9
432        Q92629 Recurrence 2.160527e-02 0.3774072  2.545353             SGCD
433        O60684 Recurrence 2.164487e-02 0.3774072 -2.544446            KPNA6
434        Q9UH65 Recurrence 2.166307e-02 0.3774072  2.544030           SWAP70
435        P02649 Recurrence 2.169746e-02 0.3774072  2.543245             APOE
436        Q8NE86 Recurrence 2.179385e-02 0.3774072 -2.541051              MCU
437        Q12929 Recurrence 2.183796e-02 0.3774072 -2.540050             EPS8
438        P20810 Recurrence 2.188387e-02 0.3774072  2.539011             CAST
439        P62820 Recurrence 2.190136e-02 0.3774072 -2.538615            RAB1A
440        Q6ZS17 Recurrence 2.192041e-02 0.3774072  2.538185           RIPOR1
441        Q70J99 Recurrence 2.196298e-02 0.3774072  2.537224           UNC13D
442        Q6WCQ1 Recurrence 2.197466e-02 0.3774072  2.536960            MPRIP
443        P05166 Recurrence 2.208546e-02 0.3774072 -2.534469             PCCB
444        Q15165 Recurrence 2.216414e-02 0.3774072 -2.532708             PON2
445        O43164 Recurrence 2.238281e-02 0.3774072  2.591049             PJA2
446        P17568 Recurrence 2.246894e-02 0.3774072 -2.525939           NDUFB7
447        O00425 Recurrence 2.256053e-02 0.3774072  2.523922          IGF2BP3
448        Q86UU1 Recurrence 2.260966e-02 0.3774072  2.522843           PHLDB1
449        O60216 Recurrence 2.264610e-02 0.3774072  2.522044            RAD21
450        Q9BYV8 Recurrence 2.266581e-02 0.3774072  2.521613            CEP41
451        Q10713 Recurrence 2.267796e-02 0.3774072 -2.521347            PMPCA
452        O14832 Recurrence 2.274536e-02 0.3774072 -2.519875             PHYH
453        Q9BQ95 Recurrence 2.276621e-02 0.3774072 -2.537243            ECSIT
454        O00291 Recurrence 2.277408e-02 0.3774072  2.519249             HIP1
455        Q86VX2 Recurrence 2.279378e-02 0.3774072 -2.518820           COMMD7
456        Q6JBY9 Recurrence 2.287136e-02 0.3774072  2.517134            RCSD1
457        P20962 Recurrence 2.292521e-02 0.3774072  2.515967             PTMS
458        Q13838 Recurrence 2.298733e-02 0.3774072 -2.514624           DDX39B
459        O43491 Recurrence 2.302440e-02 0.3774072  2.513824          EPB41L2
460        O95219 Recurrence 2.307653e-02 0.3774072 -2.512702             SNX4
461        P36776 Recurrence 2.311457e-02 0.3774072 -2.511884            LONP1
462        Q9BU89 Recurrence 2.314593e-02 0.3774072  2.511211             DOHH
463        Q9UPN3 Recurrence 2.318537e-02 0.3774072  2.510366            MACF1
464        O43865 Recurrence 2.322089e-02 0.3774072 -2.509606           AHCYL1
465        Q9Y3C6 Recurrence 2.332409e-02 0.3774072 -2.507403            PPIL1
466        O43633 Recurrence 2.334138e-02 0.3774072  2.507035           CHMP2A
467        O00273 Recurrence 2.338245e-02 0.3774072  2.506162             DFFA
468        O75607 Recurrence 2.341674e-02 0.3774072 -2.523008             NPM3
469        Q969M3 Recurrence 2.342515e-02 0.3774072 -2.522826            YIPF5
470        Q86YZ3 Recurrence 2.345242e-02 0.3774072  2.504678             HRNR
471        Q9NZT1 Recurrence 2.345557e-02 0.3774072  2.504611           CALML5
472        Q13190 Recurrence 2.348785e-02 0.3774072 -2.503928             STX5
473        Q9H930 Recurrence 2.352806e-02 0.3774072  2.503078           SP140L
474        Q9BPX6 Recurrence 2.353574e-02 0.3774072 -2.502916            MICU1
475        Q9H7B2 Recurrence 2.355333e-02 0.3774072 -2.520067             RPF2
476        Q8WYP3 Recurrence 2.364537e-02 0.3774072  2.518094             RIN2
477        Q15185 Recurrence 2.366261e-02 0.3774072 -2.500244           PTGES3
478        Q10589 Recurrence 2.370127e-02 0.3774072  2.499433             BST2
479        Q6ZNL6 Recurrence 2.376007e-02 0.3774072  2.498201             FGD5
480        Q8TDY2 Recurrence 2.378183e-02 0.3774072  2.497746           RB1CC1
481        Q7L591 Recurrence 2.384134e-02 0.3774072 -2.496503             DOK3
482        P01732 Recurrence 2.389131e-02 0.3774072 -2.512859             CD8A
483        O75935 Recurrence 2.391047e-02 0.3774072  2.495063            DCTN3
484        P84157 Recurrence 2.393899e-02 0.3774072  2.494470            MXRA7
485        P51606 Recurrence 2.400827e-02 0.3774072 -2.493033            RENBP
486        Q92995 Recurrence 2.401696e-02 0.3774072  2.530301            USP13
487        Q9UGQ3 Recurrence 2.403596e-02 0.3774072 -2.492460           SLC2A6
488        Q9NRL2 Recurrence 2.407809e-02 0.3774072 -2.491589            BAZ1A
489        O14737 Recurrence 2.410340e-02 0.3774072  2.491066            PDCD5
490        Q9BX70 Recurrence 2.412899e-02 0.3774072 -2.507848            BTBD2
491        Q8IUZ5 Recurrence 2.415172e-02 0.3774072 -2.490070           PHYKPL
492        O15121 Recurrence 2.418129e-02 0.3774072 -2.489461            DEGS1
493        Q53FV1 Recurrence 2.418585e-02 0.3774072 -2.489367           ORMDL2
494        Q7Z3U7 Recurrence 2.430363e-02 0.3776199 -2.486949             MON2
495        P82673 Recurrence 2.437257e-02 0.3776199 -2.485539           MRPS35
496        Q9UJY4 Recurrence 2.438292e-02 0.3776199 -2.485328             GGA2
497        Q9NWH9 Recurrence 2.444219e-02 0.3776199  2.484119             SLTM
498        O43772 Recurrence 2.444491e-02 0.3776199 -2.484064         SLC25A20
499        P35527 Recurrence 2.456134e-02 0.3786581  2.481698             KRT9
500        O15182 Recurrence 2.470834e-02 0.3801625  2.566410            CETN3
501        Q9NSB2 Recurrence 2.482657e-02 0.3812192  2.563816            KRT84
502        Q9NVM1 Recurrence 2.494935e-02 0.3823414  2.473889            EVA1B
503        P27816 Recurrence 2.512784e-02 0.3829622  2.470336             MAP4
504        O75306 Recurrence 2.518492e-02 0.3829622 -2.469204           NDUFS2
505        Q14802 Recurrence 2.518920e-02 0.3829622  2.469119            FXYD3
506        P07099 Recurrence 2.519048e-02 0.3829622 -2.469094            EPHX1
507        Q9Y2Q0 Recurrence 2.537910e-02 0.3829622 -2.465374           ATP8A1
508        P04259 Recurrence 2.539636e-02 0.3829622  2.465034            KRT6B
509        O96009 Recurrence 2.540046e-02 0.3829622 -2.464954            NAPSA
510        Q9BRR6 Recurrence 2.541513e-02 0.3829622 -2.464666            ADPGK
511        P07949 Recurrence 2.543838e-02 0.3829622  2.523437              RET
512        Q96CN9 Recurrence 2.548767e-02 0.3829622  2.463244             GCC1
513        Q6BCY4 Recurrence 2.554292e-02 0.3830442 -2.462163           CYB5R2
514        P22307 Recurrence 2.565566e-02 0.3832223 -2.459965             SCP2
515        Q13813 Recurrence 2.573599e-02 0.3832223  2.458405           SPTAN1
516        Q9UBC2 Recurrence 2.574727e-02 0.3832223  2.458186          EPS15L1
517        Q9H0L4 Recurrence 2.578707e-02 0.3832223  2.457415           CSTF2T
518        Q15375 Recurrence 2.580387e-02 0.3832223 -2.457090            EPHA7
519        Q96BP3 Recurrence 2.591824e-02 0.3839991 -2.454882            PPWD1
520        Q96EE3 Recurrence 2.600371e-02 0.3839991 -2.453237            SEH1L
521        Q9Y3E5 Recurrence 2.600592e-02 0.3839991 -2.453195            PTRH2
522        Q9NTK5 Recurrence 2.625313e-02 0.3844320 -2.448468             OLA1
523        Q7Z6J4 Recurrence 2.626893e-02 0.3844320 -2.464742             FGD2
524        O15056 Recurrence 2.627710e-02 0.3844320 -2.506252            SYNJ2
525        O95825 Recurrence 2.627997e-02 0.3844320 -2.447957           CRYZL1
526        O00462 Recurrence 2.633255e-02 0.3844320 -2.446958            MANBA
527        O00148 Recurrence 2.660708e-02 0.3844320 -2.441773           DDX39A
528        Q96IX5 Recurrence 2.663820e-02 0.3844320 -2.441188           ATP5MK
529        Q8TER5 Recurrence 2.684747e-02 0.3844320  2.437273         ARHGEF40
530        Q9P0M6 Recurrence 2.687527e-02 0.3844320 -2.436755        MACROH2A2
531        Q9NQ89 Recurrence 2.693593e-02 0.3844320  2.470932          C12orf4
532        Q96EI5 Recurrence 2.694565e-02 0.3844320  2.435447           TCEAL4
533        Q9NQ88 Recurrence 2.698195e-02 0.3844320 -2.434773            TIGAR
534        A1IGU5 Recurrence 2.702385e-02 0.3844320  2.469241         ARHGEF37
535        P05091 Recurrence 2.708394e-02 0.3844320 -2.432884            ALDH2
536        Q8WXE0 Recurrence 2.712799e-02 0.3844320  2.432071          CASKIN2
537        Q9UJS0 Recurrence 2.721329e-02 0.3844320 -2.430499         SLC25A13
538        Q04726 Recurrence 2.727817e-02 0.3844320  2.429306             TLE3
539        Q9Y2P8 Recurrence 2.735523e-02 0.3844320 -2.542556             RCL1
540        O75323 Recurrence 2.737550e-02 0.3844320 -2.427523         NIPSNAP2
541        P48634 Recurrence 2.737857e-02 0.3844320  2.427466           PRRC2A
542        Q5SRE7 Recurrence 2.745517e-02 0.3844320 -2.426067           PHYHD1
543        O96000 Recurrence 2.756380e-02 0.3844320 -2.424088          NDUFB10
544        Q9P107 Recurrence 2.757508e-02 0.3844320 -2.423883             GMIP
545        Q9Y608 Recurrence 2.757559e-02 0.3844320  2.423874          LRRFIP2
546        P62333 Recurrence 2.759373e-02 0.3844320  2.423544            PSMC6
547        Q96BR1 Recurrence 2.759714e-02 0.3844320 -2.423482             SGK3
548        O15228 Recurrence 2.762607e-02 0.3844320 -2.422957            GNPAT
549        O15296 Recurrence 2.763425e-02 0.3844320 -2.422809          ALOX15B
550        P27482 Recurrence 2.773192e-02 0.3844320  2.421041           CALML3
551        Q96CD2 Recurrence 2.778135e-02 0.3844320 -2.420148            PPCDC
552        Q12797 Recurrence 2.779222e-02 0.3844320 -2.419952             ASPH
553        Q96IW7 Recurrence 2.784582e-02 0.3844320 -2.453682           SEC22A
554        Q9UK55 Recurrence 2.790671e-02 0.3844320  2.417891        SERPINA10
555        Q9Y487 Recurrence 2.790762e-02 0.3844320 -2.417874         ATP6V0A2
556        Q9Y2B9 Recurrence 2.791529e-02 0.3844320  2.417736             PKIG
557        P45954 Recurrence 2.794159e-02 0.3844320 -2.417264           ACADSB
558        Q9Y6N5 Recurrence 2.803171e-02 0.3844320 -2.415649             SQOR
559        Q5SY16 Recurrence 2.812323e-02 0.3844320 -2.414014             NOL9
560        O43776 Recurrence 2.820820e-02 0.3844320 -2.412501            NARS1
561        Q53QV2 Recurrence 2.822857e-02 0.3844320  2.428104              LBH
562        Q6ULP2 Recurrence 2.823165e-02 0.3844320  2.428049            AFTPH
563        Q8IYU8 Recurrence 2.823301e-02 0.3844320 -2.412059            MICU2
564        Q04323 Recurrence 2.825602e-02 0.3844320  2.411651            UBXN1
565        P49748 Recurrence 2.831497e-02 0.3844320 -2.410605           ACADVL
566        Q9BZE1 Recurrence 2.835607e-02 0.3844320 -2.409877           MRPL37
567        P11137 Recurrence 2.843451e-02 0.3844320  2.408490             MAP2
568        Q5T749 Recurrence 2.845848e-02 0.3844320  2.442368             KPRP
569        Q9NQP4 Recurrence 2.846647e-02 0.3844320  2.407926            PFDN4
570        Q8IUH4 Recurrence 2.848385e-02 0.3844320  2.605421          ZDHHC13
571        P07942 Recurrence 2.857629e-02 0.3847680  2.405993            LAMB1
572        Q9BRX8 Recurrence 2.863881e-02 0.3847680 -2.404896           PRXL2A
573        Q8NEY8 Recurrence 2.868939e-02 0.3847680  2.404010           PPHLN1
574        Q96BP2 Recurrence 2.870881e-02 0.3847680 -2.662168           CHCHD1
575        P37840 Recurrence 2.893267e-02 0.3861189  2.399768             SNCA
576        Q99536 Recurrence 2.908668e-02 0.3861189 -2.397099             VAT1
577        Q8IXH7 Recurrence 2.909925e-02 0.3861189  2.396882           NELFCD
578        O76094 Recurrence 2.909956e-02 0.3861189 -2.396877            SRP72
579        P51398 Recurrence 2.921431e-02 0.3861189 -2.394898             DAP3
580        Q14CN2 Recurrence 2.924631e-02 0.3861189  2.394348            CLCA4
581        Q9NWT6 Recurrence 2.926459e-02 0.3861189 -2.394034           HIF1AN
582        Q12846 Recurrence 2.926592e-02 0.3861189  2.394011             STX4
583        O60762 Recurrence 2.929960e-02 0.3861189 -2.393433             DPM1
584        O15018 Recurrence 2.940092e-02 0.3861189  2.425410            PDZD2
585        Q8WV41 Recurrence 2.940640e-02 0.3861189 -2.391603            SNX33
586        Q14247 Recurrence 2.941189e-02 0.3861189  2.391509             CTTN
587        P46379 Recurrence 2.964676e-02 0.3865281  2.387508             BAG6
588        Q4G0X4 Recurrence 2.971390e-02 0.3865281 -2.386369           KCTD21
589        Q9Y490 Recurrence 2.974503e-02 0.3865281  2.385842             TLN1
590        Q9H9H4 Recurrence 2.988089e-02 0.3865281  2.383548           VPS37B
591        P41236 Recurrence 2.991909e-02 0.3865281  2.382905           PPP1R2
592        P12532 Recurrence 2.992358e-02 0.3865281  2.437204           CKMT1B
593        P55290 Recurrence 2.994011e-02 0.3865281  2.382552            CDH13
594        Q9BZF1 Recurrence 2.995623e-02 0.3865281 -2.382281           OSBPL8
595        Q96DH6 Recurrence 3.000667e-02 0.3865281 -2.381433             MSI2
596        P12235 Recurrence 3.001510e-02 0.3865281 -2.381292          SLC25A4
597        Q6P1X6 Recurrence 3.005036e-02 0.3865281 -2.380701          C8orf82
598        Q9UKM7 Recurrence 3.017527e-02 0.3865281 -2.378611           MAN1B1
599        Q9NXR1 Recurrence 3.021953e-02 0.3865281  2.377873             NDE1
600        Q14980 Recurrence 3.026304e-02 0.3865281  2.377148            NUMA1
601        Q9NZC7 Recurrence 3.028943e-02 0.3865281  2.392090             WWOX
602        P14209 Recurrence 3.033588e-02 0.3865281  2.375937             CD99
603        Q8WZA0 Recurrence 3.038282e-02 0.3865281  2.375158             LZIC
604        Q53LP3 Recurrence 3.039014e-02 0.3865281  2.375037           SOWAHC
605        Q92843 Recurrence 3.039770e-02 0.3865281  2.408029           BCL2L2
606        P28068 Recurrence 3.047059e-02 0.3868156 -2.373704          HLA-DMB
607        P21964 Recurrence 3.052403e-02 0.3868556 -2.372821             COMT
608        O75367 Recurrence 3.059808e-02 0.3871563 -2.371600        MACROH2A1
609        P12694 Recurrence 3.067306e-02 0.3874677 -2.370366           BCKDHA
610        Q8IV08 Recurrence 3.082484e-02 0.3887467 -2.367876             PLD3
611        Q96RL1 Recurrence 3.092511e-02 0.3890518  2.381448            UIMC1
612        Q13330 Recurrence 3.096265e-02 0.3890518 -2.365626             MTA1
613        Q8NBF6 Recurrence 3.101611e-02 0.3890518 -2.418088             AVL9
614        Q9Y4F3 Recurrence 3.109067e-02 0.3890518  2.378711            MARF1
615        P21127 Recurrence 3.110189e-02 0.3890518 -2.363362           CDK11B
616        P62861 Recurrence 3.120804e-02 0.3892521 -2.361643              FAU
617        P55084 Recurrence 3.125131e-02 0.3892521 -2.360943            HADHB
618        Q9NY61 Recurrence 3.127251e-02 0.3892521 -2.393216             AATF
619        P42765 Recurrence 3.132232e-02 0.3892521 -2.359797            ACAA2
620        Q6IQ49 Recurrence 3.140927e-02 0.3892521  2.358398             SDE2
621        P61077 Recurrence 3.147038e-02 0.3892521 -2.372484           UBE2D3
622        Q8TCF1 Recurrence 3.147210e-02 0.3892521 -2.357389           ZFAND1
623        O60437 Recurrence 3.163116e-02 0.3898318  2.354842              PPL
624        Q717R9 Recurrence 3.169435e-02 0.3898318  2.353834             CYS1
625        Q14202 Recurrence 3.169834e-02 0.3898318  2.430590            ZMYM3
626        Q5TZA2 Recurrence 3.178912e-02 0.3898318  2.352326            CROCC
627        Q86WB0 Recurrence 3.183435e-02 0.3898318  2.351607           ZC3HC1
628        O43556 Recurrence 3.184447e-02 0.3898318  2.351446             SGCE
629        P19013 Recurrence 3.192165e-02 0.3898318  2.350223             KRT4
630        Q13751 Recurrence 3.192435e-02 0.3898318  2.350180            LAMB3
631        O43299 Recurrence 3.202788e-02 0.3899319 -2.348544            AP5Z1
632        O95817 Recurrence 3.213404e-02 0.3899319  2.346870             BAG3
633        Q99614 Recurrence 3.216228e-02 0.3899319  2.346426             TTC1
634        Q9UJ83 Recurrence 3.217489e-02 0.3899319 -2.361116            HACL1
635        P55036 Recurrence 3.223171e-02 0.3899319  2.345336            PSMD4
636        Q15166 Recurrence 3.229226e-02 0.3899319 -2.344386             PON3
637        Q8TB36 Recurrence 3.236843e-02 0.3899319 -2.358034            GDAP1
638        Q9Y697 Recurrence 3.237193e-02 0.3899319 -2.343140             NFS1
639        Q9H0G5 Recurrence 3.240445e-02 0.3899319  2.342632            NSRP1
640        Q9BYD3 Recurrence 3.243942e-02 0.3899319 -2.342086            MRPL4
641        Q92945 Recurrence 3.251341e-02 0.3902116  2.340934            KHSRP
642        Q5PRF9 Recurrence 3.259233e-02 0.3903485  2.354491           SAMD4B
643        Q96E39 Recurrence 3.267972e-02 0.3903485  2.338351           RBMXL1
644        Q9H993 Recurrence 3.276160e-02 0.3903485 -2.337085            ARMT1
645        Q9UEW3 Recurrence 3.280844e-02 0.3903485 -2.351092            MARCO
646        O00443 Recurrence 3.282602e-02 0.3903485 -2.336090          PIK3C2A
647        Q92484 Recurrence 3.282926e-02 0.3903485 -2.411392          SMPDL3A
648        P20930 Recurrence 3.303698e-02 0.3920619  2.384370              FLG
649        P35241 Recurrence 3.322962e-02 0.3920619  2.329900              RDX
650        Q53FT3 Recurrence 3.330895e-02 0.3920619 -2.328692          HIKESHI
651        O95183 Recurrence 3.335624e-02 0.3920619  2.327973            VAMP5
652        P78559 Recurrence 3.341621e-02 0.3920619  2.327062            MAP1A
653        Q9ULR0 Recurrence 3.346199e-02 0.3920619  2.326368             ISY1
654        P55212 Recurrence 3.349529e-02 0.3920619 -2.325864            CASP6
655        Q9NP81 Recurrence 3.351729e-02 0.3920619 -2.325531            SARS2
656        P01703 Recurrence 3.354617e-02 0.3920619  2.325095         IGLV1-40
657        Q8NHG8 Recurrence 3.356671e-02 0.3920619  2.339332            ZNRF2
658        P46937 Recurrence 3.360913e-02 0.3920619  2.324144             YAP1
659        P21730 Recurrence 3.365290e-02 0.3920619 -2.323484            C5AR1
660        P62310 Recurrence 3.366845e-02 0.3920619  2.323250             LSM3
661        O00483 Recurrence 3.368685e-02 0.3920619 -2.322973           NDUFA4
662        Q27J81 Recurrence 3.390245e-02 0.3931014  2.319736             INF2
663        Q9Y485 Recurrence 3.392844e-02 0.3931014 -2.333810            DMXL1
664        Q9NVA1 Recurrence 3.396825e-02 0.3931014 -2.318753            UQCC1
665        Q13740 Recurrence 3.401992e-02 0.3931014  2.317981            ALCAM
666        O60245 Recurrence 3.403939e-02 0.3931014  2.348828            PCDH7
667        Q7Z6K5 Recurrence 3.415916e-02 0.3931014  2.315908            ARPIN
668        Q99828 Recurrence 3.423310e-02 0.3931014  2.314810             CIB1
669        Q9NZN5 Recurrence 3.429707e-02 0.3931014  2.313863         ARHGEF12
670        A6NJ78 Recurrence 3.435567e-02 0.3931014 -2.343974          METTL15
671        Q9HCC0 Recurrence 3.437269e-02 0.3931014 -2.312744            MCCC2
672        P26368 Recurrence 3.441946e-02 0.3931014 -2.312054            U2AF2
673        O00161 Recurrence 3.445462e-02 0.3931014  2.311535           SNAP23
674        Q9UNE7 Recurrence 3.456831e-02 0.3931014  2.309862            STUB1
675        Q9BYE2 Recurrence 3.459252e-02 0.3931014  2.309506         TMPRSS13
676        P20020 Recurrence 3.465424e-02 0.3931014  2.308600           ATP2B1
677        Q13561 Recurrence 3.471278e-02 0.3931014  2.307743            DCTN2
678        Q92499 Recurrence 3.473027e-02 0.3931014 -2.307487             DDX1
679        Q12923 Recurrence 3.475822e-02 0.3931014 -2.307078           PTPN13
680        Q99961 Recurrence 3.483688e-02 0.3931014  2.305929           SH3GL1
681        Q8IVE3 Recurrence 3.497062e-02 0.3931014  2.303981          PLEKHH2
682        Q6P4A8 Recurrence 3.501318e-02 0.3931014 -2.303363            PLBD1
683        Q9Y2H5 Recurrence 3.505204e-02 0.3931014  2.302799          PLEKHA6
684        Q9BTU6 Recurrence 3.506329e-02 0.3931014 -2.302636           PI4K2A
685        Q9NY46 Recurrence 3.508113e-02 0.3931014 -2.302377            SCN3A
686        Q01196 Recurrence 3.514520e-02 0.3931014 -2.435052            RUNX1
687        Q15269 Recurrence 3.514763e-02 0.3931014 -2.301414             PWP2
688        Q9P2B4 Recurrence 3.518026e-02 0.3931014  2.300942        CTTNBP2NL
689        Q9NVM9 Recurrence 3.520692e-02 0.3931014 -2.300557           INTS13
690        P49903 Recurrence 3.528930e-02 0.3934501 -2.299368           SEPHS1
691        P14317 Recurrence 3.537693e-02 0.3937069  2.298107            HCLS1
692        Q9UHP3 Recurrence 3.541469e-02 0.3937069 -2.311687            USP25
693        P30626 Recurrence 3.554320e-02 0.3945654  2.295721              SRI
694        Q6NZI2 Recurrence 3.583960e-02 0.3972825  2.291493           CAVIN1
695        P35609 Recurrence 3.603058e-02 0.3973255  2.288786            ACTN2
696        Q9Y4K0 Recurrence 3.609273e-02 0.3973255  2.386364            LOXL2
697        Q9Y4X5 Recurrence 3.633458e-02 0.3973255  2.284505            ARIH1
698        Q96PE3 Recurrence 3.636698e-02 0.3973255 -2.284051           INPP4A
699        Q9BRT3 Recurrence 3.647386e-02 0.3973255  2.282555            MIEN1
700        P53634 Recurrence 3.650519e-02 0.3973255 -2.282117             CTSC
701        Q15363 Recurrence 3.662962e-02 0.3973255 -2.280382            TMED2
702        Q8N8S7 Recurrence 3.662998e-02 0.3973255  2.280377             ENAH
703        P18827 Recurrence 3.671833e-02 0.3973255  2.279148             SDC1
704        Q9BXL7 Recurrence 3.675447e-02 0.3973255 -2.308477           CARD11
705        Q9Y228 Recurrence 3.676400e-02 0.3973255  2.278515         TRAF3IP3
706        Q13424 Recurrence 3.676411e-02 0.3973255  2.292343            SNTA1
707        Q02952 Recurrence 3.686115e-02 0.3973255  2.277168           AKAP12
708        P04264 Recurrence 3.686713e-02 0.3973255  2.277086             KRT1
709        Q01973 Recurrence 3.687777e-02 0.3973255  2.276938             ROR1
710        Q01658 Recurrence 3.689185e-02 0.3973255  2.276744              DR1
711        Q14642 Recurrence 3.697999e-02 0.3973255  2.275526           INPP5A
712        Q9BVT8 Recurrence 3.701098e-02 0.3973255  2.275098            TMUB1
713        Q9Y6D5 Recurrence 3.701448e-02 0.3973255 -2.275050          ARFGEF2
714        O60884 Recurrence 3.703696e-02 0.3973255 -2.274740           DNAJA2
715        Q93091 Recurrence 3.703766e-02 0.3973255 -2.274731           RNASE6
716        Q86WU2 Recurrence 3.714973e-02 0.3973255 -2.273189             LDHD
717        Q14258 Recurrence 3.717506e-02 0.3973255 -2.272841           TRIM25
718        Q9HB90 Recurrence 3.719610e-02 0.3973255 -2.272552            RRAGC
719        Q16831 Recurrence 3.728433e-02 0.3973255 -2.271342             UPP1
720        P52948 Recurrence 3.730746e-02 0.3973255 -2.271026            NUP98
721        Q6P3W7 Recurrence 3.735509e-02 0.3973255 -2.270374            SCYL2
722        Q14790 Recurrence 3.737958e-02 0.3973255 -2.270040            CASP8
723        O14972 Recurrence 3.741837e-02 0.3973255 -2.269510           VPS26C
724        Q9P2M7 Recurrence 3.746467e-02 0.3973255  2.268878              CGN
725        Q9UPN7 Recurrence 3.754011e-02 0.3973255  2.267851           PPP6R1
726        Q9UBF2 Recurrence 3.759477e-02 0.3973255 -2.267108            COPG2
727        Q9Y520 Recurrence 3.763054e-02 0.3973255  2.266622           PRRC2C
728        P60033 Recurrence 3.764712e-02 0.3973255 -2.266397             CD81
729        Q9H0A8 Recurrence 3.765115e-02 0.3973255 -2.266342           COMMD4
730        Q93099 Recurrence 3.772351e-02 0.3975438  2.556870              HGD
731        P78367 Recurrence 3.782143e-02 0.3980305  2.264036           NKX3-2
732        Q9H6L2 Recurrence 3.790844e-02 0.3984011  2.292174          TMEM231
733        Q8TER0 Recurrence 3.803873e-02 0.3989953  2.274674            SNED1
734        O00592 Recurrence 3.806871e-02 0.3989953  2.260705            PODXL
735        Q9P0I2 Recurrence 3.824702e-02 0.3997344 -2.258315             EMC3
736        Q9H2A7 Recurrence 3.834676e-02 0.3997344 -2.256983           CXCL16
737        Q13409 Recurrence 3.842433e-02 0.3997344  2.255949          DYNC1I2
738        Q86TU7 Recurrence 3.845472e-02 0.3997344 -2.255545            SETD3
739        Q9HCY8 Recurrence 3.855325e-02 0.3997344  2.254235          S100A14
740        O75937 Recurrence 3.858218e-02 0.3997344  2.253851           DNAJC8
741        Q5ST30 Recurrence 3.862151e-02 0.3997344 -2.348027            VARS2
742        Q96RK0 Recurrence 3.877407e-02 0.3997344  2.251312              CIC
743        Q9BR76 Recurrence 3.882088e-02 0.3997344 -2.250694           CORO1B
744        Q8TEB1 Recurrence 3.883211e-02 0.3997344  2.250546           DCAF11
745        Q9UGM3 Recurrence 3.887489e-02 0.3997344  2.249982            DMBT1
746        A5D8V6 Recurrence 3.888325e-02 0.3997344  2.249872           VPS37C
747        Q14444 Recurrence 3.897014e-02 0.3997344  2.248729          CAPRIN1
748        O95831 Recurrence 3.902854e-02 0.3997344 -2.247962            AIFM1
749        O00233 Recurrence 3.903450e-02 0.3997344  2.247884            PSMD9
750        Q16706 Recurrence 3.905944e-02 0.3997344 -2.247557           MAN2A1
751        Q8N163 Recurrence 3.922535e-02 0.3997344  2.245385            CCAR2
752        Q8IWE2 Recurrence 3.923492e-02 0.3997344  2.245260         FAM114A1
753        Q9H6R3 Recurrence 3.924064e-02 0.3997344 -2.245186            ACSS3
754 P0DI81;P0DI82 Recurrence 3.928281e-02 0.3997344  2.312711 TRAPPC2;TRAPPC2B
755        Q99487 Recurrence 3.930170e-02 0.3997344 -2.244389           PAFAH2
756        Q86WN1 Recurrence 3.931666e-02 0.3997344  2.244194           FCHSD1
757        P24311 Recurrence 3.934780e-02 0.3997344 -2.243788            COX7B
758        Q5TEJ8 Recurrence 3.938628e-02 0.3997344 -2.243287          THEMIS2
759        Q9HBL7 Recurrence 3.977286e-02 0.4014596 -2.238280           PLGRKT
760        Q9NQS1 Recurrence 3.979802e-02 0.4014596  2.266461             AVEN
761    A0A0B4J1Y8 Recurrence 3.981157e-02 0.4014596  2.266281         IGLV9-49
762        Q5VTR2 Recurrence 3.982393e-02 0.4014596  2.237622            RNF20
763        Q58WW2 Recurrence 3.992628e-02 0.4014596  2.236305            DCAF6
764        P09497 Recurrence 4.013553e-02 0.4014596  2.233623             CLTB
765        Q92974 Recurrence 4.015729e-02 0.4014596  2.233345          ARHGEF2
766        O94855 Recurrence 4.017854e-02 0.4014596 -2.233073           SEC24D
767        Q9UQ80 Recurrence 4.022927e-02 0.4014596 -2.232425            PA2G4
768        Q9NXC5 Recurrence 4.024339e-02 0.4014596 -2.232245             MIOS
769        Q32MZ4 Recurrence 4.033055e-02 0.4014596  2.231135          LRRFIP1
770        Q99567 Recurrence 4.038104e-02 0.4014596 -2.230492            NUP88
771        Q9GZN8 Recurrence 4.038265e-02 0.4014596  2.230472         C20orf27
772        Q9H5V9 Recurrence 4.043972e-02 0.4014596 -2.275699           STEEP1
773        P23528 Recurrence 4.044829e-02 0.4014596 -2.229638             CFL1
774        P58340 Recurrence 4.062227e-02 0.4014596  2.294193             MLF1
775        Q8TE02 Recurrence 4.063684e-02 0.4014596  2.273072             ELP5
776        Q92545 Recurrence 4.065461e-02 0.4014596  2.227025          TMEM131
777        O95352 Recurrence 4.071407e-02 0.4014596 -2.226274             ATG7
778        O75030 Recurrence 4.072956e-02 0.4014596  2.239124             MITF
779        Q53H12 Recurrence 4.080662e-02 0.4014596 -2.225107              AGK
780        Q01469 Recurrence 4.084187e-02 0.4014596 -2.224664            FABP5
781        Q02338 Recurrence 4.088350e-02 0.4014596 -2.224140             BDH1
782        Q96N66 Recurrence 4.089914e-02 0.4014596 -2.223944           MBOAT7
783        P13639 Recurrence 4.094382e-02 0.4014596 -2.223383             EEF2
784        Q5KU26 Recurrence 4.094977e-02 0.4014596  2.223308          COLEC12
785        Q9NUB1 Recurrence 4.097210e-02 0.4014596 -2.223028            ACSS1
786        P31946 Recurrence 4.102227e-02 0.4014596  2.222399            YWHAB
787        Q8N6H7 Recurrence 4.114407e-02 0.4014596  2.220875          ARFGAP2
788        P35270 Recurrence 4.122074e-02 0.4014596 -2.219917              SPR
789        P55854 Recurrence 4.123894e-02 0.4014596  2.219690            SUMO3
790        Q9NZL9 Recurrence 4.124495e-02 0.4014596 -2.219615            MAT2B
791        Q16719 Recurrence 4.131662e-02 0.4014596 -2.218723             KYNU
792        P29966 Recurrence 4.133056e-02 0.4014596  2.218549           MARCKS
793        K7EJ46 Recurrence 4.146716e-02 0.4022785  2.282810           SMIM22
794        Q9GZV5 Recurrence 4.157397e-02 0.4028068  2.215529            WWTR1
795        Q96G23 Recurrence 4.167500e-02 0.4032777 -2.214280            CERS2
796        Q8NDA8 Recurrence 4.178963e-02 0.4038790 -2.212866            MROH1
797        Q13753 Recurrence 4.185080e-02 0.4039627  2.212113            LAMC2
798        P25686 Recurrence 4.198523e-02 0.4047523  2.210462           DNAJB2
799        Q5JRX3 Recurrence 4.203887e-02 0.4047622 -2.209805           PITRM1
800        P09661 Recurrence 4.220512e-02 0.4055339  2.207772           SNRPA1
801        Q5T3I0 Recurrence 4.222444e-02 0.4055339 -2.252342          GPATCH4
802        O43847 Recurrence 4.240160e-02 0.4063314 -2.205379             NRDC
803        Q9NZD8 Recurrence 4.241312e-02 0.4063314 -2.217980            SPG21
804        O43708 Recurrence 4.257239e-02 0.4073500 -2.203308            GSTZ1
805        P04921 Recurrence 4.265562e-02 0.4074396  2.202301             GYPC
806        Q9H410 Recurrence 4.268768e-02 0.4074396  2.214606             DSN1
807        O60812 Recurrence 4.286871e-02 0.4078456 -2.199732         HNRNPCL1
808        O14548 Recurrence 4.291252e-02 0.4078456 -2.199205          COX7A2L
809        Q16363 Recurrence 4.293682e-02 0.4078456  2.198913            LAMA4
810        Q96MW5 Recurrence 4.309990e-02 0.4078456 -2.196957             COG8
811        P21462 Recurrence 4.310316e-02 0.4078456  2.285688             FPR1
812        Q15029 Recurrence 4.323427e-02 0.4078456 -2.195351           EFTUD2
813        P08648 Recurrence 4.328043e-02 0.4078456  2.194801            ITGA5
814        Q9UMN6 Recurrence 4.328330e-02 0.4078456 -2.194766            KMT2B
815        Q9UKM9 Recurrence 4.329219e-02 0.4078456  2.194661             RALY
816        Q99643 Recurrence 4.333247e-02 0.4078456 -2.312366             SDHC
817        Q8IZV5 Recurrence 4.335357e-02 0.4078456 -2.193929            RDH10
818        Q9BUH6 Recurrence 4.336639e-02 0.4078456  2.193777             PAXX
819        O15514 Recurrence 4.362428e-02 0.4089630 -2.190716           POLR2D
820        Q0VGL1 Recurrence 4.365806e-02 0.4089630 -2.190316          LAMTOR4
821        Q9UNW1 Recurrence 4.368156e-02 0.4089630 -2.190038           MINPP1
822        O43681 Recurrence 4.369786e-02 0.4089630 -2.189846             GET3
823        Q6P3X3 Recurrence 4.383271e-02 0.4096236 -2.215195            TTC27
824        P51956 Recurrence 4.387493e-02 0.4096236 -2.214682             NEK3
825        Q96GM5 Recurrence 4.397022e-02 0.4100156  2.213527          SMARCD1
826        P49591 Recurrence 4.438653e-02 0.4132906 -2.181765            SARS1
827        Q12907 Recurrence 4.445711e-02 0.4132906 -2.180944            LMAN2
828        P28799 Recurrence 4.456494e-02 0.4132906 -2.179691              GRN
829        Q9NRP2 Recurrence 4.460468e-02 0.4132906 -2.179230             CMC2
830        Q9H3Q1 Recurrence 4.461259e-02 0.4132906  2.179138         CDC42EP4
831        O43272 Recurrence 4.466621e-02 0.4132906 -2.190875            PRODH
832        Q15011 Recurrence 4.483354e-02 0.4132906 -2.176583          HERPUD1
833        Q9H981 Recurrence 4.485552e-02 0.4132906  2.239276            ACTR8
834        Q9Y6M1 Recurrence 4.485857e-02 0.4132906  2.176294          IGF2BP2
835        A6NI72 Recurrence 4.488498e-02 0.4132906 -2.175989            NCF1B
836        Q99942 Recurrence 4.491238e-02 0.4132906 -2.238573             RNF5
837        Q8NFP9 Recurrence 4.502413e-02 0.4138239  2.200902             NBEA
838        Q9BQ13 Recurrence 4.515249e-02 0.4145085 -2.172913           KCTD14
839        O76070 Recurrence 4.549243e-02 0.4160347  2.169029             SNCG
840        P23246 Recurrence 4.549932e-02 0.4160347  2.168951             SFPQ
841        P39880 Recurrence 4.553026e-02 0.4160347  2.168599             CUX1
842        Q9BPW8 Recurrence 4.553506e-02 0.4160347 -2.168544         NIPSNAP1
843    A0A0J9YXX1 Recurrence 4.559326e-02 0.4160723 -2.180090       IGHV5-10-1
844        P29350 Recurrence 4.569190e-02 0.4162402 -2.166762            PTPN6
845        P05067 Recurrence 4.575088e-02 0.4162402  2.166094              APP
846        Q9H0W5 Recurrence 4.595820e-02 0.4162402  2.163750            CCDC8
847        P35579 Recurrence 4.597340e-02 0.4162402  2.163579             MYH9
848        Q92759 Recurrence 4.597435e-02 0.4162402 -2.421743           GTF2H4
849        P0CAP2 Recurrence 4.602348e-02 0.4162402  2.163015           POLR2M
850        Q14019 Recurrence 4.608385e-02 0.4162402 -2.162335            COTL1
851        Q9H2V7 Recurrence 4.611189e-02 0.4162402 -2.162019            SPNS1
852        O75146 Recurrence 4.615487e-02 0.4162402  2.161536            HIP1R
853        P31350 Recurrence 4.616577e-02 0.4162402 -2.310935             RRM2
854        Q86YB7 Recurrence 4.620926e-02 0.4162402 -2.160925           ECHDC2
855        Q69YQ0 Recurrence 4.634011e-02 0.4162402  2.159459          SPECC1L
856        P09651 Recurrence 4.635312e-02 0.4162402 -2.159313          HNRNPA1
857        P49821 Recurrence 4.646166e-02 0.4162402 -2.158100           NDUFV1
858        Q10567 Recurrence 4.647547e-02 0.4162402 -2.157945            AP1B1
859        O43427 Recurrence 4.647736e-02 0.4162402 -2.157924             FIBP
860        Q5EBL4 Recurrence 4.665780e-02 0.4164041  2.155913           RILPL1
861        Q8TBC5 Recurrence 4.666476e-02 0.4164041  2.198045          ZSCAN18
862        Q99439 Recurrence 4.673296e-02 0.4164041  2.155078             CNN2
863        Q8NBL1 Recurrence 4.676952e-02 0.4164041 -2.154672          POGLUT1
864        Q9UPN6 Recurrence 4.678660e-02 0.4164041  2.166502            SCAF8
865        Q92805 Recurrence 4.686803e-02 0.4164041  2.153580           GOLGA1
866        P22304 Recurrence 4.688728e-02 0.4164041 -2.153366              IDS
867        Q6UXY8 Recurrence 4.692868e-02 0.4164041  2.300920             TMC5
868        P16435 Recurrence 4.707719e-02 0.4172406 -2.151267              POR
869        Q8TDL5 Recurrence 4.718601e-02 0.4174397  2.211113           BPIFB1
870        O95400 Recurrence 4.720819e-02 0.4174397  2.149824           CD2BP2
871        Q7Z7H5 Recurrence 4.740927e-02 0.4187365 -2.147615            TMED4
872        Q9BRR8 Recurrence 4.759513e-02 0.4196637 -2.171224          GPATCH1
873        Q9H7M9 Recurrence 4.762335e-02 0.4196637  2.145273             VSIR
874        Q96DB5 Recurrence 4.771240e-02 0.4197473 -2.144302            RMDN1
875        O75976 Recurrence 4.775815e-02 0.4197473  2.143804              CPD
876        Q96P47 Recurrence 4.779652e-02 0.4197473  2.143386            AGAP3
877        P14314 Recurrence 4.785959e-02 0.4198219  2.142700           PRKCSH
878        P52209 Recurrence 4.792072e-02 0.4198491 -2.142036              PGD
879        O75390 Recurrence 4.797184e-02 0.4198491 -2.141482               CS
880        Q7KZ85 Recurrence 4.807888e-02 0.4201339 -2.140322           SUPT6H
881        P81605 Recurrence 4.811361e-02 0.4201339  2.139947              DCD
882        Q9UBR2 Recurrence 4.819604e-02 0.4203766 -2.139056             CTSZ
883        Q9Y312 Recurrence 4.840513e-02 0.4217222 -2.136803             AAR2
884        P63098 Recurrence 4.872764e-02 0.4240518  2.133345           PPP3R1
885        P56377 Recurrence 4.903048e-02 0.4244454 -2.130117            AP1S2
886        Q96AE4 Recurrence 4.907601e-02 0.4244454  2.129633            FUBP1
887        Q9UKD2 Recurrence 4.915682e-02 0.4244454 -2.128776            MRTO4
888        Q02224 Recurrence 4.926827e-02 0.4244454  2.127595            CENPE
889        Q8TD16 Recurrence 4.928238e-02 0.4244454  2.127446            BICD2
890        Q8TES7 Recurrence 4.937218e-02 0.4244454  2.167277             FBF1
891        Q9H773 Recurrence 4.942437e-02 0.4244454  2.125946           DCTPP1
892        Q8NI60 Recurrence 4.943400e-02 0.4244454 -2.125845            COQ8A
893        P23470 Recurrence 4.943639e-02 0.4244454  2.125819            PTPRG
894        P60953 Recurrence 4.953494e-02 0.4244454 -2.124781            CDC42
895        O75608 Recurrence 4.953848e-02 0.4244454 -2.124743           LYPLA1
896        O95164 Recurrence 4.975657e-02 0.4244454  2.122452             UBL3
897        Q7RTP6 Recurrence 4.977234e-02 0.4244454  2.122287           MICAL3
898        P37235 Recurrence 4.980754e-02 0.4244454  2.121918           HPCAL1
899        P53618 Recurrence 4.986012e-02 0.4244454 -2.121367            COPB1
900        Q969G5 Recurrence 4.989664e-02 0.4244454  2.120985           CAVIN3</code></pre>
<pre class="r"><code>tFeatSigRes$Var1[413] &lt;- &#39;P0DP23;P0DP24;P0DP25&#39;
tFeatSigRes$Gene[413] &lt;- &#39;CALM1;CALM2;CALM3&#39;

# Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(Stat))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Normal tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-49-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-49-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-49-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Use probabilistic dropout model to account for data missingness and
identify differentially abundant entities that are significantly
associated with cancer recurrence. Note that features in heatmap are
ordered by their t-statistics.</p>
<pre class="r"><code># Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
# fit

# Check coefficient names for Parameter &#39;contrast&#39; in proDA::test_diff()
# proDA::result_names(fit)
# Identify significant differentially abundant entities
all_proDAProNormalRes &lt;- proDA::test_diff(fit,
                                          contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                          sort_by = &#39;pval&#39;)
proDAProNormalRes &lt;- dplyr::select(all_proDAProNormalRes, -c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::filter(pval &lt; 0.05) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(name,
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F))
proDAProNormalRes</code></pre>
<pre><code>                    name         pval  adj_pval t_statistic              Gene
1                 Q8NI37 0.0002367870 0.3384232    4.708304             PPTC7
2                 A6NKN8 0.0002806894 0.3384232   -4.625425            PCP4L1
3                 Q6NXT1 0.0002878176 0.3384232   -4.613233           ANKRD54
4                 Q6IPR1 0.0003070315 0.3384232    4.581846            ETFRF1
5                 Q96CV9 0.0003074132 0.3384232   -4.581243              OPTN
6                 Q96E40 0.0003949229 0.3384232   -4.459991            SPACA9
7                 P61326 0.0004552430 0.3384232   -4.391461             MAGOH
8                 O15382 0.0005781161 0.3384232    4.276663             BCAT2
9                 Q9NSC5 0.0006017558 0.3384232   -4.257454            HOMER3
10                Q8TDB4 0.0006053610 0.3384232   -4.254592             MGARP
11                O96008 0.0006260489 0.3384232    4.238497            TOMM40
12                Q14515 0.0006797627 0.3384232   -4.199107           SPARCL1
13                Q9UKK3 0.0007176521 0.3384232    4.173178             PARP4
14                Q92851 0.0007263143 0.3384232    4.167446            CASP10
15                A4D1E1 0.0007563768 0.3384232   -4.148075           ZNF804B
16                P21953 0.0007783478 0.3384232    4.134405            BCKDHB
17                Q17RN3 0.0007845721 0.3384232   -4.130604            FAM98C
18                Q13790 0.0008713500 0.3384232   -4.080574              APOF
19                Q96EQ0 0.0008893104 0.3384232   -4.070851              SGTB
20                Q96C36 0.0009247648 0.3384232    4.052229             PYCR2
21                Q8N9L9 0.0009335323 0.3384232   -4.047735             ACOT4
22                P13647 0.0010093660 0.3384232   -4.010561              KRT5
23                Q9NPH3 0.0010329643 0.3384232   -3.999567            IL1RAP
24                O15195 0.0010557854 0.3384232   -3.989175              VILL
25                P62979 0.0011689162 0.3596989   -3.940794            RPS27A
26                P52789 0.0012437684 0.3603022    3.911316               HK2
27                P78346 0.0012645468 0.3603022    3.903451             RPP30
28                P84243 0.0013713653 0.3648421   -3.864967             H3-3B
29                P02533 0.0014051475 0.3648421   -3.853423             KRT14
30                Q14376 0.0014227560 0.3648421    3.847516              GALE
31                P09668 0.0016452649 0.4016270    3.778633              CTSH
32                Q5M775 0.0017882840 0.4016270   -3.739143            SPECC1
33                Q9Y512 0.0018589994 0.4016270    3.720774            SAMM50
34                Q9ULP9 0.0019482436 0.4016270    3.698569           TBC1D24
35                P24821 0.0020576009 0.4016270   -3.672712               TNC
36                P52849 0.0020630635 0.4016270    3.671456             NDST2
37                Q02413 0.0022421066 0.4016270   -3.632058              DSG1
38                Q56VL3 0.0022831200 0.4016270    3.623477            OCIAD2
39                P49642 0.0022972208 0.4016270    3.620563             PRIM1
40                P35914 0.0023059842 0.4016270    3.618760             HMGCL
41                Q8IY95 0.0023397066 0.4016270    3.611888           TMEM192
42                Q7Z794 0.0024240377 0.4016270   -3.595127             KRT77
43                O60825 0.0024629539 0.4016270    3.587589            PFKFB2
44                P51649 0.0024886407 0.4016270    3.582678           ALDH5A1
45                P22735 0.0024957805 0.4016270   -3.581321              TGM1
46                Q9NVJ2 0.0025940258 0.4016270    3.563046             ARL8B
47                Q08554 0.0025962763 0.4016270   -3.562635              DSC1
48                Q86YM7 0.0026135008 0.4016270   -3.559505            HOMER1
49                P02538 0.0026706076 0.4016270   -3.549273             KRT6A
50                Q8TDQ1 0.0027445558 0.4016270    3.536344           CD300LF
51                O94808 0.0027497556 0.4016270    3.535448             GFPT2
52                P09917 0.0028417295 0.4016270    3.519872             ALOX5
53                Q5VWJ9 0.0029520453 0.4016270    3.501841             SNX30
54                Q8N4T8 0.0029641226 0.4016270    3.499908              CBR4
55                O14578 0.0030176359 0.4016270   -3.491436               CIT
56                O60828 0.0033322439 0.4016270   -3.444471             PQBP1
57                Q6P4A7 0.0033342375 0.4016270    3.444187             SFXN4
58                P61956 0.0034486988 0.4016270   -3.428198             SUMO2
59                Q05823 0.0035561518 0.4016270    3.413661            RNASEL
60                Q96BM9 0.0035585241 0.4016270    3.413345             ARL8A
61                P17405 0.0036807647 0.4016270    3.397340             SMPD1
62                O00116 0.0036873625 0.4016270    3.396492              AGPS
63                P11310 0.0036981777 0.4016270    3.395104             ACADM
64                Q6KB66 0.0037082945 0.4016270   -3.393809             KRT80
65                Q96F07 0.0037099777 0.4016270    3.393594            CYFIP2
66                Q9NP74 0.0037336899 0.4016270   -3.390574             PALMD
67                Q9Y5U9 0.0037662729 0.4016270    3.386456           IER3IP1
68                P48681 0.0037672392 0.4016270   -3.386334               NES
69                Q5JPH6 0.0038085807 0.4016270    3.381161             EARS2
70                Q8NB16 0.0039877679 0.4016270    3.359364              MLKL
71                P10620 0.0040490661 0.4016270    3.352130             MGST1
72                Q5T200 0.0040500098 0.4016270   -3.352019            ZC3H13
73                Q9H9P8 0.0041101199 0.4016270    3.345032            L2HGDH
74                Q96QE2 0.0041313759 0.4016270    3.342585           SLC2A13
75                Q0VDG4 0.0041929787 0.4016270    3.335564             SCRN3
76                Q96IQ7 0.0042314863 0.4016270   -3.331228             VSIG2
77                Q5T7V8 0.0042635244 0.4016270   -3.327649             GORAB
78                P67936 0.0043169231 0.4016270   -3.321743              TPM4
79                Q7Z3J2 0.0044514562 0.4016270    3.307180            VPS35L
80                P07910 0.0045227610 0.4016270   -3.299637            HNRNPC
81                Q96E11 0.0045282432 0.4016270    3.299062              MRRF
82                P07951 0.0045840582 0.4016270   -3.293246              TPM2
83                Q9H936 0.0045906269 0.4016270    3.292566          SLC25A22
84                O15231 0.0046721441 0.4016270   -3.284209            ZNF185
85                Q96EH3 0.0046908754 0.4016270    3.282309            MALSU1
86                P21796 0.0047164154 0.4016270    3.279730             VDAC1
87                Q8N131 0.0048065945 0.4016270   -3.270735           TMEM123
88                Q5S007 0.0048126412 0.4016270    3.270138             LRRK2
89                P15924 0.0048353627 0.4016270   -3.267900               DSP
90                O00142 0.0048363999 0.4016270    3.267799               TK2
91                P49006 0.0048393438 0.4016270   -3.267510          MARCKSL1
92                Q9H7X7 0.0049351330 0.4016270   -3.258198             IFT22
93                Q96EX1 0.0049677503 0.4016270    3.255068            SMIM12
94                Q8TCS8 0.0049806550 0.4016270    3.253835             PNPT1
95                P53602 0.0050291360 0.4016270    3.249232               MVD
96                A0AV96 0.0050605913 0.4016270    3.246268             RBM47
97                Q99805 0.0050711088 0.4016270    3.245282            TM9SF2
98                Q96P48 0.0052065422 0.4016270    3.232753             ARAP1
99                O75602 0.0052294810 0.4016270   -3.230663             SPAG6
100               Q969E8 0.0052421634 0.4016270   -3.229511              TSR2
101               Q8NBQ5 0.0052728875 0.4016270    3.226733          HSD17B11
102               Q8IZQ1 0.0053692865 0.4049600   -3.218117             WDFY3
103               Q5XXA6 0.0056240920 0.4115790   -3.196059              ANO1
104               P09493 0.0056581780 0.4115790   -3.193183              TPM1
105               Q8IU68 0.0056972321 0.4115790   -3.189909              TMC8
106               Q9H8L6 0.0059516349 0.4115790   -3.169109             MMRN2
107               P99999 0.0061167665 0.4115790    3.156072              CYCS
108               Q96RQ3 0.0061303631 0.4115790    3.155014             MCCC1
109               P43897 0.0061614830 0.4115790    3.152601              TSFM
110               Q86SQ0 0.0061883284 0.4115790   -3.150529            PHLDB2
111               O15258 0.0061950406 0.4115790    3.150012              RER1
112               Q76N32 0.0063242829 0.4115790   -3.140171             CEP68
113               P06753 0.0064805161 0.4115790   -3.128534              TPM3
114               Q9NVZ3 0.0065521135 0.4115790    3.123294            NECAP2
115               P33121 0.0065893615 0.4115790    3.120590             ACSL1
116               Q15006 0.0066342177 0.4115790    3.117353              EMC2
117               Q9Y277 0.0066403749 0.4115790    3.116910             VDAC3
118               P43405 0.0066827172 0.4115790    3.113878               SYK
119               O60341 0.0067577574 0.4115790    3.108549             KDM1A
120               Q6NY19 0.0067711281 0.4115790   -3.107606             KANK3
121               Q9Y5G0 0.0067832533 0.4115790   -3.106752           PCDHGB5
122               P07205 0.0069506757 0.4115790    3.095112              PGK2
123               Q8IVM0 0.0070129561 0.4115790   -3.090852            CCDC50
124               Q9UKU7 0.0070148844 0.4115790    3.090721             ACAD8
125               P41208 0.0070382068 0.4115790   -3.089136             CETN2
126               Q9UIG8 0.0071497167 0.4115790   -3.081627           SLCO3A1
127               Q9BXB4 0.0071996755 0.4115790    3.078301           OSBPL11
128               P23919 0.0072114524 0.4115790    3.077520             DTYMK
129               Q96DT6 0.0072169801 0.4115790    3.077154             ATG4C
130               Q15424 0.0072248593 0.4115790   -3.076632              SAFB
131               P54886 0.0073753555 0.4115790    3.066779          ALDH18A1
132               Q96C11 0.0073816390 0.4115790    3.066372              FGGY
133               Q96M27 0.0074163654 0.4115790    3.064129             PRRC1
134               Q9UKU9 0.0074537287 0.4115790   -3.061726           ANGPTL2
135               O95678 0.0075412636 0.4115790   -3.056143             KRT75
136               Q16540 0.0075614526 0.4115790    3.054865            MRPL23
137               Q9H6F5 0.0076087016 0.4115790   -3.051885            CCDC86
138               O94923 0.0076317743 0.4115790    3.050437              GLCE
139               P29374 0.0076793609 0.4115790   -3.047464            ARID4A
140               Q7Z4W1 0.0077330593 0.4115790    3.044130              DCXR
141               Q8N3P4 0.0078015708 0.4115790    3.039909              VPS8
142               Q96LZ7 0.0078239411 0.4115790    3.038539             RMDN2
143               Q14435 0.0078872319 0.4115790    3.034683            GALNT3
144               Q9H4M9 0.0079066491 0.4115790   -3.033506              EHD1
145               O95436 0.0080567573 0.4115790    3.024502           SLC34A2
146               P19878 0.0081375224 0.4115790    3.019725              NCF2
147               O43291 0.0081606810 0.4115790   -3.018364            SPINT2
148               Q8IWE5 0.0081892655 0.4115790    3.016689           PLEKHM2
149               P49327 0.0081901695 0.4115790    3.016636              FASN
150               Q6P4E1 0.0081907974 0.4115790   -3.016600             GOLM2
151               P32321 0.0081980195 0.4115790    3.016177              DCTD
152               Q16643 0.0082791658 0.4115790   -3.011459              DBN1
153               Q8N4H5 0.0082885923 0.4115790    3.010913             TOMM5
154               P17302 0.0083021715 0.4115790   -3.010129              GJA1
155               Q2TB90 0.0083305195 0.4115790    3.008496             HKDC1
156               Q96IZ0 0.0084395686 0.4115790   -3.002263              PAWR
157               O75781 0.0084834957 0.4115790   -2.999774              PALM
158               Q9BZL4 0.0084835755 0.4115790   -2.999770          PPP1R12C
159               Q5MIZ7 0.0085065719 0.4115790   -2.998472           PPP4R3B
160               P02647 0.0085843698 0.4119169   -2.994107             APOA1
161               Q9UNZ2 0.0086485158 0.4119169   -2.990537            NSFL1C
162               P08590 0.0086741902 0.4119169   -2.989115              MYL3
163               P30622 0.0088080120 0.4137770   -2.981771             CLIP1
164               Q9NSB2 0.0088602067 0.4137770   -2.978936             KRT84
165               Q99653 0.0088747177 0.4137770   -2.978151              CHP1
166               Q8N2K0 0.0089861331 0.4142372    2.972163            ABHD12
167               P61218 0.0091051372 0.4142372   -2.965847            POLR2F
168               Q96GC9 0.0091078786 0.4142372    2.965702              VMP1
169               Q9Y2G9 0.0091529354 0.4142372    2.963332             SBNO2
170               P55263 0.0091855932 0.4142372    2.961622               ADK
171               Q9Y678 0.0093854400 0.4142372    2.951282             COPG1
172               Q8WZA9 0.0094137498 0.4142372   -2.949835              IRGQ
173               Q92506 0.0094198214 0.4142372    2.949525           HSD17B8
174               P42702 0.0094577191 0.4142372   -2.947596              LIFR
175           A0A075B6H9 0.0094666929 0.4142372   -2.947140          IGLV4-69
176               Q8IWU2 0.0095124307 0.4142372   -2.944823             LMTK2
177               P50502 0.0095916949 0.4142372   -2.940834              ST13
178               Q6IAN0 0.0095936656 0.4142372    2.940735            DHRS7B
179               P26038 0.0097440631 0.4142372   -2.933256               MSN
180               Q9BTZ2 0.0098041852 0.4142372    2.930297             DHRS4
181               Q9NPG4 0.0098277539 0.4142372   -2.929142            PCDH12
182           A0A0G2JMI3 0.0099010390 0.4142372   -2.925568        IGHV1-69-2
183               Q9UG63 0.0099071541 0.4142372    2.925270             ABCF2
184               Q8IY34 0.0099356770 0.4142372    2.923887           SLC15A3
185               Q8TF09 0.0099754400 0.4142372   -2.921965           DYNLRB2
186               Q05682 0.0101165969 0.4142372   -2.915202             CALD1
187               Q9NZK5 0.0101313883 0.4142372    2.914498              ADA2
188               O75629 0.0101799177 0.4142372    2.912198             CREG1
189               Q9NNW7 0.0101878211 0.4142372    2.911824            TXNRD2
190               Q9NR45 0.0102452829 0.4142372    2.909115              NANS
191               Q8IV50 0.0102845847 0.4142372   -2.907272            LYSMD2
192               Q96BP2 0.0105000904 0.4185675    2.897281            CHCHD1
193               Q6DT37 0.0105528121 0.4185675   -2.894867          CDC42BPG
194               Q14137 0.0106221626 0.4185675    2.891710              BOP1
195               P31350 0.0106362741 0.4185675    2.891070              RRM2
196               P54252 0.0107311378 0.4185675   -2.886789             ATXN3
197               Q96RQ1 0.0109574714 0.4185675    2.876722            ERGIC2
198               Q9NYB0 0.0109648257 0.4185675   -2.876398           TERF2IP
199               Q8IY63 0.0109833587 0.4185675   -2.875583            AMOTL1
200               P16278 0.0111024063 0.4185675    2.870381              GLB1
201               Q86VS8 0.0111567166 0.4185675   -2.868025             HOOK3
202               O15254 0.0113038941 0.4185675    2.861698             ACOX3
203               Q9UBY9 0.0113265885 0.4185675   -2.860730             HSPB7
204               Q9BQB6 0.0113465148 0.4185675    2.859881            VKORC1
205               Q9NPJ3 0.0113567961 0.4185675    2.859443            ACOT13
206               Q3KQU3 0.0114258111 0.4185675   -2.856517            MAP7D1
207               Q96GS4 0.0114693801 0.4185675   -2.854678            BORCS6
208               Q9UQN3 0.0114998272 0.4185675   -2.853398            CHMP2B
209               Q0VF96 0.0115413088 0.4185675   -2.851658             CGNL1
210               P98194 0.0116211319 0.4185675    2.848327            ATP2C1
211               O15056 0.0116796782 0.4185675    2.845898             SYNJ2
212               Q9ULX6 0.0117314453 0.4185675    2.843761            AKAP8L
213               P10909 0.0117983746 0.4185675   -2.841010               CLU
214               Q8NCJ5 0.0118060886 0.4185675   -2.840694            SPRYD3
215               O14974 0.0118728002 0.4185675   -2.837970          PPP1R12A
216               Q7L5L3 0.0119263845 0.4185675   -2.835792             GDPD3
217               Q8WW59 0.0119465138 0.4185675    2.834976            SPRYD4
218               Q9NTJ4 0.0119930350 0.4185675    2.833096            MAN2C1
219               Q96C90 0.0119997905 0.4185675   -2.832824          PPP1R14B
220               Q9H2U2 0.0120667673 0.4185675    2.830131              PPA2
221               Q9NWW5 0.0121116431 0.4185675    2.828335              CLN6
222               O15182 0.0121150645 0.4185675   -2.828198             CETN3
223               O15344 0.0121446510 0.4185675   -2.827018              MID1
224               Q13825 0.0122076349 0.4185675    2.824515               AUH
225               P23588 0.0122419974 0.4185675   -2.823154             EIF4B
226               P34741 0.0123228522 0.4194677   -2.819967              SDC2
227               Q16822 0.0125136256 0.4200615    2.812528              PCK2
228               O60711 0.0125349979 0.4200615    2.811701              LPXN
229               P06727 0.0126289925 0.4200615   -2.808082             APOA4
230               Q9NWB7 0.0126868450 0.4200615   -2.805868             IFT57
231               Q8IXS6 0.0127346824 0.4200615   -2.804044             PALM2
232               P04150 0.0127425670 0.4200615   -2.803744             NR3C1
233               Q16825 0.0128483794 0.4200615   -2.799735            PTPN21
234               P04843 0.0128626992 0.4200615    2.799195              RPN1
235               P08729 0.0129237587 0.4200615   -2.796899              KRT7
236               O14656 0.0129270109 0.4200615    2.796777             TOR1A
237               P01700 0.0129409300 0.4200615   -2.796255          IGLV1-47
238               Q7Z4V5 0.0130370837 0.4200772   -2.792665            HDGFL2
239               P39656 0.0133883210 0.4200772    2.779766             DDOST
240               Q9H425 0.0134122005 0.4200772   -2.778901          C1orf198
241               O14683 0.0134568651 0.4200772    2.777287           TP53I11
242               O00308 0.0137522134 0.4200772   -2.766742              WWP2
243               Q7Z3B1 0.0137579343 0.4200772   -2.766540             NEGR1
244               Q8WVV4 0.0137666525 0.4200772   -2.766233             POF1B
245               Q9Y4K0 0.0137987297 0.4200772   -2.765102             LOXL2
246               O60936 0.0138217904 0.4200772   -2.764290              NOL3
247               Q9Y2R9 0.0138740219 0.4200772    2.762457             MRPS7
248               Q8ND56 0.0139648186 0.4200772   -2.759287            LSM14A
249               Q9BUK0 0.0140393836 0.4200772    2.756698            CHCHD7
250               Q13563 0.0140620979 0.4200772   -2.755912              PKD2
251               P26022 0.0141193678 0.4200772   -2.753935              PTX3
252               O14874 0.0141744606 0.4200772    2.752041             BCKDK
253               O60830 0.0142033272 0.4200772    2.751052           TIMM17B
254               Q5T1C6 0.0142547752 0.4200772    2.749293             THEM4
255               P07741 0.0142671972 0.4200772    2.748869              APRT
256               P61952 0.0144122807 0.4200772   -2.743946             GNG11
257               P14923 0.0144182170 0.4200772   -2.743746               JUP
258               P08727 0.0145059084 0.4200772   -2.740795             KRT19
259               Q8N999 0.0145313354 0.4200772   -2.739942          C12orf29
260               Q9UKX5 0.0145560198 0.4200772   -2.739116            ITGA11
261               Q9UJC5 0.0145765317 0.4200772    2.738430          SH3BGRL2
262               P26885 0.0147122134 0.4200772    2.733919             FKBP2
263               P49321 0.0147451854 0.4200772   -2.732828              NASP
264               Q7Z406 0.0148160170 0.4200772   -2.730494             MYH14
265               Q93050 0.0149330869 0.4200772    2.726660          ATP6V0A1
266               P05388 0.0149572013 0.4200772    2.725873             RPLP0
267               Q9P2N2 0.0149785639 0.4200772   -2.725178          ARHGAP28
268               H7BZ55 0.0149866108 0.4200772   -2.724916            CROCC2
269               Q9NUJ1 0.0152294404 0.4200772    2.717081            ABHD10
270               O00264 0.0152692195 0.4200772   -2.715809            PGRMC1
271               Q969E4 0.0153057106 0.4200772   -2.714645            TCEAL3
272               Q92828 0.0154226065 0.4200772    2.710933            CORO2A
273               Q5JTZ9 0.0154227087 0.4200772    2.710930             AARS2
274               Q86T03 0.0156047831 0.4200772    2.705203            PIP4P1
275               Q6UXY8 0.0156251098 0.4200772   -2.704568              TMC5
276               O94925 0.0156751951 0.4200772    2.703006               GLS
277               Q8N5G0 0.0157539023 0.4200772    2.700561            SMIM20
278               Q9P0K7 0.0157615744 0.4200772   -2.700323             RAI14
279               P23786 0.0157965798 0.4200772    2.699240              CPT2
280               Q14166 0.0158620680 0.4200772    2.697219            TTLL12
281               Q96P11 0.0159174725 0.4200772    2.695517             NSUN5
282               Q9BWH6 0.0159449874 0.4200772    2.694673             RPAP1
283               Q9HAN9 0.0160294508 0.4200772    2.692092            NMNAT1
284               O96005 0.0161111912 0.4200772    2.689607            CLPTM1
285               Q66LE6 0.0161287353 0.4200772    2.689075           PPP2R2D
286               O75190 0.0162943108 0.4200772   -2.684083            DNAJB6
287               Q99547 0.0163627026 0.4200772   -2.682035          MPHOSPH6
288               Q12913 0.0163852454 0.4200772   -2.681362             PTPRJ
289               Q07617 0.0164085160 0.4200772   -2.680668             SPAG1
290               Q96CM8 0.0164392469 0.4200772    2.679753             ACSF2
291               O95273 0.0165321566 0.4200772   -2.676997           CCNDBP1
292               O00115 0.0165414358 0.4200772    2.676722            DNASE2
293               Q9H1E3 0.0165416471 0.4200772   -2.676716            NUCKS1
294               P08842 0.0166063914 0.4200772   -2.674805               STS
295               Q99417 0.0167250574 0.4200772   -2.671321             MYCBP
296               Q96IW7 0.0167431312 0.4200772    2.670792            SEC22A
297               O15173 0.0168203763 0.4200772   -2.668540            PGRMC2
298               Q15027 0.0168207676 0.4200772   -2.668528             ACAP1
299               Q9H5N1 0.0168958575 0.4200772   -2.666348            RABEP2
300               O43164 0.0169722179 0.4200772   -2.664140              PJA2
301               O43765 0.0170242404 0.4200772   -2.662641              SGTA
302               Q99571 0.0170773009 0.4200772    2.661117             P2RX4
303               Q96EP5 0.0171560015 0.4200772    2.658865            DAZAP1
304               P30084 0.0172204543 0.4200772    2.657028             ECHS1
305               P07711 0.0172449051 0.4200772    2.656333              CTSL
306               Q92870 0.0172613945 0.4200772   -2.655865             APBB2
307               Q9NXN4 0.0173804626 0.4200772    2.652496             GDAP2
308               P04632 0.0174003179 0.4200772   -2.651936            CAPNS1
309               P61966 0.0174125749 0.4200772    2.651591             AP1S1
310               P09960 0.0174506632 0.4200772    2.650520             LTA4H
311               Q6IA17 0.0174513785 0.4200772   -2.650500            SIGIRR
312               P63208 0.0174646824 0.4200772   -2.650127              SKP1
313               Q9C0C2 0.0174794748 0.4200772   -2.649712          TNKS1BP1
314               Q09666 0.0176073873 0.4200772   -2.646137             AHNAK
315               P34896 0.0176194174 0.4200772    2.645802             SHMT1
316               Q92843 0.0176382451 0.4200772   -2.645278            BCL2L2
317               Q6NZ67 0.0177171565 0.4200772   -2.643089             MZT2B
318               Q9BQ95 0.0177827309 0.4200772    2.641276             ECSIT
319               P62736 0.0178451764 0.4200772   -2.639557             ACTA2
320               Q9H2J4 0.0178491334 0.4200772    2.639448             PDCL3
321               Q03426 0.0179408679 0.4200772    2.636932               MVK
322               Q16651 0.0181188288 0.4200772   -2.632087             PRSS8
323               O60760 0.0181982553 0.4200772    2.629940             HPGDS
324               O43837 0.0182043000 0.4200772    2.629777             IDH3B
325               Q9Y6A9 0.0182378081 0.4200772    2.628873             SPCS1
326               O95347 0.0182780421 0.4200772   -2.627791              SMC2
327               P32019 0.0183255122 0.4200772   -2.626517            INPP5B
328               Q96BJ3 0.0183539529 0.4200772   -2.625755              AIDA
329               Q14151 0.0184009579 0.4200772   -2.624499             SAFB2
330               P51149 0.0184028756 0.4200772    2.624448             RAB7A
331               A1A4S6 0.0184272934 0.4200772    2.623796          ARHGAP10
332               Q01804 0.0185201507 0.4200772   -2.621327             OTUD4
333               Q8TB24 0.0185541065 0.4200772    2.620426              RIN3
334               Q9NRA2 0.0186062416 0.4200772    2.619047           SLC17A5
335               Q9Y4D1 0.0186618302 0.4200772   -2.617581             DAAM1
336               Q13464 0.0186894586 0.4200772   -2.616854             ROCK1
337               Q9Y2Z4 0.0187362264 0.4200772    2.615625             YARS2
338               Q9Y6Y8 0.0187662084 0.4200772    2.614839           SEC23IP
339               P80723 0.0188163807 0.4200772   -2.613526             BASP1
340               Q86UP2 0.0188897617 0.4200772   -2.611612              KTN1
341               Q14202 0.0189003882 0.4200772   -2.611335             ZMYM3
342               Q04695 0.0190073868 0.4200772   -2.608558             KRT17
343               P06454 0.0190941216 0.4200772   -2.606318              PTMA
344               Q07075 0.0191169787 0.4200772   -2.605730             ENPEP
345               Q9NPF4 0.0191298934 0.4200772    2.605397             OSGEP
346               Q99643 0.0191313843 0.4200772    2.605359              SDHC
347               P38571 0.0192367597 0.4200772    2.602655              LIPA
348               Q9Y2P8 0.0192931038 0.4200772    2.601216              RCL1
349               P16949 0.0194541253 0.4200772   -2.597123             STMN1
350               Q99797 0.0195127345 0.4200772    2.595642             MIPEP
351               Q86W92 0.0195209701 0.4200772   -2.595434           PPFIBP1
352               Q9BRR8 0.0195977909 0.4200772    2.593500           GPATCH1
353               P18206 0.0196611509 0.4200772   -2.591909               VCL
354               Q9H7B2 0.0197331880 0.4200772    2.590107              RPF2
355               P22897 0.0197437109 0.4200772    2.589845              MRC1
356               Q9H799 0.0197578081 0.4200772   -2.589493           CPLANE1
357               Q8N766 0.0199267474 0.4200772    2.585296              EMC1
358               P02649 0.0199736425 0.4200772   -2.584137              APOE
359               Q969Q5 0.0200087109 0.4200772    2.583272             RAB24
360               Q6PI78 0.0200296075 0.4200772   -2.582758            TMEM65
361               P20810 0.0200827506 0.4200772   -2.581451              CAST
362               Q9BU89 0.0203036591 0.4200772   -2.576055              DOHH
363               Q7L5D6 0.0204120577 0.4200772    2.573427              GET4
364               Q15165 0.0204359580 0.4200772    2.572850              PON2
365               P09382 0.0205421735 0.4200772   -2.570291            LGALS1
366               Q8WVK2 0.0205861287 0.4200772    2.569235           SNRNP27
367               Q9H930 0.0209177603 0.4200772   -2.561342            SP140L
368               P17568 0.0211789733 0.4200772    2.555209            NDUFB7
369               Q9Y3A6 0.0212218930 0.4200772    2.554208             TMED5
370               Q02252 0.0212449798 0.4200772    2.553671           ALDH6A1
371               Q5RI15 0.0212533538 0.4200772    2.553476             COX20
372               O14737 0.0213997501 0.4200772   -2.550081             PDCD5
373               Q10589 0.0214216920 0.4200772   -2.549574              BST2
374               Q00722 0.0214243364 0.4200772    2.549513             PLCB2
375               Q93099 0.0214692134 0.4200772   -2.548478               HGD
376               P05362 0.0214811849 0.4200772   -2.548202             ICAM1
377               O00273 0.0215674246 0.4200772   -2.546220              DFFA
378               P01732 0.0216101048 0.4200772    2.545242              CD8A
379               O75394 0.0216857111 0.4200772    2.543513            MRPL33
380               O60245 0.0217208112 0.4200772   -2.542713             PCDH7
381               Q86UL3 0.0217515002 0.4200772    2.542014             GPAT4
382               P38606 0.0217858567 0.4200772    2.541233           ATP6V1A
383               Q9UGQ3 0.0217936728 0.4200772    2.541055            SLC2A6
384               Q9UEW3 0.0218411180 0.4200772    2.539979             MARCO
385               P05141 0.0220204650 0.4200772    2.535929           SLC25A5
386               Q9H2D6 0.0221558015 0.4200772   -2.532894            TRIOBP
387               O43491 0.0221604932 0.4200772   -2.532789           EPB41L2
388               P09132 0.0222180118 0.4200772    2.531505             SRP19
389               O96009 0.0223189165 0.4200772    2.529259             NAPSA
390               Q15185 0.0223379708 0.4200772    2.528836            PTGES3
391               P56945 0.0223440195 0.4200772   -2.528702             BCAR1
392               Q9NVM1 0.0223639098 0.4200772   -2.528261             EVA1B
393               O95865 0.0223692013 0.4200772   -2.528144             DDAH2
394               P26440 0.0223733163 0.4200772    2.528053               IVD
395               Q9NWH9 0.0223924546 0.4200772   -2.527629              SLTM
396               P23368 0.0224451688 0.4200772    2.526463               ME2
397               Q53FV1 0.0224525875 0.4200772    2.526300            ORMDL2
398               Q9Y2S7 0.0224688086 0.4200772    2.525942           POLDIP2
399               Q13137 0.0224924716 0.4200772   -2.525420          CALCOCO2
400               Q92629 0.0225058572 0.4200772   -2.525125              SGCD
401               O75436 0.0225795897 0.4200772    2.523503            VPS26A
402               P15170 0.0226773610 0.4200772    2.521360             GSPT1
403               P27816 0.0227100941 0.4200772   -2.520645              MAP4
404               Q5JVS0 0.0227974475 0.4200772    2.518740             HABP4
405               P0C7P0 0.0228867215 0.4200772    2.516801             CISD3
406               Q16549 0.0230013045 0.4200772   -2.514323             PCSK7
407               Q6JBY9 0.0231220269 0.4200772   -2.511724             RCSD1
408               Q01082 0.0231233660 0.4200772   -2.511695            SPTBN1
409               P45880 0.0231369138 0.4200772    2.511405             VDAC2
410               Q8IUZ5 0.0231466172 0.4200772    2.511196            PHYKPL
411               Q9NY61 0.0231999856 0.4200772    2.510053              AATF
412               Q92947 0.0232558626 0.4200772    2.508858              GCDH
413               P84157 0.0232591025 0.4200772   -2.508789             MXRA7
414               P20962 0.0232691437 0.4200772   -2.508575              PTMS
415               Q7L2H7 0.0233831915 0.4200772    2.506146             EIF3M
416               Q5PRF9 0.0234206028 0.4200772   -2.505352            SAMD4B
417               Q9NX74 0.0234687986 0.4200772   -2.504331              DUS2
418               Q86UU1 0.0235210492 0.4200772   -2.503226            PHLDB1
419               P98170 0.0236811207 0.4200772    2.499855              XIAP
420               P20930 0.0239085731 0.4200772   -2.495103               FLG
421               O43633 0.0239091925 0.4200772   -2.495090            CHMP2A
422               O75410 0.0239109439 0.4200772   -2.495053             TACC1
423               Q86W42 0.0239187695 0.4200772    2.494891             THOC6
424               Q96EI5 0.0239601275 0.4200772   -2.494031            TCEAL4
425               Q7L591 0.0240069656 0.4200772    2.493060              DOK3
426               Q9H0V1 0.0240269495 0.4200772    2.492646           TMEM168
427               Q8WXE0 0.0240671047 0.4200772   -2.491816           CASKIN2
428               Q9Y3E0 0.0240930278 0.4200772    2.491280            GOLT1B
429               Q9UJ83 0.0241324403 0.4200772    2.490467             HACL1
430               P05165 0.0241886799 0.4200772    2.489309              PCCA
431               Q5D862 0.0242467542 0.4200772   -2.488115              FLG2
432               Q9UBC2 0.0242507106 0.4200772   -2.488034           EPS15L1
433               O14832 0.0242947287 0.4200772    2.487132              PHYH
434               Q6YN16 0.0243063406 0.4200772    2.486894             HSDL2
435               P48634 0.0244144678 0.4200772   -2.484684            PRRC2A
436               Q9H2M9 0.0244538675 0.4200772    2.483881          RAB3GAP2
437               Q15404 0.0245161442 0.4200772   -2.482615              RSU1
438               Q96CU9 0.0245743415 0.4200772    2.481434           FOXRED1
439               Q86VX2 0.0247110736 0.4200772    2.478671            COMMD7
440               Q9UK55 0.0247544898 0.4200772   -2.477796         SERPINA10
441               Q9Y2B9 0.0247970797 0.4200772   -2.476940              PKIG
442               Q5VWQ8 0.0248248602 0.4200772   -2.476382            DAB2IP
443               Q8NE86 0.0248496292 0.4200772    2.475885               MCU
444               Q8N5K1 0.0249477441 0.4200772    2.473921             CISD2
445               O15296 0.0251289125 0.4200772    2.470314           ALOX15B
446               Q9BX70 0.0251451615 0.4200772    2.469992             BTBD2
447               O60216 0.0252059361 0.4200772   -2.468788             RAD21
448               Q15375 0.0253827282 0.4200772    2.465302             EPHA7
449               Q6BCY4 0.0255600465 0.4200772    2.461829            CYB5R2
450               Q13813 0.0255957013 0.4200772   -2.461133            SPTAN1
451               Q01196 0.0256017061 0.4200772    2.461016             RUNX1
452               Q9NZT1 0.0256202000 0.4200772   -2.460656            CALML5
453               Q5SY16 0.0257113242 0.4200772    2.458884              NOL9
454               Q03001 0.0257805124 0.4200772   -2.457542               DST
455               Q12929 0.0259464086 0.4200772    2.454339              EPS8
456               P36551 0.0260263078 0.4200772    2.452803              CPOX
457               P07942 0.0261568891 0.4200772   -2.450303             LAMB1
458               Q9H845 0.0261698757 0.4200772    2.450055             ACAD9
459               O15121 0.0262791921 0.4200772    2.447972             DEGS1
460               Q9NQP4 0.0263434616 0.4200772   -2.446751             PFDN4
461               Q86YZ3 0.0265369325 0.4200772   -2.443093              HRNR
462               P51784 0.0265701166 0.4200772    2.442468             USP11
463               Q7Z6J4 0.0265920296 0.4200772    2.442056              FGD2
464               P41236 0.0266098582 0.4200772   -2.441720            PPP1R2
465               O75439 0.0267066651 0.4200772    2.439904             PMPCB
466               Q15532 0.0268644868 0.4200772   -2.436956              SS18
467               P37840 0.0269954576 0.4200772   -2.434522              SNCA
468               Q8NEY8 0.0270095983 0.4200772   -2.434260            PPHLN1
469               P45954 0.0270702836 0.4200772    2.433137            ACADSB
470               Q14247 0.0270916343 0.4200772   -2.432742              CTTN
471               Q9NQS1 0.0271846803 0.4200772   -2.431026              AVEN
472               Q9UJS0 0.0271851872 0.4200772    2.431016          SLC25A13
473               P11137 0.0272049927 0.4200772   -2.430652              MAP2
474               P55290 0.0272296210 0.4200772   -2.430198             CDH13
475               P22307 0.0272900681 0.4200772    2.429088              SCP2
476               Q9NXR1 0.0273466463 0.4200772   -2.428051              NDE1
477               P51398 0.0273563193 0.4200772    2.427874              DAP3
478               Q9NZI7 0.0273829853 0.4200772    2.427386              UBP1
479               P12235 0.0274492184 0.4200772    2.426175           SLC25A4
480               O60684 0.0275398599 0.4200772    2.424524             KPNA6
481               P05166 0.0275512450 0.4200772    2.424316              PCCB
482               Q8WYP3 0.0276310031 0.4200772   -2.422868              RIN2
483               Q9NQ88 0.0276620949 0.4200772    2.422304             TIGAR
484               Q70J99 0.0276744548 0.4200772   -2.422080            UNC13D
485               O75182 0.0277364617 0.4200772   -2.420959             SIN3B
486               Q13107 0.0277493333 0.4200772    2.420726              USP4
487               P28068 0.0277810619 0.4200772    2.420153           HLA-DMB
488               Q969M3 0.0277828067 0.4200772    2.420122             YIPF5
489               Q15365 0.0277923195 0.4200772    2.419950             PCBP1
490               Q8WZA0 0.0278771001 0.4200772   -2.418423              LZIC
491               O75607 0.0278914439 0.4200772    2.418165              NPM3
492               Q8IV08 0.0279611555 0.4200772    2.416913              PLD3
493               Q6WCQ1 0.0279925667 0.4200772   -2.416350             MPRIP
494               O15228 0.0279993062 0.4200772    2.416229             GNPAT
495               O60941 0.0280027574 0.4200772   -2.416167              DTNB
496               Q96RL1 0.0280939771 0.4200772   -2.414536             UIMC1
497               Q6P1X6 0.0281383572 0.4200772    2.413744           C8orf82
498               Q6ZNL6 0.0281656443 0.4200772   -2.413258              FGD5
499               O00148 0.0281677267 0.4200772    2.413221            DDX39A
500               P00973 0.0282334054 0.4200772    2.412052              OAS1
501               Q14CN2 0.0282335163 0.4200772   -2.412050             CLCA4
502               Q9H0L4 0.0282344813 0.4200772   -2.412033            CSTF2T
503               Q8IUH4 0.0282427752 0.4200772   -2.411886           ZDHHC13
504               Q9NQG5 0.0283080201 0.4200772   -2.410728            RPRD1B
505               Q96IX5 0.0283272138 0.4200772    2.410388            ATP5MK
506               P54198 0.0283388842 0.4200772    2.410181              HIRA
507               P14209 0.0283587748 0.4200772   -2.409829              CD99
508               Q96BR1 0.0285467586 0.4200772    2.406512              SGK3
509               Q96CD2 0.0286013041 0.4200772    2.405554             PPCDC
510               Q5TZA2 0.0286543443 0.4200772   -2.404623             CROCC
511               Q14802 0.0287314030 0.4200772   -2.403275             FXYD3
512               O43772 0.0287658327 0.4200772    2.402673          SLC25A20
513               O43299 0.0288227046 0.4200772    2.401681             AP5Z1
514               P21964 0.0289292024 0.4200772    2.399828              COMT
515               O43865 0.0289831751 0.4200772    2.398891            AHCYL1
516               Q9NSE4 0.0290727292 0.4200772    2.397341             IARS2
517               Q9Y3B7 0.0290925291 0.4200772    2.396998            MRPL11
518               Q7Z3U7 0.0291015163 0.4200772    2.396843              MON2
519               A6NJ78 0.0292290212 0.4200772    2.394645           METTL15
520               Q9BYD3 0.0292730820 0.4200772    2.393888             MRPL4
521               P51606 0.0292827589 0.4200772    2.393722             RENBP
522               Q9Y6N5 0.0293010244 0.4200772    2.393408              SQOR
523               O15018 0.0293648058 0.4200772   -2.392315             PDZD2
524               Q14980 0.0293652260 0.4200772   -2.392308             NUMA1
525               P19013 0.0293670888 0.4200772   -2.392276              KRT4
526               Q99614 0.0293795173 0.4200772   -2.392063              TTC1
527               Q15166 0.0294960015 0.4200772    2.390073              PON3
528               Q14103 0.0295099447 0.4200772   -2.389835            HNRNPD
529               P62861 0.0295731817 0.4200772    2.388758               FAU
530               Q9H7D7 0.0296533042 0.4200772    2.387397             WDR26
531               Q9BXL7 0.0296851603 0.4200772    2.386856            CARD11
532               O00291 0.0297069856 0.4200772   -2.386486              HIP1
533               Q53FT3 0.0297525006 0.4200772    2.385716           HIKESHI
534               Q9P0M6 0.0297568874 0.4200772    2.385642         MACROH2A2
535               P78559 0.0297937671 0.4200772   -2.385018             MAP1A
536               Q9Y2Q0 0.0298234613 0.4200772    2.384517            ATP8A1
537               O00425 0.0298325445 0.4200772   -2.384364           IGF2BP3
538               Q9BYB4 0.0299847228 0.4200772    2.381802             GNB1L
539               Q9H9H4 0.0300346982 0.4200772   -2.380963            VPS37B
540               Q13751 0.0300513869 0.4200772   -2.380684             LAMB3
541               O95183 0.0301952105 0.4200772   -2.378279             VAMP5
542               Q8NBF6 0.0302051776 0.4200772    2.378112              AVL9
543               Q14498 0.0302339712 0.4200772    2.377632             RBM39
544               P62820 0.0302462845 0.4200772    2.377427             RAB1A
545               Q8WV41 0.0302586482 0.4200772    2.377221             SNX33
546               Q9BYV8 0.0303268909 0.4200772   -2.376086             CEP41
547               Q7LGA3 0.0303326937 0.4200772    2.375990            HS2ST1
548               Q6ULP2 0.0303542549 0.4200772   -2.375632             AFTPH
549               Q717R9 0.0303617986 0.4200772   -2.375507              CYS1
550               P05091 0.0303745931 0.4200772    2.375294             ALDH2
551               Q9Y608 0.0303902775 0.4200772   -2.375034           LRRFIP2
552               P82673 0.0305144870 0.4200772    2.372979            MRPS35
553               O00462 0.0306128841 0.4200772    2.371356             MANBA
554               Q9BRR6 0.0306450790 0.4200772    2.370826             ADPGK
555               Q04726 0.0306552049 0.4200772   -2.370659              TLE3
556               Q4G0X4 0.0306882348 0.4200772    2.370116            KCTD21
557               Q92484 0.0307544885 0.4200772    2.369029           SMPDL3A
558               Q9H7Z7 0.0307941726 0.4200772    2.368378            PTGES2
559               Q9BZE1 0.0308286632 0.4200772    2.367814            MRPL37
560               O43272 0.0308656910 0.4200772    2.367208             PRODH
561               O00161 0.0308949633 0.4200772   -2.366730            SNAP23
562               Q9BRX8 0.0309200790 0.4200772    2.366320            PRXL2A
563               Q9NVD7 0.0311825068 0.4200772   -2.362056             PARVA
564               O96018 0.0312104091 0.4200772   -2.361604             APBA3
565               P46937 0.0312224496 0.4200772   -2.361410              YAP1
566               Q9Y485 0.0312307493 0.4200772    2.361275             DMXL1
567               Q53QV2 0.0312557234 0.4200772   -2.360872               LBH
568               Q96DH6 0.0312805985 0.4200772    2.360470              MSI2
569               Q9NY46 0.0313207309 0.4200772    2.359823             SCN3A
570               Q99828 0.0313920827 0.4200772   -2.358674              CIB1
571               Q6ZS17 0.0314621329 0.4200772   -2.357549            RIPOR1
572               O75367 0.0314953496 0.4200772    2.357016         MACROH2A1
573               P62310 0.0315548515 0.4200772   -2.356062              LSM3
574               Q9H5V9 0.0316173573 0.4200772    2.355063            STEEP1
575               Q6IQ49 0.0316192477 0.4200772   -2.355033              SDE2
576               Q8TB36 0.0317835988 0.4200772    2.352413             GDAP1
577               Q9BYE2 0.0318089055 0.4200772   -2.352011          TMPRSS13
578               Q6NZI2 0.0318670765 0.4200772   -2.351088            CAVIN1
579               P22695 0.0320027310 0.4200772    2.348941            UQCRC2
580               P01703 0.0320109582 0.4200772   -2.348811          IGLV1-40
581               Q9NWT6 0.0320536809 0.4200772    2.348136            HIF1AN
582               O60762 0.0320569218 0.4200772    2.348085              DPM1
583               Q8TBC5 0.0321187669 0.4200772   -2.347111           ZSCAN18
584               Q12846 0.0321289791 0.4200772   -2.346950              STX4
585 P0DP23;P0DP24;P0DP25 0.0321670893 0.4200772   -2.346351 CALM1;CALM2;CALM3
586               Q13561 0.0322224231 0.4200772   -2.345481             DCTN2
587               Q10713 0.0322292311 0.4200772    2.345375             PMPCA
588               Q8TDY2 0.0322805278 0.4200772   -2.344570            RB1CC1
589               P12532 0.0322854550 0.4200772   -2.344493            CKMT1B
590               O60437 0.0322906121 0.4200772   -2.344412               PPL
591               Q9BTU6 0.0323139387 0.4200772    2.344047            PI4K2A
592               O00443 0.0323561753 0.4200772    2.343386           PIK3C2A
593               Q9Y3E5 0.0323808336 0.4200772    2.343001             PTRH2
594               Q9NRL2 0.0325211526 0.4207492    2.340813             BAZ1A
595               Q8IYU8 0.0326310207 0.4207492    2.339106             MICU2
596               Q13287 0.0326673373 0.4207492    2.338543               NMI
597               Q5T749 0.0327301185 0.4207492   -2.337571              KPRP
598               P18827 0.0327372797 0.4207492   -2.337461              SDC1
599               P55212 0.0328552837 0.4207492    2.335639             CASP6
600               Q5SWX8 0.0329209218 0.4207492    2.334628              ODR4
601               Q8IVE3 0.0329733003 0.4207492   -2.333823           PLEKHH2
602               Q9UPN3 0.0331306196 0.4207492   -2.331412             MACF1
603               Q9NQ89 0.0332213596 0.4207492   -2.330026           C12orf4
604               Q96CN9 0.0332639651 0.4207492   -2.329377              GCC1
605               P21730 0.0332654623 0.4207492    2.329354             C5AR1
606               Q99714 0.0333749345 0.4207492    2.327689          HSD17B10
607               O43776 0.0333756945 0.4207492    2.327677             NARS1
608               Q01658 0.0333786783 0.4207492   -2.327632               DR1
609               P30626 0.0334077915 0.4207492   -2.327190               SRI
610               Q709C8 0.0334472656 0.4207492    2.326592            VPS13C
611               Q86WU2 0.0334489798 0.4207492    2.326566              LDHD
612               O95817 0.0334863165 0.4207492   -2.326000              BAG3
613               Q9Y487 0.0335972696 0.4207492    2.324323          ATP6V0A2
614               Q96BP3 0.0336323801 0.4207492    2.323793             PPWD1
615               O75832 0.0336794669 0.4207492    2.323084            PSMD10
616               P14317 0.0337070346 0.4207492   -2.322669             HCLS1
617               Q9Y228 0.0337452564 0.4207492   -2.322094          TRAF3IP3
618               Q13740 0.0338413733 0.4212649   -2.320651             ALCAM
619               Q92995 0.0340392744 0.4225928   -2.317693             USP13
620               O75306 0.0340579135 0.4225928    2.317415            NDUFS2
621               Q14764 0.0342591266 0.4232106    2.314425               MVP
622               Q9UPN6 0.0342922060 0.4232106   -2.313935             SCAF8
623               Q9NVA1 0.0343045293 0.4232106    2.313752             UQCC1
624               Q9H2A7 0.0343484686 0.4232106    2.313102            CXCL16
625               Q9ULR0 0.0343827707 0.4232106   -2.312595              ISY1
626               Q6P4A8 0.0344673654 0.4235742    2.311347             PLBD1
627               Q02952 0.0345518198 0.4239348   -2.310104            AKAP12
628               Q9BRT3 0.0348955487 0.4247695   -2.305074             MIEN1
629               Q92600 0.0349494447 0.4247695    2.304289             CNOT9
630               Q8N3F0 0.0350893635 0.4247695   -2.302258             MTURN
631               Q9BVT8 0.0352569853 0.4247695   -2.299834             TMUB1
632               Q92945 0.0352813124 0.4247695   -2.299483             KHSRP
633               Q9P2M7 0.0353505143 0.4247695   -2.298487               CGN
634               Q8TEB1 0.0353625245 0.4247695   -2.298314            DCAF11
635               O00592 0.0354388212 0.4247695   -2.297217             PODXL
636               P49748 0.0354445876 0.4247695    2.297134            ACADVL
637               O15372 0.0354926069 0.4247695    2.296446             EIF3H
638               Q8NHG8 0.0355008857 0.4247695   -2.296327             ZNRF2
639               Q9Y375 0.0355258632 0.4247695    2.295969           NDUFAF1
640               Q5TEJ8 0.0355309089 0.4247695    2.295897           THEMIS2
641               Q8TBX8 0.0357882826 0.4247695    2.292223           PIP4K2C
642               P61077 0.0358243561 0.4247695    2.291710            UBE2D3
643               P78367 0.0358306359 0.4247695   -2.291620            NKX3-2
644               Q9UGM3 0.0359193961 0.4247695   -2.290360             DMBT1
645               Q58WW2 0.0359531036 0.4247695   -2.289883             DCAF6
646               P53634 0.0359606560 0.4247695    2.289776              CTSC
647               Q01469 0.0360072226 0.4247695    2.289117             FABP5
648               A1IGU5 0.0360094243 0.4247695   -2.289085          ARHGEF37
649        P0DI81;P0DI82 0.0360806165 0.4247695   -2.288079  TRAPPC2;TRAPPC2B
650               Q8N4A0 0.0360849193 0.4247695   -2.288019            GALNT4
651               Q9H0V9 0.0361211842 0.4247695    2.287507            LMAN2L
652               Q13330 0.0361770360 0.4247695    2.286720              MTA1
653               O43556 0.0362101810 0.4247695   -2.286253              SGCE
654               P60033 0.0362314775 0.4247695    2.285953              CD81
655               Q9NZC7 0.0362796259 0.4247695   -2.285277              WWOX
656               O75323 0.0362851364 0.4247695    2.285199          NIPSNAP2
657               P55084 0.0364086259 0.4247695    2.283468             HADHB
658               Q53LP3 0.0365335797 0.4247695   -2.281721            SOWAHC
659               Q9HCY8 0.0365396123 0.4247695   -2.281637           S100A14
660               Q8IWE2 0.0365438633 0.4247695   -2.281577          FAM114A1
661               Q9Y679 0.0365860268 0.4247695    2.280989              AUP1
662               Q8N8S7 0.0366396155 0.4247695   -2.280243              ENAH
663               Q04323 0.0366837194 0.4247695   -2.279630             UBXN1
664               Q32MZ4 0.0367763676 0.4247695   -2.278343           LRRFIP1
665               Q86WN1 0.0367927038 0.4247695   -2.278116            FCHSD1
666               P09497 0.0368274263 0.4247695   -2.277635              CLTB
667               P35609 0.0368467623 0.4247695   -2.277367             ACTN2
668               Q16831 0.0368836636 0.4247695    2.276857              UPP1
669               Q02338 0.0369626551 0.4250429    2.275765              BDH1
670               Q8TER5 0.0370904752 0.4258762   -2.274003          ARHGEF40
671               Q7Z6K5 0.0372504701 0.4264498   -2.271806             ARPIN
672               Q9H0G5 0.0372527178 0.4264498   -2.271775             NSRP1
673               Q99816 0.0373067364 0.4264498   -2.271036            TSG101
674               Q9GZN8 0.0374762950 0.4271629   -2.268720          C20orf27
675               Q5KU26 0.0374801722 0.4271629   -2.268667           COLEC12
676               Q16719 0.0376658408 0.4286440    2.266143              KYNU
677               P04259 0.0377865785 0.4288862   -2.264507             KRT6B
678               Q9Y520 0.0378612085 0.4288862   -2.263499            PRRC2C
679               P35241 0.0378936042 0.4288862   -2.263062               RDX
680               P29966 0.0379536246 0.4288862   -2.262253            MARCKS
681               Q13823 0.0379658776 0.4288862    2.262088              GNL2
682               Q5MNZ9 0.0380670664 0.4293987   -2.260727             WIPI1
683               Q96PE3 0.0382151689 0.4303006    2.258741            INPP4A
684               Q9Y4F3 0.0382588862 0.4303006   -2.258157             MARF1
685               Q9H0A8 0.0386029656 0.4335367    2.253576            COMMD4
686               O00233 0.0386645937 0.4335958   -2.252759             PSMD9
687               P24311 0.0388333521 0.4348544    2.250530             COX7B
688               Q96N66 0.0390204170 0.4363141    2.248069            MBOAT7
689               Q9H993 0.0392459014 0.4378236    2.245117             ARMT1
690               P25686 0.0392692472 0.4378236   -2.244812            DNAJB2
691               Q16363 0.0395739031 0.4402213   -2.240851             LAMA4
692               Q96EE3 0.0395987473 0.4402213    2.240529             SEH1L
693               Q13136 0.0396779657 0.4404655   -2.239505            PPFIA1
694               A5D8V6 0.0398512803 0.4413290   -2.237269            VPS37C
695               Q92545 0.0398704867 0.4413290   -2.237022           TMEM131
696               Q6P3X3 0.0399368996 0.4414290    2.236168             TTC27
697               Q13424 0.0400450157 0.4419890   -2.234781             SNTA1
698               Q96GM5 0.0402537370 0.4426798   -2.232113           SMARCD1
699               O60812 0.0402605609 0.4426798    2.232026          HNRNPCL1
700               Q9NRP2 0.0402967178 0.4426798    2.231565              CMC2
701               Q9Y697 0.0403377808 0.4426798    2.231043              NFS1
702               Q96GW9 0.0404978780 0.4435936    2.229009             MARS2
703               Q9H3Q1 0.0405363713 0.4435936   -2.228521          CDC42EP4
704               O14548 0.0406706212 0.4444305    2.226823           COX7A2L
705               A6NI72 0.0407344065 0.4444962    2.226017             NCF1B
706               Q9UHP3 0.0408239521 0.4448423    2.224889             USP25
707               O76070 0.0409766877 0.4458750   -2.222970              SNCG
708               Q8TER0 0.0411061540 0.4466520   -2.221349             SNED1
709               Q9UKM7 0.0412172651 0.4472277    2.219961            MAN1B1
710               P23528 0.0414717747 0.4474588    2.216795              CFL1
711               Q96E39 0.0415098779 0.4474588   -2.216322            RBMXL1
712               Q13753 0.0415232579 0.4474588   -2.216157             LAMC2
713               O95352 0.0415411332 0.4474588    2.215935              ATG7
714               P54253 0.0415777587 0.4474588   -2.215482             ATXN1
715               O00483 0.0415875499 0.4474588    2.215361            NDUFA4
716               Q8TE02 0.0417522140 0.4480004   -2.213327              ELP5
717               Q01973 0.0417585159 0.4480004   -2.213249              ROR1
718               O94855 0.0418125961 0.4480004    2.212583            SEC24D
719               Q9H0W5 0.0419077831 0.4483565   -2.211413             CCDC8
720               P28799 0.0419623892 0.4483565    2.210742               GRN
721               Q9UH65 0.0421410639 0.4487424   -2.208554            SWAP70
722               Q9H2V7 0.0421432605 0.4487424    2.208527             SPNS1
723               Q99942 0.0421948604 0.4487424    2.207897              RNF5
724               P22304 0.0422318345 0.4487424    2.207446               IDS
725               P55199 0.0423778153 0.4496725    2.205668               ELL
726               Q5JRX3 0.0425948618 0.4513530    2.203036            PITRM1
727               Q96RK0 0.0427401974 0.4519204   -2.201280               CIC
728               P42765 0.0428486610 0.4519204    2.199973             ACAA2
729               Q0VGL1 0.0428535994 0.4519204    2.199913           LAMTOR4
730               Q6UWP8 0.0428833895 0.4519204   -2.199555              SBSN
731               Q5ST30 0.0430128547 0.4524871    2.198000             VARS2
732               Q9BPW8 0.0430933610 0.4524871    2.197036          NIPSNAP1
733               Q9Y3C6 0.0431136192 0.4524871    2.196793             PPIL1
734               P05067 0.0432340051 0.4529832   -2.195355               APP
735               P35606 0.0433593398 0.4529832    2.193861             COPB2
736               P08648 0.0433888471 0.4529832   -2.193509             ITGA5
737               Q9HBL7 0.0434411810 0.4529832    2.192887            PLGRKT
738               Q14517 0.0434552925 0.4529832   -2.192720              FAT1
739               Q10570 0.0435869204 0.4536528    2.191158             CPSF1
740               Q9H6R3 0.0436605972 0.4536528    2.190286             ACSS3
741               Q9BQ13 0.0436964394 0.4536528    2.189862            KCTD14
742               Q9NS56 0.0437818834 0.4539273    2.188853            TOPORS
743               P55036 0.0438518287 0.4540405   -2.188029             PSMD4
744               Q8NDA8 0.0439852408 0.4542013    2.186460             MROH1
745               Q9UBR2 0.0440351641 0.4542013    2.185873              CTSZ
746               O95825 0.0440444810 0.4542013    2.185764            CRYZL1
747               P39880 0.0442210183 0.4552129   -2.183697              CUX1
748               Q93091 0.0442861468 0.4552129    2.182936            RNASE6
749               Q02224 0.0443200925 0.4552129   -2.182540             CENPE
750               O95400 0.0445004827 0.4560292   -2.180440            CD2BP2
751               Q9H773 0.0445181208 0.4560292   -2.180235            DCTPP1
752               P63010 0.0447021041 0.4573049    2.178101             AP2B1
753               Q9HCC0 0.0448007156 0.4577051    2.176961             MCCC2
754               Q5SRE7 0.0449286454 0.4581834    2.175486            PHYHD1
755               Q14790 0.0450156999 0.4581834    2.174484             CASP8
756               Q8TB40 0.0450964911 0.4581834    2.173556             ABHD4
757               Q9H6L2 0.0451300676 0.4581834   -2.173171           TMEM231
758               P04921 0.0451453245 0.4581834   -2.172996              GYPC
759               P07099 0.0453201295 0.4591897    2.170995             EPHX1
760               P14314 0.0454553683 0.4591897   -2.169451            PRKCSH
761               Q9HB90 0.0454655006 0.4591897    2.169336             RRAGC
762               Q69YQ0 0.0454832396 0.4591897   -2.169134           SPECC1L
763               P37235 0.0457173244 0.4602851   -2.166474            HPCAL1
764               Q86YB7 0.0457265657 0.4602851    2.166369            ECHDC2
765               P27482 0.0458216897 0.4602851   -2.165292            CALML3
766               Q8IZV5 0.0458312575 0.4602851    2.165184             RDH10
767               O43427 0.0459200665 0.4602851    2.164181              FIBP
768               Q8N653 0.0459507307 0.4602851    2.163835             LZTR1
769               Q9GZV5 0.0460502414 0.4606821   -2.162713             WWTR1
770               P63098 0.0463451003 0.4629918   -2.159403            PPP3R1
771               Q9NP81 0.0466672324 0.4629918    2.155809             SARS2
772               O00221 0.0466681306 0.4629918    2.155799            NFKBIE
773               Q9H410 0.0466842784 0.4629918   -2.155619              DSN1
774               O94768 0.0466843294 0.4629918    2.155618            STK17B
775               Q9P2B4 0.0466893718 0.4629918   -2.155562         CTTNBP2NL
776               Q14642 0.0467250489 0.4629918   -2.155166            INPP5A
777               Q9Y3A4 0.0467625907 0.4629918    2.154749             RRP7A
778               Q01459 0.0468262171 0.4630258    2.154043              CTBS
779               P60660 0.0470039884 0.4632256   -2.152075              MYL6
780               Q11201 0.0470399775 0.4632256    2.151678           ST3GAL1
781               O75146 0.0471919423 0.4632256   -2.150003             HIP1R
782               Q16706 0.0472691167 0.4632256    2.149154            MAN2A1
783               P49279 0.0472770384 0.4632256    2.149067           SLC11A1
784               Q9H0J9 0.0473879560 0.4632256    2.147849            PARP12
785               P29350 0.0474104377 0.4632256    2.147602             PTPN6
786               P80303 0.0474111699 0.4632256   -2.147594             NUCB2
787               P00441 0.0474542749 0.4632256   -2.147122              SOD1
788               Q5EBL4 0.0474828875 0.4632256   -2.146809            RILPL1
789               Q14019 0.0475288266 0.4632256    2.146306             COTL1
790               Q9Y2H5 0.0475689833 0.4632256   -2.145867           PLEKHA6
791               P20827 0.0477459281 0.4643608   -2.143937             EFNA1
792               Q8NDH2 0.0480902381 0.4663387   -2.140199           CCDC168
793               P08670 0.0480946963 0.4663387   -2.140151               VIM
794               O75935 0.0481795546 0.4663387   -2.139234             DCTN3
795               Q96G23 0.0482169568 0.4663387    2.138830             CERS2
796               Q9NZD8 0.0482523898 0.4663387    2.138448             SPG21
797               P17661 0.0483354710 0.4665556   -2.137552               DES
798               P16070 0.0485151109 0.4672922   -2.135621              CD44
799               Q02083 0.0485478869 0.4672922    2.135270              NAAA
800               Q92805 0.0485940118 0.4672922   -2.134775            GOLGA1
801               Q9UDY8 0.0487550934 0.4682558    2.133052             MALT1
802               P24844 0.0488227424 0.4683209   -2.132329              MYL9
803               Q9NVM9 0.0489564656 0.4690188    2.130904            INTS13
804               Q8IX01 0.0491801740 0.4695779   -2.128528             SUGP2
805               P56377 0.0492016154 0.4695779    2.128301             AP1S2
806               Q969G5 0.0492596237 0.4695779   -2.127687            CAVIN3
807               Q7Z404 0.0492748234 0.4695779    2.127526              TMC4
808               Q8N6H7 0.0493631577 0.4695779   -2.126592           ARFGAP2
809               P17900 0.0493810686 0.4695779    2.126403              GM2A
810               P14649 0.0496956593 0.4701202   -2.123091             MYL6B
811               O14972 0.0497089815 0.4701202    2.122951            VPS26C
812               P12694 0.0497290451 0.4701202    2.122741            BCKDHA
813               P0CAP2 0.0497523353 0.4701202   -2.122497            POLR2M
814               O15514 0.0499237798 0.4701202    2.120701            POLR2D</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(proDAProNormalRes, dplyr::desc(t_statistic))$name
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Normal Tissue proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-50-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-50-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-50-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Assess data power by p-value histogram</p>
<pre class="r"><code>p &lt;- all_proDAProNormalRes$pval
hist(p, xlab = &#39;P-value&#39;, main = &#39;Normal Tissue Proteomics (proDA)&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-51-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-51-1">
Past versions of unnamed-chunk-51-1.png
</button>
</p>
<div id="fig-unnamed-chunk-51-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-51-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-51-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/5c9ec0be03a8fabd05de4e2fa3a641d8ec0ee09c/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-51-1.png" target="_blank">5c9ec0b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Visualize data of top 6 significant features based on proDA
result</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- proDAProNormalRes[1:6,] %&gt;%
  dplyr::select(name, pval, Gene) %&gt;%
  dplyr::rename(Feature = name, pVal = pval)
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats$Feature,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = paste0(Gene, &#39;\n(p = &#39;, round(pVal, 4), &#39;)&#39;),
                Gene = factor(Gene, levels = unique(Gene))) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-52-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal   pValAdj      Stat   Test
1  PC1 Recurrence 0.01334493 0.1962674  2.781342 T-test
2  PC7 Recurrence 0.02453343 0.1962674 -2.482264 T-test</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes@sDev^2 / sum(pcaRes@sDev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Normal tissue proteomics&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-53-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-53-1">
Past versions of unnamed-chunk-53-1.png
</button>
</p>
<div id="fig-unnamed-chunk-53-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-53-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/389d24e2c5ebaa7ec54db3cf6ad0a3223fd918d0/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-53-1.png" target="_blank">389d24e</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-06-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>PC1 potentially separates recurrence and non-recurrence patients.</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC1, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Normal Tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-54-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-54-1">
Past versions of unnamed-chunk-54-1.png
</button>
</p>
<div id="fig-unnamed-chunk-54-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/128dd12ec84dc9456c33f1481db143f6784d230d/docs/figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-54-1.png" target="_blank">128dd12</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot molecular signatures captured by PC1 in input data. We took a
look at top 20 features with highest loadings and observed protein
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC1$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
# Change protein names to gene names
datMatSub &lt;- datMat[topFeatsIdx, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Normal Tissue proteomics - PC1&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-55-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display top features that build the PC1</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
dplyr::mutate(pcTopFeatTab$PC1, Gene = plyr::mapvalues(Feature,
                                                       from = featInfo$PG.Groups,
                                                       to = featInfo$PG.Genes,
                                                       warn_missing = F))</code></pre>
<pre><code>                Feature    Loading              Gene
1                Q9P2R6 0.08231263              RERE
2                Q53QV2 0.07576679               LBH
3                Q9H4X1 0.07124872              RGCC
4                P27482 0.06309854            CALML3
5                P03973 0.06243626              SLPI
6  P0DP23;P0DP24;P0DP25 0.06221882 CALM1;CALM2;CALM3
7                Q9BXV9 0.06107817              GON7
8                P12829 0.05798979              MYL4
9                Q5VWX1 0.05673529           KHDRBS2
10               Q13478 0.05654864            IL18R1
11               Q8N6G6 0.05491788          ADAMTSL1
12               P24530 0.05336776             EDNRB
13               O60220 0.05336765            TIMM8A
14               P15882 0.05309158              CHN1
15               P02741 0.05304892               CRP
16               A5D8V6 0.05245396            VPS37C
17               O00559 0.05213557             EBAG9
18               P84157 0.04864494             MXRA7
19               Q9NQA3 0.04835324            WASH6P
20               Q13790 0.04798633              APOF
21               Q8TBC3 0.04780530            SHKBP1
22               Q9NZD4 0.04742926              AHSP
23               P62760 0.04671614             VSNL1
24               P06702 0.04622401            S100A9
25               Q9NSB2 0.04618957             KRT84
26               P69892 0.04605006              HBG2
27               O75147 0.04582310             OBSL1
28               P11684 0.04576260           SCGB1A1
29               P07998 0.04565600            RNASE1
30               Q9H0I9 0.04532237             TKTL2</code></pre>
<p>Another significant PC7</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC7, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Normal Tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-57-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC7$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
# Change protein names to gene names
datMatSub &lt;- datMat[topFeatsIdx, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Normal Tissue proteomics - PC7&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-57-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
dplyr::mutate(pcTopFeatTab$PC7, Gene = plyr::mapvalues(Feature,
                                                       from = featInfo$PG.Groups,
                                                       to = featInfo$PG.Genes,
                                                       warn_missing = F))</code></pre>
<pre><code>   Feature    Loading    Gene
1   Q8TDL5 0.15234253  BPIFB1
2   Q86WV1 0.11720359   SKAP1
3   Q9P2R6 0.10859581    RERE
4   Q8N0Z6 0.09226436    TTC5
5   Q7Z589 0.09064812    EMSY
6   Q14722 0.09039718  KCNAB1
7   P02741 0.07996952     CRP
8   O15217 0.07791076   GSTA4
9   Q69YH5 0.07747341   CDCA2
10  Q6PII3 0.07185938 CCDC174
11  P69892 0.06625934    HBG2
12  Q6JQN1 0.06445783  ACAD10
13  Q9H981 0.06369119   ACTR8
14  Q9HC52 0.06340973    CBX8
15  Q5H9U9 0.06255500  DDX60L
16  Q8NHJ6 0.06193516  LILRB4
17  Q7Z6I8 0.06116830 C5orf24
18  P16050 0.06070903  ALOX15
19  P09693 0.05777846    CD3G
20  O43914 0.05658950  TYROBP
21  Q01196 0.05609539   RUNX1
22  A6NKN8 0.05545390  PCP4L1
23  Q6ZS72 0.05483372   PEAK3
24  Q8TBC3 0.05460845  SHKBP1
25  Q6P2P2 0.05451845   PRMT9
26  Q7Z6K3 0.05345797   PTAR1
27  P01880 0.05181121    IGHD
28  Q6P4A8 0.05172607   PLBD1
29  Q15652 0.05143023  JMJD1C
30  Q6PCB8 0.05141833     EMB</code></pre>
</div>
<div id="difference-in-tissue-conditions" class="section level2">
<h2>Difference in tissue conditions</h2>
<p>Differences between Tumor and Normal samples (Tumor matrix - Normal
matrix):<br />
</p>
<p>Display significant associations between learned PCs (Var1) and
metadata variables (Var2)</p>
<pre class="r"><code># Subset tumor samples
smpTumorIdx &lt;- which(colData(proTissue)$Condition == &#39;Tumor&#39;)
proTumor &lt;- proTissue[, smpTumorIdx]
# Subset normal samples
smpNormalIdx &lt;- which(colData(proTissue)$Condition == &#39;Normal&#39;)
proNormal &lt;- proTissue[, smpNormalIdx]
# Modify column names for matching two matrices
colnames(proTumor) &lt;- stringr::str_remove(colnames(proTumor), &#39;_TG&#39;)
colnames(proNormal) &lt;- stringr::str_remove(colnames(proNormal), &#39;_NG&#39;)
# Retrieve data matrix
proTumorMat &lt;- SummarizedExperiment::assay(proTumor)
proNormalMat &lt;- SummarizedExperiment::assay(proNormal)
# Subtract normal matrix by tumor matrix
if (identical(colnames(proTumorMat), colnames(proNormalMat)) &amp;
    identical(rownames(proTumorMat), rownames(proNormalMat))) {
  proTissueDiffMat &lt;- proTumorMat - proNormalMat
} else {
  print(&#39;Align samples or features first.&#39;)
}
# Retrieve samples metadata for creating SE object
colAnno &lt;- as.data.frame(colData(proTumor))
proTissueDiff &lt;- SummarizedExperiment(assays = proTissueDiffMat, colData = colAnno)

# Display significant associations between PCs and metadata variables
proTissueDiffAssoRes &lt;- doSOA(proTissueDiff, meta_var = c(&#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;,
                                                          &#39;Smoking&#39;, &#39;Stage&#39;, &#39;Metastasis&#39;),
                              do_onlyPCA = T, pca_method = &#39;ppca&#39;, num_PCs = 16)
proTissueDiffAssoRes$tPCASigRes</code></pre>
<pre><code>  Var1    Var2        pVal   pValAdj      Stat   Test
1 PC16   Stage 0.006559593 0.6297210  7.160350  ANOVA
2 PC10  Gender 0.015046298 0.7222223 -2.722979 T-test
3 PC15  Gender 0.047072551 0.8771696 -2.151318 T-test
4 PC15 Smoking 0.049886993 0.8771696  3.685694  ANOVA</code></pre>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Perform analysis
proTissueDiffRes &lt;- doSOA(proTissueDiff, meta_var = &#39;Recurrence&#39;, pca_method = &#39;ppca&#39;,
                          num_PCs = 16, num_PCfeats = 30)
datMat &lt;- proTissueDiffRes$data
smpAnno &lt;- proTissueDiffRes$smpMetadata
tFeatSigRes &lt;- proTissueDiffRes$tFeatSigRes
tPCASigRes &lt;- proTissueDiffRes$tPCASigRes

# Display features that can significantly separate recurrence and non-recurrence
# patients
dplyr::mutate(tFeatSigRes, Gene = plyr::mapvalues(Var1,
                                                  from = featInfo$PG.Groups,
                                                  to = featInfo$PG.Genes,
                                                  warn_missing = F))</code></pre>
<pre><code>          Var1       Var2         pVal   pValAdj      Stat   Test     Gene
1       Q9Y3Q0 Recurrence 0.0001865862 0.9999116 -6.507691 T-test  NAALAD2
2       Q02413 Recurrence 0.0004520629 0.9999116 -4.467039 T-test     DSG1
3       Q92843 Recurrence 0.0007984407 0.9999116 -4.342187 T-test   BCL2L2
4       Q6NXT1 Recurrence 0.0009629195 0.9999116 -4.159845 T-test  ANKRD54
5       Q9BXB4 Recurrence 0.0011833145 0.9999116  3.989928 T-test  OSBPL11
6       Q9Y3B8 Recurrence 0.0011992203 0.9999116 -3.928637 T-test    REXO2
7       P13647 Recurrence 0.0012991919 0.9999116 -3.890621 T-test     KRT5
8       Q9BQL6 Recurrence 0.0013920883 0.9999116 -3.857852 T-test   FERMT1
9       Q96C90 Recurrence 0.0014739497 0.9999116 -3.830752 T-test PPP1R14B
10      Q5TAQ9 Recurrence 0.0018170450 0.9999116  3.731585 T-test    DCAF8
11      Q7Z794 Recurrence 0.0018745618 0.9999116 -3.716826 T-test    KRT77
12      Q93063 Recurrence 0.0020060381 0.9999116  3.731362 T-test     EXT2
13      Q7Z3J2 Recurrence 0.0022094322 0.9999116  3.639007 T-test   VPS35L
14      Q99417 Recurrence 0.0024028672 0.9999116 -3.599280 T-test    MYCBP
15      Q6P4A8 Recurrence 0.0026699048 0.9999116  3.549398 T-test    PLBD1
16      P02538 Recurrence 0.0027883797 0.9999116 -3.528845 T-test    KRT6A
17      Q8N6H7 Recurrence 0.0031740386 0.9999116 -3.467508 T-test  ARFGAP2
18      P20930 Recurrence 0.0032278370 0.9999116 -3.600651 T-test      FLG
19      Q9UQN3 Recurrence 0.0032406559 0.9999116 -3.457671 T-test   CHMP2B
20      Q8TE73 Recurrence 0.0038260745 0.9999116 -3.743244 T-test    DNAH5
21      O60583 Recurrence 0.0041716068 0.9999116  3.960928 T-test    CCNT2
22      Q96BP3 Recurrence 0.0043941302 0.9999116  3.313332 T-test    PPWD1
23      Q6UWP8 Recurrence 0.0047939722 0.9999116 -3.451243 T-test     SBSN
24      Q5T749 Recurrence 0.0052424884 0.9999116 -3.347835 T-test     KPRP
25      Q96EQ0 Recurrence 0.0054317989 0.9999116 -3.245623 T-test     SGTB
26      O75190 Recurrence 0.0060442808 0.9999116 -3.161751 T-test   DNAJB6
27      Q9GZN8 Recurrence 0.0068577806 0.9999116 -3.101536 T-test C20orf27
28      Q92851 Recurrence 0.0071461481 0.9999116  3.081866 T-test   CASP10
29      Q9Y4K0 Recurrence 0.0071584115 0.9999116 -3.460212 T-test    LOXL2
30      P62195 Recurrence 0.0075845531 0.9999116  3.053406 T-test    PSMC5
31      Q9BRX8 Recurrence 0.0076516373 0.9999116  3.049194 T-test   PRXL2A
32      P08590 Recurrence 0.0080722101 0.9999116 -3.023585 T-test     MYL3
33      Q5JVS0 Recurrence 0.0084600328 0.9999116  3.468608 T-test    HABP4
34      P03915 Recurrence 0.0085072897 0.9999116 -3.141556 T-test   MT-ND5
35      P24821 Recurrence 0.0087658165 0.9999116 -2.984075 T-test      TNC
36      O75382 Recurrence 0.0090584292 0.9999116 -2.968316 T-test    TRIM3
37      P81605 Recurrence 0.0093983971 0.9999116 -2.950619 T-test      DCD
38      Q9NZL9 Recurrence 0.0096731474 0.9999116  2.936768 T-test    MAT2B
39      P20340 Recurrence 0.0099735679 0.9999116 -2.922055 T-test    RAB6A
40      O60341 Recurrence 0.0100611912 0.9999116  2.917845 T-test    KDM1A
41      K7EJ46 Recurrence 0.0101698422 0.9999116 -3.096368 T-test   SMIM22
42      Q08999 Recurrence 0.0102287828 0.9999116  4.574128 T-test     RBL2
43      Q96RQ1 Recurrence 0.0102499913 0.9999116  2.908894 T-test   ERGIC2
44      O94933 Recurrence 0.0102919732 0.9999116 -3.089678 T-test  SLITRK3
45      P54687 Recurrence 0.0104137019 0.9999116  2.901262 T-test    BCAT1
46      P02533 Recurrence 0.0109076111 0.9999116 -2.878922 T-test    KRT14
47      O14578 Recurrence 0.0113543400 0.9999116 -2.912660 T-test      CIT
48      P09327 Recurrence 0.0118760108 0.9999116  4.379881 T-test     VIL1
49      Q96HR8 Recurrence 0.0119832514 0.9999116  3.004492 T-test     NAF1
50      Q5SY16 Recurrence 0.0120277180 0.9999116  2.831699 T-test     NOL9
51      Q9H930 Recurrence 0.0120662504 0.9999116 -2.830152 T-test   SP140L
52      Q9UMZ2 Recurrence 0.0120865880 0.9999116  2.829337 T-test    SYNRG
53      Q99983 Recurrence 0.0125209468 0.9999116 -4.312535 T-test      OMD
54      Q9NVZ3 Recurrence 0.0125690638 0.9999116  2.810387 T-test   NECAP2
55      Q6P4Q7 Recurrence 0.0126261289 0.9999116 -2.808192 T-test    CNNM4
56      O96019 Recurrence 0.0133878609 0.9999116  2.779783 T-test   ACTL6A
57      P22735 Recurrence 0.0147363569 0.9999116 -2.888841 T-test     TGM1
58      A5D8V6 Recurrence 0.0151252030 0.9999116 -2.720429 T-test   VPS37C
59      Q96GS4 Recurrence 0.0151767707 0.9999116 -2.718770 T-test   BORCS6
60      Q9Y5G0 Recurrence 0.0153062525 0.9999116 -2.919779 T-test  PCDHGB5
61      Q8NBM4 Recurrence 0.0157656744 0.9999116  2.775014 T-test    UBAC2
62      Q14165 Recurrence 0.0158604970 0.9999116  2.697268 T-test     MLEC
63      Q5S007 Recurrence 0.0161937045 0.9999116  2.687110 T-test    LRRK2
64      O15195 Recurrence 0.0163010424 0.9999116 -2.757547 T-test     VILL
65      Q2TB90 Recurrence 0.0163897225 0.9999116  2.726301 T-test    HKDC1
66      Q2M1Z3 Recurrence 0.0169713620 0.9999116 -2.684698 T-test ARHGAP31
67      O15120 Recurrence 0.0170754165 0.9999116  2.681646 T-test   AGPAT2
68      Q14142 Recurrence 0.0172262233 0.9999116  2.801573 T-test   TRIM14
69      P39656 Recurrence 0.0172969616 0.9999116  2.654856 T-test    DDOST
70      Q99856 Recurrence 0.0174833656 0.9999116 -2.984373 T-test   ARID3A
71      Q13823 Recurrence 0.0175135421 0.9999116  2.719981 T-test     GNL2
72      Q9BWH6 Recurrence 0.0176963481 0.9999116  2.747123 T-test    RPAP1
73      Q6P6C2 Recurrence 0.0178806293 0.9999116  2.638583 T-test   ALKBH5
74      Q9H7D7 Recurrence 0.0181773741 0.9999116  2.630503 T-test    WDR26
75      Q9BYE2 Recurrence 0.0182155486 0.9999116 -2.629473 T-test TMPRSS13
76      Q07617 Recurrence 0.0183297719 0.9999116 -2.669166 T-test    SPAG1
77      Q49A26 Recurrence 0.0187179320 0.9999116  2.616105 T-test    GLYR1
78      A6NKN8 Recurrence 0.0189148531 0.9999116 -2.679612 T-test   PCP4L1
79      P08134 Recurrence 0.0190822009 0.9999116 -2.606626 T-test     RHOC
80      P04259 Recurrence 0.0191731411 0.9999116 -2.604286 T-test    KRT6B
81      P03973 Recurrence 0.0194125886 0.9999116  2.617444 T-test     SLPI
82      Q9H2M9 Recurrence 0.0196636638 0.9999116  2.591846 T-test RAB3GAP2
83      Q99942 Recurrence 0.0199395010 0.9999116  2.682637 T-test     RNF5
84      P55199 Recurrence 0.0201355452 0.9999116  2.892056 T-test      ELL
85      O15226 Recurrence 0.0202263254 0.9999116 -2.644393 T-test     NKRF
86      P41208 Recurrence 0.0202431680 0.9999116 -2.577527 T-test    CETN2
87      O60907 Recurrence 0.0206652005 0.9999116  2.567343 T-test    TBL1X
88      Q15046 Recurrence 0.0207889659 0.9999116  2.564394 T-test    KARS1
89      A1X283 Recurrence 0.0208675915 0.9999116  2.562529 T-test SH3PXD2B
90      Q9P2N5 Recurrence 0.0209472975 0.9999116  2.560645 T-test    RBM27
91      O75629 Recurrence 0.0211554139 0.9999116  2.555759 T-test    CREG1
92      P36955 Recurrence 0.0213824280 0.9999116 -2.550482 T-test SERPINF1
93      Q6IA17 Recurrence 0.0218294601 0.9999116 -2.540243 T-test   SIGIRR
94      Q9UN79 Recurrence 0.0220420872 0.9999116 -2.535443 T-test    SOX13
95      Q9GZX9 Recurrence 0.0221396334 0.9999116 -2.625972 T-test    TWSG1
96      P05388 Recurrence 0.0222254021 0.9999116  2.531340 T-test    RPLP0
97      Q9H6F5 Recurrence 0.0225758308 0.9999116 -2.817639 T-test   CCDC86
98      P52789 Recurrence 0.0225867200 0.9999116  2.523346 T-test      HK2
99      Q8TEU7 Recurrence 0.0228779883 0.9999116  2.516990 T-test  RAPGEF6
100     Q6P4E1 Recurrence 0.0229811427 0.9999116 -2.514758 T-test    GOLM2
101     Q9NQP4 Recurrence 0.0229898009 0.9999116 -2.514571 T-test    PFDN4
102     Q9Y548 Recurrence 0.0230296336 0.9999116  2.681594 T-test    YIPF1
103     O43765 Recurrence 0.0238947507 0.9999116 -2.495390 T-test     SGTA
104     P52849 Recurrence 0.0242229806 0.9999116  2.505880 T-test    NDST2
105     Q8NB78 Recurrence 0.0243449949 0.9999116  2.574457 T-test    KDM1B
106     P21127 Recurrence 0.0243831032 0.9999116  2.485324 T-test   CDK11B
107     Q7L5L3 Recurrence 0.0244100691 0.9999116 -2.521915 T-test    GDPD3
108     Q9UGU0 Recurrence 0.0253214631 0.9999116  2.958826 T-test    TCF20
109     Q15375 Recurrence 0.0256503559 0.9999116  2.460069 T-test    EPHA7
110     Q15652 Recurrence 0.0257935797 0.9999116 -2.665938 T-test   JMJD1C
111     Q9NPH3 Recurrence 0.0258458704 0.9999116 -2.456278 T-test   IL1RAP
112     Q5D862 Recurrence 0.0259442266 0.9999116 -2.490380 T-test     FLG2
113     P54253 Recurrence 0.0260144609 0.9999116 -2.813511 T-test    ATXN1
114     Q9NSB2 Recurrence 0.0262397568 0.9999116 -2.605564 T-test    KRT84
115     Q10589 Recurrence 0.0264031174 0.9999116 -2.445620 T-test     BST2
116     Q9UKX5 Recurrence 0.0268834929 0.9999116 -2.436602 T-test   ITGA11
117     Q01196 Recurrence 0.0270868901 0.9999116  2.636089 T-test    RUNX1
118     O15405 Recurrence 0.0272813097 0.9999116  2.512520 T-test     TOX3
119     P20810 Recurrence 0.0273345452 0.9999116 -2.428272 T-test     CAST
120     Q9UNM6 Recurrence 0.0275992153 0.9999116  2.423445 T-test   PSMD13
121     P09493 Recurrence 0.0276257743 0.9999116 -2.422963 T-test     TPM1
122     O14618 Recurrence 0.0277580538 0.9999116 -2.420568 T-test      CCS
123     P58340 Recurrence 0.0277675168 0.9999116 -2.683673 T-test     MLF1
124     Q9Y320 Recurrence 0.0280670982 0.9999116 -2.415016 T-test     TMX2
125     Q92828 Recurrence 0.0280861460 0.9999116  2.414676 T-test   CORO2A
126     Q86WN1 Recurrence 0.0284704175 0.9999116 -2.407857 T-test   FCHSD1
127     Q92620 Recurrence 0.0288233139 0.9999116  2.401670 T-test    DHX38
128     P07919 Recurrence 0.0290870606 0.9999116 -2.397093 T-test    UQCRH
129     Q6P179 Recurrence 0.0298830471 0.9999116  2.437926 T-test    ERAP2
130 A0A0B4J1Y8 Recurrence 0.0298953359 0.9999116 -2.437707 T-test IGLV9-49
131     Q9NZI7 Recurrence 0.0299558394 0.9999116  2.397759 T-test     UBP1
132     Q66PJ3 Recurrence 0.0302349180 0.9999116  2.377617 T-test  ARL6IP4
133     Q7Z406 Recurrence 0.0302969756 0.9999116 -2.376584 T-test    MYH14
134     Q9BUL5 Recurrence 0.0303411632 0.9999116 -2.993036 T-test    PHF23
135     Q9UMN6 Recurrence 0.0305181172 0.9999116  2.372919 T-test    KMT2B
136     Q9BXN1 Recurrence 0.0312976209 0.9999116 -2.360196 T-test     ASPN
137     Q7Z7L7 Recurrence 0.0313857859 0.9999116  2.546279 T-test     ZER1
138     Q9ULX6 Recurrence 0.0314042475 0.9999116  2.358478 T-test   AKAP8L
139     P35030 Recurrence 0.0314683937 0.9999116 -2.357448 T-test    PRSS3
140     Q96BP2 Recurrence 0.0316869120 0.9999116  2.598595 T-test   CHCHD1
141     P40692 Recurrence 0.0318438187 0.9999116  2.366429 T-test     MLH1
142     Q08554 Recurrence 0.0319917273 0.9999116 -2.425547 T-test     DSC1
143     Q9BXV9 Recurrence 0.0320671796 0.9999116 -2.347924 T-test     GON7
144     Q9NQG5 Recurrence 0.0322159360 0.9999116 -2.345583 T-test   RPRD1B
145     P14317 Recurrence 0.0322660991 0.9999116 -2.344796 T-test    HCLS1
146     Q9UJ83 Recurrence 0.0323663100 0.9999116  2.375239 T-test    HACL1
147     Q9C0C2 Recurrence 0.0324525645 0.9999116 -2.341881 T-test TNKS1BP1
148     Q8IWE2 Recurrence 0.0325006984 0.9999116 -2.341131 T-test FAM114A1
149     Q86SX6 Recurrence 0.0327823835 0.9999116 -2.336764 T-test    GLRX5
150     P09471 Recurrence 0.0328016451 0.9999116 -2.336466 T-test    GNAO1
151     Q99567 Recurrence 0.0330671228 0.9999116  2.332384 T-test    NUP88
152     P35251 Recurrence 0.0331588920 0.9999116  2.330980 T-test     RFC1
153     O14787 Recurrence 0.0332272171 0.9999116  2.329937 T-test    TNPO2
154     P78346 Recurrence 0.0334762609 0.9999116  2.326152 T-test    RPP30
155     Q7KZ85 Recurrence 0.0334957130 0.9999116  2.325858 T-test   SUPT6H
156     P30626 Recurrence 0.0337031489 0.9999116 -2.322727 T-test      SRI
157     Q05084 Recurrence 0.0338774305 0.9999116 -2.456524 T-test     ICA1
158     Q96QA5 Recurrence 0.0340678632 0.9999116 -2.496303 T-test    GSDMA
159     Q6IPR1 Recurrence 0.0342449960 0.9999116  2.329022 T-test   ETFRF1
160     Q9UBX1 Recurrence 0.0342776065 0.9999116 -2.328532 T-test     CTSF
161     P63010 Recurrence 0.0342794167 0.9999116  2.314124 T-test    AP2B1
162     Q6P4F2 Recurrence 0.0343431922 0.9999116 -2.313180 T-test     FDX2
163     Q9BPX7 Recurrence 0.0343605785 0.9999116  2.726161 T-test  C7orf25
164     Q01780 Recurrence 0.0347439763 0.9999116  2.307286 T-test  EXOSC10
165     Q9BYV8 Recurrence 0.0348221721 0.9999116 -2.306144 T-test    CEP41
166     Q96F07 Recurrence 0.0354082527 0.9999116  2.297656 T-test   CYFIP2
167     Q9Y485 Recurrence 0.0355386612 0.9999116  2.326185 T-test    DMXL1
168     P62861 Recurrence 0.0355699637 0.9999116  2.295338 T-test      FAU
169     Q9Y5Q8 Recurrence 0.0356375486 0.9999116  2.294371 T-test   GTF3C5
170     Q9HB58 Recurrence 0.0356932371 0.9999116 -2.293577 T-test    SP110
171     Q9H3K6 Recurrence 0.0360225836 0.9999116 -2.288899 T-test   BOLA2B
172     Q9NTI5 Recurrence 0.0360243139 0.9999116  2.288875 T-test    PDS5B
173     Q8IZE3 Recurrence 0.0361878168 0.9999116  2.687190 T-test    SCYL3
174     Q9UBK8 Recurrence 0.0368426799 0.9999116  2.277424 T-test     MTRR
175     Q96NW4 Recurrence 0.0369295571 0.9999116  2.305972 T-test  ANKRD27
176     P51911 Recurrence 0.0369929839 0.9999116 -2.275347 T-test     CNN1
177     Q08426 Recurrence 0.0376123107 0.9999116  2.266869 T-test   EHHADH
178     Q9Y5J9 Recurrence 0.0376763945 0.9999116 -2.266000 T-test   TIMM8B
179     O60828 Recurrence 0.0378765470 0.9999116 -2.276891 T-test    PQBP1
180     P48556 Recurrence 0.0378851367 0.9999116  2.263176 T-test    PSMD8
181     Q15648 Recurrence 0.0380622960 0.9999116  2.260791 T-test     MED1
182     P29353 Recurrence 0.0383071180 0.9999116 -2.257512 T-test     SHC1
183     Q9H425 Recurrence 0.0383717153 0.9999116 -2.256650 T-test C1orf198
184     P07951 Recurrence 0.0383986085 0.9999116 -2.256292 T-test     TPM2
185     P57772 Recurrence 0.0384493303 0.9999116  2.255616 T-test   EEFSEC
186     P12883 Recurrence 0.0388627640 0.9999116 -2.318642 T-test     MYH7
187     Q15036 Recurrence 0.0388711410 0.9999116  2.250032 T-test    SNX17
188     P67936 Recurrence 0.0391977508 0.9999116 -2.245746 T-test     TPM4
189     P11172 Recurrence 0.0392600000 0.9999116  2.244933 T-test     UMPS
190     Q9BY76 Recurrence 0.0393065637 0.9999116 -2.460199 T-test  ANGPTL4
191     Q7Z4R8 Recurrence 0.0395384517 0.9999116 -2.287868 T-test C6orf120
192     Q9P0I2 Recurrence 0.0398964973 0.9999116  2.236688 T-test     EMC3
193     P06731 Recurrence 0.0399093725 0.9999116 -2.236522 T-test  CEACAM5
194     P80303 Recurrence 0.0399193126 0.9999116 -2.236394 T-test    NUCB2
195     Q6P2E9 Recurrence 0.0403089922 0.9999116  2.231409 T-test     EDC4
196     Q9BXJ0 Recurrence 0.0406073099 0.9999116 -2.506449 T-test  C1QTNF5
197     Q9NSU2 Recurrence 0.0407057177 0.9999116 -2.437772 T-test    TREX1
198     P53667 Recurrence 0.0410751156 0.9999116 -2.288064 T-test    LIMK1
199     Q8IW52 Recurrence 0.0411295430 0.9999116 -2.234028 T-test  SLITRK4
200     O00487 Recurrence 0.0414168640 0.9999116  2.217476 T-test   PSMD14
201     P83110 Recurrence 0.0414791029 0.9999116 -2.216704 T-test    HTRA3
202     Q92610 Recurrence 0.0415563308 0.9999116  2.243547 T-test   ZNF592
203     Q04695 Recurrence 0.0419440969 0.9999116 -2.210967 T-test    KRT17
204     O95273 Recurrence 0.0421015929 0.9999116 -2.274407 T-test  CCNDBP1
205     Q9BZF1 Recurrence 0.0422752941 0.9999116  2.206916 T-test   OSBPL8
206     P20742 Recurrence 0.0422966460 0.9999116 -2.219418 T-test      PZP
207     Q9H6S0 Recurrence 0.0423267082 0.9999116 -2.326154 T-test   YTHDC2
208     Q10570 Recurrence 0.0424176527 0.9999116  2.205184 T-test    CPSF1
209     P54198 Recurrence 0.0426169721 0.9999116  2.267669 T-test     HIRA
210     Q9NPI6 Recurrence 0.0428141138 0.9999116  2.200389 T-test    DCP1A
211     Q96D15 Recurrence 0.0429884291 0.9999116 -2.198293 T-test     RCN3
212     O14498 Recurrence 0.0430063582 0.9999116 -2.198078 T-test     ISLR
213     Q7L2H7 Recurrence 0.0431394879 0.9999116  2.196484 T-test    EIF3M
214     Q6KB66 Recurrence 0.0432696411 0.9999116 -2.194929 T-test    KRT80
215     O95716 Recurrence 0.0433379948 0.9999116 -2.194115 T-test    RAB3D
216     O43670 Recurrence 0.0435179281 0.9999116  2.191976 T-test   ZNF207
217     Q8IZX4 Recurrence 0.0436137574 0.9999116  2.190840 T-test    TAF1L
218     Q02224 Recurrence 0.0438175134 0.9999116 -2.188433 T-test    CENPE
219     P35250 Recurrence 0.0440141681 0.9999116  2.186120 T-test     RFC2
220     P27694 Recurrence 0.0442738232 0.9999116  2.183080 T-test     RPA1
221     Q96E39 Recurrence 0.0446326357 0.9999116 -2.178906 T-test   RBMXL1
222     O15372 Recurrence 0.0447844944 0.9999116  2.177149 T-test    EIF3H
223     Q587I9 Recurrence 0.0450069280 0.9999116  2.201106 T-test   SFT2D3
224     Q6IAN0 Recurrence 0.0450115469 0.9999116  2.174532 T-test   DHRS7B
225     P46108 Recurrence 0.0453112283 0.9999116 -2.171096 T-test      CRK
226     Q9Y2R5 Recurrence 0.0453369806 0.9999116 -2.213752 T-test   MRPS17
227     O00533 Recurrence 0.0454310903 0.9999116 -2.169728 T-test     CHL1
228     P19012 Recurrence 0.0454525935 0.9999116 -2.212368 T-test    KRT15
229     P63104 Recurrence 0.0455481237 0.9999116 -2.168395 T-test    YWHAZ
230     O15182 Recurrence 0.0458084353 0.9999116 -2.251013 T-test    CETN3
231     P61077 Recurrence 0.0461767543 0.9999116  2.173406 T-test   UBE2D3
232     P43246 Recurrence 0.0462701098 0.9999116  2.160243 T-test     MSH2
233     Q92499 Recurrence 0.0466778227 0.9999116  2.155691 T-test     DDX1
234     Q9UBP6 Recurrence 0.0469162290 0.9999116 -2.214305 T-test   METTL1
235     P15882 Recurrence 0.0470170180 0.9999116 -2.264385 T-test     CHN1
236     Q9Y2C4 Recurrence 0.0474520132 0.9999116  2.147147 T-test     EXOG
237     P62736 Recurrence 0.0475423917 0.9999116 -2.146158 T-test    ACTA2
238     Q9BX69 Recurrence 0.0477020283 0.9999116  2.170023 T-test    CARD6
239     Q16651 Recurrence 0.0478431302 0.9999116 -2.142879 T-test    PRSS8
240     Q99519 Recurrence 0.0479758513 0.9999116 -2.141438 T-test     NEU1
241     Q07817 Recurrence 0.0479837649 0.9999116 -2.141352 T-test   BCL2L1
242     Q86W25 Recurrence 0.0481495066 0.9999116  2.139558 T-test   NLRP13
243     Q9UJX4 Recurrence 0.0482840111 0.9999116  2.198298 T-test   ANAPC5
244     P46937 Recurrence 0.0484074135 0.9999116 -2.136778 T-test     YAP1
245     Q9Y5N5 Recurrence 0.0485761113 0.9999116 -2.134967 T-test   N6AMT1
246     P43897 Recurrence 0.0485773587 0.9999116  2.134954 T-test     TSFM
247     P04843 Recurrence 0.0486537943 0.9999116  2.134135 T-test     RPN1
248     Q9P2G1 Recurrence 0.0488865263 0.9999116 -2.275933 T-test   ANKIB1
249     O60826 Recurrence 0.0490534341 0.9999116  2.129873 T-test   CCDC22
250     Q96M96 Recurrence 0.0491316854 0.9999116  2.129042 T-test     FGD4
251     Q8TER0 Recurrence 0.0491734817 0.9999116 -2.169481 T-test    SNED1
252     Q9UH65 Recurrence 0.0493519554 0.9999116 -2.126711 T-test   SWAP70
253     Q16718 Recurrence 0.0493852441 0.9999116 -2.126359 T-test   NDUFA5
254     Q8IX18 Recurrence 0.0499595008 0.9999116  2.160812 T-test    DHX40</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(Stat))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
# Change protein names to gene names
datMatSub &lt;- datMat[featOrder, smpOrder]
rownames(datMatSub) &lt;- plyr::mapvalues(rownames(datMatSub),
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F)
pheatmap::pheatmap(datMatSub,
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Differential tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-59-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Use probabilistic dropout model to account for data missingness and
identify differentially abundant entity differences that are
significantly associated with cancer recurrence.</p>
<pre class="r"><code># Fit linear probabilistic dropout model to normalized data
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
# fit

# Check coefficient names for Parameter &#39;contrast&#39; in proDA::test_diff()
# proDA::result_names(fit)
# Identify significant differentially abundant entities
proDAProTissueDiffRes &lt;- proDA::test_diff(fit,
                                          contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                          sort_by = &#39;pval&#39;, n_max = 20) %&gt;%
  dplyr::select(-c(diff, se, df, avg_abundance, n_approx, n_obs)) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(name,
                                       from = featInfo$PG.Groups,
                                       to = featInfo$PG.Genes,
                                       warn_missing = F))
proDAProTissueDiffRes</code></pre>
<pre><code>       name       pval  adj_pval t_statistic     Gene
6166 Q9BXB4 0.03098748 0.9999612   -2.365222  OSBPL11
4520 Q7Z3J2 0.03487549 0.9999612   -2.305366   VPS35L
4283 Q6NXT1 0.04113302 0.9999612    2.221013  ANKRD54
4126 Q5TAQ9 0.06155279 0.9999612   -2.010435    DCAF8
6026 Q9BRX8 0.07088698 0.9999612   -1.934872   PRXL2A
7487 Q9Y3B8 0.07293141 0.9999612    1.919527    REXO2
5512 Q96BP3 0.07481794 0.9999612   -1.905704    PPWD1
6954 Q9NZL9 0.07714201 0.9999612   -1.889098    MAT2B
6831 Q9NVZ3 0.09817835 0.9999612   -1.756176   NECAP2
4082 Q5SY16 0.10090785 0.9999612   -1.740815     NOL9
5379 Q92851 0.10303322 0.9999612   -1.729104   CASP10
665  O75190 0.10476235 0.9999612    1.719730   DNAJB6
3933 Q49A26 0.10488250 0.9999612   -1.719083    GLYR1
4315 Q6P4Q7 0.11399776 0.9999612    1.671825    CNNM4
5421 Q93063 0.11430111 0.9999612   -1.670310     EXT2
5654 Q96GS4 0.11860490 0.9999612    1.649170   BORCS6
690  O75382 0.12354921 0.9999612    1.625682    TRIM3
6483 Q9H7D7 0.12367027 0.9999612   -1.625117    WDR26
1342 P08134 0.12597093 0.9999612    1.614470     RHOC
6275 Q9GZN8 0.12735480 0.9999612    1.608145 C20orf27</code></pre>
<p>Visualize data of top 6 significant features based on proDA
result</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- proDAProTissueDiffRes[1:6,] %&gt;%
  dplyr::select(name, pval, Gene) %&gt;%
  dplyr::rename(Feature = name, pVal = pval)
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats$Feature,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::left_join(topSigFeats, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = paste0(Gene, &#39;\n(p = &#39;, round(pVal, 4), &#39;)&#39;),
                Gene = factor(Gene, levels = unique(Gene))) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.3, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Gene), scales = &#39;free&#39;) +
  th + theme(strip.text = element_text(size = 13, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-61-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>There is no significant association between PCs (Var1) and cancer
recurrence (Var2).</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>[1] Var1    Var2    pVal    pValAdj Stat    Test   
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<pre class="r"><code># Load and subset all Method Development datasets except DIA Tissue Proteomics from
# AG Krijgsveld
# Specific functions for subsetting data
subPlasma &lt;- function(summ_exp) {
  smpAnno &lt;- as.data.frame(colData(summ_exp))
  base &lt;- summ_exp[, smpAnno[[&#39;Condition&#39;]] == &#39;Baseline&#39;]
  folo &lt;- summ_exp[, smpAnno[[&#39;Condition&#39;]] != &#39;Baseline&#39;]
  return(list(Base = base, Folo = folo))
}

subTissue &lt;- function(summ_exp) {
  smpAnno &lt;- as.data.frame(colData(summ_exp))
  tumor &lt;- summ_exp[, smpAnno[[&#39;Condition&#39;]] == &#39;Tumor&#39;]
  normal &lt;- summ_exp[, smpAnno[[&#39;Condition&#39;]] == &#39;Normal&#39;]
  return(list(Tumor = tumor, Normal = normal))
}

# Load normalized datasets and subset them into Baseline/Follow-up and Tumor/Normal datasets
# NO MISSINGNESS
# Targeted Plasma Metabolomics
metaPlasma_Hell &lt;- readRDS(&#39;./data/AG_Hell/metaPlasmaNorm.rds&#39;)
metaBase_Hell &lt;- subPlasma(metaPlasma_Hell)$Base
metaFolo_Hell &lt;- subPlasma(metaPlasma_Hell)$Folo

# Targeted Tissue Metabolomics
metaTissue_Hell &lt;- readRDS(&#39;./data/AG_Hell/metaTissueNorm.rds&#39;)
metaTumor_Hell &lt;- subTissue(metaTissue_Hell)$Tumor
metaNormal_Hell &lt;- subTissue(metaTissue_Hell)$Normal

# DDA Plasma Proteomics
proPlasma_Kri_DDA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm.rds&#39;)
proBase_Kri_DDA &lt;- subPlasma(proPlasma_Kri_DDA)$Base
proFolo_Kri_DDA &lt;- subPlasma(proPlasma_Kri_DDA)$Folo

# DDA Tissue Proteomics
proTissue_Kri_DDA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm.rds&#39;)
proTumor_Kri_DDA &lt;- subTissue(proTissue_Kri_DDA)$Tumor
proNormal_Kri_DDA &lt;- subTissue(proTissue_Kri_DDA)$Normal

# WITH MISSINGNESS
# Untargeted Plasma Metabolomics
metaPlasma_Hopf &lt;- readRDS(&#39;./data/AG_Hopf/metaPlasmaVsn.rds&#39;)
metaBase_Hopf &lt;- subPlasma(metaPlasma_Hopf)$Base
metaFolo_Hopf &lt;- subPlasma(metaPlasma_Hopf)$Folo

# Untargeted Tissue Metabolomics
metaTissue_Hopf &lt;- readRDS(&#39;./data/AG_Hopf/metaTissueVsn.rds&#39;)
metaTumor_Hopf &lt;- subTissue(metaTissue_Hopf)$Tumor
metaNormal_Hopf &lt;- subTissue(metaTissue_Hopf)$Normal

# Untargeted Plasma Lipidomics
lipPlasma_Hopf &lt;- readRDS(&#39;./data/AG_Hopf/lipPlasmaVsn.rds&#39;)
lipBase_Hopf &lt;- subPlasma(lipPlasma_Hopf)$Base
lipFolo_Hopf &lt;- subPlasma(lipPlasma_Hopf)$Folo

# Untargeted Tissue Lipidomics
lipTissue_Hopf &lt;- readRDS(&#39;./data/AG_Hopf/lipTissueVsn.rds&#39;)
lipTumor_Hopf &lt;- subTissue(lipTissue_Hopf)$Tumor
lipNormal_Hopf &lt;- subTissue(lipTissue_Hopf)$Normal

# DIA Plasma Proteomics
proPlasma_Kri_DIA &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm_DIA.rds&#39;)
proBase_Kri_DIA &lt;- subPlasma(proPlasma_Kri_DIA)$Base
proFolo_Kri_DIA &lt;- subPlasma(proPlasma_Kri_DIA)$Folo

# DIA Tissue Proteomics
proTissue_Klin_DIA &lt;- readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)
# Extract only Method Development samples
methodDevSmps &lt;- colnames(metaTissue_Hell)
proTissue_Klin_DIA &lt;- proTissue_Klin_DIA[, colnames(proTissue_Klin_DIA) %in% methodDevSmps]
proTumor_Klin_DIA &lt;- subTissue(proTissue_Klin_DIA)$Tumor
proNormal_Klin_DIA &lt;- subTissue(proTissue_Klin_DIA)$Normal</code></pre>
<p>Visualize Recurrence-related features from DIA Plasma Proteomics
generated by AG Krijgsveld and DIA Tissue Proteomics by AG Klingmüller
through volcano plots. Log2-FC (up and down) is recurrence to
non-recurrence groups. To be clearer, entities on up side are more
abundant in recurrence group than non-recurrence group, and so forth.
For t-test between recurrence and non-recurrence groups, proDA is
engaged.</p>
<pre class="r"><code># Load normalized data
proPlasma &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm_DIA.rds&#39;)
proTissue &lt;- readRDS(&#39;./data/AG_Klingmuller/proTissueNorm_DIA.rds&#39;)
# Prepare protein-gene table for annotations
proPlasmaFeatInfo &lt;- tibble::as_tibble(rowData(proPlasma), rownames = &#39;Protein.Groups&#39;)
proTissueFeatInfo &lt;- tibble::as_tibble(rowData(proTissue), rownames = &#39;PG.Groups&#39;)

# Compute log2(FC)
# PP
proPlasmaMedi &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaMedi_DIA.rds&#39;) %&gt;%
  summExp2df(assay = &#39;mediAbundance&#39;, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;) %&gt;%
  dplyr::select(Feature, Value, Condition, Recurrence)
# BPP
proBaseLogFC &lt;- dplyr::filter(proPlasmaMedi, Condition == &#39;Baseline&#39;) %&gt;%
  dplyr::group_by(Feature, Recurrence) %&gt;%
  dplyr::summarise(Mean = mean(Value, na.rm = T)) %&gt;%
  dplyr::ungroup() %&gt;%
  tidyr::pivot_wider(names_from = &#39;Recurrence&#39;, values_from = &#39;Mean&#39;) %&gt;%
  dplyr::mutate(logFC = Yes - No) %&gt;%
  dplyr::select(Feature, logFC)
# FPP
proFoloLogFC &lt;- dplyr::filter(proPlasmaMedi, Condition != &#39;Baseline&#39;) %&gt;%
  dplyr::group_by(Feature, Recurrence) %&gt;%
  dplyr::summarise(Mean = mean(Value, na.rm = T)) %&gt;%
  dplyr::ungroup() %&gt;%
  tidyr::pivot_wider(names_from = &#39;Recurrence&#39;, values_from = &#39;Mean&#39;) %&gt;%
  dplyr::mutate(logFC = Yes - No) %&gt;%
  dplyr::select(Feature, logFC)

# Create Method Development sample list
metaTissue &lt;- readRDS(&#39;./data/AG_Hell/metaTissueNorm.rds&#39;)
methodDevSmps &lt;- colnames(metaTissue)
# TP
proTissueMedi &lt;- readRDS(&#39;./data/AG_Klingmuller/proTissueMedi_DIA.rds&#39;) %&gt;%
  summExp2df(assay = &#39;mediAbundance&#39;, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;) %&gt;%
  dplyr::filter(Sample %in% methodDevSmps) %&gt;%
  dplyr::select(Feature, Value, Condition, Recurrence)
# TTP
proTumorLogFC &lt;- dplyr::filter(proTissueMedi, Condition == &#39;Tumor&#39;) %&gt;%
  dplyr::group_by(Feature, Recurrence) %&gt;%
  dplyr::summarise(Mean = mean(Value, na.rm = T)) %&gt;%
  dplyr::ungroup() %&gt;%
  tidyr::pivot_wider(names_from = &#39;Recurrence&#39;, values_from = &#39;Mean&#39;) %&gt;%
  dplyr::mutate(logFC = Yes - No) %&gt;%
  dplyr::select(Feature, logFC)
# NTP
proNormalLogFC &lt;- dplyr::filter(proTissueMedi, Condition == &#39;Normal&#39;) %&gt;%
  dplyr::group_by(Feature, Recurrence) %&gt;%
  dplyr::summarise(Mean = mean(Value, na.rm = T)) %&gt;%
  dplyr::ungroup() %&gt;%
  tidyr::pivot_wider(names_from = &#39;Recurrence&#39;, values_from = &#39;Mean&#39;) %&gt;%
  dplyr::mutate(logFC = Yes - No) %&gt;%
  dplyr::select(Feature, logFC)

# Compute associations between features and cancer recurrence
# BPP
proBaseRes_Kri_DIA &lt;- doSOA(proBase_Kri_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proBaseRes_Kri_DIA$data
smpAnno &lt;- proBaseRes_Kri_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproBaseRes_Kri_DIA &lt;- proDA::test_diff(fit,
                                                contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                sort_by = &#39;pval&#39;)
# FPP
proFoloRes_Kri_DIA &lt;- doSOA(proFolo_Kri_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proFoloRes_Kri_DIA$data
smpAnno &lt;- proFoloRes_Kri_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproFoloRes_Kri_DIA &lt;- proDA::test_diff(fit,
                                                contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                sort_by = &#39;pval&#39;)

# TTP
proTumorRes_Klin_DIA &lt;- doSOA(proTumor_Klin_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proTumorRes_Klin_DIA$data
smpAnno &lt;- proTumorRes_Klin_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproTumorRes_Klin_DIA &lt;- proDA::test_diff(fit,
                                                  contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                  sort_by = &#39;pval&#39;)
# NTP
proNormalRes_Klin_DIA &lt;- doSOA(proNormal_Klin_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proNormalRes_Klin_DIA$data
smpAnno &lt;- proNormalRes_Klin_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproNormalRes_Klin_DIA &lt;- proDA::test_diff(fit,
                                                   contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                   sort_by = &#39;pval&#39;)

# Collect all needed information for volcano plots
# BPP
proBaseVolTab &lt;- dplyr::select(all_proDAproBaseRes_Kri_DIA, name, pval) %&gt;%
  dplyr::rename(Feature = name, pVal = pval) %&gt;%
  dplyr::left_join(proBaseLogFC, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(Feature,
                                       from = proPlasmaFeatInfo$Protein.Groups,
                                       to = proPlasmaFeatInfo$Genes,
                                       warn_missing = F))
# FPP
proFoloVolTab &lt;- dplyr::select(all_proDAproFoloRes_Kri_DIA, name, pval) %&gt;%
  dplyr::rename(Feature = name, pVal = pval) %&gt;%
  dplyr::left_join(proFoloLogFC, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(Feature,
                                       from = proPlasmaFeatInfo$Protein.Groups,
                                       to = proPlasmaFeatInfo$Genes,
                                       warn_missing = F))

# TTP
proTumorVolTab &lt;- dplyr::select(all_proDAproTumorRes_Klin_DIA, name, pval) %&gt;%
  dplyr::rename(Feature = name, pVal = pval) %&gt;%
  dplyr::left_join(proTumorLogFC, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(Feature,
                                       from = proTissueFeatInfo$PG.Groups,
                                       to = proTissueFeatInfo$PG.Genes,
                                       warn_missing = F))
# NTP
proNormalVolTab &lt;- dplyr::select(all_proDAproNormalRes_Klin_DIA, name, pval) %&gt;%
  dplyr::rename(Feature = name, pVal = pval) %&gt;%
  dplyr::left_join(proNormalLogFC, by = &#39;Feature&#39;) %&gt;%
  dplyr::mutate(Gene = plyr::mapvalues(Feature,
                                       from = proTissueFeatInfo$PG.Groups,
                                       to = proTissueFeatInfo$PG.Genes,
                                       warn_missing = F))</code></pre>
<p>Log2-FC cutoffs are 0.5 (FC=1.41) and -0.5 (FC=0.71).</p>
<pre class="r"><code># BPP
volTab &lt;- proBaseVolTab
volTab[&#39;DE&#39;] &lt;- &#39;Not significant&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &gt; 0.5] &lt;- &#39;Up&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &lt; -0.5] &lt;- &#39;Down&#39;
volTab[&#39;Label&#39;] &lt;- NA
label4SigFeats &lt;- volTab$Gene[volTab$DE != &#39;Not significant&#39;]
volTab$Label[volTab$DE != &#39;Not significant&#39;] &lt;- label4SigFeats

ggplot(volTab, aes(x=logFC, y=-log10(pVal), col=DE, label=Label)) +
  geom_point(size = 2) +
  geom_hline(yintercept = -log10(0.05), col = &#39;red&#39;, linewidth = 0.4) +
  geom_vline(xintercept = c(-0.5, 0.5), col = &#39;red&#39;, linewidth = 0.4) +
  geom_text_repel(size = 4, show.legend = F) +
  scale_color_manual(values = c(&#39;navy&#39;, &#39;grey&#39;, &#39;red&#39;)) +
  labs(x = &#39;log2(FC)&#39;, title = &#39;Baseline Plasma Proteomics&#39;) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-65-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Log2-FC cutoffs are 0.5 (FC=1.41) and -0.5 (FC=0.71).</p>
<pre class="r"><code># FPP
volTab &lt;- proFoloVolTab
volTab[&#39;DE&#39;] &lt;- &#39;Not significant&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &gt; 0.5] &lt;- &#39;Up&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &lt; -0.5] &lt;- &#39;Down&#39;
volTab[&#39;Label&#39;] &lt;- NA
label4SigFeats &lt;- volTab$Gene[volTab$DE != &#39;Not significant&#39;]
volTab$Label[volTab$DE != &#39;Not significant&#39;] &lt;- label4SigFeats

ggplot(volTab, aes(x=logFC, y=-log10(pVal), col=DE, label=Label)) +
  geom_point(size = 2) +
  geom_hline(yintercept = -log10(0.05), col = &#39;red&#39;, linewidth = 0.4) +
  geom_vline(xintercept = c(-0.5, 0.5), col = &#39;red&#39;, linewidth = 0.4) +
  geom_text_repel(size = 4, show.legend = F) +
  scale_color_manual(values = c(&#39;navy&#39;, &#39;grey&#39;, &#39;red&#39;)) +
  labs(x = &#39;log2(FC)&#39;, title = &#39;Follow-up Plasma Proteomics&#39;) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-66-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Log2-FC cutoffs are 1 (FC=2) and -1 (FC=0.5).</p>
<pre class="r"><code># TTP
volTab &lt;- proTumorVolTab
volTab[&#39;DE&#39;] &lt;- &#39;Not significant&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &gt; 1] &lt;- &#39;Up&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &lt; -1] &lt;- &#39;Down&#39;
volTab[&#39;Label&#39;] &lt;- NA
label4SigFeats &lt;- volTab$Gene[volTab$DE != &#39;Not significant&#39;]
volTab$Label[volTab$DE != &#39;Not significant&#39;] &lt;- label4SigFeats

ggplot(volTab, aes(x=logFC, y=-log10(pVal), col=DE, label=Label)) +
  geom_point(size = 2) +
  geom_hline(yintercept = -log10(0.05), col = &#39;red&#39;, linewidth = 0.4) +
  geom_vline(xintercept = c(-1, 1), col = &#39;red&#39;, linewidth = 0.4) +
  geom_text_repel(size = 4, show.legend = F) +
  scale_color_manual(values = c(&#39;navy&#39;, &#39;grey&#39;, &#39;red&#39;)) +
  labs(x = &#39;log2(FC)&#39;, title = &#39;Tumor Tissue Proteomics&#39;) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-67-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Log2-FC cutoffs are 1.5 (FC=2.83) and -1.5 (FC=0.35).</p>
<pre class="r"><code># NTP
volTab &lt;- proNormalVolTab
volTab[&#39;DE&#39;] &lt;- &#39;Not significant&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &gt; 1.5] &lt;- &#39;Up&#39;
volTab$DE[volTab$pVal &lt;= 0.05 &amp; volTab$logFC &lt; -1.5] &lt;- &#39;Down&#39;
volTab[&#39;Label&#39;] &lt;- NA
label4SigFeats &lt;- volTab$Gene[volTab$DE != &#39;Not significant&#39;]
volTab$Label[volTab$DE != &#39;Not significant&#39;] &lt;- label4SigFeats

ggplot(volTab, aes(x=logFC, y=-log10(pVal), col=DE, label=Label)) +
  geom_point(size = 2) +
  geom_hline(yintercept = -log10(0.05), col = &#39;red&#39;, linewidth = 0.4) +
  geom_vline(xintercept = c(-1.5, 1.5), col = &#39;red&#39;, linewidth = 0.4) +
  geom_text_repel(size = 4, show.legend = F) +
  scale_color_manual(values = c(&#39;navy&#39;, &#39;grey&#39;, &#39;red&#39;)) +
  labs(x = &#39;log2(FC)&#39;, title = &#39;Normal Tissue Proteomics&#39;) +
  th</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-68-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display proportions of significant features that can separate cancer
recurrence and non-recurrence patient groups for all Method Development
datasets we currently have, including Targeted/Untargeted Plasma/Tissue
Metabolomics and Lipidomics (6 datasets) and DIA/DDA Plasma/Tissue
Proteomics (4 datasets, DIA Tissue Proteomics from AG Krijgsveld is
excluded due to bad quality). Numbers above bars indicate total feature
numbers.</p>
<pre class="r"><code># Prepare significant feature lists for all datasets
# NO MISSINGNESS
# Targeted Plasma Metabolomics
metaBaseRes_Hell &lt;- doSOA(metaBase_Hell, meta_var = &#39;Recurrence&#39;)
metaFoloRes_Hell &lt;- doSOA(metaFolo_Hell, meta_var = &#39;Recurrence&#39;)

# Targeted Tissue Metabolomics
metaTumorRes_Hell &lt;- doSOA(metaTumor_Hell, meta_var = &#39;Recurrence&#39;)
metaNormalRes_Hell &lt;- doSOA(metaNormal_Hell, meta_var = &#39;Recurrence&#39;)

# DDA Plasma Proteomics
proBaseRes_Kri_DDA &lt;- doSOA(proBase_Kri_DDA, meta_var = &#39;Recurrence&#39;)
proFoloRes_Kri_DDA &lt;- doSOA(proFolo_Kri_DDA, meta_var = &#39;Recurrence&#39;)

# DDA Tissue Proteomics
proTumorRes_Kri_DDA &lt;- doSOA(proTumor_Kri_DDA, meta_var = &#39;Recurrence&#39;)
proNormalRes_Kri_DDA &lt;- doSOA(proNormal_Kri_DDA, meta_var = &#39;Recurrence&#39;)

# WITH MISSINGNESS (use proDA)
# Untargeted Plasma Metabolomics
metaBaseRes_Hopf &lt;- doSOA(metaBase_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- metaBaseRes_Hopf$data
smpAnno &lt;- metaBaseRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAmetaBaseRes_Hopf &lt;- proDA::test_diff(fit,
                                              contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                              sort_by = &#39;pval&#39;)
proDAmetaBaseRes_Hopf &lt;- dplyr::filter(all_proDAmetaBaseRes_Hopf, pval &lt; 0.05)

metaFoloRes_Hopf &lt;- doSOA(metaFolo_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- metaFoloRes_Hopf$data
smpAnno &lt;- metaFoloRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAmetaFoloRes_Hopf &lt;- proDA::test_diff(fit,
                                              contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                              sort_by = &#39;pval&#39;)
proDAmetaFoloRes_Hopf &lt;- dplyr::filter(all_proDAmetaFoloRes_Hopf, pval &lt; 0.05)

# Untargeted Tissue Metabolomics
metaTumorRes_Hopf &lt;- doSOA(metaTumor_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- metaTumorRes_Hopf$data
smpAnno &lt;- metaTumorRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAmetaTumorRes_Hopf &lt;- proDA::test_diff(fit,
                                               contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                               sort_by = &#39;pval&#39;)
proDAmetaTumorRes_Hopf &lt;- dplyr::filter(all_proDAmetaTumorRes_Hopf, pval &lt; 0.05)

metaNormalRes_Hopf &lt;- doSOA(metaNormal_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- metaNormalRes_Hopf$data
smpAnno &lt;- metaNormalRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAmetaNormalRes_Hopf &lt;- proDA::test_diff(fit,
                                                contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                sort_by = &#39;pval&#39;)
proDAmetaNormalRes_Hopf &lt;- dplyr::filter(all_proDAmetaNormalRes_Hopf, pval &lt; 0.05)

# Untargeted Plasma Lipidomics
lipBaseRes_Hopf &lt;- doSOA(lipBase_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- lipBaseRes_Hopf$data
smpAnno &lt;- lipBaseRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAlipBaseRes_Hopf &lt;- proDA::test_diff(fit,
                                             contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                             sort_by = &#39;pval&#39;)
proDAlipBaseRes_Hopf &lt;- dplyr::filter(all_proDAlipBaseRes_Hopf, pval &lt; 0.05)

lipFoloRes_Hopf &lt;- doSOA(lipFolo_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- lipFoloRes_Hopf$data
smpAnno &lt;- lipFoloRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAlipFoloRes_Hopf &lt;- proDA::test_diff(fit,
                                             contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                             sort_by = &#39;pval&#39;)
proDAlipFoloRes_Hopf &lt;- dplyr::filter(all_proDAlipFoloRes_Hopf, pval &lt; 0.05)

# Untargeted Tissue Lipidomics
lipTumorRes_Hopf &lt;- doSOA(lipTumor_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- lipTumorRes_Hopf$data
smpAnno &lt;- lipTumorRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAlipTumorRes_Hopf &lt;- proDA::test_diff(fit,
                                              contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                              sort_by = &#39;pval&#39;)
proDAlipTumorRes_Hopf &lt;- dplyr::filter(all_proDAlipTumorRes_Hopf, pval &lt; 0.05)

lipNormalRes_Hopf &lt;- doSOA(lipNormal_Hopf, meta_var = &#39;Recurrence&#39;)
datMat &lt;- lipNormalRes_Hopf$data
smpAnno &lt;- lipNormalRes_Hopf$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAlipNormalRes_Hopf &lt;- proDA::test_diff(fit,
                                               contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                               sort_by = &#39;pval&#39;)
proDAlipNormalRes_Hopf &lt;- dplyr::filter(all_proDAlipNormalRes_Hopf, pval &lt; 0.05)

# DIA Plasma Proteomics
proBaseRes_Kri_DIA &lt;- doSOA(proBase_Kri_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proBaseRes_Kri_DIA$data
smpAnno &lt;- proBaseRes_Kri_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproBaseRes_Kri_DIA &lt;- proDA::test_diff(fit,
                                                contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                sort_by = &#39;pval&#39;)
proDAproBaseRes_Kri_DIA &lt;- dplyr::filter(all_proDAproBaseRes_Kri_DIA, pval &lt; 0.05)

proFoloRes_Kri_DIA &lt;- doSOA(proFolo_Kri_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proFoloRes_Kri_DIA$data
smpAnno &lt;- proFoloRes_Kri_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproFoloRes_Kri_DIA &lt;- proDA::test_diff(fit,
                                                contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                sort_by = &#39;pval&#39;)
proDAproFoloRes_Kri_DIA &lt;- dplyr::filter(all_proDAproFoloRes_Kri_DIA, pval &lt; 0.05)

# DIA Tissue Proteomics
proTumorRes_Klin_DIA &lt;- doSOA(proTumor_Klin_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proTumorRes_Klin_DIA$data
smpAnno &lt;- proTumorRes_Klin_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproTumorRes_Klin_DIA &lt;- proDA::test_diff(fit,
                                                  contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                  sort_by = &#39;pval&#39;)
proDAproTumorRes_Klin_DIA &lt;- dplyr::filter(all_proDAproTumorRes_Klin_DIA, pval &lt; 0.05)

proNormalRes_Klin_DIA &lt;- doSOA(proNormal_Klin_DIA, meta_var = &#39;Recurrence&#39;)
datMat &lt;- proNormalRes_Klin_DIA$data
smpAnno &lt;- proNormalRes_Klin_DIA$smpMetadata
fit &lt;- proDA::proDA(datMat, design = ~ smpAnno$Recurrence)
all_proDAproNormalRes_Klin_DIA &lt;- proDA::test_diff(fit,
                                                   contrast = Intercept + `smpAnno$RecurrenceYes` - Intercept,
                                                   sort_by = &#39;pval&#39;)
proDAproNormalRes_Klin_DIA &lt;- dplyr::filter(all_proDAproNormalRes_Klin_DIA, pval &lt; 0.05)</code></pre>
<pre class="r"><code># Make barplot about proportion of significant features in each dataset
# Compute proportion of significant features
metaBaseSigFeats_Hell &lt;- nrow(metaBaseRes_Hell$tFeatSigRes) / nrow(metaBaseRes_Hell$data) * 100
metaFoloSigFeats_Hell &lt;- nrow(metaFoloRes_Hell$tFeatSigRes) / nrow(metaFoloRes_Hell$data) * 100
metaTumorSigFeats_Hell &lt;- nrow(metaTumorRes_Hell$tFeatSigRes) / nrow(metaTumorRes_Hell$data) * 100
metaNormalSigFeats_Hell &lt;- nrow(metaNormalRes_Hell$tFeatSigRes) / nrow(metaNormalRes_Hell$data) * 100
proBaseSigFeats_Kri_DDA &lt;- nrow(proBaseRes_Kri_DDA$tFeatSigRes) / nrow(proBaseRes_Kri_DDA$data) * 100
proFoloSigFeats_Kri_DDA &lt;- nrow(proFoloRes_Kri_DDA$tFeatSigRes) / nrow(proFoloRes_Kri_DDA$data) * 100
proTumorSigFeats_Kri_DDA &lt;- nrow(proTumorRes_Kri_DDA$tFeatSigRes) / nrow(proTumorRes_Kri_DDA$data) * 100
proNormalSigFeats_Kri_DDA &lt;- nrow(proNormalRes_Kri_DDA$tFeatSigRes) / nrow(proNormalRes_Kri_DDA$data) * 100

metaBaseSigFeats_Hopf &lt;- nrow(proDAmetaBaseRes_Hopf) / nrow(metaBaseRes_Hopf$data) * 100
metaFoloSigFeats_Hopf &lt;- nrow(proDAmetaFoloRes_Hopf) / nrow(metaFoloRes_Hopf$data) * 100
metaTumorSigFeats_Hopf &lt;- nrow(proDAmetaTumorRes_Hopf) / nrow(metaTumorRes_Hopf$data) * 100
metaNormalSigFeats_Hopf &lt;- nrow(proDAmetaNormalRes_Hopf) / nrow(metaNormalRes_Hopf$data) * 100
lipBaseSigFeats_Hopf &lt;- nrow(proDAlipBaseRes_Hopf) / nrow(lipBaseRes_Hopf$data) * 100
lipFoloSigFeats_Hopf &lt;- nrow(proDAlipFoloRes_Hopf) / nrow(lipFoloRes_Hopf$data) * 100
lipTumorSigFeats_Hopf &lt;- nrow(proDAlipTumorRes_Hopf) / nrow(lipTumorRes_Hopf$data) * 100
lipNormalSigFeats_Hopf &lt;- nrow(proDAlipNormalRes_Hopf) / nrow(lipNormalRes_Hopf$data) * 100
proBaseSigFeats_Kri_DIA &lt;- nrow(proDAproBaseRes_Kri_DIA) / nrow(proBaseRes_Kri_DIA$data) * 100
proFoloSigFeats_Kri_DIA &lt;- nrow(proDAproFoloRes_Kri_DIA) / nrow(proFoloRes_Kri_DIA$data) * 100
proTumorSigFeats_Klin_DIA &lt;- nrow(proDAproTumorRes_Klin_DIA) / nrow(proTumorRes_Klin_DIA$data) * 100
proNormalSigFeats_Klin_DIA &lt;- nrow(proDAproNormalRes_Klin_DIA) / nrow(proNormalRes_Klin_DIA$data) * 100

# Prepare needed information for barplot
dat &lt;- c(&#39;Untargeted Metabolomics&#39;, &#39;Untargeted Metabolomics&#39;,
         &#39;Targeted Metabolomics&#39;, &#39;Targeted Metabolomics&#39;,
         &#39;Untargeted Lipidomics&#39;, &#39;Untargeted Lipidomics&#39;,
         &#39;DIA Proteomics&#39;, &#39;DIA Proteomics&#39;,
         &#39;DDA Proteomics&#39;, &#39;DDA Proteomics&#39;,
         &#39;Untargeted Metabolomics&#39;, &#39;Untargeted Metabolomics&#39;,
         &#39;Targeted Metabolomics&#39;, &#39;Targeted Metabolomics&#39;,
         &#39;Untargeted Lipidomics&#39;, &#39;Untargeted Lipidomics&#39;,
         &#39;DIA Proteomics&#39;, &#39;DIA Proteomics&#39;,
         &#39;DDA Proteomics&#39;, &#39;DDA Proteomics&#39;)
datModal &lt;- c(rep(c(&#39;Baseline&#39;, &#39;Follow-up&#39;), 5), rep(c(&#39;Tumor&#39;, &#39;Normal&#39;), 5))
datOf &lt;- c(rep(&#39;Plasma&#39;, 10), rep(&#39;Tissue&#39;, 10))
proportion &lt;- c(metaBaseSigFeats_Hopf, metaFoloSigFeats_Hopf,
                metaBaseSigFeats_Hell, metaFoloSigFeats_Hell,
                lipBaseSigFeats_Hopf, lipFoloSigFeats_Hopf,
                proBaseSigFeats_Kri_DIA, proFoloSigFeats_Kri_DIA,
                proBaseSigFeats_Kri_DDA, proFoloSigFeats_Kri_DDA,
                metaTumorSigFeats_Hopf, metaNormalSigFeats_Hopf,
                metaTumorSigFeats_Hell, metaNormalSigFeats_Hell,
                lipTumorSigFeats_Hopf, lipNormalSigFeats_Hopf,
                proTumorSigFeats_Klin_DIA, proNormalSigFeats_Klin_DIA,
                proTumorSigFeats_Kri_DDA, proNormalSigFeats_Kri_DDA)
numFeat &lt;- c(c(nrow(metaBaseRes_Hopf$data), NA), c(nrow(metaBaseRes_Hell$data), NA),
             c(NA, nrow(lipFoloRes_Hopf$data)), c(nrow(proBaseRes_Kri_DIA$data), NA),
             c(nrow(proBaseRes_Kri_DDA$data), NA),
             c(NA, nrow(metaNormalRes_Hopf$data)), c(nrow(metaTumorRes_Hell$data), NA),
             c(NA, nrow(lipNormalRes_Hopf$data)), c(NA, nrow(proNormalRes_Klin_DIA$data)),
             c(nrow(proTumorRes_Kri_DDA$data), NA))
sigFeatPropTab &lt;- data.frame(Data = factor(dat, levels = unique(dat)),
                             Data_modality = datModal,
                             Data_of = datOf,
                             Proportion = proportion,
                             Feature_number = numFeat)

ggplot(sigFeatPropTab, aes(x=Data, y=Proportion, fill = Data_modality, label=Feature_number)) +
  geom_bar(stat = &#39;identity&#39;, position=position_dodge()) +
  geom_text(size = 5, vjust = -0.3) +
  facet_wrap(~Data_of) +
  labs(x = &#39;Data&#39;, y = &#39;Percentage (%)&#39;,
       title = &#39;Proportion of Cancer Recurrence-Related Features&#39;) +
  scale_fill_discrete(name = &#39;Sample Condition&#39;) +
  th + theme(axis.text.x = element_text(angle = 65, vjust = 1, hjust = 1),
             strip.text = element_text(size = 16, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_DIA.Rmd/unnamed-chunk-70-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(strip.text = element_text(size = 26), axis.title = element_text(size = 26),
  #       axis.text.x = element_text(size = 11.5, angle = 60, vjust = 1, hjust = 1), axis.text.y = element_text(size = 22),
  #       legend.title = element_text(size = 24), legend.text = element_text(size = 24),
  #       panel.background = element_blank(), axis.line = element_line(color = &#39;black&#39;))
# ggsave(&#39;./output/percentage_sigFeats.png&#39;, device = &#39;png&#39;, dpi = 400, height = 8, width = 18)</code></pre>
<p>=&gt; Plasma samples in general perform better than Tissue samples
and Baseline samples perform as good as Follow-up samples in terms of
separating recurrence and non-recurrence patient groups.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.2                
 [5] purrr_1.0.1                 readr_2.1.4                
 [7] tidyr_1.3.0                 tibble_3.2.1               
 [9] tidyverse_2.0.0             openxlsx_4.2.5.2           
[11] ggfortify_0.4.16            ggrepel_0.9.3              
[13] ggplot2_3.4.2               proDA_1.14.0               
[15] SummarizedExperiment_1.30.2 Biobase_2.60.0             
[17] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[19] IRanges_2.34.1              S4Vectors_0.38.1           
[21] BiocGenerics_0.46.0         MatrixGenerics_1.12.2      
[23] matrixStats_1.0.0           workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0        farver_2.1.1            bitops_1.0-7           
 [4] fastmap_1.1.1           RCurl_1.98-1.12         promises_1.2.0.1       
 [7] digest_0.6.33           timechange_0.2.0        lifecycle_1.0.3        
[10] processx_3.8.2          magrittr_2.0.3          compiler_4.3.0         
[13] rlang_1.1.1             sass_0.4.7              tools_4.3.0            
[16] utf8_1.2.3              yaml_2.3.7              ggsignif_0.6.4         
[19] knitr_1.43              S4Arrays_1.0.4          labeling_0.4.2         
[22] DelayedArray_0.26.6     RColorBrewer_1.1-3      plyr_1.8.8             
[25] abind_1.4-5             withr_2.5.0             grid_4.3.0             
[28] fansi_1.0.4             ggpubr_0.6.0            git2r_0.32.0           
[31] colorspace_2.1-0        scales_1.2.1            cli_3.6.1              
[34] rmarkdown_2.23          crayon_1.5.2            generics_0.1.3         
[37] rstudioapi_0.15.0       httr_1.4.6              tzdb_0.4.0             
[40] cachem_1.0.8            zlibbioc_1.46.0         XVector_0.40.0         
[43] vctrs_0.6.3             Matrix_1.6-0            carData_3.0-5          
[46] jsonlite_1.8.7          car_3.1-2               callr_3.7.3            
[49] hms_1.1.3               rstatix_0.7.2           jquerylib_0.1.4        
[52] glue_1.6.2              ps_1.7.5                stringi_1.7.12         
[55] gtable_0.3.3            later_1.3.1             munsell_0.5.0          
[58] pillar_1.9.0            pcaMethods_1.92.0       htmltools_0.5.5        
[61] GenomeInfoDbData_1.2.10 R6_2.5.1                rprojroot_2.0.3        
[64] evaluate_0.21           lattice_0.21-8          extraDistr_1.9.1       
[67] highr_0.10              backports_1.4.1         pheatmap_1.0.12        
[70] broom_1.0.5             httpuv_1.6.11           bslib_0.5.0            
[73] Rcpp_1.0.11             zip_2.3.0               gridExtra_2.3          
[76] whisker_0.4.1           xfun_0.39               fs_1.6.2               
[79] getPass_0.2-2           pkgconfig_2.0.3        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
