<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Single-omics analysis: Method development lung cancer patient cohort</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Single-omics analysis: Method development
lung cancer patient cohort</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongdc86335">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> dc86335
</a>
</p>
</div>
<div id="strongRepositoryversionstrongdc86335"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version dc86335.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  code/feature_selection.Rmd
    Untracked:  code/misc.R
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/corrMetaPlasmaRes_Hopf_Hell.rds
    Untracked:  data/corrMetaTissueRes_Hopf_Hell.rds
    Untracked:  data/mofa/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/preliminary_analysis_targeted_DDA.Rmd</code>) and HTML
(<code>docs/preliminary_analysis_targeted_DDA.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
e54886f
</td>
<td>
LiaoQianWu
</td>
<td>
2023-04-25
</td>
<td>
Finalize scripts about data preprocessing, preliminary analysis,
comparisons between targeted and untargeted
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Conduct univariate (association test)
and multivariate (PCA) analyses on individual datasets, Plasma and
Tissue Targeted Metabolomics from AG Hell and DDA Proteomics from AG
Krijgsveld, to take an initial look at power of each dataset in terms of
predicting patient cancer recurrences and also potentially to provide
evidence for latter multi-omics analyses. Association test between each
feature (metabolite and protein level) and cancer recurrence was
performed to capture significant features that can separate recurrence
and non-recurrence patients. PCA was performed to see if there is any
significant PC that can separate recurrence and non-recurrence patients.
</font></p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;SummarizedExperiment&#39;)
library(&#39;ggrepel&#39;)
library(&#39;ggfortify&#39;)
library(&#39;umap&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)</code></pre>
<div id="plasma-metabolomics" class="section level1">
<h1>Plasma Metabolomics</h1>
<p>Based on PCA result obtained by analyzing whole Plasma Metabolomics
dataset that includes Baseline and Follow-up samples, we found that PCA
mainly captures source of variation in time points, the unwanted
variation. Besides, we attempt to identify biomarkers that can predict
whether patients WILL suffer from cancer recurrence or not. Therefore,
we decided to analyze Baseline and Follow-up samples separately.</p>
<pre class="r"><code># Load normalized data
metaPlasmaNorm &lt;- readRDS(&#39;./data/AG_Hell/metaPlasmaNorm.rds&#39;)
# Add time point information into data
timePoint &lt;- as.data.frame(colData(metaPlasmaNorm)) %&gt;%
  dplyr::mutate(TimePoint = dplyr::case_when(
    Condition == &#39;Baseline&#39; ~ &#39;Baseline&#39;,
    Condition != &#39;Baseline&#39; ~ &#39;2 years later&#39;)) %&gt;%
  dplyr::select(TimePoint)
colData(metaPlasmaNorm)[&#39;TimePoint&#39;] &lt;- timePoint

# Perform analysis
metaPlasmaRes &lt;- SOA(metaPlasmaNorm, fac = c(&#39;Recurrence&#39;, &#39;TimePoint&#39;))
tPCASigRes &lt;- metaPlasmaRes$tPCASigRes
pcaRes &lt;- metaPlasmaRes$pcaRes
pcTab &lt;- metaPlasmaRes$pcTab</code></pre>
<p>Display associations between learned PCs (Var1) and factors (Var2).
Basically t-test was performed to test significance of difference
between two groups, e.g., recurrence and non-recurrence patient groups
or baseline and follow-up sample groups.</p>
<pre class="r"><code># Display PCs of interest
tPCASigRes</code></pre>
<pre><code>  Var1       Var2         pVal    tStats    pValAdj
1  PC2  TimePoint 0.0004253577  3.861161 0.03402862
2 PC19 Recurrence 0.0142029245 -2.570343 0.56811698
3 PC25 Recurrence 0.0301048112  2.253121 0.77425112
4 PC14 Recurrence 0.0387125560  2.141439 0.77425112</code></pre>
<p>PC2 explains quite some variation in data and separates patient
samples by time points.</p>
<pre class="r"><code># ggfortify
# ggplot2::autoplot(pcaRes, x = 1, y =2)

ggplot(pcTab, aes(x=PC1, y=PC2, col=TimePoint, shape=Recurrence,
                  group=Patient)) +
  geom_point(size = 4) +
  labs(x = &#39;PC1 (24.92%)&#39;, y = &#39;PC2 (16.79%)&#39;) +#, title = &#39;Plasma metabolomics&#39;) +
  geom_line(col = &#39;grey50&#39;, linetype = &#39;dashed&#39;) +
  scale_color_discrete(name = &#39;Time point&#39;,
                       labels = c(&#39;Baseline&#39;, &#39;2 years later&#39;)) +
  theme(axis.title = element_text(size = 16),
        legend.title = element_text(size = 14), legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaPlasma_pca.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<p>Display variance explained by each PC</p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Plasma metabolomics&#39;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="baseline-samples" class="section level2">
<h2>Baseline samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset baseline samples
smpBaseIdx &lt;- which(colData(metaPlasmaNorm)$Condition == &#39;Baseline&#39;)
metaBase &lt;- metaPlasmaNorm[, smpBaseIdx]

# Perform analysis
metaBaseRes &lt;- SOA(metaBase, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- metaBaseRes$data
smpAnno &lt;- metaBaseRes$smpMetadata
tFeatSigRes &lt;- metaBaseRes$tFeatSigRes
tPCASigRes &lt;- metaBaseRes$tPCASigRes
pcaRes &lt;- metaBaseRes$pcaRes
pcTab &lt;- metaBaseRes$pcTab
pcTopFeatTab &lt;- metaBaseRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>                  Var1       Var2         pVal    tStats   pValAdj
1          PC aa C38:3 Recurrence 0.0007201606  4.069043 0.4537012
2      Cer(d18:2/24:1) Recurrence 0.0016097226  3.707999 0.4966786
3             SM C16:1 Recurrence 0.0025488234  3.501394 0.4966786
4   HexCer(d18:2/22:0) Recurrence 0.0031535147  3.405407 0.4966786
5      Cer(d18:2/16:0) Recurrence 0.0053083586  3.169330 0.5467464
6        DG(17:0_18:1) Recurrence 0.0091328623 -2.920441 0.5467464
7          PC ae C38:6 Recurrence 0.0095671995  2.898943 0.5467464
8   HexCer(d18:2/16:0) Recurrence 0.0115590861  2.811071 0.5467464
9          PC aa C38:0 Recurrence 0.0118884791  2.797964 0.5467464
10            CE(20:3) Recurrence 0.0121374450  2.788286 0.5467464
11       TG(17:2_34:3) Recurrence 0.0124539027 -2.776257 0.5467464
12 Hex3Cer(d18:1/16:0) Recurrence 0.0138162273  2.727609 0.5467464
13         PC ae C38:5 Recurrence 0.0141186063  2.717436 0.5467464
14 Hex2Cer(d18:1/18:0) Recurrence 0.0145373427  2.703688 0.5467464
15         PC ae C38:3 Recurrence 0.0149800428  2.689558 0.5467464
16  HexCer(d18:2/24:0) Recurrence 0.0154000158  2.676517 0.5467464
17       TG(18:3_32:0) Recurrence 0.0156132290 -2.670026 0.5467464
18            CE(20:4) Recurrence 0.0156213263  2.669781 0.5467464
19         PC ae C36:5 Recurrence 0.0167773136  2.636010 0.5563004
20 Hex3Cer(d18:1/18:0) Recurrence 0.0190402795  2.575862 0.5699582
21       DG(14:0_18:2) Recurrence 0.0193386957 -2.568443 0.5699582
22 Hex3Cer(d18:1/24:1) Recurrence 0.0199033020  2.554697 0.5699582
23            SM C20:2 Recurrence 0.0213468980  2.521160 0.5847194
24            SM C18:1 Recurrence 0.0238256891  2.468275 0.6254243
25         PC ae C36:4 Recurrence 0.0273018496  2.402226 0.6581983
26         PC ae C40:6 Recurrence 0.0277950031  2.393501 0.6581983
27         PC aa C36:0 Recurrence 0.0284706111  2.381781 0.6581983
28            SM C24:1 Recurrence 0.0320498986  2.323710 0.6581983
29  HexCer(d18:1/16:0) Recurrence 0.0341216251  2.292798 0.6581983
30         PC ae C32:2 Recurrence 0.0357260764  2.270032 0.6581983
31         PC ae C34:3 Recurrence 0.0371158048  2.251064 0.6581983
32         PC aa C36:3 Recurrence 0.0412007153  2.198877 0.6581983
33         PC aa C40:3 Recurrence 0.0413903625  2.196571 0.6581983
34     Cer(d18:1/26:1) Recurrence 0.0418151242  2.191443 0.6581983
35     DG-O(14:0_18:2) Recurrence 0.0430689871 -2.176577 0.6581983
36         PC ae C38:4 Recurrence 0.0451480392  2.152781 0.6581983
37       TG(17:2_38:7) Recurrence 0.0456641180 -2.147030 0.6581983
38  HexCer(d18:1/24:1) Recurrence 0.0456652972  2.147017 0.6581983
39       TG(16:0_40:6) Recurrence 0.0474132983 -2.127968 0.6581983
40      lysoPC a C26:1 Recurrence 0.0485546935  2.115872 0.6581983
41         PC aa C42:2 Recurrence 0.0497626614  2.103350 0.6581983
42       TG(16:0_38:5) Recurrence 0.0498163681 -2.102800 0.6581983</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, #row scaling is across columns
                   main = &#39;Baseline plasma metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize data of top 6 significant features</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal    tStats   pValAdj
1 PC11 Recurrence 0.01306487 -2.753839 0.2612974
2 PC15 Recurrence 0.02757487  2.397378 0.2757487</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Baseline plasma metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>PC11 potentially separates recurrence and non-recurrence
patients.</p>
<pre class="r"><code># ggfortify
# ggplot2::autoplot(pcaRes, x = 11, y = 15)

# PC11
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC11, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(y = &#39;PC11 (2.74%)&#39;, title = &#39;Baseline plasma metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  theme(axis.title = element_text(size = 16),
        legend.title = element_text(size = 14), legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaPlasma_baseline_pca_pc11.png&#39;), device = &#39;png&#39;, dpi = 400)</code></pre>
<p>Plot molecular signatures captured by PC11 in input data. We took a
look at top 20 features with highest loadings and observed metabolite
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings. There is no single feature that
can decently hint if patients will suffer cancer recurrence or not,
which means each of these top features contributes a bit to explaining
variation between recurrence and non-recurrence patients.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC11$Feature[1:20]
# Order features by loadings
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
a &lt;- pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma metabolomics - PC11&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaPlasma_baseline_heatmap_pc11.png&#39;), a, device = &#39;png&#39;, dpi = 400)</code></pre>
<p>Display top features that build the PC11</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC11</code></pre>
<pre><code>           Feature    Loading
1       TrpBetaine 0.31571511
2            C16:2 0.23921161
3              C10 0.22381730
4    TG(16:1_30:1) 0.22333356
5    DG(21:0_22:6) 0.21131349
6             BABA 0.19295684
7  DG-O(14:0_18:2) 0.17870348
8         CE(22:1) 0.15349047
9    TG(16:1_28:0) 0.14001914
10           3-IAA 0.13908265
11   DG(18:1_18:4) 0.13559403
12     PC ae C30:1 0.12406184
13       3-Met-His 0.12133393
14        Anserine 0.11941510
15 DG-O(16:0_18:1) 0.11477589
16             DCA 0.09942035
17   DG(16:0_20:0) 0.09580647
18   DG(18:1_20:2) 0.09408512
19   TG(16:0_28:1) 0.08804546
20           C10:1 0.08465781
21   DG(14:1_18:1) 0.08292622
22   DG(14:1_20:2) 0.08256603
23        CE(22:2) 0.08197078
24   TG(16:1_32:1) 0.08052516
25    p-Cresol-SO4 0.07892452
26       Histamine 0.07727769
27  lysoPC a C18:2 0.07680191
28   DG(18:3_20:2) 0.07667813
29   TG(18:1_26:0) 0.07457925
30   DG(16:0_16:0) 0.07316404</code></pre>
<p>Another PC that significantly separates recurrence and non-recurrence
patients</p>
<pre class="r"><code># PC15
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC15, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(y = &#39;PC15 (1.71%)&#39;, title = &#39;Baseline plasma metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 6,
                             method.args = list(var.equal = T), hjust = 0.65,
                             show.legend = F) +
  theme(axis.title = element_text(size = 16),
        legend.title = element_text(size = 14), legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaPlasma_baseline_pca_pc15.png&#39;), device = &#39;png&#39;, dpi = 400)

# Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC15$Feature[1:20]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
a &lt;- pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma metabolomics - PC15&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-11-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaPlasma_baseline_heatmap_pc15.png&#39;), a, device = &#39;png&#39;, dpi = 400)

# Display top features with highest loadings of PC of interest
pcTopFeatTab$PC15</code></pre>
<pre><code>               Feature    Loading
1          PC ae C38:1 0.26532771
2  Cer(d18:0/26:1(OH)) 0.20624525
3      DG-O(14:0_18:2) 0.20269596
4        DG(14:1_20:2) 0.18904579
5         p-Cresol-SO4 0.17118411
6        TG(18:0_30:1) 0.14061407
7           ProBetaine 0.12548868
8                 BABA 0.12426900
9           DiCA(12:0) 0.12167275
10 Cer(d18:0/18:0(OH)) 0.11993035
11               C16:2 0.11952563
12               GLCAS 0.11472230
13            CE(22:2) 0.11132655
14       DG(18:1_20:2) 0.10699226
15       DG(18:1_18:4) 0.10620186
16                  C8 0.10573746
17       TG(20:1_26:1) 0.10549332
18       DG(14:0_18:2) 0.10385519
19               GUDCA 0.10103704
20               3-IPA 0.09715646
21            CE(22:1) 0.09606489
22            Cortisol 0.09449596
23       DG(18:1_18:3) 0.09154592
24         PC ae C30:1 0.09109571
25                HCys 0.09107372
26     Cer(d18:0/20:0) 0.08803444
27                GDCA 0.08728985
28       TG(17:0_32:1) 0.08447220
29          Putrescine 0.08414481
30       DG(18:2_20:0) 0.08142684</code></pre>
</div>
<div id="follow-up-samples" class="section level2">
<h2>Follow-up samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset follow-up samples
smpFoloIdx &lt;- which(colData(metaPlasmaNorm)$Condition != &#39;Baseline&#39;)
metaFolo &lt;- metaPlasmaNorm[, smpFoloIdx]

# Perform analysis
metaFoloRes &lt;- SOA(metaFolo, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- metaFoloRes$data
smpAnno &lt;- metaFoloRes$smpMetadata
tFeatSigRes &lt;- metaFoloRes$tFeatSigRes
tPCASigRes &lt;- metaFoloRes$tPCASigRes
pcaRes &lt;- metaFoloRes$pcaRes

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>              Var1       Var2        pVal    tStats   pValAdj
1         CE(18:0) Recurrence 0.004177071  3.278246 0.7418005
2         CE(16:0) Recurrence 0.004506150  3.243839 0.7418005
3             GABA Recurrence 0.006331961  3.088841 0.7418005
4         CE(18:3) Recurrence 0.006710129  3.062286 0.7418005
5       Putrescine Recurrence 0.011210911  2.825323 0.7418005
6         C18:1-OH Recurrence 0.014583130 -2.702207 0.7418005
7              Asp Recurrence 0.021263006  2.523050 0.7418005
8         CE(18:1) Recurrence 0.022178886  2.502792 0.7418005
9   lysoPC a C18:0 Recurrence 0.023623801  2.472384 0.7418005
10        CE(20:3) Recurrence 0.026772814  2.411750 0.7418005
11   DG(18:1_20:3) Recurrence 0.029590789 -2.362908 0.7418005
12             Lac Recurrence 0.032076076  2.323308 0.7418005
13        CE(17:0) Recurrence 0.033207972  2.306210 0.7418005
14        CE(14:1) Recurrence 0.033392588  2.303472 0.7418005
15   TG(22:4_34:2) Recurrence 0.034181504 -2.291931 0.7418005
16   TG(18:2_38:5) Recurrence 0.035047417 -2.279544 0.7418005
17     PC ae C40:1 Recurrence 0.038220988  2.236442 0.7418005
18        CE(14:0) Recurrence 0.039727675  2.217123 0.7418005
19         Taurine Recurrence 0.039948023  2.214354 0.7418005
20   TG(17:2_38:5) Recurrence 0.040444705 -2.208165 0.7418005
21      DiCA(14:0) Recurrence 0.041457564 -2.195757 0.7418005
22   C3-DC (C4-OH) Recurrence 0.044730899 -2.157474 0.7418005
23           C3-OH Recurrence 0.044730899 -2.157474 0.7418005
24            C4:1 Recurrence 0.044730899 -2.157474 0.7418005
25   C5-DC (C6-OH) Recurrence 0.044730899 -2.157474 0.7418005
26         C5-M-DC Recurrence 0.044730899 -2.157474 0.7418005
27 C5-OH (C3-DC-M) Recurrence 0.044730899 -2.157474 0.7418005
28         C5:1-DC Recurrence 0.044730899 -2.157474 0.7418005
29            C6:1 Recurrence 0.044730899 -2.157474 0.7418005
30           C7-DC Recurrence 0.044730899 -2.157474 0.7418005
31              C9 Recurrence 0.044730899 -2.157474 0.7418005
32          Ac-Orn Recurrence 0.044730899 -2.157474 0.7418005
33       c4-OH-Pro Recurrence 0.044730899 -2.157474 0.7418005
34             PEA Recurrence 0.044730899 -2.157474 0.7418005
35        CE(22:5) Recurrence 0.045999877  2.143320 0.7418005
36   TG(20:0_32:3) Recurrence 0.046440721 -2.138486 0.7418005
37        CE(20:4) Recurrence 0.046838060  2.134165 0.7418005
38   DG(18:1_18:3) Recurrence 0.047762464 -2.124240 0.7418005
39          C16-OH Recurrence 0.048662713 -2.114741 0.7418005
40       t4-OH-Pro Recurrence 0.049599541  2.105025 0.7418005
41   TG(22:5_34:2) Recurrence 0.049753984 -2.103439 0.7418005</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Follow-up plasma metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize data of top 6 significant features</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal   tStats   pValAdj
1 PC20 Recurrence 0.01607057 2.656383 0.3214114</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Follow-up plasma metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="changes-in-time-course" class="section level2">
<h2>Changes in time course</h2>
<p>Differences between Follow-up and Baseline samples (Follow-up matrix
- baseline matrix):<br />
</p>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset baseline samples
smpBaseIdx &lt;- which(colData(metaPlasmaNorm)$Condition == &#39;Baseline&#39;)
metaBase &lt;- metaPlasmaNorm[, smpBaseIdx]
# Subset follow-up samples
smpFoloIdx &lt;- which(colData(metaPlasmaNorm)$Condition != &#39;Baseline&#39;)
metaFolo &lt;- metaPlasmaNorm[, smpFoloIdx]
# Modify sample names for matching two datasets
colnames(metaBase) &lt;- stringr::str_remove(colnames(metaBase), &#39;_P_B&#39;)
colnames(metaFolo) &lt;- stringr::str_remove(colnames(metaFolo), &#39;_P_R&#39;)
# Retrieve data matrix
metaBaseMat &lt;- SummarizedExperiment::assay(metaBase)
metaFoloMat &lt;- SummarizedExperiment::assay(metaFolo)
# Subtract follow-up matrix by baseline matrix
if (identical(colnames(metaBaseMat), colnames(metaFoloMat)) &amp;
    identical(rownames(metaBaseMat), rownames(metaFoloMat))) {
  metaDiffMat &lt;- metaFoloMat - metaBaseMat
} else {
  print(&#39;Align samples or features first.&#39;)
}
# Retrieve samples metadata for creating SE object
colAnno &lt;- dplyr::select(as.data.frame(colData(metaBase)), Patient, Recurrence)
metaDiff &lt;- SummarizedExperiment(assays = metaDiffMat, colData = colAnno)

# Perform analysis
metaPlasmaDiffRes &lt;- SOA(metaDiff, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- metaPlasmaDiffRes$data
smpAnno &lt;- metaPlasmaDiffRes$smpMetadata
tFeatSigRes &lt;- metaPlasmaDiffRes$tFeatSigRes
tPCASigRes &lt;- metaPlasmaDiffRes$tPCASigRes
pcaRes &lt;- metaPlasmaDiffRes$pcaRes
pcTab &lt;- metaPlasmaDiffRes$pcTab
pcTopFeatTab &lt;- metaPlasmaDiffRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>              Var1       Var2        pVal    tStats   pValAdj
1       TrpBetaine Recurrence 0.005852856  3.124798 0.9982022
2    DG(17:0_18:1) Recurrence 0.009791578  2.888205 0.9982022
3  Cer(d18:1/26:1) Recurrence 0.009829049 -2.886435 0.9982022
4              Glu Recurrence 0.024507375  2.454639 0.9982022
5         SM C20:2 Recurrence 0.029262546 -2.368369 0.9982022
6    DG(18:1_20:3) Recurrence 0.029378316 -2.366436 0.9982022
7    TG(16:1_28:0) Recurrence 0.031362679  2.334374 0.9982022
8         CE(20:0) Recurrence 0.036638129  2.257508 0.9982022
9       DiCA(14:0) Recurrence 0.045920605 -2.144193 0.9982022
10        SM C22:3 Recurrence 0.047461881 -2.127448 0.9982022
11   TG(22:5_34:2) Recurrence 0.048476565 -2.116692 0.9982022</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Differential plasma metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize data of top 6 significant features</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal   tStats   pValAdj
1 PC17 Recurrence 0.006420724 3.082472 0.1284145</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Differential plasma metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>PC17 potentially separates recurrence and non-recurrence patients</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC17, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Differential plasma metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Observe molecular signatures captured by PC17 in input data</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC17$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Differential plasma metabolomics - PC17&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display top features that build PC17</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab</code></pre>
<pre><code>$PC17
               Feature    Loading
1           TrpBetaine 0.26496491
2      DG-O(14:0_18:2) 0.19193164
3  Cer(d18:0/26:1(OH)) 0.18128634
4                 TMAO 0.16050984
5        DG(14:0_18:2) 0.15776941
6         C6 (C4:1-DC) 0.14696768
7        DG(18:3_18:3) 0.14399381
8                 C3:1 0.13954501
9                3-IAA 0.12530623
10               GUDCA 0.11234694
11       TG(18:1_26:0) 0.11173160
12       DG(14:1_20:2) 0.11113362
13       DG(18:2_18:4) 0.10913552
14           3-Met-His 0.10800454
15       TG(20:1_26:1) 0.10504138
16       TG(18:0_30:1) 0.09980933
17         PC ae C30:0 0.09935600
18                 DCA 0.09881935
19       DG(16:0_20:0) 0.09699635
20               3-IPA 0.09665866
21            SM C22:3 0.09593664
22          Spermidine 0.09019993
23            Xanthine 0.09014675
24 Cer(d18:0/18:0(OH)) 0.08955950
25         PC aa C30:0 0.08911029
26                  AA 0.08894807
27                 Arg 0.08858777
28                 C16 0.08239773
29       DG(16:1_18:0) 0.08094899
30       TG(18:1_28:1) 0.08058352</code></pre>
</div>
</div>
<div id="tissue-metabolomics" class="section level1">
<h1>Tissue Metabolomics</h1>
<p>Based on PCA result obtained by analyzing whole Tissue Metabolomics
dataset that includes Tumor and Normal samples, we found that PCA mainly
captures source of variation in tissue conditions, which is not of
direct interest. In addition, Recurrence-related significant PCs do not
work that well, perhaps because Tumor and Normal samples are noise to
each other. Therefore, we decided to analyze Tumor and Normal samples
separately.</p>
<pre class="r"><code># Load normalized data
metaTissueNorm &lt;- readRDS(&#39;./data/AG_Hell/metaTissueNorm.rds&#39;)

# Perform analysis
metaTissueRes &lt;- SOA(metaTissueNorm, fac = c(&#39;Recurrence&#39;, &#39;Condition&#39;))
tPCASigRes &lt;- metaTissueRes$tPCASigRes
pcaRes &lt;- metaTissueRes$pcaRes
pcTab &lt;- metaTissueRes$pcTab</code></pre>
<p>Display associations between learned PCs (Var1) and factors (Var2).
Basically t-test was performed to test significance of difference
between two groups, e.g., recurrence and non-recurrence patient groups
or tumor and normal sample groups.</p>
<pre class="r"><code># Display PCs of interest
tPCASigRes</code></pre>
<pre><code>  Var1       Var2         pVal    tStats      pValAdj
1  PC2  Condition 6.230140e-15 12.403252 4.984112e-13
2 PC14 Recurrence 3.105732e-02  2.239454 9.839768e-01
3 PC15 Recurrence 4.812859e-02  2.042081 9.839768e-01</code></pre>
<p>PC2 explains quite some variation in data and separates patient
samples by tissue conditions.</p>
<pre class="r"><code>ggplot(pcTab, aes(x=Condition, y=PC2, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tissue metabolomics&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;, title = &#39;Tissue metabolomics&#39;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-23-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize recurrence-related significant PCs</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC14, y=PC15, col=Recurrence, group=Patient)) +
  geom_point(size = 3) +
  labs(title = &#39;Tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  geom_line(col = &#39;grey50&#39;, linetype = &#39;dashed&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Recurrence, y=PC14, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-24-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Recurrence, y=PC15, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-24-3.png" width="960" style="display: block; margin: auto;" /></p>
<div id="tumor-samples" class="section level2">
<h2>Tumor Samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset tumor samples
smpTumorIdx &lt;- which(colData(metaTissueNorm)$Condition == &#39;Tumor&#39;)
metaTumor &lt;- metaTissueNorm[, smpTumorIdx]

# Perform analysis
metaTumorRes &lt;- SOA(metaTumor, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- metaTumorRes$data
smpAnno &lt;- metaTumorRes$smpMetadata
tFeatSigRes &lt;- metaTumorRes$tFeatSigRes
tPCASigRes &lt;- metaTumorRes$tPCASigRes
pcaRes &lt;- metaTumorRes$pcaRes
pcTab &lt;- metaTumorRes$pcTab
pcTopFeatTab &lt;- metaTumorRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>           Var1       Var2       pVal    tStats   pValAdj
1 DG(16:0_20:0) Recurrence 0.01135340 -2.819440 0.9965381
2   PC aa C36:6 Recurrence 0.01348571 -2.738975 0.9965381
3         C10:2 Recurrence 0.01916791 -2.572676 0.9965381
4 DG(22:1_22:2) Recurrence 0.02480119 -2.448872 0.9965381
5 TG(20:2_34:4) Recurrence 0.02772031  2.394813 0.9965381
6  Trigonelline Recurrence 0.03527285 -2.276365 0.9965381
7      FA(20:2) Recurrence 0.04087017 -2.202918 0.9965381
8 TG(22:1_32:5) Recurrence 0.04996148 -2.101315 0.9965381</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
a &lt;- pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Tumor Tissue metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaTumor_sigFeats.png&#39;), plot = a, device = &#39;png&#39;, dpi = 400,
#        height = 8, width = 10)</code></pre>
<p>Visualize data of top 6 significant features</p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaTumor_topSigFeats.png&#39;), device = &#39;png&#39;, dpi = 400,
#        height = 8, width = 10)</code></pre>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal   tStats    pValAdj
1 PC16 Recurrence 0.003636829 3.340968 0.07273658</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Tumor tissue metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>PC16 potentially separates recurrence and non-recurrence
patients.</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC16,
                  col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Tumor tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F) +
  theme(axis.title = element_text(size = 16),
        legend.title = element_text(size = 14), legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/metaTumor_pca.png&#39;), device = &#39;png&#39;, dpi = 400,
#        height = 8, width = 10)</code></pre>
<p>Plot molecular signatures captured by PC16 in input data. We took a
look at top 10 features with highest loadings and observed metabolite
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC16$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Tumor tissue metabolomics - PC16&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display top features that build the PC16</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC16</code></pre>
<pre><code>               Feature    Loading
1        C5-DC (C6-OH) 0.17576864
2                   C8 0.17119294
3        DG(14:0_18:2) 0.14638967
4        TG(22:4_32:2) 0.13666899
5                   H1 0.13623707
6      DG-O(14:0_18:2) 0.13092075
7        DG(18:1_18:4) 0.13026247
8        PheAlaBetaine 0.12915979
9              Ind-SO4 0.12736437
10         PC aa C38:1 0.12439773
11       TG(17:1_32:1) 0.12209139
12                  C3 0.10684408
13       TG(16:1_32:0) 0.10347525
14           Sarcosine 0.10302874
15     Cer(d16:1/18:0) 0.10273602
16       TG(20:2_34:4) 0.10252979
17       DG(18:3_20:2) 0.09813467
18             Cystine 0.09356867
19        Trigonelline 0.09260150
20         PC aa C36:6 0.09121386
21                 EPA 0.09068423
22         PC ae C38:1 0.09060494
23               C10:2 0.08933812
24       DG(18:2_20:4) 0.08913299
25       TG(22:2_32:4) 0.08882532
26       TG(22:3_30:2) 0.08820029
27       DG(18:3_18:3) 0.08813895
28              Met-SO 0.08609561
29 Cer(d18:1/18:0(OH)) 0.08517335
30             Betaine 0.08510355</code></pre>
</div>
<div id="normal-samples" class="section level2">
<h2>Normal samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset normal samples
smpNormalIdx &lt;- which(colData(metaTissueNorm)$Condition == &#39;Normal&#39;)
metaNormal &lt;- metaTissueNorm[, smpNormalIdx]

# Perform analysis
metaNormalRes &lt;- SOA(metaNormal, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- metaNormalRes$data
smpAnno &lt;- metaNormalRes$smpMetadata
tFeatSigRes &lt;- metaNormalRes$tFeatSigRes
tPCASigRes &lt;- metaNormalRes$tPCASigRes
pcaRes &lt;- metaNormalRes$pcaRes
pcTab &lt;- metaNormalRes$pcTab
pcTopFeatTab &lt;- metaNormalRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>                 Var1       Var2       pVal    tStats   pValAdj
1       TG(16:0_38:7) Recurrence 0.02651824  2.416396 0.9980679
2            CE(22:1) Recurrence 0.03336133 -2.303935 0.9980679
3 Cer(d18:0/26:1(OH)) Recurrence 0.03400280 -2.294523 0.9980679
4       TG(20:2_34:4) Recurrence 0.04013662  2.211996 0.9980679
5       TG(17:2_38:5) Recurrence 0.04020299  2.211168 0.9980679</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Normal tissue metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize data of those 5 significant features</p>
<pre class="r"><code># Visualize data of those 5 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:5]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal    tStats pValAdj
1  PC6 Recurrence 0.02632357  2.419978 0.32828
2 PC17 Recurrence 0.04835676 -2.117951 0.32828
3  PC9 Recurrence 0.04924200  2.108713 0.32828</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Normal tissue metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>PC6 potentially separates recurrence and non-recurrence patients.</p>
<pre class="r"><code># PC6
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC6, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Normal tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Plot molecular signatures captured by PC6 in input data. We took a
look at top 10 features with highest loadings and observed metabolite
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC6$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Normal tissue metabolomics - PC6&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display top features that build the PC6</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC6</code></pre>
<pre><code>           Feature    Loading
1      PC aa C38:1 0.32681258
2    DG(18:2_18:3) 0.20170272
3         CE(15:1) 0.17173986
4    TG(17:0_32:1) 0.15380244
5         SM C20:2 0.15127238
6       TrpBetaine 0.14937592
7    TG(16:1_32:2) 0.14430760
8             C6:1 0.14368914
9    DG(14:0_18:2) 0.13796377
10   TG(20:1_32:2) 0.12544473
11        CE(17:0) 0.12337600
12   TG(22:6_34:3) 0.12032874
13   TG(18:0_30:1) 0.11270662
14   TG(18:0_30:0) 0.11230472
15    p-Cresol-SO4 0.10638822
16        CE(15:0) 0.10492220
17   TG(18:1_28:1) 0.10434993
18         Cystine 0.10379734
19   TG(16:1_32:0) 0.10148778
20   TG(22:6_32:1) 0.10047916
21   TG(16:0_30:2) 0.09843615
22 C5-OH (C3-DC-M) 0.09653183
23   TG(17:1_34:3) 0.09181125
24   DG(17:0_17:1) 0.08930373
25   TG(16:0_33:1) 0.08845397
26        Cortisol 0.08554376
27   TG(18:2_28:0) 0.08534346
28           3-IPA 0.08342139
29              C8 0.08258961
30   TG(18:1_30:2) 0.08244218</code></pre>
<p>Other PCs that significantly separate recurrence and non-recurrence
patients</p>
<pre class="r"><code># PC17
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC17, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Normal tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC17$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Normal tissue metabolomics - PC17&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-37-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC17</code></pre>
<pre><code>               Feature    Loading
1                 TMAO 0.23916907
2      DG-O(14:0_18:2) 0.20759778
3        DG(14:1_20:2) 0.14916716
4            3-Met-His 0.14155043
5        TG(14:0_35:1) 0.13893971
6                C7-DC 0.13303423
7        TG(20:4_33:2) 0.12991926
8        TG(16:0_28:1) 0.12556706
9        TG(14:0_35:2) 0.12352031
10       DG(17:0_17:1) 0.12176182
11               GCDCA 0.12035415
12       TG(17:1_32:1) 0.11451120
13               TCDCA 0.10899239
14            CE(14:1) 0.10776297
15            CE(17:0) 0.10553742
16       TG(18:0_30:0) 0.10354589
17       DG(14:0_18:2) 0.10187874
18       PheAlaBetaine 0.10132279
19       TG(16:0_32:3) 0.10099020
20       TG(17:0_34:2) 0.09987857
21 Cer(d18:0/26:1(OH)) 0.09730541
22       DG(18:2_18:3) 0.09680057
23            C14:2-OH 0.09583187
24       DG(18:2_20:0) 0.09495797
25       TG(18:0_32:0) 0.09460269
26            CE(15:0) 0.09380702
27       DG(14:1_18:1) 0.09344213
28       TG(20:3_36:4) 0.09213046
29       TG(18:2_36:4) 0.09203900
30       DG(18:3_20:2) 0.09089090</code></pre>
<pre class="r"><code># PC9
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC9, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Normal tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-37-3.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="difference-in-tissue-conditions" class="section level2">
<h2>Difference in tissue conditions</h2>
<p>Tumor matrix - Normal matrix<br />
</p>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset Tumor samples
smpTumorIdx &lt;- which(colData(metaTissueNorm)$Condition == &#39;Tumor&#39;)
metaTumor &lt;- metaTissueNorm[, smpTumorIdx]
# Subset Normal samples
smpNormalIdx &lt;- which(colData(metaTissueNorm)$Condition == &#39;Normal&#39;)
metaNormal &lt;- metaTissueNorm[, smpNormalIdx]
# Modify sample names for matching two datasets
colnames(metaTumor) &lt;- stringr::str_remove(colnames(metaTumor), &#39;_TG&#39;)
colnames(metaNormal) &lt;- stringr::str_remove(colnames(metaNormal), &#39;_NG&#39;)
# Retrieve data matrix
metaTumorMat &lt;- SummarizedExperiment::assay(metaTumor)
metaNormalMat &lt;- SummarizedExperiment::assay(metaNormal)
# Subtract follow-up matrix by baseline matrix
if (identical(colnames(metaTumorMat), colnames(metaNormalMat)) &amp;
    identical(rownames(metaTumorMat), rownames(metaNormalMat))) {
  metaDiffMat &lt;- metaTumorMat - metaNormalMat
} else {
  print(&#39;Align samples or features first.&#39;)
}
# Retrieve samples metadata for creating SE object
colAnno &lt;- dplyr::select(as.data.frame(colData(metaTumor)), Patient, Recurrence)
rowAnno &lt;- as.data.frame(rowData(metaTumor))
metaDiff &lt;- SummarizedExperiment(assays = metaDiffMat, colData = colAnno,
                                 rowData = rowAnno)

# Perform analysis
metaTissueDiffRes &lt;- SOA(metaDiff, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- metaTissueDiffRes$data
smpAnno &lt;- metaTissueDiffRes$smpMetadata
tFeatSigRes &lt;- metaTissueDiffRes$tFeatSigRes
tPCASigRes &lt;- metaTissueDiffRes$tPCASigRes
pcaRes &lt;- metaTissueDiffRes$pcaRes
pcTab &lt;- metaTissueDiffRes$pcTab
pcTopFeatTab &lt;- metaTissueDiffRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>                Var1       Var2       pVal    tStats   pValAdj
1      TG(17:1_34:3) Recurrence 0.02193754 -2.508053 0.9985946
2      TG(18:3_33:2) Recurrence 0.03407754 -2.293438 0.9985946
3       Trigonelline Recurrence 0.03518887 -2.277547 0.9985946
4 HexCer(d18:2/18:0) Recurrence 0.04591905  2.144211 0.9985946</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Differential tissue metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Visualize data of those 4 significant features</p>
<pre class="r"><code># Visualize data of those 4 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:4]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal    tStats    pValAdj
1 PC18 Recurrence 0.001682469 -3.688153 0.03364938</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Differential tissue metabolomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>PC18 potentially separates recurrence and non-recurrence
patients.</p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC18, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Differential tissue metabolomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-41-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Plot molecular signatures captured by PC18 in input data. We took a
look at top 10 features with highest loadings and observed metabolite
levels of these top features through heatmap. Note that features in
heatmap are ordered by their loadings.</p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC18$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Differential tissue metabolomics - PC18&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-42-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display top features that build the PC18</p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC18</code></pre>
<pre><code>               Feature    Loading
1        DG(18:3_20:2) 0.17178633
2        TG(20:0_34:1) 0.15135434
3        TG(17:1_34:3) 0.14602954
4        TG(20:1_32:2) 0.14132305
5                C7-DC 0.14049054
6             CE(22:0) 0.12455991
7                   C5 0.12405169
8      DG-O(16:0_18:1) 0.12267827
9        DG(18:2_20:4) 0.12126161
10       TG(20:1_24:3) 0.11896515
11       TG(20:4_30:0) 0.11618423
12            CE(22:1) 0.11491964
13                  C3 0.11479263
14       DG(18:2_18:3) 0.11303949
15       TG(17:2_38:6) 0.11182758
16       TG(16:1_33:1) 0.11167675
17         PC aa C38:1 0.11101286
18       TG(18:3_35:2) 0.11040101
19                  C8 0.10839350
20       DG(14:0_18:2) 0.10803467
21       TG(18:0_30:1) 0.10761459
22 Cer(d18:0/26:1(OH)) 0.10545415
23            CE(15:0) 0.10370992
24       DG(14:1_18:1) 0.10222252
25       TG(16:0_36:6) 0.10078920
26       TG(16:0_28:2) 0.09584659
27       DG(18:2_20:0) 0.09394034
28                 C10 0.09263741
29       DG(18:3_18:3) 0.09257864
30               C10:2 0.09253717</code></pre>
</div>
</div>
<div id="plasma-proteomics" class="section level1">
<h1>Plasma Proteomics</h1>
<p>As described in Plasma Metabolomics section, we analyzed Baseline and
Follow-up samples separately.</p>
<pre class="r"><code># Load normalized data
proPlasmaNorm &lt;- readRDS(&#39;./data/AG_Krijgsveld/proPlasmaNorm.rds&#39;)
colnames(proPlasmaNorm) &lt;- colData(proPlasmaNorm)$Sample
# Add time point information into data
timePoint &lt;- as.data.frame(colData(proPlasmaNorm)) %&gt;%
  dplyr::mutate(TimePoint = dplyr::case_when(
    Condition == &#39;Baseline&#39; ~ &#39;Baseline&#39;,
    Condition != &#39;Baseline&#39; ~ &#39;2 years later&#39;)) %&gt;%
  dplyr::select(TimePoint)
colData(proPlasmaNorm)[&#39;TimePoint&#39;] &lt;- timePoint

# Perform analysis
proPlasmaRes &lt;- SOA(proPlasmaNorm, fac = c(&#39;Recurrence&#39;, &#39;TimePoint&#39;))
tPCASigRes &lt;- proPlasmaRes$tPCASigRes
pcaRes &lt;- proPlasmaRes$pcaRes
pcTab &lt;- proPlasmaRes$pcTab

# Display PCs of interest
tPCASigRes</code></pre>
<pre><code>  Var1       Var2        pVal    tStats   pValAdj
1 PC26  TimePoint 0.005765987 -2.926021 0.3810590
2 PC14 Recurrence 0.009526475  2.730785 0.3810590
3 PC27  TimePoint 0.021168120  2.404632 0.5644832
4 PC18 Recurrence 0.036712606 -2.165259 0.6813483
5 PC10 Recurrence 0.042584266  2.098259 0.6813483</code></pre>
<pre class="r"><code>ggplot(pcTab, aes(x=PC26, y=PC27, col=TimePoint, shape=Recurrence, group=Patient)) +
  geom_point(size = 3) +
  labs(title = &#39;Plasma proteomics&#39;) +
  geom_line(col = &#39;grey50&#39;, linetype = &#39;dashed&#39;) +
  scale_color_discrete(name = &#39;Time point&#39;,
                       labels = c(&#39;Baseline&#39;, &#39;2 years later&#39;))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Recurrence, y=PC14, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-44-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;, title = &#39;Plasma proteomics&#39;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-44-3.png" width="960" style="display: block; margin: auto;" /></p>
<div id="baseline-samples-1" class="section level2">
<h2>Baseline samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset baseline samples
smpBaseIdx &lt;- which(colData(proPlasmaNorm)$Condition == &#39;Baseline&#39;)
proBase &lt;- proPlasmaNorm[, smpBaseIdx]

# Perform analysis
proBaseRes &lt;- SOA(proBase, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- proBaseRes$data
smpAnno &lt;- proBaseRes$smpMetadata
tFeatSigRes &lt;- proBaseRes$tFeatSigRes
tPCASigRes &lt;- proBaseRes$tPCASigRes
pcaRes &lt;- proBaseRes$pcaRes
pcTab &lt;- proBaseRes$pcTab
pcTopFeatTab &lt;- proBaseRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>            Var1       Var2         pVal    tStats   pValAdj
1     A0A0G2JMI3 Recurrence 0.0004156624  4.316448 0.1543875
2         Q08380 Recurrence 0.0007098277 -4.075537 0.1543875
3         P01833 Recurrence 0.0012217736 -3.831763 0.1771572
4     A0A0B4J1U3 Recurrence 0.0021974424  3.568155 0.2389719
5     A0A0C4DH31 Recurrence 0.0030166864  3.425428 0.2624517
6         P22626 Recurrence 0.0062404968  3.095496 0.4524360
7         P06331 Recurrence 0.0136729002 -2.732505 0.7428023
8         P0C0L5 Recurrence 0.0157309338  2.666479 0.7428023
9         P60174 Recurrence 0.0168467259  2.634053 0.7428023
10    A0A075B6Q5 Recurrence 0.0212719174 -2.522849 0.7428023
11    A0A0B4J2H0 Recurrence 0.0231713291  2.481713 0.7428023
12        O00391 Recurrence 0.0235655228 -2.473576 0.7428023
13        P10809 Recurrence 0.0238210599  2.468369 0.7428023
14        Q96PD5 Recurrence 0.0244083903 -2.456597 0.7428023
15        P08865 Recurrence 0.0291423003  2.370383 0.7428023
16        P08727 Recurrence 0.0292038163  2.369352 0.7428023
17        P0DP01 Recurrence 0.0363504725  2.261427 0.7428023
18        P06748 Recurrence 0.0393574879  2.221806 0.7428023
19        P02745 Recurrence 0.0405436993 -2.206940 0.7428023
20        P62805 Recurrence 0.0408002947  2.203776 0.7428023
21        P02746 Recurrence 0.0415452657 -2.194695 0.7428023
22        Q8N257 Recurrence 0.0436647940  2.169652 0.7428023
23        P35030 Recurrence 0.0437199457  2.169016 0.7428023
24        P40197 Recurrence 0.0450651657 -2.153710 0.7428023
25 P0DMV8;P0DMV9 Recurrence 0.0479326679  2.122431 0.7428023
26        P01764 Recurrence 0.0479792654  2.121937 0.7428023
27        P04792 Recurrence 0.0499458355  2.101475 0.7428023</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-45-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal    tStats   pValAdj
1 PC11 Recurrence 0.03732335  2.248287 0.2769264
2 PC20 Recurrence 0.03793807 -2.240147 0.2769264
3 PC13 Recurrence 0.04153895  2.194772 0.2769264</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Baseline plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-46-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># PC11
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC11, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Baseline plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-46-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC11$Feature[1:20]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma proteomics - PC11&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-46-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC11</code></pre>
<pre><code>      Feature    Loading
1  A0A0G2JMI3 0.24985749
2      P12110 0.20719066
3      P01704 0.19543585
4      P01772 0.18555743
5      P02788 0.18469386
6  A0A075B6J9 0.17483440
7  A0A0B4J1Y8 0.17239803
8      P0DJI8 0.15381027
9      Q14667 0.14644749
10     P01705 0.13893569
11     P13645 0.11953194
12     Q9Y6R7 0.11786356
13     Q15404 0.11668574
14     P61026 0.11650594
15 A0A0C4DH39 0.11306566
16     P09172 0.11102853
17 A0A0B4J1U3 0.10748606
18     P13727 0.10426196
19 A0A075B6I4 0.10173111
20 A0A0B4J2H0 0.09968231
21     Q13201 0.09947051
22     P02751 0.09787257
23     P07355 0.09746921
24     P01861 0.09643630
25     P06310 0.09391512
26 A0A0G2JS06 0.09384160
27 A0A075B6K4 0.09365570
28     P02741 0.09286585
29     P01706 0.09151497
30     P01764 0.09083255</code></pre>
<pre class="r"><code># PC13
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC13, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  # geom_text_repel(aes(label=Sample), size = 4, show.legend = F) +
  labs(title = &#39;Baseline plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-46-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC13$Feature[1:20]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Baseline plasma proteomics - PC13&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-46-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC13</code></pre>
<pre><code>                 Feature    Loading
1             A0A0B4J1Y8 0.28185126
2      A0A075B6S9;P0DSN7 0.21246339
3                 P01701 0.19405864
4                 P01817 0.19004547
5                 A0M8Q6 0.18138683
6             A0A075B6R2 0.15088162
7                 P40197 0.14905687
8                 P01718 0.14070829
9                 P69892 0.12110076
10            A0A0B4J2H0 0.12061512
11                Q99584 0.12014741
12            A0A0B4J1V2 0.11302798
13            A0A0B4J1U3 0.11010069
14            A0A0C4DH34 0.10956360
15            A0A075B6H9 0.10692747
16                P22891 0.10493922
17                P02741 0.10360699
18                P24592 0.09853841
19                P04439 0.09837370
20            A0A0G2JMI3 0.09540024
21 A0A075B6R9;A0A0C4DH68 0.09518382
22            A0A0C4DH55 0.09442313
23                P09172 0.09358080
24            A0A075B6I9 0.09304134
25                Q15166 0.09248082
26                P08637 0.09146900
27                P01834 0.08770580
28                P01833 0.08655527
29                Q6UWP8 0.08540960
30                Q04695 0.08392753</code></pre>
</div>
<div id="follow-up-samples-1" class="section level2">
<h2>Follow-up samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset follow-up samples
smpFoloIdx &lt;- which(colData(proPlasmaNorm)$Condition != &#39;Baseline&#39;)
proFolo &lt;- proPlasmaNorm[, smpFoloIdx]

# Perform analysis
proFoloRes &lt;- SOA(proFolo, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- proFoloRes$data
smpAnno &lt;- proFoloRes$smpMetadata
tFeatSigRes &lt;- proFoloRes$tFeatSigRes
tPCASigRes &lt;- proFoloRes$tPCASigRes
pcaRes &lt;- proFoloRes$pcaRes
pcTab &lt;- proFoloRes$pcTab
pcTopFeatTab &lt;- proFoloRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>         Var1       Var2         pVal    tStats   pValAdj
1      Q08380 Recurrence 0.0007077218 -4.076872 0.3078590
2  A0A0G2JMI3 Recurrence 0.0045513361  3.239309 0.7461316
3      P00736 Recurrence 0.0151039815 -2.685674 0.7461316
4      P04004 Recurrence 0.0201241217 -2.549421 0.7461316
5      P31146 Recurrence 0.0234009353  2.476958 0.7461316
6      P07737 Recurrence 0.0243673097  2.457412 0.7461316
7  A0A075B6Q5 Recurrence 0.0252744047 -2.439716 0.7461316
8      P0DP01 Recurrence 0.0259393814  2.427119 0.7461316
9      P0C0L5 Recurrence 0.0266978181  2.413115 0.7461316
10     P23528 Recurrence 0.0269542055  2.408465 0.7461316
11 A0A0J9YXX1 Recurrence 0.0318556679 -2.326702 0.7461316
12     P63104 Recurrence 0.0325473986  2.316121 0.7461316
13     P48740 Recurrence 0.0332844291 -2.305074 0.7461316
14     P50552 Recurrence 0.0337435885  2.298306 0.7461316
15     P62328 Recurrence 0.0347412288  2.283891 0.7461316
16     P27105 Recurrence 0.0362204338  2.263207 0.7461316
17     Q96PD5 Recurrence 0.0364949743 -2.259455 0.7461316
18     P02786 Recurrence 0.0371579041 -2.250499 0.7461316
19     Q13418 Recurrence 0.0401483557  2.211849 0.7461316
20 A0A0B4J2H0 Recurrence 0.0442013391  2.163490 0.7461316
21     P07195 Recurrence 0.0452225832  2.151947 0.7461316</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Follow-up plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-47-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-47-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal    tStats   pValAdj
1  PC9 Recurrence 0.04396432 -2.166204 0.5858655</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Follow-up plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-48-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC9, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Follow-up plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-48-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC9$Feature[1:20]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Follow-up plasma proteomics - PC9&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-48-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC9</code></pre>
<pre><code>      Feature    Loading
1  A0A0G2JMI3 0.20363623
2      P0DP01 0.19089840
3      P01824 0.16624714
4      P20742 0.15210239
5      Q14CN4 0.14994193
6      A0M8Q6 0.14416907
7      P01834 0.14279914
8      P01714 0.14246358
9  A0A087WSZ0 0.13657922
10 A0A087WSY4 0.13407855
11     P40197 0.12848877
12     P09172 0.12588497
13     P08519 0.12399735
14     P02741 0.12052483
15     P04264 0.11243925
16     P13224 0.11125605
17     B1AJZ9 0.10670129
18 A0A0C4DH39 0.10667783
19     Q06033 0.10505205
20     P12110 0.10470743
21     P13645 0.10396717
22     P02730 0.10101105
23     Q16610 0.09913836
24 A0A075B6K4 0.09746711
25     P35527 0.09432005
26     P01718 0.09235285
27     P11226 0.09168297
28     P49913 0.09027185
29 A0A0A0MT89 0.08775276
30 A0A0C4DH55 0.08667632</code></pre>
</div>
<div id="changes-in-time-course-1" class="section level2">
<h2>Changes in time course</h2>
<p>Follow-up matrix - baseline matrix<br />
</p>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset baseline samples
smpBaseIdx &lt;- which(colData(proPlasmaNorm)$Condition == &#39;Baseline&#39;)
proBase &lt;- proPlasmaNorm[, smpBaseIdx]
# Subset follow-up samples
smpFoloIdx &lt;- which(colData(proPlasmaNorm)$Condition != &#39;Baseline&#39;)
proFolo &lt;- proPlasmaNorm[, smpFoloIdx]
# Modify sample names for matching two datasets
colnames(proBase) &lt;- stringr::str_remove(colnames(proBase), &#39;_P_B&#39;)
colnames(proFolo) &lt;- stringr::str_remove(colnames(proFolo), &#39;_P_R&#39;)
# Retrieve data matrix
proBaseMat &lt;- SummarizedExperiment::assay(proBase)
proFoloMat &lt;- SummarizedExperiment::assay(proFolo)
# Subtract follow-up matrix by baseline matrix
# if (identical(colnames(proBaseMat), colnames(proFoloMat)) &amp;
#     identical(rownames(proBaseMat), rownames(proFoloMat))) {
#   proDiffMat &lt;- proFoloMat - proBaseMat
# } else {
#   print(&#39;Align samples or features first.&#39;)
# }
#### Manually align samples ####
if (identical(sort(colnames(proBaseMat)), sort(colnames(proFoloMat)))) {
  smpOrder &lt;- sort(colnames(proBaseMat))
  # For computing difference between Tumor and Normal samples
  proBaseMat &lt;- proBaseMat[, smpOrder]
  proFoloMat &lt;- proFoloMat[, smpOrder]
  proDiffMat &lt;- proFoloMat - proBaseMat
  # For Creating SE object
  proBase &lt;- proBase[, smpOrder]
  proFolo &lt;- proFolo[, smpOrder]
}
# Retrieve samples metadata for creating SE object
colAnno &lt;- dplyr::select(as.data.frame(colData(proBase)), Patient, Recurrence)
rowAnno &lt;- as.data.frame(rowData(proBase))
proDiff &lt;- SummarizedExperiment(assays = proDiffMat, colData = colAnno,
                                rowData = rowAnno)

# Perform analysis
proPlasmaDiffRes &lt;- SOA(proDiff, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- proPlasmaDiffRes$data
smpAnno &lt;- proPlasmaDiffRes$smpMetadata
tFeatSigRes &lt;- proPlasmaDiffRes$tFeatSigRes
tPCASigRes &lt;- proPlasmaDiffRes$tPCASigRes
pcaRes &lt;- proPlasmaDiffRes$pcaRes
pcTab &lt;- proPlasmaDiffRes$pcTab
pcTopFeatTab &lt;- proPlasmaDiffRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>         Var1       Var2        pVal    tStats  pValAdj
1      Q9H8L6 Recurrence 0.006616043 -3.068754 0.955986
2      P01871 Recurrence 0.009977778  2.879472 0.955986
3  A0A0C4DH31 Recurrence 0.011147227 -2.827975 0.955986
4      P0DOX6 Recurrence 0.015650646  2.668896 0.955986
5      P01743 Recurrence 0.026720245 -2.412706 0.955986
6      P02745 Recurrence 0.028839282  2.375494 0.955986
7      P50552 Recurrence 0.030002779  2.356135 0.955986
8      P27105 Recurrence 0.031903914  2.325957 0.955986
9      P02747 Recurrence 0.035187049  2.277573 0.955986
10     P01834 Recurrence 0.037694754 -2.243354 0.955986
11     P01714 Recurrence 0.044703969 -2.157778 0.955986</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = T,
                   scale = &#39;row&#39;, main = &#39;Differential plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-49-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-49-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal    tStats   pValAdj
1  PC9 Recurrence 0.02729361  2.402373 0.3336562
2  PC8 Recurrence 0.03336562 -2.303871 0.3336562</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;,
       title = &#39;Differential plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-50-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=PC9, y=PC8, col=Recurrence)) +
  geom_point(size = 3) +
  labs(title = &#39;Differential plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-50-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># PC9
ggplot(pcTab, aes(x=Recurrence, y=PC9, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Differential plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-50-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC9$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Differential plasma proteomics - PC9&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-50-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC9</code></pre>
<pre><code>      Feature    Loading
1      P35908 0.24016506
2      P81605 0.22664021
3      P13645 0.22158827
4      P02788 0.21982816
5      P01763 0.20104564
6      P01824 0.17800833
7      P01817 0.16662124
8      P04439 0.16352295
9      P02533 0.16015975
10     P13647 0.15667948
11     Q08830 0.15604205
12     Q13201 0.15387766
13     P40197 0.14856935
14     P61160 0.13434463
15     P02538 0.13236060
16     P35030 0.13002292
17     P06454 0.11491442
18     P01718 0.11242929
19     Q15404 0.10317074
20     P26641 0.10246731
21     P48668 0.10095937
22     P07910 0.10054399
23     P35625 0.09973387
24     P27105 0.09970923
25     P12273 0.09946012
26     P01721 0.09280813
27     Q13103 0.09120523
28     P68366 0.09050913
29     P01772 0.08814778
30 A0A075B6K6 0.08548049</code></pre>
<pre class="r"><code># PC8
ggplot(pcTab, aes(x=Recurrence, y=PC8, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Differential plasma proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-50-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC8$Feature[1:10]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Differential plasma proteomics - PC8&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-50-6.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC8</code></pre>
<pre><code>             Feature    Loading
1             P01704 0.25647495
2             P01824 0.22698206
3      P60842;Q14240 0.18354960
4             P0DP08 0.16586743
5             P0DP04 0.15494166
6             Q99584 0.15376279
7             P01763 0.14692742
8             Q92496 0.14027317
9             P01602 0.13397101
10 A0A0B4J2D9;P0DP09 0.12216541
11            P61026 0.11963694
12            P50552 0.11683960
13            P69905 0.11525317
14            P08865 0.11424998
15            Q15113 0.11186621
16            P07910 0.10880331
17            P15814 0.10565859
18        A0A075B6Q5 0.10499063
19            P68871 0.10350441
20            P01721 0.10048052
21            P13727 0.10035856
22     P02776;P10720 0.09888781
23 A0A075B6S9;P0DSN7 0.09751784
24     P01597;P04432 0.09694563
25        A0A075B6K6 0.09668535
26            P40197 0.09175417
27            P27918 0.08938352
28        A0A0C4DH24 0.08930916
29            P43652 0.08376107
30            P02751 0.08227763</code></pre>
</div>
</div>
<div id="tissue-proteomics" class="section level1">
<h1>Tissue Proteomics</h1>
<p>As described in Tissue Metabolomics section, we analyzed Tumor and
Normal samples separately.</p>
<pre class="r"><code># Load normalized data
proTissueNorm &lt;- readRDS(&#39;./data/AG_Krijgsveld/proTissueNorm.rds&#39;)
colnames(proTissueNorm) &lt;- colData(proTissueNorm)$Sample
#### Manually modify mislabeling ####
colnames(proTissueNorm)[22] &lt;- &#39;F04ERF3M_TG&#39;
colData(proTissueNorm)$Sample &lt;- colnames(proTissueNorm)
colData(proTissueNorm)$Patient[22] &lt;- &#39;F04ERF3M&#39;
colData(proTissueNorm)$Recurrence[22] &lt;- &#39;Yes&#39;

# Perform analysis
proTissueRes &lt;- SOA(proTissueNorm, fac = c(&#39;Recurrence&#39;, &#39;Condition&#39;))
tPCASigRes &lt;- proTissueRes$tPCASigRes
pcaRes &lt;- proTissueRes$pcaRes
pcTab &lt;- proTissueRes$pcTab

# Display PCs of interest
tPCASigRes</code></pre>
<pre><code>  Var1       Var2         pVal     tStats      pValAdj
1  PC1  Condition 2.521355e-18 -15.837881 2.017084e-16
2 PC40  Condition 1.778248e-04   4.154968 7.112992e-03
3 PC18 Recurrence 4.889316e-03  -2.988773 1.303818e-01
4 PC14 Recurrence 3.002505e-02   2.254282 4.695753e-01
5 PC24 Recurrence 3.020625e-02   2.251647 4.695753e-01
6 PC12 Recurrence 3.521815e-02   2.183821 4.695753e-01</code></pre>
<pre class="r"><code>ggplot(pcTab, aes(x=Condition, y=PC1, col=Condition, fill=Condition)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tissue proteomics&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-51-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pcTab, aes(x=Recurrence, y=PC18, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-51-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;, title = &#39;Tissue proteomics&#39;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-51-3.png" width="960" style="display: block; margin: auto;" /></p>
<div id="tumor-samples-1" class="section level2">
<h2>Tumor Samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset tumor samples
smpTumorIdx &lt;- which(colData(proTissueNorm)$Condition == &#39;Tumor&#39;)
proTumor &lt;- proTissueNorm[, smpTumorIdx]

# Perform analysis
proTumorRes &lt;- SOA(proTumor, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- proTumorRes$data
smpAnno &lt;- proTumorRes$smpMetadata
tFeatSigRes &lt;- proTumorRes$tFeatSigRes
tPCASigRes &lt;- proTumorRes$tPCASigRes
pcaRes &lt;- proTumorRes$pcaRes
pcTab &lt;- proTumorRes$pcTab
pcTopFeatTab &lt;- proTumorRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>          Var1       Var2         pVal    tStats   pValAdj
1       Q8NFH3 Recurrence 0.0001293816  4.847169 0.6718785
2       O00330 Recurrence 0.0005330163 -4.204373 0.9987710
3       Q9Y3B7 Recurrence 0.0013371209 -3.791280 0.9987710
4       Q5T653 Recurrence 0.0022247563 -3.562598 0.9987710
5       O75150 Recurrence 0.0022801954  3.551526 0.9987710
6       P54687 Recurrence 0.0027803896  3.462214 0.9987710
7       Q96I59 Recurrence 0.0030312931 -3.423248 0.9987710
8       Q12756 Recurrence 0.0041986280  3.275912 0.9987710
9       Q96DE0 Recurrence 0.0043555965  3.259263 0.9987710
10      O43660 Recurrence 0.0045133638  3.243113 0.9987710
11      Q96GC5 Recurrence 0.0051919026 -3.179432 0.9987710
12      P10515 Recurrence 0.0056507513 -3.140835 0.9987710
13      Q5TAQ9 Recurrence 0.0061374732  3.103106 0.9987710
14      Q14202 Recurrence 0.0070727710  3.038156 0.9987710
15      Q9Y2W6 Recurrence 0.0077478641  2.996278 0.9987710
16      P51970 Recurrence 0.0078819761 -2.988382 0.9987710
17      Q7KZ85 Recurrence 0.0085580388  2.950464 0.9987710
18      P62979 Recurrence 0.0089180456  2.931441 0.9987710
19      Q8NFJ5 Recurrence 0.0091573774 -2.919201 0.9987710
20      Q96KP4 Recurrence 0.0097920284  2.888183 0.9987710
21      Q9NX20 Recurrence 0.0099943726 -2.878702 0.9987710
22      Q9P015 Recurrence 0.0106223425 -2.850411 0.9987710
23      Q9Y512 Recurrence 0.0109811635 -2.834962 0.9987710
24      Q9BZG1 Recurrence 0.0112944887  2.821863 0.9987710
25      P11233 Recurrence 0.0113017499 -2.821564 0.9987710
26      O60716 Recurrence 0.0113865925 -2.818079 0.9987710
27      Q9NYB0 Recurrence 0.0127022451  2.767020 0.9987710
28      Q9Y3B8 Recurrence 0.0127658169 -2.764683 0.9987710
29      P18669 Recurrence 0.0128631276  2.761128 0.9987710
30      Q9BZK7 Recurrence 0.0131573546  2.750534 0.9987710
31      Q9C0B1 Recurrence 0.0132605997  2.746870 0.9987710
32      P51553 Recurrence 0.0133174304 -2.744864 0.9987710
33      Q9UBM7 Recurrence 0.0135279645 -2.737507 0.9987710
34      Q9NQH7 Recurrence 0.0140153696 -2.720886 0.9987710
35      Q99460 Recurrence 0.0140597875  2.719399 0.9987710
36      P43686 Recurrence 0.0140731132  2.718953 0.9987710
37      Q8WWQ0 Recurrence 0.0143156386  2.710919 0.9987710
38      P22670 Recurrence 0.0143487100  2.709834 0.9987710
39      P62875 Recurrence 0.0149319118  2.691075 0.9987710
40      Q9BXM0 Recurrence 0.0153458083 -2.678181 0.9987710
41      Q9BYD6 Recurrence 0.0159338677 -2.660422 0.9987710
42      P99999 Recurrence 0.0161675108 -2.653538 0.9987710
43      Q9Y5L4 Recurrence 0.0165132950 -2.643523 0.9987710
44      P62072 Recurrence 0.0173696222 -2.619555 0.9987710
45      Q14139 Recurrence 0.0173738449 -2.619440 0.9987710
46      P10398 Recurrence 0.0175113963 -2.615696 0.9987710
47      Q86Y79 Recurrence 0.0175782208 -2.613887 0.9987710
48      P09132 Recurrence 0.0176749035 -2.611281 0.9987710
49      Q16643 Recurrence 0.0182599925  2.595797 0.9987710
50      Q8WXI9 Recurrence 0.0183919497  2.592369 0.9987710
51      Q8N5M1 Recurrence 0.0187286954 -2.583728 0.9987710
52      Q5JVF3 Recurrence 0.0190096887  2.576629 0.9987710
53      Q8NBI5 Recurrence 0.0191443741 -2.573262 0.9987710
54      P50213 Recurrence 0.0191764643 -2.572463 0.9987710
55      Q15785 Recurrence 0.0193351283  2.568531 0.9987710
56      Q15334 Recurrence 0.0193828037  2.567355 0.9987710
57      P20340 Recurrence 0.0194666871 -2.565294 0.9987710
58      Q9UIS9 Recurrence 0.0197817107  2.557626 0.9987710
59      Q96SY0 Recurrence 0.0207961329  2.533695 0.9987710
60      P04075 Recurrence 0.0211914375  2.524667 0.9987710
61      Q6P5R6 Recurrence 0.0220116210  2.506432 0.9987710
62      P15927 Recurrence 0.0223997426  2.498026 0.9987710
63      Q13769 Recurrence 0.0226686676 -2.492281 0.9987710
64      Q9HBR0 Recurrence 0.0226995675 -2.491625 0.9987710
65      Q9BYD3 Recurrence 0.0239545884 -2.465669 0.9987710
66      P45985 Recurrence 0.0240044541  2.464664 0.9987710
67      Q96P70 Recurrence 0.0243206910  2.458338 0.9987710
68      O60216 Recurrence 0.0250020217  2.444966 0.9987710
69      Q8IXM3 Recurrence 0.0253389140 -2.438481 0.9987710
70      P62195 Recurrence 0.0256238126  2.433059 0.9987710
71      Q9HB71 Recurrence 0.0263308930  2.419842 0.9987710
72      Q9BXV9 Recurrence 0.0263731778 -2.419063 0.9987710
73      P27694 Recurrence 0.0268901868  2.409622 0.9987710
74      Q9H2W6 Recurrence 0.0268998814 -2.409447 0.9987710
75      Q9UPY8 Recurrence 0.0275107946  2.398511 0.9987710
76      P61964 Recurrence 0.0277059940  2.395065 0.9987710
77      Q5JTJ3 Recurrence 0.0278658455 -2.392260 0.9987710
78      Q13547 Recurrence 0.0282422816  2.385712 0.9987710
79      Q9BZE1 Recurrence 0.0291390874 -2.370437 0.9987710
80      Q9Y305 Recurrence 0.0292458057 -2.368649 0.9987710
81      Q6P6C2 Recurrence 0.0293786485  2.366431 0.9987710
82      Q9NWS8 Recurrence 0.0293932670 -2.366187 0.9987710
83      Q92878 Recurrence 0.0296192503 -2.362438 0.9987710
84      O14561 Recurrence 0.0297166874 -2.360829 0.9987710
85  A0A0C4DH31 Recurrence 0.0302209404  2.352583 0.9987710
86      Q96S66 Recurrence 0.0303823259 -2.349970 0.9987710
87      P49721 Recurrence 0.0304918296  2.348205 0.9987710
88      Q8TE77 Recurrence 0.0308502361 -2.342469 0.9987710
89      P48382 Recurrence 0.0315396381 -2.331608 0.9987710
90      O60256 Recurrence 0.0316991070  2.329127 0.9987710
91      P51809 Recurrence 0.0320442132 -2.323797 0.9987710
92      Q9H6S3 Recurrence 0.0323006659 -2.319871 0.9987710
93      Q8IYB5 Recurrence 0.0325542275 -2.316018 0.9987710
94      P25325 Recurrence 0.0328662558 -2.311314 0.9987710
95      Q9NY33 Recurrence 0.0331326899 -2.307330 0.9987710
96      Q969X6 Recurrence 0.0331568693  2.306970 0.9987710
97      Q9UM44 Recurrence 0.0332992430  2.304855 0.9987710
98      P13686 Recurrence 0.0334482959 -2.302649 0.9987710
99      O94842 Recurrence 0.0341453199  2.292455 0.9987710
100     P40937 Recurrence 0.0342147365  2.291450 0.9987710
101     Q8N7H5 Recurrence 0.0342956834  2.290281 0.9987710
102     O14980 Recurrence 0.0345346454  2.286844 0.9987710
103     Q14839 Recurrence 0.0347472931  2.283805 0.9987710
104     Q92604 Recurrence 0.0350463521 -2.279559 0.9987710
105     P82921 Recurrence 0.0354020383 -2.274552 0.9987710
106     P09104 Recurrence 0.0356416828  2.271205 0.9987710
107     Q96KC8 Recurrence 0.0362749908  2.262460 0.9987710
108     P62140 Recurrence 0.0371459138  2.250660 0.9987710
109     O00443 Recurrence 0.0374023044  2.247235 0.9987710
110     Q9NRX2 Recurrence 0.0374601627 -2.246465 0.9987710
111     Q05639 Recurrence 0.0375597304  2.245143 0.9987710
112     Q71RC2 Recurrence 0.0376186395 -2.244362 0.9987710
113     Q9H3M7 Recurrence 0.0376501598  2.243944 0.9987710
114     P47985 Recurrence 0.0381259082 -2.237684 0.9987710
115     P48449 Recurrence 0.0381867908 -2.236888 0.9987710
116     Q8TEM1 Recurrence 0.0383081512  2.235305 0.9987710
117     Q96DV4 Recurrence 0.0383621222 -2.234602 0.9987710
118     Q7LG56 Recurrence 0.0388365310 -2.228466 0.9987710
119     P60900 Recurrence 0.0390514616  2.225708 0.9987710
120     Q8N1B4 Recurrence 0.0393855130 -2.221450 0.9987710
121     Q9HAV7 Recurrence 0.0395601467 -2.219237 0.9987710
122     O95747 Recurrence 0.0396370182  2.218266 0.9987710
123     Q5VZK9 Recurrence 0.0396585008 -2.217995 0.9987710
124     P16455 Recurrence 0.0397402003  2.216965 0.9987710
125     O43772 Recurrence 0.0403793532 -2.208975 0.9987710
126     Q8NBN3 Recurrence 0.0405997937 -2.206246 0.9987710
127     Q8N1S5 Recurrence 0.0411085481 -2.200000 0.9987710
128     Q6IA69 Recurrence 0.0414389848 -2.195982 0.9987710
129     P40938 Recurrence 0.0417440831  2.192297 0.9987710
130     Q13200 Recurrence 0.0418026230  2.191593 0.9987710
131     Q9H0U6 Recurrence 0.0420202082 -2.188983 0.9987710
132     Q8N6R0 Recurrence 0.0422405648  2.186353 0.9987710
133     Q96RE7 Recurrence 0.0427485023  2.180339 0.9987710
134     P35249 Recurrence 0.0431518200  2.175609 0.9987710
135     Q9BR76 Recurrence 0.0431656739 -2.175448 0.9987710
136     Q96S44 Recurrence 0.0434465600 -2.172179 0.9987710
137     Q05655 Recurrence 0.0434904138  2.171670 0.9987710
138     Q92643 Recurrence 0.0440052952  2.165733 0.9987710
139     Q16576 Recurrence 0.0441451669  2.164132 0.9987710
140     P78362 Recurrence 0.0446796818  2.158053 0.9987710
141     P35244 Recurrence 0.0449631699  2.154856 0.9987710
142     P51531 Recurrence 0.0452560662  2.151572 0.9987710
143     Q9GZZ1 Recurrence 0.0454011468  2.149953 0.9987710
144     P07602 Recurrence 0.0455927887 -2.147821 0.9987710
145     P98179 Recurrence 0.0456908823 -2.146733 0.9987710
146     P42166 Recurrence 0.0462971092  2.140056 0.9987710
147     P35658 Recurrence 0.0465679127  2.137099 0.9987710
148     P61981 Recurrence 0.0466169660  2.136565 0.9987710
149     Q13405 Recurrence 0.0475208105 -2.126817 0.9987710
150     Q68CR1 Recurrence 0.0477090617 -2.124808 0.9987710
151     Q9NXA8 Recurrence 0.0480328969  2.121369 0.9987710
152     P09874 Recurrence 0.0481872683  2.119737 0.9987710
153     O00186 Recurrence 0.0484507090 -2.116963 0.9987710
154     Q07960 Recurrence 0.0485400016 -2.116026 0.9987710
155     P25787 Recurrence 0.0487555308  2.113771 0.9987710
156     Q8WUH2 Recurrence 0.0489180998 -2.112075 0.9987710
157     Q96A35 Recurrence 0.0491657344 -2.109503 0.9987710
158     P67870 Recurrence 0.0492697436  2.108426 0.9987710
159     Q9ULK4 Recurrence 0.0493249980  2.107855 0.9987710
160     Q14197 Recurrence 0.0493689623 -2.107400 0.9987710
161     Q96A26 Recurrence 0.0496701556  2.104299 0.9987710</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Tumor Tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-52-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-52-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal   tStats   pValAdj
1 PC14 Recurrence 0.04517205 2.152512 0.4302249</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;, title = &#39;Tumor tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-53-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC14, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Tumor tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-53-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC14$Feature[1:30]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Tumor tissue proteomics - PC14&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-53-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC14</code></pre>
<pre><code>                Feature    Loading
1                P10153 0.09724346
2                Q8WVT3 0.08761372
3                P13727 0.08326646
4                P23560 0.08279921
5                P08729 0.07873893
6            A0A1B0GTL2 0.07824092
7                Q05315 0.07812803
8                Q9HBG4 0.07126667
9                O15120 0.06771819
10               O75525 0.06680807
11               P55064 0.06643610
12               P07205 0.06568554
13               P11678 0.06416789
14 P0DTE7;P0DTE8;P0DUB6 0.06378442
15               P08514 0.06016151
16               O76070 0.06000857
17               P52895 0.05955731
18               P02042 0.05943786
19               P31513 0.05846000
20               P69905 0.05845646
21               O95715 0.05710342
22               O43927 0.05683214
23               P68871 0.05618459
24               P00749 0.05548359
25               Q9BUP0 0.05497376
26               Q53G44 0.05496474
27           A0A0C4DH43 0.05480041
28               P20742 0.05313221
29               P04264 0.05278666
30               P13762 0.05244748</code></pre>
</div>
<div id="normal-samples-1" class="section level2">
<h2>Normal samples</h2>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset normal samples
smpNormalIdx &lt;- which(colData(proTissueNorm)$Condition == &#39;Normal&#39;)
proNormal &lt;- proTissueNorm[, smpNormalIdx]

# Perform analysis
proNormalRes &lt;- SOA(proNormal, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- proNormalRes$data
smpAnno &lt;- proNormalRes$smpMetadata
tFeatSigRes &lt;- proNormalRes$tFeatSigRes
tPCASigRes &lt;- proNormalRes$tPCASigRes
pcaRes &lt;- proNormalRes$pcaRes
pcTab &lt;- proNormalRes$pcTab
pcTopFeatTab &lt;- proNormalRes$pcTopFeatTab

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>             Var1       Var2         pVal    tStats   pValAdj
1          Q9BTZ2 Recurrence 0.0008827485 -3.977623 0.9968386
2          Q9HC38 Recurrence 0.0024866051  3.512523 0.9968386
3          P29728 Recurrence 0.0050851484 -3.188890 0.9968386
4          P07942 Recurrence 0.0056991082  3.136948 0.9968386
5          P17655 Recurrence 0.0063522149  3.087380 0.9968386
6          Q4KMP7 Recurrence 0.0066149709  3.068828 0.9968386
7          Q9NQT8 Recurrence 0.0067448966  3.059918 0.9968386
8          O95833 Recurrence 0.0068062720  3.055767 0.9968386
9          Q9UBB4 Recurrence 0.0068080558  3.055647 0.9968386
10         Q2TAY7 Recurrence 0.0077370120 -2.996923 0.9968386
11         Q9UPT8 Recurrence 0.0081763909 -2.971496 0.9968386
12         Q8NBF2 Recurrence 0.0086359126  2.946284 0.9968386
13         P23258 Recurrence 0.0088532568  2.934809 0.9968386
14         Q14126 Recurrence 0.0092375083  2.915172 0.9968386
15         Q8NBQ5 Recurrence 0.0093464238 -2.909750 0.9968386
16         P54709 Recurrence 0.0098412983  2.885857 0.9968386
17         Q9Y2E5 Recurrence 0.0104218447  2.859264 0.9968386
18         Q9Y2W2 Recurrence 0.0107914208  2.843070 0.9968386
19         P84095 Recurrence 0.0108346735 -2.841209 0.9968386
20         P54289 Recurrence 0.0111969812  2.825902 0.9968386
21         P08567 Recurrence 0.0115109622 -2.813016 0.9968386
22         Q8TB37 Recurrence 0.0115999960 -2.809424 0.9968386
23         P23469 Recurrence 0.0138773382 -2.725536 0.9968386
24         P98160 Recurrence 0.0139381164  2.723483 0.9968386
25         Q3SY69 Recurrence 0.0139831205 -2.721968 0.9968386
26         Q8IVG5 Recurrence 0.0140691811 -2.719085 0.9968386
27         Q2TAA2 Recurrence 0.0142210258  2.714038 0.9968386
28         P0C0L5 Recurrence 0.0142963817  2.711552 0.9968386
29         O95817 Recurrence 0.0152394787  2.681462 0.9968386
30         O00204 Recurrence 0.0157859186  2.664830 0.9968386
31         Q9BZF1 Recurrence 0.0163455859 -2.648355 0.9968386
32         Q9UMY4 Recurrence 0.0165939363  2.641216 0.9968386
33         O15164 Recurrence 0.0171807341  2.624744 0.9968386
34         Q9Y3B8 Recurrence 0.0173129037  2.621107 0.9968386
35         Q9NPJ3 Recurrence 0.0176968217 -2.610692 0.9968386
36         O43660 Recurrence 0.0180020875  2.602563 0.9968386
37         P16422 Recurrence 0.0181600137  2.598409 0.9968386
38         P41214 Recurrence 0.0187324567  2.583632 0.9968386
39         Q6YP21 Recurrence 0.0187703643  2.582669 0.9968386
40         P52799 Recurrence 0.0188363088  2.580997 0.9968386
41         P00813 Recurrence 0.0198312784  2.556430 0.9968386
42         Q9BZL6 Recurrence 0.0203908120  2.543121 0.9968386
43         Q5J8M3 Recurrence 0.0204595020 -2.541511 0.9968386
44         P08572 Recurrence 0.0204914052  2.540765 0.9968386
45         Q8TCJ2 Recurrence 0.0211734806 -2.525074 0.9968386
46         P46108 Recurrence 0.0212732523  2.522819 0.9968386
47         Q9UNS2 Recurrence 0.0213738455 -2.520555 0.9968386
48         O15305 Recurrence 0.0213827204  2.520356 0.9968386
49         P27361 Recurrence 0.0215723428  2.516117 0.9968386
50         P23219 Recurrence 0.0216761533 -2.513812 0.9968386
51         Q7Z6K5 Recurrence 0.0221690154  2.503007 0.9968386
52         Q16363 Recurrence 0.0223708313  2.498647 0.9968386
53         Q8TCT9 Recurrence 0.0226067526 -2.493598 0.9968386
54         Q01650 Recurrence 0.0232664513  2.479738 0.9968386
55         Q14185 Recurrence 0.0237535011  2.469740 0.9968386
56         O60486 Recurrence 0.0239154747  2.466458 0.9968386
57  P20231;Q15661 Recurrence 0.0246320401 -2.452184 0.9968386
58         O95379 Recurrence 0.0249366440  2.446235 0.9968386
59         P41208 Recurrence 0.0252402767  2.440371 0.9968386
60         Q15276 Recurrence 0.0253366332  2.438524 0.9968386
61         P13498 Recurrence 0.0255163129 -2.435098 0.9968386
62         Q13619 Recurrence 0.0263526858  2.419440 0.9968386
63         Q8WUW1 Recurrence 0.0266467667 -2.414046 0.9968386
64         Q99615 Recurrence 0.0269960446  2.407710 0.9968386
65         Q69YN2 Recurrence 0.0272475782 -2.403195 0.9968386
66         O96013 Recurrence 0.0277777415  2.393804 0.9968386
67         Q9H3M7 Recurrence 0.0279314411  2.391113 0.9968386
68         Q9UGP8 Recurrence 0.0281691132 -2.386979 0.9968386
69  P21217;Q11128 Recurrence 0.0286219866  2.379190 0.9968386
70         Q15363 Recurrence 0.0287780532 -2.376533 0.9968386
71         Q9UJX2 Recurrence 0.0289110651  2.374279 0.9968386
72         O15381 Recurrence 0.0291657127  2.369990 0.9968386
73     A0A0J9YXX1 Recurrence 0.0293744420 -2.366501 0.9968386
74         Q9H330 Recurrence 0.0297266092  2.360666 0.9968386
75         Q06187 Recurrence 0.0299168324 -2.357541 0.9968386
76         Q9UHR4 Recurrence 0.0305341467  2.347525 0.9968386
77         P02462 Recurrence 0.0310544649  2.339228 0.9968386
78         P13051 Recurrence 0.0315036960  2.332168 0.9968386
79         Q9UKU9 Recurrence 0.0318705351  2.326473 0.9968386
80         Q13751 Recurrence 0.0318932024  2.326123 0.9968386
81         Q8IY37 Recurrence 0.0319783655  2.324810 0.9968386
82         P52735 Recurrence 0.0322409850  2.320782 0.9968386
83         Q14746 Recurrence 0.0325308922 -2.316371 0.9968386
84         P22792 Recurrence 0.0327910303  2.312444 0.9968386
85         Q9NZT2 Recurrence 0.0329530923  2.310012 0.9968386
86         Q13303 Recurrence 0.0330344164 -2.308796 0.9968386
87         Q8NDT2 Recurrence 0.0331543942 -2.307007 0.9968386
88         P20936 Recurrence 0.0337917673  2.297601 0.9968386
89         Q8TE77 Recurrence 0.0340116586  2.294395 0.9968386
90         Q07973 Recurrence 0.0342391199 -2.291098 0.9968386
91         Q07955 Recurrence 0.0349156314 -2.281411 0.9968386
92         O43583 Recurrence 0.0352845686  2.276201 0.9968386
93         Q9NZ43 Recurrence 0.0355104484 -2.273036 0.9968386
94         Q5TFE4 Recurrence 0.0357778736  2.269313 0.9968386
95         Q9H019 Recurrence 0.0359535596  2.266881 0.9968386
96         Q9UK41 Recurrence 0.0363656921  2.261219 0.9968386
97         P82675 Recurrence 0.0365011354  2.259371 0.9968386
98         Q05209 Recurrence 0.0367472799  2.256029 0.9968386
99         O43665 Recurrence 0.0370596270 -2.251817 0.9968386
100        Q16787 Recurrence 0.0372209445  2.249655 0.9968386
101        Q9Y478 Recurrence 0.0376695979  2.243687 0.9968386
102        Q9P2D3 Recurrence 0.0377548543 -2.242560 0.9968386
103        Q14197 Recurrence 0.0378492397 -2.241316 0.9968386
104        O15127 Recurrence 0.0385066806 -2.232725 0.9968386
105        O75563 Recurrence 0.0386340292 -2.231077 0.9968386
106        P07996 Recurrence 0.0387502672 -2.229576 0.9968386
107        O95295 Recurrence 0.0393400310 -2.222028 0.9968386
108        Q27J81 Recurrence 0.0400440557  2.213152 0.9968386
109        P50440 Recurrence 0.0405211427  2.207219 0.9968386
110        P0DJI8 Recurrence 0.0407613384 -2.204255 0.9968386
111        Q7Z4W1 Recurrence 0.0407624352 -2.204242 0.9968386
112        Q13753 Recurrence 0.0407862685  2.203949 0.9968386
113        P53007 Recurrence 0.0413724341 -2.196789 0.9968386
114        P48730 Recurrence 0.0423713387 -2.184799 0.9968386
115        Q9H223 Recurrence 0.0425542190  2.182632 0.9968386
116        Q9UKV3 Recurrence 0.0431596442 -2.175518 0.9968386
117        P52789 Recurrence 0.0433109553 -2.173754 0.9968386
118        P13591 Recurrence 0.0433835855 -2.172910 0.9968386
119        O95260 Recurrence 0.0436841635 -2.169429 0.9968386
120        Q9NVT9 Recurrence 0.0441152454 -2.164474 0.9968386
121        Q14005 Recurrence 0.0443995770 -2.161230 0.9968386
122        Q96IU4 Recurrence 0.0445194106  2.159869 0.9968386
123        Q9NS86 Recurrence 0.0450919835  2.153410 0.9968386
124        Q92556 Recurrence 0.0457974030 -2.145554 0.9968386
125        P07437 Recurrence 0.0458496482  2.144977 0.9968386
126        Q9BSH4 Recurrence 0.0459297539 -2.144093 0.9968386
127        Q9UNF1 Recurrence 0.0460960877  2.142261 0.9968386
128        Q9BTE6 Recurrence 0.0466791969  2.135888 0.9968386
129        P53365 Recurrence 0.0467578244  2.135035 0.9968386
130        Q9Y5Y2 Recurrence 0.0471979140 -2.130280 0.9968386
131        O43291 Recurrence 0.0494324630  2.106745 0.9968386
132        O95983 Recurrence 0.0494406118 -2.106661 0.9968386</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Normal tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-54-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize data of top 6 significant features
topSigFeats &lt;- tFeatSigRes$Var1[1:6]
topSigFeatData &lt;- tibble::as_tibble(datMat[topSigFeats,], rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;, values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Feature = factor(Feature, levels = topSigFeats)) %&gt;%
  dplyr::left_join(smpAnno, by = &#39;Sample&#39;)

ggplot(topSigFeatData, aes(x=Recurrence, y=Abundance, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  facet_wrap(vars(Feature), scales = &#39;free&#39;) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-54-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>  Var1       Var2       pVal    tStats   pValAdj
1  PC4 Recurrence 0.03650737 -2.259286 0.3610804
2 PC15 Recurrence 0.04936550  2.107436 0.3610804</code></pre>
<pre class="r"><code># Display proportion of total variance captured by all PCs explained by each PC
varExplained &lt;- pcaRes$sdev^2 / sum(pcaRes$sdev^2)
PC &lt;- paste0(&#39;PC&#39;, seq(length(varExplained)))
varTab &lt;- data.frame(PC = factor(PC, levels = PC),
                     Var_explained = varExplained)
ggplot(varTab, aes(x=PC, y=Var_explained*100)) +
  geom_col() +
  labs(x = &#39;&#39;, y = &#39;Variance explained (%)&#39;, title = &#39;Normal tissue proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-55-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># PC4
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC4, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Normal tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-55-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC4$Feature[1:30]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Normal tissue proteomics - PC4&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-55-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC4</code></pre>
<pre><code>      Feature    Loading
1      P11684 0.13233479
2      Q13938 0.12239622
3      P08263 0.10718630
4      O75602 0.09496052
5      P13647 0.08880350
6      P53814 0.08653623
7      P0DOX3 0.08254005
8      Q8TC71 0.08202708
9      Q8IVM0 0.07788327
10     P22792 0.07780716
11 A0A0J9YXX1 0.07720992
12     P0DJI8 0.07604096
13     Q13609 0.07603794
14     O75112 0.07318900
15 A0A0A0MS15 0.07099402
16     O95050 0.07050562
17     O15232 0.06731953
18     Q5TZA2 0.06720104
19 A0A075B6S5 0.06716921
20     Q14315 0.06659310
21     P16442 0.06462080
22     Q96QI5 0.06305502
23     P02100 0.06302400
24     P01876 0.06296850
25     Q9NQU5 0.06290597
26     P35908 0.06243532
27     A8MYP8 0.06234938
28     Q01995 0.06198790
29     P0DOX2 0.06163643
30 A0A0C4DH43 0.06140723</code></pre>
<pre class="r"><code># PC15
# Visualize PCs
ggplot(pcTab, aes(x=Recurrence, y=PC15, col=Recurrence, fill=Recurrence)) +
  geom_boxplot(alpha = 0.5, outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.2), size = 2, show.legend = F) +
  labs(title = &#39;Normal tissue proteomics&#39;) +
  scale_color_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  scale_fill_manual(values=c(&#39;#00BFC4&#39;, &#39;#F8766D&#39;)) +
  ggpubr::stat_compare_means(method = &#39;t.test&#39;, paired = F, size = 5,
                             method.args = list(var.equal = T),
                             show.legend = F)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-55-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot molecular signatures in input data
# Extract top features with highest loadings
topFeats &lt;- pcTopFeatTab$PC15$Feature[1:30]
topFeatsIdx &lt;- sapply(topFeats, function(feat) {which(rownames(datMat) == feat)})
pheatmap::pheatmap(datMat[topFeatsIdx, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F,
                   scale = &#39;row&#39;, main = &#39;Normal tissue proteomics - PC15&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-55-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Display top features with highest loadings of PC of interest
pcTopFeatTab$PC15</code></pre>
<pre><code>      Feature    Loading
1      P08047 0.11781901
2      Q9UJ83 0.11605292
3  A0A0C4DH43 0.09996109
4      P01701 0.09256788
5      Q5JTJ3 0.08882516
6      O15120 0.08263572
7      P23946 0.08090850
8      P34931 0.08026788
9      Q16602 0.07993440
10     Q9Y5U8 0.07748260
11     Q13938 0.07634917
12     P02741 0.07352199
13     P14780 0.07193395
14     O75525 0.06987063
15     Q6PIJ6 0.06820468
16     Q9P2N5 0.06783415
17     P13164 0.06779685
18     P10153 0.06596136
19     P06732 0.06539609
20 A0A075B6K4 0.06421505
21     Q9NX47 0.06384574
22     P22792 0.06270671
23     Q9NP77 0.06141847
24     Q8N4T0 0.06104694
25     P82987 0.06088295
26     P23219 0.06075763
27     Q9Y2E4 0.06047425
28     P15088 0.05964184
29     O95848 0.05948378
30     P01909 0.05910240</code></pre>
</div>
<div id="difference-in-tissue-conditions-1" class="section level2">
<h2>Difference in tissue conditions</h2>
<p>Tumor matrix - Normal matrix<br />
</p>
<p>Display significant associations between features (Var1) and cancer
recurrence (Var2) and observe molecular signatures in input data through
heatmap. Note that features in heatmap are ordered by their
t-statistics.</p>
<pre class="r"><code># Subset Tumor samples
smpTumorIdx &lt;- which(colData(proTissueNorm)$Condition == &#39;Tumor&#39;)
proTumor &lt;- proTissueNorm[, smpTumorIdx]
# Subset Normal samples
smpNormalIdx &lt;- which(colData(proTissueNorm)$Condition == &#39;Normal&#39;)
proNormal &lt;- proTissueNorm[, smpNormalIdx]
# Modify sample names for matching two datasets
colnames(proTumor) &lt;- stringr::str_remove(colnames(proTumor), &#39;_TG&#39;)
colnames(proNormal) &lt;- stringr::str_remove(colnames(proNormal), &#39;_NG&#39;)
# Retrieve data matrix
proTumorMat &lt;- SummarizedExperiment::assay(proTumor)
proNormalMat &lt;- SummarizedExperiment::assay(proNormal)
# Subtract follow-up matrix by baseline matrix
# if (identical(colnames(proTumorMat), colnames(proNormalMat)) &amp;
#     identical(rownames(proTumorMat), rownames(proNormalMat))) {
#   proDiffMat &lt;- proTumorMat - proNormalMat
# } else {
#   print(&#39;Align samples or features first.&#39;)
# }
#### Manually align samples ####
if (identical(sort(colnames(proTumorMat)), sort(colnames(proNormalMat)))) {
  smpOrder &lt;- sort(colnames(proTumorMat))
  # For computing difference between Tumor and Normal samples
  proTumorMat &lt;- proTumorMat[, smpOrder]
  proNormalMat &lt;- proNormalMat[, smpOrder]
  proDiffMat &lt;- proTumorMat - proNormalMat
  # For Creating SE object
  proTumor &lt;- proTumor[, smpOrder]
  proNormal &lt;- proNormal[, smpOrder]
}
# Retrieve samples metadata for creating SE object
colAnno &lt;- dplyr::select(as.data.frame(colData(proTumor)), Patient, Recurrence)
rowAnno &lt;- as.data.frame(rowData(proTumor))
proDiff &lt;- SummarizedExperiment(assays = proDiffMat, colData = colAnno,
                                rowData = rowAnno)

# Perform analysis
proTissueDiffRes &lt;- SOA(proDiff, fac = &#39;Recurrence&#39;, num_feats = 30)
datMat &lt;- proTissueDiffRes$data
smpAnno &lt;- proTissueDiffRes$smpMetadata
tFeatSigRes &lt;- proTissueDiffRes$tFeatSigRes
tPCASigRes &lt;- proTissueDiffRes$tPCASigRes
pcaRes &lt;- proTissueDiffRes$pcaRes

# Display features that can significantly separate recurrence and non-recurrence
# patients
tFeatSigRes</code></pre>
<pre><code>             Var1       Var2         pVal    tStats   pValAdj
1          Q96DE0 Recurrence 0.0005749216  4.170311 0.9964575
2          Q9Y3B8 Recurrence 0.0027253883 -3.471219 0.9964575
3          Q8TE77 Recurrence 0.0032217963 -3.395735 0.9964575
4          P54687 Recurrence 0.0034965023  3.358763 0.9964575
5          O95260 Recurrence 0.0039828627  3.299823 0.9964575
6          O60216 Recurrence 0.0049321187  3.202792 0.9964575
7          Q9H6S3 Recurrence 0.0056722946 -3.139099 0.9964575
8          Q8NDT2 Recurrence 0.0058682502  3.123598 0.9964575
9          P46108 Recurrence 0.0059083419 -3.120489 0.9964575
10         P82675 Recurrence 0.0060652298 -3.108518 0.9964575
11         P10515 Recurrence 0.0073932709 -3.017811 0.9964575
12         Q969X6 Recurrence 0.0085509421  2.950847 0.9964575
13         Q9Y4A5 Recurrence 0.0087572407  2.939844 0.9964575
14         O94842 Recurrence 0.0090720626  2.923528 0.9964575
15         Q8NFJ5 Recurrence 0.0097431954 -2.890500 0.9964575
16         Q2TAL8 Recurrence 0.0098495696  2.885468 0.9964575
17         Q12756 Recurrence 0.0118435814  2.799730 0.9964575
18         Q06187 Recurrence 0.0120958702  2.789889 0.9964575
19         Q9UNS2 Recurrence 0.0127318997  2.765929 0.9964575
20         Q9H9A5 Recurrence 0.0127797544  2.764173 0.9964575
21         Q9BYD3 Recurrence 0.0134250322 -2.741090 0.9964575
22         P04075 Recurrence 0.0144621667  2.706128 0.9964575
23         Q92974 Recurrence 0.0146889246  2.698804 0.9964575
24         Q53F19 Recurrence 0.0150154366  2.688446 0.9964575
25         Q8WUH2 Recurrence 0.0151457687 -2.684371 0.9964575
26         Q01650 Recurrence 0.0165260401 -2.643158 0.9964575
27         Q16363 Recurrence 0.0177238906 -2.609966 0.9964575
28         Q96SY0 Recurrence 0.0185648326  2.587914 0.9964575
29         Q96S55 Recurrence 0.0191681600  2.572670 0.9964575
30         Q14247 Recurrence 0.0191921504 -2.572073 0.9964575
31         P18669 Recurrence 0.0193318493  2.568612 0.9964575
32         Q14139 Recurrence 0.0198464374 -2.556064 0.9964575
33         Q9P0M6 Recurrence 0.0199332665  2.553978 0.9964575
34         P25787 Recurrence 0.0208596712  2.532233 0.9964575
35         Q07960 Recurrence 0.0210127785 -2.528727 0.9964575
36         Q8N983 Recurrence 0.0214351106 -2.519181 0.9964575
37         P54289 Recurrence 0.0220222764 -2.506200 0.9964575
38         Q9Y512 Recurrence 0.0223947914 -2.498132 0.9964575
39         Q9H0U6 Recurrence 0.0223962907 -2.498100 0.9964575
40         P53420 Recurrence 0.0236864461 -2.471105 0.9964575
41         P98179 Recurrence 0.0256930732 -2.431749 0.9964575
42         Q8IVS2 Recurrence 0.0259238748 -2.427409 0.9964575
43         Q05639 Recurrence 0.0260107145  2.425785 0.9964575
44         Q96A35 Recurrence 0.0262563837 -2.421219 0.9964575
45         Q12972 Recurrence 0.0262919288  2.420562 0.9964575
46         Q9C0C2 Recurrence 0.0268711961 -2.409966 0.9964575
47         P20340 Recurrence 0.0270038491 -2.407569 0.9964575
48         Q9UPT8 Recurrence 0.0271146453  2.405576 0.9964575
49         O60716 Recurrence 0.0278402636 -2.392708 0.9964575
50         Q05655 Recurrence 0.0284796478  2.381626 0.9964575
51         Q8TCJ2 Recurrence 0.0290074343  2.372652 0.9964575
52         P09104 Recurrence 0.0294570329  2.365127 0.9964575
53         P61964 Recurrence 0.0300819656  2.354843 0.9964575
54         Q14839 Recurrence 0.0307289301  2.344404 0.9964575
55         Q9UJX2 Recurrence 0.0308081081 -2.343140 0.9964575
56         O60256 Recurrence 0.0308620026  2.342282 0.9964575
57         P62195 Recurrence 0.0313501956  2.334570 0.9964575
58         P25325 Recurrence 0.0313761590 -2.334163 0.9964575
59         Q6P5R6 Recurrence 0.0316469173  2.329937 0.9964575
60         Q9BUI4 Recurrence 0.0316606622  2.329724 0.9964575
61         O00505 Recurrence 0.0318408747 -2.326931 0.9964575
62         A8MYP8 Recurrence 0.0326226570 -2.314983 0.9964575
63         E9PRG8 Recurrence 0.0328733780 -2.311207 0.9964575
64         Q9Y305 Recurrence 0.0330837871 -2.308059 0.9964575
65         Q8TC71 Recurrence 0.0334728898 -2.302286 0.9964575
66         Q2TAY7 Recurrence 0.0335117371  2.301713 0.9964575
67         Q6NXG1 Recurrence 0.0336272222 -2.300014 0.9964575
68         Q7Z6K5 Recurrence 0.0340275237 -2.294164 0.9964575
69         P30533 Recurrence 0.0341212921 -2.292803 0.9964575
70         Q9H0L4 Recurrence 0.0343581808  2.289380 0.9964575
71         O95983 Recurrence 0.0348016114  2.283031 0.9964575
72         P61019 Recurrence 0.0352534268 -2.276638 0.9964575
73         Q5TAQ9 Recurrence 0.0353883010  2.274745 0.9964575
74         Q9NS86 Recurrence 0.0356006848 -2.271776 0.9964575
75         O60486 Recurrence 0.0356916565 -2.270510 0.9964575
76         Q96MM6 Recurrence 0.0363622207 -2.261266 0.9964575
77         Q4KMP7 Recurrence 0.0364774583 -2.259693 0.9964575
78         Q9BY44 Recurrence 0.0371255403  2.250933 0.9964575
79         P22670 Recurrence 0.0375209498  2.245657 0.9964575
80         P30626 Recurrence 0.0384823559 -2.233041 0.9964575
81         Q96GC5 Recurrence 0.0386691945 -2.230622 0.9964575
82         O60885 Recurrence 0.0391683722  2.224214 0.9964575
83         Q96I59 Recurrence 0.0398252449 -2.215895 0.9964575
84         Q9Y3B7 Recurrence 0.0401369919 -2.211991 0.9964575
85         Q96DG6 Recurrence 0.0402100808 -2.211080 0.9964575
86         P31350 Recurrence 0.0405664611  2.206658 0.9964575
87         O15160 Recurrence 0.0409850287 -2.201510 0.9964575
88         Q8NBI5 Recurrence 0.0410613099 -2.200577 0.9964575
89         Q9UK41 Recurrence 0.0414860894 -2.195411 0.9964575
90         Q13247 Recurrence 0.0423686134  2.184831 0.9964575
91         Q16643 Recurrence 0.0427918840  2.179828 0.9964575
92         Q9NZQ3 Recurrence 0.0429066683  2.178479 0.9964575
93         O75602 Recurrence 0.0433543800 -2.173249 0.9964575
94         Q8N4Q0 Recurrence 0.0439942558  2.165860 0.9964575
95         Q7KZN9 Recurrence 0.0449567370  2.154928 0.9964575
96         Q92688 Recurrence 0.0457064532  2.146561 0.9964575
97         P51531 Recurrence 0.0458790083  2.144652 0.9964575
98         Q86Y79 Recurrence 0.0462523510 -2.140546 0.9964575
99         P11172 Recurrence 0.0462610301  2.140451 0.9964575
100        O15305 Recurrence 0.0462665487 -2.140391 0.9964575
101        P09874 Recurrence 0.0464309191  2.138593 0.9964575
102        Q5VZK9 Recurrence 0.0466403887 -2.136310 0.9964575
103        Q9P013 Recurrence 0.0468812591 -2.133697 0.9964575
104        Q69YL0 Recurrence 0.0469525752 -2.132926 0.9964575
105        Q15109 Recurrence 0.0472472308 -2.129750 0.9964575
106        P41208 Recurrence 0.0474513250 -2.127561 0.9964575
107        Q99460 Recurrence 0.0477198764  2.124693 0.9964575
108        O15120 Recurrence 0.0478157981  2.123673 0.9964575
109        Q14202 Recurrence 0.0478605903  2.123197 0.9964575
110        P29373 Recurrence 0.0478849379  2.122938 0.9964575
111        Q16576 Recurrence 0.0479036717  2.122739 0.9964575
112        Q92530 Recurrence 0.0480562504 -2.121122 0.9964575
113        P78347 Recurrence 0.0485484683  2.115938 0.9964575
114 Q12974;Q93096 Recurrence 0.0490058231 -2.111163 0.9964575
115        Q12965 Recurrence 0.0491431358 -2.109737 0.9964575
116        Q9P2D3 Recurrence 0.0493254240  2.107850 0.9964575
117        P23921 Recurrence 0.0493503403  2.107593 0.9964575
118        Q9C0B1 Recurrence 0.0499582671  2.101348 0.9964575</code></pre>
<pre class="r"><code># Plot molecular signatures in input data
# Order features according to t-statistics
featOrder &lt;- dplyr::arrange(tFeatSigRes, dplyr::desc(tStats))$Var1
# Arrange samples to make same groups in same clusters
smpOrder &lt;- c(which(smpAnno$Recurrence == &#39;No&#39;),
              which(smpAnno$Recurrence == &#39;Yes&#39;))
# Prepare recurrence table for annotating samples in heatmap
smpRecur &lt;- dplyr::select(smpAnno, Sample, Recurrence) %&gt;%
  tibble::column_to_rownames(&#39;Sample&#39;)
pheatmap::pheatmap(datMat[featOrder, smpOrder],
                   annotation_col = smpRecur,
                   color = colorRampPalette(c(&#39;navy&#39;, &#39;white&#39;, &#39;red&#39;))(100),
                   cluster_rows = F, cluster_cols = F, show_rownames = F,
                   scale = &#39;row&#39;, main = &#39;Differential plasma proteomics&#39;)</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-56-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display significant associations between PCs (Var1) and cancer
recurrence (Var2)</p>
<pre class="r"><code># Display PCs that can significantly separate recurrence and non-recurrence
# patients
tPCASigRes</code></pre>
<pre><code>[1] Var1    Var2    pVal    tStats  pValAdj
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Display proportion of significant features in each dataset in terms
of separating cancer recurrence and non-recurrence patient groups.
Plasma and Tissue Metabolomics has 630 features, Plasma Proteomics has
435 features, and Tissue Proteomics has 5193 features.</p>
<pre class="r"><code># Barplot about proportion of significant features in each dataset
# Compute proportion of significant features
metaBaseSigFeats &lt;- nrow(metaBaseRes$tFeatSigRes) / nrow(metaBaseRes$data) * 100
metaFoloSigFeats &lt;- nrow(metaFoloRes$tFeatSigRes) / nrow(metaFoloRes$data) * 100
# metaPlasmaDiffSigFeats &lt;- nrow(metaPlasmaDiffRes$tFeatSigRes) / nrow(metaPlasmaDiffRes$data) * 100
proBaseSigFeats &lt;- nrow(proBaseRes$tFeatSigRes) / nrow(proBaseRes$data) * 100
proFoloSigFeats &lt;- nrow(proFoloRes$tFeatSigRes) / nrow(proFoloRes$data) * 100
# proPlasmaDiffSigFeats &lt;- nrow(proPlasmaDiffRes$tFeatSigRes) / nrow(proPlasmaDiffRes$data) * 100
metaTumorSigFeats &lt;- nrow(metaTumorRes$tFeatSigRes) / nrow(metaTumorRes$data) * 100
metaNormalSigFeats &lt;- nrow(metaNormalRes$tFeatSigRes) / nrow(metaNormalRes$data) * 100
# metaTissueDiffSigFeats &lt;- nrow(metaTissueDiffRes$tFeatSigRes) / nrow(metaTissueDiffRes$data) * 100
proTumorSigFeats &lt;- nrow(proTumorRes$tFeatSigRes) / nrow(proTumorRes$data) * 100
proNormalSigFeats &lt;- nrow(proNormalRes$tFeatSigRes) / nrow(proNormalRes$data) * 100
# proTissueDiffSigFeats &lt;- nrow(proTissueDiffRes$tFeatSigRes) /nrow(proTissueDiffRes$data) * 100

# Prepare needed information for barplot
dat &lt;- rev(c(&#39;Baseline Plasma Metabolomics&#39;, &#39;Follow-up Plasma Metabolomics&#39;,#&#39;Differential Plasma Metabolomics&#39;,
             &#39;Baseline Plasma Proteomics&#39;, &#39;Follow-up Plasma Proteomics&#39;,#&#39;Differential Plasma Proteomics&#39;,
             &#39;Tumor Tissue Metabolomics&#39;, &#39;Normal Tissue Metabolomics&#39;,#&#39;Differential Tissue Metabolomics&#39;,
             &#39;Tumor Tissue Proteomics&#39;, &#39;Normal Tissue Proteomics&#39;))#,&#39;Differential Tissue Proteomics&#39;
datModal &lt;- rev(c(rep(&#39;Plasma Metabolomics&#39;, 2), rep(&#39;Plasma Proteomics&#39;, 2),
                  rep(&#39;Tissue Metabolomics&#39;, 2), rep(&#39;Tissue Proteomics&#39;, 2)))
proportion &lt;- rev(c(metaBaseSigFeats, metaFoloSigFeats, #metaPlasmaDiffSigFeats,
                    proBaseSigFeats, proFoloSigFeats, #proPlasmaDiffSigFeats,
                    metaTumorSigFeats, metaNormalSigFeats, #metaTissueDiffSigFeats,
                    proTumorSigFeats, proNormalSigFeats)) #,proTissueDiffSigFeats
sigFeatPropTab &lt;- data.frame(Data = factor(dat, levels = dat),
                             Data_modality = datModal,
                             Proportion = proportion)

ggplot(sigFeatPropTab, aes(x=Data, y=Proportion, fill=Data_modality)) +
  geom_bar(stat = &#39;identity&#39;) +
  coord_flip() +
  labs(y = &#39;Percentage (%)&#39;,
       title = &#39;Proportion of Cancer Recurrence-Related Features&#39;) +
  scale_fill_discrete(name = &#39;Data modality&#39;) +
  theme(axis.title = element_text(size = 16), axis.text = element_text(size = 12),
        legend.title = element_text(size = 14), legend.text = element_text(size = 12))</code></pre>
<p><img src="figure/preliminary_analysis_targeted_DDA.Rmd/unnamed-chunk-58-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste0(result_path, &#39;AG_Hell/proporSigFeats.png&#39;), device = &#39;png&#39;, dpi = 400,
#        height = 8, width = 10)</code></pre>
<ol style="list-style-type: decimal">
<li>All sorts of datasets in this script provide information for
predicting patient cancer recurrence and the PCA models built on them
have more or less similar performance. If I must pick one most
impressive, I would say Targeted (Tumor and Differential) Tissue
Metabolomics from AG Hell is a little bit more surprising. Targeted
Plasma and Tissue Metabolomics contains the same feature list and Plasma
even results in more significant features, but Tissue can somehow
slightly better predict cancer recurrences. It might be that Tissue
Metabolomics has fewer but more robust Recurrence-related
features.<br />
</li>
<li>Overall, Plasma datasets got higher proportions of
Recurrence-related features than Tissue when they have the same feature
list (Metabolomics) or even Tissue have much larger feature space
(Proteomics). Hence, Plasma samples seem to be more informative and
Tissue samples are more noisy. Yet, since Targeted and DDA methods are
biased (and sample size is very small), these hypotheses have to be
validated by unbiased methods, like Untargeted (please check analysis
results of the data from AG Hopf) and DIA.</li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.2.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.1                
 [5] purrr_1.0.1                 readr_2.1.4                
 [7] tidyr_1.3.0                 tibble_3.2.1               
 [9] tidyverse_2.0.0             umap_0.2.10.0              
[11] ggfortify_0.4.16            ggrepel_0.9.3              
[13] ggplot2_3.4.1               SummarizedExperiment_1.26.1
[15] Biobase_2.56.0              GenomicRanges_1.48.0       
[17] GenomeInfoDb_1.32.4         IRanges_2.30.1             
[19] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[21] MatrixGenerics_1.8.1        matrixStats_0.63.0         
[23] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7           fs_1.6.1               RColorBrewer_1.1-3    
 [4] httr_1.4.5             rprojroot_2.0.3        backports_1.4.1       
 [7] tools_4.2.1            bslib_0.4.2            utf8_1.2.3            
[10] R6_2.5.1               colorspace_2.1-0       withr_2.5.0           
[13] tidyselect_1.2.0       gridExtra_2.3          processx_3.8.0        
[16] compiler_4.2.1         git2r_0.31.0           cli_3.6.1             
[19] DelayedArray_0.22.0    labeling_0.4.2         sass_0.4.5            
[22] scales_1.2.1           callr_3.7.3            askpass_1.1           
[25] digest_0.6.31          rmarkdown_2.21         XVector_0.36.0        
[28] pkgconfig_2.0.3        htmltools_0.5.5        fastmap_1.1.1         
[31] highr_0.10             rlang_1.1.0            rstudioapi_0.14       
[34] jquerylib_0.1.4        generics_0.1.3         farver_2.1.1          
[37] jsonlite_1.8.4         car_3.1-1              RCurl_1.98-1.10       
[40] magrittr_2.0.3         GenomeInfoDbData_1.2.8 Matrix_1.5-3          
[43] Rcpp_1.0.10            munsell_0.5.0          fansi_1.0.4           
[46] abind_1.4-5            reticulate_1.28        lifecycle_1.0.3       
[49] stringi_1.7.12         whisker_0.4.1          yaml_2.3.7            
[52] carData_3.0-5          zlibbioc_1.42.0        grid_4.2.1            
[55] promises_1.2.0.1       lattice_0.20-45        hms_1.1.3             
[58] knitr_1.42             ps_1.7.3               pillar_1.9.0          
[61] ggpubr_0.6.0           ggsignif_0.6.4         glue_1.6.2            
[64] evaluate_0.20          getPass_0.2-2          png_0.1-8             
[67] vctrs_0.6.1            tzdb_0.3.0             httpuv_1.6.9          
[70] gtable_0.3.3           openssl_2.0.6          cachem_1.0.7          
[73] xfun_0.38              broom_1.0.4            RSpectra_0.16-1       
[76] rstatix_0.7.2          later_1.3.0            pheatmap_1.0.12       
[79] timechange_0.2.0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
