<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Qian-Wu Liao" />


<title>Preprocessing: Method development lung cancer patient cohort</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMART-CARE: Lung Cancer Method Development Cohort</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Preprocessing: Method development lung
cancer patient cohort</h1>
<h4 class="author">Qian-Wu Liao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>SMART-CARE_LungCancer_MethodDev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/qianwu/Desktop/SMART-CARE_LungCancer_MethodDev
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree717cb2321703f047d6659449429f744bc5e74644targetblank717cb23a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/717cb2321703f047d6659449429f744bc5e74644" target="_blank">717cb23</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLiaoQianWuSMARTCARELungCancerMethodDevtree717cb2321703f047d6659449429f744bc5e74644targetblank717cb23a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/tree/717cb2321703f047d6659449429f744bc5e74644" target="_blank">717cb23</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/feature_selection.Rmd
    Untracked:  analysis/mofa_downstream_analysis.Rmd
    Untracked:  code/Junyan/
    Untracked:  code/about.Rmd
    Untracked:  code/dataset_list.R
    Untracked:  code/feature_selection_old.Rmd
    Untracked:  code/license.Rmd
    Untracked:  code/mofa_downstream_analysis.Rmd
    Untracked:  code/mofa_factors_investigation.Rmd
    Untracked:  code/mofa_varied_omics_combinations.Rmd
    Untracked:  code/workflowr_commands.R
    Untracked:  data/AG_Hell/
    Untracked:  data/AG_Hopf/
    Untracked:  data/AG_Klingmuller/
    Untracked:  data/AG_Krijgsveld/
    Untracked:  data/MethodDev_LungCancer_Datasets.RData
    Untracked:  data/mofa/
    Untracked:  data/patient_metadata.tsv
    Untracked:  data/rf/
    Untracked:  data/stats/
    Untracked:  output/TAC_2023/
    Untracked:  output/YIG_seminar/
    Untracked:  output/preliminary_analysis_untargeted.html

Unstaged changes:
    Deleted:    analysis/mofa_varied_omics_combinations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/preprocessing_DIA_new.Rmd</code>)
and HTML (<code>docs/preprocessing_DIA_new.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/9a0840b76aceb404bc0147e8caca1cd3a5a42beb/analysis/preprocessing_DIA_new.Rmd" target="_blank">9a0840b</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-08-04
</td>
<td>
Redo feature filtering for DIA proteomics and rerun subsequent analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/83a849a244396dfe32e55476a8e333eebc667fb4/analysis/preprocessing_DIA_new.Rmd" target="_blank">83a849a</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-31
</td>
<td>
Do single-omics analysis of new DIA plasma and tissue proteomics from AG
Krijgsveld and compare it with other proteomics we have
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LiaoQianWu/SMART-CARE_LungCancer_MethodDev/blob/1a03bdbbda2ba728fd9c129eff4e0a9d9029978a/analysis/preprocessing_DIA_new.Rmd" target="_blank">1a03bdb</a>
</td>
<td>
LiaoQianWu
</td>
<td>
2023-07-27
</td>
<td>
Preprocess new DIA plasma and tissue proteomics from AG Krijgsveld
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><font size='4'> Description: Preprocess New Plasma and Tissue DIA
Proteomics generated by Karim Aljakouch from AG Krijgsveld, simply
including data cleansing and data normalization (VSN). All needed
information of each dataset (2 in total) was then stored in
SummarizedExperiment objects for further analyses.<br />
</p>
<p>Update: Modify feature filtering criterion: Remove features
quantified in less than 2/3 of samples to avoid analysis results that
one subgroup is fully missing since we got a great deal of significant
cancer recurrence-related features. </font></p>
<p>Load libraries</p>
<pre class="r"><code>library(&#39;readr&#39;)
library(&#39;readxl&#39;)
library(&#39;vsn&#39;)
library(&#39;visdat&#39;)
library(&#39;SummarizedExperiment&#39;)
library(&#39;tidyverse&#39;)
# Load user-defined functions
source(&#39;./code/misc.R&#39;)

# Set plot theme
th &lt;- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = &#39;bold&#39;),
        axis.text = element_text(face = &#39;bold&#39;),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))

# List sample annotations to keep in SE objects
anno &lt;- c(&#39;Condition&#39;, &#39;Patient&#39;, &#39;Recurrence&#39;, &#39;Gender&#39;, &#39;Age&#39;, &#39;Smoking&#39;,
          &#39;Stage&#39;, &#39;Identifier&#39;)</code></pre>
<div id="plasma-proteomics" class="section level1">
<h1>Plasma Proteomics</h1>
<pre class="r"><code># Load unprocessed data
proPlasmaTab &lt;- readr::read_delim(&#39;./data/AG_Krijgsveld/20230726_Thorax_Method_Est_plasma.txt&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Identifier&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Identifier = stringr::str_remove_all(Identifier, &#39;_4\\.raw$&#39;))

# Load sample metadata from preprocessed Targeted Metabolomics SE object
smpAnno &lt;- colData(readRDS(&#39;./data/AG_Hell/metaPlasmaNorm.rds&#39;)) %&gt;%
  tibble::as_tibble(rownames = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Identifier = stringr::str_remove(Identifier, &#39;_3$&#39;),
                TimePoint = dplyr::case_when(Condition == &#39;Baseline&#39; ~ &#39;Bseline&#39;,
                                             Condition != &#39;Baseline&#39; ~ &#39;2 years later&#39;))

# Combine all information into a table
proPlasmaTab &lt;- dplyr::left_join(proPlasmaTab, smpAnno, by = &#39;Identifier&#39;) %&gt;%
  # Exclude Sample &#39;G045W7J8_P_B&#39; due to extremely high missingness (72%)
  dplyr::filter(Sample != &#39;G045W7J8_P_B&#39;)

# Convert long data to SE object (wide data)
proPlasma &lt;- df2SummExp(proPlasmaTab, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;,
                        values = &#39;Abundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                        col_anno = anno)</code></pre>
<p>Display dimensions of data (39 samples and 1097 features)</p>
<pre class="r"><code>dim(proPlasma)</code></pre>
<pre><code>[1] 1097   39</code></pre>
<p>=&gt; Baseline sample ‘G045W7J8_P_B’ is excluded due to extremely
high missingness (72%). <br/> <br/></p>
<p>Display data missingness</p>
<pre class="r"><code>exprMat &lt;- SummarizedExperiment::assay(proPlasma)
visdat::vis_miss(exprMat, cluster = T, sort_miss = T) +
  labs(y = &#39;Features&#39;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 0, size = 11, face = &#39;bold&#39;),
        axis.text.y = element_blank(),
        axis.title.y = element_text(size = 13, face = &#39;bold&#39;),
        legend.text = element_text(size = 12, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display distribution of original data</p>
<pre class="r"><code>ggplot(proPlasmaTab, aes(x=Sample, y=Abundance)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(y = &#39;Protein abundance&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Remove features quantified in less than 2/3 of samples for
more reliable data normalization</strong></p>
<pre class="r"><code># Remove features quantified in less than 2/3 of samples
rmFeats &lt;- dplyr::group_by(proPlasmaTab, Protein.Group) %&gt;%
  # Compute proportion of observed data points of each group
  dplyr::summarise(frac_nonNA = round(sum(!is.na(Abundance)) / length(Abundance), 2)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Keep proteins observed in less than 2/3 of samples of groups
  dplyr::filter(frac_nonNA &lt; 0.67) %&gt;%
  dplyr::pull(Protein.Group)
proPlasmaTab &lt;- dplyr::filter(proPlasmaTab, !Protein.Group %in% rmFeats)

# Convert long data to SE object
proPlasma &lt;- df2SummExp(proPlasmaTab, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;,
                        values = &#39;Abundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                        col_anno = anno)</code></pre>
<pre class="r"><code># Remove features quantified in less than 2/3 of samples in all four sample groups
# of interest (Baseline/Follow-up Recurrence/Non-recurrence) for more reliable data normalization\
# Missing data is more likely to happen in proteins with small abundances due to detection
# limit. Image comparison between two sample groups, certain proteins are highly missing
# in one group and present in the other

# Remove features quantified in less than 2/3 of samples in all sample groups of interest
# Grouping like this since Baseline and Follow-up samples will be analyzed separately
rmFeats &lt;- dplyr::group_by(proPlasmaTab, Protein.Group, TimePoint, Recurrence) %&gt;%
  # Compute proportion of observed data points of each group
  dplyr::summarise(frac_nonNA = round(sum(!is.na(Abundance)) / length(Abundance), 2)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Keep proteins observed in less than 2/3 of samples of groups
  dplyr::filter(frac_nonNA &lt; 0.67) %&gt;%
  dplyr::group_by(Protein.Group) %&gt;%
  dplyr::summarise(Count = length(Protein.Group)) %&gt;%
  # Keep proteins observed in less than 2/3 of samples of all groups
  dplyr::filter(Count == 4) %&gt;%
  dplyr::pull(Protein.Group)
proPlasmaTab &lt;- dplyr::filter(proPlasmaTab, !Protein.Group %in% rmFeats)

# Convert long data to SE object
proPlasma &lt;- df2SummExp(proPlasmaTab, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;,
                        values = &#39;Abundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                        col_anno = anno)

# Got 648 features after filtering</code></pre>
<p>Display dimensions of filtered data (39 samples and 590 features)</p>
<pre class="r"><code>dim(proPlasma)</code></pre>
<pre><code>[1] 590  39</code></pre>
<p>Display data missingness</p>
<pre class="r"><code>exprMat &lt;- SummarizedExperiment::assay(proPlasma)
visdat::vis_miss(exprMat, cluster = T, sort_miss = T) +
  labs(y = &#39;Features&#39;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 0, size = 11, face = &#39;bold&#39;),
        axis.text.y = element_blank(),
        axis.title.y = element_text(size = 13, face = &#39;bold&#39;),
        legend.text = element_text(size = 12, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display distribution of filtered data</p>
<pre class="r"><code>ggplot(proPlasmaTab, aes(x=Sample, y=Abundance)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(y = &#39;Protein abundance&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Perform VSN and display distribution of normalized data</p>
<pre class="r"><code># Perform VSN
exprMat &lt;- as.matrix(SummarizedExperiment::assay(proPlasma))
fit &lt;- vsnMatrix(exprMat)
proPlasmaNorm &lt;- proPlasma
SummarizedExperiment::assay(proPlasmaNorm) &lt;- predict(fit, exprMat)

# Save normalized data
# saveRDS(proPlasmaNorm, &#39;./data/AG_Krijgsveld/new_proPlasmaNorm_DIA.rds&#39;)

# Convert SE object to long data for plotting
proPlasmaTabNorm &lt;- summExp2df(proPlasmaNorm, assay = &#39;Abundance&#39;,
                               row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;)

ggplot(proPlasmaTabNorm, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(y = &#39;Normalized protein abundance&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Store median normalized data for computing log2(FC)
exprMat &lt;- as.matrix(assay(proPlasma))
sampleAnno &lt;- tibble::as_tibble(colData(proPlasma), rownames = &#39;Sample&#39;)
featAnno &lt;- tibble::as_tibble(rowData(proPlasma), rownames = &#39;Feature&#39;)
proPlasmaMediTab &lt;- limma::normalizeBetweenArrays(exprMat, method = &#39;scale&#39;) %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;,
                      values_to = &#39;mediAbundance&#39;) %&gt;%
  dplyr::mutate(mediAbundance = log2(mediAbundance)) %&gt;%
  dplyr::left_join(sampleAnno, by = &#39;Sample&#39;) %&gt;%
  dplyr::left_join(featAnno, by = &#39;Feature&#39;)

ggplot(proPlasmaMediTab, aes(x=Sample, y=mediAbundance)) +
  geom_boxplot() +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

proPlasmaMedi &lt;- df2SummExp(proPlasmaMediTab, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;,
                            values = &#39;mediAbundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                            col_anno = anno)
# saveRDS(proPlasmaMedi, &#39;./data/AG_Krijgsveld/new_proPlasmaMedi_DIA.rds&#39;)</code></pre>
</div>
<div id="tissue-proteomics" class="section level1">
<h1>Tissue Proteomics</h1>
<pre class="r"><code># Load unprocessed data
proTissueTab &lt;- readr::read_delim(&#39;./data/AG_Krijgsveld/20230726_Thorax_Method_Est_tissue.txt&#39;) %&gt;%
  dplyr::select(-c(Protein.Ids, First.Protein.Description)) %&gt;%
  tidyr::pivot_longer(cols = -c(&#39;Protein.Group&#39;, &#39;Protein.Names&#39;, &#39;Genes&#39;),
                      names_to = &#39;Identifier&#39;,
                      values_to = &#39;Abundance&#39;) %&gt;%
  dplyr::mutate(Protein.Names = stringr::str_remove_all(Protein.Names, &#39;_HUMAN&#39;),
                Identifier = stringr::str_remove_all(Identifier, &#39;_4\\.raw$&#39;))

# Load sample metadata from preprocessed Targeted Metabolomics SE object
smpAnno &lt;- colData(readRDS(&#39;./data/AG_Hell/metaTissueNorm.rds&#39;)) %&gt;%
  tibble::as_tibble(rownames = &#39;Sample&#39;) %&gt;%
  dplyr::mutate(Identifier = stringr::str_remove(Identifier, &#39;_3$&#39;))

# Combine all information into a table
proTissueTab &lt;- dplyr::left_join(proTissueTab, smpAnno, by = &#39;Identifier&#39;)

# Convert long data to SE object (wide data)
proTissue &lt;- df2SummExp(proTissueTab, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;,
                        values = &#39;Abundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                        col_anno = anno)</code></pre>
<p>Display dimensions of data (40 samples and 7537 features)</p>
<pre class="r"><code>dim(proTissue)</code></pre>
<pre><code>[1] 7537   40</code></pre>
<p>Display data missingness</p>
<pre class="r"><code>exprMat &lt;- SummarizedExperiment::assay(proTissue)
visdat::vis_miss(exprMat, cluster = T, sort_miss = T) +
  labs(y = &#39;Features&#39;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 0, size = 11, face = &#39;bold&#39;),
        axis.text.y = element_blank(),
        axis.title.y = element_text(size = 13, face = &#39;bold&#39;),
        legend.text = element_text(size = 12, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display distribution of original data</p>
<pre class="r"><code>ggplot(proTissueTab, aes(x=Sample, y=Abundance)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(y = &#39;Protein abundance&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><strong>Remove features quantified in less than 2/3 of samples for
more reliable data normalization</strong></p>
<pre class="r"><code># Remove features quantified in less than 2/3 of samples
rmFeats &lt;- dplyr::group_by(proTissueTab, Protein.Group) %&gt;%
  # Compute proportion of observed data points of each group
  dplyr::summarise(frac_nonNA = round(sum(!is.na(Abundance)) / length(Abundance), 2)) %&gt;%
  dplyr::ungroup() %&gt;%
  # Keep proteins observed in less than 2/3 of samples of groups
  dplyr::filter(frac_nonNA &lt; 0.67) %&gt;%
  dplyr::pull(Protein.Group)
proTissueTab &lt;- dplyr::filter(proTissueTab, !Protein.Group %in% rmFeats)

# Convert long data to SE object
proTissue &lt;- df2SummExp(proTissueTab, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;,
                        values = &#39;Abundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                        col_anno = anno)</code></pre>
<pre class="r"><code># Remove features quantified in less than 2/3 of samples in all four sample groups
# of interest (Normal/Tumor Recurrence/Non-recurrence) for more reliable data normalization

# Remove features quantified in less than 2/3 of samples in all sample groups of interest
rmFeats &lt;- dplyr::group_by(proTissueTab, Protein.Group, Condition, Recurrence) %&gt;%
  dplyr::summarise(frac_nonNA = round(sum(!is.na(Abundance)) / length(Abundance), 2)) %&gt;%
  dplyr::ungroup() %&gt;%
  dplyr::filter(frac_nonNA &lt; 0.67) %&gt;%
  dplyr::group_by(Protein.Group) %&gt;%
  dplyr::summarise(Count = length(Protein.Group)) %&gt;%
  dplyr::filter(Count == 4) %&gt;%
  dplyr::pull(Protein.Group)
proTissueTab &lt;- dplyr::filter(proTissueTab, !Protein.Group %in% rmFeats)

# Convert long data to SE object
proTissue &lt;- df2SummExp(proTissueTab, row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;,
                        values = &#39;Abundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                        col_anno = anno)

# Got 6873 features after filtering</code></pre>
<p>Display dimensions of filtered data (40 samples and 6065
features)</p>
<pre class="r"><code>dim(proTissue)</code></pre>
<pre><code>[1] 6065   40</code></pre>
<p>Display data missingness</p>
<pre class="r"><code>exprMat &lt;- SummarizedExperiment::assay(proTissue)
visdat::vis_miss(exprMat, cluster = T, sort_miss = T) +
  labs(y = &#39;Features&#39;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 0, size = 11, face = &#39;bold&#39;),
        axis.text.y = element_blank(),
        axis.title.y = element_text(size = 13, face = &#39;bold&#39;),
        legend.text = element_text(size = 12, face = &#39;bold&#39;))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Display distribution of filtered data</p>
<pre class="r"><code>ggplot(proTissueTab, aes(x=Sample, y=Abundance)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(y = &#39;Protein abundance&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Perform VSN and display distribution of normalized data</p>
<pre class="r"><code># Perform VSN
exprMat &lt;- as.matrix(SummarizedExperiment::assay(proTissue))
fit &lt;- vsnMatrix(exprMat)
proTissueNorm &lt;- proTissue
SummarizedExperiment::assay(proTissueNorm) &lt;- predict(fit, exprMat)

# Save normalized data
# saveRDS(proTissueNorm, &#39;./data/AG_Krijgsveld/new_proTissueNorm_DIA.rds&#39;)

# Convert SE object to long data for plotting
proTissueTabNorm &lt;- summExp2df(proTissueNorm, assay = &#39;Abundance&#39;,
                               row_id = &#39;Protein.Group&#39;, col_id = &#39;Sample&#39;)

ggplot(proTissueTabNorm, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(y = &#39;Normalized protein abundance&#39;) +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))</code></pre>
<p><img src="figure/preprocessing_DIA_new.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Store median normalized data for computing log2(FC)
exprMat &lt;- as.matrix(assay(proTissue))
sampleAnno &lt;- tibble::as_tibble(colData(proTissue), rownames = &#39;Sample&#39;)
featAnno &lt;- tibble::as_tibble(rowData(proTissue), rownames = &#39;Feature&#39;)
proTissueMediTab &lt;- limma::normalizeBetweenArrays(exprMat, method = &#39;scale&#39;) %&gt;%
  tibble::as_tibble(rownames = &#39;Feature&#39;) %&gt;%
  tidyr::pivot_longer(cols = -&#39;Feature&#39;,
                      names_to = &#39;Sample&#39;,
                      values_to = &#39;mediAbundance&#39;) %&gt;%
  dplyr::mutate(mediAbundance = log2(mediAbundance)) %&gt;%
  dplyr::left_join(sampleAnno, by = &#39;Sample&#39;) %&gt;%
  dplyr::left_join(featAnno, by = &#39;Feature&#39;)

ggplot(proTissueMediTab, aes(x=Sample, y=mediAbundance)) +
  geom_boxplot() +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

proTissueMedi &lt;- df2SummExp(proTissueMediTab, row_id = &#39;Feature&#39;, col_id = &#39;Sample&#39;,
                            values = &#39;mediAbundance&#39;, row_anno = c(&#39;Protein.Names&#39;, &#39;Genes&#39;),
                            col_anno = anno)
# saveRDS(proTissueMedi, &#39;./data/AG_Krijgsveld/new_proTissueMedi_DIA.rds&#39;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lubridate_1.9.2             forcats_1.0.0              
 [3] stringr_1.5.0               dplyr_1.1.2                
 [5] purrr_1.0.1                 tidyr_1.3.0                
 [7] tibble_3.2.1                ggplot2_3.4.2              
 [9] tidyverse_2.0.0             SummarizedExperiment_1.30.2
[11] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[13] IRanges_2.34.1              S4Vectors_0.38.1           
[15] MatrixGenerics_1.12.2       matrixStats_1.0.0          
[17] visdat_0.6.0                vsn_3.68.0                 
[19] Biobase_2.60.0              BiocGenerics_0.46.0        
[21] readxl_1.4.3                readr_2.1.4                
[23] workflowr_1.7.0            

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0        farver_2.1.1            bitops_1.0-7           
 [4] fastmap_1.1.1           RCurl_1.98-1.12         promises_1.2.0.1       
 [7] digest_0.6.33           timechange_0.2.0        lifecycle_1.0.3        
[10] processx_3.8.2          magrittr_2.0.3          compiler_4.3.0         
[13] rlang_1.1.1             sass_0.4.7              tools_4.3.0            
[16] utf8_1.2.3              yaml_2.3.7              knitr_1.43             
[19] labeling_0.4.2          S4Arrays_1.0.4          bit_4.0.5              
[22] DelayedArray_0.26.6     withr_2.5.0             grid_4.3.0             
[25] preprocessCore_1.62.1   fansi_1.0.4             git2r_0.32.0           
[28] colorspace_2.1-0        scales_1.2.1            cli_3.6.1              
[31] rmarkdown_2.23          crayon_1.5.2            generics_0.1.3         
[34] rstudioapi_0.15.0       httr_1.4.6              tzdb_0.4.0             
[37] cachem_1.0.8            affy_1.78.1             zlibbioc_1.46.0        
[40] parallel_4.3.0          BiocManager_1.30.21     cellranger_1.1.0       
[43] XVector_0.40.0          vctrs_0.6.3             Matrix_1.6-0           
[46] jsonlite_1.8.7          callr_3.7.3             hms_1.1.3              
[49] bit64_4.0.5             limma_3.56.2            jquerylib_0.1.4        
[52] affyio_1.70.0           glue_1.6.2              ps_1.7.5               
[55] stringi_1.7.12          gtable_0.3.3            later_1.3.1            
[58] munsell_0.5.0           pillar_1.9.0            htmltools_0.5.5        
[61] GenomeInfoDbData_1.2.10 R6_2.5.1                rprojroot_2.0.3        
[64] vroom_1.6.3             evaluate_0.21           lattice_0.21-8         
[67] highr_0.10              httpuv_1.6.11           bslib_0.5.0            
[70] Rcpp_1.0.11             whisker_0.4.1           xfun_0.39              
[73] fs_1.6.2                getPass_0.2-2           pkgconfig_2.0.3        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
