---
title: 'Preprocessing: Targeted Metabolomics and DDA Proteomics of Method Development cohort'
author: "Qian-Wu Liao"
output:
  workflowr::wflow_html:
    toc: yes
    toc_float: yes
    code_folding: hide
---

<font size='4'> Description: Preprocess Tissue and Plasma Targeted Metabolomics
data generated by AG Hell and Tissue and Plasma DDA Proteomics from AG Krijgsveld,
which includes data cleansing, data normalization, and in the end, storing all needed
information of each dataset (4 in total) in SummarizedExperiment objects for further
analyses. </font>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.height = 8, fig.width = 10, warning = F)
knitr::opts_knit$set(root.dir = '/Users/qianwu/Desktop/SMART-CARE_LungCancer')
```

Load libraries
```{r library loading, message = FALSE}
library('readr')
library('readxl')
library('vsn')
library('pwr')
library('SummarizedExperiment')
library('ggrepel')
library('tidyverse')
# Load user-defined functions
source('./code/misc.R')

# Set plot theme
th <- theme_bw(base_size = 15) +
  theme(axis.title = element_text(face = 'bold'),
        axis.text = element_text(face = 'bold'),
        axis.ticks = element_line(linewidth = 0.8),
        legend.text = element_text(size = 15))
```

```{r prepare global info, message=F}
# Load sample metadata
smpMetadat <- readxl::read_excel('./data/sample_metadata.xlsx')
colnames(smpMetadat) <- smpMetadat[3,, drop = F]
smpMetadat <- dplyr::slice(smpMetadat, -c(1:3)) %>%
  dplyr::select(c(Code, Parents, Visit, `Material submitted`, `Date and time of collection or surgery`,
                  `SMART-CARE cohort identifier`)) %>%
  dplyr::rename(Sample = Code, Patient = Parents, TimePoint = Visit, SmpType = `Material submitted`,
                Date = `Date and time of collection or surgery`, Cohort = `SMART-CARE cohort identifier`) %>%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, '^SC_T_S_|^SC_DIS_S_|^SC_S_|_P'),
                Patient = stringr::str_remove(Patient, '^/THRX_SPACE/THRX_DB/'),
                TimePoint = dplyr::case_when(TimePoint == 'PRETHERAPEUTIC' ~ 'Baseline',
                                             TimePoint == 'FOLLOW-UP' ~ 'Follow-up',
                                             TimePoint == 'RECURRENCE' ~ 'Recurrence'),
                SmpType = dplyr::case_when(SmpType == 'EDTA_PLASMA' ~ 'Plasma',
                                           SmpType == 'FRESH_FROZEN_TISSUE' ~ 'Tissue'),
                Condition = dplyr::case_when(grepl('_TU|_TG', Sample) ~ 'Tumor',
                                             grepl('_NG', Sample) ~ 'Normal',
                                             !grepl('_TU|_TG|_NG', Sample) ~ TimePoint),
                TimePoint = dplyr::case_when(TimePoint %in% 'Recurrence' ~ 'Follow-up',
                                             !TimePoint %in% 'Recurrence' ~ TimePoint),
                Date = stringr::str_extract(Date, '^\\d+-\\d+-\\d+'),
                Cohort = dplyr::case_when(Cohort == 'DISCOVERY_COHORT' ~ 'Discovery',
                                          Cohort == 'METHOD_DEVELOPMENT_COHORT' ~ 'MethodDev'),
                Date = as.Date(Date, format = '%Y-%m-%d'))

# Load patient metadata
patientMetadat <- readxl::read_excel('./data/patient_metadata.xlsx')
colnames(patientMetadat) <- patientMetadat[3,, drop = F]
patientMetadat <- dplyr::slice(patientMetadat, -c(1:3)) %>%
  dplyr::select(c(Code, Gender, `Age at diagnosis`, `Pathological stage`, `Smoking status`,
                  `Adjuvant chemotherapy`,)) %>%
  dplyr::rename(Patient = Code, Age = `Age at diagnosis`, Stage = `Pathological stage`,
                Smoking = `Smoking status`, Adjuvant = `Adjuvant chemotherapy`) %>%
  dplyr::mutate(Gender = dplyr::case_when(Gender == 'MALE' ~ 'Male',
                                          Gender == 'FEMALE' ~ 'Female'),
                Smoking = dplyr::case_when(Smoking == 'SMOKER' ~ 'Smoker',
                                           Smoking == 'EX-SMOKER' ~ 'Ex-smoker',
                                           Smoking == 'NON-SMOKER' ~ 'Non-smoker'),
                Adjuvant = dplyr::case_when(Adjuvant == 'true' ~ 'True',
                                            Adjuvant == 'false' ~ 'False'),
                Age = as.numeric(Age))
# Include patient recurrence information in patient metadata
recurPats <- dplyr::filter(smpMetadat, TimePoint == 'Recurrence') %>%
  dplyr::pull(Patient)
patientMetadat <- dplyr::mutate(patientMetadat, Recurrence = dplyr::case_when(Patient %in% recurPats ~ 'Yes',
                                                                              !Patient %in% recurPats ~ 'No'))

# Load aliquot metadata
aliquotMetadat <- readxl::read_excel('./data/aliquot_metadata.xlsx')
colnames(aliquotMetadat) <- aliquotMetadat[3,, drop = F]
aliquotMetadat <- dplyr::slice(aliquotMetadat, -c(1:3)) %>%
  dplyr::select(Code, Parents, `Tumor Cell Content (%)`, `Cohort Identifier`, `Submission to`,
                `Delivered?`) %>%
  dplyr::rename(Aliquot = Code, Sample = Parents, TumorPurity = `Tumor Cell Content (%)`,
                Cohort = `Cohort Identifier`, To = `Submission to`, Delivered = `Delivered?`) %>%
  dplyr::mutate(Sample = stringr::str_remove_all(Sample, '^/THRX_SPACE/THRX_DB/SC_T_S_|_P'),
                Sample = stringr::str_remove(Sample, '^/THRX_SPACE/THRX_DB/SC_DIS_S_'),
                Sample = stringr::str_remove(Sample, '^/THRX_SPACE/THRX_DB/SC_S_'),
                Sample = stringr::str_remove(Sample, '^/THRX_SPACE/THRX_DB/SC_DIS_'),
                Sample = dplyr::case_when(Sample == 'SDIR55_TU' ~ 'PDIR55_TU',
                                          Sample == 'SDIR55_NG' ~ 'PDIR55_NG',
                                          !Sample %in% c('SDIR55_TU', 'SDIR55_NG') ~ Sample),
                Cohort = dplyr::case_when(Cohort == 'method establishment' ~ 'MethodDev',
                                          Cohort == 'DISCOVERY_COHORT' ~ 'Discovery'),
                TumorPurity = as.numeric(TumorPurity),
                Sample = dplyr::case_when(grepl('^SC_DIS_A_Y4W4L7_P_V2', Aliquot) ~ 'Y4W4L7_V2',
                                          !grepl('^SC_DIS_A_Y4W4L7_P_V2', Aliquot) ~ Sample)) %>%
  dplyr::filter(To %in% c('HELL', 'KRIJGSVELD'))
# Prepare aliquot metadata containing tumor purity information
tumorPurityInfo <- dplyr::select(aliquotMetadat, Aliquot, Sample, TumorPurity)

# Combine all needed metadata
summMetadat <- dplyr::left_join(tumorPurityInfo, smpMetadat, by = 'Sample') %>%
  dplyr::left_join(patientMetadat, by = 'Patient')

# List sample annotations to keep in SE objects
smpAnno <- c('Patient', 'SmpType', 'TimePoint', 'Date', 'Cohort', 'Condition',
             'Recurrence', 'Gender', 'Age', 'Smoking', 'Stage', 'Adjuvant')
```

# Plasma Metabolomics

```{r message = F}
# Clean up data and convert data to long data
metaPlasmaTab <- readxl::read_excel(
  './data/MethodDev/AG_Hell/Thorax cohort_not-normalized_subset plasma.xlsx',
  skip = 1) %>%
  filter(!is.na(parents)) %>%
  dplyr::select(-c(Sample_identification_cut, parents, patients_tissue, Individual_ID, Status)) %>%
  rename(Aliquot = 'Sample Identification') %>%
  pivot_longer(cols = -'Aliquot',
               names_to = 'Metabolite',
               values_to = 'Abundance') %>%
  mutate(Abundance = as.numeric(Abundance)) %>%
  left_join(summMetadat, by = 'Aliquot')
# Convert long data to SE object
metaPlasma <- df2SummExp(metaPlasmaTab, row_id = 'Metabolite', col_id = 'Sample',
                         values = 'Abundance', col_anno = smpAnno)
```

Display dimensions of processed data (40 samples and 630 features)
```{r}
dim(metaPlasma)
```

Display distribution of original data
```{r warning = F}
metaPlasma4Plot <- summExp2df(metaPlasma, assay = 'Abundance',
                              row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaPlasma4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

# Zero-valued data points are removed.
```

## Perform VSN

Perform VSN (variance stabilizing normalization) on original data
```{r}
exprMat <- as.matrix(assay(metaPlasma))
fit <- vsnMatrix(exprMat)
metaPlasmaNorm <- metaPlasma
assay(metaPlasmaNorm) <- predict(fit, exprMat)
# Save vsn normalized data
saveRDS(metaPlasmaNorm, './data/MethodDev/AG_Hell/metaPlasmaVsn.rds')

metaPlasmaNorm4Plot <- summExp2df(metaPlasmaNorm, assay = 'Abundance',
                                  row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaPlasmaNorm4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Show feature mean-variance relationship of vsn normalized data
```{r}
vsn::meanSdPlot(assay(metaPlasmaNorm), ranks = T, plot = F, bins = 30)$gg +
  labs(x = 'Rank of mean', y = 'SD') +
  th
```

```{r message=F}
# Store median normalized data for computing log2(FC) later
exprMat <- as.matrix(assay(metaPlasma))
sampleAnno <- tibble::as_tibble(colData(metaPlasma), rownames = 'Sample')
metaPlasmaMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance+0.0001)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample')

# ggplot(metaPlasmaMediTab, aes(x=Sample, y=mediAbundance)) +
#   geom_boxplot() +
#   th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

metaPlasmaMedi <- df2SummExp(metaPlasmaMediTab, row_id = 'Feature', col_id = 'Sample',
                             values = 'mediAbundance', col_anno = smpAnno)
# Save median normalized data
saveRDS(metaPlasmaMedi, './data/MethodDev/AG_Hell/metaPlasmaMedi.rds')
```

## Perform PCA

Use only top 100 metabolites that have high variance
```{r}
exprMat <- assay(metaPlasmaNorm)
sd <- rowSds(exprMat)
exprMatSub <- exprMat[order(sd, decreasing = TRUE)[1:100],]
resPC <- prcomp(t(exprMatSub), center = TRUE, scale. = FALSE)

anno <- distinct(metaPlasmaTab, Sample, Patient, Condition)
metaPlasmaNorm4Plot <- resPC$x[,1:10] %>%
  as.data.frame() %>%
  rownames_to_column('Sample') %>%
  left_join(anno, by = 'Sample')

ggplot(metaPlasmaNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
<br/>
<br/>

Use all metabolites
```{r}
resPC2 <- prcomp(t(exprMat), center = TRUE, scale. = FALSE)

metaPlasmaNorm4Plot2 <- resPC2$x[, 1:10] %>%
  as.data.frame() %>%
  rownames_to_column('Sample')%>%
  left_join(anno, by = 'Sample')

ggplot(metaPlasmaNorm4Plot2, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```




# Tissue Metabolomics

```{r message = F, warning = F}
# Clean up data and convert data to long data
metaTissueTab <- readxl::read_excel(
  './data/MethodDev/AG_Hell/Thorax cohort_not-normalized_subset tissue.xlsx',
  skip = 1) %>%
  dplyr::filter(!is.na(parents)) %>%
  dplyr::select(-c(Sample_identification_cut, parents, patients_tissue, Individual_ID, Status)) %>%
  dplyr::rename(Aliquot = `Sample Identification`) %>%
  tidyr::pivot_longer(cols = -'Aliquot',
                      names_to = 'Metabolite',
                      values_to = 'Abundance') %>%
  dplyr::mutate(Abundance = as.numeric(Abundance)) %>%
  dplyr::left_join(summMetadat, by = 'Aliquot')
# Convert long data to SE object
metaTissue <- df2SummExp(metaTissueTab, row_id = 'Metabolite', col_id = 'Sample',
                         values = 'Abundance', col_anno = c(smpAnno, 'TumorPurity'))
```

Display dimensions of processed data (40 samples and 630 features)
```{r}
dim(metaTissue)
```

Display 3 NA data points in this dataset
```{r}
dplyr::filter(metaTissueTab, is.na(Abundance)) %>%
  dplyr::select(Metabolite, Abundance, Sample, Condition)

# Values of these data points were infinite (character) before converting them to
# numeric class.
```

Display distribution of original data
```{r warning = F}
metaTissue4Plot <- summExp2df(metaTissue, assay = 'Abundance',
                              row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaTissue4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

# There are 3 NA data points and 756 zeros.
```

## Perform VSN

```{r warning = F}
exprMat <- as.matrix(assay(metaTissue))
fit <- vsnMatrix(exprMat)
metaTissueNorm <- metaTissue
assay(metaTissueNorm) <- predict(fit, exprMat)
# Save vsn normalized data
saveRDS(metaTissueNorm, './data/MethodDev/AG_Hell/metaTissueVsn.rds')

metaTissueNorm4Plot <- summExp2df(metaTissueNorm, assay = 'Abundance',
                                  row_id = 'Metabolite', col_id = 'Sample')
ggplot(metaTissueNorm4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Metabolite abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Show feature mean-variance relationship of vsn normalized data
```{r}
vsn::meanSdPlot(assay(metaTissueNorm), ranks = T, plot = F, bins = 30)$gg +
  labs(x = 'Rank of mean', y = 'SD') +
  th
```

```{r message=F}
# Store median normalized data for computing log2(FC)
exprMat <- as.matrix(assay(metaTissue))
sampleAnno <- tibble::as_tibble(colData(metaTissue), rownames = 'Sample')
metaTissueMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance+0.0001)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample')

# ggplot(metaTissueMediTab, aes(x=Sample, y=mediAbundance)) +
#   geom_boxplot() +
#   th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

metaTissueMedi <- df2SummExp(metaTissueMediTab, row_id = 'Feature', col_id = 'Sample',
                             values = 'mediAbundance', col_anno = c(smpAnno, 'TumorPurity'))
# Save median normalized data
saveRDS(metaTissueMedi, './data/MethodDev/AG_Hell/metaTissueMedi.rds')
```

## Perform PCA
Take an initial look at dataset for quality control
<br/>
<br/>

Use only top 100 metabolites that have high variance
```{r}
exprMat <- assay(metaTissueNorm)
# Remove the metabolites containing non-applicable values (try to impute the
# missing values later)
exprMat <- exprMat[which(rownames(exprMat) != c('Ala', 'Histamine')),]
sd <- rowSds(exprMat)
exprMatSub <- exprMat[order(sd, decreasing = TRUE)[1:100],]
resPC <- prcomp(t(exprMatSub), center = TRUE, scale. = FALSE) # Each row is an
# observation, i.e., samples

anno <- dplyr::distinct(metaTissueTab, Sample, Patient, Condition)
metaTissueNorm4Plot <- resPC$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('Sample') %>%
  dplyr::left_join(anno, by = 'Sample')

ggplot(metaTissueNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
<br/>
<br/>

Use all metabolites except those three containing missing values
```{r}
resPC2 <- prcomp(t(exprMat), center = TRUE, scale. = FALSE)

metaTissueNorm4Plot2 <- resPC2$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('Sample') %>%
  dplyr::left_join(anno, by = 'Sample')

ggplot(metaTissueNorm4Plot2, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```

```{r include = F, eval = F}
# Perform power analysis to estimate the required number of samples for varied
# statistical power
dTab <- summExp2df(metaTissueNorm, assay = 'Abundance',
                   row_id = 'Metabolite', col_id = 'Sample') %>%
  dplyr::select(-Sample) %>%
  tidyr::pivot_wider(names_from = 'Condition', values_from = 'Value') #%>%
  dplyr::mutate(diff = abs(Normal - Tumor)) %>%
  dplyr::group_by(Metabolite) %>%
  dplyr::summarise(d = mean(diff) / sd(diff))

dTab4Plot <- lapply(seq(0.01, 1.5, length.out = 100), function(i){
  tibble(nCall = nrow(dplyr::filter(dTab, d > i)),
         nSample = pwr.t.test(d = i, sig.level = 0.05,
                              power = 0.8, type = 'paired')$n)}) %>%
  dplyr::bind_rows() %>%
  dplyr::arrange(nCall) %>%
  dplyr::filter(nCall > 0)

ggplot(dTab4Plot, aes(x=nCall, y=nSample)) +
  geom_line() +
  geom_point() +
  ylim(0, 100)
```




# Tissue Proteomics

```{r message = F}
# Abundance data
proTissueTab <- readxl::read_excel(
  './data/MethodDev/AG_Krijgsveld/20211119_151605_20211104_TMMS_Tissue_Report.xlsx',
  sheet = 1) %>%
  dplyr::select(PG.Genes, PG.ProteinGroups, contains('oecf4')) %>%
  # Convert messy data to tidy table
  tidyr::pivot_longer(cols = -c('PG.Genes', 'PG.ProteinGroups'),
                      names_to = 'Sample_IDs',
                      values_to = 'Abundance') %>%
  dplyr::mutate(Abundance = as.numeric(Abundance)) %>%
  dplyr::group_by(PG.Genes, PG.ProteinGroups, Sample_IDs) %>%
  # Determine a single numeric value for a protein of a sample if multiple
  # identical protein measures of a sample exist
  dplyr::summarise(Abundance = mean(Abundance, na.rm = TRUE)) %>%
  dplyr::ungroup() %>%
  dplyr::filter(!is.na(Abundance))

# Column PG.ProteinGroups has no overlap and column PG.Genes has few overlaps (
# TMPO:2 and NA:5) because some genes can encode multiple protein isoforms through
# RNA splicing.
# 19 protein groups are not present in any samples (Abundance = NaN).
```

```{r message = F}
# Annotation data
proTissueAnno <- readxl::read_excel(
  './data/MethodDev/AG_Krijgsveld/20211119_151605_20211104_TMMS_Tissue_Report.xlsx',
  sheet = 2) %>%
  dplyr::select(Sample_IDs, Identifier) %>%
  rename(Aliquot = Identifier) %>%
  dplyr::mutate(Aliquot = str_remove(Aliquot, '/THRX_SPACE/'),
                id = paste0('smp', seq(nrow(.)))) %>%
  dplyr::left_join(summMetadat, by = 'Aliquot')
```

Show dimensions of processed data (40 samples and 5193 features)
```{r}
# Combine processed Abundance and annotation data
proTissueTab <- dplyr::left_join(proTissueTab, proTissueAnno, by = 'Sample_IDs')
# Convert tidy table to value matrix for further analysis
proTissue <- df2SummExp(proTissueTab, row_id = 'PG.ProteinGroups', col_id = 'id',
                        values = 'Abundance', row_anno = 'PG.Genes',
                        col_anno = c(smpAnno, 'TumorPurity', 'Sample'))
dim(proTissue)
```

View distribution of original data to see if there is technical noise
```{r}
# Convert value matrix to long data for making plot
proTissue4Plot <- summExp2df(proTissue, assay = 'Abundance',
                             row_id = 'PG.ProteinGroups', col_id = 'id')
ggplot(proTissue4Plot, aes(x=id, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Protein abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

```{r include = F, eval = F}
# Log2-transform original data
ggplot(proTissue4Plot, aes(x=id, y=log2(Value))) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Protein abundance') +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

## Perform VSN

```{r}
exprMat <- as.matrix(assay(proTissue))
fit <- vsnMatrix(exprMat)
proTissueNorm <- proTissue
assay(proTissueNorm) <- predict(fit, exprMat)

proTissueNorm4Plot <- summExp2df(proTissueNorm, assay = 'Abundance',
                                 row_id = 'PG.ProteinGroups', col_id = 'id')
ggplot(proTissueNorm4Plot, aes(x=id, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Protein abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Show feature mean-variance relationship of vsn normalized data
```{r}
vsn::meanSdPlot(assay(proTissueNorm), ranks = T, plot = F)$gg +
  labs(x = 'Rank of mean', y = 'SD') +
  th
```

## Perform PCA
Take an initial look at dataset for quality control

```{r}
# prcomp()'s parameter *center* is set to TRUE (Check effects of parameters *center*
# and *scale.*)

exprMat <- assay(proTissueNorm)
sd <- rowSds(exprMat)
exprMatSub <- exprMat[order(sd, decreasing = TRUE)[1:1000],]
resPC <- prcomp(t(exprMatSub), center = TRUE, scale. = FALSE) # Each row is an
# observation, i.e., samples

anno <- dplyr::distinct(proTissueTab, id, Patient, Condition)
proTissueNorm4Plot <- resPC$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('id') %>%
  dplyr::left_join(anno, by = 'id')

ggplot(proTissueNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_text_repel(aes(label=id), show.legend = F) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
Patient "AHTYU035" has two tumor samples (smp24, smp29) and one normal sample (smp9).\
Patient "P3457YLC" does not have a paired tumor sample (smp22).\
It is very probably mislabeling. If not, check correlation between two tumor samples
from the same patient to get further idea.
<br/>
<br/>

```{r}
# Manually modify potential mislabeling for now
ids <- colnames(proTissueNorm)
colnames(proTissueNorm) <- colData(proTissueNorm)$Sample
colnames(proTissueNorm)[which(ids == 'smp29')] <- 'F04ERF3M_TG'
colData(proTissueNorm)$Sample <- colnames(proTissueNorm)
colData(proTissueNorm)$Patient[which(ids == 'smp29')] <- 'P3457YLC'
colData(proTissueNorm)$Recurrence[which(ids == 'smp29')] <- 'Yes'
colData(proTissueNorm)$Age[which(ids == 'smp29')] <- 68
colData(proTissueNorm)$Adjuvant[which(ids == 'smp29')] <- 'False'
# Remove redundant Sample column in column annotation table
colData(proTissueNorm) <- colData(proTissueNorm)[, -which(colnames(colData(proTissueNorm)) == 'Sample')]

# Save vsn normalized data
saveRDS(proTissueNorm, './data/MethodDev/AG_Krijgsveld/proTissueVsn_DDA.rds')
```

```{r}
# Store median normalized data for computing log2(FC)
# Manually modify potential mislabeling for now
ids <- colnames(proTissue)
colnames(proTissue) <- colData(proTissue)$Sample
colnames(proTissue)[which(ids == 'smp29')] <- 'F04ERF3M_TG'
colData(proTissue)$Sample <- colnames(proTissue)
colData(proTissue)$Patient[which(ids == 'smp29')] <- 'P3457YLC'
colData(proTissue)$Recurrence[which(ids == 'smp29')] <- 'Yes'
colData(proTissue)$Age[which(ids == 'smp29')] <- 68
colData(proTissue)$Adjuvant[which(ids == 'smp29')] <- 'False'

exprMat <- as.matrix(assay(proTissue))
sampleAnno <- tibble::as_tibble(colData(proTissue))
featAnno <- tibble::as_tibble(rowData(proTissue), rownames = 'Feature')
proTissueMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample') %>%
  dplyr::left_join(featAnno, by = 'Feature')

# ggplot(proTissueMediTab, aes(x=Sample, y=mediAbundance)) +
#   geom_boxplot() +
#   th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

proTissueMedi <- df2SummExp(proTissueMediTab, row_id = 'Feature', col_id = 'Sample',
                            values = 'mediAbundance', row_anno = 'PG.Genes',
                            col_anno = c(smpAnno, 'TumorPurity'))
# Save median normalized data
saveRDS(proTissueMedi, './data/MethodDev/AG_Krijgsveld/proTissueMedi_DDA.rds')
```

```{r include = F, eval = F}
# prcomp()'s parameter *center* is set to FALSE

resPC2 <- prcomp(t(exprMatSub), center = FALSE, scale. = FALSE)

proTissueNorm4Plot2 <- resPC2$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('id') %>%
  dplyr::left_join(anno, by = 'id')

ggplot(proTissueNorm4Plot2, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed')
```

```{r include = F, eval = F}
### Perform power analysis

compD <- function(tab, n = 40){
  #' Compute effect size for each variable
  
  meanDiff <- abs(tab$avg[1] - tab$avg[2])
  fc <- abs(log(tab$avg[1]/tab$avg[2]))
  sdPool <- sqrt((tab$sd[1]^2 + tab$sd[2]^2) / 2)
  corFac <- ((n-3)/(n-2.25))*sqrt((n-2)/n) # Correction factor for small datasets
  d = corFac * meanDiff / sdPool
  return(tibble(d = d, logFC = fc))
}
```

```{r message = F, include = F, eval = F}
proTissueNormTab <- summExp2df(proTissueNorm, assay = 'Abundance',
                               row_id = 'PG.ProteinGroups', col_id = 'id') %>%
  dplyr::group_by(Condition, PG.ProteinGroups) %>%
  dplyr::summarise(avg = mean(Value), sd = sd(Value))

dTab <- dplyr::group_by(proTissueNormTab, PG.ProteinGroups) %>%
  tidyr::nest() %>%
  dplyr::mutate(results = map(data, compD)) %>%
  tidyr::unnest(results) %>%
  dplyr::select(-data)
```

```{r include = F, eval = F}
# Estimate required number of samples for varied statistical power

dTab4Plot <- lapply(seq(0.01, 1.5, length.out = 100), function(i){
  tibble(nCall = nrow(dplyr::filter(dTab, d > i)),
         nSample = pwr.t.test(d = i, sig.level = 0.05, power = 0.8,
                              type = 'two.sample')$n)}) %>%
  dplyr::bind_rows() %>%
  dplyr::arrange(nCall) %>%
  dplyr::filter(nCall > 0)

ggplot(dTab4Plot, aes(x=nCall, y=nSample)) +
  geom_line() +
  geom_point() +
  scale_y_log10() +
  scale_x_log10()

# nSample is number of samples required in each group.
```




# Plasma Proteomics

```{r message = F}
# Abundance data
proPlasmaTab <- readxl::read_excel(
  './data/MethodDev/AG_Krijgsveld/20211119_133842_20211018_TMMS_TRX_plasma_Report.xlsx'
  , sheet = 1) %>%
  dplyr::select(PG.Genes, PG.ProteinGroups, contains('oecf4')) %>%
  pivot_longer(cols = -c('PG.Genes', 'PG.ProteinGroups'),
               names_to = 'Sample_IDs',
               values_to = 'Abundance') %>%
  mutate(Abundance = as.numeric(Abundance)) %>%
  group_by(PG.Genes, PG.ProteinGroups, Sample_IDs) %>%
  summarise(Abundance = mean(Abundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(!is.na(Abundance))

# Column PG.ProteinGroups has no overlap and column PG.Genes has few overlaps (NA:5).
```

```{r}
# Annotation data
proPlasmaAnno <- readxl::read_excel(
  './data/MethodDev/AG_Krijgsveld/20211119_133842_20211018_TMMS_TRX_plasma_Report.xlsx',
  sheet = 2) %>%
  dplyr::select(Sample_IDs, Identifier) %>%
  rename(Aliquot = Identifier) %>%
  mutate(Aliquot = str_remove(Aliquot, '/THRX_SPACE/')) %>%
  left_join(summMetadat, by = 'Aliquot')
```

Show dimensions of processed data (40 samples and 435 features)
```{r}
proPlasmaTab <- left_join(proPlasmaTab, proPlasmaAnno, by = 'Sample_IDs')
proPlasma <- df2SummExp(proPlasmaTab, row_id = 'PG.ProteinGroups', col_id = 'Sample',
                        values = 'Abundance', row_anno = 'PG.Genes', col_anno = smpAnno)
dim(proPlasma)
```

Display distribution of original data
```{r}
proPlasma4Plot <- summExp2df(proPlasma, assay = 'Abundance',
                             row_id = 'PG.ProteinGroups', col_id = 'Sample')
ggplot(proPlasma4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  scale_y_log10() +
  labs(x = 'Sample ID', y = 'Protein Abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

## perform VSN

```{r}
exprMat <- as.matrix(assay(proPlasma))
fit <- vsnMatrix(exprMat)
proPlasmaNorm <- proPlasma
assay(proPlasmaNorm) <- predict(fit, exprMat)
# Save vsn normalized data
saveRDS(proPlasmaNorm, './data/MethodDev/AG_Krijgsveld/proPlasmaVsn_DDA.rds')

proPlasmaNorm4Plot <- summExp2df(proPlasmaNorm, assay = 'Abundance',
                                 row_id = 'PG.ProteinGroups', col_id = 'Sample')
ggplot(proPlasmaNorm4Plot, aes(x=Sample, y=Value)) +
  geom_boxplot() +
  labs(x = 'Sample ID', y = 'Protein Abundance') +
  th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```

Show feature mean-variance relationship of vsn normalized data
```{r}
vsn::meanSdPlot(assay(proPlasmaNorm), ranks = T, plot = F, bins = 30)$gg +
  labs(x = 'Rank of mean', y = 'SD') +
  th
```

```{r}
# Store median normalized data for computing log2(FC)
exprMat <- as.matrix(assay(proPlasma))
sampleAnno <- tibble::as_tibble(colData(proPlasma), rownames = 'Sample')
featAnno <- tibble::as_tibble(rowData(proPlasma), rownames = 'Feature')
proPlasmaMediTab <- limma::normalizeBetweenArrays(exprMat, method = 'scale') %>%
  tibble::as_tibble(rownames = 'Feature') %>%
  tidyr::pivot_longer(cols = -'Feature',
                      names_to = 'Sample',
                      values_to = 'mediAbundance') %>%
  dplyr::mutate(mediAbundance = log2(mediAbundance)) %>%
  dplyr::left_join(sampleAnno, by = 'Sample') %>%
  dplyr::left_join(featAnno, by = 'Feature')

# ggplot(proPlasmaMediTab, aes(x=Sample, y=mediAbundance)) +
#   geom_boxplot() +
#   th + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

proPlasmaMedi <- df2SummExp(proPlasmaMediTab, row_id = 'Feature', col_id = 'Sample',
                            values = 'mediAbundance', row_anno = 'PG.Genes', col_anno = smpAnno)
# Save median normalized data
saveRDS(proPlasmaMedi, './data/MethodDev/AG_Krijgsveld/proPlasmaMedi_DDA.rds')
```

## Perform PCA

```{r}
exprMat <- assay(proPlasmaNorm)
resPC <- prcomp(t(exprMat), center = TRUE, scale. = FALSE)

anno <- dplyr::distinct(proPlasmaTab, Sample, Patient, Condition)
proPlasmaNorm4Plot <- resPC$x[, 1:10] %>%
  as.data.frame() %>%
  tibble::rownames_to_column('Sample') %>%
  dplyr::left_join(anno, by = 'Sample')

ggplot(proPlasmaNorm4Plot, aes(x=PC1, y=PC2, col=Condition, group=Patient)) +
  geom_point() +
  geom_line(col = 'grey50', linetype = 'dashed') +
  th
```
